<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="学习生物信息学的笔记本">
<meta property="og:type" content="website">
<meta property="og:title" content="生信技工">
<meta property="og:url" content="http://example.com/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="学习生物信息学的笔记本">
<meta property="og:locale" content="zh_CN">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoftware">
<meta property="article:tag" content="skills">
<meta property="article:tag" content="tools">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main">
  
    <article id="post-omics_genome.submit" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2022/03/22/omics_genome.submit/" class="article-date">
  <time datetime="2022-03-22T13:30:00.000Z" itemprop="datePublished">2022-03-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/genome/">genome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=31721697&auto=1&height=32"></iframe></div>

<h1 id="1-基因组数据库"><a href="#1-基因组数据库" class="headerlink" title="1. 基因组数据库"></a>1. 基因组数据库</h1><p>提交基因组组装的数据到公共数据库以实现共享，在发表提供基因组数据文章时在文章内提供基因组的accession number。</p>
<p>常用的公共数据库包括：</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/">NCBI-GenBank</a>：美国生物信息学国家中心</li>
<li><a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/">EMBL-EBI</a>：欧洲生物信息学研究所</li>
<li><a target="_blank" rel="noopener" href="https://www.ddbj.nig.ac.jp/index-e.html">DDBJ</a>：日本DNA数据银行</li>
<li><a target="_blank" rel="noopener" href="https://ngdc.cncb.ac.cn/?lang=zh">中国的国家基因组科学数据中心, National Genomics Data Center, NGDC</a>: 中国</li>
</ul>
<h1 id="2-基因组提交"><a href="#2-基因组提交" class="headerlink" title="2. 基因组提交"></a>2. 基因组提交</h1><p>可以单独提交基因组，也可以基因组和基因组注释一同提交。</p>
<p>单独提交基因组会比加上注释信息简单很多，因为加上注释信息一般都会面临多种注释错误(我的例子是4万个基因注释中包含了将近一千个错误需要手动修改)，但是现在基因组注释是非常基础的操作，所以还是建议加上注释文件更完整。如果发表时间紧张，可以考虑先提交基因组获得accession number(在文章里提供)，再提交注释(accession number不变)。</p>
<ul>
<li>单独提交基因组只需要准备基因组文件(fasta格式)，在提交网站填写基因组相关信息，上传基因组文件即可。</li>
<li>基因组与注释文件共同提交则需要对注释文件(gff或者bed格式)做检查和过滤后，用table2asn软件从基因组文件(fasta格式)和注释文件(gff或者bed格式)生成一个sqn文件，再提交sqn文件到提交网站上。当然，单独的基因组文件也可以转化成sqn格式再上传。</li>
</ul>
<p>在<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/">NCBI的提交网站</a>可以选择各种数据类型，提交基因组就输入<strong>genome</strong>。</p>
<h1 id="3-文件准备"><a href="#3-文件准备" class="headerlink" title="3. 文件准备"></a>3. 文件准备</h1><p>在整理基因组文件，基因组结构注释和功能注释文件时，可以先用table2asn检测错误，以指导基因组改善(例如可能删除污染片段)，以及合并注释文件。</p>
<h2 id="3-1-基因组文件"><a href="#3-1-基因组文件" class="headerlink" title="3.1. 基因组文件"></a>3.1. 基因组文件</h2><p>基因组文件用fa格式储存，最好做一做污染过滤。</p>
<h2 id="3-2-注释文件"><a href="#3-2-注释文件" class="headerlink" title="3.2. 注释文件"></a>3.2. 注释文件</h2><ul>
<li>注释文件可以是tbl格式或者gff格式，我的习惯是用gff格式(许多注释软件结果也是gff格式)。</li>
<li>提交的注释文件可以只有结构注释(不提供功能注释，或者通过Dryad【收费】或Figshare【免费】等文件分享网站提供功能注释文件)，也可以把功能注释整理后添加到结构注释文件的product属性值。</li>
</ul>
<h3 id="3-2-1-功能注释"><a href="#3-2-1-功能注释" class="headerlink" title="3.2.1. 功能注释"></a>3.2.1. 功能注释</h3><p>功能注释以product属性值的形式提供，添加到结构注释gff文件的第九列。</p>
<p>合并功能注释和结构注释可以参考博客<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/05/17/omics_genome.functional.annotation/">基因组注释(三)：基因功能注释</a></p>
<p>product值的不规范情况：</p>
<ul>
<li>包含逗号,（因为逗号是分隔符）</li>
<li>包含竖杠|</li>
<li>短横杠-开头或结尾</li>
<li>冒号:开头或结尾</li>
<li>斜杠&#x2F;开头或结尾</li>
<li>不包含字母</li>
<li>括号（包括圆括号,中括号和大括号）不完整的</li>
</ul>
<h1 id="4-提交基因组"><a href="#4-提交基因组" class="headerlink" title="4. 提交基因组"></a>4. 提交基因组</h1><h2 id="4-1-提交基因组到NCBI的GenBank"><a href="#4-1-提交基因组到NCBI的GenBank" class="headerlink" title="4.1. 提交基因组到NCBI的GenBank"></a>4.1. 提交基因组到NCBI的GenBank</h2><p>单独提交基因组只需要基因组(fasta格式)文件，在基因组提交网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/subs/genome/">genome Submit Portal</a>操作。</p>
<ol>
<li>注册和登录NCBI账号</li>
<li>选择提交类型Submission Type：单个基因组Single genome或者批量多个基因组Batch&#x2F;multiple genomes</li>
</ol>
<p>然后进入信息填写页面：</p>
<ol>
<li><p>提交者信息Submitter</p>
<ul>
<li>包括姓名First and Last name，邮箱Email，单位Submitting organization(学校)，部门Department(学院)，街道Street，城市City，省&#x2F;州State&#x2F;Province，邮编Postal code，国家Country。</li>
<li>在最后可以选择<strong>Update my contact information in profile</strong>会更新填写的信息到账号，下一次新填写就默认填入提交者信息了。</li>
</ul>
</li>
<li><p>一般性的信息General info</p>
<ul>
<li>BioProject和BioSample可以先注册在这里提供accession number；也可以选择还未注册提交之后会增加填写BioProject和BioSample信息的步骤然后自动注册新的；</li>
<li>释放日期Release date可以指定日期，也可以提交通过后立即释放；释放日期可以提交后发邮件更改的；</li>
<li>基因组信息Genome info：如果保存到了sqn文件也可以选择**Genome Assembly structured comment is in the contig .sqn file(s)**就不需要填写(但我选了之后不识别还是不选填入)。包括组装日期Assembly date，组装方法Assembly method和版本，组装名称Assembly name，基因组覆盖度Genome coverage(这里其实是depth)，测序技术Sequencing technology；</li>
<li>确认信息：是否完整基因组，是否最终版，是否de novo从头组装即无参考组装，是否已有提交的更新，是否自动移除被系统认为是污染的序列(建议自动移除，否则可能发邮件让手动移除污染序列)；</li>
<li>Submission title可选填，建议填上以区别于其他提交。</li>
</ul>
</li>
<li><p>BioProject General Info</p>
<ul>
<li>提供创建新的BioProject的信息，项目题目Project title，项目描述Project description，和领域Relevance</li>
<li>可选项：相关的外部链接External links，基金grants</li>
</ul>
</li>
<li><p>Publications【only batch&#x2F;multiple genomes】：选填，发表文章的PubMed ID或者DOI号</p>
</li>
<li><p>BioSample Type：样品类型，选择Human&#x2F;Plant等</p>
</li>
<li><p>BioSample Attributes</p>
<ul>
<li>如果是批量提交基因组，可以用excel或csv表格提交多个BioSamples信息</li>
<li>样品名称Sample Name：区别于其他样品的唯一名称</li>
<li>生物名称Organism，可以填物种学名；个体描述isolate&#x2F;栽培名称cultivar&#x2F;生态型ecotype三选一填，年龄age&#x2F;发育阶段development stage二选一填</li>
<li>地理位置geographic location：可以填国家China</li>
<li>组织tissue：填写取样部位，eg leaf，flower，root，stem</li>
</ul>
</li>
<li><p>Genome Info【only batch&#x2F;multiple genomes】</p>
<ul>
<li>批量提交的基因组应该有相同的类型：要么只有染色体序列，要么都包含非染色体的序列(会被处理到WGD genome程序)</li>
<li>确认信息：是否完整基因组，是否最终版，是否de novo从头组装即无参考组装，是否已有提交的更新，是否自动移除被系统认为是污染的序列(建议自动移除，否则可能发邮件让手动移除污染序列)；</li>
</ul>
</li>
<li><p>Files</p>
<ul>
<li>提交基因组文件，单个基因组提交fasta格式或者sqn格式，批量基因组提交fasta格式或者asn.1格式</li>
<li>FTP或Aspera命令行上传文件夹(&gt;10GB)，或者网页提交(&lt;2GB)或Aspera Connect plugin(2-10GB)上传文件</li>
<li>提交之后会检查信息是否错误，提示修改</li>
</ul>
</li>
<li><p>空隙Gaps【only batch&#x2F;multiple genomes】</p>
<ul>
<li>基因组序列中的N的含义，是否随机合并序列(不用组装软件)，感觉一般都不会随机合并吧</li>
<li>指定代表gap的N的最小数量，是否有代表未知长度gap的N的数量(常用大于100个N代表未知长度的gap)，组装gap的证据的数据源</li>
</ul>
</li>
<li><p>Assignment【only single genomes】：确认是否有染色体序列，填写染色体序列信息，是否有质体或线粒体或质粒序列    </p>
</li>
<li><p>References【only single genomes】：序列作者和相关文章信息</p>
</li>
<li><p>Review&amp;Submit：最后检查一遍信息没错误就确认提交</p>
</li>
</ol>
<h2 id="4-2-提交基因组和注释"><a href="#4-2-提交基因组和注释" class="headerlink" title="4.2. 提交基因组和注释"></a>4.2. 提交基因组和注释</h2><p>提交带有注释的基因组的大致步骤：</p>
<ol>
<li>注册新的BioProject和BioSample</li>
<li>生成template.sbt文件</li>
<li>用table2asn把基因组(fsa)和注释文件(gff&#x2F;tbl)生成用于提交的sqn文件</li>
<li>根据生成的验证结果文件查看table2asn检查出来的错误，并修正基因组和注释文件中包含的错误</li>
<li>重新运行table2asn和修正错误(重复步骤3和4)直至没有错误</li>
<li>根据上一个部分<strong>提交基因组到NCBI的GenBank</strong>的步骤提交sqn文件到NCBI</li>
<li>提交后可能会收到邮件需要再次更改，更改后重新运行table2asn和修正错误(重复步骤3和4)直至没有错误在NCBI修正错误(提交新的sqn文件)</li>
</ol>
<h3 id="4-2-1-准备工作"><a href="#4-2-1-准备工作" class="headerlink" title="4.2.1. 准备工作"></a>4.2.1. 准备工作</h3><h4 id="4-2-1-1-注册新的BioProject和BioSample"><a href="#4-2-1-1-注册新的BioProject和BioSample" class="headerlink" title="4.2.1.1. 注册新的BioProject和BioSample"></a>4.2.1.1. 注册新的BioProject和BioSample</h4><ul>
<li>在网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/subs/">submit</a>提交新的BioProject和BioSample</li>
<li>根据项目和样品信息，计划发表的信息填写，勾选自动生成Locus Tag Prefixes。</li>
<li>提交后马上可以获得BioProject ID(PRJNAxxxxxx)和BioSample ID(SAMNxxxxxxxx)，以及Locus Tag Prefixes(xxxxx)三个编号信息用于genome的提交填写。</li>
</ul>
<h4 id="4-2-1-2-生成template-sbt文件"><a href="#4-2-1-2-生成template-sbt文件" class="headerlink" title="4.2.1.2. 生成template.sbt文件"></a>4.2.1.2. 生成template.sbt文件</h4><p>在网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/genbank/template/submission/">template</a>提交基因组信息和BioProject ID(PRJNAxxxxxx)和BioSample ID(SAMNxxxxxxxx)，生成template.sbt文件。</p>
<h3 id="4-2-2-生成提交文件"><a href="#4-2-2-生成提交文件" class="headerlink" title="4.2.2. 生成提交文件"></a>4.2.2. 生成提交文件</h3><p>用table2asn把基因组(fsa)和注释文件(gff&#x2F;tbl)生成用于提交的sqn文件。</p>
<p>table2asn还有一个功能，是在整理基因组文件，基因组结构注释和功能注释文件时，可以先用table2asn检测错误，以指导基因组(例如可能删除污染片段)，以及合并注释文件。</p>
<h4 id="4-2-2-1-table2asn简介"><a href="#4-2-2-1-table2asn简介" class="headerlink" title="4.2.2.1. table2asn简介"></a>4.2.2.1. table2asn简介</h4><p>table2asn用于把基因组和其他信息转换成NCBI的GenBank认可的ASN.1 (Abstract Syntax Notation 1)文本格式文件(sqn后缀)和其他用于协助修改的文件，是命令行格式的软件，取代了原来的老款软件tbl2asn。</p>
<p>另一个相同功能的图形界面软件是Genome Workbench(尝试了一下，数据量大的情况下Window版本会卡顿，还是推荐table2asn)。</p>
<h4 id="4-2-2-2-安装table2asn"><a href="#4-2-2-2-安装table2asn" class="headerlink" title="4.2.2.2. 安装table2asn"></a>4.2.2.2. 安装table2asn</h4><p>在网址下载对应系统(linux64,mac,win64)版本的table2asn安装包，解压缩，重命名为table2asn并设置权限(linux系统下运行<code>chmod 711 table2asn</code>)。</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/">table2asn下载网址</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/table2asn/">table2asn 用法简介</a></li>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/DOCUMENTATION/table2asn_readme.txt">table2asn read me</a></li>
</ul>
<h4 id="4-2-2-3-运行table2asn"><a href="#4-2-2-3-运行table2asn" class="headerlink" title="4.2.2.3. 运行table2asn"></a>4.2.2.3. 运行table2asn</h4><ol>
<li>用法</li>
</ol>
<ul>
<li>最基本的用法：<code>table2asn -t template.sbt -indir ./</code></li>
<li>注释的基因组提交的基础用法：<code>table2asn -t template.sbt -indir ./ -outdir ./ -M n -Z -locus-tag-prefix L6164</code></li>
<li>注释的基因组提交推荐用法：<code>table2asn -t template.sbt -indir ./ -outdir ./ -M n -Z -a s -V vb -c fx -euk -augustus-fix -l paired-ends -gaps-min 1 -gaps-unknown 100 -locus-tag-prefix L6164 -j &quot;[organism=Bauhinia variegata][isolate=pink_petal]&quot; -logfile table2asn.log</code></li>
</ul>
<ol start="2">
<li>参数解释</li>
</ol>
<ul>
<li>-indir .&#x2F;: 指定存放sample.fsa和sample.gff&#x2F;.tbl文件的目录。自动识别目录下所有*.fsa和*.gff&#x2F;tbl文件，需要sample.fsa和sample.gff有相同前缀</li>
<li>-outdir .&#x2F;: 指定结果文件目录，默认与输入一致</li>
<li>-t: 指定template.sbt文件</li>
<li>-i sample.fsa：指定目录下的单个sample.fsa文件(单个基因组提交)，有基因组的最大限制（大概是2G内），超出限制可以只用-indir不用-i -f指定</li>
<li>-f sample.gff：指定目录下的sample.gff&#x2F;tbl文件，不能和-indir同用</li>
<li>-M n：用作原核或真核基因组提交，代替(-a s -V v -c f)，加上-Z就包含基因组特定的差异验证; -M t则是用作TSA提交，代替(-a s -V v -c f，加上TSA特定的验证)</li>
<li>-Z: 生成差异报告(Discrepancy report)，保存到sample.dr中；只推荐注释基因组或者转录组提交使用</li>
<li>-a s: 指定文件类型；-a a指任意格式，包括单个fasta或ASN.1(default)； -a s指一套不相关的序列，例如一个基因组组装；-a s1指物种内的居群集；-a s2指不同物种的系统集</li>
<li>-V: 验证数据，后面可以跟vbr的任意组合；v验证数据记录，结果保存在sample.val(含有错误的类型和严重程度)，错误总结保存在sample.stats；b生成sample.gbf的GenBank文件；r在不做国家检查的情况下验证(Validates without Country Check)</li>
<li>-c: 清理数据，后面可以跟fxdD的任意组合；f修正差异报告中特定类别的产物名称，被修正的产物名称保存在sample.fixedproducts；x扩展注释的features的1-2个核苷酸边界，使得连接gaps或序列末端；D&#x2F;d修正采样日期</li>
<li>-euk：为差异报告的测试指定是真核生物(eukaryotic lineage)</li>
<li>-augustus-fix：针对augustus的错误进行修正</li>
<li>-locus-tag-prefix L6164: 在向NCBI申请BioProject(以及BioSample)后会分配一个locus-tag-prefix前缀，在这里指定，不指定可能在sample.dr中报错FATAL: INCONSISTENT_PROTEIN_ID。</li>
<li>-gaps-min 1 -gaps-unknown 100: 指定序列中的N的含义，如未正确指定，可能在submit.stats中报错SEQ_INST.InternalNsInSeqRaw。-gaps-min 指定代表gap的连续Ns的最小数量，1指把大于等于1个的连续Ns碱基都转换成assembly_gaps。-gaps-unknown 指定代表未知长度的gap的连续Ns的</li>
<li>-l paired-ends: </li>
<li>-j “[organism&#x3D;Bauhinia variegata][isolate&#x3D;pink_petal]”: 用于指定样本名[organism]和个体名[isolate]，会添加进每一条序列的ID里；不指定可能在sample.stats中报错SEQ_DESCR.BioSourceMissing和SEQ_DESCR.NoSourceDescriptor。</li>
<li>-logfile bv.log:指定log文件，如果运q行有错误存放在log文件</li>
<li>-y：添加评论到每一条提交记录，例如-y “Contigs larger than 2kb have been annotated, representing approximately 87% of the total genome”</li>
</ul>
<h3 id="4-2-3-修正错误"><a href="#4-2-3-修正错误" class="headerlink" title="4.2.3. 修正错误"></a>4.2.3. 修正错误</h3><p>根据生成的验证结果修正注释错误，重复table2asn生成sqn文件和修正错误的步骤直至没有错误。</p>
<h4 id="4-2-3-1-查看错误"><a href="#4-2-3-1-查看错误" class="headerlink" title="4.2.3.1. 查看错误"></a>4.2.3.1. 查看错误</h4><p>查看运行table2asn生成的sample.stats，如果有ERROR-level的错误和部分特定的WARNING-level的错误就需要修正，通过sample.val查看错误的具体描述。<br>另外差异报告文件sample.dr中标注为FATAL的错误也需要修正。</p>
<p>下面是一个sample.stats文件的例子：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">Total messages:		86696</span><br><span class="line"></span><br><span class="line">=================================================================</span><br><span class="line">86259 WARNING-level messages exist</span><br><span class="line"></span><br><span class="line">SEQ_FEAT.CDSmRNArange:	219</span><br><span class="line">SEQ_FEAT.NotSpliceConsensusDonor:	300</span><br><span class="line">SEQ_FEAT.NotSpliceConsensusAcceptor:	585</span><br><span class="line">SEQ_FEAT.SeqFeatXrefFeatureMissing:	84578</span><br><span class="line">SEQ_FEAT.ShortExon:	577</span><br><span class="line"></span><br><span class="line">=================================================================</span><br><span class="line">437 ERROR-level messages exist</span><br><span class="line"></span><br><span class="line">SEQ_INST.StopInProtein:	109</span><br><span class="line">SEQ_INST.InternalNsInSeqRaw:	169</span><br><span class="line">SEQ_DESCR.BioSourceMissing:	47</span><br><span class="line">SEQ_DESCR.NoSourceDescriptor:	1</span><br><span class="line">SEQ_FEAT.InternalStop:	109</span><br><span class="line">SEQ_FEAT.NoStop:	2</span><br></pre></td></tr></table></figure>

<h4 id="4-2-3-2-注释规范"><a href="#4-2-3-2-注释规范" class="headerlink" title="4.2.3.2. 注释规范"></a>4.2.3.2. 注释规范</h4><p>报错常常是注释文件的注释不规范导致的，根据报错我总结出几点注释规范：</p>
<ol>
<li>同一个mRNA包含的exons与CDSs的数量和位置几乎一致；区别在于同一个mRNA的exons的起始和终止位置(第一个exon的首和最后一个exon的尾)与mRNA的起始和终止相同，包含5’UTR和3’UTR，CDSs则不包含5’UTR和3’UTR，所以CDSs的起始和终止位置不一定和exon相同。</li>
<li>第9列的产物属性product值不能包括逗号,(因为逗号是多个属性值的分隔符)，竖杠|，不能不包含字母，不能以短横杠-或冒号:结尾。最好先做好product值的过滤filter，再添加到gff注释文件。</li>
</ol>
<p>product值的不规范情况：</p>
<ul>
<li>包含逗号,（因为逗号是分隔符）</li>
<li>包含竖杠|</li>
<li>短横杠-开头或结尾</li>
<li>冒号:开头或结尾</li>
<li>斜杠&#x2F;开头或结尾</li>
<li>不包含字母</li>
<li>括号（包括圆括号和中括号）不完整的</li>
</ul>
<p>在整理基因组结构注释和功能注释时，可以先用table2asn检测错误，以指导注释合并。</p>
<h4 id="4-2-3-3-报错和修正方案"><a href="#4-2-3-3-报错和修正方案" class="headerlink" title="4.2.3.3. 报错和修正方案"></a>4.2.3.3. 报错和修正方案</h4><p>table2asn运行结束后要根据输出文件的报错进行调整，并重新运行直至没有报错再提交给NCBI。</p>
<p>NCBI给出了<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/validation/">Validation and Discrepancy Report Error Explanations</a>和<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genome_validation/">Validation Error Explanations for Genomes</a>可以参考报错含义和解决方案。</p>
<p>下面我总结了一下我遇到的报错类型和修正方案：</p>
<h5 id="4-2-3-3-1-sample-stats-x2F-sample-val文件中报错ERROR"><a href="#4-2-3-3-1-sample-stats-x2F-sample-val文件中报错ERROR" class="headerlink" title="4.2.3.3.1. sample.stats&#x2F;sample.val文件中报错ERROR"></a>4.2.3.3.1. sample.stats&#x2F;sample.val文件中报错ERROR</h5><p>添加<code>pseudo=true</code>属性值的命令举例：<code>sed -i &#39;/ID=Bv00639;/ &#123;s/$/pseudo=true;/g&#125;&#39; sample.gff</code></p>
<ol>
<li><strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong><ul>
<li>当gff的CDS位置有中间终止密码子时报错</li>
<li>可以修改CDS序列位置，去除中间终止密码子。如果无法去除中间终止密码子，可以在sample.gff文件对应的gene行(第三列为gene的那行)的第九列添加<code>pseudo=true</code>属性值代表序列无法被翻译。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.NoStop</strong><ul>
<li>当gff的CDS位置无终止密码子时</li>
<li>可以延长CDS序列位置，直至出现终止密码子。如果无法找到终止密码子，可以在sample.gff文件对应的gene行(第三列为gene的那行)的第九列添加<code>pseudo=true</code>属性值代表序列无法被翻译。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.AbuttingIntervals</strong><ul>
<li>当gff注释的一个基因的exons位置相邻时。</li>
<li>可以合并两个相邻的exons，更改位置。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.SeqLocOrder</strong><ul>
<li>当gff的exon位置有重叠时。</li>
<li>可以参考mRNA和CDS序列的位置更改重叠的exon位置；exons的边界和mRNA一致，数量和CDS一致。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.CDSmRNAXrefLocationProblem</strong>或<strong>SEQ_FEAT.CDSwithMultipleMRNAs</strong><ul>
<li>当gff注释CDS和exon的位置或数量不一致时</li>
<li>可以更改exon或CDS位置和数量。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.ShortIntron</strong><ul>
<li>含有短于11bp的内含子时</li>
<li>可以修正intron的位置使它长于11bp，或者在gene那行第九列添加<code>pseudo=true</code>属性值，或者在table2asn的命令中中加上参数<code>-c s</code>代表标记为”LOW QUALITY PROTEIN”。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.TransLen</strong><ul>
<li>表示蛋白质长度与预测的蛋白质长度不匹配，运行错误</li>
<li>建议重跑table2asn，报错持续存在就写邮件把sample.sqn和运行的命令行发给NCBI(<a href="mailto:&#103;&#101;&#x6e;&#x6f;&#109;&#101;&#115;&#64;&#110;&#x63;&#98;&#105;&#x2e;&#110;&#x6c;&#x6d;&#46;&#110;&#105;&#104;&#x2e;&#103;&#x6f;&#118;">&#103;&#101;&#x6e;&#x6f;&#109;&#101;&#115;&#64;&#110;&#x63;&#98;&#105;&#x2e;&#110;&#x6c;&#x6d;&#46;&#110;&#105;&#104;&#x2e;&#103;&#x6f;&#118;</a>)让帮忙修改这个错误。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.BadInternalCharacter</strong><ul>
<li>表示gff的第9列的属性值attributes中包含不符合规定的字符，例如”|”符号。</li>
<li>删除第9列属性值包含的”|”符号</li>
</ul>
</li>
<li><strong>SEQ_FEAT.BadTrailingHyphen</strong></li>
</ol>
<ul>
<li>当gff的第9列的属性值attributes中以连字符”-“结尾时</li>
<li>删除第9列属性值包含的”-“符号</li>
</ul>
<ol start="11">
<li><strong>SEQ_FEAT.BadTrailingCharacter</strong><ul>
<li>当gff的第9列的属性值attributes中以连字符”:”结尾时。</li>
<li>删除第9列属性值包含的”:”结尾符号</li>
</ul>
</li>
<li>当gff的product属性不符合蛋白产物名称规定时，会生成Adiantum.nelumboides.fixedproducts文件，记录把不符合规定的product属性值改为<strong>hypothetical protein</strong>。可以提取不符合规定的product属性值用于product的filter。</li>
</ol>
<h5 id="4-2-3-3-2-sample-stats-x2F-sample-val文件中的报警WARNING"><a href="#4-2-3-3-2-sample-stats-x2F-sample-val文件中的报警WARNING" class="headerlink" title="4.2.3.3.2. sample.stats&#x2F;sample.val文件中的报警WARNING"></a>4.2.3.3.2. sample.stats&#x2F;sample.val文件中的报警<strong>WARNING</strong></h5><p>部分WARNING也最好修复；比如<strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong>和<strong>SEQ_FEAT.DuplicateFeat</strong>虽然是在WARNING-level messages中，但上传给NCBI后会发邮件让修改后重新提交。</p>
<ol>
<li><strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong><ul>
<li>exon和CDS的位置或数量不一致且有中间终止密码子导致</li>
<li>可以调整exon和CDS的位置和数量；可能引入<strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong>错误，若引入则在gene行的第九列添加<code>pseudo=true</code>属性值；</li>
</ul>
</li>
<li><strong>SEQ_FEAT.DuplicateFeat</strong><ul>
<li>当gff在同样的位置注释了多个gene时</li>
<li>需要删除其中一个。</li>
</ul>
</li>
</ol>
<h5 id="4-2-3-3-3-sample-dr文件中的报错FATAL"><a href="#4-2-3-3-3-sample-dr文件中的报错FATAL" class="headerlink" title="4.2.3.3.3. sample.dr文件中的报错FATAL"></a>4.2.3.3.3. sample.dr文件中的报错FATAL</h5><p><code>cat sample.dr |grep &quot;FATAL&quot;</code>查看FATAL报错信息，如果物种不是细菌则可以忽略BACTERIAL_开头的报错信息；CONTAINED_CDS开头的信息不影响提交，也可暂时忽略。<br>如果product属性值中有错误格式会报错，比如含有不完整的括号，会显示<strong>FATAL: SUSPECT_PRODUCT_NAMES: 9 features contain unbalanced brackets or parentheses</strong>，需要处理(添加括号或者删除括号)。</p>
<h3 id="4-2-4-在线提交"><a href="#4-2-4-在线提交" class="headerlink" title="4.2.4. 在线提交"></a>4.2.4. 在线提交</h3><p>根据上一个部分<strong>提交基因组到NCBI的GenBank</strong>的步骤提交sqn文件到NCBI</p>
<h3 id="4-2-5-提交后修改"><a href="#4-2-5-提交后修改" class="headerlink" title="4.2.5. 提交后修改"></a>4.2.5. 提交后修改</h3><p>提交给NCBI后会收到邮件通知处理进度，如果有错误，会在提交网站显示Error提交状态，并收到需要更改的邮件。根据邮件提示修正注释错误，重复table2asn生成sqn文件和修正错误的步骤直至没有错误在NCBI修正错误(提交新的sqn文件)。</p>
<p>下面记录了我遇到的邮件中提到的错误和修改方案：</p>
<ol>
<li><p><strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong><br>exon和CDS的位置或数量不一致且有中间终止密码子导致。可以调整exon和CDS的位置和数量；可能引入<strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong>错误，若引入则在gene行的第九列添加<code>pseudo=true</code>属性值；</p>
</li>
<li><p><strong>SEQ_FEAT.DuplicateFeat</strong><br>这个错误是同一位置注释到多个基因引起的，可以提取邮件中的gene ID(重复基因的第二个)信息，保存成gene ID list，然后用命令<code>grep -v -f geneID.list sample.gff &gt;sample.revised.gff</code>删除list里的注释信息。</p>
</li>
<li><p><strong>Discrepancy</strong><br>Discrepancy.txt文件中保存了一些不符合标准的报错，比如product属性值中有横杠起始的元素，或者括号不完全，根据Discrepancy.txt文件指出的错误一条条修改。</p>
</li>
</ol>
<p><code>cat sample.dr |grep &quot;FATAL&quot;</code>查看FATAL报错信息，如果物种不是细菌则可以忽略BACTERIAL_开头的报错信息；CONTAINED_CDS开头的信息不影响提交，也可暂时忽略。<br>如果product属性值中有不完整的括号，会显示<strong>FATAL: SUSPECT_PRODUCT_NAMES: 9 features contain unbalanced brackets or parentheses</strong>，需要处理需要处理(添加括号或者删除括号)。</p>
<h2 id="4-3-提交状态"><a href="#4-3-提交状态" class="headerlink" title="4.3. 提交状态"></a>4.3. 提交状态</h2><p>在提交网页，提交后会显示提交状态，经过一轮机器自动验证和两轮人工检查后可公开数据。</p>
<p>提交状态含义：</p>
<ol>
<li>Unfinished</li>
</ol>
<ul>
<li>提交未完成</li>
<li>提交如果没有最后确认，会保存已填写内容，随时可以登陆账号点击未完成的提交号继续提交。</li>
</ul>
<ol start="2">
<li>Queued</li>
</ol>
<ul>
<li>提交在等待自动验证</li>
<li>刚提交的任务会进入自动验证环节</li>
</ul>
<ol start="3">
<li>Error</li>
</ol>
<ul>
<li>错误</li>
<li>有文件有错误需要修正或者再次提交；会提供错误具体信息的文件和邮件通知。</li>
</ul>
<ol start="4">
<li>Processing</li>
</ol>
<ul>
<li>处理中</li>
<li>提交通过自动验证，等待NCBI员工手动检查</li>
</ul>
<ol start="5">
<li>Processing and accession number present</li>
</ol>
<ul>
<li>处理中，并且显示了序列号</li>
<li>提交通过了NCBI员工的首次检查，等待NCBI员工的最终检查</li>
<li>如果提交有问题最终检查员工会发邮件告知</li>
<li>如果指定了释放日期，提交会在释放日期前再做最终检查；也就是说释放日期前都保持在这个提交状态</li>
</ul>
<ol start="6">
<li>Processed</li>
</ol>
<ul>
<li>已处理</li>
<li>数据已被公开释放，在这之前做的任何更改都会包含在释放数据中，但在提交网站不会显示修改。</li>
</ul>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genomesubmit/">genomes submission guide</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/table2asn/">table2asn</a></li>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/DOCUMENTATION/">table2asn documentation</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/validation/">Validation and Discrepancy Report Error Explanations</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genome_validation/">Validation Error Explanations for Genomes</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/asndisc/">Discrepancy Report</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/genome-vendor/sequin/blob/master/errmsg/valid.msg">MODULE valid</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genomes_gff/">annotating genomes with gff3</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2022/03/22/omics_genome.submit/" data-id="cl1aiqcrz0046m4vke1i4cmup" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_geneflow_treemix" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2022/03/20/bioinfo_geneflow_treemix/" class="article-date">
  <time datetime="2022-03-20T14:00:00.000Z" itemprop="datePublished">2022-03-20</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1296539261&auto=1&height=32"></iframe></div>

<p>这篇博客记录的分析步骤是：</p>
<ol>
<li>使用基因组尺度的snp数据计算等位基因频率</li>
<li>转换成TreeMix的输入文件格式</li>
<li>用TreeMix推断基因流</li>
<li>确定最佳基因流数量</li>
<li>绘制系统发育网络图</li>
</ol>
<h1 id="1-TreeMix"><a href="#1-TreeMix" class="headerlink" title="1. TreeMix"></a>1. TreeMix</h1><p>TreeMix是一个利用全基因组尺度的等位基因频率数据推断居群的分化和和杂合（基因流动或基因渗入）的一个软件，适用于推断一个物种多个居群的系统发育网。TreeMix用多居群的等位基因频率来推测最大似然树(ML树)，并可推断指定数量基因流事件。</p>
<h2 id="1-1-TreeMix-install"><a href="#1-1-TreeMix-install" class="headerlink" title="1.1. TreeMix install"></a>1.1. TreeMix install</h2><ol>
<li><p><code>conda install -c bioconda treemix</code> #现在conda可以下到最新版本treemix 1.13</p>
</li>
<li><p>在<a target="_blank" rel="noopener" href="https://bitbucket.org/nygcresearch/treemix/downloads/">treemix software</a>下载压缩包再安装</p>
</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">tar -xvf treemix-1.0.tar.gz</span><br><span class="line">cd treemix-1.0</span><br><span class="line">./configure</span><br><span class="line">make</span><br><span class="line">make install</span><br></pre></td></tr></table></figure>

<h2 id="1-2-准备文件"><a href="#1-2-准备文件" class="headerlink" title="1.2. 准备文件"></a>1.2. 准备文件</h2><ol>
<li>sample.snp.vcf.gz：基因组尺度的snp文件，vcf格式</li>
<li>sample.tre【optional】：居群的系统发育树文件，Newick格式，把支持率等信息删除；也可以不提供，用TreeMix推断ML树。<strong>这个树文件是居群树文件，不是个体树文件</strong>。</li>
</ol>
<h2 id="1-3-分析步骤"><a href="#1-3-分析步骤" class="headerlink" title="1.3. 分析步骤"></a>1.3. 分析步骤</h2><h3 id="1-3-1-计算等位基因频率"><a href="#1-3-1-计算等位基因频率" class="headerlink" title="1.3.1. 计算等位基因频率"></a>1.3.1. 计算等位基因频率</h3><ol>
<li><p><code>vcftools --gzvcf sample.snp.vcf.gz --plink-tped --out sample</code> # 转换为tped格式，生成sample.tped和sample.tfam文件。<br>如果报错<strong>Unrecognized values used for CHROM: scaffold_1 - Replacing with 0</strong>，可以用<code>bcftools view -H sample.snp.vcf.gz | cut -f 1 | uniq | awk &#39;&#123;print $0&quot;\t&quot;$0&#125;&#39; &gt; chrom-map.txt</code>生成染色体ID的文件，然后再用–chrom-map指定文件来运行<code>vcftools --gzvcf sample.snp.vcf.gz --plink-tped --chrom-map chrom-map.txt --out sample</code>。</p>
</li>
<li><p>sample.tfam文件共6列，前两列都为样本ID，修改第一列数据为群体ID。（可以是同一物种不同种群作为单独群体，也可以是一个物种作为一个群体，这里sample的一个物种作为一个群体，群体ID为物种名）。</p>
</li>
<li><p><code>cat sample.tfam |awk &#39;&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$1&#125;&#39; &gt;sample.pop.cov</code>编辑文件sample.pop.cov，格式为：共三列，前两列与修改后的sample.tfam前两列一样，为群体ID和样本ID，第三列和第一列一致，tab分隔。</p>
</li>
<li><p>计算等位基因组的频率</p>
</li>
</ol>
<ul>
<li><code>plink --threads 12 --tfile sample --freq --allow-extra-chr --within sample.pop.cov</code> # 计算等位基因组的频率，生成plink.frq.strat和plink.nosex文件，加上–allow-extra-chr是防止识别不了染色体名称。</li>
<li><code>pigz plink.frq.strat</code> #压缩等位基因频率文件</li>
</ul>
<ol start="5">
<li>转换格式【耗时小时计】</li>
</ol>
<ul>
<li><code>python2 /path/to/plink2treemix.py plink.frq.strat.gz sample.treemix.in.gz</code> #用treemix自带脚本进行格式转换，notes：输入输出都为压缩文件，plink2treemix.py使用python2并需要绝对路径（否则报错）。</li>
</ul>
<h3 id="1-3-2-treemix推断基因流"><a href="#1-3-2-treemix推断基因流" class="headerlink" title="1.3.2. treemix推断基因流"></a>1.3.2. treemix推断基因流</h3><ol>
<li><p>treemix默认功能<br><code>treemix -i sample.treemix.in.gz -o sample</code> # treemix的默认功能是在所有位点都独立的假设下建一棵种群ML树</p>
</li>
<li><p>TreeMix参数解释</p>
</li>
</ol>
<ul>
<li>-i 指定基因频率输入文件</li>
<li>-o 指定输出文件前缀</li>
<li>-tf 【可选】指定树文件，指定后就使用指定树的拓扑结构(最好把支持率和其他无关信息都删除，只留最简单的Newick格式)，否则treemix会推断拓扑</li>
<li>-root 【可选】指定外类群(指定的是居群名称)，多个用逗号分隔；最好指定，否则后面plot_tree画树没找到更换外类群的参数会很麻烦</li>
<li>-m 为the number of migration edges即基因渗入的次数</li>
<li>-k 1000 因为SNP之间不是独立位点，为了避免连锁不平衡，用k参数指定SNP数量有连锁，比如这里指定用1000个SNP组成的blocks评估协方差矩阵</li>
<li>-se 计算迁移权重的标准误差(计算成本高)，如果想省时间可以不用这个参数</li>
<li>-bootstrap 为了判断给定树拓扑的可信度，在blocks运行bootstrap重复，没太理解</li>
<li>-global 在增加所有种群后做一轮全局重组。</li>
<li>noss 关闭样本量校正。TreeMix计算协方差会考虑每个种群的样本量，有些情况(如果有种群的样本只有1个)会过度校正，可以关闭。</li>
<li>-g old.vertices.gz old.edges.gz #使用之前生成的树和图结果，用-g指定之前的两个结果文件</li>
<li>-cor_mig known_events and -climb #合并已知的迁移事件</li>
</ul>
<ol start="3">
<li><p>推荐参数<br>检测物种间基因流，每个物种一个个体或多个个体的数据，用的参数：<br><code>treemix -i sample.treemix.in.gz -o sample -tf sample.tre -root outgroup1,outgroup2 -m 3 -k 1000 -se -bootstrap -global -noss</code></p>
</li>
<li><p>多次分析以评估最佳m值<br>比如m取1-10(常用1-5,1-10)，每个m值重复5次(至少两次)。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">for m in &#123;1..10&#125;</span><br><span class="line">do</span><br><span class="line">	&#123;</span><br><span class="line">	for i in &#123;1..5&#125;</span><br><span class="line">	do</span><br><span class="line">		&#123;</span><br><span class="line">			nohup time treemix -i sample.treemix.in.gz -o sample.$&#123;m&#125;.$&#123;i&#125; -tf sample.tre -root outgroup1,outgroup2 -m $&#123;m&#125; -k 1000 -se -bootstrap -global -noss&gt;$&#123;m&#125;.$&#123;i&#125;.out 2&gt;&amp;1 &amp;</span><br><span class="line">		&#125;</span><br><span class="line">	done</span><br><span class="line">	&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>
</li>
<li><p>运行结果<br>每一次运行都将生成一套后缀为llik,modelcov,treeout,vertices,cov,covse,edges的文件，这些文件都不大(KB级别)，但还会生成一个300+MB的out文件(out文件没报错就可以删除了)。</p>
</li>
</ol>
<ul>
<li>.llik：保存了从0到指定基因渗入次数m的始末似然值</li>
<li>.treeout.gz: 拟合树模型(fitted tree model)和迁移事件(migration events)。第一行是Newick格式的ML树，剩下行包含迁移边界(migration edges)</li>
<li>.cov.gz: 种群间的协方差矩阵(covariance matrix)</li>
<li>.covse.gz: 协方差矩阵的标准误差(standard errors)</li>
<li>.modelcov.gz: 模型的拟合协方差(fitted covariance)</li>
<li>.vertices.gz和.edges.gz: 包含推断图的内部结构</li>
</ul>
<h3 id="1-3-3-确认最佳m值"><a href="#1-3-3-确认最佳m值" class="headerlink" title="1.3.3. 确认最佳m值"></a>1.3.3. 确认最佳m值</h3><p>R包OptM分析treemix的结果，确认最佳m值(Deltam值最小情况下的m值)。</p>
<p>在treemix运行的结果目录下运行R，需要使用每一次运行的.llik,.cov.gz,.modelcov.gz三个配套结果文件，每个m值至少重复&#x2F;迭代两次，否则报错。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span>library<span class="punctuation">(</span>OptM<span class="punctuation">)</span> <span class="comment">#载入OptM包</span></span><br><span class="line"><span class="operator">&gt;</span>evanno<span class="operator">=</span>optM<span class="punctuation">(</span><span class="string">&quot;./&quot;</span><span class="punctuation">,</span>method <span class="operator">=</span> <span class="string">&quot;Evanno&quot;</span>，tsv<span class="operator">=</span><span class="string">&quot;evanno.txt&quot;</span><span class="punctuation">)</span> <span class="comment">#用Evanno方法(默认)分析treemix结果文件，保存结果到evanno.txt文件，里面就有每个m对应的Deltam值(第15列)，Deltam值最小时对应的m即时最佳m值，这里的例子是m=3时最低。</span></span><br><span class="line">  m runs   mean<span class="punctuation">(</span>Lm<span class="punctuation">)</span>   sd<span class="punctuation">(</span>Lm<span class="punctuation">)</span>   <span class="built_in">min</span><span class="punctuation">(</span>Lm<span class="punctuation">)</span>   <span class="built_in">max</span><span class="punctuation">(</span>Lm<span class="punctuation">)</span></span><br><span class="line"><span class="number">1</span> <span class="number">0</span>   <span class="number">32</span> <span class="operator">-</span><span class="number">329764.72</span> <span class="number">4802.416</span> <span class="operator">-</span><span class="number">336972.0</span> <span class="operator">-</span><span class="number">320838.0</span></span><br><span class="line"><span class="number">2</span> <span class="number">1</span>    <span class="number">5</span> <span class="operator">-</span><span class="number">239329.60</span> <span class="number">3018.974</span> <span class="operator">-</span><span class="number">243918.0</span> <span class="operator">-</span><span class="number">236221.0</span></span><br><span class="line"><span class="number">3</span> <span class="number">2</span>    <span class="number">5</span> <span class="operator">-</span><span class="number">139018.20</span> <span class="number">5332.074</span> <span class="operator">-</span><span class="number">147433.0</span> <span class="operator">-</span><span class="number">134177.0</span></span><br><span class="line"><span class="number">4</span> <span class="number">3</span>    <span class="number">5</span> <span class="operator">-</span><span class="number">116765.80</span> <span class="number">1533.134</span> <span class="operator">-</span><span class="number">119301.0</span> <span class="operator">-</span><span class="number">115329.0</span></span><br><span class="line"><span class="number">5</span> <span class="number">4</span>    <span class="number">5</span>  <span class="operator">-</span><span class="number">94811.20</span> <span class="number">2239.830</span>  <span class="operator">-</span><span class="number">97001.2</span>  <span class="operator">-</span><span class="number">91664.1</span></span><br><span class="line"><span class="number">6</span> <span class="number">5</span>    <span class="number">2</span>  <span class="operator">-</span><span class="number">86727.45</span> <span class="number">1860.469</span>  <span class="operator">-</span><span class="number">88043.0</span>  <span class="operator">-</span><span class="number">85411.9</span></span><br><span class="line"><span class="number">7</span> <span class="number">6</span>    <span class="number">5</span>  <span class="operator">-</span><span class="number">79769.74</span> <span class="number">1545.658</span>  <span class="operator">-</span><span class="number">82041.1</span>  <span class="operator">-</span><span class="number">77782.6</span></span><br><span class="line"><span class="number">8</span> <span class="number">7</span>    <span class="number">5</span>  <span class="operator">-</span><span class="number">68844.08</span> <span class="number">2924.953</span>  <span class="operator">-</span><span class="number">73560.4</span>  <span class="operator">-</span><span class="number">66420.4</span></span><br><span class="line">      L<span class="string">&#x27;(m)   sdL&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span>  minL<span class="string">&#x27;(m)   maxL&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span>    L<span class="string">&#x27;&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span></span><br><span class="line"><span class="number">1</span>        <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span>         <span class="literal">NA</span>        <span class="literal">NA</span></span><br><span class="line"><span class="number">2</span>  <span class="number">90435.12</span> <span class="number">1783.4423</span> <span class="number">88651.676</span>  <span class="number">92218.561</span>  <span class="number">9876.281</span></span><br><span class="line"><span class="number">3</span> <span class="number">100311.40</span> <span class="number">2313.0997</span> <span class="number">97998.300</span> <span class="number">102624.500</span> <span class="number">78059.000</span></span><br><span class="line"><span class="number">4</span>  <span class="number">22252.40</span> <span class="number">3798.9398</span> <span class="number">18453.460</span>  <span class="number">26051.340</span>   <span class="number">297.800</span></span><br><span class="line"><span class="number">5</span>  <span class="number">21954.60</span>  <span class="number">706.6962</span> <span class="number">21247.904</span>  <span class="number">22661.296</span> <span class="number">13870.850</span></span><br><span class="line"><span class="number">6</span>   <span class="number">8083.75</span>  <span class="number">379.3613</span>  <span class="number">7704.389</span>   <span class="number">8463.111</span>  <span class="number">1126.040</span></span><br><span class="line"><span class="number">7</span>   <span class="number">6957.71</span>  <span class="number">314.8111</span>  <span class="number">6642.899</span>   <span class="number">7272.521</span>  <span class="number">3967.950</span></span><br><span class="line"><span class="number">8</span>  <span class="number">10925.66</span> <span class="number">1379.2958</span>  <span class="number">9546.364</span>  <span class="number">12304.956</span>        <span class="literal">NA</span></span><br><span class="line">    sdL<span class="string">&#x27;&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span> minL<span class="string">&#x27;&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span> maxL<span class="string">&#x27;&#x27;</span><span class="punctuation">(</span>m<span class="punctuation">)</span>     Deltam</span><br><span class="line"><span class="number">1</span>         <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span>         <span class="literal">NA</span></span><br><span class="line"><span class="number">2</span>  <span class="number">529.65744</span>  <span class="number">9346.624</span> <span class="number">10405.939</span>  <span class="number">3.2714033</span></span><br><span class="line"><span class="number">3</span> <span class="number">1485.84010</span> <span class="number">76573.160</span> <span class="number">79544.840</span> <span class="number">14.6395203</span></span><br><span class="line"><span class="number">4</span> <span class="number">3092.24366</span> <span class="operator">-</span><span class="number">2794.444</span>  <span class="number">3390.044</span>  <span class="number">0.1942427</span></span><br><span class="line"><span class="number">5</span>  <span class="number">327.33487</span> <span class="number">13543.515</span> <span class="number">14198.185</span>  <span class="number">6.1928138</span></span><br><span class="line"><span class="number">6</span>   <span class="number">64.55023</span>  <span class="number">1061.490</span>  <span class="number">1190.590</span>  <span class="number">0.6052453</span></span><br><span class="line"><span class="number">7</span> <span class="number">1064.48476</span>  <span class="number">2903.465</span>  <span class="number">5032.435</span>  <span class="number">2.5671598</span></span><br><span class="line"><span class="number">8</span>         <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span>         <span class="literal">NA</span></span><br><span class="line">    mean<span class="punctuation">(</span>f<span class="punctuation">)</span>        sd<span class="punctuation">(</span>f<span class="punctuation">)</span></span><br><span class="line"><span class="number">1</span>        <span class="literal">NA</span>           <span class="literal">NA</span></span><br><span class="line"><span class="number">2</span> <span class="number">0.9776741</span> <span class="number">1.214193e-04</span></span><br><span class="line"><span class="number">3</span> <span class="number">0.9837528</span> <span class="number">7.790282e-04</span></span><br><span class="line"><span class="number">4</span> <span class="number">0.9867460</span> <span class="number">2.483868e-04</span></span><br><span class="line"><span class="number">5</span> <span class="number">0.9897553</span> <span class="number">8.854187e-05</span></span><br><span class="line"><span class="number">6</span> <span class="number">0.9909923</span> <span class="number">5.825696e-05</span></span><br><span class="line"><span class="number">7</span> <span class="number">0.9915973</span> <span class="number">8.194263e-05</span></span><br><span class="line"><span class="number">8</span> <span class="number">0.9923055</span> <span class="number">2.114830e-04</span></span><br><span class="line"></span><br><span class="line"><span class="operator">&gt;</span>plot_optM<span class="punctuation">(</span>evanno<span class="punctuation">,</span>method <span class="operator">=</span> <span class="string">&quot;Evanno&quot;</span><span class="punctuation">,</span>pdf<span class="operator">=</span><span class="string">&quot;evanno.pdf&quot;</span><span class="punctuation">)</span> <span class="comment">#把结果作图，更直观判断最佳m值，保存成evanno.pdf</span></span><br><span class="line"></span><br><span class="line"><span class="operator">&gt;</span>linear<span class="operator">=</span>optM<span class="punctuation">(</span><span class="string">&quot;./&quot;</span><span class="punctuation">,</span>method <span class="operator">=</span> <span class="string">&quot;linear&quot;</span><span class="punctuation">,</span>tsv<span class="operator">=</span><span class="string">&quot;linear.txt&quot;</span><span class="punctuation">)</span> <span class="comment">#还可以换linear线性回归方法分析一下treemix结果，m作为横坐标，Log Likelihood做纵坐标，保存结果到linear.txt文件，但不知道为啥没报错但没有结果文件输出</span></span><br><span class="line"><span class="operator">&gt;</span>plot_optM<span class="punctuation">(</span>linear<span class="punctuation">,</span>method <span class="operator">=</span> <span class="string">&quot;linear&quot;</span><span class="punctuation">,</span>pdf<span class="operator">=</span><span class="string">&quot;linear.pdf&quot;</span><span class="punctuation">)</span> <span class="comment">#结果作图，这个结果我没找到怎么看最佳m值，猜测是在图中斜率变化最大的位置对应的m值，标记了黑色的change points处，这里是m在2-3的位置，接近2。</span></span><br></pre></td></tr></table></figure>

<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_geneflow.treemix_evanno.png?raw=true" width=80% title="linear" align=center/>

<p><strong><p align="center">Figure 1. evanno 示例</p></strong></p>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_geneflow.treemix_linear.png?raw=true" width=80% title="linear" align=center/>

<p><strong><p align="center">Figure 2. linear 示例</p></strong></p>
<h3 id="1-3-4-最佳m值作图"><a href="#1-3-4-最佳m值作图" class="headerlink" title="1.3.4. 最佳m值作图"></a>1.3.4. 最佳m值作图</h3><p>用最佳m值对应的结果文件作图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span>source<span class="punctuation">(</span><span class="string">&quot;path/to/plotting_funcs.R&quot;</span><span class="punctuation">)</span> <span class="comment">#载入treemix/src文件夹中的R脚本</span></span><br><span class="line"><span class="operator">&gt;</span>plot_tree<span class="punctuation">(</span><span class="string">&quot;sample.treemix.3.1&quot;</span><span class="punctuation">)</span> <span class="comment">#用sample.treemix.3.1结果作图</span></span><br></pre></td></tr></table></figure>



<p><strong>notes</strong>：</p>
<ul>
<li>虽然TreeMix的输入是个体的snp.vcf计算的基因型频率数据，但推断的系统发育网是居群的(居群名称组成的系统结构)，不是个体的系统发育。</li>
<li>sample.tre是居群树(据群名称组成的树)，不是个体树</li>
<li>treemix的参数-root指定的也是居群名称</li>
</ul>
<h1 id="2-reference"><a href="#2-reference" class="headerlink" title="2. reference"></a>2. reference</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.nature.com/articles/npre.2012.6956.1">TreeMix paper</a></li>
<li><a target="_blank" rel="noopener" href="https://bitbucket.org/nygcresearch/treemix/downloads/">TreeMix software</a></li>
<li><a target="_blank" rel="noopener" href="https://bitbucket.org/nygcresearch/treemix/downloads/treemix_manual_10_1_2012.pdf">TreeMix manual</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2022/03/20/bioinfo_geneflow_treemix/" data-id="cl1aiqcrl002km4vk4t3nhduc" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_evolutionary.tree" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2022/01/24/bioinfo_evolutionary.tree/" class="article-date">
  <time datetime="2022-01-24T13:00:00.000Z" itemprop="datePublished">2022-01-24</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=283100&auto=1&height=32"></iframe></div>

<h1 id="1-进化树"><a href="#1-进化树" class="headerlink" title="1. 进化树"></a>1. 进化树</h1><p>进化树的相关知识可以参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/20/bio_evolutionary.tree/">博文进化树相关知识</a>。</p>
<h1 id="2-进化树的绘制"><a href="#2-进化树的绘制" class="headerlink" title="2. 进化树的绘制"></a>2. 进化树的绘制</h1><p>进化树的绘制是指把newick格式的树文件转换成树图形，有许多软件可以使用，在线网页版（itol，EvolView），也有R包（ggtree），或者软件figtree。<br>如果是简单的树绘制，推荐在线网页<strong>itol</strong>；如果需要精细的注释，推荐R包<strong>ggtree</strong>。</p>
<h2 id="2-1-在线绘制进化树"><a href="#2-1-在线绘制进化树" class="headerlink" title="2.1. 在线绘制进化树"></a>2.1. 在线绘制进化树</h2><ol>
<li><a target="_blank" rel="noopener" href="https://itol.embl.de/">itol</a></li>
<li><a target="_blank" rel="noopener" href="https://www.evolgenius.info/evolview/#login">EvolView</a></li>
</ol>
<h2 id="2-2-R包ggtree【推荐】"><a href="#2-2-R包ggtree【推荐】" class="headerlink" title="2.2. R包ggtree【推荐】"></a>2.2. R包ggtree【推荐】</h2><p>R包treeio用于解析各种格式的进化树文件，ggtree用于绘制进化树和注释。</p>
<h3 id="2-2-1-树文件的导入"><a href="#2-2-1-树文件的导入" class="headerlink" title="2.2.1. 树文件的导入"></a>2.2.1. 树文件的导入</h3><h4 id="2-2-1-1-treeio导入树"><a href="#2-2-1-1-treeio导入树" class="headerlink" title="2.2.1.1. treeio导入树"></a>2.2.1.1. treeio导入树</h4><p>treeio支持标准格式树（包括Newick树格式，NEXUS树格式，New Hampshire eXtended(NHX)树格式，Jplace树格式）和建树软件输出树（RAxML，iqtree，BEAST，MrBayes，PAML(BASEML,CODEML)，r8s，MEGA，HyPhy，Phylip，以及生成Jplace格式树的EPA和PPLACER），可以用文本输入和文件输入两种形式导入和解析树。</p>
<p><a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/ggtree-book/chapter-treeio.html">treeio tutorial</a>解释了各种建树软件输出的树包含的信息。</p>
<ol>
<li>文本输入<br>当树文件的文本比较简单时，可以直接把文本读到变量中，再解析。</li>
</ol>
<ul>
<li><p>Newick格式树<br><code>read.newick()</code>：用read.newick读取Newick格式的树文本，有枝长信息，缺点是没有办法添加注释信息</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>treeio<span class="punctuation">)</span></span><br><span class="line">treetext<span class="operator">&lt;-</span><span class="string">&quot;((((Mc:0.418672,Eg:0.201388):0.148853,((Pt:0.219587,(Cs:0.209405,(Gr:0.224188,At:0.492461):0.027774):0.012564):0.019343,(Pp:0.194515,(Mt:0.354580,Cl:0.338644:0.022063):0.019111):0.011710):0.029914,Vv:0.169100):0.241611,Mg:0.241611);&quot;</span>  </span><br><span class="line">tree <span class="operator">&lt;-</span> read.newick<span class="punctuation">(</span>textConnection<span class="punctuation">(</span>treetext<span class="punctuation">)</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>NHX格式树<br><code>read.nhx()</code>：用read.nhx()读取NHX格式的树文本，节点后中括号内[&amp;&amp;NHX]添加值注释信息，用冒号:分隔多个注释信息，缺点是没有办法添加范围注释，比如95%置信区间注释</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">treetext<span class="operator">&lt;</span><span class="string">&quot;((((Mc:0.418672[&amp;&amp;NHX:I=5728:D=1170:G=Malvids],Eg:0.201388[&amp;&amp;NHX:I=1420:D=2519:G=Malvids])[&amp;&amp;NHX:C=1]:0.148853,((Pt:0.219587[&amp;&amp;NHX:I=5355:D=1003:G=Fabids],(Cs:0.209405[&amp;&amp;NHX:I=1026:D=3516:G=Malvids],(Gr:0.224188[&amp;&amp;NHX:I=4875:D=998:G=Malvids],At:0.492461[&amp;&amp;NHX:I=1776:D=2891:G=Malvids]):0.027774[&amp;&amp;NHX:C=2]):0.012564[&amp;&amp;NHX:C=3]):0.019343[&amp;&amp;NHX:C=4],(Pp:0.194515[&amp;&amp;NHX:I=1159:D=2949:G=Fabids],(Mt:0.354580[&amp;&amp;NHX:I=3310:D=1984:G=Fabids],Cl:0.338644[&amp;&amp;NHX:I=794:D=3735:G=Fabids]):0.022063[&amp;&amp;NHX:C=5]):0.019111[&amp;&amp;NHX:C=6]):0.011710[&amp;&amp;NHX:C=7]):0.029914[&amp;&amp;NHX:C=8],Vv:0.169100[&amp;&amp;NHX:I=1522:D=3045:G=Vitales]):0.241611[&amp;&amp;NHX:C=9],Mg:0.241611[&amp;&amp;NHX:I=2278:D=1909:G=Asterids])[&amp;&amp;NHX:C=10];&quot;</span>  </span><br><span class="line">tree <span class="operator">&lt;-</span> read.nhx<span class="punctuation">(</span>textConnection<span class="punctuation">(</span>treetext<span class="punctuation">)</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure></li>
</ul>
<ol start="2">
<li>treeio导入树文件<br>treeio支持解析大部分软件输出的树文件</li>
</ol>
<table>
<thead>
<tr>
<th>解析函数Parser function</th>
<th>功能</th>
</tr>
</thead>
<tbody><tr>
<td>read.newick</td>
<td>解析Newick格式文件</td>
</tr>
<tr>
<td>read.nexus</td>
<td>解析NEXUS格式文件</td>
</tr>
<tr>
<td>read.nhx</td>
<td>解析NHX格式文件</td>
</tr>
<tr>
<td>read.jplace</td>
<td>解析jplace格式文件，包括EPA和pplacer的输出文件</td>
</tr>
<tr>
<td>read.jtree</td>
<td>解析jtree格式文件</td>
</tr>
<tr>
<td>read.fasta</td>
<td>解析fasta序列</td>
</tr>
<tr>
<td>read.iqtree</td>
<td>解析IQ-Tree的Newick格式输出，支持解析SH-aLRT和UFBoot支持率</td>
</tr>
<tr>
<td>read.raxml</td>
<td>解析RAxML的输出</td>
</tr>
<tr>
<td>read.r8s</td>
<td>解析r8s的输出</td>
</tr>
<tr>
<td>read.mrbayes</td>
<td>解析MrBayes的输出</td>
</tr>
<tr>
<td>read.beast</td>
<td>解析BEAST的输出</td>
</tr>
<tr>
<td>read.astral</td>
<td>解析ASTRAL的输出</td>
</tr>
<tr>
<td>read.mega</td>
<td>解析MEGA的Nexus格式输出</td>
</tr>
<tr>
<td>read.codeml</td>
<td>解析CodeML的输出，rst和mlc文件</td>
</tr>
<tr>
<td>read.paml_rst</td>
<td>解析rst文件，BaseML或CodeML的输出</td>
</tr>
<tr>
<td>read.hyphy</td>
<td>解析HyPhy输出文件</td>
</tr>
<tr>
<td>…</td>
<td>…</td>
</tr>
</tbody></table>
<p>用对应的解析函数读取相应格式的文件，例如读取Newick格式文件用<code>tree&lt;-read.newick(&quot;newick.tre&quot;)</code>,读取beast输出的树文件用<code>tree&lt;-read.beast(&quot;beast.tre&quot;)</code>。</p>
<ol start="3">
<li>read.mega&#x2F;read.beast读取NEXUS格式文件</li>
</ol>
<ul>
<li>NEXUS格式把Newick树和其他信息组合，每种信息单独成块(blocks)，可扩展块来实现添加新类别的信息，NEXUS格式可以看作Newick树加注释信息的组合。</li>
<li>NEXUS格式树包括TAXA块和TREE块，常常也会包括储存序列比对(Phylip格式)的DATA块，有些软件有自己特定的块。</li>
<li>每个块都是<code>Begin Blockname;</code>开头，<code>End;</code>加一个空行结尾。</li>
<li>NEXUS格式的优点是可以实现尽可能地注释树，包括值注释和区间注释(例如95%置信区间)，缺点是格式稍微复杂，必须文件读取。</li>
<li>我自己尝试用read.nexus读取NEXUS文件，读取没问题，但如果想使用注释信息(比如geom_text&#x2F;geom_label2使用注释，geom_range来显示区间注释)就会报错；但是用read.mega&#x2F;read.beast读取NEXUS文件就可以顺利使用注释信息，查看了treeio里的示例文件，read.mega和read.beast里的NEXUS文件格式是一样的，区别在于beast只能读NEXUS文件，mega还可以都Newick文件。</li>
</ul>
<p>一个NEXUS文件的例子，treeio包自带的mtCDNA_timetree.nex文件，可以参考示例文件格式自己写一个NEXUS格式的树文件用于ggtree：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">#NEXUS</span><br><span class="line"></span><br><span class="line">Begin Taxa;</span><br><span class="line">	Dimensions ntax=7;</span><br><span class="line">	TaxLabels</span><br><span class="line">		homo_sapiens</span><br><span class="line">		chimpanzee</span><br><span class="line">		bonobo</span><br><span class="line">		gorilla</span><br><span class="line">		orangutan</span><br><span class="line">		sumatran</span><br><span class="line">		gibbon</span><br><span class="line">		;</span><br><span class="line">End;</span><br><span class="line"></span><br><span class="line">Begin Trees;</span><br><span class="line">	Translate</span><br><span class="line">		1 homo_sapiens,</span><br><span class="line">		2 chimpanzee,</span><br><span class="line">		3 bonobo,</span><br><span class="line">		4 gorilla,</span><br><span class="line">		5 orangutan,</span><br><span class="line">		6 sumatran,</span><br><span class="line">		7 gibbon</span><br><span class="line">		;</span><br><span class="line">tree TREE1 = [&amp;R] (((((2[&amp;rate=8.21387638458966E-001,branch_length=6.27635980985610E-002]:7.76603755686346E-002,3[&amp;rate=8.08047027678532E-001,branch_length=6.17442197999256E-002]:7.76603755686346E-002)[&amp;rate=8.14690026837604E-001,branch_length=1.06551791913040E-001,reltime=7.76603755686346E-002,reltime_stderr=1.66322225004909E-002,reltime_95%_CI=&#123;4.50612194676725E-002,1.10259531669597E-001&#125;,data_coverage=100%]:1.32925460715114E-001,1[&amp;rate=1.14839817218159E+000,branch_length=2.37947876847083E-001]:2.10585836283749E-001)[&amp;rate=9.67258258023612E-001,branch_length=6.82392048602499E-002,reltime=2.10585836283749E-001,reltime_stderr=3.66695942566917E-002,reltime_95%_CI=&#123;1.38713431540633E-001,2.82458241026865E-001&#125;,data_coverage=100%]:7.17020095880709E-002,4[&amp;rate=9.49741644253964E-001,branch_length=2.63789706511460E-001]:2.82287845871820E-001)[&amp;rate=9.58459935726878E-001,branch_length=2.94031112947650E-001,reltime=2.82287845871820E-001,reltime_stderr=4.16686354669857E-002,reltime_95%_CI=&#123;2.00617320356528E-001,3.63958371387112E-001&#125;,data_coverage=100%]:3.11787816307430E-001,(5[&amp;rate=1.34822289815278E+000,branch_length=1.52886612438902E-001]:1.15251762438322E-001,6[&amp;rate=8.34007596053647E-001,branch_length=9.45753081954446E-002]:1.15251762438322E-001)[&amp;rate=1.06039055929072E+000,branch_length=4.99576332711077E-001,reltime=1.15251762438322E-001,reltime_stderr=1.98603068330366E-002,reltime_95%_CI=&#123;7.63255610455702E-002,1.54177963831074E-001&#125;,data_coverage=100%]:4.78823899740928E-001)[&amp;rate=1.00813782158154E+000,branch_length=2.10619528131437E-001,reltime=5.94075662179250E-001,reltime_stderr=7.90987567922010E-002,reltime_95%_CI=&#123;4.39042098866536E-001,7.49109225491964E-001&#125;,data_coverage=100%]:2.12333512272958E-001,7[&amp;rate=1.00813782158154E+000,branch_length=7.99899733140789E-001]:8.06409174452208E-001);</span><br><span class="line">End;</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>在代表样品的序号后用中括号[&amp;]提供注释信息，多个注释用逗号分隔，如果提供的注释信息值是区间，用大括号包含区间。</p>
<h3 id="2-2-2-ggtree画树"><a href="#2-2-2-ggtree画树" class="headerlink" title="2.2.2. ggtree画树"></a>2.2.2. ggtree画树</h3><p>ggtree函数是ggplot()的扩展，ggtree()相当于<code>ggplot() + geom_tree() + xlab(NA) + ylab(NA) + theme_tree()</code>的简单组合。ggplot2可添加的图层都可以直接应用于ggtree。</p>
<h4 id="2-2-2-1-基础画树"><a href="#2-2-2-1-基础画树" class="headerlink" title="2.2.2.1. 基础画树"></a>2.2.2.1. 基础画树</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>ggtree<span class="punctuation">)</span></span><br><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span> <span class="comment">#默认参数画树</span></span><br></pre></td></tr></table></figure>

<h4 id="2-2-2-2-基础注释"><a href="#2-2-2-2-基础注释" class="headerlink" title="2.2.2.2. 基础注释"></a>2.2.2.2. 基础注释</h4><ol>
<li><p>基础注释函数用法<br>包含注释函数的基本参数用法</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment">#画树 </span></span><br><span class="line"><span class="operator">+</span> geom_tiplab<span class="punctuation">(</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment"># 末端节点标上物种</span></span><br><span class="line"><span class="operator">+</span> geom_rootedge<span class="punctuation">(</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#添加根节点的枝长线，如果树文件中有根节点枝长则直接可使用</span></span><br><span class="line"><span class="operator">+</span> geom_point2<span class="punctuation">(</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment">#节点上标记圆点</span></span><br><span class="line"><span class="operator">+</span> geom_text2<span class="punctuation">(</span>aes<span class="punctuation">(</span>label<span class="operator">=</span>node<span class="punctuation">)</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment">#节点上做文字标记。label指定内容，这里指定节点上标记节点号，可以用这个确定树图上对应的节点号</span></span><br><span class="line"><span class="operator">+</span> geom_label2<span class="punctuation">(</span>aes<span class="punctuation">(</span>label<span class="operator">=</span>node<span class="punctuation">)</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#节点上做标签标记。label指定内容，这里指定节点上标记节点号。</span></span><br><span class="line"><span class="operator">+</span> geom_cladelabel<span class="punctuation">(</span><span class="number">20</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;legumes&quot;</span><span class="punctuation">,</span>angle<span class="operator">=</span><span class="number">90</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用条带和文本注释一个clade(node20)，label指定文本内容，angle指定文本角度</span></span><br><span class="line"><span class="operator">+</span> geom_strip<span class="punctuation">(</span><span class="number">9</span><span class="punctuation">,</span><span class="number">10</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;legumes&quot;</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用条带指定node9到node10的位置，label指定文字内容</span></span><br><span class="line"><span class="operator">+</span> geom_hilight<span class="punctuation">(</span>node<span class="operator">=</span><span class="number">20</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用矩形高亮node20号分支，用法和geom_highlight()好像是一样的</span></span><br><span class="line"><span class="operator">+</span> geom_range<span class="punctuation">(</span><span class="string">&quot;length_0.95_HPD&quot;</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#在节点上画长条形的分歧时间的95%置信范围，值来自树文件中的length_0.95_HPD注释值</span></span><br><span class="line"><span class="operator">+</span> geom_taxallink<span class="punctuation">(</span>taxa1<span class="operator">=</span><span class="number">12</span><span class="punctuation">,</span> taxa2<span class="operator">=</span><span class="number">20</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用曲线连接node12和node20</span></span><br><span class="line"><span class="operator">+</span> geom_treescale<span class="punctuation">(</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#添加比例尺</span></span><br><span class="line"><span class="operator">+</span> xlim<span class="punctuation">(</span><span class="literal">NA</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#如果内容超出边界，就用xlim拓展显示x轴的范围</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>常用注释参数<br>包含注释函数的常用参数用法</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">,</span> layout<span class="operator">=</span><span class="string">&quot;circular&quot;</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">0.8</span><span class="punctuation">,</span> branch.length<span class="operator">=</span><span class="string">&#x27;none&#x27;</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment">#画树，layout指定布局类型，默认是rectangular；size指定枝的粗细；branch.length指定无枝长信息  </span></span><br><span class="line"><span class="operator">+</span> geom_tiplab<span class="punctuation">(</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span> size <span class="operator">=</span> <span class="number">4</span><span class="punctuation">,</span> hjust <span class="operator">=</span> <span class="operator">-</span><span class="number">0.1</span><span class="punctuation">,</span> align<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">,</span> linesize<span class="operator">=</span><span class="number">0.7</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment"># 末端节点标上物种，size文字大小，hjust是离进化枝线条的距离，align=T文字间左对齐，linesize设置虚线尺寸。</span></span><br><span class="line"><span class="operator">+</span> geom_rootedge<span class="punctuation">(</span>rootedge <span class="operator">=</span> <span class="number">1</span><span class="punctuation">,</span>size<span class="operator">=</span><span class="number">1</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#添加根节点的枝长线，如果树文件中有根节点枝长则直接可使用，否则可以用rootedge设置根节点枝长(不设置为0)，size设置粗细</span></span><br><span class="line"><span class="operator">+</span> geom_point2<span class="punctuation">(</span>color<span class="operator">=</span><span class="string">&quot;#6FE1F8&quot;</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">5</span><span class="punctuation">,</span> alpha<span class="operator">=</span><span class="number">0.7</span><span class="punctuation">)</span> <span class="punctuation">\</span> <span class="comment">#所有节点上标记圆点，配色和透明度</span></span><br><span class="line"><span class="operator">+</span> geom_point2<span class="punctuation">(</span>aes<span class="punctuation">(</span>subset<span class="operator">=</span>node<span class="operator">==</span><span class="number">16</span><span class="punctuation">)</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&#x27;darkgreen&#x27;</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">5</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#在16号节点标记绿色圆点。其中subset是取子集，符合条件的才注释；color设置颜色，size设置大小。</span></span><br><span class="line"><span class="operator">+</span> geom_text2<span class="punctuation">(</span>aes<span class="punctuation">(</span>label<span class="operator">=</span>support<span class="punctuation">,</span>hjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.1</span><span class="punctuation">)</span><span class="punctuation">,</span>size<span class="operator">=</span><span class="number">3</span><span class="punctuation">,</span> fontface <span class="operator">=</span> <span class="string">&quot;bold&quot;</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#节点上标记支持率；size设置大小，fontface设置文本加粗显示</span></span><br><span class="line"><span class="operator">+</span> geom_text2<span class="punctuation">(</span>aes<span class="punctuation">(</span>subset<span class="operator">=</span><span class="operator">!</span>isTip<span class="punctuation">,</span> label<span class="operator">=</span>node<span class="punctuation">)</span><span class="punctuation">,</span> hjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.3</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">3</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&quot;deepskyblue4&quot;</span><span class="punctuation">)</span>  <span class="punctuation">\</span> <span class="comment">#非tip节点标记上node号</span></span><br><span class="line"><span class="operator">+</span> geom_label2<span class="punctuation">(</span>aes<span class="punctuation">(</span>x<span class="operator">=</span><span class="number">0.79</span><span class="punctuation">,</span>label<span class="operator">=</span>D<span class="punctuation">)</span><span class="punctuation">,</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span>vjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.2</span><span class="punctuation">,</span>fill<span class="operator">=</span><span class="string">&quot;#7FFF00&quot;</span><span class="punctuation">,</span>alpha<span class="operator">=</span><span class="number">0.6</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#做标记,aes里的x指定位置，label指定内容</span></span><br><span class="line"><span class="operator">+</span> geom_strip<span class="punctuation">(</span><span class="number">9</span><span class="punctuation">,</span><span class="number">10</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;legumes&quot;</span><span class="punctuation">,</span>offset <span class="operator">=</span> <span class="number">0.06</span><span class="punctuation">,</span>offset.text <span class="operator">=</span> <span class="number">0.012</span><span class="punctuation">,</span>color <span class="operator">=</span> <span class="string">&quot;#FF6100&quot;</span><span class="punctuation">,</span>barsize <span class="operator">=</span> <span class="number">1.5</span><span class="punctuation">,</span>fontsize <span class="operator">=</span> <span class="number">5</span><span class="punctuation">,</span>angle <span class="operator">=</span> <span class="number">90</span><span class="punctuation">,</span>hjust <span class="operator">=</span> <span class="number">0.5</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用条带指定node9和node10之间的分支clade，label指定文字内容，offset指定条带和文字与末端节点的距离，offset.text指定文字与条带的距离，barsize指定条带尺寸，fontsize指定文字尺寸，hjust指定文字位置</span></span><br><span class="line"><span class="operator">+</span> geom_hilight<span class="punctuation">(</span>node<span class="operator">=</span><span class="number">15</span><span class="punctuation">,</span>fill<span class="operator">=</span><span class="string">&quot;firebrick&quot;</span><span class="punctuation">,</span>alpha<span class="operator">=</span><span class="number">0.6</span><span class="punctuation">)</span> <span class="punctuation">\</span><span class="comment">#用矩形高亮一个分支</span></span><br><span class="line"><span class="operator">+</span> geom_range<span class="punctuation">(</span><span class="string">&quot;length_0.95_HPD&quot;</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&#x27;red&#x27;</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">2</span><span class="punctuation">,</span> alpha<span class="operator">=</span><span class="number">.5</span><span class="punctuation">,</span> center<span class="operator">=</span><span class="string">&#x27;length&#x27;</span><span class="punctuation">)</span> <span class="comment">#在节点上画长条形的分歧时间的95%置信范围，值来自树文件中的length_0.95_HPD注释值，让中心点位于length注释值</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>其他设置</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span></span><br><span class="line"><span class="operator">+</span> scale_x_reverse<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#反转x轴</span></span><br><span class="line"><span class="operator">+</span> coord_flip<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#置换x轴和y轴</span></span><br></pre></td></tr></table></figure></li>
</ol>
<h4 id="2-2-2-3-改变树的结构"><a href="#2-2-2-3-改变树的结构" class="headerlink" title="2.2.2.3. 改变树的结构"></a>2.2.2.3. 改变树的结构</h4><p>改变树的结构包括取子树(viewClade)，坍塌(collapse)，扩展(expand)，旋转(rotate)，分组(groupClade)等操作，下面两种函数用法是等价的：</p>
<ul>
<li>rotate(p,node&#x3D;12)</li>
<li>p %&gt;% rotate(node&#x3D;12)</li>
</ul>
<ol>
<li><p>显示子树</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">p <span class="operator">&lt;-</span> ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span><span class="operator">+</span>geom_tiplab<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line">viewClade<span class="punctuation">(</span>p<span class="punctuation">,</span> MRCA<span class="punctuation">(</span>p<span class="punctuation">,</span> <span class="number">10</span>，<span class="number">15</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#显示10-15节点的子树部分</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>旋转枝</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">flip<span class="punctuation">(</span>p<span class="punctuation">,</span> <span class="number">15</span><span class="punctuation">,</span> <span class="number">17</span><span class="punctuation">)</span> <span class="comment">#交换15号节点和17号节点代表的枝的位置，15和17号共享同一个祖先节点16。</span></span><br><span class="line">rotate<span class="punctuation">(</span>p<span class="punctuation">,</span> <span class="number">16</span><span class="punctuation">)</span> <span class="comment">#在16号节点把上下两个枝镜像翻转180°，与flip的区别是rotate同时改变了15和17号节点内所有子节点的顺序</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>坍塌和扩展枝</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">p2 <span class="operator">&lt;-</span> collapse<span class="punctuation">(</span>p<span class="punctuation">,</span> node<span class="operator">=</span><span class="number">16</span><span class="punctuation">)</span> <span class="operator">+</span> geom_point2<span class="punctuation">(</span>aes<span class="punctuation">(</span>subset<span class="operator">=</span><span class="punctuation">(</span>node<span class="operator">==</span><span class="number">16</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> shape<span class="operator">=</span><span class="number">21</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">5</span><span class="punctuation">,</span> fill<span class="operator">=</span><span class="string">&#x27;green&#x27;</span><span class="punctuation">)</span> <span class="comment">#坍塌21号节点，用绿色圆点标记</span></span><br><span class="line">expand<span class="punctuation">(</span>p2<span class="punctuation">,</span> node<span class="operator">=</span><span class="number">21</span><span class="punctuation">)</span> <span class="comment">#把p2坍塌的21号节点扩展开</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>缩小和放大枝</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">scaleClade<span class="punctuation">(</span>p<span class="punctuation">,</span> node<span class="operator">=</span><span class="number">15</span><span class="punctuation">,</span> scale<span class="operator">=</span><span class="number">0.1</span><span class="punctuation">)</span> <span class="comment">#缩小15号节点及子节点的枝，压缩尺度为0.1，拓扑和枝长仍然保留，但枝间距缩小为原来的0.1倍。常用于tip比较多的系统树上缩小不重要的部分，但不坍塌。</span></span><br><span class="line">scaleClade<span class="punctuation">(</span>p<span class="punctuation">,</span> node<span class="operator">=</span><span class="number">17</span><span class="punctuation">,</span> scale<span class="operator">=</span><span class="number">1.5</span><span class="punctuation">)</span> <span class="comment">#放大17号节点及子节点的枝，放大尺度为1.5，枝间距放大为原来的1.5倍。常用于突出重要的枝。</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>分组<br>单系分组用groupClade，多系&#x2F;并系分组用groupOTU。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">p3 <span class="operator">&lt;-</span> groupClade<span class="punctuation">(</span>p<span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">15</span><span class="punctuation">,</span><span class="number">17</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#节点15及子节点分为一组，节点17及子节点分为一组，赋予group属性，适用单系分组。</span></span><br><span class="line">ggtree<span class="punctuation">(</span>p3<span class="punctuation">,</span> aes<span class="punctuation">(</span>color<span class="operator">=</span>group<span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span> scale_color_manual<span class="punctuation">(</span>values<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span> <span class="string">&quot;red&quot;</span><span class="punctuation">,</span> <span class="string">&quot;blue&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#用group属性注释枝的颜色，默认颜色为黑色，15节点代表的组为红色，17节点代表的组为蓝色</span></span><br><span class="line"></span><br><span class="line">p4 <span class="operator">&lt;-</span> groupOTU<span class="punctuation">(</span>p<span class="punctuation">,</span> focus<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;A&quot;</span><span class="punctuation">,</span><span class="string">&quot;D&quot;</span><span class="punctuation">,</span><span class="string">&quot;G&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#A,D,G分为一组，赋予group属性</span></span><br><span class="line">ggtree<span class="punctuation">(</span>p4<span class="punctuation">,</span> aes<span class="punctuation">(</span>color<span class="operator">=</span>group<span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#用group属性注释枝的颜色，A,D,G所在的枝和共享的枝一种颜色，其他枝另一种颜色。A,D,G可以是多系/并系。</span></span><br></pre></td></tr></table></figure></li>
</ol>
<h4 id="2-2-2-4-可视化数据并与系统发育树关联-——-facet-plot"><a href="#2-2-2-4-可视化数据并与系统发育树关联-——-facet-plot" class="headerlink" title="2.2.2.4. 可视化数据并与系统发育树关联 —— facet_plot"></a>2.2.2.4. 可视化数据并与系统发育树关联 —— facet_plot</h4><p>可能有不同的数据类型，并希望将它们可视化并与树关联对齐，可用facet_plot函数来处理。</p>
<p>可视化图的类型：点图dotplot&#x2F;直方图barplot&#x2F;堆叠图stacked barplot&#x2F;热图&#x2F;箱线图boxplot。</p>
<p>ggtree中定义了操作符%&lt;+%，来添加数据。添加之后，用户的数据对ggplot是可见的。可以用于树的注释。</p>
<ol>
<li>可视化点图dotplot —— geom_point<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">tr <span class="operator">&lt;-</span> rtree<span class="punctuation">(</span><span class="number">30</span><span class="punctuation">)</span> <span class="comment"># 用treeio的rtree函数生成30个tip的随机树  </span></span><br><span class="line">p <span class="operator">&lt;-</span> ggtree<span class="punctuation">(</span>tr<span class="punctuation">)</span></span><br><span class="line">d1 <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>id<span class="operator">=</span>tr<span class="operator">$</span>tip.label<span class="punctuation">,</span> location<span class="operator">=</span>sample<span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;GZ&quot;</span><span class="punctuation">,</span> <span class="string">&quot;HK&quot;</span><span class="punctuation">,</span> <span class="string">&quot;CZ&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="number">30</span><span class="punctuation">,</span> replace<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#从(&quot;GZ&quot;, &quot;HK&quot;, &quot;CZ&quot;)中随机生成30个采样地点location数据  </span></span><br><span class="line">p1<span class="operator">&lt;-</span> p <span class="operator">%&lt;+%</span> d1 <span class="operator">+</span> geom_tippoint<span class="punctuation">(</span>aes<span class="punctuation">(</span>color<span class="operator">=</span>location<span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#给树的端点根据location着色  </span></span><br><span class="line"></span><br><span class="line">d2 <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>id<span class="operator">=</span>tr<span class="operator">$</span>tip.label<span class="punctuation">,</span> val<span class="operator">=</span>rnorm<span class="punctuation">(</span><span class="number">30</span><span class="punctuation">,</span> sd<span class="operator">=</span><span class="number">3</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#生成两列数据，分别是tip.label和随机生成的30个sd为3的数据值，均值mean默认为0  </span></span><br><span class="line">p2 <span class="operator">&lt;-</span> facet_plot<span class="punctuation">(</span>p1<span class="punctuation">,</span> panel<span class="operator">=</span><span class="string">&quot;dot&quot;</span><span class="punctuation">,</span> data<span class="operator">=</span>d2<span class="punctuation">,</span> geom<span class="operator">=</span>geom_point<span class="punctuation">,</span> aes<span class="punctuation">(</span>x<span class="operator">=</span>val<span class="punctuation">)</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&#x27;firebrick&#x27;</span><span class="punctuation">)</span> <span class="operator">+</span> theme_tree2<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#用ggtree的facet_plot的geom_point函数画点图  </span></span><br></pre></td></tr></table></figure></li>
</ol>
<img src="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_dot2.png" width=100% height=70% title="facet_plot_p2.png" align=center/>

<p><strong><p align="center">Figure 1. p2 dotplot</strong><br>from <a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_dot2.png">guangchuangyu’s blog</a></p></p>
<ol start="2">
<li>可视化堆叠直方图stacked barplot —— geom_barh<br>大多ggplot2的geom绘制垂直版本的图形对象，通过ggstance包可以制作水平版本的数据图geom，方便画直方图&#x2F;堆叠直方图&#x2F;热图&#x2F;箱线图与系统发育树对应。<br>包括geom_barh()，geom_histogramh()，geom_linerangeh()，geom_pointrangeh()，geom_errorbarh()，geom_crossbarh()，geom_boxploth()，geom_violinh()。<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>ggstance<span class="punctuation">)</span></span><br><span class="line">d3 <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>id <span class="operator">=</span> <span class="built_in">rep</span><span class="punctuation">(</span>tr<span class="operator">$</span>tip.label<span class="punctuation">,</span> each<span class="operator">=</span><span class="number">2</span><span class="punctuation">)</span><span class="punctuation">,</span>  </span><br><span class="line">					value <span class="operator">=</span> <span class="built_in">abs</span><span class="punctuation">(</span>rnorm<span class="punctuation">(</span><span class="number">60</span><span class="punctuation">,</span> mean<span class="operator">=</span><span class="number">100</span><span class="punctuation">,</span> sd<span class="operator">=</span><span class="number">50</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span>  </span><br><span class="line">					category <span class="operator">=</span> <span class="built_in">rep</span><span class="punctuation">(</span><span class="built_in">LETTERS</span><span class="punctuation">[</span><span class="number">1</span><span class="operator">:</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">,</span> <span class="number">30</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#随机生成3列数据，分别是tip.label,随机值,代表category的A/B  </span></span><br><span class="line">p3 <span class="operator">&lt;-</span> facet_plot<span class="punctuation">(</span>p2<span class="punctuation">,</span> panel <span class="operator">=</span> <span class="string">&#x27;Stacked Barplot&#x27;</span><span class="punctuation">,</span> data <span class="operator">=</span> d3<span class="punctuation">,</span>   </span><br><span class="line">				geom <span class="operator">=</span> geom_barh<span class="punctuation">,</span> <span class="comment">#用geom_barh画水平版本的堆叠直方图  </span></span><br><span class="line">				mapping <span class="operator">=</span> aes<span class="punctuation">(</span>x <span class="operator">=</span> value<span class="punctuation">,</span> fill <span class="operator">=</span> as.factor<span class="punctuation">(</span>category<span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span>   </span><br><span class="line">				stat<span class="operator">=</span><span class="string">&#x27;identity&#x27;</span> <span class="punctuation">)</span>   </span><br></pre></td></tr></table></figure></li>
</ol>
<img src="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_bar2.png" title="facet_plot_p3.png" width="100%" height="70%" />

<p><strong><p align="center">Figure 2. p3 barplot</strong><br>from <a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_bar2.png">guangchuangyu blog</a></p></p>
<ol start="3">
<li>可视化箱线图boxplot —— geom_boxploth<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">d4 <span class="operator">=</span> data.frame<span class="punctuation">(</span>id<span class="operator">=</span><span class="built_in">rep</span><span class="punctuation">(</span>tr<span class="operator">$</span>tip.label<span class="punctuation">,</span> each<span class="operator">=</span><span class="number">20</span><span class="punctuation">)</span><span class="punctuation">,</span>   </span><br><span class="line">				val<span class="operator">=</span>as.vector<span class="punctuation">(</span>sapply<span class="punctuation">(</span><span class="number">1</span><span class="operator">:</span><span class="number">30</span><span class="punctuation">,</span> <span class="keyword">function</span><span class="punctuation">(</span>i<span class="punctuation">)</span>   </span><br><span class="line">								rnorm<span class="punctuation">(</span><span class="number">20</span><span class="punctuation">,</span> mean<span class="operator">=</span>i<span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">)</span>  </span><br><span class="line">				<span class="punctuation">)</span>				</span><br><span class="line">p4 <span class="operator">&lt;-</span> facet_plot<span class="punctuation">(</span>p3<span class="punctuation">,</span> panel<span class="operator">=</span><span class="string">&quot;Boxplot&quot;</span><span class="punctuation">,</span> data<span class="operator">=</span>d4<span class="punctuation">,</span> geom_boxploth<span class="punctuation">,</span>   </span><br><span class="line">			mapping <span class="operator">=</span> aes<span class="punctuation">(</span>x<span class="operator">=</span>val<span class="punctuation">,</span> group<span class="operator">=</span>label<span class="punctuation">,</span> color<span class="operator">=</span>location<span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#用geom_boxploth画水平版本的箱线图</span></span><br></pre></td></tr></table></figure></li>
</ol>
<img src="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_boxplot2.png" title="facet_plot_p4.png" width="100%" height="70%" />

<p><strong><p align="center">Figure 3. p4 boxplot</strong><br>from <a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/facet_plot_boxplot2.png">guangchuangyu blog</a></p></p>
<h4 id="2-2-2-5-时序树-chronogram"><a href="#2-2-2-5-时序树-chronogram" class="headerlink" title="2.2.2.5. 时序树(chronogram)"></a>2.2.2.5. 时序树(chronogram)</h4><p>时序树上每个节点的数字代表的是分歧时间，端点一般对齐（因为端点的所有样本都对应当下的采样时间），画时序图需要先有时间标定的树文件，树上的枝长代表时长，一般会再添加时间尺度轴。</p>
<ol>
<li>画时序树</li>
</ol>
<ul>
<li>通过<code>ggtree(tree, mrsd=&quot;2020-01-01&quot;)</code>中的mrsd指定采样日期(即tip端点时间)，mrsd代表most recent sampling date,指定采样时间</li>
<li>通过<code>+theme_tree2()</code>来在树下添加一个时间尺度轴，默认是以年为单位</li>
<li>可用deeptime包向ggtree添加地质时间尺度</li>
</ul>
<p>实现采样时间(mrsd)和内部节点时间来缩放树</p>
<ol start="2">
<li>读取时序树<br>读取时序树与读取任何系统发育树的方法一样，读取任何时间代表枝长的树文件即可。</li>
</ol>
<ul>
<li>可以用treeio解析估计时间的软件(例如beast，r8s)输出的时序树；</li>
<li>也可以把时间写入枝长位置，以文本的形式读取时序树；</li>
</ul>
<ol start="3">
<li>时间尺度轴的变化</li>
</ol>
<ul>
<li><p>直接用<code>theme_tree2() +scale_x_continuous()</code>来设置时间尺度轴的显示</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span> <span class="operator">+</span> theme_tree2<span class="punctuation">(</span><span class="punctuation">)</span> <span class="operator">+</span>scale_x_continuous<span class="punctuation">(</span>limits<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">1</span><span class="punctuation">,</span><span class="number">4</span><span class="punctuation">)</span><span class="punctuation">,</span>breaks <span class="operator">=</span> seq<span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span><span class="number">4</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">)</span><span class="punctuation">,</span>labels <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">4</span><span class="punctuation">,</span><span class="number">3</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">,</span><span class="number">0</span><span class="punctuation">)</span><span class="punctuation">,</span>expand <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">0.1</span><span class="punctuation">,</span><span class="number">0.1</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span> geom_treescale<span class="punctuation">(</span>x<span class="operator">=</span><span class="number">3</span><span class="punctuation">,</span>y<span class="operator">=</span><span class="number">20</span><span class="punctuation">,</span>color<span class="operator">=</span><span class="string">&quot;red&quot;</span><span class="punctuation">)</span> <span class="comment">#画树，theme_tree2()在树下方加时间尺度轴，这里假设是[0,4]，用scale_x_continuous()限制时间尺度轴的显示和刻度，limits=c(-1,4)设置尺度轴显示范围为-1到4(同样也是图的横向显示范围)，breaks设置值的显示范围为0到4，每隔1显示一个值，labels设置显示的值，可以直接设定c(4,3,2,1,0)使时间尺度轴倒序显示，expand设置绘图左右范围的空白大小；geom_treescale()在右上角添加比例尺</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>用<code>revts</code>设置时间尺度轴翻转</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">p <span class="operator">&lt;-</span> ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span> <span class="operator">+</span> theme_tree2<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line">p1 <span class="operator">&lt;-</span> revts<span class="punctuation">(</span>p<span class="punctuation">)</span> <span class="comment">#翻转时间尺度轴，比如[0,4]翻转成[-4,0]；注意使用revts后，xlim的范围也要相应调整。</span></span><br><span class="line">library<span class="punctuation">(</span>ggplot2<span class="punctuation">)</span> <span class="comment">#scale_x_continuous是ggplot2的函数</span></span><br><span class="line">p2 <span class="operator">&lt;-</span> p <span class="operator">+</span> scale_x_continuous<span class="punctuation">(</span>labels <span class="operator">=</span> <span class="built_in">abs</span><span class="punctuation">)</span> <span class="comment">#把时间尺度轴的标签显示为绝对值abs的形式，scale_x_continuous是ggplot的函数</span></span><br><span class="line">p3 <span class="operator">&lt;-</span> revts<span class="punctuation">(</span>p2<span class="punctuation">)</span> <span class="comment">#把p2时间尺度轴翻转，由于p2用了绝对值，效果是把[0,4]翻转成了[4,0]</span></span><br><span class="line">p4 <span class="operator">&lt;-</span> p1 <span class="operator">+</span> scale_x_continuous<span class="punctuation">(</span>breaks<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">4</span><span class="operator">:</span><span class="number">0</span><span class="punctuation">)</span><span class="punctuation">,</span> labels<span class="operator">=</span><span class="built_in">abs</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">4</span><span class="operator">:</span><span class="number">0</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#把时间尺度轴的标签显示为绝对值abs的形式，即把[-4,0]显示成[4,0]</span></span><br></pre></td></tr></table></figure></li>
</ul>
<h4 id="2-2-2-6-系统发育网-phylogenetic-reticulum-——-geom-taxalink"><a href="#2-2-2-6-系统发育网-phylogenetic-reticulum-——-geom-taxalink" class="headerlink" title="2.2.2.6. 系统发育网(phylogenetic reticulum) —— geom_taxalink"></a>2.2.2.6. 系统发育网(phylogenetic reticulum) —— geom_taxalink</h4><p>系统发育网一般是在系统发育树绘制的基础上添加物种间的杂交和基因流关系。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ggtree<span class="punctuation">(</span>tree<span class="punctuation">)</span> <span class="operator">+</span> geom_tiplab<span class="punctuation">(</span><span class="punctuation">)</span>  </span><br><span class="line"><span class="operator">+</span> geom_taxalink<span class="punctuation">(</span><span class="string">&#x27;A&#x27;</span><span class="punctuation">,</span> <span class="string">&#x27;E&#x27;</span><span class="punctuation">)</span> <span class="comment">#在tipA和E之间添加关联线，默认是黑色实线。  </span></span><br><span class="line"><span class="operator">+</span> geom_taxalink<span class="punctuation">(</span><span class="string">&#x27;F&#x27;</span><span class="punctuation">,</span> <span class="string">&#x27;K&#x27;</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&#x27;red&#x27;</span><span class="punctuation">,</span> linetype <span class="operator">=</span> <span class="string">&#x27;dashed&#x27;</span><span class="punctuation">,</span> arrow<span class="operator">=</span>grid<span class="operator">::</span>arrow<span class="punctuation">(</span><span class="built_in">length</span><span class="operator">=</span>grid<span class="operator">::</span>unit<span class="punctuation">(</span><span class="number">0.02</span><span class="punctuation">,</span> <span class="string">&quot;npc&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#在tipF和K之间添加关联线，红色虚线，并添加F到K的箭头，箭头大小为0.02npc。  </span></span><br></pre></td></tr></table></figure>

<h4 id="2-2-2-7-多棵图共同展示"><a href="#2-2-2-7-多棵图共同展示" class="headerlink" title="2.2.2.7. 多棵图共同展示"></a>2.2.2.7. 多棵图共同展示</h4><ol>
<li><p>multiPhylo图<br>多棵树并列展示。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">trees <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="number">10</span><span class="punctuation">,</span> <span class="number">20</span><span class="punctuation">,</span> <span class="number">40</span><span class="punctuation">)</span><span class="punctuation">,</span> rtree<span class="punctuation">)</span> <span class="comment">#随机生成3棵树，节点分别为10，20，40  </span></span><br><span class="line"><span class="built_in">class</span><span class="punctuation">(</span>trees<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="string">&quot;multiPhylo&quot;</span>  </span><br><span class="line">ggtree<span class="punctuation">(</span>trees<span class="punctuation">)</span> <span class="operator">+</span> facet_wrap<span class="punctuation">(</span><span class="operator">~</span>.id<span class="punctuation">,</span> scale<span class="operator">=</span><span class="string">&quot;free&quot;</span><span class="punctuation">)</span> <span class="operator">+</span> geom_tiplab<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#分面展示3棵树  </span></span><br><span class="line"></span><br><span class="line">btrees <span class="operator">=</span> read.tree<span class="punctuation">(</span>system.file<span class="punctuation">(</span><span class="string">&quot;extdata/RAxML&quot;</span><span class="punctuation">,</span> <span class="string">&quot;RAxML_bootstrap.H3&quot;</span><span class="punctuation">,</span> package<span class="operator">=</span><span class="string">&quot;ggtree&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#读取系统的100棵树  </span></span><br><span class="line">ggtree<span class="punctuation">(</span>btrees<span class="punctuation">)</span> <span class="operator">+</span> facet_wrap<span class="punctuation">(</span><span class="operator">~</span>.id<span class="punctuation">,</span> ncol<span class="operator">=</span><span class="number">10</span><span class="punctuation">)</span> <span class="comment">#分面展示100棵树，每行10棵。  </span></span><br></pre></td></tr></table></figure>
</li>
<li><p>DensityTree图<br>多棵树重叠展示。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">trees <span class="operator">&lt;-</span> read.tree<span class="punctuation">(</span>system.file<span class="punctuation">(</span><span class="string">&quot;extdata/RAxML&quot;</span><span class="punctuation">,</span> <span class="string">&quot;RAxML_bootstrap.H3&quot;</span><span class="punctuation">,</span> package<span class="operator">=</span><span class="string">&quot;treeio&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment">#读取系统自带树文件RAxML_bootstrap.H3，包含100棵树。  </span></span><br><span class="line">ggdensitree<span class="punctuation">(</span>trees<span class="punctuation">,</span> alpha<span class="operator">=</span><span class="number">.3</span><span class="punctuation">,</span> colour<span class="operator">=</span><span class="string">&#x27;steelblue&#x27;</span><span class="punctuation">)</span> <span class="operator">+</span> geom_tiplab<span class="punctuation">(</span>size<span class="operator">=</span><span class="number">3</span><span class="punctuation">)</span> <span class="operator">+</span> xlim<span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span> <span class="number">45</span><span class="punctuation">)</span> <span class="comment">#重叠多棵树。bootstrap值高的线条一致性，低的线条较乱。  </span></span><br></pre></td></tr></table></figure></li>
</ol>
<h4 id="2-2-2-8-一个生成树的例子"><a href="#2-2-2-8-一个生成树的例子" class="headerlink" title="2.2.2.8. 一个生成树的例子"></a>2.2.2.8. 一个生成树的例子</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>treeio<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ggtree<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">treetext<span class="operator">&lt;-</span><span class="string">&quot;((((Mc:0.418672[&amp;&amp;NHX:I=+5728:D=-1170:G=Malvids],Eg:0.201388[&amp;&amp;NHX:I=+1420:D=-2519:G=Malvids])[&amp;&amp;NHX:C=1]:0.148853,((Pt:0.219587[&amp;&amp;NHX:I=+5355:D=-1003:G=Fabids],(Cs:0.209405[&amp;&amp;NHX:I=+1026:D=-3516:G=Malvids],(Gr:0.224188[&amp;&amp;NHX:I=+4875:D=-998:G=Malvids],At:0.492461[&amp;&amp;NHX:I=+1776:D=-2891:G=Malvids]):0.027774[&amp;&amp;NHX:C=2]):0.012564[&amp;&amp;NHX:C=3]):0.019343[&amp;&amp;NHX:C=4],(Pp:0.194515[&amp;&amp;NHX:I=+1159:D=-2949:G=Fabids],(Mt:0.354580[&amp;&amp;NHX:I=+3310:D=-1984:G=Fabids],Cl:0.338644[&amp;&amp;NHX:I=+794:D=-3735:G=Fabids]):0.022063[&amp;&amp;NHX:C=5]):0.019111[&amp;&amp;NHX:C=6]):0.011710[&amp;&amp;NHX:C=7]):0.029914[&amp;&amp;NHX:C=8],Vv:0.169100[&amp;&amp;NHX:I=+1522:D=-3045:G=Vitales]):0.241611[&amp;&amp;NHX:C=9],Mg:0.241611[&amp;&amp;NHX:I=+2278:D=-1909:G=Asterids])[&amp;&amp;NHX:C=10];&quot;</span>  </span><br><span class="line">tree <span class="operator">&lt;-</span> read.nhx<span class="punctuation">(</span>textConnection<span class="punctuation">(</span>treetext<span class="punctuation">)</span><span class="punctuation">)</span>  </span><br><span class="line"></span><br><span class="line">p1 <span class="operator">&lt;-</span> ggtree<span class="punctuation">(</span>tree<span class="punctuation">,</span>size<span class="operator">=</span><span class="number">0.8</span><span class="punctuation">)</span> <span class="comment">#画树，size指定线条粗细  </span></span><br><span class="line"><span class="operator">+</span> geom_tiplab<span class="punctuation">(</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span>size <span class="operator">=</span> <span class="number">4.5</span><span class="punctuation">,</span>hjust <span class="operator">=</span> <span class="operator">-</span><span class="number">0.1</span><span class="punctuation">,</span> align<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span> linesize<span class="operator">=</span><span class="number">0.7</span><span class="punctuation">)</span> <span class="comment">#添加tiplab-物种名  </span></span><br><span class="line"><span class="operator">+</span> geom_nodepoint<span class="punctuation">(</span>color<span class="operator">=</span><span class="string">&quot;#DCDCDC&quot;</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">6</span><span class="punctuation">,</span> alpha<span class="operator">=</span><span class="number">0.9</span><span class="punctuation">)</span> <span class="comment">#节点处添加圆点，指定颜色灰色和尺寸6  </span></span><br><span class="line"><span class="operator">+</span> geom_nodelab<span class="punctuation">(</span>aes<span class="punctuation">(</span>label<span class="operator">=</span>C<span class="punctuation">)</span><span class="punctuation">,</span> size<span class="operator">=</span><span class="number">3.5</span><span class="punctuation">,</span> hjust <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span> <span class="comment">#在注释了标签C的节点处添加C的值，这里的标签C用于注释节点序号  </span></span><br><span class="line"><span class="operator">+</span> geom_label2<span class="punctuation">(</span>aes<span class="punctuation">(</span>x<span class="operator">=</span><span class="number">0.79</span><span class="punctuation">,</span>label<span class="operator">=</span>D<span class="punctuation">)</span><span class="punctuation">,</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span>vjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.2</span><span class="punctuation">,</span>fill<span class="operator">=</span><span class="string">&quot;lightgreen&quot;</span><span class="punctuation">,</span>alpha<span class="operator">=</span><span class="number">0.6</span><span class="punctuation">)</span> <span class="comment"># 在注释了D的位置添加D的值，这里的标签D用于注释收缩基因的数量，x的值用于指定添加位置。  </span></span><br><span class="line"><span class="operator">+</span> geom_label2<span class="punctuation">(</span>aes<span class="punctuation">(</span>x<span class="operator">=</span><span class="number">0.73</span><span class="punctuation">,</span>label<span class="operator">=</span>I<span class="punctuation">)</span><span class="punctuation">,</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">,</span>vjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.2</span><span class="punctuation">,</span>fill<span class="operator">=</span><span class="string">&quot;pink&quot;</span><span class="punctuation">,</span>alpha<span class="operator">=</span><span class="number">0.6</span><span class="punctuation">)</span> <span class="comment"># 在注释了I的位置添加I的值，这里的标签I用于注释扩张基因的数量，x的值用于指定添加位置。  </span></span><br><span class="line"><span class="operator">+</span> xlim<span class="punctuation">(</span><span class="literal">NA</span><span class="punctuation">,</span><span class="number">1.1</span><span class="punctuation">)</span> <span class="comment"># 如果物种名或者其他标签超出显示范围，用xlim增大x轴显示。  </span></span><br><span class="line"></span><br><span class="line">p2<span class="operator">&lt;-</span> ggtree<span class="operator">::</span>rotate<span class="punctuation">(</span>p1<span class="punctuation">,</span><span class="number">13</span><span class="punctuation">)</span> <span class="comment">#旋转节点13的上下分支clade  </span></span><br><span class="line">p2<span class="operator">&lt;-</span> p1 <span class="operator">%&gt;%</span> rotate<span class="punctuation">(</span>node<span class="operator">=</span><span class="number">14</span><span class="punctuation">)</span> <span class="operator">%&gt;%</span> rotate<span class="punctuation">(</span>node<span class="operator">=</span><span class="number">16</span><span class="punctuation">)</span> <span class="comment"># 旋转节点14和节点16  </span></span><br><span class="line"></span><br><span class="line">p2<span class="operator">+</span>geom_strip<span class="punctuation">(</span><span class="number">10</span><span class="punctuation">,</span><span class="number">10</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;Vitales&quot;</span><span class="punctuation">,</span>offset <span class="operator">=</span> <span class="number">0.14</span><span class="punctuation">,</span> offset.text <span class="operator">=</span><span class="number">0.01</span><span class="punctuation">,</span>color <span class="operator">=</span> <span class="string">&quot;grey31&quot;</span><span class="punctuation">,</span>barsize <span class="operator">=</span> <span class="number">1.5</span><span class="punctuation">,</span>fontsize <span class="operator">=</span> <span class="number">5</span><span class="punctuation">)</span><span class="operator">+</span>geom_strip<span class="punctuation">(</span><span class="number">3</span><span class="punctuation">,</span><span class="number">7</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;Fabids&quot;</span><span class="punctuation">,</span>offset <span class="operator">=</span> <span class="number">0.14</span><span class="punctuation">,</span> offset.text <span class="operator">=</span><span class="number">0.01</span><span class="punctuation">,</span>color <span class="operator">=</span> <span class="string">&quot;grey31&quot;</span><span class="punctuation">,</span>barsize <span class="operator">=</span> <span class="number">1.5</span><span class="punctuation">,</span>fontsize <span class="operator">=</span> <span class="number">5</span><span class="punctuation">)</span><span class="operator">+</span>geom_strip<span class="punctuation">(</span><span class="number">1</span><span class="punctuation">,</span><span class="number">4</span><span class="punctuation">,</span>label<span class="operator">=</span><span class="string">&quot;Malvids&quot;</span><span class="punctuation">,</span>offset <span class="operator">=</span> <span class="number">0.14</span><span class="punctuation">,</span> offset.text <span class="operator">=</span><span class="number">0.01</span><span class="punctuation">,</span>color <span class="operator">=</span> <span class="string">&quot;grey31&quot;</span><span class="punctuation">,</span>barsize <span class="operator">=</span> <span class="number">1.5</span><span class="punctuation">,</span>fontsize <span class="operator">=</span> <span class="number">5</span><span class="punctuation">)</span> <span class="comment"># 给图加上分支clade的条带和文字注释</span></span><br></pre></td></tr></table></figure>

<h1 id="3-reference"><a href="#3-reference" class="headerlink" title="3. reference"></a>3. reference</h1><ol>
<li><a target="_blank" rel="noopener" href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12628">ggtree paper</a></li>
<li><a target="_blank" rel="noopener" href="https://yulab-smu.top/treedata-book/index.html">book_visualization of phylogenetic trees</a></li>
<li><a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html">ggtree tutorial</a></li>
<li><a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/ggtree-book/chapter-treeio.html">treeio tutorial</a></li>
<li><a target="_blank" rel="noopener" href="https://guangchuangyu.github.io/2016/10/facet_plot-a-general-solution-to-associate-data-with-phylogenetic-tree/">facet_plot</a></li>
<li><a target="_blank" rel="noopener" href="https://www.codenong.com/js9fda2b82cedb/">不同树的绘制</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2022/01/24/bioinfo_evolutionary.tree/" data-id="cl1aiqcrg0028m4vk141fdgqy" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-linux_command" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2022/01/16/linux_command/" class="article-date">
  <time datetime="2022-01-16T11:30:54.000Z" itemprop="datePublished">2022-01-16</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/linux/">linux</a>►<a class="article-category-link" href="/categories/linux/shell/">shell</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2022/01/16/linux_command/">shell的常用命令</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=2117115&auto=1&height=32"></iframe></div>

<h1 id="1-常用命令符号"><a href="#1-常用命令符号" class="headerlink" title="1. 常用命令符号"></a>1. 常用命令符号</h1><h2 id="1-1-常用命令连接符号-——-amp-amp"><a href="#1-1-常用命令连接符号-——-amp-amp" class="headerlink" title="1.1. 常用命令连接符号 —— ;,&amp;&amp;,||"></a>1.1. 常用命令连接符号 —— <code>;</code>,<code>&amp;&amp;</code>,<code>||</code></h2><p>多条命令放在一行时，需要在命令之间使用命令连接符，包括<code>;</code>,<code>&amp;&amp;</code>,<code>||</code>等。</p>
<h3 id="1-1-1"><a href="#1-1-1" class="headerlink" title="1.1.1. ;"></a>1.1.1. <code>;</code></h3><p><code>;</code>是最简单的命令连接符。</p>
<p><code>command1; command2; command3</code>表示多个命令顺序执行，命令间没有逻辑联系。</p>
<h3 id="1-1-2-amp-amp"><a href="#1-1-2-amp-amp" class="headerlink" title="1.1.2. &amp;&amp;"></a>1.1.2. <code>&amp;&amp;</code></h3><p><code>&amp;&amp;</code>逻辑与命令连接符。</p>
<p><code>command1 &amp;&amp; command2 &amp;&amp; command3</code>表示从左往右按顺序执行命令，如果中间有命令执行失败，则中断在失败的命令处不继续往下执行。即左边的命令执行成功才执行右边的命令；当左边命令执行失败就不会执行右边命令。</p>
<h3 id="1-1-3"><a href="#1-1-3" class="headerlink" title="1.1.3. ||"></a>1.1.3. <code>||</code></h3><p><code>||</code>逻辑或命令连接符。</p>
<p><code>command1 ||command2 || command3</code>表示从左往右按顺序执行命令，如果中间有命令执行成功，则中断在成功的命令处不继续往下执行。即左边的命令执行成功时，停止执行右边的命令；只有当左边的命令执行失败时，才执行右边的命令。</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2022/01/16/linux_command/" data-id="cl1aiqcru003om4vkb4mufl6z" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/" rel="tag">;</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/linux/" rel="tag">linux</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-R_install" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/12/28/R_install/" class="article-date">
  <time datetime="2021-12-28T08:30:00.000Z" itemprop="datePublished">2021-12-28</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer-language/">computer language</a>►<a class="article-category-link" href="/categories/computer-language/R/">R</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/12/28/R_install/">R及相关软件的安装</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=27853350&auto=1&height=32"></iframe></div>

<h1 id="1-R"><a href="#1-R" class="headerlink" title="1. R"></a>1. R</h1><p>R是常用于统计和画图的计算机语言。</p>
<h2 id="1-1-R包存储库"><a href="#1-1-R包存储库" class="headerlink" title="1.1. R包存储库"></a>1.1. R包存储库</h2><p>R 语言可以通过各种宏包来拓展功能，一般通过中央软件存储库下载R包，如R基金会资助的最常用的<a target="_blank" rel="noopener" href="https://cran.r-project.org/"><strong>The Comprehensive R Archive Network (CRAN)</strong></a>和存储R开发的生物数据分析包的生物专业软件存储库<a target="_blank" rel="noopener" href="https://www.bioconductor.org/"><strong>Bioconductor</strong></a>。</p>
<h2 id="1-2-R相关"><a href="#1-2-R相关" class="headerlink" title="1.2. R相关"></a>1.2. R相关</h2><h3 id="1-2-1-RStudio"><a href="#1-2-1-RStudio" class="headerlink" title="1.2.1. RStudio"></a>1.2.1. RStudio</h3><ul>
<li>RStudio 是 R 的集成开发环境 (IDE)。</li>
<li>它包括一个控制台、支持直接代码执行的语法高亮编辑器，以及用于绘图、历史、调试和工作区管理的工具。</li>
<li>有两种格式：RStudio Desktop 是一个常规的桌面应用程序，而 RStudio Server 运行在远程服务器上，并允许使用Web 浏览器访问 RStudio。</li>
</ul>
<h3 id="1-2-2-Rtools"><a href="#1-2-2-Rtools" class="headerlink" title="1.2.2. Rtools"></a>1.2.2. Rtools</h3><ul>
<li>Rtools是 Windows 上从源代码构建 R 包所需的一组程序。</li>
<li>Rtools为 Windows 平台提供了一个与 R 兼容的工具链，使得Windows平台R的使用与UNIX-ish平台R的使用许多命令一致。它主要包括 GNU make、GNU gcc 和其他在 UNIX-ish 平台上常用的实用程序。</li>
<li>如果只使用CRAN 或 Bioconductor 上提供的R包，则无需Rtools，如果使用需要编译的R包（比如ROracle）或者制作自己的R包，则需要Rtools。</li>
</ul>
<h1 id="2-R的安装和配置"><a href="#2-R的安装和配置" class="headerlink" title="2. R的安装和配置"></a>2. R的安装和配置</h1><h2 id="2-1-R-for-Windows"><a href="#2-1-R-for-Windows" class="headerlink" title="2.1. R for Windows"></a>2.1. R for Windows</h2><h3 id="2-1-1-安装R"><a href="#2-1-1-安装R" class="headerlink" title="2.1.1. 安装R"></a>2.1.1. 安装R</h3><p><a target="_blank" rel="noopener" href="https://www.r-project.org/">R官网</a>或者<a target="_blank" rel="noopener" href="https://mirrors.tuna.tsinghua.edu.cn/CRAN/bin/">R清华镜像</a>下载安装最新版本的R(现在最新版是R4.1.2)。<br>双击之后按步骤安装。</p>
<h3 id="2-1-2-安装RStudio-Desktop"><a href="#2-1-2-安装RStudio-Desktop" class="headerlink" title="2.1.2. 安装RStudio Desktop"></a>2.1.2. 安装RStudio Desktop</h3><p><a target="_blank" rel="noopener" href="https://www.rstudio.com/">RStudio官网</a>下载安装最新版本的Rstudio Desktop。<br>双击之后按步骤安装。</p>
<h3 id="2-1-3-安装Rtools"><a href="#2-1-3-安装Rtools" class="headerlink" title="2.1.3. 安装Rtools"></a>2.1.3. 安装Rtools</h3><p><a target="_blank" rel="noopener" href="https://cran.r-project.org/bin/windows/Rtools/rtools40.html">Rtools4</a>下载安装最新版本的Rtools。<br>双击之后按步骤安装。</p>
<h2 id="2-2-R-for-Linux"><a href="#2-2-R-for-Linux" class="headerlink" title="2.2. R for Linux"></a>2.2. R for Linux</h2><h3 id="2-2-1-安装R"><a href="#2-2-1-安装R" class="headerlink" title="2.2.1. 安装R"></a>2.2.1. 安装R</h3><h4 id="2-2-1-1-安装R"><a href="#2-2-1-1-安装R" class="headerlink" title="2.2.1.1. 安装R"></a>2.2.1.1. 安装R</h4><p><a target="_blank" rel="noopener" href="https://cran.r-project.org/">R 官网</a>里参考对应系统的R安装教程。</p>
<ol>
<li>Ubuntu<br><code>apt update</code>,<code>apt upgrade</code>,<code>apt install r-base</code></li>
<li>Centos<br><code>dnf install R</code></li>
<li>Debian<br><code>apt-get update</code>,<code>apt-get install r-base r-base-dev</code></li>
<li>conda安装<br><code>conda create -n r r-essentials r-base</code> #创建新环境r，并从CRAN安装所有的r-essentials包，R Essentials 包包含大约 200 个最流行的数据科学 R 包，包括 IRKernel、dplyr、shiny、ggplot2、tidyr、caret 和 nnet。</li>
</ol>
<p>安装R后<code>R --version</code>查看版本，可能非最新版。</p>
<h4 id="2-2-1-2-手动安装R"><a href="#2-2-1-2-手动安装R" class="headerlink" title="2.2.1.2. 手动安装R"></a>2.2.1.2. 手动安装R</h4><p><a target="_blank" rel="noopener" href="https://blog.51cto.com/kuxingseng2016/1846326">R source installation ref</a>;<a target="_blank" rel="noopener" href="https://xieduo7.github.io/2018-04-02-R%E5%AE%89%E8%A3%85.html">ref2</a>。</p>
<ol>
<li>R的依赖<br>参考R官方文档<a target="_blank" rel="noopener" href="https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Essential-and-useful-other-programs-under-a-Unix_002dalike">R installation and administration</a>，包括zlib,bzip,liblzma(XZ),PCRE等。</li>
<li>R下载<br><a target="_blank" rel="noopener" href="https://www.r-project.org/">R官网</a>或者<a target="_blank" rel="noopener" href="https://mirrors.tuna.tsinghua.edu.cn/CRAN/bin/">R清华镜像</a>下载安装最新版本的R(现在最新版是R4.1.2)。</li>
<li>R源码安装<br>R源码是C语言写的，需要编译安装。<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">tar zxvf R-4.1.2.tar.gz</span><br><span class="line">cd R-4.1.2</span><br><span class="line">./configure --enable-R-shlib #配置，生成Makefile</span><br><span class="line">make #编译</span><br><span class="line">make install #安装</span><br></pre></td></tr></table></figure></li>
</ol>
<h3 id="2-2-2-安装RStudio-Server"><a href="#2-2-2-安装RStudio-Server" class="headerlink" title="2.2.2. 安装RStudio Server"></a>2.2.2. 安装RStudio Server</h3><h4 id="2-2-2-1-安装RStudio-Server"><a href="#2-2-2-1-安装RStudio-Server" class="headerlink" title="2.2.2.1. 安装RStudio Server"></a>2.2.2.1. 安装RStudio Server</h4><p>RStudio Server 运行在远程服务器上，并允许使用Web 浏览器访问 RStudio。</p>
<p>在<a target="_blank" rel="noopener" href="https://www.rstudio.com/products/rstudio/download/">RStodio官网</a>上选择RStudio Server，选相应的系统版本下载。</p>
<ol>
<li>Ubuntu<br><code>sudo apt-get install r-base</code> # for Ubuntu 16</li>
<li>CentOS<br><code>wget https://download2.rstudio.org/server/centos8/x86_64/rstudio-server-rhel-2021.09.1-372-x86_64.rpm</code>,<br><code>sudo yum install rstudio-server-rhel-2021.09.1-372-x86_64.rpm</code> # for CentOS 8</li>
</ol>
<h4 id="2-2-2-2-卸载-x2F-更新RStudio-Server"><a href="#2-2-2-2-卸载-x2F-更新RStudio-Server" class="headerlink" title="2.2.2.2. 卸载&#x2F;更新RStudio Server"></a>2.2.2.2. 卸载&#x2F;更新RStudio Server</h4><p>更新也需要先卸载再安装新的版本。</p>
<ol>
<li><code>rstudio-server stop</code> #暂停当前RStudio Server服务</li>
<li><code>yum remove rstudio-server</code> #CentOS卸载，或者 <code>apt-get remove rstudio-server</code> #Ubuntu卸载</li>
</ol>
<h4 id="2-2-2-3-安装后配置RStudio-Server"><a href="#2-2-2-3-安装后配置RStudio-Server" class="headerlink" title="2.2.2.3. 安装后配置RStudio Server"></a>2.2.2.3. 安装后配置RStudio Server</h4><ol>
<li>安装后，通过修改&#x2F;etc&#x2F;rstudio&#x2F;rserver.conf更改端口和R所在路径</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">www-port=8787 # 端口, 默认8787</span><br><span class="line">www-address=0.0.0.0</span><br><span class="line">rsession-which-r=/opt/sysoft/R-4.1.2/bin/R # 安装R的路径</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>用<code>rstudio-server restart</code>重启服务<br>没有任何信息就表示安装成功了，还可以<code>rstudio-server verify-installation</code>验证，没有输出就是安装成功。</li>
</ol>
<h4 id="2-2-2-4-使用RStudio-Server"><a href="#2-2-2-4-使用RStudio-Server" class="headerlink" title="2.2.2.4. 使用RStudio Server"></a>2.2.2.4. 使用RStudio Server</h4><p><code>ps -aux|grep rstudio-server</code>查看服务是否运行，<code>ipconfig</code>查看服务器IP地址。</p>
<p>安装配置好RStudio Server后，直接在浏览器打开<a href="http://IP:8787，输入服务器的用户名和密码，即可访问RStudio">http://IP:8787，输入服务器的用户名和密码，即可访问RStudio</a> Server服务。</p>
<h1 id="3-R包安装"><a href="#3-R包安装" class="headerlink" title="3. R包安装"></a>3. R包安装</h1><p>R 语言可以通过各种宏包来拓展功能，一般通过中央软件存储库下载R包，如最常用的R基金会资助的<a target="_blank" rel="noopener" href="https://cran.r-project.org/"><strong>The Comprehensive R Archive Network (CRAN)</strong></a>，存储R开发的生物数据分析包的生物专业软件存储库<a target="_blank" rel="noopener" href="https://www.bioconductor.org/"><strong>Bioconductor</strong></a>，以及<a target="_blank" rel="noopener" href="https://anaconda.org/"><strong>anaconda</strong></a>。</p>
<h2 id="3-1-通过CRAN安装R包"><a href="#3-1-通过CRAN安装R包" class="headerlink" title="3.1. 通过CRAN安装R包"></a>3.1. 通过CRAN安装R包</h2><ol>
<li><code>install.packages(&quot;packagename&quot;)</code> R默认的安装方式，会从CRAN下载包并安装。</li>
<li>同时安装多个包: <code>install.packages(c(&quot;packagename1&quot;,&quot;packagename2&quot;))</code></li>
<li>使用R包下载地址安装: <code>install.packages(&quot;http://cran.r-project.org/src/contrib/Archive/ggplot2/ggplot2_0.9.1.tar.gz&quot;, repos=NULL, type=&quot;source&quot;)</code></li>
<li>下载包到本地后手动安装: <code>install.packages(&quot;ggplot2_0.9.1.tar.gz&quot;, repos = NULL)</code></li>
</ol>
<h2 id="3-2-通过Bioconductor后安装R包"><a href="#3-2-通过Bioconductor后安装R包" class="headerlink" title="3.2. 通过Bioconductor后安装R包"></a>3.2. 通过Bioconductor后安装R包</h2><ol>
<li>安装Bioconductor<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quietly <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">    install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span>version <span class="operator">=</span> <span class="string">&quot;3.13&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
BiocManager的版本与R版本一一对应，安装时如果版本不对会有提示，根据提示安装对应版本即可。</li>
<li>查看可用的Bioconductor包：<code>BiocManager::available()</code></li>
<li>安装Bioconductor库里有的R包：<code>BiocManager::install(&quot;packagename&quot;)</code></li>
<li>同时安装多个包：<code>BiocManager::install(c(&quot;packagename1&quot;,&quot;packagename2&quot;))</code></li>
</ol>
<h2 id="3-3-通过conda安装R包"><a href="#3-3-通过conda安装R包" class="headerlink" title="3.3. 通过conda安装R包"></a>3.3. 通过conda安装R包</h2><p>在<a target="_blank" rel="noopener" href="https://anaconda.org/">anaconda官网</a>查询想要安装的R包，如果库里有可以使用conda安装。</p>
<p>一般需要在包名称前添加r-，比如安装rbokeh和rjava。<br><code>conda install r-rbokeh</code>,<code>conda install r-rjava</code></p>
<h2 id="3-4-更新和卸载R包"><a href="#3-4-更新和卸载R包" class="headerlink" title="3.4. 更新和卸载R包"></a>3.4. 更新和卸载R包</h2><ol>
<li>更新：<code>update.packages()</code>；conda更新R包caret<code>conda update r-caret</code></li>
<li>卸载：<code>remove.packages()</code>。</li>
</ol>
<h1 id="4-R包载入"><a href="#4-R包载入" class="headerlink" title="4. R包载入"></a>4. R包载入</h1><ol>
<li>library(packagename)<br>如果包不存在，library会停止执行，不返回任何值。</li>
<li>require(packagename)<br>require会根据包存在与否返回true或false，并继续下面的语句执行。</li>
</ol>
<h1 id="5-R包使用查询"><a href="#5-R包使用查询" class="headerlink" title="5. R包使用查询"></a>5. R包使用查询</h1><p>R包使用方法: <code>help(package=&quot;packagename&quot;)</code></p>
<h1 id="6-reference"><a href="#6-reference" class="headerlink" title="6. reference"></a>6. reference</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/435/">https://www.bioinfo-scrounger.com/archives/435/</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/12/28/R_install/" data-id="cl1aiqcqz000ym4vk6clbg901" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_GSEA_clusterProfiler" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/12/13/bioinfo_GSEA_clusterProfiler/" class="article-date">
  <time datetime="2021-12-13T14:00:00.000Z" itemprop="datePublished">2021-12-13</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/12/13/bioinfo_GSEA_clusterProfiler/">基因富集分析(gene set enrichment analysis, GSEA) —— clusterProfiler</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1336867002&auto=1&height=32"></iframe></div>

<h1 id="1-基因富集分析-gene-set-enrichment-analysis-GSEA"><a href="#1-基因富集分析-gene-set-enrichment-analysis-GSEA" class="headerlink" title="1. 基因富集分析(gene set enrichment analysis, GSEA)"></a>1. 基因富集分析(gene set enrichment analysis, GSEA)</h1><p>基因富集分析(gene set enrichment analysis, GSEA)的介绍参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/12/bioinfo_GSEA/">博文基因富集分析</a>。</p>
<h1 id="2-clusterProfiler介绍"><a href="#2-clusterProfiler介绍" class="headerlink" title="2. clusterProfiler介绍"></a>2. clusterProfiler介绍</h1><p>clusterProfiler是一个R包，是一个解释组学数据的通用富集工具许多基因集的功能注释和富集分析，以及富集分析结果的可视化。</p>
<p>2021年07月发布了clusterProfiler 4.0版本。</p>
<img src="https://yulab-smu.top/biomedical-knowledge-mining-book/figures/clusterProfiler-diagram.png" width=100% title="clusterProfiler function and workflow" alt="clusterProfiler function and workflow" align=center/>

<p><strong><p align="center">Figure 1. clusterProfiler function and workflow</strong><br>from <a target="_blank" rel="noopener" href="https://yulab-smu.top/biomedical-knowledge-mining-book/">yulab’s book</a></p></p>
<h1 id="3-clusterProfiler支持的基因集-gene-sets-或基因通路数据库"><a href="#3-clusterProfiler支持的基因集-gene-sets-或基因通路数据库" class="headerlink" title="3. clusterProfiler支持的基因集(gene sets)或基因通路数据库"></a>3. clusterProfiler支持的基因集(gene sets)或基因通路数据库</h1><ol>
<li>Gene Ontology(GO)</li>
<li>Kyoto Encyclopedia of Genes and Genomes(KEGG)</li>
<li>Disease Ontology(DO)</li>
<li>Disease Gene Network(DisGeNET)</li>
<li>Molecular Signatures Database(MSigDb)</li>
<li>wikiPathways<br>… …</li>
</ol>
<h1 id="4-clusterProfiler功能-——-富集分析-enrichment-analysis"><a href="#4-clusterProfiler功能-——-富集分析-enrichment-analysis" class="headerlink" title="4. clusterProfiler功能 —— 富集分析(enrichment analysis)"></a>4. clusterProfiler功能 —— 富集分析(enrichment analysis)</h1><h2 id="4-1-过表达分析-Over-Representation-Analysis-ORA"><a href="#4-1-过表达分析-Over-Representation-Analysis-ORA" class="headerlink" title="4.1. 过表达分析(Over Representation Analysis, ORA)"></a>4.1. 过表达分析(Over Representation Analysis, ORA)</h2><p>过表达分析是用于采用超几何分布检验判断已知的生物功能或过程(例如GO&#x2F;KEGG)在实验产生的基因列表(例如差异表达基因列表: differentially expressed genes, DEGs)中是否过表达(over-represented&#x3D;enriched)的常用方法。</p>
<p>所有基因（或全基因组）作为背景总数N，被注释到已知基因集(例如GO&#x2F;KEGG)某一子集的基因总数是M，差异表达的基因数量是n，差异表达基因中属于M的数量是k。<br>过表达分析就是用统计学的方法判断GO&#x2F;KEGG的每一个子类中是否k&#x2F;n显著高于M&#x2F;N，显著高于就是这个子类过表达。</p>
<p>差异表达基因通常是前期通过比较转录组分析等手段获取的不同转录组间有表达差异的基因列表。</p>
<h2 id="4-2-基因集富集分析-Gene-Set-Enrichment-Analysis-GSEA"><a href="#4-2-基因集富集分析-Gene-Set-Enrichment-Analysis-GSEA" class="headerlink" title="4.2. 基因集富集分析(Gene Set Enrichment Analysis, GSEA)"></a>4.2. 基因集富集分析(Gene Set Enrichment Analysis, GSEA)</h2><p>对于差异很大的基因，可以使用ORA分析，但差异较小的基因也可能具有意义，很多相关表型差异是通过一组基因中微小但一致的变化(small but consistent changes)来表现的。</p>
<p>基因集富集分析汇总一个基因集中每个基因的统计数据，可以检测预先定义的基因集(例如GO&#x2F;KEGG)中所有基因以一种较小但协调的(small but coordinated)方式发生变化的情况。</p>
<p>基因根据表型进行排序，给定先验定义的基因集(例如共享相同GO&#x2F;KEGG类别的基因)，GSEA的目标是确定S的成员是随机分布在排序的基因列表L中，还是主要分布在顶部或者底部。</p>
<h2 id="4-3-leading-edge分析和核心富集基因-Leading-edge-analysis-and-core-enriched-genes"><a href="#4-3-leading-edge分析和核心富集基因-Leading-edge-analysis-and-core-enriched-genes" class="headerlink" title="4.3. leading edge分析和核心富集基因(Leading edge analysis and core enriched genes)"></a>4.3. leading edge分析和核心富集基因(Leading edge analysis and core enriched genes)</h2><p>包（clusterProfiler,DOSE,meshes,ReactomePA）支持Leading edge分析并可以提供GSEA分析中的核心富集基因core enriched genes结果。</p>
<ol>
<li>Leading edge分析可以获取：</li>
</ol>
<ul>
<li>Tags显示对富集分数有贡献的基因的百分比</li>
<li>List显示获得富集分数在列表中的位置</li>
<li>Signal显示富集信号强度</li>
</ul>
<h1 id="5-clusterProfiler安装"><a href="#5-clusterProfiler安装" class="headerlink" title="5. clusterProfiler安装"></a>5. clusterProfiler安装</h1><p>启动R后输入下面命令安装</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quiet <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">    install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;clusterProfiler&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h1 id="6-clusterProfiler富集分析"><a href="#6-clusterProfiler富集分析" class="headerlink" title="6. clusterProfiler富集分析"></a>6. clusterProfiler富集分析</h1><p>clusterProfiler支持对许多ontology&#x2F;pathway的hypergeometric test和gene set enrichment analyses，包括数据库GO,KEGG,DO,DisGeNET,MSigDb,wikiPathways。</p>
<p>使用支持的数据库时，需要检查分析样本是否在已有物种列表中。不在物种列表的物种数据，以及其他不在数据库列表的数据库，做分析时使用通用的富集分析(Universal enrichment analysis)模块。</p>
<p>分析前，先用命令<code>library(clusterProfiler)</code>载入clusterProfiler包。</p>
<h2 id="6-1-GO富集分析"><a href="#6-1-GO富集分析" class="headerlink" title="6.1. GO富集分析"></a>6.1. GO富集分析</h2><h3 id="6-1-1-支持的物种"><a href="#6-1-1-支持的物种" class="headerlink" title="6.1.1. 支持的物种"></a>6.1.1. 支持的物种</h3><p>GO富集分析需要物种的注释数据库，来源可以是Bioconductor，AnnotationHub或者用户提供GO注释文件。</p>
<p>groupGO(),enrichGO(),gseGO()函数支持具有OrgDb数据库的生物物种，enricher()和gseGO()函数支持用户自己提供的GO注释文件。</p>
<ol>
<li><p>clusterProfiler的默认GO注释信息来自Bioconductor，目前Bioconductor自带20个，植物只有拟南芥org.At.tair.db一个数据库。<a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">OrgDb数据库</a></p>
</li>
<li><p>还可以通过AnnotationHub包在线获取OrgDb对象，并制作Org.Db库。</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">source<span class="punctuation">(</span><span class="string">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="punctuation">)</span></span><br><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quietly <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">  install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;AnnotationHub&quot;</span><span class="punctuation">)</span> <span class="comment"># 一个包含大量注释信息的数据库，里面有很多物种及来源于很多数据库的注释信息。</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;biomaRt&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">library<span class="punctuation">(</span>AnnotationHub<span class="punctuation">)</span> </span><br><span class="line">library<span class="punctuation">(</span>biomaRt<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">hub <span class="operator">&lt;-</span> AnnotationHub<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#建立AnnotationHub对象（视人品，网不行加载不了）</span></span><br><span class="line"><span class="comment"># unique(hub$species) #查看AnonotationHub里面物种</span></span><br><span class="line">hub<span class="operator">$</span>species<span class="punctuation">[</span>which<span class="punctuation">(</span>hub<span class="operator">$</span>species<span class="operator">==</span><span class="string">&quot;Solanum&quot;</span><span class="punctuation">)</span><span class="punctuation">]</span> <span class="comment">#看AnonotationHub里是否包含想要的物种</span></span><br><span class="line"><span class="comment"># Solanum是番茄的拉丁名</span></span><br><span class="line">query<span class="punctuation">(</span>hub<span class="punctuation">,</span> <span class="string">&quot;Solanum&quot;</span><span class="punctuation">)</span>  <span class="comment">#查看该物种信息</span></span><br><span class="line">hub<span class="punctuation">[</span>hub<span class="operator">$</span>species<span class="operator">==</span><span class="string">&quot;Solanum&quot;</span> <span class="operator">&amp;</span> hub<span class="operator">$</span>rdataclass <span class="operator">==</span> <span class="string">&quot;OrgDb&quot;</span><span class="punctuation">]</span> <span class="comment">#OrgDb属于rdataclass中，因此查看下该物种有没有OrgDb</span></span><br><span class="line">Solanum.OrgDb <span class="operator">&lt;-</span> hub<span class="punctuation">[[</span><span class="string">&quot;AH59087&quot;</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="comment">#AH59087是番茄对应的编号</span></span><br><span class="line"><span class="comment">#制作为标准注释库，就可和模式生物一样使用了</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>用户提供GO注释数据<br>如果分析的物种没有OrgDb数据库，可以自己做GO注释后提供背景注释数据库，比如interproscan、eggnog-mapper和blas2go等软件获取的GO注释结果，整理成clusterProfiler支持的输入格式即可。</li>
</ol>
<p>clusterProfiler需要导入的GO注释文件go_annotation.txt的格式如下：</p>
<table>
<thead>
<tr>
<th>GeneID</th>
<th>GO</th>
<th>GO_Description</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>GO:0005819</td>
<td>spindle</td>
</tr>
<tr>
<td>2</td>
<td>GO:0072686</td>
<td>mitotic spindle</td>
</tr>
<tr>
<td>3</td>
<td>GO:0000776</td>
<td>kinetochore</td>
</tr>
</tbody></table>
<p>data.frame格式，包含三列，第一列为Gene ID，第二列为 GO ID，第三列为GO_Description，顺序无要求。</p>
<p>可以使用enricher()和gseGO()函数进行ORA和GSEA分析。</p>
<p>如果基因是通过直接注释来注释的，那么也应该用它们的祖先GO节点来注释（间接注释）。如果用户只有直接注释，他们可以将他们的注释传递给buildGOmap函数，该函数将推断间接注释并生成data.frame同时适用于enricher()和 的gseGO()。</p>
<h3 id="6-1-2-GO的ORA分析"><a href="#6-1-2-GO的ORA分析" class="headerlink" title="6.1.2. GO的ORA分析"></a>6.1.2. GO的ORA分析</h3><h4 id="6-1-2-1-输入文件"><a href="#6-1-2-1-输入文件" class="headerlink" title="6.1.2.1. 输入文件"></a>6.1.2.1. 输入文件</h4><p>ORA分析的输入文件是gene ID list，比如差异表达分析(DESeq2)获得的差异表达基因列表，保存为内容为一列数据的文本文件gene.list，数据内容可以是OrgDb支持的任意ID类型（常用的都支持，ENSEMBL，ENTREZID，GENETYPE，GO，PFAM），具体参考<a target="_blank" rel="noopener" href="http://yulab-smu.top/biomedical-knowledge-mining-book/useful-utilities.html#id-convert">ID</a>。<br>分析的是列表中基因在GO&#x2F;KEGG各个子分类单元是否被过度代表还是代表不足。</p>
<h4 id="6-1-2-2-bitr格式转换"><a href="#6-1-2-2-bitr格式转换" class="headerlink" title="6.1.2.2. bitr格式转换"></a>6.1.2.2. bitr格式转换</h4><p>如果需要，也可以用bitr功能函数实现各种ID格式的转换。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">)</span> <span class="comment">#单列基因名文件</span></span><br><span class="line">data<span class="operator">$</span>V1 <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#需要character格式，然后进行ID转化</span></span><br><span class="line">test <span class="operator">=</span> bitr<span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">,</span> fromType<span class="operator">=</span><span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">,</span> toType<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;ENSEMBL&quot;</span><span class="punctuation">,</span> <span class="string">&quot;ENTREZID&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> OrgDb<span class="operator">=</span> org.At.tair.db<span class="punctuation">)</span> <span class="comment">#将SYMBOL格式转为ENSEMBL和ENTERZID格式 </span></span><br><span class="line">head<span class="punctuation">(</span>test<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">    SYMBOL         ENSEMBL ENTREZID</span><br><span class="line"><span class="number">1</span>    AASDH ENSG00000157426   <span class="number">132949</span></span><br><span class="line"><span class="number">2</span>   ABCB11 ENSG00000073734     <span class="number">8647</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-2-3-GO分类"><a href="#6-1-2-3-GO分类" class="headerlink" title="6.1.2.3. GO分类"></a>6.1.2.3. GO分类</h4><p>groupGO()函数可以基于GO在指定水平范围内的分布进行基因分类。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br><span class="line">ggo <span class="operator">&lt;-</span> groupGO<span class="punctuation">(</span>gene <span class="operator">=</span> genes<span class="punctuation">,</span> OrgDb <span class="operator">=</span> org.Hs.eg.db<span class="punctuation">,</span> ont <span class="operator">=</span> <span class="string">&quot;CC&quot;</span><span class="punctuation">,</span> level <span class="operator">=</span> <span class="number">3</span><span class="punctuation">,</span> readable <span class="operator">=</span> TURE<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>ggo<span class="punctuation">,</span><span class="number">3</span><span class="punctuation">)</span></span><br><span class="line">                   ID       Description Count GeneRatio geneID</span><br><span class="line">GO<span class="operator">:</span><span class="number">0000003</span> GO<span class="operator">:</span><span class="number">0000003</span>      reproduction     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span>       </span><br><span class="line">GO<span class="operator">:</span><span class="number">0008152</span> GO<span class="operator">:</span><span class="number">0008152</span> metabolic process     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span>       </span><br><span class="line">GO<span class="operator">:</span><span class="number">0001906</span> GO<span class="operator">:</span><span class="number">0001906</span>      cell killing     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-2-4-ORA分析"><a href="#6-1-2-4-ORA分析" class="headerlink" title="6.1.2.4. ORA分析"></a>6.1.2.4. ORA分析</h4><ol>
<li>读取输入文件<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br></pre></td></tr></table></figure></li>
<li>支持物种的标准分析 enrichGO()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">ego <span class="operator">&lt;-</span> enrichGO<span class="punctuation">(</span>gene          <span class="operator">=</span> genes<span class="punctuation">,</span> <span class="comment"># list of entrez gene id</span></span><br><span class="line">                OrgDb         <span class="operator">=</span> org.At.tair.db<span class="punctuation">,</span> <span class="comment"># 背景使用分析物种的org包，这里示例使用拟南芥的数据库</span></span><br><span class="line">                keyType       <span class="operator">=</span> <span class="string">&#x27;ENSEMBL&#x27;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">                ont           <span class="operator">=</span> <span class="string">&quot;CC&quot;</span><span class="punctuation">,</span> <span class="comment"># &quot;BP&quot;, &quot;MF&quot;, &quot;CC&quot;, &quot;ALL&quot;。GO三个子类里选</span></span><br><span class="line">                pAdjustMethod <span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span> <span class="comment"># 指定多重假设检验矫正的方法，选项包含 &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</span></span><br><span class="line">                pvalueCutoff  <span class="operator">=</span> <span class="number">0.01</span><span class="punctuation">,</span> <span class="comment"># 富集分析的pvalue</span></span><br><span class="line">                qvalueCutoff  <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span> <span class="comment"># 富集分析显著性的qvalue</span></span><br><span class="line">                readable      <span class="operator">=</span> <span class="literal">TRUE</span> <span class="punctuation">)</span> <span class="comment"># 是否将gene ID转换到 gene symbol</span></span><br><span class="line">head<span class="punctuation">(</span>ego<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line">     ONTOLOGY         ID                                                Description</span><br><span class="line">GO<span class="operator">:</span><span class="number">0002887</span>       BP GO<span class="operator">:</span><span class="number">0002887</span> negative regulation of myeloid leukocyte mediated immunity</span><br><span class="line">GO<span class="operator">:</span><span class="number">0033004</span>       BP GO<span class="operator">:</span><span class="number">0033004</span>                negative regulation of mast cell activation</span><br><span class="line">           GeneRatio  BgRatio      pvalue  p.adjust    qvalue                   geneID Count</span><br><span class="line">GO<span class="operator">:</span><span class="number">0002887</span>     <span class="number">2</span><span class="operator">/</span><span class="number">121</span> <span class="number">10</span><span class="operator">/</span><span class="number">11461</span> <span class="number">0.004706555</span> <span class="number">0.7796682</span> <span class="number">0.7796682</span>              CD300A<span class="operator">/</span>CD84     <span class="number">2</span></span><br><span class="line">GO<span class="operator">:</span><span class="number">0033004</span>     <span class="number">2</span><span class="operator">/</span><span class="number">121</span> <span class="number">10</span><span class="operator">/</span><span class="number">11461</span> <span class="number">0.004706555</span> <span class="number">0.7796682</span> <span class="number">0.7796682</span>              CD300A<span class="operator">/</span>CD84     <span class="number">2</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li><p>提供注释文件go_annotation.txt的非模式物种分析 enricher()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">go_anno <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;go_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> go_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> go_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">ego <span class="operator">&lt;-</span> enricher<span class="punctuation">(</span>genes<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>ego结果文件解释</p>
</li>
</ol>
<ul>
<li>ONTOLOGY：CC  BP  MF </li>
<li>ID: Gene Ontology数据库中唯一的标号信息</li>
<li>Description ：Gene Ontology功能的描述信息</li>
<li>GeneRatio：差异基因中与该Term相关的基因数与整个差异基因总数的比值</li>
<li>BgRatio：所有（ bg）基因中与该Term相关的基因数与所有（ bg）基因的比值</li>
<li>pvalue: 富集分析统计学显著水平，一般情况下， P-value &lt; 0.05 该功能为富集项</li>
<li>p.adjust 矫正后的P-Value</li>
<li>qvalue：对p值进行统计学检验的q值</li>
<li>geneID：与该Term相关的基因</li>
<li>Count：与该Term相关的基因数</li>
</ul>
<ol start="5">
<li>输出结果保存为csv表格<br><code>write.csv(summary(ego),&quot;enrich.csv&quot;,row.names =FALSE)</code></li>
</ol>
<h4 id="6-1-2-5-结果整理和筛选"><a href="#6-1-2-5-结果整理和筛选" class="headerlink" title="6.1.2.5. 结果整理和筛选"></a>6.1.2.5. 结果整理和筛选</h4><ol>
<li>如果没使用keyType参数，可以在得到结果后使用</li>
<li>如果没使用readable&#x3D;True参数，可以在得到结果后用函数setReadable()将GeneID转换为symbol。<code>ego &lt;- setReadable(ego, OrgDb = org.At.tair.db)</code></li>
<li>函数dropGO可以移除enrichGO结果中特定的GO term或GO level</li>
<li>函数gofilter()可以将结果限定在特定的GO level</li>
<li>函数simplify可以去除冗余，<code>ego_rm &lt;- simplify(ego, cutoff=0.7, by=&quot;p.adjust&quot;, select_fun=min)</code></li>
</ol>
<h3 id="6-1-3-GO的GSEA分析"><a href="#6-1-3-GO的GSEA分析" class="headerlink" title="6.1.3. GO的GSEA分析"></a>6.1.3. GO的GSEA分析</h3><p>GSEA分析通过置换检验来计算p值</p>
<h4 id="6-1-3-1-GSEA的输入文件"><a href="#6-1-3-1-GSEA的输入文件" class="headerlink" title="6.1.3.1. GSEA的输入文件"></a>6.1.3.1. GSEA的输入文件</h4><ol>
<li>GSEA分析的输入文件是一个基因排序列表，有三个要点：</li>
</ol>
<ul>
<li>numeric vector：倍数变化或者其他类型的数字变量，比如差异表达分析里的logFC值</li>
<li>named vector：每个数字倍对应的gene ID命名</li>
<li>sorted vector；数字应该以降序排序<br>即包含两列，一列基因ID名称，一列数据，并以数据降序排序。</li>
</ul>
<ol start="2">
<li>获取输入文件的示例：</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">d <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span>your_csv_file<span class="punctuation">)</span></span><br><span class="line"><span class="comment">## assume that 1st column is ID (no duplicated allowed)</span></span><br><span class="line"><span class="comment">## 2nd column is fold change</span></span><br><span class="line"><span class="comment">## feature 1: numeric vector</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="comment">## feature 2: named vector</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">## feature 3: decreasing order</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>mydata<span class="punctuation">,</span><span class="number">3</span><span class="punctuation">)</span></span><br><span class="line">  gene_name     female     male    logFC</span><br><span class="line"><span class="number">1</span>   CG32548 <span class="number">0.02310383</span> <span class="number">72.43205</span> <span class="number">11.61428</span></span><br><span class="line"><span class="number">2</span>   CG15892 <span class="number">0.02624160</span> <span class="number">57.22716</span> <span class="number">11.09063</span></span><br><span class="line"><span class="number">3</span>   CR43803 <span class="number">0.02474626</span> <span class="number">34.09726</span> <span class="number">10.42823</span></span><br><span class="line"><span class="operator">&gt;</span> <span class="comment">#GSEA的基因排序列表</span></span><br><span class="line"><span class="operator">&gt;</span> genelist <span class="operator">&lt;-</span> mydata<span class="operator">$</span>logFC <span class="comment">#numeric vector</span></span><br><span class="line"><span class="operator">&gt;</span> <span class="built_in">names</span><span class="punctuation">(</span>genelist<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>mydata<span class="operator">$</span>gene_name<span class="punctuation">)</span> <span class="comment">#named vector</span></span><br><span class="line"><span class="operator">&gt;</span> genelist <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>genelist<span class="punctuation">,</span>decreasing<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">)</span> <span class="comment">#decreasing order</span></span><br><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>genelist<span class="punctuation">)</span></span><br><span class="line"> CG32548  CG15892  CR43803  CG15136   CG4983  CG13989 </span><br><span class="line"><span class="number">11.61428</span> <span class="number">11.09063</span> <span class="number">10.42823</span> <span class="number">10.34305</span> <span class="number">10.29130</span> <span class="number">10.00569</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-3-2-GSEA分析"><a href="#6-1-3-2-GSEA分析" class="headerlink" title="6.1.3.2. GSEA分析"></a>6.1.3.2. GSEA分析</h4><ol>
<li><p>读取输入文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span><span class="string">&quot;geneList.csv&quot;</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>支持物种的标准分析 gseGO()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">gsego <span class="operator">&lt;-</span> gseGO<span class="punctuation">(</span>geneList     <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">              OrgDb        <span class="operator">=</span> org.At.tair.db<span class="punctuation">,</span></span><br><span class="line">              ont          <span class="operator">=</span> <span class="string">&quot;MF&quot;</span><span class="punctuation">,</span> <span class="comment"># GO子类，包含&quot;BP&quot;,&quot;MF&quot;,&quot;CC&quot;,&quot;ALL&quot;</span></span><br><span class="line">              keyType      <span class="operator">=</span> <span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">,</span>  <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">              nPerm        <span class="operator">=</span> <span class="number">1000</span><span class="punctuation">,</span> <span class="comment"># permutation numbers 置换次数</span></span><br><span class="line">              minGSSize    <span class="operator">=</span> <span class="number">100</span><span class="punctuation">,</span> <span class="comment"># 富集基因数量的下限</span></span><br><span class="line">              maxGSSize    <span class="operator">=</span> <span class="number">500</span><span class="punctuation">,</span> <span class="comment"># 富集基因数量的上限</span></span><br><span class="line">              pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span></span><br><span class="line">              verbose      <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> <span class="comment"># 不输出结果</span></span><br><span class="line">              by           <span class="operator">=</span> <span class="string">&quot;fgsea&quot;</span><span class="punctuation">)</span> <span class="comment"># 选项fgsea或DOSE</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>gsego<span class="punctuation">,</span><span class="number">1</span><span class="punctuation">)</span>;<span class="built_in">dim</span><span class="punctuation">(</span>gsego<span class="punctuation">)</span></span><br><span class="line">                   ID        Description setSize enrichmentScore</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> GO<span class="operator">:</span><span class="number">0003674</span> molecular_function     <span class="number">323</span>       <span class="number">0.9462411</span></span><br><span class="line">                NES      pvalue    p.adjust qvalues rank</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> <span class="number">1.716837</span> <span class="number">0.000999001</span> <span class="number">0.000999001</span>      <span class="literal">NA</span>  <span class="number">579</span></span><br><span class="line">                              leading_edge</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> tags<span class="operator">=</span><span class="number">100</span><span class="operator">%, list=11%</span><span class="punctuation">,</span> signal<span class="operator">=</span><span class="number">95</span>%</span><br><span class="line">core_enrichment</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> CG15892<span class="operator">/</span>CG15136<span class="operator">/</span>CG4983<span class="operator">/</span>CG43851<span class="operator">/</span>…</span><br><span class="line"><span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span> <span class="number">53</span> <span class="number">11</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>data.frame<span class="punctuation">(</span>gsego<span class="operator">$</span>ID<span class="punctuation">,</span>gsego<span class="operator">$</span>Description<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">   gsego.ID gsego.Description</span><br><span class="line"><span class="number">1</span> GO<span class="operator">:</span><span class="number">0003674</span> molecular_function</span><br><span class="line"><span class="number">2</span> GO<span class="operator">:</span><span class="number">0005488</span>            binding</span><br></pre></td></tr></table></figure>
</li>
<li><p>提供注释文件go_annotation.txt的非模式物种分析 GSEA()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;go_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">gsego <span class="operator">&lt;-</span> GSEA<span class="punctuation">(</span>genes<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>gsego结果文件解释</p>
</li>
</ol>
<ul>
<li>ID: Gene Ontology数据库中唯一的标号信息</li>
<li>Description：Gene Ontology功能的描述信息</li>
<li>setSize</li>
<li>enrichmentScore：富集分数</li>
<li>NES</li>
<li>pvalue: 富集分析统计学显著水平，一般情况下， P-value &lt; 0.05 该功能为富集项</li>
<li>p.adjust：矫正后的P-Value</li>
<li>qvalues：对p值进行统计学检验的q值</li>
<li>rank</li>
<li>leading_edge</li>
<li>core_enrichment</li>
</ul>
<h3 id="6-1-4-GO富集结果可视化"><a href="#6-1-4-GO富集结果可视化" class="headerlink" title="6.1.4. GO富集结果可视化"></a>6.1.4. GO富集结果可视化</h3><p><code>goplot(ego)</code>简单可视化结果为有向无环图。</p>
<h2 id="6-2-KEGG富集分析"><a href="#6-2-KEGG富集分析" class="headerlink" title="6.2. KEGG富集分析"></a>6.2. KEGG富集分析</h2><p>clusterProfiler通过<a target="_blank" rel="noopener" href="https://www.kegg.jp/kegg/rest/keggapi.html">KEGG数据库的API</a>来获取KEGG的注释信息，包括一个物种所有基因对应的pathway注释文件，比如人的：<a target="_blank" rel="noopener" href="http://rest.kegg.jp/link/hsa/pathway%EF%BC%9B%E5%92%8Cpathway%E5%AF%B9%E5%BA%94%E7%9A%84%E6%8F%8F%E8%BF%B0%E4%BF%A1%E6%81%AF%EF%BC%8C%E6%AF%94%E5%A6%82%E4%BA%BA%E7%9A%84%EF%BC%9Ahttp://rest.kegg.jp/list/pathway/hsa%E3%80%82">http://rest.kegg.jp/link/hsa/pathway；和pathway对应的描述信息，比如人的：http://rest.kegg.jp/list/pathway/hsa。</a></p>
<h3 id="6-2-1-支持的物种"><a href="#6-2-1-支持的物种" class="headerlink" title="6.2.1. 支持的物种"></a>6.2.1. 支持的物种</h3><ol>
<li>clusterProfiler包支持的物种<br>只需要将物种缩写输入给clusterProfiler，clusterProfiler包支持自动联网调取<a target="_blank" rel="noopener" href="https://www.genome.jp/kegg/catalog/org_list.html">kegg注释物种</a>列出物种的pathway注释信息，网站可以查看物种列表和缩写，或者用clusterProfiler包提供search_kegg_organism()函数来帮助搜索支持的生物。</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">search_kegg_organism<span class="punctuation">(</span><span class="string">&#x27;osa&#x27;</span><span class="punctuation">,</span> by<span class="operator">=</span><span class="string">&#x27;kegg_code&#x27;</span><span class="punctuation">)</span> <span class="comment">#查询kegg_code为osa的记录，水稻</span></span><br><span class="line">search_kegg_organism<span class="punctuation">(</span><span class="string">&#x27;Escherichia coli&#x27;</span><span class="punctuation">,</span> by<span class="operator">=</span><span class="string">&#x27;scientific_name&#x27;</span><span class="punctuation">)</span> <span class="comment">#查询学名为Escherichia coli的记录</span></span><br></pre></td></tr></table></figure>

<ol start="2">
<li>用户提供KEGG pathway注释数据<br>如果分析的物种不支持自动调取，可以自己做KEGG pathway注释后提供注释文件，比如interproscan、eggnog-mapper等软件的注释结果，整理成clusterProfiler支持的输入格式即可。</li>
</ol>
<p>clusterProfiler需要导入的KEGG pathway注释文件pathway_annotation.txt的格式如下：</p>
<table>
<thead>
<tr>
<th>GeneID</th>
<th>Pathway</th>
<th>Path_Description</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>ko:00001</td>
<td>spindle</td>
</tr>
<tr>
<td>2</td>
<td>ko:00002</td>
<td>mitotic spindle</td>
</tr>
<tr>
<td>3</td>
<td>ko:00003</td>
<td>kinetochore</td>
</tr>
</tbody></table>
<p>data.frame格式，包含三列，第一列为Gene ID，第二列为 KEGG Pathway ID，第三列为Path_Description，顺序无要求。</p>
<h3 id="6-2-2-KEGG-pathway的ORA分析"><a href="#6-2-2-KEGG-pathway的ORA分析" class="headerlink" title="6.2.2. KEGG pathway的ORA分析"></a>6.2.2. KEGG pathway的ORA分析</h3><h4 id="6-2-2-1-KEGG输入ID的格式转换"><a href="#6-2-2-1-KEGG输入ID的格式转换" class="headerlink" title="6.2.2.1. KEGG输入ID的格式转换"></a>6.2.2.1. KEGG输入ID的格式转换</h4><p>ID转换函数</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>clusterProfiler<span class="punctuation">)</span></span><br><span class="line">bitr_kegg<span class="punctuation">(</span><span class="string">&quot;1&quot;</span><span class="punctuation">,</span>fromType <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span>toType <span class="operator">=</span> <span class="string">&#x27;ncbi-proteinid&#x27;</span><span class="punctuation">,</span>organism<span class="operator">=</span><span class="string">&#x27;hsa&#x27;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">library<span class="punctuation">(</span>org.Hs.eg.db<span class="punctuation">)</span></span><br><span class="line">keytypes<span class="punctuation">(</span>org.Hs.eg.db<span class="punctuation">)</span> <span class="comment">#查看支持的ID类型</span></span><br><span class="line">bitr<span class="punctuation">(</span>gene<span class="punctuation">,</span> fromType <span class="operator">=</span> <span class="string">&quot;ENTREZID&quot;</span><span class="punctuation">,</span> toType <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;ENSEMBL&quot;</span><span class="punctuation">,</span> <span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> OrgDb <span class="operator">=</span> org.Hs.eg.db<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#以上看出ID转换输入时，可以向量的形式，也可以单列基因名list导入</span></span><br><span class="line">gene <span class="operator">&lt;-</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;AASDH&quot;</span><span class="punctuation">,</span><span class="string">&quot;ABCB11&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAM12&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAMTS16&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAMTS18&quot;</span><span class="punctuation">)</span></span><br><span class="line">genes  <span class="operator">&lt;-</span>  gene<span class="operator">$</span>V1 <span class="comment">#字符串</span></span><br><span class="line"><span class="comment">#也可以是内置数据</span></span><br><span class="line">data<span class="punctuation">(</span>geneList<span class="punctuation">,</span> package<span class="operator">=</span><span class="string">&quot;DOSE&quot;</span><span class="punctuation">)</span> <span class="comment">#富集分析的背景基因集</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span><span class="punctuation">[</span><span class="built_in">abs</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&gt;</span> <span class="number">2</span><span class="punctuation">]</span></span><br></pre></td></tr></table></figure>

<h4 id="6-2-2-2-KEGG-pathway的ORA分析"><a href="#6-2-2-2-KEGG-pathway的ORA分析" class="headerlink" title="6.2.2.2. KEGG pathway的ORA分析"></a>6.2.2.2. KEGG pathway的ORA分析</h4><p>输入文件与GO的ORA分析输入文件一样。</p>
<ol>
<li>导入输入文件</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br></pre></td></tr></table></figure>

<ol start="2">
<li>支持物种的标准分析 enrichKEGG()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">kk <span class="operator">&lt;-</span> enrichKEGG<span class="punctuation">(</span>gene         <span class="operator">=</span> genes<span class="punctuation">,</span></span><br><span class="line">                 keyType      <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">                 organism     <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                 pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span></span><br><span class="line">                 pAdjustMethod<span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span></span><br><span class="line">                 qvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>kk<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line">              ID                                      Description GeneRatio  BgRatio</span><br><span class="line">hsa04750 hsa04750 Inflammatory mediator regulation of TRP channels      <span class="number">5</span><span class="operator">/</span><span class="number">53</span>  <span class="number">97</span><span class="operator">/</span><span class="number">7387</span></span><br><span class="line">hsa04020 hsa04020                        Calcium signaling pathway      <span class="number">6</span><span class="operator">/</span><span class="number">53</span> <span class="number">182</span><span class="operator">/</span><span class="number">7387</span></span><br><span class="line">               pvalue   p.adjust    qvalue                              geneID Count</span><br><span class="line">hsa04750 <span class="number">0.0006135305</span> <span class="number">0.08589427</span> <span class="number">0.0807277</span>             <span class="number">40</span><span class="operator">/</span><span class="number">3556</span><span class="operator">/</span><span class="number">3708</span><span class="operator">/</span><span class="number">5608</span><span class="operator">/</span><span class="number">79054</span>     <span class="number">5</span></span><br><span class="line">hsa04020 <span class="number">0.0018078040</span> <span class="number">0.12654628</span> <span class="number">0.1189345</span>        <span class="number">493</span><span class="operator">/</span><span class="number">1129</span><span class="operator">/</span><span class="number">2066</span><span class="operator">/</span><span class="number">3707</span><span class="operator">/</span><span class="number">3708</span><span class="operator">/</span><span class="number">4842</span>     <span class="number">6</span></span><br></pre></td></tr></table></figure>

<p>输入的ID类型默认是kegg gene id，也可以是ncbi-geneid或ncbi-proteinid或者uniprot等，可以通过上一步格式转换转换ID类型。<br>organism对应物种的三字母缩写，其他参数与GO的ORA分析参数一致。</p>
<p>不像enrichGO()函数有readable()参数，enrichKEGG()函数没有这个参数，当待分析物种在OrgDb数据库中可用时，可以用setReadable()函数把gene ID转换成gene Name。</p>
<ol start="3">
<li>提供注释文件pathway_annotation.txt的非模式物种分析 enricher()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">pathway_anno <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;pathway_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> pathway_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> pathway_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">kk <span class="operator">&lt;-</span> enricher<span class="punctuation">(</span>gene<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span>TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="4">
<li>结果输出到csv文件<br><code>write.csv(summary(kk),&quot;KEGG-enrich.csv&quot;,row.names =FALSE)</code></li>
</ol>
<h3 id="6-2-3-KEGG-pathway的GSEA分析"><a href="#6-2-3-KEGG-pathway的GSEA分析" class="headerlink" title="6.2.3. KEGG pathway的GSEA分析"></a>6.2.3. KEGG pathway的GSEA分析</h3><ol>
<li><p>读取输入文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span><span class="string">&quot;geneList.csv&quot;</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>支持物种的标准分析 gseKEGG()</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">kks <span class="operator">&lt;-</span> gseKEGG<span class="punctuation">(</span>geneList     <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">               keyType      <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">               organism     <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">               nPerm        <span class="operator">=</span> <span class="number">1000</span><span class="punctuation">,</span></span><br><span class="line">               minGSSize    <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span></span><br><span class="line">               maxGSSize    <span class="operator">=</span> <span class="number">500</span><span class="punctuation">,</span></span><br><span class="line">               pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span></span><br><span class="line">               pAdjustMethod<span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span></span><br><span class="line">               verbose      <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>kk2<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>提供注释文件pathway_annotation.txt的非模式物种分析 GSEA()<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;pathway_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">kks <span class="operator">&lt;-</span> GSEA<span class="punctuation">(</span>gene<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure></li>
</ol>
<h3 id="6-2-4-KEGG-module的ORA分析"><a href="#6-2-4-KEGG-module的ORA分析" class="headerlink" title="6.2.4. KEGG module的ORA分析"></a>6.2.4. KEGG module的ORA分析</h3><p><a target="_blank" rel="noopener" href="https://www.genome.jp/kegg/module.html">KEGG Module</a>是手动定义的功能单元的集合。在某些情况下，KEGG 模块有更直接的解释。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">mkk <span class="operator">&lt;-</span> enrichMKEGG<span class="punctuation">(</span>gene <span class="operator">=</span> gene<span class="punctuation">,</span></span><br><span class="line">                   organism <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                   pvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">,</span></span><br><span class="line">                   qvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>mkk<span class="punctuation">)</span></span><br><span class="line"><span class="comment">##            ID                                             Description GeneRatio</span></span><br><span class="line"><span class="comment">## M00912 M00912      NAD biosynthesis, tryptophan =&gt; quinolinate =&gt; NAD       2/9</span></span><br><span class="line"><span class="comment">## M00095 M00095          C5 isoprenoid biosynthesis, mevalonate pathway       1/9</span></span><br><span class="line"><span class="comment">## M00053 M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP =&gt; dCTP       1/9</span></span><br><span class="line"><span class="comment">## M00938 M00938 Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP       1/9</span></span><br><span class="line"><span class="comment">## M00003 M00003            Gluconeogenesis, oxaloacetate =&gt; fructose-6P       1/9</span></span><br><span class="line"><span class="comment">## M00049 M00049     Adenine ribonucleotide biosynthesis, IMP =&gt; ADP,ATP       1/9</span></span><br><span class="line"><span class="comment">##        BgRatio      pvalue   p.adjust     qvalue     geneID Count</span></span><br><span class="line"><span class="comment">## M00912  12/829 0.006541756 0.03925053 0.03443029 23475/3620     2</span></span><br><span class="line"><span class="comment">## M00095  10/829 0.103949536 0.20710097 0.18166751       3158     1</span></span><br><span class="line"><span class="comment">## M00053  11/829 0.113796244 0.20710097 0.18166751       6241     1</span></span><br><span class="line"><span class="comment">## M00938  14/829 0.142761862 0.20710097 0.18166751       6241     1</span></span><br><span class="line"><span class="comment">## M00003  18/829 0.180072577 0.20710097 0.18166751       5105     1</span></span><br><span class="line"><span class="comment">## M00049  21/829 0.207100966 0.20710097 0.18166751      26289     1             </span></span><br></pre></td></tr></table></figure>

<h3 id="6-2-5-KEGG-module的GSEA分析"><a href="#6-2-5-KEGG-module的GSEA分析" class="headerlink" title="6.2.5. KEGG module的GSEA分析"></a>6.2.5. KEGG module的GSEA分析</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">mkk2 <span class="operator">&lt;-</span> gseMKEGG<span class="punctuation">(</span>geneList <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">                 organism <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                 pvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>mkk2<span class="punctuation">)</span></span><br><span class="line"><span class="comment">##            ID                                                      Description</span></span><br><span class="line"><span class="comment">## M00001 M00001        Glycolysis (Embden-Meyerhof pathway), glucose =&gt; pyruvate</span></span><br><span class="line"><span class="comment">## M00035 M00035                                           Methionine degradation</span></span><br><span class="line"><span class="comment">## M00002 M00002         Glycolysis, core module involving three-carbon compounds</span></span><br><span class="line"><span class="comment">## M00938 M00938          Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP</span></span><br><span class="line"><span class="comment">## M00104 M00104 Bile acid biosynthesis, cholesterol =&gt; cholate/chenodeoxycholate</span></span><br><span class="line"><span class="comment">## M00009 M00009                           Citrate cycle (TCA cycle, Krebs cycle)</span></span><br><span class="line"><span class="comment">##        setSize enrichmentScore       NES      pvalue  p.adjust   qvalues rank</span></span><br><span class="line"><span class="comment">## M00001      24       0.5739036  1.683936 0.005050296 0.1616095 0.1488508 2886</span></span><br><span class="line"><span class="comment">## M00035      10       0.6784636  1.637462 0.025966367 0.2710124 0.2496167 1555</span></span><br><span class="line"><span class="comment">## M00002      11       0.6421781  1.585841 0.032455383 0.2710124 0.2496167 1381</span></span><br><span class="line"><span class="comment">## M00938      10       0.6648004  1.604486 0.033876549 0.2710124 0.2496167  648</span></span><br><span class="line"><span class="comment">## M00104      10      -0.5876900 -1.338409 0.128747795 0.6564103 0.6045884  961</span></span><br><span class="line"><span class="comment">## M00009      22       0.4504911  1.315039 0.138271605 0.6564103 0.6045884 3514</span></span><br><span class="line"><span class="comment">##                          leading_edge</span></span><br><span class="line"><span class="comment">## M00001 tags=54%, list=23%, signal=42%</span></span><br><span class="line"><span class="comment">## M00035 tags=50%, list=12%, signal=44%</span></span><br><span class="line"><span class="comment">## M00002 tags=55%, list=11%, signal=49%</span></span><br><span class="line"><span class="comment">## M00938  tags=40%, list=5%, signal=38%</span></span><br><span class="line"><span class="comment">## M00104  tags=50%, list=8%, signal=46%</span></span><br><span class="line"><span class="comment">## M00009 tags=50%, list=28%, signal=36%</span></span><br><span class="line"><span class="comment">##                                                         core_enrichment</span></span><br><span class="line"><span class="comment">## M00001 5214/3101/2821/7167/2597/5230/2023/5223/5315/3099/5232/2027/5211</span></span><br><span class="line"><span class="comment">## M00035                                           875/1789/191/1788/1786</span></span><br><span class="line"><span class="comment">## M00002                                    7167/2597/5230/2023/5223/5315</span></span><br><span class="line"><span class="comment">## M00938                                              6241/7298/4830/1841</span></span><br><span class="line"><span class="comment">## M00104                                        6342/10998/1581/3295/8309</span></span><br><span class="line"><span class="comment">## M00009            3418/50/4190/3419/2271/3421/55753/3417/1431/6389/4191</span></span><br></pre></td></tr></table></figure>

<h3 id="6-2-6-KEGG-pathways富集结果的可视化"><a href="#6-2-6-KEGG-pathways富集结果的可视化" class="headerlink" title="6.2.6. KEGG pathways富集结果的可视化"></a>6.2.6. KEGG pathways富集结果的可视化</h3><p>enrichplot包可以实现几种方法，可以用在GO,KEGG,MSigDb等基因集注释上。</p>
<ol>
<li><p>browseKEGG()函数会打开网络浏览器突出显示富集的基因的KEGG通路，用hsa04110基因举例：<br><code>browseKEGG(kk, &#39;hsa04110&#39;)</code></p>
</li>
<li><p>pathview::pathview()可视化富集的KEGG通路</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">library(&quot;pathview&quot;)</span><br><span class="line">hsa04110 &lt;- pathview(gene.data  = geneList,</span><br><span class="line">                     pathway.id = &quot;hsa04110&quot;,</span><br><span class="line">                     species    = &quot;hsa&quot;,</span><br><span class="line">                     limit      = list(gene=max(abs(geneList)), cpd=1))</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="6-3-通用的富集分析-Universal-enrichment-analysis"><a href="#6-3-通用的富集分析-Universal-enrichment-analysis" class="headerlink" title="6.3. 通用的富集分析(Universal enrichment analysis)"></a>6.3. 通用的富集分析(Universal enrichment analysis)</h2><p>除GO,KEGG基因集外，clusterProfiler还支持WikiPathways，Reactome，Disease，MeSH等数据库的富集分析。</p>
<p>虽然clusterProfiler支持对许多ontology&#x2F;pathway的hypergeometric test和gene set enrichment analyses，但是还有很多数据，包括不支持的物种、不支持的ontologies&#x2F;pathways或自定义注释等。<br>clusterProfiler提供了用于hypergeometric test的enricher()函数和用于基因集富集分析的GSEA()函数，用于接受用户定义的注释。</p>
<p>另外两个参数TERM2GENE和TERM2NAME:</p>
<ul>
<li>TERM2GENE是一个必需的data.frame，第一列为term ID，第二列为对应映射基因；</li>
<li>TERM2NAME是一个可选的data.frame，第一列为term ID，第二列为对应term name。</li>
</ul>
<p>在GO和KEGG富集分析的用户自行提供注释文件的分析部分，使用的enricher()和GSEA()函数。</p>
<h1 id="7-功能富集结果可视化"><a href="#7-功能富集结果可视化" class="headerlink" title="7. 功能富集结果可视化"></a>7. 功能富集结果可视化</h1><h2 id="7-1-enrichplot包"><a href="#7-1-enrichplot包" class="headerlink" title="7.1. enrichplot包"></a>7.1. enrichplot包</h2><p>enrichplot包有几种可视化方法来解释富集结果，支持clusterProfiler获得的ORA和GSEA富集结果。</p>
<ol>
<li><p>安装和载入enrichplot包</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;enrichplot&quot;</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>enrichplot<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>可视化 —— 条形图<br>将富集分数（例如p 值）和基因计数或比率描述为条形高度和颜色。横轴为该GO term下的差异基因个数，纵轴为富集到的GO Terms的描述信息， showCategory指定展示的GO Terms的个数为20个，默认展示显著富集的top10个，即p.adjust最小的10个。</p>
</li>
</ol>
<p><code>barplot(ego, showCategory=20, title=&quot;EnrichmentGO_MF&quot;)</code></p>
<p>使用mutate导出的其他变量也可以用作条形高度或颜色。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mutate<span class="punctuation">(</span>edo<span class="punctuation">,</span> qscore <span class="operator">=</span> <span class="operator">-</span><span class="built_in">log</span><span class="punctuation">(</span>p.adjust<span class="punctuation">,</span> base<span class="operator">=</span><span class="number">10</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">%&gt;%</span> </span><br><span class="line">    barplot<span class="punctuation">(</span>x<span class="operator">=</span><span class="string">&quot;qscore&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>可视化 —— 点阵图<br><code>dotplot(ego, showCategory = 10, title=&quot;EnrichmentGO_MF_dot&quot;)</code></li>
</ol>
<p>散点图，横坐标为GeneRatio，纵坐标为富集到的GO Terms的描述信息，showCategory指定展示的GO Terms的个数，默认展示显著富集的top10个，即p.adjust最小的10个。</p>
<ol start="4">
<li>可视化 —— 类别网络图<br>cnetplot 将基因和生物学概念（例如 GO 术语或 KEGG 通路）的联系描述为一个网络（有助于查看哪些基因涉及富集通路和可能属于多个注释类别的基因）。对于基因和富集的GO terms之间的对应关系进行展示，如果一个基因位于一个GO Terms下，则将该基因与GO连线。图中灰色的点代表基因，黄色的点代表富集到的GO terms, 默认画top5富集到的GO terms, GO 节点的大小对应富集到的基因个数。</li>
</ol>
<p><code>cnetplot(ego, categorySize = &quot;pvalue&quot;, foldChange = gene_list</code></p>
<ol start="5">
<li>可视化 —— 类热图功能分类</li>
</ol>
<p>heatplot类似cnetplot，而显示为热图的关系。如果用户想要显示大量重要术语，那么类别网络图可能会过于复杂。在heatplot能够简化结果和更容易识别的表达模式。</p>
<p><code>heatplot(ego, foldChange=geneList, showCategory=5)</code></p>
<ol start="6">
<li>可视化 —— 树状图<br>treeplot()函数执行丰富术语的层次聚类。它依赖于pairwise_termsim()函数计算的丰富项的成对相似性，默认情况下使用 Jaccard 的相似性指数 (JC)。如果支持，用户还可以使用语义相似度值（例如，GO、DO和MeSH）。</li>
</ol>
<p>默认聚合方法treeplot()是ward.D，用户可以通过hclust_method参数指定其他方法（例如，’average’、’complete’、’median’、’centroid’等。</p>
<p>treeplot()函数会将树切割成几个子树（由nCluster参数指定（默认为 5））并使用高频词标记子树。</p>
<p><code>ego2 &lt;- pairwise_termsim(ego)</code>;<br><code>treeplot(ego2, hclust_method = &quot;average&quot;)</code></p>
<ol start="7">
<li>可视化 —— 富集图<br>对于富集到的GO terms之间的基因重叠关系进行展示，如果两个GO terms系的差异基因存在重叠，说明这两个节点存在overlap关系，在图中用线条连接起来。每个节点是一个富集到的GO term, 默认画top30个富集到的GO terms, 节点大小对应该GO terms下富集到的差异基因个数，节点的颜色对应p.adjust的值，从小到大，对应蓝色到红色。</li>
</ol>
<p><code>ego2 &lt;- pairwise_termsim(ego)</code>;<br><code>emapplot(ego2, cex_category=1.5, layout=&quot;kk&quot;)</code></p>
<ol start="8">
<li>可视化 —— upset图<br>upsetplot是cnetplot可视化基因和基因集之间复杂关联的替代方法。它强调不同基因集之间的基因重叠。</li>
</ol>
<p><code>upsetplot(ego)</code></p>
<ol start="9">
<li>可视化 —— 脊线图<br>ridgeplot将可视化核心富集基因的表达分布为GSEA富集类别。它帮助用户解释上调&#x2F;下调的途径。</li>
</ol>
<p><code>ridgeplot(ego)</code></p>
<h2 id="7-2-可视化-——-有向无环图"><a href="#7-2-可视化-——-有向无环图" class="headerlink" title="7.2. 可视化 —— 有向无环图"></a>7.2. 可视化 —— 有向无环图</h2><ol>
<li><p><code>plotGOgraph(ego)</code><br>矩形代表富集到的top10个GO Terms，颜色从黄到红，对应p值从大到小。</p>
</li>
<li><p><code>goplot(ego, showCategory = 10)</code><br>igraph布局方式的有向无环图</p>
</li>
</ol>
<h2 id="7-3-可视化-——-wordcloud"><a href="#7-3-可视化-——-wordcloud" class="headerlink" title="7.3. 可视化 —— wordcloud"></a>7.3. 可视化 —— wordcloud</h2><p>词云的方式显示结果</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wcdf <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span>text <span class="operator">=</span> ego<span class="operator">$</span>GeneRatio<span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&quot;/&quot;</span><span class="punctuation">)</span><span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span></span><br><span class="line">wcdf<span class="operator">$</span>term <span class="operator">&lt;-</span>  ego<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line">wordcloud<span class="punctuation">(</span>words <span class="operator">=</span> wcdf<span class="operator">$</span>term<span class="punctuation">,</span> freq <span class="operator">=</span> wcdf<span class="operator">$</span>V1<span class="punctuation">,</span> scale<span class="operator">=</span><span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="number">4</span><span class="punctuation">,</span> <span class="number">.1</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> colors<span class="operator">=</span>brewer.pal<span class="punctuation">(</span><span class="number">8</span><span class="punctuation">,</span> <span class="string">&quot;Dark2&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> max.words <span class="operator">=</span> <span class="number">25</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<h1 id="8-references"><a href="#8-references" class="headerlink" title="8. references"></a>8. references</h1><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Gene_set_enrichment_analysis">GSEA wiki</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/47b5ea646932">enrichment</a><br><a target="_blank" rel="noopener" href="https://github.com/YuLab-SMU/clusterProfiler">clusterProfiler github</a><br><a target="_blank" rel="noopener" href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">universal enrichment analysis using clusterProfiler</a><br><a target="_blank" rel="noopener" href="https://www.cell.com/the-innovation/fulltext/S2666-6758(21)00066-7?_returnURL=https://linkinghub.elsevier.com/retrieve/pii/S2666675821000667?showall=true">clusterProfiler paper</a><br><a target="_blank" rel="noopener" href="https://www.cnblogs.com/jessepeng/p/12159139.html">tutorial</a><br><a target="_blank" rel="noopener" href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">函数simplify</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/12/13/bioinfo_GSEA_clusterProfiler/" data-id="cl1aiqcr5001jm4vk3hl4hxvo" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_OrthoFinder2" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/12/06/bioinfo_OrthoFinder2/" class="article-date">
  <time datetime="2021-12-06T09:20:00.000Z" itemprop="datePublished">2021-12-06</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/12/06/bioinfo_OrthoFinder2/">用OrthoFinder做直系同源推断</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=108151&auto=1&height=32"></iframe></div>

<h1 id="1-直系同源-orthology-推断"><a href="#1-直系同源-orthology-推断" class="headerlink" title="1. 直系同源(orthology)推断"></a>1. 直系同源(orthology)推断</h1><h2 id="1-1-直系同源推断"><a href="#1-1-直系同源推断" class="headerlink" title="1.1. 直系同源推断"></a>1.1. 直系同源推断</h2><p>直系同源(orthology)是指直系同源基因的直系同源这种属性，相关的概念可以参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/06/23/concept_homology/">博文-同源性相关概念</a>。</p>
<p>直系同源推断(orthology inference)常常在多个物种间进行，通常是获取多个物种的直系同源群(orthogroups)和直系同源群间的关系。</p>
<p>直系同源基因判定的方法有两种：</p>
<ul>
<li>一种是将物种的直系同源基因两两比较，不同物种的每一对都是直系同源，但存在基因重复，使得用这种方法找到的同源关系是不具有传递性的（A和B是同源的，B和C是同源的，但是A和C不是同源的）。multiparanoid和OMA方法都是这个原理。所以需要每个同源基因都在多个集合中存在才行，进行两两比较。</li>
<li>另一种是识别完整的正交群，一个同源组orthogroup包含所有物种共同祖先基因，同时包含直系／旁系同源基因，OrthoMCL是这个原理，它是通过正交组做推断，用BLAST计算多个物种序列之间的序列相似性得分，然后用MCL聚类算法，识别该数据集中高度相似的序列组。</li>
</ul>
<h2 id="1-2-直系同源基因推断常用软件：OrthoMCL和OrthoFinder"><a href="#1-2-直系同源基因推断常用软件：OrthoMCL和OrthoFinder" class="headerlink" title="1.2. 直系同源基因推断常用软件：OrthoMCL和OrthoFinder"></a>1.2. 直系同源基因推断常用软件：OrthoMCL和OrthoFinder</h2><h3 id="1-2-1-OrthoMCL"><a href="#1-2-1-OrthoMCL" class="headerlink" title="1.2.1. OrthoMCL"></a>1.2.1. OrthoMCL</h3><p>OrthoMCL是2003年发表的工具。发表以后得到广泛使用，但存在一些问题：</p>
<ul>
<li>2013年之后就没更新，安装需要使用MySQL。</li>
<li>在BLAST序列相似度时发现，长序列比短序列更准确，有时会留下较长的质量差的序列，而丢弃较短的单质量好的序列。</li>
<li>orthoBench是唯一可用的公开的正交群标准数据集，用orthoBench数据集评估orthoMCL性能，发现它对序列长度有依赖性，太长或者太短的序列都不准确。</li>
<li>recall率有待提高。</li>
</ul>
<h3 id="1-2-2-OrthoFinder"><a href="#1-2-2-OrthoFinder" class="headerlink" title="1.2.2. OrthoFinder"></a>1.2.2. OrthoFinder</h3><p>OrthoFinder是2015年才出现，2019年正式发表的工具。OrthoFinder解决了一些OrthoMCL中存在的问题，展现了优势：</p>
<ul>
<li>持续更新，安装友好，运行速度快。</li>
<li>OrthoFinder解决了这个问题，它将BLAST评分做了转换，将原来的e值准换成bit值（比特值）来减少序列长度对准确度的影响，因为e值存在阈值，e 小于10的－180次方的都被算作0，而比特值没有阈值。</li>
<li>OrthoFinder用RBNH代替了传统的RBH提高了recall。</li>
</ul>
<p>这篇博文主要介绍OrthoFinder这个工具。</p>
<h1 id="2-OrthoFinder"><a href="#2-OrthoFinder" class="headerlink" title="2. OrthoFinder"></a>2. OrthoFinder</h1><h2 id="2-1-OrthoFinder的功能"><a href="#2-1-OrthoFinder的功能" class="headerlink" title="2.1. OrthoFinder的功能"></a>2.1. OrthoFinder的功能</h2><p>OrthoFinder是用于推断系统发育直系同源关系的基于python的软件。OrthoFinder1模块推断直系同源群，OrthoFinder2模块在直系同源群的基础上构建物种系统发育树，步骤如下：</p>
<ol>
<li>推断所有物种的直系同源群和直系同源基因。</li>
</ol>
<ul>
<li>用all vs all比对所有输入的物种蛋白序列；</li>
<li>RBNHs方法确定阈值；</li>
<li>MCL聚类获得直系同源组。</li>
</ul>
<ol start="2">
<li>推断找到的所有直系同源基因的有根基因树。</li>
</ol>
<ul>
<li>为每个直系同源组构建基因系统发育树；</li>
<li>使用STAG算法从无根基因树上构建无根物种树；</li>
<li>使用STRIDE算法构建有根物种树；</li>
<li>有根物种树进一步辅助构建有根基因树。</li>
</ul>
<ol start="3">
<li>用基因树推断基因间的所有直系同源群间的关系。</li>
<li>推断基因复制事件并交叉引用到对应的基因树和物种树上的对应节点。</li>
<li>提供比较基因组学统计结果。</li>
</ol>
<img src="https://github.com/davidemms/OrthoFinder/raw/master/assets/Workflow.png" width=100% title="The OrthoFinder workflow" alt="The OrthoFinder workflow" align=center/>

<p><strong><p align="center">Figure 1. The OrthoFinder workflow</strong><br>from <a target="_blank" rel="noopener" href="https://github.com/davidemms/OrthoFinder">github: orthofinder</a></p></p>
<h2 id="2-2-OrthoFinder的分步原理"><a href="#2-2-OrthoFinder的分步原理" class="headerlink" title="2.2. OrthoFinder的分步原理"></a>2.2. OrthoFinder的分步原理</h2><h3 id="2-2-1-OrthoFinder1推断直系同源群的基本原理"><a href="#2-2-1-OrthoFinder1推断直系同源群的基本原理" class="headerlink" title="2.2.1. OrthoFinder1推断直系同源群的基本原理"></a>2.2.1. OrthoFinder1推断直系同源群的基本原理</h3><ul>
<li>首先BLAST all vs all 做所有物种蛋白序列的比对。选best hits Top5%，用比特值代替e值，用最小二乘法将Top5%的分数值拟合成一个线性模型，将比特值用这个模型转换之后，较长的质量差的hit将不再优于短的质量好的序列；</li>
<li>然后将基因长度和系统发育距离标准化。用的是RBNH方法，传统的是用RBH方法，它是一种高精度鉴定直系同源基因对的方法，用BLAST best hit的倒数做的，但是RBNH是用标准话的比特值来做的。</li>
</ul>
<h3 id="2-2-2-OrthoFinder1推断直系同源群的工作流程"><a href="#2-2-2-OrthoFinder1推断直系同源群的工作流程" class="headerlink" title="2.2.2. OrthoFinder1推断直系同源群的工作流程"></a>2.2.2. OrthoFinder1推断直系同源群的工作流程</h3><p>OrthoFinder1推断直系同源群的分析过程分为如下几步:</p>
<ol>
<li>BLAST all-vs-all搜索。使用BLASTP以evalue&#x3D;10e-3进行搜索，寻找潜在的同源基因。(除了BLAST, 还可以选择DIAMOND和MMSeq2等其他搜索工具)。</li>
<li>基于基因长度和系统发育距离对BLAST bit得分进行标准化。</li>
<li>使用RBNHs方法，确定一个同源组序列性相似度的阈值，筛选出质量好的直系同源组(orthogroup graph)。</li>
<li>构建直系同源组图(orthogroup graph)，通过归一化的比特值画出正交组的边缘连接，用作MCL的输入。</li>
<li>使用MCL对基因进行聚类，划分直系同源组。</li>
</ol>
<h3 id="2-2-3-OrthoFinder2在OrthoFinder1的基础上构建物种系统发育树的流程"><a href="#2-2-3-OrthoFinder2在OrthoFinder1的基础上构建物种系统发育树的流程" class="headerlink" title="2.2.3. OrthoFinder2在OrthoFinder1的基础上构建物种系统发育树的流程"></a>2.2.3. OrthoFinder2在OrthoFinder1的基础上构建物种系统发育树的流程</h3><ol>
<li>为每个直系同源组构建基因系统发育树；</li>
<li>使用STAG算法从无根基因树上构建无根物种树；</li>
<li>使用STRIDE算法构建有根物种树；</li>
<li>有根物种树进一步辅助构建有根基因树；</li>
<li>基于DLC(Duplication-Loss-Coalescent)分析模型，有根基因树可以用来推断物种形成和基因复制事件；</li>
<li>最后把结果记录在统计信息中。</li>
</ol>
<h2 id="2-3-OrthoFinder使用"><a href="#2-3-OrthoFinder使用" class="headerlink" title="2.3. OrthoFinder使用"></a>2.3. OrthoFinder使用</h2><h3 id="2-3-1-OrthoFinder安装"><a href="#2-3-1-OrthoFinder安装" class="headerlink" title="2.3.1. OrthoFinder安装"></a>2.3.1. OrthoFinder安装</h3><ol>
<li><p>conda安装【推荐】<br>conda安装一次解决依赖问题：<code>conda install orthofinder</code></p>
</li>
<li><p>下载解压缩<br><a target="_blank" rel="noopener" href="https://github.com/davidemms/OrthoFinder/releases">OrthoFinder最新版下载地址</a></p>
</li>
</ol>
<p>下载后解压缩：<code>tar xzf OrthoFinder_source.tar.gz</code> or <code>tar xzf OrthoFinder.tar.gz</code></p>
<p>即可使用：<code>python OrthoFinder_source/orthofinder.py -h</code> or <code>./OrthoFinder/orthofinder -h</code></p>
<h3 id="2-3-2-OrthoFinder运行"><a href="#2-3-2-OrthoFinder运行" class="headerlink" title="2.3.2. OrthoFinder运行"></a>2.3.2. OrthoFinder运行</h3><h4 id="2-3-2-1-输入文件"><a href="#2-3-2-1-输入文件" class="headerlink" title="2.3.2.1. 输入文件"></a>2.3.2.1. 输入文件</h4><p>把需要分析的物种的组学蛋白序列文件(.fa,.faa,.fasta,.fas,.pep后缀)放在同一个目录下(例如workdirectory目录下)，然后用OrthoFinder的-f参数指定目录做分析。</p>
<p>如果是下载的蛋白数据，使用最长转录本的蛋白序列即可。</p>
<h4 id="2-3-2-2-简单版"><a href="#2-3-2-2-简单版" class="headerlink" title="2.3.2.2. 简单版"></a>2.3.2.2. 简单版</h4><p><code>orthofinder -f ./workdirectory</code> # -f指定数据目录</p>
<h4 id="2-3-2-3-参数版"><a href="#2-3-2-3-参数版" class="headerlink" title="2.3.2.3. 参数版"></a>2.3.2.3. 参数版</h4><p><code>orthofinder -f ./workdirectory -t 24 -a 8 -M msa -S blast -A mafft -T raxml-ng</code></p>
<ul>
<li>-t指定比对线程;</li>
<li>-a指定分析线程;</li>
<li>-M指定推断基因树的方法: dendroblast(default)&#x2F;msa;</li>
<li>-S指定比对软件: diamond(default)&#x2F;blast&#x2F;blast_gz&#x2F;mmseqs&#x2F;blast_nucl</li>
<li>-A指定多序列比对(MSA)使用软件：mafft(default)&#x2F;muscle；需要指定-M msa</li>
<li>-T指定画树软件：fasttree(default)&#x2F;raxml&#x2F;raxml-ng&#x2F;iqtree</li>
</ul>
<h4 id="2-3-2-4-修改参数文件config-json"><a href="#2-3-2-4-修改参数文件config-json" class="headerlink" title="2.3.2.4. 修改参数文件config.json"></a>2.3.2.4. 修改参数文件config.json</h4><ol>
<li>如果需要做更细致的参数设定，比如对调用软件blast的参数evalue阈值的修改，可以通过修改参数文件config.json来实现。</li>
<li>config.json文件位置：下载安装的在安装目录下，conda安装的在conda的bin目录下。</li>
<li>推荐修改值：</li>
</ol>
<ul>
<li>raxml-ng参数修改为–model LG+G8+F –threads 2 –bs-trees 1000（default是–model LG+G4 –threads 1)；</li>
<li>blastp和blastn的参数修改为 -evalue 0.00001（default是0.001）</li>
</ul>
<h4 id="2-3-2-5-RESTART"><a href="#2-3-2-5-RESTART" class="headerlink" title="2.3.2.5. RESTART"></a>2.3.2.5. RESTART</h4><p>如果需要重新运行，可以使用以下参数设定从哪个步骤开始运行。</p>
<ul>
<li>-b  <dir> # Start OrthoFinder from pre-computed BLAST results in <dir></li>
<li>-fg <dir> # Start OrthoFinder from pre-computed orthogroups in <dir></li>
<li>-ft <dir> # Start OrthoFinder from pre-computed gene trees in <dir></li>
</ul>
<ol>
<li>添加新物种<br>运行之后想要添加物种可以用<code>orthofinder -b previous_orthofinder_directory -f new_fasta_directory</code>而无需重新运行先前计算的 BLAST 搜索。</li>
</ol>
<p>这会将“new_fasta_directory”中的每个物种添加到现有物种集中，重用所有以前的 BLAST 结果，仅执行新物种所需的新 BLAST 搜索并重新计算正交群。’previous_orthofinder_directory’ 是包含文件 ‘SpeciesIDs.txt’ 的 OrthoFinder ‘WorkingDirectory&#x2F;‘。</p>
<ol start="2">
<li>移除物种<br>OrthoFinder 允许您从以前的分析中删除物种。<br>在先前分析的“WorkingDirectory&#x2F;”中有一个名为“SpeciesIDs.txt”的文件。使用“#”字符注释掉要从分析中删除的任何物种，然后使用以下命令运行 OrthoFinder：<code>orthofinder -b previous_orthofinder_directory</code></li>
</ol>
<p>其中 ‘previous_orthofinder_directory’ 是包含文件 ‘SpeciesIDs.txt’ 的 OrthoFinder ‘WorkingDirectory&#x2F;‘。</p>
<ol start="3">
<li>同时添加和删除物种<br>前面两个选项可以结合起来，如上所述注释掉要删除的物种，然后使用命令：<br><code>orthofinder -b previous_orthofinder_directory -f new_fasta_directory</code></li>
</ol>
<h4 id="2-3-2-6-结果文件"><a href="#2-3-2-6-结果文件" class="headerlink" title="2.3.2.6. 结果文件"></a>2.3.2.6. 结果文件</h4><p>OrthoFinder的标准输出包括：直系同源组，直系同源基因，有根基因树，解析基因树，无根物种树、有根物种树，基因重复事件以及相关的统计数据。</p>
<ol>
<li><p><strong>Phylogenetic_Hierarchical_Orthogroups 文件夹</strong></p>
<p> 从 2.4.0 版本开始，主要文件储存在这个目录下面，orthogroups目录被弃用。</p>
<p> OrthoFinder 通过分析有根基因树来推断每个层次级别（即物种树中的每个节点）的 HOG、正交群。这是一种比所有其他方法使用的和以前由 OrthoFinder 使用的基于基因相似性&#x2F;图形的方法（已弃用的 Orthogroups&#x2F;Orthogroups.tsv 文件）更准确的正交群推断方法。</p>
<p> 根据 Orthobench 基准，这些新的正交群比 OrthoFinder 2 正交群 (Orthogroups&#x2F;Orthogroups.tsv) 准确 12%。通过包括外群物种，可以进一步提高准确度（在 Orthobench 上准确度提高 20%），这有助于解释有根基因树。</p>
</li>
</ol>
<ul>
<li>参阅“Species_Tree&#x2F;SpeciesTree_rooted_node_labels.txt”以确定哪个 N?.tsv 文件包含您需要的正交群。</li>
<li>N0.tsv是制表符分隔的文本文件。每行包含属于单个正交群的基因。来自每个正交群的基因被组织成列，每个物种一个。附加列给出了 HOG（分层正交群）ID 和基因树中确定 HOG 的节点（注意，这可能位于包含基因的进化枝的根上方）。该文件有效地替换了使用 MCL 进行马尔可夫聚类的Orthogroups&#x2F;Orthogroups.tsv 中的正交群。</li>
<li>N1.txt, N2.tsv, … : Orthogroups 从对应物种树 N1、N2 等物种进化枝的基因树推断出来。现在可以在分析中包含外群物种，然后使用 HOG 文件获取为物种树中所选进化枝定义的正交群。</li>
</ul>
<ol start="2">
<li><strong>Orthologues 文件夹</strong></li>
</ol>
<ul>
<li>Orthologues 目录包含每个物种的一个子目录，该子目录又包含每个成对物种比较的文件，列出该物种对之间的直向同源物。</li>
<li>直向同源物可以是一对一、一对多或多对多，这取决于直向同源物分化后的基因复制事件。</li>
<li>文件中的每一行都包含一个物种中的基因，这些基因是另一个物种中基因的直向同源物，并且每一行都与包含这些基因的直向群交叉引用。</li>
</ul>
<ol start="3">
<li><strong>Orthogroups 文件夹【2.4.0版本后弃用】</strong><br>从 2.4.0 版本开始，主要文件储存在Phylogenetic_Hierarchical_Orthogroups目录下面，orthogroups目录被弃用。</li>
</ol>
<ul>
<li>Orthogroups.tsv【2.4.0版本后弃用】：制表符分隔的文本文件。每行包含属于单个正交群的基因。来自每个正交群的基因被组织成列，每个物种一个。应改用 Phylogenetic_Hierarchical_Orthogroups&#x2F;N0.tsv 中的正交群。</li>
<li>Orthogroups.txt【旧格式】：包含 Orthogroups.tsv 文件内容的 OrthoMCL 输出格式文件。</li>
<li>Orthogroups_UnassignedGenes.tsv：是制表符分隔的文本文件，其格式与 Orthogroups.csv 相同，但包含未分配给任何正交群的所有基因，即 MCL 中 未成功聚类（直系同源组中基因数 &gt;&#x3D; 1）的离散基因。</li>
<li>Orthogroups.GeneCount.tsv：是一个制表符分隔的文本文件，其格式与 Orthogroups.csv 相同，但包含每个正交群中每个物种的基因数计数，可用于基因收缩扩张分析。</li>
<li>Orthogroups_SingleCopyOrthologues.txt：是一个正交群列表，每个物种只包含一个基因，即它们包含一对一的直向同源物（单拷贝直系同源基因）。它们非常适合物种间比较和物种树推断。</li>
</ul>
<ol start="4">
<li><strong>Gene_Trees 文件夹</strong></li>
</ol>
<ul>
<li>记录了每个 orthogroup（gene_num &gt;&#x3D; 4）的有根基因树结构。</li>
</ul>
<ol start="5">
<li><strong>Resolved Gene Trees文件夹</strong></li>
</ol>
<ul>
<li>为具有 4 个或更多序列的每个正交群推断出有根的系统发育树，并使用OrthoFinder的杂种-重叠&#x2F;重复-丢失溯祖模型(hybrid species-overlap&#x2F;duplication-loss coalescent model)进行解析。</li>
</ul>
<ol start="6">
<li><strong>Species_Tree 文件夹</strong></li>
</ol>
<ul>
<li>SpeciesTree_rooted.txt：STAG物种树从所有orthogroups推断，含有内部节点STAG支持值，并使用STRIDE生根，计算出的有根物种树结构。</li>
<li>SpeciesTree_rooted_node_labels.txt：与上述相同的树，但节点被赋予标签(N0,N1, . . . , N m N_0,N_1,…,N_mN0,N1,…,Nm)（而不是支持值）以允许其他结果文件交叉引用物种树中的分支&#x2F;节点（例如基因复制事件的位置）。</li>
<li>Orthogroups_for_concatenated_alignment.txt：仅在 -M msa 模式下输出，列出了所有串联起来用于推断物种树的 orthogroup ID。</li>
</ul>
<ol start="7">
<li><strong>Comparative_Genomics_Statistics 文件夹</strong></li>
</ol>
<ul>
<li><p>Duplications_per_Orthogroup.tsv：是一个制表符分隔的文本文件，它给出了每个正交群中标识的重复数。此数据的主文件是 Gene_Duplication_Events&#x2F;Duplications.tsv。</p>
</li>
<li><p>Duplications_per_Species_Tree_Node.tsv：是一个制表符分隔的文本文件，它给出了识别为沿着物种树的每个分支发生的重复数。此数据的主文件是 Gene_Duplication_Events&#x2F;Duplications.tsv。</p>
</li>
<li><p>Orthogroups_SpeciesOverlaps.tsv：是一个制表符分隔的文本文件，其中包含作为方阵的每个物种对之间共享的正交群的数量。</p>
</li>
<li><p>OrthologuesStats_*.tsv：是制表符分隔的文本文件，包含矩阵给出每对物种之间一对一、一对多和多对多关系中的直向同源物数量。</p>
<ul>
<li>OrthologuesStats_one-to-one.tsv是每个物种对之间一对一直向同源物的数量。</li>
<li>OrthologuesStats_many-to-many.tsv包含每个物种对的多对多关系中的直向同源物的数量（由于物种形成后两个谱系中的基因重复事件）。条目 (i,j) 是物种 i 中与物种 j 中的基因存在多对多直系关系的基因数。</li>
<li>OrthologuesStats_one-to-many.tsv：条目 (i,j) 给出物种 i 中与物种 j 的基因处于一对多直系关系的基因数量。这里有一个示例结果文件的演练： https : &#x2F;&#x2F;github.com&#x2F;davidemms&#x2F;OrthoFinder&#x2F;issues&#x2F;259。</li>
<li>OrthologuesStats_many-to-one.tsv：条目 (i,j) 给出物种 i 中与物种 j 的基因处于多对一直系关系的基因数量。这里有一个示例结果文件的演练： https : &#x2F;&#x2F;github.com&#x2F;davidemms&#x2F;OrthoFinder&#x2F;issues&#x2F;259。</li>
<li>OrthologuesStats_Total.tsv包含任何多样性的每个物种的直向同源物对的总数。条目 (i,j) 是物种 i 中在物种 j 中具有直向同源物的基因总数。</li>
</ul>
</li>
<li><p>Statistics_Overall.tsv：是一个制表符分隔的文本文件，其中包含有关正交群大小和分配给正交群的基因比例的一般统计信息。</p>
</li>
<li><p>Statistics_PerSpecies.tsv：是一个制表符分隔的文本文件，包含与 Statistics_Overall.csv 文件相同的信息，但针对每个单独的物种。</p>
</li>
</ul>
<p>文件“Statistics_Overall.csv”和“Statistics_PerSpecies.csv”中的大部分术语都是不言自明的，其余的定义如下。</p>
<ul>
<li>Species-specific orthogroup：完全由来自一个物种的基因组成的正交群。</li>
<li>G50：正交群中的基因数量，使得 50% 的基因处于该大小或更大的正交群中。</li>
<li>O50：正交群的最小数量，使得 50% 的基因处于该大小或更大的正交群中。</li>
<li>单拷贝正群：每个物种只有一个基因（并且没有更多）的正群。这些正交群是推断物种树和许多其他分析的理想选择。</li>
<li>未分配基因：未与任何其他基因放在同一个群中的基因。</li>
</ul>
<ol start="8">
<li><strong>Gene_Duplication_Events 文件夹</strong></li>
</ol>
<ul>
<li>Duplications.tsv：是一个制表符分隔的文本文件，它列出了通过检查每个正群基因树的每个节点识别出的所有基因复制事件。列是“Orthogroup”，“Species Tree node”（发生复制的物种树的分支，参见Species_Tree&#x2F;SpeciesTree_rooted_node_labels.txt），“Gene tree node”（与基因复制事件对应的节点，参见相应的orthogroup Resolved_Gene_Trees&#x2F;) 中的树；”Support”（存在复制基因的两个副本的预期物种的比例）；“Type”（”Terminal”：物种树终端分支上的重复，”Non-Terminal”：物种树内部分支上的重复，因此被多个物种共享，”Non-Terminal”：STRIDE检查基因树的拓扑结构在复制后应该是什么）；“Genes 1”（基因列表来自复制基因的一个副本），“Genes 2”（基因列表来自复制基因的另一个副本）。</li>
<li>SpeciesTree_Gene_Duplications_0.5_Support.txt：提供了物种树分支上的上述重复的总和。它是一个 newick 格式的文本文件。每个节点或物种名称后面的数字是在导致节点&#x2F;物种的分支上发生的具有至少 50% 支持度的基因复制事件的数量。分支长度是标准分支长度，如 Species_Tree&#x2F;SpeciesTree_rooted.txt 中给出的。</li>
</ul>
<ol start="9">
<li><strong>Orthogroup_Sequences 文件夹</strong></li>
</ol>
<ul>
<li>FASTA格式文件，每个正交群的 FASTA 文件给出了正交群中每个基因的氨基酸序列。</li>
</ul>
<ol start="10">
<li><strong>Single_Copy_Orthologue_Sequences 文件夹</strong></li>
</ol>
<ul>
<li>FASTA格式文件，包含每个单拷贝 Orthogroup 所包含的氨基酸的序列信息。</li>
</ul>
<ol start="11">
<li><strong>WorkingDirectory 文件夹</strong></li>
</ol>
<ul>
<li>OrthoFinder 运行所需的必需中间文件，如 DIAMOND 比对结果，STAG 输出的无根物种树等。</li>
</ul>
<ol start="12">
<li><strong>MultipleSequenceAlignments 文件夹</strong></li>
</ol>
<ul>
<li>此文件夹仅在 -M msa 模式下输出，均为 FASTA 格式文件。</li>
<li>记录了每个 orthogroup 中序列间的多序列比对结果。</li>
<li>记录了程序通过 CMSA 算法过滤后的 orthogroup 中各序列串联后的多序列比对结果，同时比对结果中空位数 &gt; 50% 的列已被删除。</li>
</ul>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://davidemms.github.io/">tutorial</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/davidemms/OrthoFinder">github</a></li>
<li><a target="_blank" rel="noopener" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1832-y">paper</a></li>
<li><a target="_blank" rel="noopener" href="https://xuzhougeng.top/archives/OrthoFinder2-fast-and-accurate-phylogenomic-orthology-analysis-from-gene-sequences">xuzhougeng’s blog</a></li>
<li><a target="_blank" rel="noopener" href="https://blog.csdn.net/weixin_44614936/article/details/101640473">website</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/12/06/bioinfo_OrthoFinder2/" data-id="cl1aiqcrb001xm4vk2mi12mup" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_WGD" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/27/bioinfo_WGD/" class="article-date">
  <time datetime="2021-11-27T09:50:00.000Z" itemprop="datePublished">2021-11-27</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/27/bioinfo_WGD/">基因组WGD事件的鉴定和时间估算</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=29808885&auto=1&height=32"></iframe></div>

<h1 id="1-MCScanX介绍"><a href="#1-MCScanX介绍" class="headerlink" title="1. MCScanX介绍"></a>1. MCScanX介绍</h1><p>MCScanX调整<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/05/bioinfo_MCscan/">MCScan算法</a>进行检测基因组间或基因组内的同线性，并附加了14个下游分析和可视化的脚本。</p>
<h1 id="2-MCScanX同线性分析"><a href="#2-MCScanX同线性分析" class="headerlink" title="2. MCScanX同线性分析"></a>2. MCScanX同线性分析</h1><h2 id="2-1-准备输入文件"><a href="#2-1-准备输入文件" class="headerlink" title="2.1. 准备输入文件"></a>2.1. 准备输入文件</h2><p>MCScanX做同线性分析需要两个输入文件sample.gff(四列数据)和sample.blast。</p>
<p>如果是多个物种，则把多个物种的gff3文件和pep.fa文件合并后再准备sample.gff和sample.blast输入文件。</p>
<ol>
<li>sample.gff<br><code>cat sample.gene.gff3 |awk &#39;&#123;if($3==&quot;gene&quot;)&#123;print $1,$9,$4,$5&#125;&#125;&#39;|sed &quot;s/;.*;//g&quot;|sed &quot;s/ID=//g&quot;|sed &quot;s/ /\t/g&quot; &gt;sample.gff #准备gff文件</code></li>
<li>sample.blast<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">makeblastdb -in sample.pep.fa -dbtype prot -out index/sample.pep #给蛋白序列建库</span><br><span class="line">blastp -query sample.pep.fa -db index/sample.pep -out sample.blast -evalue 1e-5 -num_threads 12 -outfmt 6 -num_alignments 5 &amp; #进行自我比对，生成6号格式的比对结果sample.blast</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="2-2-运行MCScanX"><a href="#2-2-运行MCScanX" class="headerlink" title="2.2. 运行MCScanX"></a>2.2. 运行MCScanX</h2><p>在有sample.gff和sample.blast两个文件的目录下，指定前缀sample运行<code>MCScanX sample</code>。</p>
<p>重要参数解释：</p>
<ul>
<li>-s MATCH_SIZE,default: 5。每个共线性区块包含的基因数量的下限。</li>
<li>-m MAX_GAPS，default：25。在共线性区块中允许的最大gaps数量。</li>
<li>-b patterns of collinear blocks。0:intra- and inter-species (default); 1:intra-species; 2:inter-species。</li>
</ul>
<h2 id="2-3-结果文件"><a href="#2-3-结果文件" class="headerlink" title="2.3. 结果文件"></a>2.3. 结果文件</h2><ul>
<li>主要的结果文件是<strong>sample.collinearity</strong>。以#开头的参数(parameters)和基本统计信息(statistics)，以及Alignment 0起始的共线性区块列表信息。</li>
<li>sample.html #目录，内含多个文件</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">############### Parameters ###############</span><br><span class="line"># MATCH_SCORE: 50</span><br><span class="line"># MATCH_SIZE: 5</span><br><span class="line"># GAP_PENALTY: -1</span><br><span class="line"># OVERLAP_WINDOW: 5</span><br><span class="line"># E_VALUE: 1e-05</span><br><span class="line"># MAX GAPS: 25</span><br><span class="line">############### Statistics ###############</span><br><span class="line"># Number of collinear genes: 21371, Percentage: 56.25</span><br><span class="line"># Number of all genes: 37996</span><br><span class="line">##########################################</span><br><span class="line">## Alignment 0: score=352.0 e_value=1.2e-17 N=8 scaf014&amp;scaf014 plus</span><br><span class="line">477-  0:	bv37796	bv37889	  3e-06</span><br><span class="line">477-  1:	bv37820	bv37905	  6e-24</span><br><span class="line">477-  2:	bv37822	bv37926	  9e-29</span><br><span class="line">477-  3:	bv37826	bv37930	  4e-62</span><br><span class="line">477-  4:	bv37828	bv37931	  1e-74</span><br><span class="line">477-  5:	bv37829	bv37932	 3e-142</span><br><span class="line">477-  6:	bv37830	bv37933	  6e-78</span><br><span class="line">477-  7:	bv37832	bv37934	  3e-76</span><br><span class="line">## Alignment 1: score=300.0 e_value=6.1e-08 N=6 scaf014&amp;scaf014 minus</span><br><span class="line">478-  0:	bv35514	bv35525	      0</span><br><span class="line">478-  1:	bv35515	bv35524	      0</span><br><span class="line">478-  2:	bv35516	bv35523	  1e-81</span><br><span class="line">478-  3:	bv35517	bv35522	      0</span><br><span class="line">478-  4:	bv35518	bv35521	 1e-115</span><br><span class="line">478-  5:	bv35519	bv35520	      0</span><br><span class="line">... ...</span><br><span class="line">... ...</span><br></pre></td></tr></table></figure>

<h1 id="3-MCScanX下游分析"><a href="#3-MCScanX下游分析" class="headerlink" title="3. MCScanX下游分析"></a>3. MCScanX下游分析</h1><h2 id="3-1-计算共线性区块Ks"><a href="#3-1-计算共线性区块Ks" class="headerlink" title="3.1. 计算共线性区块Ks"></a>3.1. 计算共线性区块Ks</h2><p>tips:需要检查得到的结果中Ks值是否有负值或NA等无效数据，并过滤无效数据。</p>
<h3 id="3-1-1-方案一：MCScanX的downstream-analyses目录包含一些下游脚本，其中有计算kaks的脚本add-ka-and-ks-to-collinearity-pl"><a href="#3-1-1-方案一：MCScanX的downstream-analyses目录包含一些下游脚本，其中有计算kaks的脚本add-ka-and-ks-to-collinearity-pl" class="headerlink" title="3.1.1. 方案一：MCScanX的downstream_analyses目录包含一些下游脚本，其中有计算kaks的脚本add_ka_and_ks_to_collinearity.pl"></a>3.1.1. 方案一：MCScanX的downstream_analyses目录包含一些下游脚本，其中有计算kaks的脚本add_ka_and_ks_to_collinearity.pl</h3><p><code>add_ka_and_ks_to_collinearity.pl -i sample.collinearity -d sample.cds.fa -o sample.kaks &gt; out.log 2&gt;&amp;1</code> #注意算出的部分kaks值为-2的问题，未找到解决方案</p>
<h3 id="3-1-2-方案二：ParaAT-pl-KaKs-Calculator2-0"><a href="#3-1-2-方案二：ParaAT-pl-KaKs-Calculator2-0" class="headerlink" title="3.1.2. 方案二：ParaAT.pl+KaKs_Calculator2.0"></a>3.1.2. 方案二：ParaAT.pl+KaKs_Calculator2.0</h3><ul>
<li>ParaAT.pl用于根据同源基因对list生成比对的gene对cds序列，并可以指定输出格式，如axt格式；</li>
<li>KaKs_Calculator用于计算基因对的kaks。</li>
</ul>
<p>用还会用到两个脚本:</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/scbgfengchao/4DTv/blob/master/axt2one-line.py">axt2one-line.py</a>转换axt格式为单行</li>
<li><a target="_blank" rel="noopener" href="https://github.com/JinfengChen/Scripts/blob/master/FFgenome/03.evolution/distance_kaks_4dtv/bin/calculate_4DTV_correction.pl">calculate_4DTV_correction.pl</a>计算4dtv。</li>
</ul>
<ol>
<li>安装ParaAT.pl<br>参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/10/29/bioinfo_align.pep2cds/">blog.ParaAT</a></li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://download.big.ac.cn/bigd/tools/ParaAT2.0.tar.gz</span><br><span class="line">tar -zxf ParaAT2.0.tar.gz</span><br><span class="line">cd ParaAT2.0</span><br><span class="line">ParaAT.pl -h</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>安装KaKs_Calculator2.0<br><a target="_blank" rel="noopener" href="https://sourceforge.net/projects/kakscalculator2/?source=typ_redirect">KaKs_Calculator2.0下载地址</a></li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget https://altushost-swe.dl.sourceforge.net/project/kakscalculator2/KaKs_Calculator2.0.tar.gz</span><br><span class="line">tar -zxf KaKs_Calculator2.0.tar.gz</span><br><span class="line">chmod 777 ./KaKs_Calculator2.0/bin/Linux/KaKs_Calculator</span><br><span class="line">chmod 777 ./KaKs_Calculator2.0/src/AXTConvertor</span><br></pre></td></tr></table></figure>

<p>然后把.&#x2F;KaKs_Calculator2.0&#x2F;bin&#x2F;Linux&#x2F;和.&#x2F;KaKs_Calculator2.0&#x2F;src&#x2F;添加到环境变量即可使用KaKs_Calculator和AXTConvertor命令。</p>
<ol start="3">
<li><p>用ParaAT.pl获取共线性基因对比对序列</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cat sample.collinearity |grep -v &quot;^#&quot;|cut -f2,3 &gt;sample.homolog #提取blocks的同源gene对</span><br><span class="line">echo &quot;24&quot; &gt;proc #指定ParaAT.pl使用线程</span><br><span class="line">ParaAT.pl -h sample.homolog -n sample.cds.fa -a sample.pep.fa -m mafft -p proc -f axt -g -k -o sample.paraat 2&gt; paraat.log &amp; #用ParaAT.pl调用mafft做每对共线性基因的蛋白比对和蛋白转cds比对，输出axt格式。-g移除比对有gap的密码子；-o指定生成目录；加上-k参数可以在获得axt文件后自动调用KaKs_Calculator计算kaks值，使用MA模型，比YN模型慢，推荐手动用KaKs_Calculator的YN模型，生成sample.axt.kaks文件。</span><br></pre></td></tr></table></figure>
</li>
<li><p>用KaKs_Calculator手动计算共线性基因对的KaKs和4dtv值<br>ParaAT.pl的-k参数只能指定KaKs_Calculator的MA模型计算kaks值，如果需要指定其他的模型，则可以手动运行计算。</p>
</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">cd sample.paraat</span><br><span class="line">for i in `ls |grep &quot;axt&quot;`;do KaKs_Calculator -i $i -o $&#123;i&#125;.kaks -m YN;done #用YN模型计算每个gene对的KaKs，生成四列数据，gene对，Ka，Ks，Ka/Ks</span><br><span class="line">for i in `ls |grep &quot;kaks&quot;`;do awk &#x27;NR&gt;1&#123;print $1&quot;\t&quot;$3&quot;\t&quot;$4&quot;\t&quot;$5&#125;&#x27; $i &gt;&gt;../all.kaks;done #合并kaks到all.kaks文件</span><br><span class="line">- 计算4dtv</span><br><span class="line">for i in `ls |grep &quot;axt&quot;`;do axt2one-line.py $i $&#123;i&#125;.one-line;done #多行axt文件转换成单行</span><br><span class="line">ls |grep &quot;axt.one-line&quot;|while read id;do calculate_4DTV_correction.pl $id &gt; $&#123;id%%one-line&#125;4dtv;done #计算4dtv值，生成两列数据，gene对，4dtv</span><br><span class="line">for i in `ls |grep &quot;4dtv&quot;`;do awk &#x27;NR&gt;1&#123;print $1&quot;\t&quot;$3&#125;&#x27; $i &gt;&gt;../all.4dtv;done #合并4dtv值到all.4dtv</span><br><span class="line">cd ..</span><br><span class="line">join all.kaks all.4dtv |sed &quot;s/ /\t/g&quot; |awk &#x27;$3 != &quot;NA&quot; &#123;print $0&#125;&#x27; |sed &#x27;1i\genepair\tKa\tKs\tKa/Ks\t4dtv_corrected&#x27; &gt;all.results #以gene对为基准，join合并kaks和4dtv值到一个文件，然后过滤Ks值为NA的无效数据，添加标题行。</span><br><span class="line"></span><br><span class="line">rm all.kaks* all.4dtv* #删除中间文件</span><br></pre></td></tr></table></figure>

<h2 id="3-2-ggplot2画密度分布图和峰值"><a href="#3-2-ggplot2画密度分布图和峰值" class="headerlink" title="3.2. ggplot2画密度分布图和峰值"></a>3.2. ggplot2画密度分布图和峰值</h2><h3 id="3-2-1-ggplot2画密度分布图"><a href="#3-2-1-ggplot2画密度分布图" class="headerlink" title="3.2.1. ggplot2画密度分布图"></a>3.2.1. ggplot2画密度分布图</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>ggplot2<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>ggpmisc<span class="punctuation">)</span></span><br><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;all.results&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">)</span></span><br><span class="line">p <span class="operator">&lt;-</span> ggplot<span class="punctuation">(</span>data<span class="punctuation">,</span> aes<span class="punctuation">(</span>Ks<span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span> geom_density<span class="punctuation">(</span>size<span class="operator">=</span><span class="number">1</span><span class="punctuation">,</span>color<span class="operator">=</span><span class="string">&quot;black&quot;</span><span class="punctuation">)</span><span class="operator">+</span>xlab<span class="punctuation">(</span><span class="string">&quot;Synonymous substitution rate(Ks)&quot;</span><span class="punctuation">)</span><span class="operator">+</span>ylab<span class="punctuation">(</span><span class="string">&quot;Percent of Total Paralogs&quot;</span><span class="punctuation">)</span><span class="operator">+</span>theme_classic<span class="punctuation">(</span><span class="punctuation">)</span><span class="operator">+</span>scale_y_continuous<span class="punctuation">(</span>breaks<span class="operator">=</span>seq<span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span> <span class="number">2.5</span><span class="punctuation">,</span> <span class="number">0.2</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="operator">+</span>xlim<span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span><span class="number">3</span><span class="punctuation">)</span></span><br><span class="line">ggsave<span class="punctuation">(</span>file<span class="operator">=</span><span class="string">&quot;Ks.pdf&quot;</span><span class="punctuation">,</span>plot<span class="operator">=</span>p<span class="punctuation">,</span>width<span class="operator">=</span><span class="number">10</span><span class="punctuation">,</span>height<span class="operator">=</span><span class="number">5</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h3 id="3-2-2-ggplot2峰值标定"><a href="#3-2-2-ggplot2峰值标定" class="headerlink" title="3.2.2. ggplot2峰值标定"></a>3.2.2. ggplot2峰值标定</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pb <span class="operator">&lt;-</span> ggplot_build<span class="punctuation">(</span>p<span class="punctuation">)</span></span><br><span class="line">pic <span class="operator">&lt;-</span> p <span class="operator">+</span> stat_peaks<span class="punctuation">(</span>data <span class="operator">=</span> pb<span class="punctuation">[[</span><span class="string">&#x27;data&#x27;</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">[[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="punctuation">,</span> aes<span class="punctuation">(</span>x<span class="operator">=</span>x<span class="punctuation">,</span> y<span class="operator">=</span>density<span class="punctuation">)</span><span class="punctuation">,</span> geom<span class="operator">=</span> <span class="string">&#x27;text&#x27;</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&quot;red&quot;</span><span class="punctuation">,</span> <span class="punctuation">,</span>vjust<span class="operator">=</span><span class="operator">-</span><span class="number">0.5</span><span class="punctuation">)</span></span><br><span class="line">ggsave<span class="punctuation">(</span>file<span class="operator">=</span><span class="string">&quot;Ks_peaks.pdf&quot;</span><span class="punctuation">,</span>plot<span class="operator">=</span>pic<span class="punctuation">,</span>width<span class="operator">=</span><span class="number">10</span><span class="punctuation">,</span>height<span class="operator">=</span><span class="number">5</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p>在密度分布图里得到红色标记的峰值。</p>
<h3 id="3-2-3-多个密度图"><a href="#3-2-3-多个密度图" class="headerlink" title="3.2.3. 多个密度图"></a>3.2.3. 多个密度图</h3><p>同时展示多组Ks数据分布，可以把数据合并，添加一列作为分类标签，通过colour颜色参数指定这个分类标签列，则可以实现一张图上展示多个密度图。</p>
<p><code>ggplot(data)+geom_density(aes(x=V3,colour=V5),adjust=2)+xlim(0,0.5)+theme_classic()</code> #指定第三列V3为数据，第五列V5为分类依据并赋予不同颜色。</p>
<h3 id="3-2-4-通过密度分布图判断WGD"><a href="#3-2-4-通过密度分布图判断WGD" class="headerlink" title="3.2.4. 通过密度分布图判断WGD"></a>3.2.4. 通过密度分布图判断WGD</h3><p>一般认为，Ks密度分布图如果有明显的峰，一个峰代表对应一次WGD事件。<br>通过峰的数量和对应的Ks大小可以判断WGD事件的次数和发生时间。</p>
<h2 id="3-3-WGD发生的时间"><a href="#3-3-WGD发生的时间" class="headerlink" title="3.3. WGD发生的时间"></a>3.3. WGD发生的时间</h2><p>通过Ks的密度分布图鉴定得到WGD事件发生的证据之后，有多种方法可以估算WGD发生的时间。</p>
<h3 id="3-3-1-通过Ks-x3D-2μT计算时间"><a href="#3-3-1-通过Ks-x3D-2μT计算时间" class="headerlink" title="3.3.1. 通过Ks&#x3D;2μT计算时间"></a>3.3.1. 通过Ks&#x3D;2μT计算时间</h3><p>依据公式Ks&#x3D;2μT来计算时间T，依赖准确的进化速率μ，μ通常引用近缘类群的已有权威研究。</p>
<p>进化速率μ：</p>
<ul>
<li>蕨类植物的同义突变率：4.79e-9 subst.&#x2F;syn. site&#x2F;year :the first estimate of fern nuclear genome evolutionary rates with polypodiaceous nuclear genomes (Barker 2009)</li>
<li>水稻：2.5e-8 subst.&#x2F;syn. site&#x2F;year</li>
</ul>
<h3 id="3-3-2-通过进化树估算时间"><a href="#3-3-2-通过进化树估算时间" class="headerlink" title="3.3.2. 通过进化树估算时间"></a>3.3.2. 通过进化树估算时间</h3><ul>
<li>用代表WGD的Ks值附近的同线性基因对制作两套样品的单倍型数据，然后选取一到多个近缘种的蛋白序列与其中一套单倍型做直系同源基因分析，用找到的直系同源单拷贝基因建树（把两套单倍型数据当作两个物种来建树），通过化石或者二次标定的方法用paml的mcmctree模块计算两套单倍型的分化时间，即为WGD发生的时间。</li>
<li>如果物种分化时间的尺度太大而不能准确估算WGD时间，可以先用大尺度的系统发育树做近缘种和目标物种的分化时间的估算，然后在做WGD时间估算时只用一个近缘种加上近缘种与目标物种的分化时间标定来获得更准确的WGD时间估计结果。</li>
</ul>
<h4 id="3-3-2-1-获取sample发生WGD的两套基因"><a href="#3-3-2-1-获取sample发生WGD的两套基因" class="headerlink" title="3.3.2.1. 获取sample发生WGD的两套基因"></a>3.3.2.1. 获取sample发生WGD的两套基因</h4><ol>
<li>前面分析中计算Ks的分布峰值假设在<strong>0.05</strong>，取Ks值在0.05正负25%的范围，即[0.0375-0.0625]之间的基因对。<br><code>cat all.results |awk &#39;$3&gt;0.0375 &amp;&amp; $3&gt;0.0625 &#123;print $1&#125; |sed &quot;s/-/\t/g&quot; &gt;wgd.homologs</code></li>
<li>基因对中若有一个基因对多个基因的情况，则选取Ks最接近0.05的那对。</li>
<li>获取两套基因list。<br><code>cut -f1 wgd.homologs &gt;wgd_H1.homologs</code>;<br><code>cut -f2 wgd.homologs &gt;wgd_H2.homologs</code></li>
</ol>
<h4 id="3-3-2-2-选近缘种和分化时间"><a href="#3-3-2-2-选近缘种和分化时间" class="headerlink" title="3.3.2.2. 选近缘种和分化时间"></a>3.3.2.2. 选近缘种和分化时间</h4><ol>
<li>参考近年发表的权威组学系统发育文章选取近缘种和物种分化时间数据。</li>
<li>这里假设选取了两个近缘物种sampleA和sampleB。sample发生WGD形成的两套单倍型H1和H2，目标物种sample与近缘种sampleA的分化时间已知为88Ma。</li>
<li>确定物种树拓扑结构：<code>(((sampleH1,sampleH2),sampleA)&#39;&lt;0.88&gt;0.88),sampleB;</code>，以sampleB为外类群。</li>
</ol>
<h4 id="3-3-2-3-orthofinder找直系同源基因"><a href="#3-3-2-3-orthofinder找直系同源基因" class="headerlink" title="3.3.2.3. orthofinder找直系同源基因"></a>3.3.2.3. orthofinder找直系同源基因</h4><ol>
<li>根据基因对提取蛋白序列，任意用一个（这里用wgd_H1.homologs)提取蛋白序列：<code>seqkit grep -n -f wgd_H1.homologs sample.pep &gt; wgd_H1.pep</code></li>
<li>与其他两个近缘种的蛋白序列一起，用orthofinder找直系同源基因。</li>
</ol>
<ul>
<li>orthofinder使用方法：把所有物种的pep蛋白序列（两个近缘物种加上wgd_H1.pep）放进一个文件夹(directory)，注意蛋白序列中不能有.&#x2F;*等符号（常用来代表终止密码子）。</li>
<li>运行<code>orthofinder -f directory -t 24</code> # -t指定线程</li>
<li>结果文件的Orthogroups目录下，Orthogroups.txt包含所有直系同源基因，Orthogroups_SingleCopyOrthologues.txt包含单拷贝的直系同源基因。</li>
<li>根据Orthogroups_SingleCopyOrthologues.txt把Orthogroups.txt里基因信息提取出来,<code>grep -f Orthogroups_SingleCopyOrthologues.txt Orthogroups.tsv &gt;OG.txt</code>。OG.txt文件内容是singlecopyorthologues列表，第一列orthogroupID，后面3列每套蛋白序列ID，共4列，这个文件的每行数据为一个homologs对应的每套数据的蛋白ID。</li>
<li>合并wgd.homologs和OG.txt文件为OG.list：<code>join -1 1 -2 2 wgd.homologs OG.txt &gt; OG.list</code>。OG.list文件是在OG.txt基础上增加了wgd_H2.homologs的一套蛋白ID数据，共有四套蛋白ID数据。</li>
</ul>
<h4 id="3-3-2-4-建单基因树"><a href="#3-3-2-4-建单基因树" class="headerlink" title="3.3.2.4. 建单基因树"></a>3.3.2.4. 建单基因树</h4><ol>
<li>建树脚本<br>用下面的脚本singlegenetree.sh，提取每个homologs序列，并通过raxml-ng（raxmlHPC建氨基酸树只有bootstrap结果，没有bestTree）为每个homologs基因比对（mafft）和建树。<br>OG.list文件包含第一列ogID和后4列4套样本ID。</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">singlegenetree.sh</span></span><br><span class="line">cat ./OG.list | while read line</span><br><span class="line">do</span><br><span class="line">        sample_id=$(echo $line |awk &#x27;&#123;print $1&#125;&#x27;)</span><br><span class="line">        sample_a=$(echo $line |awk &#x27;&#123;print $2&#125;&#x27;)</span><br><span class="line">        sample_b=$(echo $line |awk &#x27;&#123;print $3&#125;&#x27;)</span><br><span class="line">        sample_c=$(echo $line |awk &#x27;&#123;print $4&#125;&#x27;)        </span><br><span class="line">        sample_d=$(echo $line |awk &#x27;&#123;print $5&#125;&#x27;)</span><br><span class="line">        mkdir ./singlegenetree/$&#123;sample_id&#125;</span><br><span class="line">        seqkit grep -n $&#123;sample_a&#125; sample.pep &gt;./singlegenetree/$&#123;sample_id&#125;/H1_$&#123;sample_a&#125;.pep</span><br><span class="line">        seqkit grep -n $&#123;sample_b&#125; sample.pep &gt;./singlegenetree/$&#123;sample_id&#125;/H2_$&#123;sample_b&#125;.pep</span><br><span class="line">        seqkit grep -n $&#123;sample_c&#125; sampleA.pep &gt;./singlegenetree/$&#123;sample_id&#125;/$&#123;sample_c&#125;.pep</span><br><span class="line">        seqkit grep -n $&#123;sample_d&#125; sampleB.pep &gt;./singlegenetree/$&#123;sample_id&#125;/$&#123;sample_d&#125;.pep</span><br><span class="line">        cd ./singlegenetree/$&#123;sample_id&#125;/</span><br><span class="line">        cat H1_$&#123;sample_a&#125;.pep H2_$&#123;sample_b&#125;.pep $&#123;sample_c&#125;.pep $&#123;sample_d&#125;.pep &gt; $&#123;sample_id&#125;.pep</span><br><span class="line">        mafft $&#123;sample_id&#125;.pep &gt;$&#123;sample_id&#125;.mafft.pep</span><br><span class="line">        raxml-ng --all --msa $&#123;sample_id&#125;.mafft.pep --data-type AA --threads 8 --bs-trees 100 --model LG+G8+F --tree pars&#123;10&#125; --outgroup $&#123;sample_d&#125; --redo</span><br><span class="line">        sed -E &quot;s/:[0-9]\.[0-9]*//g&quot; $&#123;sample_id&#125;.mafft.pep.raxml.bestTree  &gt;besttree.$&#123;sample_id&#125; #删除不必要的枝长等信息</span><br><span class="line">        cd ../../</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>筛选homologs<br>判断生成的每个OG的树文件besttree.${sample_id}，只保留符合物种树拓扑结构<code>(((sampleH1,sampleH2),sampleA)),sampleB;</code>的基因，保存成OG.filtered。</li>
</ol>
<h4 id="3-3-2-5-paml估计分化时间"><a href="#3-3-2-5-paml估计分化时间" class="headerlink" title="3.3.2.5. paml估计分化时间"></a>3.3.2.5. paml估计分化时间</h4><p>此节参考博客<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/03/25/bioinfo_caculate.divergence.time/">估算系统树分歧时间</a>的paml部分。</p>
<ol>
<li>在OG.filtered最后一列添加每个OG的序列长度信息，保存成OG.filtered.list。</li>
<li>准备输入文件sample.phy<br>上一步脚本提取的每个homologs的比对后的四套数据蛋白序列${sample_id}.mafft.pep，用脚本fa2phylip.sh换成phylip格式（用python的SeqIO库换的格式mcmctree识别不了）。<br>并合并提取的所有homologs序列。</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">fa2phylip.sh</span></span><br><span class="line">cat ./OG.filtered.list | while read line</span><br><span class="line">do</span><br><span class="line">        sample_id=$(echo $line |awk &#x27;&#123;print $1&#125;&#x27;) #获取OG.ID</span><br><span class="line">        sample_a=$(echo $line |awk &#x27;&#123;print $6&#125;&#x27;) #获取序列长度</span><br><span class="line">        sed &quot;s/_.*//g&quot; ../singlegenetree/$&#123;sample_id&#125;/$&#123;sample_id&#125;.mafft.pep|seqkit seq -w 0|sed -E &quot;:a;N;s/\n/ /g;ta&quot; |sed &quot;s/ &gt;/\n/g&quot; |sed &quot;s/&gt;//g&quot;|sed &quot;s/ /  /g&quot;|sed &quot;1i\5  $&#123;sample_a&#125;&quot; |sed &#x27;1i\ &#x27; &gt; ./phy/$&#123;sample_id&#125;.phy #把上一步获取的$&#123;sample_id&#125;.mafft.pep改为phylip格式，并在首行添加空格行（为了合并后每个OG用空行隔开）</span><br><span class="line"></span><br><span class="line">done</span><br><span class="line">cat ./phy/*phy &gt;sample.phy #合并所有phy文件为一个sample.phy文件，假设共有261个子phy文件被合并</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>准备输入文件sample.tre</li>
</ol>
<ul>
<li>sample.tre包含两行，第一行表述树中有5个样本，共计1个树，两个数值之间用空格分割；<br>第二行则是Newick格式树信息。</li>
<li>其中第二行Newick格式树包含有校准点信息，校准点信息一般指95%HPD（Highest Posterior Density）对应的置信区间，这里用的校准点时间只有一个点，所以写成’&lt;0.88&gt;0.88’形式；校准点单位是100MYA（软件说明文档中使用该单位，也推荐使用该单位，若使用其它单位，后续配置文件中的相关参数也需要对应修改）。</li>
<li>此外，Newick格式的树尾部一定要有分号，没有的话程序可能不能正常运行。</li>
<li>sample.tre内容为：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">4 1</span><br><span class="line">(((sampleH1,sampleH2),sampleA)&#x27;&lt;0.88&gt;0.88),sampleB;</span><br></pre></td></tr></table></figure>

<ol start="4">
<li>准备配置文件mcmctree3.ctl<br>ndata &#x3D; 261表示有261个数据，与sample.phy包含的子phy数量对应；seqtype &#x3D; 2表示这里用的氨基酸序列；model &#x3D; 0暂时选用0，后面要改；</li>
</ol>
<figure class="highlight plaintext"><figcaption><span>mcmctree3.ctl</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line">          seed = -1</span><br><span class="line">       seqfile = sample.phy</span><br><span class="line">      treefile = sample.tre</span><br><span class="line">       outfile = mcmc.out</span><br><span class="line">      mcmcfile = mcmc.txt</span><br><span class="line"></span><br><span class="line">       seqtype = 2    *0: DNA data; 1: codon data; 2: AAs data</span><br><span class="line">        ndata = 261</span><br><span class="line">       usedata = 3    * 0: no data; 1:seq like; 2:normal approximation</span><br><span class="line">         clock = 3    * 1: global clock; 2: independent rates; 3: correlated rates</span><br><span class="line">*      TipDate = 1 100</span><br><span class="line">        RootAge = &#x27;&lt;10&#x27;  * constraint on root age, used if no fossil for root.</span><br><span class="line"></span><br><span class="line">         model = 0    * 0:JC69, 1:K80, 2:F81, 3:F84, 4:HKY85</span><br><span class="line">         alpha = 0.5   * alpha for gamma rates at sites</span><br><span class="line">         ncatG = 4    * No. categories in discrete gamma</span><br><span class="line"></span><br><span class="line">     cleandata = 0    * remove sites with ambiguity data (1:yes, 0:no)?</span><br><span class="line"></span><br><span class="line">       BDparas = 1 1 0.1   * birth, death, sampling</span><br><span class="line">   kappa_gamma = 6 2      * gamma prior for kappa</span><br><span class="line">   alpha_gamma = 1 1      * gamma prior for alpha</span><br><span class="line"></span><br><span class="line">   rgene_gamma = 2 20 1   * gamma prior for rate for genes</span><br><span class="line">  sigma2_gamma = 1 10 1    * gamma prior for sigma^2     (for clock=2 or 3)</span><br><span class="line"></span><br><span class="line">      finetune = 1: .1 .1 .1 .1 .1 .1  * times, rates, mixing, paras, RateParas</span><br><span class="line"></span><br><span class="line">         print = 1</span><br><span class="line">        burnin = 1000000</span><br><span class="line">      sampfreq = 10</span><br><span class="line">       nsample = 500000</span><br><span class="line"></span><br><span class="line">*** Note: Make your window wider (100 columns) when running this program.</span><br></pre></td></tr></table></figure>

<ol start="5">
<li>运行<code>mcmctree mcmctree3.ctl</code></li>
</ol>
<ul>
<li>此步骤的目的是为了获取in.BV文件；</li>
<li>用命令<code>mcmctree mcmctree3.ctl</code>运行起来后，ctrl+C多次中断运行（共261个序列，会依次调用261次codeml，中断后会自动进入下一次调用，目的是完成261次调用生成261个序列的临时文件）；由于单个序列短，很快会完成分析，也可以不ctrl+C中断，等待完成，只要生成261套tmp*文件即可。</li>
<li>删除rst2文件和mcmc.out,mcmc.txt,out.BV文件；把paml软件下的wag.dat文件复制到当前目录下<code>cp ~/software/anaconda3/dat/wag.dat ./</code>；</li>
<li>用命令<code>sed -i -e &quot;s/model = 0/model =2\naaRatefile = wag.dat/&quot; -e &quot;s/method = 0/method = 1/&quot; tmp*.ctl</code>把所有tmp*.ctl文件的model修改成model &#x3D; 2，method修改成method &#x3D; 1,并添加一行aaRatefile &#x3D; wag.dat；</li>
<li>用命令<code>for i in $(ls tmp*.ctl);do codeml $&#123;i&#125; ; mv rst2 $&#123;i&#125;.rst2; done</code>调用codeml进行261次分析（每次codeml都生成rst2文件，所以每次生成后把rst2改名，否则rst2被覆盖）；</li>
<li>然后<code>cat *rst2 &gt;in.BV</code>合并所有rst2文件到in.BV文件；</li>
</ul>
<ol start="6">
<li>运行<code>mcmctree mcmctree2.ctl</code></li>
</ol>
<ul>
<li>新建目录并复制四个文件sample.phy,sample.tre,mcmctree3.ctl,in.BV到新目录<code>mkdir mcmctree2 &amp;&amp; cd mcmctree2 &amp;&amp; cp ../Ane.* ./ &amp;&amp; cp ../mcmctree3.ctl ./ &amp;&amp; cp ../in.BV ./</code>;</li>
<li>把mcmctree3.ctl文件内的usedata &#x3D; 3改为usedata &#x3D; 2，并重命名成mcmctree2.ctl；</li>
<li>然后运行<code>mcmctree mcmctree2.ctl</code>；</li>
<li>等待两三天获取结果，FigTree.tre文件包含了所有节点的95%HPD的时间，是nexus格式。</li>
<li>根据FigTree.tre结果：<code>((sampleA: 0.408365, (sampleH1: 0.107333, sampleH2: 0.107333) [&amp;95%=&#123;0.0749432, 0.126365&#125;]: 0.301032) [&amp;95%=&#123;0.285365, 0.478163&#125;]: 0.362000, sampleB: 0.770366) [&amp;95%=&#123;0.53864, 0.90012&#125;];</code>的sampleH1和sampleH2分化的时间，可以看出WGD发生的时间在**10.7Ma[7.5-12.6]**之间。</li>
</ul>
<h1 id="4-references"><a href="#4-references" class="headerlink" title="4. references"></a>4. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3326336/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3326336/</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/wyp1125/MCScanX">https://github.com/wyp1125/MCScanX</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/27/bioinfo_WGD/" data-id="cl1aiqcrd0020m4vkbbvy93bx" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/paml/" rel="tag">paml</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bio_evolutionary.tree" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/20/bio_evolutionary.tree/" class="article-date">
  <time datetime="2021-11-20T03:50:00.000Z" itemprop="datePublished">2021-11-20</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/evolution/">evolution</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/20/bio_evolutionary.tree/">进化树相关知识</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=26657608&auto=1&height=32"></iframe></div>

<h1 id="1-进化树"><a href="#1-进化树" class="headerlink" title="1. 进化树"></a>1. 进化树</h1><p>系统树(phylogenetic tree&#x2F;phylogeny)，又叫进化树(evolutionary tree)，是基于形态或遗传特征差异展示物种或其他实体间的进化关系的分支图或树图。</p>
<h1 id="2-进化树分类"><a href="#2-进化树分类" class="headerlink" title="2. 进化树分类"></a>2. 进化树分类</h1><h2 id="2-1-根据有无祖先根分类"><a href="#2-1-根据有无祖先根分类" class="headerlink" title="2.1. 根据有无祖先根分类"></a>2.1. 根据有无祖先根分类</h2><ol>
<li>有根树(rooted tree)<br> 根节点是所有其他节点的父节点，每个带有后代的节点代表这些后代的最近共同祖先。在有根树中，只有根节点的度数（指传入和传出边的总数）是2，其他节点的最小度数都为3。使树生根的常见方式是指定无争议的外类群。</li>
<li>无根树(unrooted tree)<br> 仅说明叶节点的相关性，无需推断祖先。</li>
</ol>
<h2 id="2-2-根据分歧数量分类"><a href="#2-2-根据分歧数量分类" class="headerlink" title="2.2. 根据分歧数量分类"></a>2.2. 根据分歧数量分类</h2><p>有根树和无根树都可以是二叉的，也可以是多叉的。</p>
<ol>
<li>二叉树(bifurcated tree)<br> 有根二叉树每个内部节点都有两个后代，无根二叉树每个内部节点有三个邻近的自由树。</li>
<li>多叉树(multifurcated tree)<br> 有根多叉树在某些内部节点可能有两个以上后代，无根多叉树在某些内部节点可能有三个以上邻近自由树。</li>
</ol>
<h2 id="2-3-根据有无标签"><a href="#2-3-根据有无标签" class="headerlink" title="2.3. 根据有无标签"></a>2.3. 根据有无标签</h2><ol>
<li>标签树(labeled tree)<br> 标签树在叶节点有特定的值。</li>
<li>非标签树(unlabeled tree)<br> 只有树型。</li>
</ol>
<h2 id="2-4-特殊的树的类型"><a href="#2-4-特殊的树的类型" class="headerlink" title="2.4. 特殊的树的类型"></a>2.4. 特殊的树的类型</h2><ol>
<li>树状图(dendrogram)<br> 树状图的总称，无论有无系统发生关系。</li>
<li>分支树(cladogram)<br>只有分支模式，枝长不包含变化量信息，中间节点也不代表祖先。</li>
<li>系统树(phylogram)<br>枝长与变化量&#x2F;碱基替换数成比例。</li>
<li>超度量树&#x2F;时序树(chronogram)<br> 枝长与时长成比例。</li>
<li>速率树(ratogram)<br> 枝长表示替换速率。</li>
<li>达尔格伦树(dahlgrenogram)<br> 系统发育树的部分样品的图。</li>
<li>系统发育网(phylogenetic network)<br> 有向无环图。</li>
<li>罗马树(romerogram)&#x2F;纺锤树(spindle diagram)&#x2F;气泡树(bubble diagram)<br> 横向宽度代表生物的分类学多样性，纵坐标是地质时间。为了反映多个类群的丰度随着时间的变化。由于不适合处理并系类群，被提议不再使用。</li>
<li>生命珊瑚(coral of life)<br> 达尔文提出珊瑚可能比树更适合描绘生命的进化。</li>
</ol>
<h1 id="3-读进化树"><a href="#3-读进化树" class="headerlink" title="3. 读进化树"></a>3. 读进化树</h1><h2 id="3-1-根据树型"><a href="#3-1-根据树型" class="headerlink" title="3.1. 根据树型"></a>3.1. 根据树型</h2><ul>
<li>当树的所有叶节点没有对齐的时候，枝长可能代表变化量&#x2F;碱基替换数(phylogram)&#x2F;替换速率(ratogram)；</li>
<li>当树的所有叶节点对齐的时候，枝长如果一致则无信息(cladogram)，枝长如果不一致可能代表进化时长(chronogram)。</li>
</ul>
<h1 id="4-reference"><a href="#4-reference" class="headerlink" title="4. reference"></a>4. reference</h1><ol>
<li><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Phylogenetic_tree">wiki:Phylogenetic_tree</a></li>
<li><a target="_blank" rel="noopener" href="https://www.nature.com/scitable/topicpage/reading-a-phylogenetic-tree-the-meaning-of-41956/">https://www.nature.com/scitable/topicpage/reading-a-phylogenetic-tree-the-meaning-of-41956/</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/20/bio_evolutionary.tree/" data-id="cl1aiqcr20017m4vkelg08frw" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-omics_transcriptome.RNA-seq" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/19/omics_transcriptome.RNA-seq/" class="article-date">
  <time datetime="2021-11-19T12:50:00.000Z" itemprop="datePublished">2021-11-19</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/transcriptome/">transcriptome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/19/omics_transcriptome.RNA-seq/">RNA-sequencing数据分析工具比较</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=176028&auto=1&height=32"></iframe></div>

<h1 id="1-background"><a href="#1-background" class="headerlink" title="1. background"></a>1. background</h1><p>2017年nature communications上发表的Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis，对RNA-sequencing(RNA-seq)数据的39种分析工具做了比较和总结。</p>
<p>RNA-seq分析流程主要有四大模块：</p>
<ul>
<li>RNA-seq变异分析(Genomic variants)</li>
<li>短读长数据的亚型检测(Short-read isoform detection)</li>
<li>长读长数据的亚型jiance(Long-read isoform detection)</li>
<li>表达分析(Expression analysis)</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp" title="RNACocktail分析协议" width="90%" />

<p><strong><p align="center">Figure 1. RNACocktail分析协议</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="2-比较结果总结"><a href="#2-比较结果总结" class="headerlink" title="2. 比较结果总结"></a>2. 比较结果总结</h1><ol>
<li>比对工具：<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</li>
<li>有参组装工具：<strong>StringTie</strong>在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</li>
<li>从头组装工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；<strong>Oases</strong>鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</li>
<li>三代测序错误纠正工具：<strong>LoRDEC</strong>在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</li>
<li>全长转录本亚型检测工具：注重质量选GMAP，注重速度选<strong>STARlong</strong>。</li>
<li>转录本的定量：<strong>Salmon-SMEM</strong>(不经过比对)运行速度和表现都更好；StringTie(基于基因组)的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>差异表达的工具：<strong>DESeq2</strong>在各项得分中均优于其他的工具。</li>
<li>检测基因组和转录组的突变：<strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</li>
</ol>
<p>作者总结了每一步的高精度工具，作为RNA-seq分析工具选择的一般建议。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig8_HTML.jpg?as=webp" title="RNACocktail计算流程" width="90%" />

<p><strong><p align="center">Figure 2. RNACocktail计算流程</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="3-比较结果"><a href="#3-比较结果" class="headerlink" title="3. 比较结果"></a>3. 比较结果</h1><h2 id="3-1-比对工具"><a href="#3-1-比对工具" class="headerlink" title="3.1. 比对工具"></a>3.1. 比对工具</h2><h3 id="3-1-1-比对工具用法"><a href="#3-1-1-比对工具用法" class="headerlink" title="3.1.1. 比对工具用法"></a>3.1.1. 比对工具用法</h3><p>RNA-seq分析的第一步通常是转录本的鉴定，需要把RNA-seq reads比对到合适的参考序列上。</p>
<p>如果用基因组作为参考序列可以检测到新的转录本，但可能需要耗费更多的计算资源；如果用转录组作为参考则无法找出新的转录本，但速度更快。如果研究物种没有可靠的参考序列，可以重头组装对转录本进行鉴定。</p>
<p>比对工具可以把RNA-seq short reads往参考基因组上进行mapping，做比对(alignment)，进行剪切点预测(junction prediction)和exon-intron边界的预测。</p>
<h3 id="3-1-2-比对结果"><a href="#3-1-2-比对结果" class="headerlink" title="3.1.2. 比对结果"></a>3.1.2. 比对结果</h3><p>比较了三款比对主流软件，<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</p>
<ul>
<li><strong>HISAT2</strong>在所有样品中能检测到的剪接位点验证率最高，但找到的剪接位点总数则比TopHat和STAR都要少。在运行速度方面，HISAT2则有较大的优势，比STAR快大约2.5倍并且比TopHat快大约100倍。</li>
<li><strong>STAR</strong>对于成对reads的唯一比对表现则比较好，特别是对于有较长读长的MCF7-300细胞系的数据，不过STAR会有更多含有soft-clipped和碱基错配的较低质量比对情况。</li>
<li>如果单看有soft-clipped的reads，<strong>TopHat</strong>会把这部分reads全部舍弃。</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig2_HTML.jpg?as=webp" title="比对工具比较结果" width="90%" />

<p><strong><p align="center">Figure 3. 比对工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-2-有参组装工具"><a href="#3-2-有参组装工具" class="headerlink" title="3.2. 有参组装工具"></a>3.2. 有参组装工具</h2><p>比对完成后，接着组装转录本。比较了三款组装工具，StringTie在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</p>
<ul>
<li>转录本数量：StringTie&gt;Cufflinks~IDP</li>
<li>基因水平准确性和灵敏度：IDP&gt;Cufflinks&gt;StringTie</li>
<li>转录本水平准确性：StringTie&gt;IDP&gt;Cufflink</li>
<li>运行速度：StringTie&gt;Cufflinks~IDP</li>
<li>IDP则会忽略单外显子转录本</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig3_HTML.jpg?as=webp" title="有参组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 4. 有参组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-3-从头组装工具"><a href="#3-3-从头组装工具" class="headerlink" title="3.3. 从头组装工具"></a>3.3. 从头组装工具</h2><p>比较了三款从头组装转录本的工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；Oases鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</p>
<ul>
<li>Trinity：组装的转录本较长，灵敏度高。</li>
<li>SOAPdenovo-Trans：组装到转录本数量多，准确性最高，对于高表达的转录本有明显的组装偏好性；花费的内存和CPU最低。</li>
<li>Oases：在所有样品中都有较好的N10-N50长度的表现，鉴定长转录本有优势；能够较好地涵盖到低表达的基因。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig4_HTML.jpg?as=webp" title="从头组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 5. 从头组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-4-三代测序错误纠正工具"><a href="#3-4-三代测序错误纠正工具" class="headerlink" title="3.4. 三代测序错误纠正工具"></a>3.4. 三代测序错误纠正工具</h2><p>LoRDEC在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</p>
<ul>
<li>LoRDEC：纠正后的reads质量更高，运行速度比LSC快100倍。</li>
<li>LSC：纠正后的reads比对率提高更多。</li>
</ul>
<h2 id="3-5-全长转录本亚型检测工具"><a href="#3-5-全长转录本亚型检测工具" class="headerlink" title="3.5. 全长转录本亚型检测工具"></a>3.5. 全长转录本亚型检测工具</h2><p>注重质量选GMAP，注重速度选STARlong。</p>
<ul>
<li>GMAP：比对到参考序列的reads数量更多。</li>
<li>STARlong：运行速度快68倍。</li>
</ul>
<h2 id="3-6-转录本的定量"><a href="#3-6-转录本的定量" class="headerlink" title="3.6. 转录本的定量"></a>3.6. 转录本的定量</h2><h3 id="3-6-1-转录本定量软件"><a href="#3-6-1-转录本定量软件" class="headerlink" title="3.6.1. 转录本定量软件"></a>3.6.1. 转录本定量软件</h3><ol>
<li>基于比对的转录本定量</li>
</ol>
<ul>
<li>传统方法是将read比对(spliced -aligned)到参考基因组，然后利用Cufflinks和StringTie进行转录本组装，最后进行定量。</li>
<li>如果具有参考转录本序列，reads可以直接跟转录本序列比对(aligned)，然后使用Salmon-Aln和eXpress进行定量。</li>
</ul>
<ol start="2">
<li>不经过比对(alignment-free)的转录本定量</li>
</ol>
<ul>
<li>主要提供了四个工具：Sailfish、Salmon-SMEM、quasi-mapping和kallisto。</li>
</ul>
<ol start="3">
<li>基于长读长技术的IDP（使用不同的短读长和长读长比对工具）</li>
</ol>
<h3 id="3-6-2-比较结果"><a href="#3-6-2-比较结果" class="headerlink" title="3.6.2. 比较结果"></a>3.6.2. 比较结果</h3><ul>
<li>StringTie的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>基于转录组比对的两款工具则和Salmon-SMEM的相关性更高，考虑到Salmon-SMEM更快的运行速度，eXpress和Salmon-Aln可能使用体验不如Salmon-SMEM。</li>
<li>kallisto和Salmon-SMEM表现最好。</li>
<li>基于比对基因组的工具则能够看到，使用HISAT2或者TopHat对具有较长读长的样品进行比对所得到的定量结果要比STAR要好。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig5_HTML.jpg?as=webp" title="转录本表达定量比较结果" width="90%" />

<p><strong><p align="center">Figure 6. 转录本表达定量比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-7-差异表达的工具"><a href="#3-7-差异表达的工具" class="headerlink" title="3.7. 差异表达的工具"></a>3.7. 差异表达的工具</h2><p>比较了基于计数的工具（DESeq2、limma和edgeR）和基于组装的工具（Cuffdiff和Ballgown）。</p>
<p><strong>DESeq2</strong>在各项得分中均优于其他的工具，而Cuffdiff和Ballgown的表现则相对比较差。<br>总的来说，基于计数的工具要比基于组装的工具效果要好；另外从运行时间上看的话，Cuffdiff也是最慢的工具。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig6_HTML.jpg?as=webp" title="差异表达比较结果" width="90%" />

<p><strong><p align="center">Figure 7. 差异表达比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-8-检测基因组和转录组的突变"><a href="#3-8-检测基因组和转录组的突变" class="headerlink" title="3.8. 检测基因组和转录组的突变"></a>3.8. 检测基因组和转录组的突变</h2><p>两款突变检测的工具：GATK HaplotypeCaller和Samtools mpileup。</p>
<p><strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig7_HTML.jpg?as=webp" title="突变检测比较结果" width="90%" />

<p><strong><p align="center">Figure 8. 突变检测比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="4-39个RNA-seq分析工具版本号、重要参数及下载地址"><a href="#4-39个RNA-seq分析工具版本号、重要参数及下载地址" class="headerlink" title="4. 39个RNA-seq分析工具版本号、重要参数及下载地址"></a>4. 39个RNA-seq分析工具版本号、重要参数及下载地址</h1><h2 id="4-1-比对工具-——-Reference-based-transcript-identification"><a href="#4-1-比对工具-——-Reference-based-transcript-identification" class="headerlink" title="4.1. 比对工具 —— Reference-based transcript identification"></a>4.1. 比对工具 —— Reference-based transcript identification</h2><ol>
<li>TopHat2： –no-coverage-search<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/index.shtml">http://ccb.jhu.edu/software/tophat/index.shtml</a></li>
<li>STAR: -twopassMode Basic –outFilterType BySJout<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
<li>HISAT2 2.0.1-beta –dta (or –dta-cufflinks)<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/hisat/index.shtml">http://www.ccb.jhu.edu/software/hisat/index.shtml</a></li>
<li>RASER 0.52 -b 0.03<br><a target="_blank" rel="noopener" href="https://www.ibp.ucla.edu/research/xiao/RASER.html">https://www.ibp.ucla.edu/research/xiao/RASER.html</a></li>
</ol>
<h2 id="4-2-有参考转录本组装工具"><a href="#4-2-有参考转录本组装工具" class="headerlink" title="4.2. 有参考转录本组装工具"></a>4.2. 有参考转录本组装工具</h2><ol>
<li>Cufflinks 2.2.1 –frag-bias-correct<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>StringTie 1.2.1 -v -B<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/stringtie/">http://www.ccb.jhu.edu/software/stringtie/</a></li>
</ol>
<h2 id="4-3-无参考转录本组装工具"><a href="#4-3-无参考转录本组装工具" class="headerlink" title="4.3. 无参考转录本组装工具"></a>4.3. 无参考转录本组装工具</h2><ol>
<li>SOAPdenovoTrans 1.04 -K 25<br><a target="_blank" rel="noopener" href="https://github.com/aquaskyline/SOAPdenovo-Trans/">https://github.com/aquaskyline/SOAPdenovo-Trans/</a></li>
<li>Oases 0.2.09 (Velvetv1.2.10) (velveth haslength: 25) (velvetg options: -read trkg yes)<br><a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/~zerbino/oases/">http://www.ebi.ac.uk/~zerbino/oases/</a></li>
<li>Trinity 2.1.1 –normalize reads<br><a target="_blank" rel="noopener" href="http://trinityrnaseq.sourceforge.net/">http://trinityrnaseq.sourceforge.net/</a></li>
</ol>
<h2 id="4-4-三代长read分析工具"><a href="#4-4-三代长read分析工具" class="headerlink" title="4.4. 三代长read分析工具"></a>4.4. 三代长read分析工具</h2><ol>
<li>LoRDEC 0.6 -k 23 -s 3<br><a target="_blank" rel="noopener" href="http://atgc.lirmm.fr/lordec/">http://atgc.lirmm.fr/lordec/</a></li>
<li>GMAP 12&#x2F;31&#x2F;15 -f 1<br><a target="_blank" rel="noopener" href="http://research-pub.gene.com/gmap/">http://research-pub.gene.com/gmap/</a></li>
<li>STARlong 2.5.1b<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
</ol>
<p>Followed the recommended options : </p>
<ul>
<li>–outSAMattributes NH HI NM MD</li>
<li>–readNameSeparator space</li>
<li>–outFilterMultimapScoreRange 1</li>
<li>–outFilterMismatchNmax 2000</li>
<li>–scoreGapNoncan -20</li>
<li>–scoreGapGCAG -4</li>
<li>–scoreGapATAC -8</li>
<li>–scoreDelOpen -1</li>
<li>–scoreDelBase -1</li>
<li>–scoreInsOpen -1</li>
<li>–scoreInsBase -1</li>
<li>–alignEndsType Local</li>
<li>–seedSearchStartLmax 50</li>
<li>–seedPerReadNmax 100000</li>
<li>–seedPerWindowNmax 1000</li>
<li>–alignTranscriptsPerReadNmax 100000</li>
<li>–alignTranscriptsPerWindowNmax 10000</li>
<li>–outSAMstrandField intronMotif</li>
<li>–outSAMunmapped Within</li>
</ul>
<ol start="4">
<li>IDP 0.1.9<br><a target="_blank" rel="noopener" href="https://www.healthcare.uiowa.edu/labs/au/IDP/">https://www.healthcare.uiowa.edu/labs/au/IDP/</a></li>
</ol>
<h2 id="4-5-定量工具"><a href="#4-5-定量工具" class="headerlink" title="4.5. 定量工具"></a>4.5. 定量工具</h2><ol>
<li>eXpress 1.5.1 (bowtie2 v2.2.7) (bowtie2 options: -a -X 600 –rdg 6,5 –rfg 6,5 –score-min L,-.6,-.4 –no-discordant –no-mixed)<br><a target="_blank" rel="noopener" href="https://pachterlab.github.io/eXpress/index.html">https://pachterlab.github.io/eXpress/index.html</a></li>
<li>kallisto 0.42.4<br><a target="_blank" rel="noopener" href="http://pachterlab.github.io/kallisto/about.html">http://pachterlab.github.io/kallisto/about.html</a></li>
<li>Sailfish 0.9.0<br><a target="_blank" rel="noopener" href="http://www.cs.cmu.edu/~ckingsf/software/sailfish/">http://www.cs.cmu.edu/~ckingsf/software/sailfish/</a></li>
<li>Salmon-Aln 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a></li>
<li>Salmon-SMEM 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type fmd<br>quant: -k,19 </li>
<li>Salmon-Quasi 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type quasi -k 31 </li>
<li>featureCounts 1.5.0-p1 -p -B -C<br><a target="_blank" rel="noopener" href="http://subread.sourceforge.net/">http://subread.sourceforge.net/</a></li>
</ol>
<h2 id="4-6-差异表达分析工具"><a href="#4-6-差异表达分析工具" class="headerlink" title="4.6. 差异表达分析工具"></a>4.6. 差异表达分析工具</h2><ol>
<li>DESeq2 1.14.1<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html">http://bioconductor.org/packages/release/bioc/html/DESeq2.html</a></li>
<li>edgeR 3.16.5<br><a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/bioc/html/edgeR.html">http://www.bioconductor.org/packages/release/bioc/html/edgeR.html</a></li>
<li>limma 3.30.7<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/limma.html">http://bioconductor.org/packages/release/bioc/html/limma.html</a></li>
<li>Cuffdiff 2.2.1<br>–frag-bias-correct –emit-count-tables<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>Ballgown 2.6.0<br><a target="_blank" rel="noopener" href="https://github.com/alyssafrazee/ballgown">https://github.com/alyssafrazee/ballgown</a></li>
<li>sleuth 0.28.1<br><a target="_blank" rel="noopener" href="https://github.com/pachterlab/sleuth">https://github.com/pachterlab/sleuth</a></li>
</ol>
<h2 id="4-7-变异分析工具"><a href="#4-7-变异分析工具" class="headerlink" title="4.7. 变异分析工具"></a>4.7. 变异分析工具</h2><ol>
<li>SAMtools 1.2 (bcftools v1.2)<br>samtools mpileup -C50 -d 100000<br><a target="_blank" rel="noopener" href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</a></li>
<li>bcftools filter -s LowQual -e ‘%QUAL&lt;20 —— DP&gt;10000’<br><a target="_blank" rel="noopener" href="https://github.com/samtools/bcftools">https://github.com/samtools/bcftools</a></li>
<li>GATK v3.5-0-g36282e4 (picard 1.129)<br><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/download/">https://software.broadinstitute.org/gatk/download/</a><br>Picard AddOrReplaceReadGroups: SO&#x3D;coordinate<br>Picard MarkDuplicates: CREATE INDEX&#x3D;true VALIDATION STRINGENCY&#x3D;SILENTGATK<br>SplitNCigarReads: -rf ReassignOneMappingQuality -RMQF 255 -RMQT 60</li>
</ol>
<p>-U ALLOW N CIGAR READSGATK<br>HaplotypeCaller: -stand call conf 20.0<br>-stand emit conf 20.0 -A StrandBiasBySample<br>-A StrandAlleleCountsBySampleGATK<br>VariantFiltration: -window 35 -cluster 3 -filterName FS -filter<br>“FS &gt;30.0” -filterName QD -filter “QD &lt;2.0”</p>
<h2 id="4-8-RNA编辑"><a href="#4-8-RNA编辑" class="headerlink" title="4.8. RNA编辑"></a>4.8. RNA编辑</h2><ol>
<li>GIREMI 0.2.1<br><a target="_blank" rel="noopener" href="https://github.com/zhqingit/giremi">https://github.com/zhqingit/giremi</a></li>
<li>Varsim 0.5.1<br><a target="_blank" rel="noopener" href="https://github.com/bioinform/varsim">https://github.com/bioinform/varsim</a></li>
</ol>
<h2 id="4-9-基因融合"><a href="#4-9-基因融合" class="headerlink" title="4.9. 基因融合"></a>4.9. 基因融合</h2><ol>
<li>FusionCatcher 0.99.5a beta<br><a target="_blank" rel="noopener" href="https://github.com/ndaniel/fusioncatcher">https://github.com/ndaniel/fusioncatcher</a></li>
<li>JAFFA 1.0.6<br><a target="_blank" rel="noopener" href="https://github.com/Oshlack/JAFFA">https://github.com/Oshlack/JAFFA</a></li>
<li>SOAPfuse 1.27<br><a target="_blank" rel="noopener" href="http://soap.genomics.org.cn/soapfuse.html">http://soap.genomics.org.cn/soapfuse.html</a></li>
<li>STAR-Fusion 0.7.0<br><a target="_blank" rel="noopener" href="https://github.com/STAR-Fusion/STAR-Fusion">https://github.com/STAR-Fusion/STAR-Fusion</a></li>
<li>TopHat-Fusion 2.0.14<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/fusion_index.shtml">http://ccb.jhu.edu/software/tophat/fusion_index.shtml</a><br>s</li>
</ol>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper</a></li>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw">https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw</a></li>
<li><a target="_blank" rel="noopener" href="https://www.cnblogs.com/wangprince2017/p/9959008.html">https://www.cnblogs.com/wangprince2017/p/9959008.html</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/" data-id="cl1aiqcs0004am4vk9ss1b9oc" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li></ul>

    </footer>
  </div>
  
</article>


  


  <nav id="page-nav">
    
    <span class="page-number current">1</span><a class="page-number" href="/page/2/">2</a><a class="page-number" href="/page/3/">3</a><span class="space">&hellip;</span><a class="page-number" href="/page/5/">5</a><a class="extend next" rel="next" href="/page/2/">下一页 &amp;raquo;</a>
  </nav>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>