<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="基于多组学的vcf格式文件，计算群体的pi，dxy，fst值。">
<meta property="og:type" content="article">
<meta property="og:title" content="使用pixy计算群体的pi，dxy，fst">
<meta property="og:url" content="http://example.com/2023/03/13/bioinfo_population.genetics_pixy/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="基于多组学的vcf格式文件，计算群体的pi，dxy，fst值。">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2023-03-12T16:00:00.000Z">
<meta property="article:modified_time" content="2023-03-26T07:10:20.618Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="pixy">
<meta property="article:tag" content="pi">
<meta property="article:tag" content="dxy">
<meta property="article:tag" content="fst">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">

<link rel="canonical" href="http://example.com/2023/03/13/bioinfo_population.genetics_pixy/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>使用pixy计算群体的pi，dxy，fst | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --></head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/03/13/bioinfo_population.genetics_pixy/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          使用pixy计算群体的pi，dxy，fst
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2023-03-13 00:00:00" itemprop="dateCreated datePublished" datetime="2023-03-13T00:00:00+08:00">2023-03-13</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2023-03-26 15:10:20" itemprop="dateModified" datetime="2023-03-26T15:10:20+08:00">2023-03-26</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/" itemprop="url" rel="index"><span itemprop="name">bioinfo</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/population-genetics/" itemprop="url" rel="index"><span itemprop="name">population genetics</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2023/03/13/bioinfo_population.genetics_pixy/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2023/03/13/bioinfo_population.genetics_pixy/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>5k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>5 mins.</span>
            </span>
            <div class="post-description">基于多组学的vcf格式文件，计算群体的pi，dxy，fst值。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=105146&auto=1&height=32"></iframe></div>

<h1 id="1-用pixy计算群体的pi，dxy，fst"><a href="#1-用pixy计算群体的pi，dxy，fst" class="headerlink" title="1. 用pixy计算群体的pi，dxy，fst"></a>1. 用pixy计算群体的pi，dxy，fst</h1><p>pixy可以基于多组学的vcf格式文件来计算群体的pi，dxy，fst值。</p>
<h2 id="1-1-安装pixy"><a href="#1-1-安装pixy" class="headerlink" title="1.1. 安装pixy"></a>1.1. 安装pixy</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda install --yes -c conda-forge pixy</span><br><span class="line">conda install --yes -c bioconda htslib</span><br><span class="line">pixy --help</span><br></pre></td></tr></table></figure>

<h2 id="1-2-准备vcf文件"><a href="#1-2-准备vcf文件" class="headerlink" title="1.2. 准备vcf文件"></a>1.2. 准备vcf文件</h2><p>pixy需要的vcf文件与只包含snp的vcf文件不同，是同时包含变异位点（variant sites）和非变异位点（invariant sites）的全位点（all sites）文件。</p>
<ol>
<li>获取all-sites的vcf文件</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">gatk --java-options &quot;-Xmx32g&quot; HaplotypeCaller -R ref.fa -I input.bam -O output.g.vcf -ERC GVCF # 生成单个样本gvcf文件</span><br><span class="line">gatk --java-options &quot;-Xmx32g&quot; GenomicsDBImport -V output1.g.vcf -V output2.g.vcf ... --genomicsdb-workspace-path &lt;allsamples_genomicsdb&gt;</span><br><span class="line">for chr in $(cat chr.list);do gatk --java-options &quot;-Xmx32g&quot; GenotypeGVCFs -R ref.fa -V samples.g.vcf -L $&#123;chr&#125; -all-sites -O $&#123;chr&#125;_allsites.vcf.gz;done  # joint-calling这里一定要加参数-all-sites(把non-variant位点也保存到vcf文件中，后面才能进行pixy的分析）。必须使用-L指定染色体，否则有报错风险。耗时步骤。</span><br><span class="line">ls *_allsites.vcf.gz &gt;vcf.list # 把生成的染色体的vcf文件名保存到vcf.list</span><br><span class="line">gatk MergeVcfs -I vcf.list -O samples_allsites.vcf.gz # 合并所有染色体vcf文件到一个vcf文件，耗时步骤</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>分别获取invariant和variant位点，过滤后，再合并</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">vcftools --gzvcf samples_allsites.vcf.gz --max-maf 0 --minQ 30 --max-missing 0.9 --min-meanDP 10 --max-meanDP 43 --remove filter.indv --recode --stdout | bgzip -c &gt; samples_invariant_filtered.vcf.gz &amp; # 提取invariant位点并过滤，同时生成out.log名称的log文件</span><br><span class="line">vcftools --gzvcf samples_allsites.vcf.gz --mac 1 --min-alleles 2 --max-alleles 2 --minQ 30 --max-missing 0.9 --min-meanDP 10 --max-meanDP 43 --remove filter.indv --recode --stdout | bgzip -c &gt; samples_variant_filtered.vcf.gz &amp; # 提取variant位点并过滤，同时也生成out.log名称的log文件，最好与上一步骤分开在两个文件夹运行</span><br><span class="line">tabix samples_invariant_filtered.vcf.gz</span><br><span class="line">tabix samples_variant_filtered.vcf.gz</span><br><span class="line">bcftools concat --allow-overlaps samples_variant.vcf.gz samples_invariant.vcf.gz -O z -o samples_allsites_filtered.vcf.gz # 合并invariant和variant位点到一个文件</span><br></pre></td></tr></table></figure>

<h2 id="1-3-pixy计算dxy，pi和fst"><a href="#1-3-pixy计算dxy，pi和fst" class="headerlink" title="1.3. pixy计算dxy，pi和fst"></a>1.3. pixy计算dxy，pi和fst</h2><ol>
<li>popfile.txt群体文件</li>
</ol>
<ul>
<li>popfile.txt文件包含样本个体分组信息，两列内容，tab分隔。第一列样本ID，第二列分组名称（常用物种作为分组依据）。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">ERS223827   BFS</span><br><span class="line">ERS223759   BFS</span><br><span class="line">ERS223750   BFS</span><br><span class="line">ERS223967   AFS</span><br><span class="line">ERS223970   AFS</span><br><span class="line">ERS223924   AFS</span><br><span class="line">ERS224300   AFS</span><br><span class="line">ERS224168   KES</span><br><span class="line">ERS224314   KES</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>运行</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">bgzip samples_allsites_filtered.vcf</span><br><span class="line">tabix samples_allsites_filtered.vcf.gz</span><br><span class="line">nohup pixy --stats pi fst dxy --vcf sample_allsites_filtered.vcf.gz --populations popfile.txt --window_size 10000 --n_cores 16 &amp;</span><br></pre></td></tr></table></figure>

<h2 id="1-4-pixy的结果"><a href="#1-4-pixy的结果" class="headerlink" title="1.4. pixy的结果"></a>1.4. pixy的结果</h2><p>根据–stats的设定，生成三个统计量的结果文件。</p>
<ol>
<li>pixy_pi.txt：居群内的核苷酸多样性（nucleotide diversity (pi)）</li>
</ol>
<ul>
<li>pop：popfile.txt群体文件中群体ID</li>
<li>chromosome：染色体或contig的ID</li>
<li>window_pos_1：基因组window的起始位置</li>
<li>window_pos_2：基因组window的终止位置</li>
<li>avg_pi：window的位点的平均核苷酸多样性。更具体地说，pixy 计算window中所有位点的每个位点的加权平均核苷酸多样性，其中权重由每个位点的基因分型样本数决定。</li>
<li>no_sites：window中至少具有一种有效基因型的位点总数。此统计信息被提供给用户，不会直接用于任何计算。</li>
<li>count_diffs：window中所有基因型之间成对差异的原始数量。这是计算 avg_pi 的分子。</li>
<li>count_comparisons：window中所有基因型之间非缺失成对比较的原始数量（即比较两种基因型且均有效的情况）。这是计算 avg_pi 的分母。</li>
<li>count_missing：window中所有基因型之间缺失配对比较的原始数量（即比较两种基因型且至少缺失一种的情况）。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">pop	chromosome	window_pos_1	window_pos_2	avg_pi	no_sites	count_diffs	count_comparisons	count_missing</span><br><span class="line">BFS	HiC_scaffold_1	1	10000	0.0	6	0	6	0</span><br><span class="line">AFS	HiC_scaffold_1	1	10000	0.0	6	0	6	0</span><br><span class="line">... ...</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>pixy_dxy.txt：居群间的核苷酸差异（nucleotide divergence (dxy)）</li>
</ol>
<ul>
<li>pop1：popfile.txt群体文件中群体ID，比较的第一个居群</li>
<li>pop2：popfile.txt群体文件中群体ID，比较的第二个居群</li>
<li>chromosome：染色体或contig的ID</li>
<li>window_pos_1：基因组window的起始位置</li>
<li>window_pos_2：基因组window的终止位置</li>
<li>avg_dxy：window的平均每个位点的核苷酸差异</li>
<li>no_sites：window中两个群体中至少具有一种有效基因型的位点总数。此统计信息提供给用户，不会直接用于任何计算。</li>
<li>count_diffs：所有基因型之间的成对交叉种群差异的原始数量。这是计算 avg_dxy 的分子。</li>
<li>count_comparisons：window中所有基因型之间非缺失成对交叉种群比较的原始数量（即比较两种基因型且均有效的情况）。这是计算 avg_dxy 的分母。</li>
<li>count_missing：window中所有基因型之间缺失的成对跨种群比较的原始数量（即两个基因型被比较而至少有一个缺失的情况）。这个统计数字提供给用户，不直接用于任何计算。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pop1	pop2	chromosome	window_pos_1	window_pos_2	avg_dxy	no_sites    count_diffs	count_comparisons	count_missing</span><br><span class="line">BFS	AFS	HiC_scaffold_1	1	10000	0.0	6	24	0</span><br><span class="line">CFS	DES	HiC_scaffold_1	1	10000	0.0	6	24	0</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>pixy_fst.txt：Weir and Cockerham’s FST</li>
</ol>
<ul>
<li>pop1：popfile.txt群体文件中群体ID，比较的第一个居群</li>
<li>pop2：popfile.txt群体文件中群体ID，比较的第二个居群</li>
<li>chromosome：染色体或contig的ID</li>
<li>window_pos_1：基因组window的起始位置</li>
<li>window_pos_2：基因组window的终止位置</li>
<li>avg_wc_fst：window的平均每个SNP（不是每个位点）的 Weir and Cockerham’s FST </li>
<li>no_snps：window内变异位点（SNPs）的总数</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pop1	pop2	chromosome	window_pos_1	window_pos_2	avg_wc_fst	no_snps</span><br><span class="line">BFS	AFS	HiC_scaffold_1	30001	40000	NA	67</span><br><span class="line">CFS	DES	HiC_scaffold_1	30001	40000	0.0	67</span><br></pre></td></tr></table></figure>

<h2 id="结果的后续处理"><a href="#结果的后续处理" class="headerlink" title="结果的后续处理"></a>结果的后续处理</h2><ol>
<li>计算平均值的注意事项</li>
</ol>
<p>值得注意的是，如果想用pixy的结果继续计算更大的window或者整个基因组的平均值不能直接用window的pi&#x2F;dxy值来计算。</p>
<p>正确的计算方法是使用原始计数重新计算differences&#x2F;comparisons ratios，公式如下：</p>
<p>$$(window 1 count_diffs + window 2 count_diffs + … + window n count_diffs) &#x2F; (window 1 comparisons + window 2 comparisons + … + window n comparisons)$$</p>
<ol start="2">
<li>计算整个基因组的pi值</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">cat pixy_pi.txt |cut -f1,2 |sort|uniq &gt;pops.list</span><br><span class="line">cat pops.list | while read line</span><br><span class="line">do</span><br><span class="line">    a=$(echo $line | awk &#x27;&#123;print $1&#125;&#x27;);</span><br><span class="line">    b=$(echo $line | awk &#x27;&#123;print $2&#125;&#x27;);</span><br><span class="line">    cat pixy_dxy.txt |awk &#x27;$1 == &quot;&#x27;$a&#x27;&quot; &amp;&amp; $2 == &quot;&#x27;$b&#x27;&quot; &#123;suma+=$8;sumb+=$9&#125; END &#123;print &quot;&#x27;$a&#x27;&quot;, &quot;&#x27;$b&#x27;&quot;, suma, sumb, suma/sumb&#125;&#x27; &gt;&gt; dxy_average.txt</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>计算整个基因组的dxy平均值</li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">cat pixy_dxy.txt |cut -f1 |sort|uniq &gt;pop.list</span><br><span class="line">cat pop.list | while read line</span><br><span class="line">do</span><br><span class="line">    a=$(echo $line | awk &#x27;&#123;print $1&#125;&#x27;);</span><br><span class="line">    cat pixy_pi.txt |awk &#x27;$1 == &quot;&#x27;$a&#x27;&quot; &#123;suma+=$7;sumb+=$8&#125; END &#123;print &quot;&#x27;$a&#x27;&quot;, suma, sumb, suma/sumb&#125;&#x27; &gt;&gt; pi_average.txt</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<h2 id="结果可视化"><a href="#结果可视化" class="headerlink" title="结果可视化"></a>结果可视化</h2><h1 id="2-references"><a href="#2-references" class="headerlink" title="2. references"></a>2. references</h1><ol>
<li>pixy manual：<a target="_blank" rel="noopener" href="https://pixy.readthedocs.io/en/latest/index.html">https://pixy.readthedocs.io/en/latest/index.html</a></li>
<li>pixy github：<a target="_blank" rel="noopener" href="https://github.com/ksamuk/pixy">https://github.com/ksamuk/pixy</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=30% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    
        <div class="reward-container">
  <div>真诚赞赏，手留余香</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    Donate
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/images/wechatpay.jpg" alt="南方枝叶 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
      
      <div style="display: inline-block;">
        <img src="/images/alipay.jpg" alt="南方枝叶 Alipay">
        <p>Alipay</p>
      </div>

  </div>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/pixy/" rel="tag"># pixy</a>
              <a href="/tags/pi/" rel="tag"># pi</a>
              <a href="/tags/dxy/" rel="tag"># dxy</a>
              <a href="/tags/fst/" rel="tag"># fst</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/12/31/omics_genome.survey_02.Smudgeplot/" rel="prev" title="用k-mer分析进行基因组调查：（二）用Smudgeplot估计倍性">
      <i class="fa fa-chevron-left"></i> 用k-mer分析进行基因组调查：（二）用Smudgeplot估计倍性
    </a></div>
      <div class="post-nav-item">
    <a href="/2023/03/13/bioinfo_phylogeny_MSCquartets/" rel="next" title="用R包MSCquartets评估不完全谱系分选的程度">
      用R包MSCquartets评估不完全谱系分选的程度 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-%E7%94%A8pixy%E8%AE%A1%E7%AE%97%E7%BE%A4%E4%BD%93%E7%9A%84pi%EF%BC%8Cdxy%EF%BC%8Cfst"><span class="nav-text">1. 用pixy计算群体的pi，dxy，fst</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#1-1-%E5%AE%89%E8%A3%85pixy"><span class="nav-text">1.1. 安装pixy</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-2-%E5%87%86%E5%A4%87vcf%E6%96%87%E4%BB%B6"><span class="nav-text">1.2. 准备vcf文件</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-3-pixy%E8%AE%A1%E7%AE%97dxy%EF%BC%8Cpi%E5%92%8Cfst"><span class="nav-text">1.3. pixy计算dxy，pi和fst</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-4-pixy%E7%9A%84%E7%BB%93%E6%9E%9C"><span class="nav-text">1.4. pixy的结果</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%93%E6%9E%9C%E7%9A%84%E5%90%8E%E7%BB%AD%E5%A4%84%E7%90%86"><span class="nav-text">结果的后续处理</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%93%E6%9E%9C%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-text">结果可视化</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-references"><span class="nav-text">2. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">100</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">53</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">320</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

      <div class="wechat_channel">
        <br>
        <!-- 这里添加你的二维码图片 -->
        <img src ="/images/wechat_channel.jpg?raw=true">
        <span>欢迎关注微信公众号</span>
      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">553k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">8:23</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2023/03/13/bioinfo_population.genetics_pixy/";
    this.page.identifier = "2023/03/13/bioinfo_population.genetics_pixy/";
    this.page.title = "使用pixy计算群体的pi，dxy，fst";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
