<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="介绍了分析两个基因组间的结构变异的软件MUMandCo，用它检测插入，缺失，倒位，易位，串联重复复制，串联重复收缩等结构变异。">
<meta property="og:type" content="article">
<meta property="og:title" content="结构变异分析软件：MUMandCo">
<meta property="og:url" content="http://example.com/2022/07/17/bioinfo_variantion_SV_MUMandCo/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="介绍了分析两个基因组间的结构变异的软件MUMandCo，用它检测插入，缺失，倒位，易位，串联重复复制，串联重复收缩等结构变异。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/structural_variants_sum.png?raw=true">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2022-07-16T16:00:00.000Z">
<meta property="article:modified_time" content="2022-07-17T09:05:16.800Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="structural variation">
<meta property="article:tag" content="SV">
<meta property="article:tag" content="insertion">
<meta property="article:tag" content="deletion">
<meta property="article:tag" content="MUMandCo">
<meta property="article:tag" content="duplication">
<meta property="article:tag" content="contraction">
<meta property="article:tag" content="inversion">
<meta property="article:tag" content="translocation">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/structural_variants_sum.png?raw=true">

<link rel="canonical" href="http://example.com/2022/07/17/bioinfo_variantion_SV_MUMandCo/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>结构变异分析软件：MUMandCo | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>关于</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file-user fa-fw"></i>resume</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/07/17/bioinfo_variantion_SV_MUMandCo/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          结构变异分析软件：MUMandCo
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>
              

              <time title="创建时间：2022-07-17 00:00:00 / 修改时间：17:05:16" itemprop="dateCreated datePublished" datetime="2022-07-17T00:00:00+08:00">2022-07-17</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/" itemprop="url" rel="index"><span itemprop="name">bioinfo</span></a>
                </span>
                  ，
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/variantion/" itemprop="url" rel="index"><span itemprop="name">variantion</span></a>
                </span>
                  ，
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/variantion/structural-variation/" itemprop="url" rel="index"><span itemprop="name">structural variation</span></a>
                </span>
            </span>

          
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus：</span>
    
    <a title="disqus" href="/2022/07/17/bioinfo_variantion_SV_MUMandCo/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2022/07/17/bioinfo_variantion_SV_MUMandCo/" itemprop="commentCount"></span>
    </a>
  </span>
  
  
            <div class="post-description">介绍了分析两个基因组间的结构变异的软件MUMandCo，用它检测插入，缺失，倒位，易位，串联重复复制，串联重复收缩等结构变异。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=18031133&auto=1&height=32"></iframe></div>

<h1 id="1-MUMandCo简介"><a href="#1-MUMandCo简介" class="headerlink" title="1. MUMandCo简介"></a>1. MUMandCo简介</h1><p>MUMandCo是用于<strong>两个基因组间</strong>的<strong>结构变异（structural variant，SV）</strong>检测分析的软件，基于MUMmer(v4)的nucmer比对结果检测全基因组范围内的插入，缺失，串联重复，倒位和易位等结构变异。</p>
<p>2020年发表在Bioinformatics。</p>
<h1 id="2-MUMandCo原理"><a href="#2-MUMandCo原理" class="headerlink" title="2. MUMandCo原理"></a>2. MUMandCo原理</h1><p>MUMandCo调用MUMmer的nucmer模块做互换的全基因组比对（whole-genome alignmetn，WGA），delta-filter模块做全局（g-）和多对多（m-）的过滤，show-coords模块解析坐标。</p>
<ol>
<li>选出与参考基因组有最精确、非重叠的alignments的query contigs。其余的alignment都不考虑。</li>
<li>利用这些alignments统计超过50bp的SV的类型和特征。</li>
<li>全局（g-）比对 基于多个contig-chromosome pairings来检测易位（translocation）片段，基于alignment orientation检测大的倒位（inversions），基于alignment gaps检测可能的插入缺失（insertion and deletion），利用互换的比对，gaps会在reference和query基因组间同时被考虑。</li>
<li>多对多（m-）比对用于检测潜在的倒位（inversions）和重复（duplications）。</li>
<li>用全局（g-）和多对多（m-）比对的过滤去除假阳性。</li>
<li>参数<code>-b</code>可以调用BLAST来检测插入和缺失是重复的（mobile）还是新的（novel）。</li>
<li>生成保存了reference和query基因组的SV的类型、坐标等细节的tsv文件。</li>
</ol>
<h1 id="3-MUMandCo可以检测的内容"><a href="#3-MUMandCo可以检测的内容" class="headerlink" title="3. MUMandCo可以检测的内容"></a>3. MUMandCo可以检测的内容</h1><ul>
<li>插入（insertion）：&gt;&#x3D;50bp &amp; &lt;&#x3D;150kb</li>
<li>缺失（deletion）：&gt;&#x3D;50bp &amp; &lt;&#x3D;150kb</li>
<li>串联重复复制（tandem duplication）：&gt;&#x3D;50bp &amp; &lt;&#x3D;150kb</li>
<li>串联重复收缩（tandem contraction）：&gt;&#x3D;50bp &amp; &lt;&#x3D;150kb</li>
<li>倒位（inversion）：&gt;&#x3D;1kb</li>
<li>易位（translocation）：&gt;&#x3D;10kb</li>
</ul>
<h1 id="4-MUMandCo安装"><a href="#4-MUMandCo安装" class="headerlink" title="4. MUMandCo安装"></a>4. MUMandCo安装</h1><ul>
<li>MUMandCo是个bash脚本，下载后赋予执行权限即可使用。</li>
<li>依赖的软件包括MUMmer(v4)和samtools，如果使用<code>-b</code>参数则还需要blast。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/SAMtoBAM/MUMandCo.git</span><br><span class="line">chmod 711 ./MUMandCo/mumandco_v3.8.sh</span><br></pre></td></tr></table></figure>

<h1 id="5-MUMandCo使用"><a href="#5-MUMandCo使用" class="headerlink" title="5. MUMandCo使用"></a>5. MUMandCo使用</h1><ol>
<li>运行</li>
</ol>
<p><code>bash mumandco_v3.8.sh -r genome.fa -q query.fa -g 125500000 -t 24 -b -o out</code></p>
<ol start="2">
<li>参数</li>
</ol>
<ul>
<li>-r genome.fa：参考基因组</li>
<li>-q query.fa：被检测的基因组</li>
<li>-g 125500000：参考基因组大小，单位是bp</li>
<li>-t 24：线程24，默认1</li>
<li>-b：添加blast选项用来确认插入&#x2F;缺失是mobile&#x2F;repetitive还是novel。我的理解是mobile&#x2F;repetitive代表别的地方有这个插入&#x2F;缺失，novel代表是一个新产生的插入&#x2F;缺失。</li>
<li>-ml：alignments的最小长度；默认50bp</li>
<li>-o out：输出文件的前缀；默认是mumandco</li>
</ul>
<ol start="3">
<li>时间</li>
</ol>
<ul>
<li>两个都是约300Mbp的基因组的SV分析，用了24线程，运行总耗时5小时。</li>
</ul>
<h1 id="6-MUMandCo结果"><a href="#6-MUMandCo结果" class="headerlink" title="6. MUMandCo结果"></a>6. MUMandCo结果</h1><ol>
<li>out.summary.txt</li>
</ol>
<ul>
<li>总结SV的类型和数量的文件</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">mumandco</span><br><span class="line">Total_SVs	8252</span><br><span class="line">Deletions	3337</span><br><span class="line">Insertions	4363</span><br><span class="line">Duplications	44</span><br><span class="line">Contractions	8</span><br><span class="line">Inversions	144</span><br><span class="line">Translocations	356</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>out.SVs_all.tsv</li>
</ol>
<ul>
<li>保存了所有检测到的SVs的每一个SV的类型、坐标、长度等详细信息。</li>
<li>共有9列，包括ref_chr    query_chr    ref_start    ref_stop    size    SV_type    query_start  query_stop    info。</li>
<li>第六列SV_type包括contraction，deletion_mobile，deletion_novel，duplication，insertion_mobile，insertion_novel，inversion，transloc。</li>
<li>第九列info包含以下三种情况：<ul>
<li>‘complicated’: 在一个区域（region）有多个calls，一般是插入和缺失的重叠造成。</li>
<li>‘double’: 在相同的坐标位置（coordinates）有几个calls; 一般是串联重复复制或者串联重复收缩有多个拷贝的改变造成。</li>
<li>‘]chrX:xxxxxx]’ : 标记易位片段与其他片段的关联的VCF指示符。</li>
</ul>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ref_chr	query_chr	ref_start	ref_stop	size	SV_type	query_start  query_stop	info</span><br><span class="line">MCscaf001	Contig151	257669	257671	188	insertion_mobile	18955     19143</span><br><span class="line">MCscaf001	LG12	674171	698315	10734	insertion_mobile	12964720     12975454</span><br><span class="line">... ...</span><br><span class="line">MCscaf254	LG05	9960	45987	36027	transloc	10452396	10497136	]MCscaf012:3067777]</span><br><span class="line">MCscaf254	LG05	13407	16116	2709	deletion_mobile	10455842	10467408	complicated</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>out.SVs_all.withfragment.tsv</li>
</ol>
<ul>
<li>与out.SVs_all.tsv格式和内容类似，多一列fragment信息，保存了对应的参考基因组的碱基序列（插入的序列来自query基因组）。</li>
</ul>
<ol start="4">
<li>out.SVs_all.vcf</li>
<li>out_alignments文件夹</li>
</ol>
<ul>
<li>保存了alignments文件，包括ref和query各自的的delta，delta_filter，delta_filter.coords，delta_filter.coordsg文件</li>
</ul>
<h1 id="7-结果整理和统计"><a href="#7-结果整理和统计" class="headerlink" title="7. 结果整理和统计"></a>7. 结果整理和统计</h1><ol>
<li>通过out.SVs_all.tsv文件的第六列SV_type值把不同的SV类型分开，例如单独获取translocations的信息：<code>cat out.SVs_all.tsv| awk &#39;$6==&quot;transloc&quot; &#123;print $0&#125;&#39; &gt;transloc.tsv</code></li>
<li>大部分类型都是每个一行记录，所以行数就是总数，总数与out.summary.txt文件是一致的。</li>
<li>translocations这种类型大部分有两行记录，所以行数不是总数，如果想要计算总数可以这样：<code>cat transloc.tsv |cut -f1-8|uniq |wc -l</code>。</li>
</ol>
<p>我整理了一个这样的表格，用来一览SV的概况，供参考。</p>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/structural_variants_sum.png?raw=true" title="结构变异概况" width="90%" align=center/>

<p><strong><p align="center">结构变异概况</p></strong></p>
<p>感慨一句，选择什么计算机语言来写代码真没那么重要，人家bash脚本也可以发Bioinformatics。</p>
<h1 id="8-references"><a href="#8-references" class="headerlink" title="8. references"></a>8. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://github.com/SAMtoBAM/MUMandCo">https://github.com/SAMtoBAM/MUMandCo</a></li>
<li>paper: <a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/36/10/3242/5756209">https://academic.oup.com/bioinformatics/article/36/10/3242/5756209</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=50% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/structural-variation/" rel="tag"># structural variation</a>
              <a href="/tags/SV/" rel="tag"># SV</a>
              <a href="/tags/insertion/" rel="tag"># insertion</a>
              <a href="/tags/deletion/" rel="tag"># deletion</a>
              <a href="/tags/MUMandCo/" rel="tag"># MUMandCo</a>
              <a href="/tags/duplication/" rel="tag"># duplication</a>
              <a href="/tags/contraction/" rel="tag"># contraction</a>
              <a href="/tags/inversion/" rel="tag"># inversion</a>
              <a href="/tags/translocation/" rel="tag"># translocation</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/07/10/omics_genome_quality.assessment_intro/" rel="prev" title="基因组质量评估：（一）概述">
      <i class="fa fa-chevron-left"></i> 基因组质量评估：（一）概述
    </a></div>
      <div class="post-nav-item">
    <a href="/2022/07/20/omics_genome_quality.assessment_LAI/" rel="next" title="基因组质量评估：（四）用LAI评估基因组组装的连贯性">
      基因组质量评估：（四）用LAI评估基因组组装的连贯性 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-MUMandCo%E7%AE%80%E4%BB%8B"><span class="nav-number">1.</span> <span class="nav-text">1. MUMandCo简介</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-MUMandCo%E5%8E%9F%E7%90%86"><span class="nav-number">2.</span> <span class="nav-text">2. MUMandCo原理</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-MUMandCo%E5%8F%AF%E4%BB%A5%E6%A3%80%E6%B5%8B%E7%9A%84%E5%86%85%E5%AE%B9"><span class="nav-number">3.</span> <span class="nav-text">3. MUMandCo可以检测的内容</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-MUMandCo%E5%AE%89%E8%A3%85"><span class="nav-number">4.</span> <span class="nav-text">4. MUMandCo安装</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-MUMandCo%E4%BD%BF%E7%94%A8"><span class="nav-number">5.</span> <span class="nav-text">5. MUMandCo使用</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-MUMandCo%E7%BB%93%E6%9E%9C"><span class="nav-number">6.</span> <span class="nav-text">6. MUMandCo结果</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-%E7%BB%93%E6%9E%9C%E6%95%B4%E7%90%86%E5%92%8C%E7%BB%9F%E8%AE%A1"><span class="nav-number">7.</span> <span class="nav-text">7. 结果整理和统计</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-references"><span class="nav-number">8.</span> <span class="nav-text">8. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">69</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">34</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">232</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
</div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2022/07/17/bioinfo_variantion_SV_MUMandCo/";
    this.page.identifier = "2022/07/17/bioinfo_variantion_SV_MUMandCo/";
    this.page.title = "结构变异分析软件：MUMandCo";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
