<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>基因富集分析(gene set enrichment analysis, GSEA) —— clusterProfiler | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="介绍了基因富集分析R包clusterProfiler。">
<meta property="og:type" content="article">
<meta property="og:title" content="基因富集分析(gene set enrichment analysis, GSEA) —— clusterProfiler">
<meta property="og:url" content="http://example.com/2021/12/13/bioinfo_GSEA_clusterProfiler/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="介绍了基因富集分析R包clusterProfiler。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://yulab-smu.top/biomedical-knowledge-mining-book/figures/clusterProfiler-diagram.png">
<meta property="article:published_time" content="2021-12-13T14:00:00.000Z">
<meta property="article:modified_time" content="2022-03-28T02:43:39.818Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="gene set enrichment analysis">
<meta property="article:tag" content="GSEA">
<meta property="article:tag" content="clusterProfiler">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://yulab-smu.top/biomedical-knowledge-mining-book/figures/clusterProfiler-diagram.png">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-bioinfo_GSEA_clusterProfiler" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/12/13/bioinfo_GSEA_clusterProfiler/" class="article-date">
  <time datetime="2021-12-13T14:00:00.000Z" itemprop="datePublished">2021-12-13</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      基因富集分析(gene set enrichment analysis, GSEA) —— clusterProfiler
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1336867002&auto=1&height=32"></iframe></div>

<h1 id="1-基因富集分析-gene-set-enrichment-analysis-GSEA"><a href="#1-基因富集分析-gene-set-enrichment-analysis-GSEA" class="headerlink" title="1. 基因富集分析(gene set enrichment analysis, GSEA)"></a>1. 基因富集分析(gene set enrichment analysis, GSEA)</h1><p>基因富集分析(gene set enrichment analysis, GSEA)的介绍参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/12/bioinfo_GSEA/">博文基因富集分析</a>。</p>
<h1 id="2-clusterProfiler介绍"><a href="#2-clusterProfiler介绍" class="headerlink" title="2. clusterProfiler介绍"></a>2. clusterProfiler介绍</h1><p>clusterProfiler是一个R包，是一个解释组学数据的通用富集工具许多基因集的功能注释和富集分析，以及富集分析结果的可视化。</p>
<p>2021年07月发布了clusterProfiler 4.0版本。</p>
<img src="https://yulab-smu.top/biomedical-knowledge-mining-book/figures/clusterProfiler-diagram.png" width=100% title="clusterProfiler function and workflow" alt="clusterProfiler function and workflow" align=center/>

<p><strong><p align="center">Figure 1. clusterProfiler function and workflow</strong><br>from <a target="_blank" rel="noopener" href="https://yulab-smu.top/biomedical-knowledge-mining-book/">yulab’s book</a></p></p>
<h1 id="3-clusterProfiler支持的基因集-gene-sets-或基因通路数据库"><a href="#3-clusterProfiler支持的基因集-gene-sets-或基因通路数据库" class="headerlink" title="3. clusterProfiler支持的基因集(gene sets)或基因通路数据库"></a>3. clusterProfiler支持的基因集(gene sets)或基因通路数据库</h1><ol>
<li>Gene Ontology(GO)</li>
<li>Kyoto Encyclopedia of Genes and Genomes(KEGG)</li>
<li>Disease Ontology(DO)</li>
<li>Disease Gene Network(DisGeNET)</li>
<li>Molecular Signatures Database(MSigDb)</li>
<li>wikiPathways<br>… …</li>
</ol>
<h1 id="4-clusterProfiler功能-——-富集分析-enrichment-analysis"><a href="#4-clusterProfiler功能-——-富集分析-enrichment-analysis" class="headerlink" title="4. clusterProfiler功能 —— 富集分析(enrichment analysis)"></a>4. clusterProfiler功能 —— 富集分析(enrichment analysis)</h1><h2 id="4-1-过表达分析-Over-Representation-Analysis-ORA"><a href="#4-1-过表达分析-Over-Representation-Analysis-ORA" class="headerlink" title="4.1. 过表达分析(Over Representation Analysis, ORA)"></a>4.1. 过表达分析(Over Representation Analysis, ORA)</h2><p>过表达分析是用于采用超几何分布检验判断已知的生物功能或过程(例如GO&#x2F;KEGG)在实验产生的基因列表(例如差异表达基因列表: differentially expressed genes, DEGs)中是否过表达(over-represented&#x3D;enriched)的常用方法。</p>
<p>所有基因（或全基因组）作为背景总数N，被注释到已知基因集(例如GO&#x2F;KEGG)某一子集的基因总数是M，差异表达的基因数量是n，差异表达基因中属于M的数量是k。<br>过表达分析就是用统计学的方法判断GO&#x2F;KEGG的每一个子类中是否k&#x2F;n显著高于M&#x2F;N，显著高于就是这个子类过表达。</p>
<p>差异表达基因通常是前期通过比较转录组分析等手段获取的不同转录组间有表达差异的基因列表。</p>
<h2 id="4-2-基因集富集分析-Gene-Set-Enrichment-Analysis-GSEA"><a href="#4-2-基因集富集分析-Gene-Set-Enrichment-Analysis-GSEA" class="headerlink" title="4.2. 基因集富集分析(Gene Set Enrichment Analysis, GSEA)"></a>4.2. 基因集富集分析(Gene Set Enrichment Analysis, GSEA)</h2><p>对于差异很大的基因，可以使用ORA分析，但差异较小的基因也可能具有意义，很多相关表型差异是通过一组基因中微小但一致的变化(small but consistent changes)来表现的。</p>
<p>基因集富集分析汇总一个基因集中每个基因的统计数据，可以检测预先定义的基因集(例如GO&#x2F;KEGG)中所有基因以一种较小但协调的(small but coordinated)方式发生变化的情况。</p>
<p>基因根据表型进行排序，给定先验定义的基因集(例如共享相同GO&#x2F;KEGG类别的基因)，GSEA的目标是确定S的成员是随机分布在排序的基因列表L中，还是主要分布在顶部或者底部。</p>
<h2 id="4-3-leading-edge分析和核心富集基因-Leading-edge-analysis-and-core-enriched-genes"><a href="#4-3-leading-edge分析和核心富集基因-Leading-edge-analysis-and-core-enriched-genes" class="headerlink" title="4.3. leading edge分析和核心富集基因(Leading edge analysis and core enriched genes)"></a>4.3. leading edge分析和核心富集基因(Leading edge analysis and core enriched genes)</h2><p>包（clusterProfiler,DOSE,meshes,ReactomePA）支持Leading edge分析并可以提供GSEA分析中的核心富集基因core enriched genes结果。</p>
<ol>
<li>Leading edge分析可以获取：</li>
</ol>
<ul>
<li>Tags显示对富集分数有贡献的基因的百分比</li>
<li>List显示获得富集分数在列表中的位置</li>
<li>Signal显示富集信号强度</li>
</ul>
<h1 id="5-clusterProfiler安装"><a href="#5-clusterProfiler安装" class="headerlink" title="5. clusterProfiler安装"></a>5. clusterProfiler安装</h1><p>启动R后输入下面命令安装</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quiet <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">    install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;clusterProfiler&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h1 id="6-clusterProfiler富集分析"><a href="#6-clusterProfiler富集分析" class="headerlink" title="6. clusterProfiler富集分析"></a>6. clusterProfiler富集分析</h1><p>clusterProfiler支持对许多ontology&#x2F;pathway的hypergeometric test和gene set enrichment analyses，包括数据库GO,KEGG,DO,DisGeNET,MSigDb,wikiPathways。</p>
<p>使用支持的数据库时，需要检查分析样本是否在已有物种列表中。不在物种列表的物种数据，以及其他不在数据库列表的数据库，做分析时使用通用的富集分析(Universal enrichment analysis)模块。</p>
<p>分析前，先用命令<code>library(clusterProfiler)</code>载入clusterProfiler包。</p>
<h2 id="6-1-GO富集分析"><a href="#6-1-GO富集分析" class="headerlink" title="6.1. GO富集分析"></a>6.1. GO富集分析</h2><h3 id="6-1-1-支持的物种"><a href="#6-1-1-支持的物种" class="headerlink" title="6.1.1. 支持的物种"></a>6.1.1. 支持的物种</h3><p>GO富集分析需要物种的注释数据库，来源可以是Bioconductor，AnnotationHub或者用户提供GO注释文件。</p>
<p>groupGO(),enrichGO(),gseGO()函数支持具有OrgDb数据库的生物物种，enricher()和gseGO()函数支持用户自己提供的GO注释文件。</p>
<ol>
<li><p>clusterProfiler的默认GO注释信息来自Bioconductor，目前Bioconductor自带20个，植物只有拟南芥org.At.tair.db一个数据库。<a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">OrgDb数据库</a></p>
</li>
<li><p>还可以通过AnnotationHub包在线获取OrgDb对象，并制作Org.Db库。</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">source<span class="punctuation">(</span><span class="string">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="punctuation">)</span></span><br><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quietly <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">  install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;AnnotationHub&quot;</span><span class="punctuation">)</span> <span class="comment"># 一个包含大量注释信息的数据库，里面有很多物种及来源于很多数据库的注释信息。</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;biomaRt&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">library<span class="punctuation">(</span>AnnotationHub<span class="punctuation">)</span> </span><br><span class="line">library<span class="punctuation">(</span>biomaRt<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">hub <span class="operator">&lt;-</span> AnnotationHub<span class="punctuation">(</span><span class="punctuation">)</span> <span class="comment">#建立AnnotationHub对象（视人品，网不行加载不了）</span></span><br><span class="line"><span class="comment"># unique(hub$species) #查看AnonotationHub里面物种</span></span><br><span class="line">hub<span class="operator">$</span>species<span class="punctuation">[</span>which<span class="punctuation">(</span>hub<span class="operator">$</span>species<span class="operator">==</span><span class="string">&quot;Solanum&quot;</span><span class="punctuation">)</span><span class="punctuation">]</span> <span class="comment">#看AnonotationHub里是否包含想要的物种</span></span><br><span class="line"><span class="comment"># Solanum是番茄的拉丁名</span></span><br><span class="line">query<span class="punctuation">(</span>hub<span class="punctuation">,</span> <span class="string">&quot;Solanum&quot;</span><span class="punctuation">)</span>  <span class="comment">#查看该物种信息</span></span><br><span class="line">hub<span class="punctuation">[</span>hub<span class="operator">$</span>species<span class="operator">==</span><span class="string">&quot;Solanum&quot;</span> <span class="operator">&amp;</span> hub<span class="operator">$</span>rdataclass <span class="operator">==</span> <span class="string">&quot;OrgDb&quot;</span><span class="punctuation">]</span> <span class="comment">#OrgDb属于rdataclass中，因此查看下该物种有没有OrgDb</span></span><br><span class="line">Solanum.OrgDb <span class="operator">&lt;-</span> hub<span class="punctuation">[[</span><span class="string">&quot;AH59087&quot;</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="comment">#AH59087是番茄对应的编号</span></span><br><span class="line"><span class="comment">#制作为标准注释库，就可和模式生物一样使用了</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>用户提供GO注释数据<br>如果分析的物种没有OrgDb数据库，可以自己做GO注释后提供背景注释数据库，比如interproscan、eggnog-mapper和blas2go等软件获取的GO注释结果，整理成clusterProfiler支持的输入格式即可。</li>
</ol>
<p>clusterProfiler需要导入的GO注释文件go_annotation.txt的格式如下：</p>
<table>
<thead>
<tr>
<th>GeneID</th>
<th>GO</th>
<th>GO_Description</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>GO:0005819</td>
<td>spindle</td>
</tr>
<tr>
<td>2</td>
<td>GO:0072686</td>
<td>mitotic spindle</td>
</tr>
<tr>
<td>3</td>
<td>GO:0000776</td>
<td>kinetochore</td>
</tr>
</tbody></table>
<p>data.frame格式，包含三列，第一列为Gene ID，第二列为 GO ID，第三列为GO_Description，顺序无要求。</p>
<p>可以使用enricher()和gseGO()函数进行ORA和GSEA分析。</p>
<p>如果基因是通过直接注释来注释的，那么也应该用它们的祖先GO节点来注释（间接注释）。如果用户只有直接注释，他们可以将他们的注释传递给buildGOmap函数，该函数将推断间接注释并生成data.frame同时适用于enricher()和 的gseGO()。</p>
<h3 id="6-1-2-GO的ORA分析"><a href="#6-1-2-GO的ORA分析" class="headerlink" title="6.1.2. GO的ORA分析"></a>6.1.2. GO的ORA分析</h3><h4 id="6-1-2-1-输入文件"><a href="#6-1-2-1-输入文件" class="headerlink" title="6.1.2.1. 输入文件"></a>6.1.2.1. 输入文件</h4><p>ORA分析的输入文件是gene ID list，比如差异表达分析(DESeq2)获得的差异表达基因列表，保存为内容为一列数据的文本文件gene.list，数据内容可以是OrgDb支持的任意ID类型（常用的都支持，ENSEMBL，ENTREZID，GENETYPE，GO，PFAM），具体参考<a target="_blank" rel="noopener" href="http://yulab-smu.top/biomedical-knowledge-mining-book/useful-utilities.html#id-convert">ID</a>。<br>分析的是列表中基因在GO&#x2F;KEGG各个子分类单元是否被过度代表还是代表不足。</p>
<h4 id="6-1-2-2-bitr格式转换"><a href="#6-1-2-2-bitr格式转换" class="headerlink" title="6.1.2.2. bitr格式转换"></a>6.1.2.2. bitr格式转换</h4><p>如果需要，也可以用bitr功能函数实现各种ID格式的转换。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">)</span> <span class="comment">#单列基因名文件</span></span><br><span class="line">data<span class="operator">$</span>V1 <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#需要character格式，然后进行ID转化</span></span><br><span class="line">test <span class="operator">=</span> bitr<span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">,</span> fromType<span class="operator">=</span><span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">,</span> toType<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;ENSEMBL&quot;</span><span class="punctuation">,</span> <span class="string">&quot;ENTREZID&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> OrgDb<span class="operator">=</span> org.At.tair.db<span class="punctuation">)</span> <span class="comment">#将SYMBOL格式转为ENSEMBL和ENTERZID格式 </span></span><br><span class="line">head<span class="punctuation">(</span>test<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">    SYMBOL         ENSEMBL ENTREZID</span><br><span class="line"><span class="number">1</span>    AASDH ENSG00000157426   <span class="number">132949</span></span><br><span class="line"><span class="number">2</span>   ABCB11 ENSG00000073734     <span class="number">8647</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-2-3-GO分类"><a href="#6-1-2-3-GO分类" class="headerlink" title="6.1.2.3. GO分类"></a>6.1.2.3. GO分类</h4><p>groupGO()函数可以基于GO在指定水平范围内的分布进行基因分类。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br><span class="line">ggo <span class="operator">&lt;-</span> groupGO<span class="punctuation">(</span>gene <span class="operator">=</span> genes<span class="punctuation">,</span> OrgDb <span class="operator">=</span> org.Hs.eg.db<span class="punctuation">,</span> ont <span class="operator">=</span> <span class="string">&quot;CC&quot;</span><span class="punctuation">,</span> level <span class="operator">=</span> <span class="number">3</span><span class="punctuation">,</span> readable <span class="operator">=</span> TURE<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>ggo<span class="punctuation">,</span><span class="number">3</span><span class="punctuation">)</span></span><br><span class="line">                   ID       Description Count GeneRatio geneID</span><br><span class="line">GO<span class="operator">:</span><span class="number">0000003</span> GO<span class="operator">:</span><span class="number">0000003</span>      reproduction     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span>       </span><br><span class="line">GO<span class="operator">:</span><span class="number">0008152</span> GO<span class="operator">:</span><span class="number">0008152</span> metabolic process     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span>       </span><br><span class="line">GO<span class="operator">:</span><span class="number">0001906</span> GO<span class="operator">:</span><span class="number">0001906</span>      cell killing     <span class="number">0</span>     <span class="number">0</span><span class="operator">/</span><span class="number">929</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-2-4-ORA分析"><a href="#6-1-2-4-ORA分析" class="headerlink" title="6.1.2.4. ORA分析"></a>6.1.2.4. ORA分析</h4><ol>
<li>读取输入文件<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br></pre></td></tr></table></figure></li>
<li>支持物种的标准分析 enrichGO()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">ego <span class="operator">&lt;-</span> enrichGO<span class="punctuation">(</span>gene          <span class="operator">=</span> genes<span class="punctuation">,</span> <span class="comment"># list of entrez gene id</span></span><br><span class="line">                OrgDb         <span class="operator">=</span> org.At.tair.db<span class="punctuation">,</span> <span class="comment"># 背景使用分析物种的org包，这里示例使用拟南芥的数据库</span></span><br><span class="line">                keyType       <span class="operator">=</span> <span class="string">&#x27;ENSEMBL&#x27;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">                ont           <span class="operator">=</span> <span class="string">&quot;CC&quot;</span><span class="punctuation">,</span> <span class="comment"># &quot;BP&quot;, &quot;MF&quot;, &quot;CC&quot;, &quot;ALL&quot;。GO三个子类里选</span></span><br><span class="line">                pAdjustMethod <span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span> <span class="comment"># 指定多重假设检验矫正的方法，选项包含 &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</span></span><br><span class="line">                pvalueCutoff  <span class="operator">=</span> <span class="number">0.01</span><span class="punctuation">,</span> <span class="comment"># 富集分析的pvalue</span></span><br><span class="line">                qvalueCutoff  <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span> <span class="comment"># 富集分析显著性的qvalue</span></span><br><span class="line">                readable      <span class="operator">=</span> <span class="literal">TRUE</span> <span class="punctuation">)</span> <span class="comment"># 是否将gene ID转换到 gene symbol</span></span><br><span class="line">head<span class="punctuation">(</span>ego<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line">     ONTOLOGY         ID                                                Description</span><br><span class="line">GO<span class="operator">:</span><span class="number">0002887</span>       BP GO<span class="operator">:</span><span class="number">0002887</span> negative regulation of myeloid leukocyte mediated immunity</span><br><span class="line">GO<span class="operator">:</span><span class="number">0033004</span>       BP GO<span class="operator">:</span><span class="number">0033004</span>                negative regulation of mast cell activation</span><br><span class="line">           GeneRatio  BgRatio      pvalue  p.adjust    qvalue                   geneID Count</span><br><span class="line">GO<span class="operator">:</span><span class="number">0002887</span>     <span class="number">2</span><span class="operator">/</span><span class="number">121</span> <span class="number">10</span><span class="operator">/</span><span class="number">11461</span> <span class="number">0.004706555</span> <span class="number">0.7796682</span> <span class="number">0.7796682</span>              CD300A<span class="operator">/</span>CD84     <span class="number">2</span></span><br><span class="line">GO<span class="operator">:</span><span class="number">0033004</span>     <span class="number">2</span><span class="operator">/</span><span class="number">121</span> <span class="number">10</span><span class="operator">/</span><span class="number">11461</span> <span class="number">0.004706555</span> <span class="number">0.7796682</span> <span class="number">0.7796682</span>              CD300A<span class="operator">/</span>CD84     <span class="number">2</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li><p>提供注释文件go_annotation.txt的非模式物种分析 enricher()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">go_anno <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;go_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> go_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> go_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">ego <span class="operator">&lt;-</span> enricher<span class="punctuation">(</span>genes<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>ego结果文件解释</p>
</li>
</ol>
<ul>
<li>ONTOLOGY：CC  BP  MF </li>
<li>ID: Gene Ontology数据库中唯一的标号信息</li>
<li>Description ：Gene Ontology功能的描述信息</li>
<li>GeneRatio：差异基因中与该Term相关的基因数与整个差异基因总数的比值</li>
<li>BgRatio：所有（ bg）基因中与该Term相关的基因数与所有（ bg）基因的比值</li>
<li>pvalue: 富集分析统计学显著水平，一般情况下， P-value &lt; 0.05 该功能为富集项</li>
<li>p.adjust 矫正后的P-Value</li>
<li>qvalue：对p值进行统计学检验的q值</li>
<li>geneID：与该Term相关的基因</li>
<li>Count：与该Term相关的基因数</li>
</ul>
<ol start="5">
<li>输出结果保存为csv表格<br><code>write.csv(summary(ego),&quot;enrich.csv&quot;,row.names =FALSE)</code></li>
</ol>
<h4 id="6-1-2-5-结果整理和筛选"><a href="#6-1-2-5-结果整理和筛选" class="headerlink" title="6.1.2.5. 结果整理和筛选"></a>6.1.2.5. 结果整理和筛选</h4><ol>
<li>如果没使用keyType参数，可以在得到结果后使用</li>
<li>如果没使用readable&#x3D;True参数，可以在得到结果后用函数setReadable()将GeneID转换为symbol。<code>ego &lt;- setReadable(ego, OrgDb = org.At.tair.db)</code></li>
<li>函数dropGO可以移除enrichGO结果中特定的GO term或GO level</li>
<li>函数gofilter()可以将结果限定在特定的GO level</li>
<li>函数simplify可以去除冗余，<code>ego_rm &lt;- simplify(ego, cutoff=0.7, by=&quot;p.adjust&quot;, select_fun=min)</code></li>
</ol>
<h3 id="6-1-3-GO的GSEA分析"><a href="#6-1-3-GO的GSEA分析" class="headerlink" title="6.1.3. GO的GSEA分析"></a>6.1.3. GO的GSEA分析</h3><p>GSEA分析通过置换检验来计算p值</p>
<h4 id="6-1-3-1-GSEA的输入文件"><a href="#6-1-3-1-GSEA的输入文件" class="headerlink" title="6.1.3.1. GSEA的输入文件"></a>6.1.3.1. GSEA的输入文件</h4><ol>
<li>GSEA分析的输入文件是一个基因排序列表，有三个要点：</li>
</ol>
<ul>
<li>numeric vector：倍数变化或者其他类型的数字变量，比如差异表达分析里的logFC值</li>
<li>named vector：每个数字倍对应的gene ID命名</li>
<li>sorted vector；数字应该以降序排序<br>即包含两列，一列基因ID名称，一列数据，并以数据降序排序。</li>
</ul>
<ol start="2">
<li>获取输入文件的示例：</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">d <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span>your_csv_file<span class="punctuation">)</span></span><br><span class="line"><span class="comment">## assume that 1st column is ID (no duplicated allowed)</span></span><br><span class="line"><span class="comment">## 2nd column is fold change</span></span><br><span class="line"><span class="comment">## feature 1: numeric vector</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="comment">## feature 2: named vector</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">## feature 3: decreasing order</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>mydata<span class="punctuation">,</span><span class="number">3</span><span class="punctuation">)</span></span><br><span class="line">  gene_name     female     male    logFC</span><br><span class="line"><span class="number">1</span>   CG32548 <span class="number">0.02310383</span> <span class="number">72.43205</span> <span class="number">11.61428</span></span><br><span class="line"><span class="number">2</span>   CG15892 <span class="number">0.02624160</span> <span class="number">57.22716</span> <span class="number">11.09063</span></span><br><span class="line"><span class="number">3</span>   CR43803 <span class="number">0.02474626</span> <span class="number">34.09726</span> <span class="number">10.42823</span></span><br><span class="line"><span class="operator">&gt;</span> <span class="comment">#GSEA的基因排序列表</span></span><br><span class="line"><span class="operator">&gt;</span> genelist <span class="operator">&lt;-</span> mydata<span class="operator">$</span>logFC <span class="comment">#numeric vector</span></span><br><span class="line"><span class="operator">&gt;</span> <span class="built_in">names</span><span class="punctuation">(</span>genelist<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>mydata<span class="operator">$</span>gene_name<span class="punctuation">)</span> <span class="comment">#named vector</span></span><br><span class="line"><span class="operator">&gt;</span> genelist <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>genelist<span class="punctuation">,</span>decreasing<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">)</span> <span class="comment">#decreasing order</span></span><br><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>genelist<span class="punctuation">)</span></span><br><span class="line"> CG32548  CG15892  CR43803  CG15136   CG4983  CG13989 </span><br><span class="line"><span class="number">11.61428</span> <span class="number">11.09063</span> <span class="number">10.42823</span> <span class="number">10.34305</span> <span class="number">10.29130</span> <span class="number">10.00569</span></span><br></pre></td></tr></table></figure>

<h4 id="6-1-3-2-GSEA分析"><a href="#6-1-3-2-GSEA分析" class="headerlink" title="6.1.3.2. GSEA分析"></a>6.1.3.2. GSEA分析</h4><ol>
<li><p>读取输入文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span><span class="string">&quot;geneList.csv&quot;</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>支持物种的标准分析 gseGO()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">gsego <span class="operator">&lt;-</span> gseGO<span class="punctuation">(</span>geneList     <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">              OrgDb        <span class="operator">=</span> org.At.tair.db<span class="punctuation">,</span></span><br><span class="line">              ont          <span class="operator">=</span> <span class="string">&quot;MF&quot;</span><span class="punctuation">,</span> <span class="comment"># GO子类，包含&quot;BP&quot;,&quot;MF&quot;,&quot;CC&quot;,&quot;ALL&quot;</span></span><br><span class="line">              keyType      <span class="operator">=</span> <span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">,</span>  <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">              nPerm        <span class="operator">=</span> <span class="number">1000</span><span class="punctuation">,</span> <span class="comment"># permutation numbers 置换次数</span></span><br><span class="line">              minGSSize    <span class="operator">=</span> <span class="number">100</span><span class="punctuation">,</span> <span class="comment"># 富集基因数量的下限</span></span><br><span class="line">              maxGSSize    <span class="operator">=</span> <span class="number">500</span><span class="punctuation">,</span> <span class="comment"># 富集基因数量的上限</span></span><br><span class="line">              pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span></span><br><span class="line">              verbose      <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> <span class="comment"># 不输出结果</span></span><br><span class="line">              by           <span class="operator">=</span> <span class="string">&quot;fgsea&quot;</span><span class="punctuation">)</span> <span class="comment"># 选项fgsea或DOSE</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>gsego<span class="punctuation">,</span><span class="number">1</span><span class="punctuation">)</span>;<span class="built_in">dim</span><span class="punctuation">(</span>gsego<span class="punctuation">)</span></span><br><span class="line">                   ID        Description setSize enrichmentScore</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> GO<span class="operator">:</span><span class="number">0003674</span> molecular_function     <span class="number">323</span>       <span class="number">0.9462411</span></span><br><span class="line">                NES      pvalue    p.adjust qvalues rank</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> <span class="number">1.716837</span> <span class="number">0.000999001</span> <span class="number">0.000999001</span>      <span class="literal">NA</span>  <span class="number">579</span></span><br><span class="line">                              leading_edge</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> tags<span class="operator">=</span><span class="number">100</span><span class="operator">%, list=11%</span><span class="punctuation">,</span> signal<span class="operator">=</span><span class="number">95</span>%</span><br><span class="line">core_enrichment</span><br><span class="line">GO<span class="operator">:</span><span class="number">0003674</span> CG15892<span class="operator">/</span>CG15136<span class="operator">/</span>CG4983<span class="operator">/</span>CG43851<span class="operator">/</span>…</span><br><span class="line"><span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span> <span class="number">53</span> <span class="number">11</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>data.frame<span class="punctuation">(</span>gsego<span class="operator">$</span>ID<span class="punctuation">,</span>gsego<span class="operator">$</span>Description<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">   gsego.ID gsego.Description</span><br><span class="line"><span class="number">1</span> GO<span class="operator">:</span><span class="number">0003674</span> molecular_function</span><br><span class="line"><span class="number">2</span> GO<span class="operator">:</span><span class="number">0005488</span>            binding</span><br></pre></td></tr></table></figure>
</li>
<li><p>提供注释文件go_annotation.txt的非模式物种分析 GSEA()</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;go_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">gsego <span class="operator">&lt;-</span> GSEA<span class="punctuation">(</span>genes<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>gsego结果文件解释</p>
</li>
</ol>
<ul>
<li>ID: Gene Ontology数据库中唯一的标号信息</li>
<li>Description：Gene Ontology功能的描述信息</li>
<li>setSize</li>
<li>enrichmentScore：富集分数</li>
<li>NES</li>
<li>pvalue: 富集分析统计学显著水平，一般情况下， P-value &lt; 0.05 该功能为富集项</li>
<li>p.adjust：矫正后的P-Value</li>
<li>qvalues：对p值进行统计学检验的q值</li>
<li>rank</li>
<li>leading_edge</li>
<li>core_enrichment</li>
</ul>
<h3 id="6-1-4-GO富集结果可视化"><a href="#6-1-4-GO富集结果可视化" class="headerlink" title="6.1.4. GO富集结果可视化"></a>6.1.4. GO富集结果可视化</h3><p><code>goplot(ego)</code>简单可视化结果为有向无环图。</p>
<h2 id="6-2-KEGG富集分析"><a href="#6-2-KEGG富集分析" class="headerlink" title="6.2. KEGG富集分析"></a>6.2. KEGG富集分析</h2><p>clusterProfiler通过<a target="_blank" rel="noopener" href="https://www.kegg.jp/kegg/rest/keggapi.html">KEGG数据库的API</a>来获取KEGG的注释信息，包括一个物种所有基因对应的pathway注释文件，比如人的：<a target="_blank" rel="noopener" href="http://rest.kegg.jp/link/hsa/pathway%EF%BC%9B%E5%92%8Cpathway%E5%AF%B9%E5%BA%94%E7%9A%84%E6%8F%8F%E8%BF%B0%E4%BF%A1%E6%81%AF%EF%BC%8C%E6%AF%94%E5%A6%82%E4%BA%BA%E7%9A%84%EF%BC%9Ahttp://rest.kegg.jp/list/pathway/hsa%E3%80%82">http://rest.kegg.jp/link/hsa/pathway；和pathway对应的描述信息，比如人的：http://rest.kegg.jp/list/pathway/hsa。</a></p>
<h3 id="6-2-1-支持的物种"><a href="#6-2-1-支持的物种" class="headerlink" title="6.2.1. 支持的物种"></a>6.2.1. 支持的物种</h3><ol>
<li>clusterProfiler包支持的物种<br>只需要将物种缩写输入给clusterProfiler，clusterProfiler包支持自动联网调取<a target="_blank" rel="noopener" href="https://www.genome.jp/kegg/catalog/org_list.html">kegg注释物种</a>列出物种的pathway注释信息，网站可以查看物种列表和缩写，或者用clusterProfiler包提供search_kegg_organism()函数来帮助搜索支持的生物。</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">search_kegg_organism<span class="punctuation">(</span><span class="string">&#x27;osa&#x27;</span><span class="punctuation">,</span> by<span class="operator">=</span><span class="string">&#x27;kegg_code&#x27;</span><span class="punctuation">)</span> <span class="comment">#查询kegg_code为osa的记录，水稻</span></span><br><span class="line">search_kegg_organism<span class="punctuation">(</span><span class="string">&#x27;Escherichia coli&#x27;</span><span class="punctuation">,</span> by<span class="operator">=</span><span class="string">&#x27;scientific_name&#x27;</span><span class="punctuation">)</span> <span class="comment">#查询学名为Escherichia coli的记录</span></span><br></pre></td></tr></table></figure>

<ol start="2">
<li>用户提供KEGG pathway注释数据<br>如果分析的物种不支持自动调取，可以自己做KEGG pathway注释后提供注释文件，比如interproscan、eggnog-mapper等软件的注释结果，整理成clusterProfiler支持的输入格式即可。</li>
</ol>
<p>clusterProfiler需要导入的KEGG pathway注释文件pathway_annotation.txt的格式如下：</p>
<table>
<thead>
<tr>
<th>GeneID</th>
<th>Pathway</th>
<th>Path_Description</th>
</tr>
</thead>
<tbody><tr>
<td>1</td>
<td>ko:00001</td>
<td>spindle</td>
</tr>
<tr>
<td>2</td>
<td>ko:00002</td>
<td>mitotic spindle</td>
</tr>
<tr>
<td>3</td>
<td>ko:00003</td>
<td>kinetochore</td>
</tr>
</tbody></table>
<p>data.frame格式，包含三列，第一列为Gene ID，第二列为 KEGG Pathway ID，第三列为Path_Description，顺序无要求。</p>
<h3 id="6-2-2-KEGG-pathway的ORA分析"><a href="#6-2-2-KEGG-pathway的ORA分析" class="headerlink" title="6.2.2. KEGG pathway的ORA分析"></a>6.2.2. KEGG pathway的ORA分析</h3><h4 id="6-2-2-1-KEGG输入ID的格式转换"><a href="#6-2-2-1-KEGG输入ID的格式转换" class="headerlink" title="6.2.2.1. KEGG输入ID的格式转换"></a>6.2.2.1. KEGG输入ID的格式转换</h4><p>ID转换函数</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>clusterProfiler<span class="punctuation">)</span></span><br><span class="line">bitr_kegg<span class="punctuation">(</span><span class="string">&quot;1&quot;</span><span class="punctuation">,</span>fromType <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span>toType <span class="operator">=</span> <span class="string">&#x27;ncbi-proteinid&#x27;</span><span class="punctuation">,</span>organism<span class="operator">=</span><span class="string">&#x27;hsa&#x27;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">library<span class="punctuation">(</span>org.Hs.eg.db<span class="punctuation">)</span></span><br><span class="line">keytypes<span class="punctuation">(</span>org.Hs.eg.db<span class="punctuation">)</span> <span class="comment">#查看支持的ID类型</span></span><br><span class="line">bitr<span class="punctuation">(</span>gene<span class="punctuation">,</span> fromType <span class="operator">=</span> <span class="string">&quot;ENTREZID&quot;</span><span class="punctuation">,</span> toType <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;ENSEMBL&quot;</span><span class="punctuation">,</span> <span class="string">&quot;SYMBOL&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> OrgDb <span class="operator">=</span> org.Hs.eg.db<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#以上看出ID转换输入时，可以向量的形式，也可以单列基因名list导入</span></span><br><span class="line">gene <span class="operator">&lt;-</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;AASDH&quot;</span><span class="punctuation">,</span><span class="string">&quot;ABCB11&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAM12&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAMTS16&quot;</span><span class="punctuation">,</span><span class="string">&quot;ADAMTS18&quot;</span><span class="punctuation">)</span></span><br><span class="line">genes  <span class="operator">&lt;-</span>  gene<span class="operator">$</span>V1 <span class="comment">#字符串</span></span><br><span class="line"><span class="comment">#也可以是内置数据</span></span><br><span class="line">data<span class="punctuation">(</span>geneList<span class="punctuation">,</span> package<span class="operator">=</span><span class="string">&quot;DOSE&quot;</span><span class="punctuation">)</span> <span class="comment">#富集分析的背景基因集</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span><span class="punctuation">[</span><span class="built_in">abs</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&gt;</span> <span class="number">2</span><span class="punctuation">]</span></span><br></pre></td></tr></table></figure>

<h4 id="6-2-2-2-KEGG-pathway的ORA分析"><a href="#6-2-2-2-KEGG-pathway的ORA分析" class="headerlink" title="6.2.2.2. KEGG pathway的ORA分析"></a>6.2.2.2. KEGG pathway的ORA分析</h4><p>输入文件与GO的ORA分析输入文件一样。</p>
<ol>
<li>导入输入文件</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene.list&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span> <span class="comment">#读取gene ID list，内容为一列ENSEMBL格式的基因ID名称</span></span><br><span class="line">genes <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>data<span class="operator">$</span>V1<span class="punctuation">)</span> <span class="comment">#转换成字符格式</span></span><br></pre></td></tr></table></figure>

<ol start="2">
<li>支持物种的标准分析 enrichKEGG()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">kk <span class="operator">&lt;-</span> enrichKEGG<span class="punctuation">(</span>gene         <span class="operator">=</span> genes<span class="punctuation">,</span></span><br><span class="line">                 keyType      <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">                 organism     <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                 pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span></span><br><span class="line">                 pAdjustMethod<span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span></span><br><span class="line">                 qvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>kk<span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span></span><br><span class="line">              ID                                      Description GeneRatio  BgRatio</span><br><span class="line">hsa04750 hsa04750 Inflammatory mediator regulation of TRP channels      <span class="number">5</span><span class="operator">/</span><span class="number">53</span>  <span class="number">97</span><span class="operator">/</span><span class="number">7387</span></span><br><span class="line">hsa04020 hsa04020                        Calcium signaling pathway      <span class="number">6</span><span class="operator">/</span><span class="number">53</span> <span class="number">182</span><span class="operator">/</span><span class="number">7387</span></span><br><span class="line">               pvalue   p.adjust    qvalue                              geneID Count</span><br><span class="line">hsa04750 <span class="number">0.0006135305</span> <span class="number">0.08589427</span> <span class="number">0.0807277</span>             <span class="number">40</span><span class="operator">/</span><span class="number">3556</span><span class="operator">/</span><span class="number">3708</span><span class="operator">/</span><span class="number">5608</span><span class="operator">/</span><span class="number">79054</span>     <span class="number">5</span></span><br><span class="line">hsa04020 <span class="number">0.0018078040</span> <span class="number">0.12654628</span> <span class="number">0.1189345</span>        <span class="number">493</span><span class="operator">/</span><span class="number">1129</span><span class="operator">/</span><span class="number">2066</span><span class="operator">/</span><span class="number">3707</span><span class="operator">/</span><span class="number">3708</span><span class="operator">/</span><span class="number">4842</span>     <span class="number">6</span></span><br></pre></td></tr></table></figure>

<p>输入的ID类型默认是kegg gene id，也可以是ncbi-geneid或ncbi-proteinid或者uniprot等，可以通过上一步格式转换转换ID类型。<br>organism对应物种的三字母缩写，其他参数与GO的ORA分析参数一致。</p>
<p>不像enrichGO()函数有readable()参数，enrichKEGG()函数没有这个参数，当待分析物种在OrgDb数据库中可用时，可以用setReadable()函数把gene ID转换成gene Name。</p>
<ol start="3">
<li>提供注释文件pathway_annotation.txt的非模式物种分析 enricher()</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">pathway_anno <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;pathway_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> pathway_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> pathway_anno<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">kk <span class="operator">&lt;-</span> enricher<span class="punctuation">(</span>gene<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span>TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="4">
<li>结果输出到csv文件<br><code>write.csv(summary(kk),&quot;KEGG-enrich.csv&quot;,row.names =FALSE)</code></li>
</ol>
<h3 id="6-2-3-KEGG-pathway的GSEA分析"><a href="#6-2-3-KEGG-pathway的GSEA分析" class="headerlink" title="6.2.3. KEGG pathway的GSEA分析"></a>6.2.3. KEGG pathway的GSEA分析</h3><ol>
<li><p>读取输入文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span><span class="string">&quot;geneList.csv&quot;</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>geneList<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">as.character</span><span class="punctuation">(</span>d<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">geneList <span class="operator">&lt;-</span> sort<span class="punctuation">(</span>geneList<span class="punctuation">,</span> decreasing <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>支持物种的标准分析 gseKEGG()</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">kks <span class="operator">&lt;-</span> gseKEGG<span class="punctuation">(</span>geneList     <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">               keyType      <span class="operator">=</span> <span class="string">&quot;kegg&quot;</span><span class="punctuation">,</span> <span class="comment"># 输入基因的类型，命令keytypes(org.Hs.eg.db)会列出可用的所有类型；</span></span><br><span class="line">               organism     <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">               nPerm        <span class="operator">=</span> <span class="number">1000</span><span class="punctuation">,</span></span><br><span class="line">               minGSSize    <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span></span><br><span class="line">               maxGSSize    <span class="operator">=</span> <span class="number">500</span><span class="punctuation">,</span></span><br><span class="line">               pvalueCutoff <span class="operator">=</span> <span class="number">0.05</span><span class="punctuation">,</span></span><br><span class="line">               pAdjustMethod<span class="operator">=</span> <span class="string">&quot;BH&quot;</span><span class="punctuation">,</span></span><br><span class="line">               verbose      <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>kk2<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>提供注释文件pathway_annotation.txt的非模式物种分析 GSEA()<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">data <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;pathway_annotation.txt&quot;</span><span class="punctuation">,</span>header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">)</span></span><br><span class="line">go2gene <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">go2name <span class="operator">&lt;-</span> data<span class="punctuation">[</span><span class="punctuation">,</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">2</span><span class="punctuation">,</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">kks <span class="operator">&lt;-</span> GSEA<span class="punctuation">(</span>gene<span class="punctuation">,</span> TERM2GENE <span class="operator">=</span> go2gene<span class="punctuation">,</span> TERM2NAME <span class="operator">=</span> go2name<span class="punctuation">)</span></span><br></pre></td></tr></table></figure></li>
</ol>
<h3 id="6-2-4-KEGG-module的ORA分析"><a href="#6-2-4-KEGG-module的ORA分析" class="headerlink" title="6.2.4. KEGG module的ORA分析"></a>6.2.4. KEGG module的ORA分析</h3><p><a target="_blank" rel="noopener" href="https://www.genome.jp/kegg/module.html">KEGG Module</a>是手动定义的功能单元的集合。在某些情况下，KEGG 模块有更直接的解释。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">mkk <span class="operator">&lt;-</span> enrichMKEGG<span class="punctuation">(</span>gene <span class="operator">=</span> gene<span class="punctuation">,</span></span><br><span class="line">                   organism <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                   pvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">,</span></span><br><span class="line">                   qvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>mkk<span class="punctuation">)</span></span><br><span class="line"><span class="comment">##            ID                                             Description GeneRatio</span></span><br><span class="line"><span class="comment">## M00912 M00912      NAD biosynthesis, tryptophan =&gt; quinolinate =&gt; NAD       2/9</span></span><br><span class="line"><span class="comment">## M00095 M00095          C5 isoprenoid biosynthesis, mevalonate pathway       1/9</span></span><br><span class="line"><span class="comment">## M00053 M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP =&gt; dCTP       1/9</span></span><br><span class="line"><span class="comment">## M00938 M00938 Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP       1/9</span></span><br><span class="line"><span class="comment">## M00003 M00003            Gluconeogenesis, oxaloacetate =&gt; fructose-6P       1/9</span></span><br><span class="line"><span class="comment">## M00049 M00049     Adenine ribonucleotide biosynthesis, IMP =&gt; ADP,ATP       1/9</span></span><br><span class="line"><span class="comment">##        BgRatio      pvalue   p.adjust     qvalue     geneID Count</span></span><br><span class="line"><span class="comment">## M00912  12/829 0.006541756 0.03925053 0.03443029 23475/3620     2</span></span><br><span class="line"><span class="comment">## M00095  10/829 0.103949536 0.20710097 0.18166751       3158     1</span></span><br><span class="line"><span class="comment">## M00053  11/829 0.113796244 0.20710097 0.18166751       6241     1</span></span><br><span class="line"><span class="comment">## M00938  14/829 0.142761862 0.20710097 0.18166751       6241     1</span></span><br><span class="line"><span class="comment">## M00003  18/829 0.180072577 0.20710097 0.18166751       5105     1</span></span><br><span class="line"><span class="comment">## M00049  21/829 0.207100966 0.20710097 0.18166751      26289     1             </span></span><br></pre></td></tr></table></figure>

<h3 id="6-2-5-KEGG-module的GSEA分析"><a href="#6-2-5-KEGG-module的GSEA分析" class="headerlink" title="6.2.5. KEGG module的GSEA分析"></a>6.2.5. KEGG module的GSEA分析</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">mkk2 <span class="operator">&lt;-</span> gseMKEGG<span class="punctuation">(</span>geneList <span class="operator">=</span> geneList<span class="punctuation">,</span></span><br><span class="line">                 organism <span class="operator">=</span> <span class="string">&#x27;hsa&#x27;</span><span class="punctuation">,</span></span><br><span class="line">                 pvalueCutoff <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>mkk2<span class="punctuation">)</span></span><br><span class="line"><span class="comment">##            ID                                                      Description</span></span><br><span class="line"><span class="comment">## M00001 M00001        Glycolysis (Embden-Meyerhof pathway), glucose =&gt; pyruvate</span></span><br><span class="line"><span class="comment">## M00035 M00035                                           Methionine degradation</span></span><br><span class="line"><span class="comment">## M00002 M00002         Glycolysis, core module involving three-carbon compounds</span></span><br><span class="line"><span class="comment">## M00938 M00938          Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP</span></span><br><span class="line"><span class="comment">## M00104 M00104 Bile acid biosynthesis, cholesterol =&gt; cholate/chenodeoxycholate</span></span><br><span class="line"><span class="comment">## M00009 M00009                           Citrate cycle (TCA cycle, Krebs cycle)</span></span><br><span class="line"><span class="comment">##        setSize enrichmentScore       NES      pvalue  p.adjust   qvalues rank</span></span><br><span class="line"><span class="comment">## M00001      24       0.5739036  1.683936 0.005050296 0.1616095 0.1488508 2886</span></span><br><span class="line"><span class="comment">## M00035      10       0.6784636  1.637462 0.025966367 0.2710124 0.2496167 1555</span></span><br><span class="line"><span class="comment">## M00002      11       0.6421781  1.585841 0.032455383 0.2710124 0.2496167 1381</span></span><br><span class="line"><span class="comment">## M00938      10       0.6648004  1.604486 0.033876549 0.2710124 0.2496167  648</span></span><br><span class="line"><span class="comment">## M00104      10      -0.5876900 -1.338409 0.128747795 0.6564103 0.6045884  961</span></span><br><span class="line"><span class="comment">## M00009      22       0.4504911  1.315039 0.138271605 0.6564103 0.6045884 3514</span></span><br><span class="line"><span class="comment">##                          leading_edge</span></span><br><span class="line"><span class="comment">## M00001 tags=54%, list=23%, signal=42%</span></span><br><span class="line"><span class="comment">## M00035 tags=50%, list=12%, signal=44%</span></span><br><span class="line"><span class="comment">## M00002 tags=55%, list=11%, signal=49%</span></span><br><span class="line"><span class="comment">## M00938  tags=40%, list=5%, signal=38%</span></span><br><span class="line"><span class="comment">## M00104  tags=50%, list=8%, signal=46%</span></span><br><span class="line"><span class="comment">## M00009 tags=50%, list=28%, signal=36%</span></span><br><span class="line"><span class="comment">##                                                         core_enrichment</span></span><br><span class="line"><span class="comment">## M00001 5214/3101/2821/7167/2597/5230/2023/5223/5315/3099/5232/2027/5211</span></span><br><span class="line"><span class="comment">## M00035                                           875/1789/191/1788/1786</span></span><br><span class="line"><span class="comment">## M00002                                    7167/2597/5230/2023/5223/5315</span></span><br><span class="line"><span class="comment">## M00938                                              6241/7298/4830/1841</span></span><br><span class="line"><span class="comment">## M00104                                        6342/10998/1581/3295/8309</span></span><br><span class="line"><span class="comment">## M00009            3418/50/4190/3419/2271/3421/55753/3417/1431/6389/4191</span></span><br></pre></td></tr></table></figure>

<h3 id="6-2-6-KEGG-pathways富集结果的可视化"><a href="#6-2-6-KEGG-pathways富集结果的可视化" class="headerlink" title="6.2.6. KEGG pathways富集结果的可视化"></a>6.2.6. KEGG pathways富集结果的可视化</h3><p>enrichplot包可以实现几种方法，可以用在GO,KEGG,MSigDb等基因集注释上。</p>
<ol>
<li><p>browseKEGG()函数会打开网络浏览器突出显示富集的基因的KEGG通路，用hsa04110基因举例：<br><code>browseKEGG(kk, &#39;hsa04110&#39;)</code></p>
</li>
<li><p>pathview::pathview()可视化富集的KEGG通路</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">library(&quot;pathview&quot;)</span><br><span class="line">hsa04110 &lt;- pathview(gene.data  = geneList,</span><br><span class="line">                     pathway.id = &quot;hsa04110&quot;,</span><br><span class="line">                     species    = &quot;hsa&quot;,</span><br><span class="line">                     limit      = list(gene=max(abs(geneList)), cpd=1))</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="6-3-通用的富集分析-Universal-enrichment-analysis"><a href="#6-3-通用的富集分析-Universal-enrichment-analysis" class="headerlink" title="6.3. 通用的富集分析(Universal enrichment analysis)"></a>6.3. 通用的富集分析(Universal enrichment analysis)</h2><p>除GO,KEGG基因集外，clusterProfiler还支持WikiPathways，Reactome，Disease，MeSH等数据库的富集分析。</p>
<p>虽然clusterProfiler支持对许多ontology&#x2F;pathway的hypergeometric test和gene set enrichment analyses，但是还有很多数据，包括不支持的物种、不支持的ontologies&#x2F;pathways或自定义注释等。<br>clusterProfiler提供了用于hypergeometric test的enricher()函数和用于基因集富集分析的GSEA()函数，用于接受用户定义的注释。</p>
<p>另外两个参数TERM2GENE和TERM2NAME:</p>
<ul>
<li>TERM2GENE是一个必需的data.frame，第一列为term ID，第二列为对应映射基因；</li>
<li>TERM2NAME是一个可选的data.frame，第一列为term ID，第二列为对应term name。</li>
</ul>
<p>在GO和KEGG富集分析的用户自行提供注释文件的分析部分，使用的enricher()和GSEA()函数。</p>
<h1 id="7-功能富集结果可视化"><a href="#7-功能富集结果可视化" class="headerlink" title="7. 功能富集结果可视化"></a>7. 功能富集结果可视化</h1><h2 id="7-1-enrichplot包"><a href="#7-1-enrichplot包" class="headerlink" title="7.1. enrichplot包"></a>7.1. enrichplot包</h2><p>enrichplot包有几种可视化方法来解释富集结果，支持clusterProfiler获得的ORA和GSEA富集结果。</p>
<ol>
<li><p>安装和载入enrichplot包</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;enrichplot&quot;</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>enrichplot<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
</li>
<li><p>可视化 —— 条形图<br>将富集分数（例如p 值）和基因计数或比率描述为条形高度和颜色。横轴为该GO term下的差异基因个数，纵轴为富集到的GO Terms的描述信息， showCategory指定展示的GO Terms的个数为20个，默认展示显著富集的top10个，即p.adjust最小的10个。</p>
</li>
</ol>
<p><code>barplot(ego, showCategory=20, title=&quot;EnrichmentGO_MF&quot;)</code></p>
<p>使用mutate导出的其他变量也可以用作条形高度或颜色。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mutate<span class="punctuation">(</span>edo<span class="punctuation">,</span> qscore <span class="operator">=</span> <span class="operator">-</span><span class="built_in">log</span><span class="punctuation">(</span>p.adjust<span class="punctuation">,</span> base<span class="operator">=</span><span class="number">10</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">%&gt;%</span> </span><br><span class="line">    barplot<span class="punctuation">(</span>x<span class="operator">=</span><span class="string">&quot;qscore&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<ol start="3">
<li>可视化 —— 点阵图<br><code>dotplot(ego, showCategory = 10, title=&quot;EnrichmentGO_MF_dot&quot;)</code></li>
</ol>
<p>散点图，横坐标为GeneRatio，纵坐标为富集到的GO Terms的描述信息，showCategory指定展示的GO Terms的个数，默认展示显著富集的top10个，即p.adjust最小的10个。</p>
<ol start="4">
<li>可视化 —— 类别网络图<br>cnetplot 将基因和生物学概念（例如 GO 术语或 KEGG 通路）的联系描述为一个网络（有助于查看哪些基因涉及富集通路和可能属于多个注释类别的基因）。对于基因和富集的GO terms之间的对应关系进行展示，如果一个基因位于一个GO Terms下，则将该基因与GO连线。图中灰色的点代表基因，黄色的点代表富集到的GO terms, 默认画top5富集到的GO terms, GO 节点的大小对应富集到的基因个数。</li>
</ol>
<p><code>cnetplot(ego, categorySize = &quot;pvalue&quot;, foldChange = gene_list</code></p>
<ol start="5">
<li>可视化 —— 类热图功能分类</li>
</ol>
<p>heatplot类似cnetplot，而显示为热图的关系。如果用户想要显示大量重要术语，那么类别网络图可能会过于复杂。在heatplot能够简化结果和更容易识别的表达模式。</p>
<p><code>heatplot(ego, foldChange=geneList, showCategory=5)</code></p>
<ol start="6">
<li>可视化 —— 树状图<br>treeplot()函数执行丰富术语的层次聚类。它依赖于pairwise_termsim()函数计算的丰富项的成对相似性，默认情况下使用 Jaccard 的相似性指数 (JC)。如果支持，用户还可以使用语义相似度值（例如，GO、DO和MeSH）。</li>
</ol>
<p>默认聚合方法treeplot()是ward.D，用户可以通过hclust_method参数指定其他方法（例如，’average’、’complete’、’median’、’centroid’等。</p>
<p>treeplot()函数会将树切割成几个子树（由nCluster参数指定（默认为 5））并使用高频词标记子树。</p>
<p><code>ego2 &lt;- pairwise_termsim(ego)</code>;<br><code>treeplot(ego2, hclust_method = &quot;average&quot;)</code></p>
<ol start="7">
<li>可视化 —— 富集图<br>对于富集到的GO terms之间的基因重叠关系进行展示，如果两个GO terms系的差异基因存在重叠，说明这两个节点存在overlap关系，在图中用线条连接起来。每个节点是一个富集到的GO term, 默认画top30个富集到的GO terms, 节点大小对应该GO terms下富集到的差异基因个数，节点的颜色对应p.adjust的值，从小到大，对应蓝色到红色。</li>
</ol>
<p><code>ego2 &lt;- pairwise_termsim(ego)</code>;<br><code>emapplot(ego2, cex_category=1.5, layout=&quot;kk&quot;)</code></p>
<ol start="8">
<li>可视化 —— upset图<br>upsetplot是cnetplot可视化基因和基因集之间复杂关联的替代方法。它强调不同基因集之间的基因重叠。</li>
</ol>
<p><code>upsetplot(ego)</code></p>
<ol start="9">
<li>可视化 —— 脊线图<br>ridgeplot将可视化核心富集基因的表达分布为GSEA富集类别。它帮助用户解释上调&#x2F;下调的途径。</li>
</ol>
<p><code>ridgeplot(ego)</code></p>
<h2 id="7-2-可视化-——-有向无环图"><a href="#7-2-可视化-——-有向无环图" class="headerlink" title="7.2. 可视化 —— 有向无环图"></a>7.2. 可视化 —— 有向无环图</h2><ol>
<li><p><code>plotGOgraph(ego)</code><br>矩形代表富集到的top10个GO Terms，颜色从黄到红，对应p值从大到小。</p>
</li>
<li><p><code>goplot(ego, showCategory = 10)</code><br>igraph布局方式的有向无环图</p>
</li>
</ol>
<h2 id="7-3-可视化-——-wordcloud"><a href="#7-3-可视化-——-wordcloud" class="headerlink" title="7.3. 可视化 —— wordcloud"></a>7.3. 可视化 —— wordcloud</h2><p>词云的方式显示结果</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wcdf <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span>text <span class="operator">=</span> ego<span class="operator">$</span>GeneRatio<span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&quot;/&quot;</span><span class="punctuation">)</span><span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span></span><br><span class="line">wcdf<span class="operator">$</span>term <span class="operator">&lt;-</span>  ego<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span></span><br><span class="line">wordcloud<span class="punctuation">(</span>words <span class="operator">=</span> wcdf<span class="operator">$</span>term<span class="punctuation">,</span> freq <span class="operator">=</span> wcdf<span class="operator">$</span>V1<span class="punctuation">,</span> scale<span class="operator">=</span><span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="number">4</span><span class="punctuation">,</span> <span class="number">.1</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> colors<span class="operator">=</span>brewer.pal<span class="punctuation">(</span><span class="number">8</span><span class="punctuation">,</span> <span class="string">&quot;Dark2&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> max.words <span class="operator">=</span> <span class="number">25</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<h1 id="8-references"><a href="#8-references" class="headerlink" title="8. references"></a>8. references</h1><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Gene_set_enrichment_analysis">GSEA wiki</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/47b5ea646932">enrichment</a><br><a target="_blank" rel="noopener" href="https://github.com/YuLab-SMU/clusterProfiler">clusterProfiler github</a><br><a target="_blank" rel="noopener" href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">universal enrichment analysis using clusterProfiler</a><br><a target="_blank" rel="noopener" href="https://www.cell.com/the-innovation/fulltext/S2666-6758(21)00066-7?_returnURL=https://linkinghub.elsevier.com/retrieve/pii/S2666675821000667?showall=true">clusterProfiler paper</a><br><a target="_blank" rel="noopener" href="https://www.cnblogs.com/jessepeng/p/12159139.html">tutorial</a><br><a target="_blank" rel="noopener" href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">函数simplify</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/12/13/bioinfo_GSEA_clusterProfiler/" data-id="cl1aidmwj001j4gvk2nj86gc5" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/12/28/R_install/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          R及相关软件的安装
        
      </div>
    </a>
  
  
    <a href="/2021/12/06/bioinfo_OrthoFinder2/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">用OrthoFinder做直系同源推断</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>