<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="mapping法评估基因组组装质量。这篇文章主要介绍了软件bamdst的安装和使用，可全面统计BAM文件的相关参数，包括mapping rate、coverage和depth。">
<meta property="og:type" content="article">
<meta property="og:title" content="基因组质量评估：（五）mapping法：6. 用软件bamdst统计BAM文件">
<meta property="og:url" content="http://example.com/2022/07/31/omics_genome_quality.assessment_mapping_bamdst/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="mapping法评估基因组组装质量。这篇文章主要介绍了软件bamdst的安装和使用，可全面统计BAM文件的相关参数，包括mapping rate、coverage和depth。">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2022-07-30T16:00:00.000Z">
<meta property="article:modified_time" content="2022-07-31T07:45:21.981Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoft">
<meta property="article:tag" content="genome">
<meta property="article:tag" content="quality assessment">
<meta property="article:tag" content="mapping">
<meta property="article:tag" content="sam">
<meta property="article:tag" content="bam">
<meta property="article:tag" content="depth">
<meta property="article:tag" content="bamdst">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">

<link rel="canonical" href="http://example.com/2022/07/31/omics_genome_quality.assessment_mapping_bamdst/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>基因组质量评估：（五）mapping法：6. 用软件bamdst统计BAM文件 | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --></head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/07/31/omics_genome_quality.assessment_mapping_bamdst/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          基因组质量评估：（五）mapping法：6. 用软件bamdst统计BAM文件
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-07-31 00:00:00 / Modified: 15:45:21" itemprop="dateCreated datePublished" datetime="2022-07-31T00:00:00+08:00">2022-07-31</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/" itemprop="url" rel="index"><span itemprop="name">omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/" itemprop="url" rel="index"><span itemprop="name">genome</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/quality-assessment/" itemprop="url" rel="index"><span itemprop="name">quality assessment</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2022/07/31/omics_genome_quality.assessment_mapping_bamdst/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2022/07/31/omics_genome_quality.assessment_mapping_bamdst/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>9.3k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>8 mins.</span>
            </span>
            <div class="post-description">mapping法评估基因组组装质量。这篇文章主要介绍了软件bamdst的安装和使用，可全面统计BAM文件的相关参数，包括mapping rate、coverage和depth。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=5351733&auto=1&height=32"></iframe></div>

<h1 id="1-bamdst简介"><a href="#1-bamdst简介" class="headerlink" title="1. bamdst简介"></a>1. bamdst简介</h1><p>bamdst（<a target="_blank" rel="noopener" href="https://github.com/shiquan/bamdst%EF%BC%89%E6%98%AF%E4%B8%80%E4%B8%AA%E7%BB%9F%E8%AE%A1bam%E6%96%87%E4%BB%B6%E7%9B%AE%E6%A0%87%E5%8C%BA%E5%9F%9F%E6%B7%B1%E5%BA%A6%E5%92%8C%E8%A6%86%E7%9B%96%E5%BA%A6%E7%9A%84%E5%B7%A5%E5%85%B7%EF%BC%8C%E7%BB%9F%E8%AE%A1%E6%95%B0%E6%8D%AE%E5%85%A8%E9%9D%A2%EF%BC%8C%E5%8C%85%E6%8B%ACcoverage%EF%BC%8Cdepth%EF%BC%8Cmapping">https://github.com/shiquan/bamdst）是一个统计bam文件目标区域深度和覆盖度的工具，统计数据全面，包括coverage，depth，mapping</a> rate。</p>
<h1 id="2-下载安装"><a href="#2-下载安装" class="headerlink" title="2. 下载安装"></a>2. 下载安装</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/shiquan/bamdst.git</span><br><span class="line">cd bamdst</span><br><span class="line">make</span><br><span class="line">./bamdst -h</span><br></pre></td></tr></table></figure>

<h1 id="3-分析"><a href="#3-分析" class="headerlink" title="3. 分析"></a>3. 分析</h1><h2 id="3-1-输入文件"><a href="#3-1-输入文件" class="headerlink" title="3.1. 输入文件"></a>3.1. 输入文件</h2><ol>
<li>sample.bed</li>
</ol>
<ul>
<li>sample.bed文件包含分析的位置，共有三列，tab分隔。包括染色体名称，起始位置，终止位置。</li>
<li>如果计算基因组全部数据，则包含基因组所有染色体的位置。建议用genome.fa.fai索引文件转化生成<code>cat genome.fa.fai |awk &#39;&#123;print $1&quot;\t1\t&quot;$2&#125;&#39; &gt; sample.bed</code>。</li>
</ul>
<ol start="2">
<li>sample.bam</li>
</ol>
<ul>
<li>sample.bam文件需要已排序（samtools sort可实现）</li>
</ul>
<h2 id="3-2-运行"><a href="#3-2-运行" class="headerlink" title="3.2. 运行"></a>3.2. 运行</h2><p><code>bamdst -p sample.bed sample.bam -o ./</code></p>
<ul>
<li>-p sample.bed：指定bed文件</li>
<li>-o .&#x2F; ：输出目录</li>
</ul>
<h2 id="3-3-结果"><a href="#3-3-结果" class="headerlink" title="3.3. 结果"></a>3.3. 结果</h2><ol>
<li>coverage.report</li>
</ol>
<ul>
<li>总结的统计信息，包括reads的mapping rate，genome coverage，average depth等数据。示例：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br></pre></td><td class="code"><pre><span class="line">## The file was created by bamdst</span><br><span class="line">## Version : 1.0.9</span><br><span class="line">## Files : ./sample.bam</span><br><span class="line">                               [Total] Raw Reads (All reads)	86430326</span><br><span class="line">                                       [Total] QC Fail reads	0</span><br><span class="line">                                        [Total] Raw Data(Mb)	10773.87</span><br><span class="line">                                        [Total] Paired Reads	86430326</span><br><span class="line">                                        [Total] Mapped Reads	84639699</span><br><span class="line">                            [Total] Fraction of Mapped Reads	97.93% # mapping rate</span><br><span class="line">                                     [Total] Mapped Data(Mb)	10550.19</span><br><span class="line">                         [Total] Fraction of Mapped Data(Mb)	97.92%</span><br><span class="line">                                     [Total] Properly paired	82169742</span><br><span class="line">                         [Total] Fraction of Properly paired	95.07%</span><br><span class="line">                                [Total] Read and mate paired	84430784</span><br><span class="line">                    [Total] Fraction of Read and mate paired	97.69%</span><br><span class="line">                                          [Total] Singletons	208915</span><br><span class="line">                       [Total] Read and mate map to diff chr	1568692</span><br><span class="line">                                               [Total] Read1	43215163</span><br><span class="line">                                               [Total] Read2	43215163</span><br><span class="line">                                        [Total] Read1(rmdup)	42319713</span><br><span class="line">                                        [Total] Read2(rmdup)	42319986</span><br><span class="line">                                [Total] forward strand reads	42323338</span><br><span class="line">                               [Total] backward strand reads	42316361</span><br><span class="line">                                 [Total] PCR duplicate reads	0</span><br><span class="line">                     [Total] Fraction of PCR duplicate reads	0.00%</span><br><span class="line">                            [Total] Map quality cutoff value	20</span><br><span class="line">                       [Total] MapQuality above cutoff reads	76685486</span><br><span class="line">                 [Total] Fraction of MapQ reads in all reads	88.73%</span><br><span class="line">              [Total] Fraction of MapQ reads in mapped reads	90.60%</span><br><span class="line">                                       [Target] Target Reads	84639699</span><br><span class="line">              [Target] Fraction of Target Reads in all reads	97.93%</span><br><span class="line">           [Target] Fraction of Target Reads in mapped reads	100.00%</span><br><span class="line">                                    [Target] Target Data(Mb)	10457.94</span><br><span class="line">                              [Target] Target Data Rmdup(Mb)	9505.01</span><br><span class="line">                [Target] Fraction of Target Data in all data	97.07%</span><br><span class="line">             [Target] Fraction of Target Data in mapped data	99.13%</span><br><span class="line">                                      [Target] Len of region	256218203</span><br><span class="line">                                      [Target] Average depth	40.82 # average depth</span><br><span class="line">                               [Target] Average depth(rmdup)	37.10</span><br><span class="line">                                     [Target] Coverage (&gt;0x)	99.58% # genome coverage</span><br><span class="line">                                    [Target] Coverage (&gt;=4x)	99.12%</span><br><span class="line">                                   [Target] Coverage (&gt;=10x)	97.41%</span><br><span class="line">                                   [Target] Coverage (&gt;=30x)	68.46%</span><br><span class="line">                                  [Target] Coverage (&gt;=100x)	0.83%</span><br><span class="line">                                [Target] Target Region Count	266</span><br><span class="line">                                [Target] Region covered &gt; 0x	261</span><br><span class="line">                       [Target] Fraction Region covered &gt; 0x	98.12%</span><br><span class="line">                      [Target] Fraction Region covered &gt;= 4x	97.74%</span><br><span class="line">                     [Target] Fraction Region covered &gt;= 10x	94.36%</span><br><span class="line">                     [Target] Fraction Region covered &gt;= 30x	53.01%</span><br><span class="line">                    [Target] Fraction Region covered &gt;= 100x	12.78%</span><br><span class="line">                                          [flank] flank size	200</span><br><span class="line">           [flank] Len of region (not include target region)	256271403</span><br><span class="line">                                       [flank] Average depth	40.79</span><br><span class="line">                                         [flank] flank Reads	84639699</span><br><span class="line">                [flank] Fraction of flank Reads in all reads	97.93%</span><br><span class="line">             [flank] Fraction of flank Reads in mapped reads	100.00%</span><br><span class="line">                                      [flank] flank Data(Mb)	10452.76</span><br><span class="line">                  [flank] Fraction of flank Data in all data	97.02%</span><br><span class="line">               [flank] Fraction of flank Data in mapped data	99.08%</span><br><span class="line">                                      [flank] Coverage (&gt;0x)	99.54%</span><br><span class="line">                                     [flank] Coverage (&gt;=4x)	99.08%</span><br><span class="line">                                    [flank] Coverage (&gt;=10x)	97.36%</span><br><span class="line">                                    [flank] Coverage (&gt;=30x)	68.41%</span><br><span class="line">                                   [flank] Coverage (&gt;=100x)	0.83%</span><br></pre></td></tr></table></figure>

<ul>
<li>每个值具体的解释：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line"> [Total] Raw Reads (All reads) // All reads in the bam file(s).</span><br><span class="line"> [Total] QC Fail reads // Reads number failed QC, this flag is marked by other software,like bwa. See flag in the bam structure.</span><br><span class="line"> [Total] Raw Data(Mb) // Total reads data in the bam file(s).</span><br><span class="line">[Total] Paired Reads // Paired reads numbers.</span><br><span class="line">[Total] Mapped Reads // Mapped reads numbers.</span><br><span class="line">[Total] Fraction of Mapped Reads // Ratio of mapped reads against raw reads.</span><br><span class="line">[Total] Mapped Data(Mb) // Mapped data in the bam file(s).</span><br><span class="line">[Total] Fraction of Mapped Data(Mb) // Ratio of mapped data against raw data.</span><br><span class="line">[Total] Properly paired // Paired reads with properly insert size. See bam format protocol for details.</span><br><span class="line">[Total] Fraction of Properly paired // Ratio of properly paired reads against mapped reads</span><br><span class="line">[Total] Read and mate paired // Read (read1) and mate read (read2) paired.</span><br><span class="line">[Total] Fraction of Read and mate paired // Ratio of read and mate paired against mapped reads</span><br><span class="line">[Total] Singletons // Read mapped but mate read unmapped, and vice versa.</span><br><span class="line">[Total] Read and mate map to diff chr // Read and mate read mapped to different chromosome, usually because mapping error and structure variants.</span><br><span class="line">[Total] Read1 // First reads in mate paired sequencing</span><br><span class="line">[Total] Read2 // Mate reads</span><br><span class="line">[Total] Read1(rmdup) // First reads after remove duplications.</span><br><span class="line">[Total] Read2(rmdup) // Mate reads after remove duplications.</span><br><span class="line">[Total] forward strand reads // Number of forward strand reads.</span><br><span class="line">[Total] backward strand reads // Number of backward strand reads.</span><br><span class="line">[Total] PCR duplicate reads // PCR duplications.</span><br><span class="line">[Total] Fraction of PCR duplicate reads // Ratio of PCR duplications.</span><br><span class="line">[Total] Map quality cutoff value // Cutoff map quality score, this value can be set by -q. default is 20, because some variants caller like GATK only consider high quality reads.</span><br><span class="line">[Total] MapQuality above cutoff reads // Number of reads with higher or equal quality score than cutoff value.</span><br><span class="line">[Total] Fraction of MapQ reads in all reads // Ratio of reads with higher or equal Q score against raw reads.</span><br><span class="line">[Total] Fraction of MapQ reads in mapped reads // Ratio of reads with higher or equal Q score against mapped reads.</span><br><span class="line">[Target] Target Reads // Number of reads covered target region (specified by bed file).</span><br><span class="line">[Target] Fraction of Target Reads in all reads // Ratio of target reads against raw reads.</span><br><span class="line">[Target] Fraction of Target Reads in mapped reads // Ratio of target reads against mapped reads.</span><br><span class="line">[Target] Target Data(Mb) // Total bases covered target region. If a read covered target region partly, only the covered bases will be counted.</span><br><span class="line">[Target] Target Data Rmdup(Mb) // Total bases covered target region after remove PCR duplications. </span><br><span class="line">[Target] Fraction of Target Data in all data // Ratio of target bases against raw bases.</span><br><span class="line">[Target] Fraction of Target Data in mapped data // Ratio of target bases against mapped bases.</span><br><span class="line">[Target] Len of region // The length of target regions.</span><br><span class="line">[Target] Average depth // Average depth of target regions. Calculated by &quot;target bases / length of regions&quot;.</span><br><span class="line">[Target] Average depth(rmdup) // Average depth of target regions after remove PCR duplications.</span><br><span class="line">[Target] Coverage (&gt;0x) // Ratio of bases with depth greater than 0x in target regions, which also means the ratio of covered regions in target regions.</span><br><span class="line">[Target] Coverage (&gt;=4x) // Ratio of bases with depth greater than or equal to 4x in target regions.</span><br><span class="line">[Target] Coverage (&gt;=10x) // Ratio of bases with depth greater than or equal to 10x in target regions.</span><br><span class="line">[Target] Coverage (&gt;=30x) // Ratio of bases with depth greater than or equal to 30x in target regions.</span><br><span class="line">[Target] Coverage (&gt;=100x) // Ratio of bases with depth greater than or equal to 100x in target regions.</span><br><span class="line">[Target] Coverage (&gt;=Nx) // This is addtional line for user self-defined cutoff value, see --cutoffdepth</span><br><span class="line">[Target] Target Region Count // Number of target regions. In normal practise,it is the total number of exomes.</span><br><span class="line">[Target] Region covered &gt; 0x // The number of these regions with average depth greater than 0x.</span><br><span class="line">[Target] Fraction Region covered &gt; 0x // Ratio of these regions with average depth greater than 0x.</span><br><span class="line">[Target] Fraction Region covered &gt;= 4x // Ratio of these regions with average depth greater than or equal to 4x.</span><br><span class="line">[Target] Fraction Region covered &gt;= 10x // Ratio of these regions with average depth greater than or equal to 10x.</span><br><span class="line">[Target] Fraction Region covered &gt;= 30x // Ratio of these regions with average depth greater than or equal to 30x.</span><br><span class="line">[Target] Fraction Region covered &gt;= 100x // Ratio of these regions with average depth greater than or equal to 100x.</span><br><span class="line">[flank] flank size // The flank size will be count. 200 bp in default. Oligos could also capture the nearby regions of target regions.</span><br><span class="line">[flank] Len of region (not include target region) // The length of flank regions (target regions will not be count).</span><br><span class="line">[flank] Average depth // Average depth of flank regions.</span><br><span class="line">[flank] flank Reads // The total number of reads covered the flank regions. Note: some reads covered the edge of target regions, will be count in flank regions also. </span><br><span class="line">[flank] Fraction of flank Reads in all reads // Ratio of reads covered in flank regions against raw reads.</span><br><span class="line">[flank] Fraction of flank Reads in mapped reads // Ration of reads covered in flank regions against mapped reads.</span><br><span class="line">[flank] flank Data(Mb) // Total bases in the flank regions.</span><br><span class="line">[flank] Fraction of flank Data in all data // Ratio of total bases in the flank regions against raw data.</span><br><span class="line">[flank] Fraction of flank Data in mapped data // Ratio of total bases in the flank regions against mapped data.</span><br><span class="line">[flank] Coverage (&gt;0x) // Ratio of flank bases with depth greater than 0x.</span><br><span class="line">[flank] Coverage (&gt;=4x) // Ratio of flank bases with depth greater than or equal to 4x.</span><br><span class="line">[flank] Coverage (&gt;=10x) // Ratio of flank bases with depth greater than or equal to 10x.</span><br><span class="line">[flank] Coverage (&gt;=30x) // Ratio of flank bases with depth greater than or equal to 30x.</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>chromosomes.report</li>
</ol>
<ul>
<li>每条染色体的depth和coverage信息，示例：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">#Chromosome	   DATA(%)	 Avg depth	    Median	 Coverage%	  Cov 4x %	 Cov 10x %	 Cov 30x %	Cov 100x %</span><br><span class="line">  MCscaf061	    0.00	   42.69	     41.0	  100.00	  100.00	   99.30	   86.09	    0.00</span><br><span class="line">  MCscaf062	    0.02	   66.31	     37.0	  100.0099.85	   99.26	   71.85	    0.40</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>insertsize.plot</li>
</ol>
<ul>
<li>推断的insert size分布。示例：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">0	0	0.000000	41302164	1.000000</span><br><span class="line">1	0	0.000000	41302164	1.000000</span><br><span class="line">2	169	0.000004	41301995	0.999996</span><br><span class="line">3	154	0.000004	41301841	0.999992</span><br><span class="line">4	179	0.000004	41301662	0.999988</span><br><span class="line">5	220	0.000005	41301442	0.999982</span><br></pre></td></tr></table></figure>

<ol start="4">
<li>uncover.bed</li>
</ol>
<ul>
<li>文件包含sample.bam在sample.bed上bad covered或uncovered region的区域。示例：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">MCscaf062	3308	3324</span><br><span class="line">MCscaf062	13172	13219</span><br><span class="line">MCscaf063	1966	1988</span><br><span class="line">MCscaf064	154	177</span><br><span class="line">MCscaf065	4380	4388</span><br><span class="line">MCscaf065	15265	15278</span><br><span class="line">MCscaf065	49992	50000</span><br></pre></td></tr></table></figure>

<ol start="5">
<li>depth_distribution.plot</li>
</ol>
<ul>
<li>示例：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">0	1080969	0.004219	255137234	0.995781</span><br><span class="line">1	375560	0.001466	254761674	0.994315</span><br><span class="line">2	376656	0.001470	254385018	0.992845</span><br><span class="line">3	420120	0.001640	253964898	0.991206</span><br><span class="line">4	471142	0.001839	253493756	0.989367</span><br><span class="line">5	548760	0.002142	252944996	0.987225</span><br></pre></td></tr></table></figure>

<ol start="6">
<li>depth.tsv.gz</li>
</ol>
<p>文件包含输入的sample.bed的每个位置的三种深度：</p>
<ul>
<li>raw depth：没有过滤的从bam文件直接提取的深度。coverage.report文件是用raw depth统计得到的。</li>
<li>rmdup depth：过滤掉duplicated reads，secondary alignment reads，low map quality reads(mapQ&lt;20)后计算的depth。开发者说类似<code>samtools depth</code>的结果，但应该是raw depth类似<code>samtools depth</code>，rmdup depth类似<code>samtools mpileup</code>才更准确。如果想用rmdup depth统计coverage.report，运行时加上参数“–use_rmdup”。</li>
<li>coverage depth：考虑deletion区域的raw depth，所以值会大于或等于raw depth的值。</li>
</ul>
<ol start="7">
<li>region.tsv.gz</li>
</ol>
<ul>
<li>文件包含输入的sample.bed的每个区域的average depth，median depth，coverage。</li>
</ul>
<h1 id="4-references"><a href="#4-references" class="headerlink" title="4. references"></a>4. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://github.com/shiquan/bamdst">https://github.com/shiquan/bamdst</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=50% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/biosoft/" rel="tag"># biosoft</a>
              <a href="/tags/genome/" rel="tag"># genome</a>
              <a href="/tags/quality-assessment/" rel="tag"># quality assessment</a>
              <a href="/tags/mapping/" rel="tag"># mapping</a>
              <a href="/tags/sam/" rel="tag"># sam</a>
              <a href="/tags/bam/" rel="tag"># bam</a>
              <a href="/tags/depth/" rel="tag"># depth</a>
              <a href="/tags/bamdst/" rel="tag"># bamdst</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/07/31/omics_genome_quality.assessment_mapping_mosdepth/" rel="prev" title="基因组质量评估：（五）mapping法：7. 用软件mosdepth统计BAM文件的深度">
      <i class="fa fa-chevron-left"></i> 基因组质量评估：（五）mapping法：7. 用软件mosdepth统计BAM文件的深度
    </a></div>
      <div class="post-nav-item">
    <a href="/2022/07/31/omics_genome_quality.assessment_mapping_QualiMap/" rel="next" title="基因组质量评估：（五）mapping法：5. 用软件QualiMap统计BAM文件">
      基因组质量评估：（五）mapping法：5. 用软件QualiMap统计BAM文件 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-bamdst%E7%AE%80%E4%BB%8B"><span class="nav-text">1. bamdst简介</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-%E4%B8%8B%E8%BD%BD%E5%AE%89%E8%A3%85"><span class="nav-text">2. 下载安装</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-%E5%88%86%E6%9E%90"><span class="nav-text">3. 分析</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#3-1-%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6"><span class="nav-text">3.1. 输入文件</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-2-%E8%BF%90%E8%A1%8C"><span class="nav-text">3.2. 运行</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-3-%E7%BB%93%E6%9E%9C"><span class="nav-text">3.3. 结果</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-references"><span class="nav-text">4. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">94</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">47</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">294</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">496k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">7:31</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2022/07/31/omics_genome_quality.assessment_mapping_bamdst/";
    this.page.identifier = "2022/07/31/omics_genome_quality.assessment_mapping_bamdst/";
    this.page.title = "基因组质量评估：（五）mapping法：6. 用软件bamdst统计BAM文件";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
