<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>RNA-sequencing数据分析工具比较 | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="RNA-sequencing数据分析工具的比较，学习文章内容的笔记。">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-sequencing数据分析工具比较">
<meta property="og:url" content="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="RNA-sequencing数据分析工具的比较，学习文章内容的笔记。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig8_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig2_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig3_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig4_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig5_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig6_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig7_HTML.jpg?as=webp">
<meta property="article:published_time" content="2021-11-19T12:50:00.000Z">
<meta property="article:modified_time" content="2022-03-28T03:46:53.051Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="RNA-sequencing">
<meta property="article:tag" content="RNA-seq">
<meta property="article:tag" content="HISAT2">
<meta property="article:tag" content="StringTie">
<meta property="article:tag" content="Oases">
<meta property="article:tag" content="LoRDEC">
<meta property="article:tag" content="STARlong">
<meta property="article:tag" content="Salmon-SMEM">
<meta property="article:tag" content="DESeq2">
<meta property="article:tag" content="GATK">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-omics_transcriptome.RNA-seq" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/19/omics_transcriptome.RNA-seq/" class="article-date">
  <time datetime="2021-11-19T12:50:00.000Z" itemprop="datePublished">2021-11-19</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/transcriptome/">transcriptome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      RNA-sequencing数据分析工具比较
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=176028&auto=1&height=32"></iframe></div>

<h1 id="1-background"><a href="#1-background" class="headerlink" title="1. background"></a>1. background</h1><p>2017年nature communications上发表的Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis，对RNA-sequencing(RNA-seq)数据的39种分析工具做了比较和总结。</p>
<p>RNA-seq分析流程主要有四大模块：</p>
<ul>
<li>RNA-seq变异分析(Genomic variants)</li>
<li>短读长数据的亚型检测(Short-read isoform detection)</li>
<li>长读长数据的亚型jiance(Long-read isoform detection)</li>
<li>表达分析(Expression analysis)</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp" title="RNACocktail分析协议" width="90%" />

<p><strong><p align="center">Figure 1. RNACocktail分析协议</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="2-比较结果总结"><a href="#2-比较结果总结" class="headerlink" title="2. 比较结果总结"></a>2. 比较结果总结</h1><ol>
<li>比对工具：<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</li>
<li>有参组装工具：<strong>StringTie</strong>在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</li>
<li>从头组装工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；<strong>Oases</strong>鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</li>
<li>三代测序错误纠正工具：<strong>LoRDEC</strong>在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</li>
<li>全长转录本亚型检测工具：注重质量选GMAP，注重速度选<strong>STARlong</strong>。</li>
<li>转录本的定量：<strong>Salmon-SMEM</strong>(不经过比对)运行速度和表现都更好；StringTie(基于基因组)的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>差异表达的工具：<strong>DESeq2</strong>在各项得分中均优于其他的工具。</li>
<li>检测基因组和转录组的突变：<strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</li>
</ol>
<p>作者总结了每一步的高精度工具，作为RNA-seq分析工具选择的一般建议。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig8_HTML.jpg?as=webp" title="RNACocktail计算流程" width="90%" />

<p><strong><p align="center">Figure 2. RNACocktail计算流程</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="3-比较结果"><a href="#3-比较结果" class="headerlink" title="3. 比较结果"></a>3. 比较结果</h1><h2 id="3-1-比对工具"><a href="#3-1-比对工具" class="headerlink" title="3.1. 比对工具"></a>3.1. 比对工具</h2><h3 id="3-1-1-比对工具用法"><a href="#3-1-1-比对工具用法" class="headerlink" title="3.1.1. 比对工具用法"></a>3.1.1. 比对工具用法</h3><p>RNA-seq分析的第一步通常是转录本的鉴定，需要把RNA-seq reads比对到合适的参考序列上。</p>
<p>如果用基因组作为参考序列可以检测到新的转录本，但可能需要耗费更多的计算资源；如果用转录组作为参考则无法找出新的转录本，但速度更快。如果研究物种没有可靠的参考序列，可以重头组装对转录本进行鉴定。</p>
<p>比对工具可以把RNA-seq short reads往参考基因组上进行mapping，做比对(alignment)，进行剪切点预测(junction prediction)和exon-intron边界的预测。</p>
<h3 id="3-1-2-比对结果"><a href="#3-1-2-比对结果" class="headerlink" title="3.1.2. 比对结果"></a>3.1.2. 比对结果</h3><p>比较了三款比对主流软件，<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</p>
<ul>
<li><strong>HISAT2</strong>在所有样品中能检测到的剪接位点验证率最高，但找到的剪接位点总数则比TopHat和STAR都要少。在运行速度方面，HISAT2则有较大的优势，比STAR快大约2.5倍并且比TopHat快大约100倍。</li>
<li><strong>STAR</strong>对于成对reads的唯一比对表现则比较好，特别是对于有较长读长的MCF7-300细胞系的数据，不过STAR会有更多含有soft-clipped和碱基错配的较低质量比对情况。</li>
<li>如果单看有soft-clipped的reads，<strong>TopHat</strong>会把这部分reads全部舍弃。</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig2_HTML.jpg?as=webp" title="比对工具比较结果" width="90%" />

<p><strong><p align="center">Figure 3. 比对工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-2-有参组装工具"><a href="#3-2-有参组装工具" class="headerlink" title="3.2. 有参组装工具"></a>3.2. 有参组装工具</h2><p>比对完成后，接着组装转录本。比较了三款组装工具，StringTie在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</p>
<ul>
<li>转录本数量：StringTie&gt;Cufflinks~IDP</li>
<li>基因水平准确性和灵敏度：IDP&gt;Cufflinks&gt;StringTie</li>
<li>转录本水平准确性：StringTie&gt;IDP&gt;Cufflink</li>
<li>运行速度：StringTie&gt;Cufflinks~IDP</li>
<li>IDP则会忽略单外显子转录本</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig3_HTML.jpg?as=webp" title="有参组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 4. 有参组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-3-从头组装工具"><a href="#3-3-从头组装工具" class="headerlink" title="3.3. 从头组装工具"></a>3.3. 从头组装工具</h2><p>比较了三款从头组装转录本的工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；Oases鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</p>
<ul>
<li>Trinity：组装的转录本较长，灵敏度高。</li>
<li>SOAPdenovo-Trans：组装到转录本数量多，准确性最高，对于高表达的转录本有明显的组装偏好性；花费的内存和CPU最低。</li>
<li>Oases：在所有样品中都有较好的N10-N50长度的表现，鉴定长转录本有优势；能够较好地涵盖到低表达的基因。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig4_HTML.jpg?as=webp" title="从头组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 5. 从头组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-4-三代测序错误纠正工具"><a href="#3-4-三代测序错误纠正工具" class="headerlink" title="3.4. 三代测序错误纠正工具"></a>3.4. 三代测序错误纠正工具</h2><p>LoRDEC在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</p>
<ul>
<li>LoRDEC：纠正后的reads质量更高，运行速度比LSC快100倍。</li>
<li>LSC：纠正后的reads比对率提高更多。</li>
</ul>
<h2 id="3-5-全长转录本亚型检测工具"><a href="#3-5-全长转录本亚型检测工具" class="headerlink" title="3.5. 全长转录本亚型检测工具"></a>3.5. 全长转录本亚型检测工具</h2><p>注重质量选GMAP，注重速度选STARlong。</p>
<ul>
<li>GMAP：比对到参考序列的reads数量更多。</li>
<li>STARlong：运行速度快68倍。</li>
</ul>
<h2 id="3-6-转录本的定量"><a href="#3-6-转录本的定量" class="headerlink" title="3.6. 转录本的定量"></a>3.6. 转录本的定量</h2><h3 id="3-6-1-转录本定量软件"><a href="#3-6-1-转录本定量软件" class="headerlink" title="3.6.1. 转录本定量软件"></a>3.6.1. 转录本定量软件</h3><ol>
<li>基于比对的转录本定量</li>
</ol>
<ul>
<li>传统方法是将read比对(spliced -aligned)到参考基因组，然后利用Cufflinks和StringTie进行转录本组装，最后进行定量。</li>
<li>如果具有参考转录本序列，reads可以直接跟转录本序列比对(aligned)，然后使用Salmon-Aln和eXpress进行定量。</li>
</ul>
<ol start="2">
<li>不经过比对(alignment-free)的转录本定量</li>
</ol>
<ul>
<li>主要提供了四个工具：Sailfish、Salmon-SMEM、quasi-mapping和kallisto。</li>
</ul>
<ol start="3">
<li>基于长读长技术的IDP（使用不同的短读长和长读长比对工具）</li>
</ol>
<h3 id="3-6-2-比较结果"><a href="#3-6-2-比较结果" class="headerlink" title="3.6.2. 比较结果"></a>3.6.2. 比较结果</h3><ul>
<li>StringTie的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>基于转录组比对的两款工具则和Salmon-SMEM的相关性更高，考虑到Salmon-SMEM更快的运行速度，eXpress和Salmon-Aln可能使用体验不如Salmon-SMEM。</li>
<li>kallisto和Salmon-SMEM表现最好。</li>
<li>基于比对基因组的工具则能够看到，使用HISAT2或者TopHat对具有较长读长的样品进行比对所得到的定量结果要比STAR要好。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig5_HTML.jpg?as=webp" title="转录本表达定量比较结果" width="90%" />

<p><strong><p align="center">Figure 6. 转录本表达定量比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-7-差异表达的工具"><a href="#3-7-差异表达的工具" class="headerlink" title="3.7. 差异表达的工具"></a>3.7. 差异表达的工具</h2><p>比较了基于计数的工具（DESeq2、limma和edgeR）和基于组装的工具（Cuffdiff和Ballgown）。</p>
<p><strong>DESeq2</strong>在各项得分中均优于其他的工具，而Cuffdiff和Ballgown的表现则相对比较差。<br>总的来说，基于计数的工具要比基于组装的工具效果要好；另外从运行时间上看的话，Cuffdiff也是最慢的工具。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig6_HTML.jpg?as=webp" title="差异表达比较结果" width="90%" />

<p><strong><p align="center">Figure 7. 差异表达比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-8-检测基因组和转录组的突变"><a href="#3-8-检测基因组和转录组的突变" class="headerlink" title="3.8. 检测基因组和转录组的突变"></a>3.8. 检测基因组和转录组的突变</h2><p>两款突变检测的工具：GATK HaplotypeCaller和Samtools mpileup。</p>
<p><strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig7_HTML.jpg?as=webp" title="突变检测比较结果" width="90%" />

<p><strong><p align="center">Figure 8. 突变检测比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="4-39个RNA-seq分析工具版本号、重要参数及下载地址"><a href="#4-39个RNA-seq分析工具版本号、重要参数及下载地址" class="headerlink" title="4. 39个RNA-seq分析工具版本号、重要参数及下载地址"></a>4. 39个RNA-seq分析工具版本号、重要参数及下载地址</h1><h2 id="4-1-比对工具-——-Reference-based-transcript-identification"><a href="#4-1-比对工具-——-Reference-based-transcript-identification" class="headerlink" title="4.1. 比对工具 —— Reference-based transcript identification"></a>4.1. 比对工具 —— Reference-based transcript identification</h2><ol>
<li>TopHat2： –no-coverage-search<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/index.shtml">http://ccb.jhu.edu/software/tophat/index.shtml</a></li>
<li>STAR: -twopassMode Basic –outFilterType BySJout<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
<li>HISAT2 2.0.1-beta –dta (or –dta-cufflinks)<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/hisat/index.shtml">http://www.ccb.jhu.edu/software/hisat/index.shtml</a></li>
<li>RASER 0.52 -b 0.03<br><a target="_blank" rel="noopener" href="https://www.ibp.ucla.edu/research/xiao/RASER.html">https://www.ibp.ucla.edu/research/xiao/RASER.html</a></li>
</ol>
<h2 id="4-2-有参考转录本组装工具"><a href="#4-2-有参考转录本组装工具" class="headerlink" title="4.2. 有参考转录本组装工具"></a>4.2. 有参考转录本组装工具</h2><ol>
<li>Cufflinks 2.2.1 –frag-bias-correct<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>StringTie 1.2.1 -v -B<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/stringtie/">http://www.ccb.jhu.edu/software/stringtie/</a></li>
</ol>
<h2 id="4-3-无参考转录本组装工具"><a href="#4-3-无参考转录本组装工具" class="headerlink" title="4.3. 无参考转录本组装工具"></a>4.3. 无参考转录本组装工具</h2><ol>
<li>SOAPdenovoTrans 1.04 -K 25<br><a target="_blank" rel="noopener" href="https://github.com/aquaskyline/SOAPdenovo-Trans/">https://github.com/aquaskyline/SOAPdenovo-Trans/</a></li>
<li>Oases 0.2.09 (Velvetv1.2.10) (velveth haslength: 25) (velvetg options: -read trkg yes)<br><a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/~zerbino/oases/">http://www.ebi.ac.uk/~zerbino/oases/</a></li>
<li>Trinity 2.1.1 –normalize reads<br><a target="_blank" rel="noopener" href="http://trinityrnaseq.sourceforge.net/">http://trinityrnaseq.sourceforge.net/</a></li>
</ol>
<h2 id="4-4-三代长read分析工具"><a href="#4-4-三代长read分析工具" class="headerlink" title="4.4. 三代长read分析工具"></a>4.4. 三代长read分析工具</h2><ol>
<li>LoRDEC 0.6 -k 23 -s 3<br><a target="_blank" rel="noopener" href="http://atgc.lirmm.fr/lordec/">http://atgc.lirmm.fr/lordec/</a></li>
<li>GMAP 12&#x2F;31&#x2F;15 -f 1<br><a target="_blank" rel="noopener" href="http://research-pub.gene.com/gmap/">http://research-pub.gene.com/gmap/</a></li>
<li>STARlong 2.5.1b<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
</ol>
<p>Followed the recommended options : </p>
<ul>
<li>–outSAMattributes NH HI NM MD</li>
<li>–readNameSeparator space</li>
<li>–outFilterMultimapScoreRange 1</li>
<li>–outFilterMismatchNmax 2000</li>
<li>–scoreGapNoncan -20</li>
<li>–scoreGapGCAG -4</li>
<li>–scoreGapATAC -8</li>
<li>–scoreDelOpen -1</li>
<li>–scoreDelBase -1</li>
<li>–scoreInsOpen -1</li>
<li>–scoreInsBase -1</li>
<li>–alignEndsType Local</li>
<li>–seedSearchStartLmax 50</li>
<li>–seedPerReadNmax 100000</li>
<li>–seedPerWindowNmax 1000</li>
<li>–alignTranscriptsPerReadNmax 100000</li>
<li>–alignTranscriptsPerWindowNmax 10000</li>
<li>–outSAMstrandField intronMotif</li>
<li>–outSAMunmapped Within</li>
</ul>
<ol start="4">
<li>IDP 0.1.9<br><a target="_blank" rel="noopener" href="https://www.healthcare.uiowa.edu/labs/au/IDP/">https://www.healthcare.uiowa.edu/labs/au/IDP/</a></li>
</ol>
<h2 id="4-5-定量工具"><a href="#4-5-定量工具" class="headerlink" title="4.5. 定量工具"></a>4.5. 定量工具</h2><ol>
<li>eXpress 1.5.1 (bowtie2 v2.2.7) (bowtie2 options: -a -X 600 –rdg 6,5 –rfg 6,5 –score-min L,-.6,-.4 –no-discordant –no-mixed)<br><a target="_blank" rel="noopener" href="https://pachterlab.github.io/eXpress/index.html">https://pachterlab.github.io/eXpress/index.html</a></li>
<li>kallisto 0.42.4<br><a target="_blank" rel="noopener" href="http://pachterlab.github.io/kallisto/about.html">http://pachterlab.github.io/kallisto/about.html</a></li>
<li>Sailfish 0.9.0<br><a target="_blank" rel="noopener" href="http://www.cs.cmu.edu/~ckingsf/software/sailfish/">http://www.cs.cmu.edu/~ckingsf/software/sailfish/</a></li>
<li>Salmon-Aln 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a></li>
<li>Salmon-SMEM 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type fmd<br>quant: -k,19 </li>
<li>Salmon-Quasi 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type quasi -k 31 </li>
<li>featureCounts 1.5.0-p1 -p -B -C<br><a target="_blank" rel="noopener" href="http://subread.sourceforge.net/">http://subread.sourceforge.net/</a></li>
</ol>
<h2 id="4-6-差异表达分析工具"><a href="#4-6-差异表达分析工具" class="headerlink" title="4.6. 差异表达分析工具"></a>4.6. 差异表达分析工具</h2><ol>
<li>DESeq2 1.14.1<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html">http://bioconductor.org/packages/release/bioc/html/DESeq2.html</a></li>
<li>edgeR 3.16.5<br><a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/bioc/html/edgeR.html">http://www.bioconductor.org/packages/release/bioc/html/edgeR.html</a></li>
<li>limma 3.30.7<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/limma.html">http://bioconductor.org/packages/release/bioc/html/limma.html</a></li>
<li>Cuffdiff 2.2.1<br>–frag-bias-correct –emit-count-tables<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>Ballgown 2.6.0<br><a target="_blank" rel="noopener" href="https://github.com/alyssafrazee/ballgown">https://github.com/alyssafrazee/ballgown</a></li>
<li>sleuth 0.28.1<br><a target="_blank" rel="noopener" href="https://github.com/pachterlab/sleuth">https://github.com/pachterlab/sleuth</a></li>
</ol>
<h2 id="4-7-变异分析工具"><a href="#4-7-变异分析工具" class="headerlink" title="4.7. 变异分析工具"></a>4.7. 变异分析工具</h2><ol>
<li>SAMtools 1.2 (bcftools v1.2)<br>samtools mpileup -C50 -d 100000<br><a target="_blank" rel="noopener" href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</a></li>
<li>bcftools filter -s LowQual -e ‘%QUAL&lt;20 —— DP&gt;10000’<br><a target="_blank" rel="noopener" href="https://github.com/samtools/bcftools">https://github.com/samtools/bcftools</a></li>
<li>GATK v3.5-0-g36282e4 (picard 1.129)<br><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/download/">https://software.broadinstitute.org/gatk/download/</a><br>Picard AddOrReplaceReadGroups: SO&#x3D;coordinate<br>Picard MarkDuplicates: CREATE INDEX&#x3D;true VALIDATION STRINGENCY&#x3D;SILENTGATK<br>SplitNCigarReads: -rf ReassignOneMappingQuality -RMQF 255 -RMQT 60</li>
</ol>
<p>-U ALLOW N CIGAR READSGATK<br>HaplotypeCaller: -stand call conf 20.0<br>-stand emit conf 20.0 -A StrandBiasBySample<br>-A StrandAlleleCountsBySampleGATK<br>VariantFiltration: -window 35 -cluster 3 -filterName FS -filter<br>“FS &gt;30.0” -filterName QD -filter “QD &lt;2.0”</p>
<h2 id="4-8-RNA编辑"><a href="#4-8-RNA编辑" class="headerlink" title="4.8. RNA编辑"></a>4.8. RNA编辑</h2><ol>
<li>GIREMI 0.2.1<br><a target="_blank" rel="noopener" href="https://github.com/zhqingit/giremi">https://github.com/zhqingit/giremi</a></li>
<li>Varsim 0.5.1<br><a target="_blank" rel="noopener" href="https://github.com/bioinform/varsim">https://github.com/bioinform/varsim</a></li>
</ol>
<h2 id="4-9-基因融合"><a href="#4-9-基因融合" class="headerlink" title="4.9. 基因融合"></a>4.9. 基因融合</h2><ol>
<li>FusionCatcher 0.99.5a beta<br><a target="_blank" rel="noopener" href="https://github.com/ndaniel/fusioncatcher">https://github.com/ndaniel/fusioncatcher</a></li>
<li>JAFFA 1.0.6<br><a target="_blank" rel="noopener" href="https://github.com/Oshlack/JAFFA">https://github.com/Oshlack/JAFFA</a></li>
<li>SOAPfuse 1.27<br><a target="_blank" rel="noopener" href="http://soap.genomics.org.cn/soapfuse.html">http://soap.genomics.org.cn/soapfuse.html</a></li>
<li>STAR-Fusion 0.7.0<br><a target="_blank" rel="noopener" href="https://github.com/STAR-Fusion/STAR-Fusion">https://github.com/STAR-Fusion/STAR-Fusion</a></li>
<li>TopHat-Fusion 2.0.14<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/fusion_index.shtml">http://ccb.jhu.edu/software/tophat/fusion_index.shtml</a><br>s</li>
</ol>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper</a></li>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw">https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw</a></li>
<li><a target="_blank" rel="noopener" href="https://www.cnblogs.com/wangprince2017/p/9959008.html">https://www.cnblogs.com/wangprince2017/p/9959008.html</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/" data-id="cl1aiqcs0004am4vk9ss1b9oc" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/11/20/bio_evolutionary.tree/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          进化树相关知识
        
      </div>
    </a>
  
  
    <a href="/2021/11/16/biosoft_Circos/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Circos圈图绘制</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>