<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="RNA-sequencing数据分析工具的比较，学习文章内容的笔记。">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-sequencing数据分析工具比较">
<meta property="og:url" content="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="RNA-sequencing数据分析工具的比较，学习文章内容的笔记。">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig8_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig2_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig3_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig4_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig5_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig6_HTML.jpg?as=webp">
<meta property="og:image" content="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig7_HTML.jpg?as=webp">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2021-11-19T12:50:00.000Z">
<meta property="article:modified_time" content="2022-07-15T11:37:46.915Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="RNA-sequencing">
<meta property="article:tag" content="RNA-seq">
<meta property="article:tag" content="HISAT2">
<meta property="article:tag" content="StringTie">
<meta property="article:tag" content="Oases">
<meta property="article:tag" content="LoRDEC">
<meta property="article:tag" content="STARlong">
<meta property="article:tag" content="Salmon-SMEM">
<meta property="article:tag" content="DESeq2">
<meta property="article:tag" content="GATK">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp">

<link rel="canonical" href="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>RNA-sequencing数据分析工具比较 | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2021/11/19/omics_transcriptome.RNA-seq/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNA-sequencing数据分析工具比较
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-11-19 20:50:00" itemprop="dateCreated datePublished" datetime="2021-11-19T20:50:00+08:00">2021-11-19</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-07-15 19:37:46" itemprop="dateModified" datetime="2022-07-15T19:37:46+08:00">2022-07-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/" itemprop="url" rel="index"><span itemprop="name">omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/transcriptome/" itemprop="url" rel="index"><span itemprop="name">transcriptome</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2021/11/19/omics_transcriptome.RNA-seq/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2021/11/19/omics_transcriptome.RNA-seq/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>7.8k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>7 mins.</span>
            </span>
            <div class="post-description">RNA-sequencing数据分析工具的比较，学习文章内容的笔记。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=176028&auto=1&height=32"></iframe></div>

<h1 id="1-background"><a href="#1-background" class="headerlink" title="1. background"></a>1. background</h1><p>2017年nature communications上发表的Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis，对RNA-sequencing(RNA-seq)数据的39种分析工具做了比较和总结。</p>
<p>RNA-seq分析流程主要有四大模块：</p>
<ul>
<li>RNA-seq变异分析(Genomic variants)</li>
<li>短读长数据的亚型检测(Short-read isoform detection)</li>
<li>长读长数据的亚型jiance(Long-read isoform detection)</li>
<li>表达分析(Expression analysis)</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig1_HTML.jpg?as=webp" title="RNACocktail分析协议" width="90%" />

<p><strong><p align="center">Figure 1. RNACocktail分析协议</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="2-比较结果总结"><a href="#2-比较结果总结" class="headerlink" title="2. 比较结果总结"></a>2. 比较结果总结</h1><ol>
<li>比对工具：<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</li>
<li>有参组装工具：<strong>StringTie</strong>在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</li>
<li>从头组装工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；<strong>Oases</strong>鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</li>
<li>三代测序错误纠正工具：<strong>LoRDEC</strong>在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</li>
<li>全长转录本亚型检测工具：注重质量选GMAP，注重速度选<strong>STARlong</strong>。</li>
<li>转录本的定量：<strong>Salmon-SMEM</strong>(不经过比对)运行速度和表现都更好；StringTie(基于基因组)的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>差异表达的工具：<strong>DESeq2</strong>在各项得分中均优于其他的工具。</li>
<li>检测基因组和转录组的突变：<strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</li>
</ol>
<p>作者总结了每一步的高精度工具，作为RNA-seq分析工具选择的一般建议。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig8_HTML.jpg?as=webp" title="RNACocktail计算流程" width="90%" />

<p><strong><p align="center">Figure 2. RNACocktail计算流程</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="3-比较结果"><a href="#3-比较结果" class="headerlink" title="3. 比较结果"></a>3. 比较结果</h1><h2 id="3-1-比对工具"><a href="#3-1-比对工具" class="headerlink" title="3.1. 比对工具"></a>3.1. 比对工具</h2><h3 id="3-1-1-比对工具用法"><a href="#3-1-1-比对工具用法" class="headerlink" title="3.1.1. 比对工具用法"></a>3.1.1. 比对工具用法</h3><p>RNA-seq分析的第一步通常是转录本的鉴定，需要把RNA-seq reads比对到合适的参考序列上。</p>
<p>如果用基因组作为参考序列可以检测到新的转录本，但可能需要耗费更多的计算资源；如果用转录组作为参考则无法找出新的转录本，但速度更快。如果研究物种没有可靠的参考序列，可以重头组装对转录本进行鉴定。</p>
<p>比对工具可以把RNA-seq short reads往参考基因组上进行mapping，做比对(alignment)，进行剪切点预测(junction prediction)和exon-intron边界的预测。</p>
<h3 id="3-1-2-比对结果"><a href="#3-1-2-比对结果" class="headerlink" title="3.1.2. 比对结果"></a>3.1.2. 比对结果</h3><p>比较了三款比对主流软件，<strong>HISAT2</strong>在准确性和运算速度上表现最好值得推荐，STAR和TopHat在剪切位点总数上表现更好。</p>
<ul>
<li><strong>HISAT2</strong>在所有样品中能检测到的剪接位点验证率最高，但找到的剪接位点总数则比TopHat和STAR都要少。在运行速度方面，HISAT2则有较大的优势，比STAR快大约2.5倍并且比TopHat快大约100倍。</li>
<li><strong>STAR</strong>对于成对reads的唯一比对表现则比较好，特别是对于有较长读长的MCF7-300细胞系的数据，不过STAR会有更多含有soft-clipped和碱基错配的较低质量比对情况。</li>
<li>如果单看有soft-clipped的reads，<strong>TopHat</strong>会把这部分reads全部舍弃。</li>
</ul>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig2_HTML.jpg?as=webp" title="比对工具比较结果" width="90%" />

<p><strong><p align="center">Figure 3. 比对工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-2-有参组装工具"><a href="#3-2-有参组装工具" class="headerlink" title="3.2. 有参组装工具"></a>3.2. 有参组装工具</h2><p>比对完成后，接着组装转录本。比较了三款组装工具，StringTie在组装的转录本数量，转录本水平的准确性和运行速度上都表现最好，Cufflinks表现一般，isoform detection and prediction(IDP)在基因水平准确性上表现最好。</p>
<ul>
<li>转录本数量：StringTie&gt;Cufflinks~IDP</li>
<li>基因水平准确性和灵敏度：IDP&gt;Cufflinks&gt;StringTie</li>
<li>转录本水平准确性：StringTie&gt;IDP&gt;Cufflink</li>
<li>运行速度：StringTie&gt;Cufflinks~IDP</li>
<li>IDP则会忽略单外显子转录本</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig3_HTML.jpg?as=webp" title="有参组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 4. 有参组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-3-从头组装工具"><a href="#3-3-从头组装工具" class="headerlink" title="3.3. 从头组装工具"></a>3.3. 从头组装工具</h2><p>比较了三款从头组装转录本的工具：Trinity转录本长，灵敏度高；SOAPdenovo-Trans转录本多，准确性高；Oases鉴定长转录本有优势，能够较好地涵盖到低表达的基因。</p>
<ul>
<li>Trinity：组装的转录本较长，灵敏度高。</li>
<li>SOAPdenovo-Trans：组装到转录本数量多，准确性最高，对于高表达的转录本有明显的组装偏好性；花费的内存和CPU最低。</li>
<li>Oases：在所有样品中都有较好的N10-N50长度的表现，鉴定长转录本有优势；能够较好地涵盖到低表达的基因。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig4_HTML.jpg?as=webp" title="从头组装工具比较结果" width="90%" />

<p><strong><p align="center">Figure 5. 从头组装工具比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-4-三代测序错误纠正工具"><a href="#3-4-三代测序错误纠正工具" class="headerlink" title="3.4. 三代测序错误纠正工具"></a>3.4. 三代测序错误纠正工具</h2><p>LoRDEC在纠错质量和速度上更有优势，LSC在纠正后reads比对率的改善上表现更好。</p>
<ul>
<li>LoRDEC：纠正后的reads质量更高，运行速度比LSC快100倍。</li>
<li>LSC：纠正后的reads比对率提高更多。</li>
</ul>
<h2 id="3-5-全长转录本亚型检测工具"><a href="#3-5-全长转录本亚型检测工具" class="headerlink" title="3.5. 全长转录本亚型检测工具"></a>3.5. 全长转录本亚型检测工具</h2><p>注重质量选GMAP，注重速度选STARlong。</p>
<ul>
<li>GMAP：比对到参考序列的reads数量更多。</li>
<li>STARlong：运行速度快68倍。</li>
</ul>
<h2 id="3-6-转录本的定量"><a href="#3-6-转录本的定量" class="headerlink" title="3.6. 转录本的定量"></a>3.6. 转录本的定量</h2><h3 id="3-6-1-转录本定量软件"><a href="#3-6-1-转录本定量软件" class="headerlink" title="3.6.1. 转录本定量软件"></a>3.6.1. 转录本定量软件</h3><ol>
<li>基于比对的转录本定量</li>
</ol>
<ul>
<li>传统方法是将read比对(spliced -aligned)到参考基因组，然后利用Cufflinks和StringTie进行转录本组装，最后进行定量。</li>
<li>如果具有参考转录本序列，reads可以直接跟转录本序列比对(aligned)，然后使用Salmon-Aln和eXpress进行定量。</li>
</ul>
<ol start="2">
<li>不经过比对(alignment-free)的转录本定量</li>
</ol>
<ul>
<li>主要提供了四个工具：Sailfish、Salmon-SMEM、quasi-mapping和kallisto。</li>
</ul>
<ol start="3">
<li>基于长读长技术的IDP（使用不同的短读长和长读长比对工具）</li>
</ol>
<h3 id="3-6-2-比较结果"><a href="#3-6-2-比较结果" class="headerlink" title="3.6.2. 比较结果"></a>3.6.2. 比较结果</h3><ul>
<li>StringTie的定量结果更接近于不基于基因组比对的工具结果。</li>
<li>基于转录组比对的两款工具则和Salmon-SMEM的相关性更高，考虑到Salmon-SMEM更快的运行速度，eXpress和Salmon-Aln可能使用体验不如Salmon-SMEM。</li>
<li>kallisto和Salmon-SMEM表现最好。</li>
<li>基于比对基因组的工具则能够看到，使用HISAT2或者TopHat对具有较长读长的样品进行比对所得到的定量结果要比STAR要好。</li>
</ul>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig5_HTML.jpg?as=webp" title="转录本表达定量比较结果" width="90%" />

<p><strong><p align="center">Figure 6. 转录本表达定量比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-7-差异表达的工具"><a href="#3-7-差异表达的工具" class="headerlink" title="3.7. 差异表达的工具"></a>3.7. 差异表达的工具</h2><p>比较了基于计数的工具（DESeq2、limma和edgeR）和基于组装的工具（Cuffdiff和Ballgown）。</p>
<p><strong>DESeq2</strong>在各项得分中均优于其他的工具，而Cuffdiff和Ballgown的表现则相对比较差。<br>总的来说，基于计数的工具要比基于组装的工具效果要好；另外从运行时间上看的话，Cuffdiff也是最慢的工具。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig6_HTML.jpg?as=webp" title="差异表达比较结果" width="90%" />

<p><strong><p align="center">Figure 7. 差异表达比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h2 id="3-8-检测基因组和转录组的突变"><a href="#3-8-检测基因组和转录组的突变" class="headerlink" title="3.8. 检测基因组和转录组的突变"></a>3.8. 检测基因组和转录组的突变</h2><p>两款突变检测的工具：GATK HaplotypeCaller和Samtools mpileup。</p>
<p><strong>GATK</strong>在突变检测方面具有较高的准确性，在运行时间方面GATK与Samtools没有明显的差异。</p>
<img src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-017-00050-4/MediaObjects/41467_2017_50_Fig7_HTML.jpg?as=webp" title="突变检测比较结果" width="90%" />

<p><strong><p align="center">Figure 8. 突变检测比较结果</strong><br>from <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">paper: Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis</a></p></p>
<h1 id="4-39个RNA-seq分析工具版本号、重要参数及下载地址"><a href="#4-39个RNA-seq分析工具版本号、重要参数及下载地址" class="headerlink" title="4. 39个RNA-seq分析工具版本号、重要参数及下载地址"></a>4. 39个RNA-seq分析工具版本号、重要参数及下载地址</h1><h2 id="4-1-比对工具-——-Reference-based-transcript-identification"><a href="#4-1-比对工具-——-Reference-based-transcript-identification" class="headerlink" title="4.1. 比对工具 —— Reference-based transcript identification"></a>4.1. 比对工具 —— Reference-based transcript identification</h2><ol>
<li>TopHat2： –no-coverage-search<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/index.shtml">http://ccb.jhu.edu/software/tophat/index.shtml</a></li>
<li>STAR: -twopassMode Basic –outFilterType BySJout<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
<li>HISAT2 2.0.1-beta –dta (or –dta-cufflinks)<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/hisat/index.shtml">http://www.ccb.jhu.edu/software/hisat/index.shtml</a></li>
<li>RASER 0.52 -b 0.03<br><a target="_blank" rel="noopener" href="https://www.ibp.ucla.edu/research/xiao/RASER.html">https://www.ibp.ucla.edu/research/xiao/RASER.html</a></li>
</ol>
<h2 id="4-2-有参考转录本组装工具"><a href="#4-2-有参考转录本组装工具" class="headerlink" title="4.2. 有参考转录本组装工具"></a>4.2. 有参考转录本组装工具</h2><ol>
<li>Cufflinks 2.2.1 –frag-bias-correct<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>StringTie 1.2.1 -v -B<br><a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/stringtie/">http://www.ccb.jhu.edu/software/stringtie/</a></li>
</ol>
<h2 id="4-3-无参考转录本组装工具"><a href="#4-3-无参考转录本组装工具" class="headerlink" title="4.3. 无参考转录本组装工具"></a>4.3. 无参考转录本组装工具</h2><ol>
<li>SOAPdenovoTrans 1.04 -K 25<br><a target="_blank" rel="noopener" href="https://github.com/aquaskyline/SOAPdenovo-Trans/">https://github.com/aquaskyline/SOAPdenovo-Trans/</a></li>
<li>Oases 0.2.09 (Velvetv1.2.10) (velveth haslength: 25) (velvetg options: -read trkg yes)<br><a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/~zerbino/oases/">http://www.ebi.ac.uk/~zerbino/oases/</a></li>
<li>Trinity 2.1.1 –normalize reads<br><a target="_blank" rel="noopener" href="http://trinityrnaseq.sourceforge.net/">http://trinityrnaseq.sourceforge.net/</a></li>
</ol>
<h2 id="4-4-三代长read分析工具"><a href="#4-4-三代长read分析工具" class="headerlink" title="4.4. 三代长read分析工具"></a>4.4. 三代长read分析工具</h2><ol>
<li>LoRDEC 0.6 -k 23 -s 3<br><a target="_blank" rel="noopener" href="http://atgc.lirmm.fr/lordec/">http://atgc.lirmm.fr/lordec/</a></li>
<li>GMAP 12&#x2F;31&#x2F;15 -f 1<br><a target="_blank" rel="noopener" href="http://research-pub.gene.com/gmap/">http://research-pub.gene.com/gmap/</a></li>
<li>STARlong 2.5.1b<br><a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/releases">https://github.com/alexdobin/STAR/releases</a></li>
</ol>
<p>Followed the recommended options : </p>
<ul>
<li>–outSAMattributes NH HI NM MD</li>
<li>–readNameSeparator space</li>
<li>–outFilterMultimapScoreRange 1</li>
<li>–outFilterMismatchNmax 2000</li>
<li>–scoreGapNoncan -20</li>
<li>–scoreGapGCAG -4</li>
<li>–scoreGapATAC -8</li>
<li>–scoreDelOpen -1</li>
<li>–scoreDelBase -1</li>
<li>–scoreInsOpen -1</li>
<li>–scoreInsBase -1</li>
<li>–alignEndsType Local</li>
<li>–seedSearchStartLmax 50</li>
<li>–seedPerReadNmax 100000</li>
<li>–seedPerWindowNmax 1000</li>
<li>–alignTranscriptsPerReadNmax 100000</li>
<li>–alignTranscriptsPerWindowNmax 10000</li>
<li>–outSAMstrandField intronMotif</li>
<li>–outSAMunmapped Within</li>
</ul>
<ol start="4">
<li>IDP 0.1.9<br><a target="_blank" rel="noopener" href="https://www.healthcare.uiowa.edu/labs/au/IDP/">https://www.healthcare.uiowa.edu/labs/au/IDP/</a></li>
</ol>
<h2 id="4-5-定量工具"><a href="#4-5-定量工具" class="headerlink" title="4.5. 定量工具"></a>4.5. 定量工具</h2><ol>
<li>eXpress 1.5.1 (bowtie2 v2.2.7) (bowtie2 options: -a -X 600 –rdg 6,5 –rfg 6,5 –score-min L,-.6,-.4 –no-discordant –no-mixed)<br><a target="_blank" rel="noopener" href="https://pachterlab.github.io/eXpress/index.html">https://pachterlab.github.io/eXpress/index.html</a></li>
<li>kallisto 0.42.4<br><a target="_blank" rel="noopener" href="http://pachterlab.github.io/kallisto/about.html">http://pachterlab.github.io/kallisto/about.html</a></li>
<li>Sailfish 0.9.0<br><a target="_blank" rel="noopener" href="http://www.cs.cmu.edu/~ckingsf/software/sailfish/">http://www.cs.cmu.edu/~ckingsf/software/sailfish/</a></li>
<li>Salmon-Aln 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a></li>
<li>Salmon-SMEM 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type fmd<br>quant: -k,19 </li>
<li>Salmon-Quasi 0.6.1<br><a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon">https://github.com/COMBINE-lab/salmon</a><br>index: –type quasi -k 31 </li>
<li>featureCounts 1.5.0-p1 -p -B -C<br><a target="_blank" rel="noopener" href="http://subread.sourceforge.net/">http://subread.sourceforge.net/</a></li>
</ol>
<h2 id="4-6-差异表达分析工具"><a href="#4-6-差异表达分析工具" class="headerlink" title="4.6. 差异表达分析工具"></a>4.6. 差异表达分析工具</h2><ol>
<li>DESeq2 1.14.1<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html">http://bioconductor.org/packages/release/bioc/html/DESeq2.html</a></li>
<li>edgeR 3.16.5<br><a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/bioc/html/edgeR.html">http://www.bioconductor.org/packages/release/bioc/html/edgeR.html</a></li>
<li>limma 3.30.7<br><a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/limma.html">http://bioconductor.org/packages/release/bioc/html/limma.html</a></li>
<li>Cuffdiff 2.2.1<br>–frag-bias-correct –emit-count-tables<br><a target="_blank" rel="noopener" href="http://cole-trapnell-lab.github.io/cufflinks/">http://cole-trapnell-lab.github.io/cufflinks/</a></li>
<li>Ballgown 2.6.0<br><a target="_blank" rel="noopener" href="https://github.com/alyssafrazee/ballgown">https://github.com/alyssafrazee/ballgown</a></li>
<li>sleuth 0.28.1<br><a target="_blank" rel="noopener" href="https://github.com/pachterlab/sleuth">https://github.com/pachterlab/sleuth</a></li>
</ol>
<h2 id="4-7-变异分析工具"><a href="#4-7-变异分析工具" class="headerlink" title="4.7. 变异分析工具"></a>4.7. 变异分析工具</h2><ol>
<li>SAMtools 1.2 (bcftools v1.2)<br>samtools mpileup -C50 -d 100000<br><a target="_blank" rel="noopener" href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</a></li>
<li>bcftools filter -s LowQual -e ‘%QUAL&lt;20 —— DP&gt;10000’<br><a target="_blank" rel="noopener" href="https://github.com/samtools/bcftools">https://github.com/samtools/bcftools</a></li>
<li>GATK v3.5-0-g36282e4 (picard 1.129)<br><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/download/">https://software.broadinstitute.org/gatk/download/</a><br>Picard AddOrReplaceReadGroups: SO&#x3D;coordinate<br>Picard MarkDuplicates: CREATE INDEX&#x3D;true VALIDATION STRINGENCY&#x3D;SILENTGATK<br>SplitNCigarReads: -rf ReassignOneMappingQuality -RMQF 255 -RMQT 60</li>
</ol>
<p>-U ALLOW N CIGAR READSGATK<br>HaplotypeCaller: -stand call conf 20.0<br>-stand emit conf 20.0 -A StrandBiasBySample<br>-A StrandAlleleCountsBySampleGATK<br>VariantFiltration: -window 35 -cluster 3 -filterName FS -filter<br>“FS &gt;30.0” -filterName QD -filter “QD &lt;2.0”</p>
<h2 id="4-8-RNA编辑"><a href="#4-8-RNA编辑" class="headerlink" title="4.8. RNA编辑"></a>4.8. RNA编辑</h2><ol>
<li>GIREMI 0.2.1<br><a target="_blank" rel="noopener" href="https://github.com/zhqingit/giremi">https://github.com/zhqingit/giremi</a></li>
<li>Varsim 0.5.1<br><a target="_blank" rel="noopener" href="https://github.com/bioinform/varsim">https://github.com/bioinform/varsim</a></li>
</ol>
<h2 id="4-9-基因融合"><a href="#4-9-基因融合" class="headerlink" title="4.9. 基因融合"></a>4.9. 基因融合</h2><ol>
<li>FusionCatcher 0.99.5a beta<br><a target="_blank" rel="noopener" href="https://github.com/ndaniel/fusioncatcher">https://github.com/ndaniel/fusioncatcher</a></li>
<li>JAFFA 1.0.6<br><a target="_blank" rel="noopener" href="https://github.com/Oshlack/JAFFA">https://github.com/Oshlack/JAFFA</a></li>
<li>SOAPfuse 1.27<br><a target="_blank" rel="noopener" href="http://soap.genomics.org.cn/soapfuse.html">http://soap.genomics.org.cn/soapfuse.html</a></li>
<li>STAR-Fusion 0.7.0<br><a target="_blank" rel="noopener" href="https://github.com/STAR-Fusion/STAR-Fusion">https://github.com/STAR-Fusion/STAR-Fusion</a></li>
<li>TopHat-Fusion 2.0.14<br><a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/tophat/fusion_index.shtml">http://ccb.jhu.edu/software/tophat/fusion_index.shtml</a><br>s</li>
</ol>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li>paper：<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-017-00050-4">https://www.nature.com/articles/s41467-017-00050-4</a></li>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw">https://mp.weixin.qq.com/s/hp_wxiLDI9EVB8Z3L1bxbw</a></li>
<li><a target="_blank" rel="noopener" href="https://www.cnblogs.com/wangprince2017/p/9959008.html">https://www.cnblogs.com/wangprince2017/p/9959008.html</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=50% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    
        <div class="reward-container">
  <div>真诚赞赏，手留余香</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    Donate
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/images/wechatpay.jpg" alt="南方枝叶 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
      
      <div style="display: inline-block;">
        <img src="/images/alipay.jpg" alt="南方枝叶 Alipay">
        <p>Alipay</p>
      </div>

  </div>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/RNA-sequencing/" rel="tag"># RNA-sequencing</a>
              <a href="/tags/RNA-seq/" rel="tag"># RNA-seq</a>
              <a href="/tags/HISAT2/" rel="tag"># HISAT2</a>
              <a href="/tags/StringTie/" rel="tag"># StringTie</a>
              <a href="/tags/Oases/" rel="tag"># Oases</a>
              <a href="/tags/LoRDEC/" rel="tag"># LoRDEC</a>
              <a href="/tags/STARlong/" rel="tag"># STARlong</a>
              <a href="/tags/Salmon-SMEM/" rel="tag"># Salmon-SMEM</a>
              <a href="/tags/DESeq2/" rel="tag"># DESeq2</a>
              <a href="/tags/GATK/" rel="tag"># GATK</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2021/11/16/bio_plot_Circos/" rel="prev" title="Circos圈图绘制">
      <i class="fa fa-chevron-left"></i> Circos圈图绘制
    </a></div>
      <div class="post-nav-item">
    <a href="/2021/11/20/bio_evolutionary.tree/" rel="next" title="进化树相关知识">
      进化树相关知识 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-background"><span class="nav-text">1. background</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-%E6%AF%94%E8%BE%83%E7%BB%93%E6%9E%9C%E6%80%BB%E7%BB%93"><span class="nav-text">2. 比较结果总结</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-%E6%AF%94%E8%BE%83%E7%BB%93%E6%9E%9C"><span class="nav-text">3. 比较结果</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#3-1-%E6%AF%94%E5%AF%B9%E5%B7%A5%E5%85%B7"><span class="nav-text">3.1. 比对工具</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-1-%E6%AF%94%E5%AF%B9%E5%B7%A5%E5%85%B7%E7%94%A8%E6%B3%95"><span class="nav-text">3.1.1. 比对工具用法</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-2-%E6%AF%94%E5%AF%B9%E7%BB%93%E6%9E%9C"><span class="nav-text">3.1.2. 比对结果</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-2-%E6%9C%89%E5%8F%82%E7%BB%84%E8%A3%85%E5%B7%A5%E5%85%B7"><span class="nav-text">3.2. 有参组装工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-3-%E4%BB%8E%E5%A4%B4%E7%BB%84%E8%A3%85%E5%B7%A5%E5%85%B7"><span class="nav-text">3.3. 从头组装工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-4-%E4%B8%89%E4%BB%A3%E6%B5%8B%E5%BA%8F%E9%94%99%E8%AF%AF%E7%BA%A0%E6%AD%A3%E5%B7%A5%E5%85%B7"><span class="nav-text">3.4. 三代测序错误纠正工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-5-%E5%85%A8%E9%95%BF%E8%BD%AC%E5%BD%95%E6%9C%AC%E4%BA%9A%E5%9E%8B%E6%A3%80%E6%B5%8B%E5%B7%A5%E5%85%B7"><span class="nav-text">3.5. 全长转录本亚型检测工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-6-%E8%BD%AC%E5%BD%95%E6%9C%AC%E7%9A%84%E5%AE%9A%E9%87%8F"><span class="nav-text">3.6. 转录本的定量</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-6-1-%E8%BD%AC%E5%BD%95%E6%9C%AC%E5%AE%9A%E9%87%8F%E8%BD%AF%E4%BB%B6"><span class="nav-text">3.6.1. 转录本定量软件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-6-2-%E6%AF%94%E8%BE%83%E7%BB%93%E6%9E%9C"><span class="nav-text">3.6.2. 比较结果</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-7-%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E7%9A%84%E5%B7%A5%E5%85%B7"><span class="nav-text">3.7. 差异表达的工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-8-%E6%A3%80%E6%B5%8B%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%92%8C%E8%BD%AC%E5%BD%95%E7%BB%84%E7%9A%84%E7%AA%81%E5%8F%98"><span class="nav-text">3.8. 检测基因组和转录组的突变</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-39%E4%B8%AARNA-seq%E5%88%86%E6%9E%90%E5%B7%A5%E5%85%B7%E7%89%88%E6%9C%AC%E5%8F%B7%E3%80%81%E9%87%8D%E8%A6%81%E5%8F%82%E6%95%B0%E5%8F%8A%E4%B8%8B%E8%BD%BD%E5%9C%B0%E5%9D%80"><span class="nav-text">4. 39个RNA-seq分析工具版本号、重要参数及下载地址</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#4-1-%E6%AF%94%E5%AF%B9%E5%B7%A5%E5%85%B7-%E2%80%94%E2%80%94-Reference-based-transcript-identification"><span class="nav-text">4.1. 比对工具 —— Reference-based transcript identification</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-2-%E6%9C%89%E5%8F%82%E8%80%83%E8%BD%AC%E5%BD%95%E6%9C%AC%E7%BB%84%E8%A3%85%E5%B7%A5%E5%85%B7"><span class="nav-text">4.2. 有参考转录本组装工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-3-%E6%97%A0%E5%8F%82%E8%80%83%E8%BD%AC%E5%BD%95%E6%9C%AC%E7%BB%84%E8%A3%85%E5%B7%A5%E5%85%B7"><span class="nav-text">4.3. 无参考转录本组装工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-4-%E4%B8%89%E4%BB%A3%E9%95%BFread%E5%88%86%E6%9E%90%E5%B7%A5%E5%85%B7"><span class="nav-text">4.4. 三代长read分析工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-5-%E5%AE%9A%E9%87%8F%E5%B7%A5%E5%85%B7"><span class="nav-text">4.5. 定量工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-6-%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90%E5%B7%A5%E5%85%B7"><span class="nav-text">4.6. 差异表达分析工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-7-%E5%8F%98%E5%BC%82%E5%88%86%E6%9E%90%E5%B7%A5%E5%85%B7"><span class="nav-text">4.7. 变异分析工具</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-8-RNA%E7%BC%96%E8%BE%91"><span class="nav-text">4.8. RNA编辑</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-9-%E5%9F%BA%E5%9B%A0%E8%9E%8D%E5%90%88"><span class="nav-text">4.9. 基因融合</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-references"><span class="nav-text">5. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">115</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">65</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">417</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="/atom.xml" title="RSS → &#x2F;atom.xml"><i class="fa fa-rss fa-fw"></i>RSS</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

      <div class="wechat_channel">
        <br>
        <!-- 这里添加你的二维码图片 -->
        <img src ="/images/wechat_channel.jpg?raw=true">
        <span>欢迎关注微信公众号</span>
      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2024</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">634k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">9:37</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2021/11/19/omics_transcriptome.RNA-seq/";
    this.page.identifier = "2021/11/19/omics_transcriptome.RNA-seq/";
    this.page.title = "RNA-sequencing数据分析工具比较";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
