<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单概述了mapping法的其中一个评估指标：depth，主要介绍了统计mapped reads的深度分布的一般方">
<meta property="og:type" content="article">
<meta property="og:title" content="基因组质量评估：（五）mapping法：3. 统计mapped reads的深度分布">
<meta property="og:url" content="http://example.com/2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单概述了mapping法的其中一个评估指标：depth，主要介绍了统计mapped reads的深度分布的一般方">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2022-07-26T16:00:00.000Z">
<meta property="article:modified_time" content="2022-07-27T13:02:13.496Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="organelle">
<meta property="article:tag" content="mitogenome">
<meta property="article:tag" content="genome">
<meta property="article:tag" content="quality assessment">
<meta property="article:tag" content="mapping">
<meta property="article:tag" content="sam">
<meta property="article:tag" content="bam">
<meta property="article:tag" content="depth">
<meta property="article:tag" content="plastome">
<meta property="article:tag" content="samtools">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">

<link rel="canonical" href="http://example.com/2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>基因组质量评估：（五）mapping法：3. 统计mapped reads的深度分布 | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --></head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          基因组质量评估：（五）mapping法：3. 统计mapped reads的深度分布
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-07-27 00:00:00 / Modified: 21:02:13" itemprop="dateCreated datePublished" datetime="2022-07-27T00:00:00+08:00">2022-07-27</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/" itemprop="url" rel="index"><span itemprop="name">omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/" itemprop="url" rel="index"><span itemprop="name">genome</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/quality-assessment/" itemprop="url" rel="index"><span itemprop="name">quality assessment</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>1.5k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>1 mins.</span>
            </span>
            <div class="post-description">mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单概述了mapping法的其中一个评估指标：depth，主要介绍了统计mapped reads的深度分布的一般方法。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=2117115&auto=1&height=32"></iframe></div>

<h1 id="1-前情提要：mapping"><a href="#1-前情提要：mapping" class="headerlink" title="1. 前情提要：mapping"></a>1. 前情提要：mapping</h1><p>此篇博客在已通过mapping获得SAM&#x2F;BAM文件的基础上进行分析，</p>
<ol>
<li>关于mapping获得SAM&#x2F;BAM文件的操作可以参考博客：</li>
</ol>
<p><a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/07/23/omics_genome_quality.assessment_mapping_intro/">基因组质量评估：（五）mapping法：1. 简介</a></p>
<h1 id="2-深度（depth）分布"><a href="#2-深度（depth）分布" class="headerlink" title="2. 深度（depth）分布"></a>2. 深度（depth）分布</h1><ul>
<li>在测序是随机分布的情况下，期望在基因组的所有染色体上，mapped reads的depth是均匀分布的。</li>
<li>把reads回mapping到组装好的基因组，通过计算depth，观察depth在基因组上的分布来判断组装的质量。</li>
</ul>
<h1 id="3-统计深度（depth）"><a href="#3-统计深度（depth）" class="headerlink" title="3. 统计深度（depth）"></a>3. 统计深度（depth）</h1><h2 id="3-1-samtools统计"><a href="#3-1-samtools统计" class="headerlink" title="3.1. samtools统计"></a>3.1. samtools统计</h2><p><code>samtools</code>的<code>depth</code>调用<code>mpileup</code>模块进行mapped reads的深度统计。</p>
<h3 id="3-1-1-samtools-depth统计"><a href="#3-1-1-samtools-depth统计" class="headerlink" title="3.1.1. samtools depth统计"></a>3.1.1. samtools depth统计</h3><ol>
<li>samtools depth统计</li>
</ol>
<ul>
<li><code>samtools depth illumina.bam &gt; depth.out</code></li>
</ul>
<ol start="2">
<li>输出</li>
</ol>
<ul>
<li>depth.out有三列数据，tab分隔。</li>
<li>第一列参考序列（染色体）名称；</li>
<li>第二列位置；</li>
<li>第三列比对上的reads数量（即depth）。</li>
</ul>
<h3 id="3-1-2-samtools-mpileup统计"><a href="#3-1-2-samtools-mpileup统计" class="headerlink" title="3.1.2. samtools mpileup统计"></a>3.1.2. samtools mpileup统计</h3><ol>
<li>运行</li>
</ol>
<ul>
<li><code>samtools mpileup -A -Q sample.bam &gt; mpileup.out</code></li>
</ul>
<ol start="2">
<li>输出</li>
</ol>
<ul>
<li>mpileup.out共有6列数据，tab分隔。</li>
<li>第一列参考序列（染色体）名称；</li>
<li>第二列位置；</li>
<li>第三列参考序列的碱基；</li>
<li>第四列比对上的reads数量（即depth）；</li>
<li>第五列比对上的情况；</li>
<li>第六列比对上的碱基的质量。</li>
</ul>
<ol start="3">
<li>第五列比对上的情况，具体解释：</li>
</ol>
<ul>
<li>*表示模糊碱基</li>
<li>大写表示在正链不匹配</li>
<li>小写表示在负链不匹配</li>
<li>^表示匹配的碱基是一个reads的开始，^后紧跟的ascii码减去33代表比对质量，修饰的是后面的碱基，后面紧跟的碱基代表该read的第一个碱基</li>
<li>$代表一个read的结束，该符号修饰前面的碱基</li>
<li>正则表达式<code>+[0-9]+[ACGTNacgtn]+</code>代表在该位点后插入的碱基。举例中chr1的2003928A后面有个+6GGGCCG，很可能是indel</li>
<li>正则表达式<code>-[0-9]+[ACGTNacgtn]+</code>代表在该位点后缺失的碱基</li>
</ul>
<h3 id="3-1-3-samtools-mpileup和samtools-depth的差异"><a href="#3-1-3-samtools-mpileup和samtools-depth的差异" class="headerlink" title="3.1.3. samtools mpileup和samtools depth的差异"></a>3.1.3. samtools mpileup和samtools depth的差异</h3><ol>
<li>差异</li>
</ol>
<ul>
<li><code>samtools depth</code>是调用了<code>samtools mpileup</code>进行的。</li>
<li>所以<code>samtools mpileup</code>可以设置更多参数进行过滤，也有一些默认的过滤参数；而<code>samtools depth</code>没有进行过滤。</li>
</ul>
<ol start="2">
<li><code>samtools mpileup</code>的默认过滤</li>
</ol>
<ul>
<li><code>samtools mpileup</code>默认过滤掉测序质量&lt;13的碱基；</li>
<li><code>samtools mpileup</code>默认过滤掉PE reads中比对异常的reads（包括双端都比上，但是两条配对reads之间的比对距离明显偏离了插入片段的长度分布，或者一端比对上而另一端没比对上）。除非加上-A参数保留异常reads，才与<code>samtools depth</code>一致。</li>
</ul>
<h2 id="3-2-其他方法"><a href="#3-2-其他方法" class="headerlink" title="3.2. 其他方法"></a>3.2. 其他方法</h2><p>除了这里提到的统计深度的方法，还有mosdepth,bamdst,quallimap等软件可以实现，在接下来的博客里会详细介绍。</p>
<h1 id="4-深度分布作图"><a href="#4-深度分布作图" class="headerlink" title="4. 深度分布作图"></a>4. 深度分布作图</h1><p>统计获得滑窗或者位点的深度后，可以作点图&#x2F;折线图来直观地观察深度在整个染色体上的分布。</p>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li>samtools mpileup和samtools depth计算的depth不同的解释：<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/73208822">https://zhuanlan.zhihu.com/p/73208822</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=50% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/organelle/" rel="tag"># organelle</a>
              <a href="/tags/mitogenome/" rel="tag"># mitogenome</a>
              <a href="/tags/genome/" rel="tag"># genome</a>
              <a href="/tags/quality-assessment/" rel="tag"># quality assessment</a>
              <a href="/tags/mapping/" rel="tag"># mapping</a>
              <a href="/tags/sam/" rel="tag"># sam</a>
              <a href="/tags/bam/" rel="tag"># bam</a>
              <a href="/tags/depth/" rel="tag"># depth</a>
              <a href="/tags/plastome/" rel="tag"># plastome</a>
              <a href="/tags/samtools/" rel="tag"># samtools</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/07/23/omics_genome_quality.assessment_mapping_intro/" rel="prev" title="基因组质量评估：（五）mapping法：1. 简介">
      <i class="fa fa-chevron-left"></i> 基因组质量评估：（五）mapping法：1. 简介
    </a></div>
      <div class="post-nav-item">
    <a href="/2022/07/27/omics_genome_quality.assessment_mapping_depth_observe/" rel="next" title="基因组质量评估：（五）mapping法：4. 观察mapped reads的深度分布">
      基因组质量评估：（五）mapping法：4. 观察mapped reads的深度分布 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-%E5%89%8D%E6%83%85%E6%8F%90%E8%A6%81%EF%BC%9Amapping"><span class="nav-text">1. 前情提要：mapping</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-%E6%B7%B1%E5%BA%A6%EF%BC%88depth%EF%BC%89%E5%88%86%E5%B8%83"><span class="nav-text">2. 深度（depth）分布</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-%E7%BB%9F%E8%AE%A1%E6%B7%B1%E5%BA%A6%EF%BC%88depth%EF%BC%89"><span class="nav-text">3. 统计深度（depth）</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#3-1-samtools%E7%BB%9F%E8%AE%A1"><span class="nav-text">3.1. samtools统计</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-1-samtools-depth%E7%BB%9F%E8%AE%A1"><span class="nav-text">3.1.1. samtools depth统计</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-2-samtools-mpileup%E7%BB%9F%E8%AE%A1"><span class="nav-text">3.1.2. samtools mpileup统计</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-3-samtools-mpileup%E5%92%8Csamtools-depth%E7%9A%84%E5%B7%AE%E5%BC%82"><span class="nav-text">3.1.3. samtools mpileup和samtools depth的差异</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-2-%E5%85%B6%E4%BB%96%E6%96%B9%E6%B3%95"><span class="nav-text">3.2. 其他方法</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-%E6%B7%B1%E5%BA%A6%E5%88%86%E5%B8%83%E4%BD%9C%E5%9B%BE"><span class="nav-text">4. 深度分布作图</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-references"><span class="nav-text">5. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">98</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">50</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">312</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">542k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">8:13</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/";
    this.page.identifier = "2022/07/27/omics_genome_quality.assessment_mapping_depth_samtools/";
    this.page.title = "基因组质量评估：（五）mapping法：3. 统计mapped reads的深度分布";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
