<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>scripts | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="store some scripts using in biology and bioinformatics">
<meta property="og:type" content="article">
<meta property="og:title" content="scripts">
<meta property="og:url" content="http://example.com/2021/04/01/scripts/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="store some scripts using in biology and bioinformatics">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2021-04-01T02:56:00.000Z">
<meta property="article:modified_time" content="2022-01-11T13:20:43.129Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoft">
<meta property="article:tag" content="scripts">
<meta property="article:tag" content="batch_run.sh">
<meta property="article:tag" content="ROUSFinder2.0.py">
<meta property="article:tag" content="extract_intron_info.pl">
<meta property="article:tag" content="multi-transcriptome_assembly_with_ref.sh">
<meta property="article:tag" content="extract_single_copy_genes.sh">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-scripts" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/04/01/scripts/" class="article-date">
  <time datetime="2021-04-01T02:56:00.000Z" itemprop="datePublished">2021-04-01</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer/">computer</a>►<a class="article-category-link" href="/categories/computer/scripts/">scripts</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      scripts
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=476751845&auto=1&height=32"></iframe></div>

<h1 id="1-我写的脚本"><a href="#1-我写的脚本" class="headerlink" title="1. 我写的脚本"></a>1. 我写的脚本</h1><h2 id="1-1-多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测"><a href="#1-1-多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测" class="headerlink" title="1.1. 多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测"></a>1.1. 多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/myscripts/multi-transcriptome_assembly_with_ref.sh">multi-transcriptome_assembly_with_ref.sh</a></p>
<h2 id="1-2-提取，合并，和比对单拷贝基因和蛋白"><a href="#1-2-提取，合并，和比对单拷贝基因和蛋白" class="headerlink" title="1.2. 提取，合并，和比对单拷贝基因和蛋白"></a>1.2. 提取，合并，和比对单拷贝基因和蛋白</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/my_scripts/extract_single_copy_genes.sh">extract_single_copy_genes.sh</a></p>
<h1 id="2-我修改的脚本（其他人写的）"><a href="#2-我修改的脚本（其他人写的）" class="headerlink" title="2. 我修改的脚本（其他人写的）"></a>2. 我修改的脚本（其他人写的）</h1><h2 id="2-1-从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息"><a href="#2-1-从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息" class="headerlink" title="2.1. 从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息"></a>2.1. 从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/modifiedscripts/extract_intron_info.pl">extract_intron_info.pl</a></p>
<h1 id="3-保存的脚本"><a href="#3-保存的脚本" class="headerlink" title="3. 保存的脚本"></a>3. 保存的脚本</h1><h2 id="3-1-batch-run-sh-——-多线程并行运行批量化命令"><a href="#3-1-batch-run-sh-——-多线程并行运行批量化命令" class="headerlink" title="3.1. batch_run.sh —— 多线程并行运行批量化命令"></a>3.1. batch_run.sh —— 多线程并行运行批量化命令</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/saved_scripts/batch_run.sh">batch_run.sh</a></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">!/bin/bash</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">batch_run.sh脚本用于多线程并行批量化循环命令，并控制并行数量始终为<span class="variable">$&#123;thread&#125;</span>。</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">usage: 修改thread=后的值为想要设定的并行线程数，把需要并行的命令替换23行的run something。运行脚本batch_run.sh，实现并行运行循环任务。</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">ref：网上找的脚本，来源找不到了，侵删。</span></span><br><span class="line"></span><br><span class="line">thread=16</span><br><span class="line"></span><br><span class="line">start_time=`date +%s`              # 定义脚本运行的开始时间</span><br><span class="line">echo $&#123;start_time&#125;</span><br><span class="line">tmpFifo=/tmp/$$.fifo # 声明管道名称，$$表示脚本当前运行的进程PID</span><br><span class="line">mkfifo $&#123;tmpFifo&#125; # 创建有名管道</span><br><span class="line">exec 3&lt;&gt;$&#123;tmpFifo&#125;                   #创建文件描述符，以可读（&lt;）可写（&gt;）的方式关联管道文件，这时候文件描述符3就有了有名管道文件的所有特性</span><br><span class="line">rm -rf $&#123;tmpFifo&#125;                    #关联后的文件描述符拥有管道文件的所有特性,所以这时候管道文件可以删除，我们留下文件描述符来用就可以了</span><br><span class="line">for ((i=1;i&lt;=$&#123;thread&#125;;i++))</span><br><span class="line">do</span><br><span class="line">        echo &quot;&quot; &gt;&amp;3                   #&amp;3代表引用文件描述符3，这条命令代表往管道里面放入了一个&quot;令牌&quot;</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">for sample in $(cat sample_list.txt) #并行运行的循环</span><br><span class="line">do</span><br><span class="line">        read -u3                           #代表从管道中读取一个令牌</span><br><span class="line">        &#123;</span><br><span class="line">                echo $&#123;sample&#125;</span><br><span class="line">                run something # 需要并行运行的真实命令</span><br><span class="line">                echo &#x27;success&#x27; $&#123;sample&#125;       </span><br><span class="line">                echo &quot;&quot; &gt;&amp;3                   #代表我这一次命令执行到最后，把令牌放回管道</span><br><span class="line">        &#125; &amp;</span><br><span class="line">done</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash"><span class="comment"># ps:可以使用while read line; do echo $line; done &lt; file.txt代替for i in $(cat file.txt);do echo $i; done循环。但是不能用cat file.txt | while read line; do echo $line; done代替。</span></span></span><br><span class="line"></span><br><span class="line">wait</span><br><span class="line"> </span><br><span class="line">stop_time=`date +%s`  #定义脚本运行的结束时间</span><br><span class="line"> </span><br><span class="line">echo &quot;TIME:`expr $stop_time - $start_time`&quot;</span><br><span class="line">exec 3&lt;&amp;-                       #关闭文件描述符的读</span><br><span class="line">exec 3&gt;&amp;-                       #关闭文件描述符的写</span><br></pre></td></tr></table></figure>

<h2 id="3-2-ROUSFinder2-0-py-——-植物线粒体基因组重复序列注释脚本"><a href="#3-2-ROUSFinder2-0-py-——-植物线粒体基因组重复序列注释脚本" class="headerlink" title="3.2. ROUSFinder2.0.py —— 植物线粒体基因组重复序列注释脚本"></a>3.2. ROUSFinder2.0.py —— 植物线粒体基因组重复序列注释脚本</h2><ol>
<li>背景<br>文章<a target="_blank" rel="noopener" href="https://academic.oup.com/g3journal/article/9/2/549/6026745">Repeats of Unusual Size in plant mitochondrial genomes: identification, incidence and evolution</a>总结了植物线粒体基因组的重复序列的特征。研究表明，植物线粒体基因组比动物的要大，包含大量的非编码DNA，突变率低，重排率高。</li>
</ol>
<p>文章里提供了一个python脚本<a target="_blank" rel="noopener" href="https://gsajournals.figshare.com/articles/figure/Supplemental_Material_for_Wynn_and_Christensen_2018/7425680?file=13750433">ROUSFinder2.0.py</a>用于注释植物线粒体基因组中的非串联重复序列(non-tandem repeat)。</p>
<ol start="2">
<li>脚本<br>脚本是python2写的，依赖主要是blastn。</li>
</ol>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ROUSFinder2.0.py用于植物线粒体基因组重复序列的注释</span></span><br><span class="line"><span class="comment">#! /usr/bin/env python  </span></span><br><span class="line"><span class="keyword">import</span> sys, math, os, argparse, csv  </span><br><span class="line">csv.field_size_limit(sys.maxsize)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Version 2.0, November 21, 2018  </span></span><br><span class="line"><span class="comment"># Changes: uses variable parameters  </span></span><br><span class="line"><span class="comment"># Find dispersed repeated sequences in genomes.   </span></span><br><span class="line"><span class="comment"># Designed for plant mitochondrial genomes of up to a few Mbp.  </span></span><br><span class="line"><span class="comment"># May be very slow with larger genomes.   </span></span><br><span class="line"><span class="comment"># Blast can also sometimes give odd results with large or highly repetetive genomes.  </span></span><br><span class="line"><span class="comment"># Gaps, or runs of &#x27;N&#x27;s in the sequence will definitely give weird results.   </span></span><br><span class="line"><span class="comment"># The program assumes there aren&#x27;t any, and that the longest repeat will be the full sequence to itself.  </span></span><br><span class="line"><span class="comment"># If there are long repeats in the output that are listed as being only at one location, this is probably what happened.  </span></span><br><span class="line"><span class="comment"># If there are a lot of repeats within repeats the results can also be odd.  </span></span><br><span class="line"><span class="comment"># Copyright Alan C. Christensen, University of Nebraska, 2018  </span></span><br><span class="line"><span class="comment"># No guarantees, warranties, support, or anything else is implicit or explicit.  </span></span><br><span class="line"><span class="comment"># Input is a fasta format file of a sequence. Genbank format works but generates lots of error messages in stdout.  </span></span><br><span class="line"><span class="comment"># Output is a list of unique, ungapped repeated sequences, fasta formatted.  </span></span><br><span class="line"><span class="comment"># The names are in the format &#x27;&gt;Repeat/ROUS_name_start_end_length&#x27;.  </span></span><br><span class="line"><span class="comment"># A table of repeats with the coordinates of each one is generated.  </span></span><br><span class="line"><span class="comment"># A list of repeat name, length and copy number is generated.  </span></span><br><span class="line"><span class="comment"># A binned table of the total number of repeats in size ranges is generated.  </span></span><br><span class="line"><span class="comment">#  </span></span><br><span class="line"><span class="comment"># PARAMETERS  </span></span><br><span class="line"><span class="comment">#   REQUIRED:  </span></span><br><span class="line"><span class="comment">#      input file in fasta format  </span></span><br><span class="line"><span class="comment">#   Optional  </span></span><br><span class="line"><span class="comment">#      -o output file name  </span></span><br><span class="line"><span class="comment">#      -m minimum length of exact matches to keep  </span></span><br><span class="line"><span class="comment">#      -b path to blastn (default is /usr/bin/)  </span></span><br><span class="line"><span class="comment">#      -k keep temp files  </span></span><br><span class="line"><span class="comment">#      -gb to write the repeats to a genbank format file  </span></span><br><span class="line"><span class="comment">#      -rew reward for match (default is 1)  </span></span><br><span class="line"><span class="comment">#      -pen penalty for mismatch (default is 20)  </span></span><br><span class="line">  </span><br><span class="line">parser = argparse.ArgumentParser(description=<span class="string">&#x27;Find repeats in a fasta sequence file&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;infile&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Input .fasta file&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-o&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;outfile&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Output file name seed, default is input_repeats&#x27;</span>, default=<span class="string">&#x27;default&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-m&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;minlen&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Minimum length of matches to keep, default=50&#x27;</span>, default=<span class="string">&#x27;50&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-b&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;blast_path&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Path to blastn program, default is /usr/bin/&#x27;</span>, default=<span class="string">&#x27;/usr/bin/&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-k&#x27;</span>, action=<span class="string">&#x27;store_true&#x27;</span>, dest=<span class="string">&#x27;keep&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;True to keep temp files&#x27;</span>, default=<span class="literal">False</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-gb&#x27;</span>, action=<span class="string">&#x27;store_true&#x27;</span>, dest=<span class="string">&#x27;genbank&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;True to write GenBank format file&#x27;</span>, default=<span class="literal">False</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-rew&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;reward&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Reward for match&#x27;</span>, default=<span class="string">&#x27;1&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-pen&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;penalty&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Penalty for mismatch&#x27;</span>, default=<span class="string">&#x27;20&#x27;</span>)  </span><br><span class="line">results = parser.parse_args()  </span><br><span class="line">infile = results.infile  </span><br><span class="line">outfile = results.outfile  </span><br><span class="line">minlen = <span class="built_in">int</span>(results.minlen)  </span><br><span class="line">blast_path = results.blast_path  </span><br><span class="line">keep = results.keep  </span><br><span class="line">genbank = results.genbank  </span><br><span class="line">reward = results.reward  </span><br><span class="line">penalty = results.penalty  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># It might be useful to define the wordsize as something less than minlen, so both variables are used.  </span></span><br><span class="line"><span class="comment"># Wordsize smaller than minlen would give smaller core identical sequences in the middle of repeats.  </span></span><br><span class="line"><span class="comment"># An example might be to change this to wordsize = str(int(minlen/2)).  </span></span><br><span class="line">wordsize = <span class="built_in">str</span>(minlen)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># If no output file seed is specified, make one by stripping leading directory information  </span></span><br><span class="line"><span class="comment"># and stripping trailing .fa or .fasta from the input file name and using that.  </span></span><br><span class="line"><span class="keyword">if</span> outfile == <span class="string">&#x27;default&#x27;</span>:  </span><br><span class="line">    outfile = infile  </span><br><span class="line">    <span class="keyword">if</span> outfile.count(<span class="string">&#x27;/&#x27;</span>) &gt; <span class="number">0</span>:  </span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(outfile.count(<span class="string">&#x27;/&#x27;</span>)):  </span><br><span class="line">            index = outfile.index(<span class="string">&#x27;/&#x27;</span>)  </span><br><span class="line">            outfile = outfile[index+<span class="number">1</span>:]  </span><br><span class="line">    <span class="keyword">if</span> outfile.endswith(<span class="string">&#x27;.fa&#x27;</span>) <span class="keyword">or</span> outfile.endswith(<span class="string">&#x27;.fasta&#x27;</span>):  </span><br><span class="line">        outfile = outfile.rstrip(<span class="string">&#x27;fasta&#x27;</span>)  </span><br><span class="line">    outfile = outfile.rstrip(<span class="string">&#x27;.&#x27;</span>)  </span><br><span class="line">outfa = outfile+<span class="string">&#x27;_rep.fasta&#x27;</span>  </span><br><span class="line">outtab = outfile+<span class="string">&#x27;_rep_table.txt&#x27;</span>  </span><br><span class="line">outbin = outfile+<span class="string">&#x27;_binned.txt&#x27;</span>  </span><br><span class="line">outcount = outfile+<span class="string">&#x27;_rep_counts.txt&#x27;</span>  </span><br><span class="line">outgb = outfile+<span class="string">&#x27;_repeats.gb.txt&#x27;</span>  </span><br><span class="line">tempblast = outfile+<span class="string">&#x27;_tempblast.txt&#x27;</span>  </span><br><span class="line">temprepeats = outfile+<span class="string">&#x27;_temprepeats.txt&#x27;</span>  </span><br><span class="line">tempparse = outfile+<span class="string">&#x27;_sequence_parsing.txt&#x27;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Get sequence name and length from fasta file.  </span></span><br><span class="line">seq = <span class="built_in">open</span>(infile, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">seqname = seq.readline()  </span><br><span class="line">seqname = seqname.lstrip(<span class="string">&#x27;&gt; &#x27;</span>)  </span><br><span class="line">seqname = seqname.rstrip()  </span><br><span class="line">seqlen = <span class="number">0</span>   </span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> seq:  </span><br><span class="line">    <span class="keyword">if</span>(line[<span class="number">0</span>] == <span class="string">&quot;&gt;&quot;</span>):  </span><br><span class="line">        <span class="keyword">continue</span>  </span><br><span class="line">    seqlen += <span class="built_in">len</span>(line.strip())  </span><br><span class="line">seq.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># run blastn with query file plus strand (removing first line which is full length sequence), minus strand, and concatenate  </span></span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Performing self-blastn comparison with &#x27;</span>+seqname      </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+infile+<span class="string">&#x27; -strand plus -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward &#x27;</span>+reward+<span class="string">&#x27; -penalty -&#x27;</span>+penalty+<span class="string">&#x27; -ungapped -dust no -soft_masking false -evalue 10  -outfmt &quot;10 qstart qend length sstart send mismatch sstrand qseq&quot; | tail -n+2 &gt; tempblast1.txt&#x27;</span>)  </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+infile+<span class="string">&#x27; -strand minus -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward &#x27;</span>+reward+<span class="string">&#x27; -penalty -&#x27;</span>+penalty+<span class="string">&#x27; -ungapped -dust no -soft_masking false -evalue 10 -outfmt &quot;10 qstart qend length sstart send mismatch sstrand qseq&quot; &gt; tempblast2.txt&#x27;</span>)  </span><br><span class="line">os.system(<span class="string">&#x27;cat tempblast1.txt tempblast2.txt &gt; &#x27;</span>+tempblast)  </span><br><span class="line">os.system(<span class="string">&#x27;rm tempblast1.txt tempblast2.txt&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># open tempblast.txt, convert to list of lists, and sort by length and position descending  </span></span><br><span class="line"><span class="comment"># This is necessary because blastn does not output every possible pair of hits when there are more than 2 copies of a repeat  </span></span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Sorting alignments...&#x27;</span>  </span><br><span class="line">f = <span class="built_in">open</span>(tempblast, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">reader = csv.reader(f)  </span><br><span class="line">alignments = <span class="built_in">list</span>(reader)  </span><br><span class="line">f.close()  </span><br><span class="line">alignments = <span class="built_in">sorted</span>(alignments, key=<span class="keyword">lambda</span> x: (-<span class="number">1</span>*<span class="built_in">int</span>(x[<span class="number">2</span>]), -<span class="number">1</span>*<span class="built_in">int</span>(x[<span class="number">0</span>])))  </span><br><span class="line">alignments.append([<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;0&#x27;</span>,<span class="string">&#x27;A&#x27;</span>,<span class="string">&#x27;X&#x27;</span>])  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># New list of uniques  </span></span><br><span class="line"><span class="comment"># Text file &#x27;_sequence_parsing.txt&#x27; includes the information on how duplicates were found.  </span></span><br><span class="line"><span class="comment"># Start at row 0. Compare to subsequent rows.   </span></span><br><span class="line"><span class="comment"># If repeat length is different from the next row, it has passed all the tests, write it to the file.  </span></span><br><span class="line"><span class="comment"># If query or subject coordinates are the same as the query or subject or reversed coordinates  </span></span><br><span class="line"><span class="comment"># of a subsequent row, it is not unique, so go to the next row and do the comparisons again.  </span></span><br><span class="line"><span class="comment"># Thanks to Alex Kozik for repeatedly testing and finding bugs in the algorithm.  </span></span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Finding unique repeats...&#x27;</span>  </span><br><span class="line">uniques = []  </span><br><span class="line">sp = <span class="built_in">open</span>(tempparse, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> row <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(alignments)):  </span><br><span class="line">    sp.write(<span class="string">&#x27;row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">      </span><br><span class="line">    <span class="keyword">if</span> <span class="built_in">int</span>(alignments[row][<span class="number">2</span>]) &lt; minlen:  </span><br><span class="line">        <span class="comment"># This won&#x27;t happen unless the word_size is defined as something other than minlen.  </span></span><br><span class="line">        <span class="comment"># That could be useful under some circumstances.  </span></span><br><span class="line">        sp.write(<span class="string">&#x27;row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is less than minlength&#x27;</span>)  </span><br><span class="line">        <span class="keyword">break</span>  </span><br><span class="line">    <span class="keyword">else</span>:  </span><br><span class="line">      </span><br><span class="line">        <span class="keyword">for</span> compare <span class="keyword">in</span> <span class="built_in">range</span>(row+<span class="number">1</span>,<span class="built_in">len</span>(alignments)):  </span><br><span class="line">            <span class="keyword">if</span> alignments[row][<span class="number">2</span>] != alignments[compare][<span class="number">2</span>]:   </span><br><span class="line">                uniques.append(alignments[row])  </span><br><span class="line">                sp.write(<span class="string">&#x27;\tadding row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; to unique list\n&#x27;</span>)  </span><br><span class="line">                <span class="keyword">break</span>  </span><br><span class="line">            <span class="keyword">else</span>:  </span><br><span class="line">                sp.write(<span class="string">&#x27;\tcomparing to &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">      </span><br><span class="line">                <span class="keyword">if</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">0</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">1</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; and &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27; are the same\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">1</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">0</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qend and qstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">3</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">4</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as sstart and send of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">4</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">3</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as send and sstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">0</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">1</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qstart and qend of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">1</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">0</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qend and qstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">3</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">4</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as sstart and send of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">4</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">3</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as send and sstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">else</span>:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is different\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">sp.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Write uniques into output file  </span></span><br><span class="line"><span class="comment"># Start list for copy number table  </span></span><br><span class="line">rous_count = <span class="number">0</span>  </span><br><span class="line">g = <span class="built_in">open</span>(outfa, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">repcopies = []  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(uniques)):  </span><br><span class="line">    qstart = uniques[i][<span class="number">0</span>]  </span><br><span class="line">    qend = uniques[i][<span class="number">1</span>]  </span><br><span class="line">    length = uniques[i][<span class="number">2</span>]  </span><br><span class="line">    seq = uniques[i][<span class="number">7</span>]  </span><br><span class="line">      </span><br><span class="line">    rous_count += <span class="number">1</span>  </span><br><span class="line">    g.write(<span class="string">&#x27;&gt;Repeat_&#x27;</span>+<span class="built_in">str</span>(rous_count)+<span class="string">&#x27;\n&#x27;</span>+seq+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">    repcopies.append([<span class="string">&#x27;Repeat_&#x27;</span>+<span class="built_in">str</span>(rous_count),length])  </span><br><span class="line">          </span><br><span class="line"><span class="keyword">if</span> rous_count == <span class="number">0</span>:  </span><br><span class="line">    <span class="built_in">print</span> <span class="string">&quot;\tRepeats of unusual size? I don&#x27;t think they exist&quot;</span>  </span><br><span class="line">g.close()  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Repeat fasta file is done, as you wish.&#x27;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Now find each copy of each repeat. Again, this is because the blastn output file does not have every possible alignment.  </span></span><br><span class="line"><span class="comment"># It is also because the information on locations and strand is not organized well in the blastn output.  </span></span><br><span class="line"><span class="comment"># In addition, this subroutine eliminates duplicates of nested repeats.  </span></span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Finding all copies of repeats...&quot;</span>  </span><br><span class="line">g = <span class="built_in">open</span>(outfa, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+outfa+<span class="string">&#x27; -strand both -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward 1 -penalty -20 -ungapped -dust no -soft_masking false -evalue 1000 -outfmt &quot;10 qseqid length sstart send sstrand qcovhsp&quot; &gt; &#x27;</span>+temprepeats)  </span><br><span class="line">g.close()  </span><br><span class="line">  </span><br><span class="line">tempr = <span class="built_in">open</span>(temprepeats, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">reader = csv.reader(tempr)  </span><br><span class="line">replist = <span class="built_in">list</span>(reader)  </span><br><span class="line">tempr.close()  </span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Making a table of the repeats...&quot;</span>  </span><br><span class="line">sum_rep_len = <span class="number">0</span>  </span><br><span class="line">bin_dict = &#123;&#125;  </span><br><span class="line">binned = [seqname,seqlen,<span class="number">0</span>]  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># defining the bins  </span></span><br><span class="line">i = <span class="number">0</span>  </span><br><span class="line">j = <span class="number">50</span>  </span><br><span class="line"><span class="keyword">while</span> j &lt; <span class="number">1000</span>:  </span><br><span class="line">    bin_dict[i] = j  </span><br><span class="line">    binned.append(<span class="number">0</span>)  </span><br><span class="line">    i += <span class="number">1</span>  </span><br><span class="line">    j += <span class="number">50</span>  </span><br><span class="line"><span class="keyword">while</span> j &lt;= <span class="number">10000</span>:  </span><br><span class="line">    bin_dict[i] = j  </span><br><span class="line">    binned.append(<span class="number">0</span>)  </span><br><span class="line">    i +=<span class="number">1</span>  </span><br><span class="line">    j += <span class="number">250</span>  </span><br><span class="line">      </span><br><span class="line"><span class="comment"># make list for entire sequence, set each position as 0  </span></span><br><span class="line">posit = []  </span><br><span class="line"><span class="keyword">for</span> n <span class="keyword">in</span> <span class="built_in">range</span>(seqlen):  </span><br><span class="line">    posit.append(<span class="number">0</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Thanks to Emily Wynn for suggesting qcovhsp for this loop.  </span></span><br><span class="line"><span class="comment"># if qcovhsp is &gt;98%, write to the file  </span></span><br><span class="line"><span class="comment"># write tab separated values of repeat name, length, start, end, strand to outtab  </span></span><br><span class="line"><span class="comment"># make list for genbank file  </span></span><br><span class="line"><span class="comment"># Keep stats on lengths  </span></span><br><span class="line">rt = <span class="built_in">open</span>(outtab, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">rt.write(seqname+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(seqlen)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">templist = []  </span><br><span class="line">gblist =[]  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># look at each repeat in turn  </span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(replist)):  </span><br><span class="line">    <span class="comment"># if repeat is good (&gt;98% identical to another one), write it to the file, and put the name in a list  </span></span><br><span class="line">    <span class="keyword">if</span> <span class="built_in">int</span>(replist[i][<span class="number">5</span>])&gt;<span class="number">98</span>:  </span><br><span class="line">        rt.write(<span class="built_in">str</span>(replist[i][<span class="number">0</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">1</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">2</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">3</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">4</span>])+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">        <span class="keyword">if</span> replist[i][<span class="number">4</span>] == <span class="string">&#x27;minus&#x27;</span>:  </span><br><span class="line">            location = <span class="string">&#x27;complement(&#x27;</span>+replist[i][<span class="number">3</span>]+<span class="string">&#x27;..&#x27;</span>+replist[i][<span class="number">2</span>]+<span class="string">&#x27;)&#x27;</span>  </span><br><span class="line">        <span class="keyword">else</span>:  </span><br><span class="line">            location = replist[i][<span class="number">2</span>]+<span class="string">&#x27;..&#x27;</span>+replist[i][<span class="number">3</span>]  </span><br><span class="line">        gblist.append(<span class="string">&#x27;     repeat_region   &#x27;</span>+location+<span class="string">&#x27;\n                     /rpt_type=dispersed\n                     /label=&#x27;</span>+replist[i][<span class="number">0</span>]+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">        templist.append(replist[i][<span class="number">0</span>])  </span><br><span class="line">        <span class="comment"># then write 1&#x27;s at every position in the sequence covered by that repeat  </span></span><br><span class="line">        <span class="comment"># these can then be summed to get total bases of repeats  </span></span><br><span class="line">        <span class="comment"># bases in overlapping repeats are only counted once  </span></span><br><span class="line">        <span class="keyword">for</span> n <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">int</span>(replist[i][<span class="number">2</span>]), <span class="built_in">int</span>(replist[i][<span class="number">3</span>])):  </span><br><span class="line">            posit[n] = <span class="number">1</span>  </span><br><span class="line">        <span class="comment"># then scan through bin sizes and if a repeat is greater than the  </span></span><br><span class="line">        <span class="comment"># bin_dict size cutoff, add one to the bin  </span></span><br><span class="line">        <span class="keyword">for</span> j <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(binned)-<span class="number">4</span>, -<span class="number">1</span>, -<span class="number">1</span>):  </span><br><span class="line">            <span class="keyword">if</span> <span class="built_in">int</span>(replist[i][<span class="number">1</span>]) &gt;= bin_dict[j]:  </span><br><span class="line">                binned[j+<span class="number">3</span>] +=<span class="number">1</span>  </span><br><span class="line">                <span class="keyword">break</span>  </span><br><span class="line">sum_rep_len = posit.count(<span class="number">1</span>)  </span><br><span class="line">binned[<span class="number">2</span>] = sum_rep_len  </span><br><span class="line">rt.close()  </span><br><span class="line"><span class="keyword">if</span> genbank == <span class="literal">True</span>:  </span><br><span class="line">    gb = <span class="built_in">open</span>(outgb, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">    <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(gblist)):  </span><br><span class="line">        gb.write(gblist[i])  </span><br><span class="line">    gb.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># write tab separated values of repeat name, length, copy number to outcount  </span></span><br><span class="line"><span class="comment"># first two lines are also a table of stats on repeats  </span></span><br><span class="line">rc = <span class="built_in">open</span>(outcount,<span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">rc.write(<span class="string">&#x27;Sequence\tGenome_size\tNumROUS\tAvgSize\tAvgCopyNum\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">numrous = <span class="number">0</span>  </span><br><span class="line">sizerous = <span class="number">0</span>  </span><br><span class="line">copyrous = <span class="number">0</span>  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(repcopies)):  </span><br><span class="line">    repname = repcopies[i][<span class="number">0</span>]  </span><br><span class="line">    replen = <span class="built_in">float</span>(repcopies[i][<span class="number">1</span>])  </span><br><span class="line">    repcop = <span class="built_in">float</span>(templist.count(repname))  </span><br><span class="line">  </span><br><span class="line">    numrous += <span class="number">1</span>  </span><br><span class="line">    sizerous += replen  </span><br><span class="line">    copyrous += repcop  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">if</span> numrous == <span class="number">0</span>:  </span><br><span class="line">    avsizerous = <span class="string">&#x27;NA&#x27;</span>  </span><br><span class="line">    avcopyrous = <span class="string">&#x27;NA&#x27;</span>  </span><br><span class="line"><span class="keyword">else</span>:  </span><br><span class="line">    avsizerous = sizerous/numrous  </span><br><span class="line">    avcopyrous = copyrous/numrous  </span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">rc.write(seqname+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(seqlen)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(numrous)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(avsizerous)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(avcopyrous)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(repcopies)):  </span><br><span class="line">    rc.write(repcopies[i][<span class="number">0</span>]+<span class="string">&#x27;\t&#x27;</span>+repcopies[i][<span class="number">1</span>]+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(templist.count(repcopies[i][<span class="number">0</span>]))+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">rc.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Write binned table headers, then stats for this sequence.  </span></span><br><span class="line">binfile = <span class="built_in">open</span>(outbin, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;Sequence\tSeq_len\tRep_len\t&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(bin_dict)):  </span><br><span class="line">    binfile.write(<span class="built_in">str</span>(bin_dict[i])+<span class="string">&#x27;\t&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(binned)):  </span><br><span class="line">    binfile.write(<span class="built_in">str</span>(binned[i])+<span class="string">&#x27;\t&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">binfile.close()  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Repeat tables are done, as you wish.&quot;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Removing temp files if necessary  </span></span><br><span class="line"><span class="keyword">if</span> keep == <span class="literal">False</span>:  </span><br><span class="line">    os.system(<span class="string">&#x27;rm &#x27;</span>+tempblast+<span class="string">&#x27; &#x27;</span>+temprepeats+<span class="string">&#x27; &#x27;</span>+tempparse)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Rachael Schulte, William Goldman and Rob Reiner inspired this section of code  </span></span><br><span class="line">quote_dict = &#123;<span class="number">0</span>:<span class="string">&quot;48656c6c6f2e204d79206e616d6520697320496e69676f204d6f6e746f79612e20596f75206b696c6c6564206d79206661746865722e205072657061726520746f206469652e&quot;</span>, <span class="number">1</span>:<span class="string">&quot;5768656e20492077617320796f7572206167652c2074656c65766973696f6e207761732063616c6c656420626f6f6b732e&quot;</span>, <span class="number">2</span>:<span class="string">&quot;486176652066756e2073746f726d696e2720646120636173746c6521&quot;</span>, <span class="number">3</span>:<span class="string">&quot;4d79207761792773206e6f7420766572792073706f7274736d616e6c696b652e&quot;</span>, <span class="number">4</span>:<span class="string">&quot;596f75206b656570207573696e67207468617420776f72642e204920646f206e6f74207468696e6b206974206d65616e73207768617420796f75207468696e6b206974206d65616e732e&quot;</span>, <span class="number">5</span>:<span class="string">&quot;4d75726465726564206279207069726174657320697320676f6f642e&quot;</span>,<span class="number">6</span>:<span class="string">&quot;496e636f6e6365697661626c6521&quot;</span>, <span class="number">7</span>:<span class="string">&quot;5468657265277320612062696720646966666572656e6365206265747765656e206d6f73746c79206465616420616e6420616c6c20646561642e&quot;</span>, <span class="number">8</span>:<span class="string">&quot;596f7520727573682061206d697261636c65206d616e2c20796f752067657420726f7474656e206d697261636c65732e&quot;</span>, <span class="number">9</span>:<span class="string">&quot;476f6f64206e696768742c20576573746c65792e20476f6f6420776f726b2e20536c6565702077656c6c2e2049276c6c206d6f7374206c696b656c79206b696c6c20796f7520696e20746865206d6f726e696e672e&quot;</span>,<span class="number">10</span>:<span class="string">&quot;4e6f206d6f7265207268796d65732c2049206d65616e2069742120416e79626f64792077616e742061207065616e75743f&quot;</span>&#125;  </span><br><span class="line"><span class="keyword">import</span> random, binascii  </span><br><span class="line">z = random.randint(<span class="number">0</span>,<span class="number">10</span>)  </span><br><span class="line"><span class="built_in">print</span> binascii.unhexlify(quote_dict[z])+<span class="string">&#x27;\n&#x27;</span>  </span><br></pre></td></tr></table></figure>

<ol start="3">
<li>运行<br><code>python2 ROUSFinder2.0.py mito.genome.fa</code></li>
</ol>
<ul>
<li>-m参数指定注释重复序列的最小长度，默认是50bp</li>
<li>-b参数指定blastn的所在路径</li>
<li>-o指定输出文件</li>
<li>-gb生成GenBank格式文件</li>
</ul>
<ol start="4">
<li>输出文件<br>输出文件有四个，mito.genome.fa_binned.txt, mito.genome.fa_rep.fasta, mito.genome.fa_rep_counts.txt, mito.genome.fa_rep_table.txt。</li>
</ol>
<p>如果使用了-gb还会生成mito.genome.fa_repeats.gb.txt。</p>
<ul>
<li>mito.genome.fa_rep_table.txt文件，内容包括重复序列的长度，起始位置，正负链等信息。</li>
<li>mito.genome.fa_rep.fasta是鉴定出来的重复序列。</li>
<li>mito.genome.fa_rep_counts.txt是重复序列的数量信息，包括序列ID-Sequence,基因组尺寸-Genome_size,重复序列数量-NumROUS,重复序列平均尺寸-AvgSize,平均重复次数-AvgCopyNum，以及每个重复序列的长度信息。</li>
<li>mito.genome.fa_binned.txt，其他统计信息。</li>
<li>mito.genome.fa_repeats.gb.txt是GeneBank格式的重复序列文件。</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/04/01/scripts/" data-id="cl1aidmxg004e4gvkema3557f" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/04/20/linux_PATH/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          在Linux系统中设置环境变量PATH
        
      </div>
    </a>
  
  
    <a href="/2021/03/27/bio_experiment/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">分子生物学实验</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>