<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="学习生物信息学的笔记本">
<meta property="og:type" content="website">
<meta property="og:title" content="生信技工">
<meta property="og:url" content="http://example.com/page/4/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="学习生物信息学的笔记本">
<meta property="og:locale" content="zh_CN">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoftware">
<meta property="article:tag" content="skills">
<meta property="article:tag" content="tools">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main">
  
    <article id="post-omics_genome.functional.annotation" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/05/17/omics_genome.functional.annotation/" class="article-date">
  <time datetime="2021-05-17T02:36:00.000Z" itemprop="datePublished">2021-05-17</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/genome/">genome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/05/17/omics_genome.functional.annotation/">基因组注释（三）：基因功能注释</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1842025914&auto=1&height=32"></iframe></div>


<p>基因功能的注释依赖于上一步的基因结构预测，根据预测结果从基因组上提取翻译后的<strong>蛋白序列</strong>和主流的数据库进行比对，完成功能注释。(有些也支持用DNA序列做功能注释的，但比较少。)</p>
<p>比较推荐的方案是：eggNOG-mapper网页注释和PANNZER2网页注释，如果有余力再加上interpro的本地注释。</p>
<p>功能注释的一般流程：1.选择数据库并下载，2.构建blastp索引，3.用blastp比对蛋白序列到数据库，4.结果整理。</p>
<p>有些软件集成了数据库和功能注释的多个流程，甚至提供服务器用于网页版注释，例如eggNOG-mapper和interproscan分别集成了eggNOG和interpro数据库，PANNZER2使用最新的UniProt数据库实现网页注释。</p>
<h1 id="1-数据库概述"><a href="#1-数据库概述" class="headerlink" title="1. 数据库概述"></a>1. 数据库概述</h1><p>根据数据库中已知编码基因的注释信息（包括motif、domain），基于同源比对，对基因中的模序和结构域、新基因编码的蛋白质功能、所参与的信号传导通路和代谢途径等的预测。基因组注释内容还可涉及蛋白激酶、病原与宿主互作、致病毒力因子预测、抗性基因等。<br>常用数据库：</p>
<ul>
<li><p>Nr：NCBI官方非冗余蛋白数据库，包括PDB, Swiss-Prot, PIR, PRF; 如果要用DNA序列，就是nt库。</p>
</li>
<li><p>UniProt：分为两部分。Swiss-Prot是UniProt数据库中经过人工校正的高质量蛋白数据库，蛋白序列得到实验的验证。TrEBL是UniProt数据库中未经过人工校正、机器自动注释的蛋白数据库。</p>
</li>
<li><p>eggNOG：EMBL创建，是对NCBI的COG数据库的拓展，提供了不同分类水平蛋白的直系同源分组（Orthologous Groups，OG）。</p>
</li>
<li><p>InterPro：EBI开发的一个整合的蛋白家族功能注释数据库，包括Gene3D、CDD、Pfam等10几个数据库。</p>
</li>
<li><p>Pfam: 蛋白结构域注释的分类系统。</p>
</li>
<li><p>GO: 基因本体论注释数据库。</p>
</li>
<li><p>KEGG: 代谢通路注释数据库。</p>
</li>
</ul>
<p>数据库的下载非常耗时，不管是eggnog还是interproscan，linux下几十KB&#x2F;s，window下几百KB&#x2F;s(需科学上网)，耗时长度以天计算，可以尝试先在window系统下载，再转移到linux下。</p>
<h2 id="1-1-Nr-x2F-Nt数据库"><a href="#1-1-Nr-x2F-Nt数据库" class="headerlink" title="1.1. Nr&#x2F;Nt数据库"></a>1.1. Nr&#x2F;Nt数据库</h2><p><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/nr.gz">nr数据库</a>；<br><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/nt.gz">nt数据库</a>；</p>
<h2 id="1-2-UniProt数据库（Swiss-Prot和TrEMBL）"><a href="#1-2-UniProt数据库（Swiss-Prot和TrEMBL）" class="headerlink" title="1.2. UniProt数据库（Swiss-Prot和TrEMBL）"></a>1.2. UniProt数据库（Swiss-Prot和TrEMBL）</h2><p>UniProt数据库包括人工校正的高质量数据库Swiss-Prot和软件自动注释的TrEMBL数据库。</p>
<p><a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz">Swiss-Prot数据库</a>：高质量，手工注释，非冗余的蛋白质序列数据库。非常可靠。</p>
<p><a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.fasta.gz">TrEMBL数据库</a>：自动注释的蛋白质序列数据库。</p>
<p>UniProt将完整数据库分类拆分成几个<a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/">子库</a>。包括古菌，真菌，植物，人，哺乳动物，脊椎动物，无脊椎动物，啮齿动物等。</p>
<h2 id="1-3-GO数据库"><a href="#1-3-GO数据库" class="headerlink" title="1.3. GO数据库"></a>1.3. GO数据库</h2><p>[GO数据库](<a target="_blank" rel="noopener" href="http://geneontology.org/docs/download-ontology/%EF%BC%89">http://geneontology.org/docs/download-ontology/）</a></p>
<h2 id="1-4-KEGG数据库"><a href="#1-4-KEGG数据库" class="headerlink" title="1.4. KEGG数据库"></a>1.4. KEGG数据库</h2><p>KEGG(Kyoto Encyclopedia of Genes and Genomes)京都基因和基因组百科全书，是研究Pathway代谢通路最主要的数据库，整合了基因组、化学、系统及疾病和健康的信息。</p>
<h2 id="1-5-COG直系同源蛋白数据库"><a href="#1-5-COG直系同源蛋白数据库" class="headerlink" title="1.5. COG直系同源蛋白数据库"></a>1.5. COG直系同源蛋白数据库</h2><p><a href="ftp://ftp.ncbi.nlm.nih.gov/pub/COG/COG2014/data/">COG数据库</a>:NCBI开发的用于同源蛋白注释的数据库。<br>fun2003-2014.tab是COG的分类信息，将COG的功能分为26个类别，每个类别用一个字母表示。<br>cognames2003-2014.tab是COG的详细信息，包括编号，对应的分类，功能描述等。<br>cog2003-2014.csv是蛋白和COG对应关系。<br>prot2003-2014.fa.gz是fa格式的蛋白序列和注释信息。</p>
<h2 id="1-6-KOG数据库"><a href="#1-6-KOG数据库" class="headerlink" title="1.6. KOG数据库"></a>1.6. KOG数据库</h2><p>KOG（Clusters of orthologous groups for eukaryotic complete genomes）真核生物蛋白相邻类的聚簇。构成每个KOG的蛋白都是被假定来自一个祖先蛋白，要么是orthologs，要么是paralogs。orthologs是指来自不同物种的由垂直家系（物种形成）进化而来的蛋白，并且保留与原始蛋白相同的功能；paralogs是指在一定物种中来源于基因复制的蛋白，可能会进化成新的与原来有关的功能。<br><a href="ftp://ftp.ncbi.nlm.nih.gov/pub/COG/KOG/kyva">KOG数据库</a></p>
<h2 id="1-7-eggNOG数据库"><a href="#1-7-eggNOG数据库" class="headerlink" title="1.7. eggNOG数据库"></a>1.7. eggNOG数据库</h2><p>eggNOG（evolutionary genealogy of genes：non-supervised orthologous groups）由EMBL创建，是对NCBI的COG数据库的拓展，提供了不同分类水平蛋白的直系同源分组（Orthologous Groups，OG），包括真核生物，原核生物，病毒的信息。拓展了COG的分类，采用无监督聚类算法在全基因组范围内推导基因功能，更适合谱系特征基因的分析。</p>
<p><a target="_blank" rel="noopener" href="http://eggnog5.embl.de/">eggNOG数据库</a>中，e5.proteomes.faa是所有的蛋白组序列，e5.taxid_info.tsv是taxid对应的物种名及完整的谱系信息，e5.og_annotations.tsv是所有NOG group信息（第一列为Taxid，第二列为NOG groups，第三列为COG归属，第四列为Function）。per_tax_level下不同taxonomy level的members.tsv文件分别是相应level的蛋白序列id和NOG group的对应关系（第一列taxid，第二列NOG groups，第三列为该NOG group所包含的蛋白序列数目，第四列为该NOG group所包含的物种数目，第五列是该NOG group所包含的蛋白序列id，第六列是该NOG group所包含的物种taxid。</p>
<h2 id="1-8-pfam数据库"><a href="#1-8-pfam数据库" class="headerlink" title="1.8. pfam数据库"></a>1.8. pfam数据库</h2><p><a target="_blank" rel="noopener" href="http://pfam.xfam.org/">pfam数据库</a>是蛋白质家族数据库，根据多序列比对结果和隐马尔科夫模型，将蛋白分成不同家族。</p>
<h2 id="1-9-string数据库"><a href="#1-9-string数据库" class="headerlink" title="1.9. string数据库"></a>1.9. string数据库</h2><p><a target="_blank" rel="noopener" href="https://string-db.org/">string数据库</a>。<br>搜寻已知蛋白质和预测蛋白质之间的相关关系的系统，包括蛋白质之间的物理作用和间接的功能相关性。基于染色体临近，系统进化谱，基因融合和基因芯片数据等计算基因或蛋白的共表达。<br>包括动物转录因子数据AnimalTFDB3.0，植物转录因子数据<a target="_blank" rel="noopener" href="http://plntfdb.bio.uni-potsdam.de/v3.0/">PlnTFDB</a>，真菌转录因子数据库<a target="_blank" rel="noopener" href="http://ftfd.snu.ac.kr/index.php?a=view">Fungal TFDB</a>。</p>
<p>植物转录因子数据库收录了大部分植物模式物种共20个种的84个转录因子家族，包含28193 protein models，26184 distinct protein sequences，支持在线blast和本地blast。<br>分析工具：<a target="_blank" rel="noopener" href="http://itak.feilab.net/cgi-bin/itak/index.cgi">iTAK</a>，软件内置了PlantTFDB的数据库，可直接用于预测植物转录因子。<br>结果在classification.txt文件中。</p>
<h1 id="2-基因功能注释"><a href="#2-基因功能注释" class="headerlink" title="2. 基因功能注释"></a>2. 基因功能注释</h1><p>notes: 注释分析中要保证蛋白序列中没有代表氨基酸字符以外的字符，比如说有些软件会把最后一个终止密码子翻译成”.”或者”*”，需要删除。</p>
<h2 id="2-1-下载数据库手动注释–以Uniprot为例"><a href="#2-1-下载数据库手动注释–以Uniprot为例" class="headerlink" title="2.1. 下载数据库手动注释–以Uniprot为例"></a>2.1. 下载数据库手动注释–以Uniprot为例</h2><h3 id="2-1-1-下载数据库"><a href="#2-1-1-下载数据库" class="headerlink" title="2.1.1. 下载数据库"></a>2.1.1. 下载数据库</h3><p>这里以Uniprot为例，下载其他数据库后可用同样方法进行手动注释。</p>
<p>可以uniprot_sprot和uniprot_trembl的植物数据库合并在一起。<br><code>wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/uniprot_sprot_plants.dat.gz #</code>;<br><code>wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/uniprot_trembl_plants.dat.gz # 11.5G</code>;<br><code>zcat uniprot_sprot_plants.dat.gz uniprot_trembl_plants.dat.gz &gt; uniprot_plants.dat</code>;</p>
<p>转换格式：<br>用python的SeqIO模块转换</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> Bio <span class="keyword">import</span> SeqIO</span><br><span class="line">count = SeqIO.convert(<span class="string">&quot;uniprot_plants.dat&quot;</span>,<span class="string">&quot;swiss&quot;</span>,<span class="string">&quot;uniprot_plants.fa&quot;</span>,<span class="string">&quot;fasta&quot;</span>)</span><br><span class="line"><span class="built_in">print</span>(<span class="string">&quot;Converted %i records&quot;</span> % count)</span><br></pre></td></tr></table></figure>

<h3 id="2-1-2-构建blastp索引"><a href="#2-1-2-构建blastp索引" class="headerlink" title="2.1.2. 构建blastp索引"></a>2.1.2. 构建blastp索引</h3><p><code>makeblastdb -in uniprot_plants.fasta -dbtype prot -title uniprot_plants -parse_seqids</code><br>或者更快的<br><code>diamond makedb --in uniprot_plants.fasta -d uniprot_plants</code></p>
<h3 id="2-1-3-用blastp比对蛋白序列protein-fa和数据库uniprot-plants"><a href="#2-1-3-用blastp比对蛋白序列protein-fa和数据库uniprot-plants" class="headerlink" title="2.1.3. 用blastp比对蛋白序列protein.fa和数据库uniprot_plants"></a>2.1.3. 用blastp比对蛋白序列protein.fa和数据库uniprot_plants</h3><p><code>blastp -query protein.fa -out uniprot_plants.xml -db uniprot_plants.fasta -evalue 1e-5 -outfmt 6 -num_threads 24 &amp;</code><br>或者更快的<br><code>diamond blastp -d uniprot_plants.dmnd -q proteins.fasta --evalue 1e-5 &gt; blastp.outfmt6</code></p>
<h3 id="2-1-4-比对结果中筛选每个query的最佳subject"><a href="#2-1-4-比对结果中筛选每个query的最佳subject" class="headerlink" title="2.1.4. 比对结果中筛选每个query的最佳subject"></a>2.1.4. 比对结果中筛选每个query的最佳subject</h3><p>jcvi可用conda安装<br><code>python -m jcvi.formats.blast best -n 1 blastp.outfmt6</code><br>生成blastp.outfmt6.best</p>
<h3 id="2-1-5-使用add-annotation-from-dat-py根据blastp输出从dat提取GO-x2F-KEGG-x2F-同源基因"><a href="#2-1-5-使用add-annotation-from-dat-py根据blastp输出从dat提取GO-x2F-KEGG-x2F-同源基因" class="headerlink" title="2.1.5. 使用add_annotation_from_dat.py根据blastp输出从dat提取GO&#x2F;KEGG&#x2F;同源基因"></a>2.1.5. 使用add_annotation_from_dat.py根据blastp输出从dat提取GO&#x2F;KEGG&#x2F;同源基因</h3><p>python add_annotation_from_dat.py blastp.outfmt6.best uniprot_plants.dat<br>生成swiss_annotation.tsv文件；包含以下几列。</p>
<ul>
<li>gene id</li>
<li>uniprot accession</li>
<li>identity</li>
<li>homology species</li>
<li>EnsemblPlants</li>
<li>GO ID</li>
<li>GO component, CC&#x2F;MF&#x2F;BP</li>
<li>evidence</li>
</ul>
<p>参考脚本<a target="_blank" rel="noopener" href="https://github.com/xuzhougeng/myscripts/blob/master/annotation/add_annotation_from_dat.py">add_annotation_from_dat.py</a></p>
<h3 id="2-1-6-blast2go"><a href="#2-1-6-blast2go" class="headerlink" title="2.1.6. blast2go"></a>2.1.6. blast2go</h3><p>blast2go的功能是将blast注释结果转换成GO注释，但使用较为复杂，下载数据库耗时长，不推荐使用。<br><a target="_blank" rel="noopener" href="https://www.plob.org/article/1299.html">blast2go教程</a>；<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/ad44b2a837c9">blast2go教程</a><br>下载数据库<br>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz">ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz</a> #650M<br>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2accession.gz">ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2accession.gz</a> #2.3G<br>wget <a href="ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz">ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz</a> #10G<br>wget <a target="_blank" rel="noopener" href="http://release.geneontology.org/2017-01-01/mysql_dumps/go_monthly-assocdb-data.gz">http://release.geneontology.org/2017-01-01/mysql_dumps/go_monthly-assocdb-data.gz</a> #6.3G</p>
<h2 id="2-2-eggnog-mapper注释"><a href="#2-2-eggnog-mapper注释" class="headerlink" title="2.2. eggnog-mapper注释"></a>2.2. eggnog-mapper注释</h2><p>eggNOG(evolutionary genealogy of genes: Non-supervised Orthologous Groups)。<br><a target="_blank" rel="noopener" href="http://eggnog5.embl.de/download/emapperdb-5.0.0/">eggNOG数据库</a>是数据库，eggnog-mapper是使用eggnog数据库作为参考的注释软件或网页，<a target="_blank" rel="noopener" href="http://eggnog-mapper.embl.de/">eggnog-mapper网页版</a>。</p>
<h3 id="2-2-1-eggNOG本地注释"><a href="#2-2-1-eggNOG本地注释" class="headerlink" title="2.2.1. eggNOG本地注释"></a>2.2.1. eggNOG本地注释</h3><p>下载数据库本地注释的方法：<br>在eggnog网站下载对应使用的数据库。<br><a target="_blank" rel="noopener" href="http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog.db.gz">http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog.db.gz</a><br><a target="_blank" rel="noopener" href="http://eggnogdb.embl.de/download/emapperdb-5.0.0/eggnog_proteins.dmnd.gz">http://eggnogdb.embl.de/download/emapperdb-5.0.0/eggnog_proteins.dmnd.gz</a><br>注释命令：<br>python2 &#x2F;home&#x2F;leon&#x2F;software&#x2F;3.Function&#x2F;eggnog-mapper.orginal&#x2F;emapper.py -i Tov.final.all.maker.transcripts.fasta -o run_egg –override  -m diamond  –cpu 80</p>
<h3 id="2-2-2-eggNOG-mapper网页版注释"><a href="#2-2-2-eggNOG-mapper网页版注释" class="headerlink" title="2.2.2. eggNOG-mapper网页版注释"></a>2.2.2. eggNOG-mapper网页版注释</h3><h4 id="2-2-2-1-注释原理"><a href="#2-2-2-1-注释原理" class="headerlink" title="2.2.2.1. 注释原理"></a>2.2.2.1. 注释原理</h4><p>注释分为三步：</p>
<ol>
<li>序列比对<br>用hmmer3对所有蛋白序列在eggNOG数据库中搜索，找到最佳匹配的HMM；再用phmmer在最佳匹配的HMM对应的一组eggNOG蛋白中进一步搜索，每条蛋白序列的最佳匹配结果以seed orthlog形式存放。<br>除了HMMER3还可选DIAMOND直接对所有eggNOG蛋白序列进行搜索，速度更快，若数据大或在eggNOG搜集的物种库中有近缘种可选择DIAMOND。</li>
<li>推测直系同源基因<br>基于预分析的eggNOG进化树数据库，提取最佳匹配seed orthlog的蛋白的一组更精细的直系同源基因，根据bit-score或E-value对结果过滤，剔除同源性不高的结果。</li>
<li>功能注释<br>用于搜索的蛋白序列对应的直系同源基因的功能描述就是最终的注释结果。比如说GO, KEGG, COG等。</li>
</ol>
<h4 id="2-2-2-2-注释"><a href="#2-2-2-2-注释" class="headerlink" title="2.2.2.2. 注释"></a>2.2.2.2. 注释</h4><p>推荐使用<a target="_blank" rel="noopener" href="http://eggnog-mapper.embl.de/">eggnog-mapper网页版</a>注释，避免下载数据库的繁琐。限制10万条蛋白序列，数量超过可以切分后分开注释。</p>
<p>只需要上传氨基酸序列，点击提交；然后<strong>在收到的邮件里点击链接，开始任务</strong>即可。</p>
<h4 id="2-2-2-3-注释结果"><a href="#2-2-2-3-注释结果" class="headerlink" title="2.2.2.3. 注释结果"></a>2.2.2.3. 注释结果</h4><p>eggnog-mapper会生成五个文件:</p>
<ul>
<li>[project_name].emapper.gff: gff注释文件。</li>
<li>[project_name].emapper.orthologs: 搜索匹配到的物种和orthologs的信息。</li>
<li>[project_name].emapper.seed_orthologs: 记录每个用于搜索序列对的的最佳的OG，也就是</li>
<li>[project_name].emapper.annotations.tsv: 该文件提供了最终的注释结果。大部分需要的内容都可以通过写脚本从从提取，一共有13列。</li>
<li>[project_name].emapper.annotations.xlsx：最终注释结果，另一种格式。</li>
</ul>
<p>其中[project_name].emapper.annotations.tsv每一列对应的记录如下：</p>
<ol>
<li>query: 检索的基因名或者其他ID</li>
<li>seed_ortholog: eggNOG中最佳的蛋白匹配</li>
<li>evalue: 最佳匹配的e-value</li>
<li>score: 最佳匹配的bit-score</li>
<li>eggNOG_OGs：匹配到的eggNOG数据库里的OGs</li>
<li>max_annot_lvl：</li>
<li>COG_category：COG类别</li>
<li>Description：功能描述</li>
<li>Preferred_name: 预测的基因名，特别指的是类似AP2有一定含义的基因名，而不是- AT2G17950这类编号</li>
<li>GOs: 推测的GO的词条， 未必最新</li>
<li>EC：</li>
<li>KEGG_ko: 推测的KEGG KO词条， 未必最新</li>
<li>KEGG_Pathway：推测的KEGG通路词条</li>
<li>KEGG_Module：</li>
<li>KEGG_Reaction：</li>
<li>KEGG_rclass：</li>
<li>BRITE:</li>
<li>KEGG_TC:</li>
<li>CAZy</li>
<li>BiGG_Reaction: BiGG代谢反应的预测结果</li>
<li>PFAMs:注释到PFAM数据</li>
</ol>
<h4 id="2-2-2-4-结果整理"><a href="#2-2-2-4-结果整理" class="headerlink" title="2.2.2.4. 结果整理"></a>2.2.2.4. 结果整理</h4><ul>
<li><code>cat eggnog.web/*.emapper.annotations.tsv |sed &#39;/#.*/d&#39; |cut -f1,8,9 |awk -F&quot;\t&quot; &#39;&#123;print $1&quot;\t&quot;$3&quot;: &quot;$2&#125;&#39; |sed &#39;/-: -/d&#39;|sort -k 1.3n |uniq |sed &#39;1i\gene\teggnog.preferred.name: description&#39; &gt;eggnog.anno</code> 提取信息列：1（query），8（Description），9（Preferred name），10（GOs），12（KEGG_ko）列。然后提取1（query），8（Description），9（Preferred name）注释信息。eggnog注释是每个基因一行信息，所以eggnog.anno的行数就是注释到的基因数量。</li>
<li><code>cat eggnog.web/*.emapper.annotations.tsv |sed &#39;/#.*/d&#39; |cut -f1,10 |sed &#39;/\t-$/d&#39; |sort -k 1.3n |uniq |sed &#39;1i\gene\teggnog.go&#39; &gt;eggnog.go.anno</code> 提取信息列：1（query），10（GOs）的eggnog的GO注释信息。</li>
<li><code>cat eggnog.web/*.emapper.annotations.tsv |sed &#39;/#.*/d&#39; |cut -f1,12 |sed &#39;/\t-$/d&#39; |sort -k 1.3n |uniq |sed &#39;1i\gene\teggnog.kegg&#39; &gt;eggnog.kegg.anno</code> 提取信息列：1（query），12（KEGG_ko）的eggnog的KEGG注释信息。</li>
</ul>
<h2 id="2-3-InterProScan"><a href="#2-3-InterProScan" class="headerlink" title="2.3. InterProScan"></a>2.3. InterProScan</h2><h3 id="2-3-1-Interproscan"><a href="#2-3-1-Interproscan" class="headerlink" title="2.3.1. Interproscan"></a>2.3.1. Interproscan</h3><p>interpro是数据库，interproscan是可以使用interpro的注释软件。<br>如果序列少，可以使用<a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/interpro/search/sequence/">interproscan网页版</a>注释，避免下载数据库的繁琐。限制一次注释一条，长度小于40000个蛋白序列，超过可以切分后分开注释。</p>
<p>interproscan可实现多个数据库同时注释，包括：</p>
<ul>
<li>InterPro注释</li>
<li>Pfam数据库注释(可以通过hmmscan搜索pfam数据库完成)</li>
<li>PANTHER数据库注释</li>
<li>GO注释(可以基于NR和Pfam等数据库，然后BLAST2GO完成)</li>
<li>Reactome通路注释，不同于KEGG<br>还有很多数据库。</li>
</ul>
<h3 id="2-3-2-Interproscan的安装和数据库下载"><a href="#2-3-2-Interproscan的安装和数据库下载" class="headerlink" title="2.3.2. Interproscan的安装和数据库下载"></a>2.3.2. Interproscan的安装和数据库下载</h3><p>如果注释的文件比较大或者比较多，可以下载本地版注释，下载过程非常慢。根据你的linux版本和发布日期来选择最适版本，软件很大最近版大约9.1G。建议下载对应的md5文件，用 md5sum -c *.md5来检查下载的是否完全。</p>
<h4 id="2-3-2-1-Interproscan安装"><a href="#2-3-2-1-Interproscan安装" class="headerlink" title="2.3.2.1. Interproscan安装"></a>2.3.2.1. Interproscan安装</h4><p>interproscan软件地址：<a href="ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/">ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/</a>, 选择最新版本，比如5.47-82.0，则<code>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.47-82.0/*</code>，版本号目录下有两个文件，一个软件的tar.gz压缩文件，一个对应的md5文件，都下下来。</p>
<h4 id="2-3-2-2-软件自带数据库"><a href="#2-3-2-2-软件自带数据库" class="headerlink" title="2.3.2.2. 软件自带数据库"></a>2.3.2.2. 软件自带数据库</h4><p>软件本身自带了很多数据库，不需要安装，5.47-82.0版本带有CDD-3.17,Coils-2.2.1,Gene3D-4.2.0,Hamap-2020_01,MobiDBLite-2.0,PANTHER-15.0,Pfam-33.1,PIRSF-3.10,PRINTS-42.0,ProSitePatterns-2019_11,ProSiteProfiles-2019_11,SFLD-4,SMART-7.1,SUPERFAMILY-1.75,TIGRFAM-15.0数据库。</p>
<h4 id="2-3-2-3-关于PANHTER数据库"><a href="#2-3-2-3-关于PANHTER数据库" class="headerlink" title="2.3.2.3. 关于PANHTER数据库"></a>2.3.2.3. 关于PANHTER数据库</h4><p>PANHTER(Protein ANalysis THrough Evolutionary Relationships) 数据库是Gene Ontology Phylogenetic Annotation Project的一部分。</p>
<ol>
<li>发现interproscan 5.47-82.0软件自带PANHTER数据库，所以不用单独下载啦。</li>
<li>若是没有带PANHTER数据库，则下载构建本地PANHTER数据库，需要下载并解压到软件安装目录下的 &#x2F;path of interproscan&#x2F;data&#x2F;。</li>
</ol>
<ul>
<li>PANHTER下载地址：<a href="ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/">ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/</a>,</li>
<li>选择最新版本，比如14.1，则<code>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/panther-data-14.1.tar.gz*</code>，下载14.1和对应的md5文件。</li>
<li>md5验证两个文件下载完全后，tar zxf解压缩，并把PANHTER数据库放进interproscan的data目录下。</li>
</ul>
<h4 id="2-3-2-4-java和python版本检查"><a href="#2-3-2-4-java和python版本检查" class="headerlink" title="2.3.2.4. java和python版本检查"></a>2.3.2.4. java和python版本检查</h4><p>软件的运行依赖java11和python3，<code>java -version</code>和<code>python --version</code>检查版本是否正确。</p>
<h4 id="2-3-2-5-测试安装"><a href="#2-3-2-5-测试安装" class="headerlink" title="2.3.2.5. 测试安装"></a>2.3.2.5. 测试安装</h4><p>解压缩后运行 interproscan.sh 测试是否成功安装，弹出help界面就是成功安装，用测试文件运行下，<code>interproscan.sh -i test_proteins.fasta</code>，没报错则软件可正常运行。</p>
<h4 id="2-3-2-6-tips"><a href="#2-3-2-6-tips" class="headerlink" title="2.3.2.6. tips"></a>2.3.2.6. tips</h4><ul>
<li>如果本地化运行，可以把interproscan.properties文件里的<a href="precalculated.match.lookup.service.url=http://www.ebi.ac.uk/interpro/match-lookup">这行代码</a>注释掉，就不进行match.lookup检查，这个应该是更新数据库的检查。</li>
<li>如果不注释，在运行时存储interproscan数据库端网络慢时（大部分情况下）会输出一些Connection timed out和ERROR - Lookup version check failed的报错信息，但the analysis will continue to run locally.本地运行仍在继续。</li>
</ul>
<h3 id="2-3-3-Interproscan运行"><a href="#2-3-3-Interproscan运行" class="headerlink" title="2.3.3. Interproscan运行"></a>2.3.3. Interproscan运行</h3><p>常用运行参数<br><code>interproscan -i pep.fa -b out.iprscan -goterms -iprlookup -pa -dp -cpu 24</code></p>
<p>interproscan的参数：</p>
<ul>
<li><p>-i,–input 输入文件，一般要为fasta格式，注意不要带有除氨基酸符号的其他特殊符号（比如代表终止密码子的*），gene&#x2F;氨基酸的名称不能为空。</p>
</li>
<li><p>-b,–output-file-base 指定输出文件的路径和文件名，默认是输入文件的路径</p>
</li>
<li><p>-appl,–applications 指定使用Interpro中哪些数据库，默认使用全部数据库</p>
</li>
<li><p>-f,–formats 用于指定输出文件的后缀，蛋白序列默认输出TSV, XML and GFF3</p>
</li>
<li><p>-t,–seqtype 输入文件的序列类型，p为protein，n为dna&#x2F;rna，默认是p</p>
</li>
<li><p>-cpu 指定使用线程数</p>
</li>
<li><p>-goterms 打开GO注释，依赖iprlookup参数，switch on lookup of corresponding Gene Ontology annotation (IMPLIES -iprlookup option)</p>
</li>
<li><p>-iprlookup Also include lookup of corresponding InterPro annotation in the TSV and GFF3 output formats.</p>
</li>
<li><p>-pa Optional, switch on lookup of corresponding Pathway annotation (IMPLIES -iprlookup option)</p>
</li>
<li><p>-dp 关闭precalculated match lookup service，默认的是开启。根据md5值来快速检验这上传的数据是否已经被注释了，如果是已经注释了就直接出结果。</p>
</li>
</ul>
<h3 id="2-3-4-Interproscan结果解释"><a href="#2-3-4-Interproscan结果解释" class="headerlink" title="2.3.4. Interproscan结果解释"></a>2.3.4. Interproscan结果解释</h3><p>生成的out.tsv和out.gff3等是不同格式的注释文件，这里使用tsv格式。</p>
<p>tsv格式文件每一列含义：</p>
<ol>
<li>蛋白名字</li>
<li>序列MD5 disest</li>
<li>序列长度</li>
<li>所用的库（Pfam&#x2F;PANTHER&#x2F;RPINTS etc）</li>
<li>匹配的数据编号（PF00001&#x2F;PTHR00001 etc)</li>
<li>功能描述</li>
<li>起始位置</li>
<li>终止位置</li>
<li>e-value得分</li>
<li>匹配的状态T: true</li>
<li>日期</li>
<li>interPro 注释编号</li>
<li>interPro 注释描述</li>
<li>GO注释</li>
<li>Pathways 注释</li>
</ol>
<p>每行是一条蛋白序列注释到的每个数据库的每一个匹配，所以每个基因会有多行。<br>根据第四列-数据库名称来提取不同数据库的注释结果。这里提取pfam和PANTHER两个数据库的结果。</p>
<h3 id="2-3-5-Interproscan结果整理"><a href="#2-3-5-Interproscan结果整理" class="headerlink" title="2.3.5. Interproscan结果整理"></a>2.3.5. Interproscan结果整理</h3><ol>
<li>提取interpro数据库注释内容并转化成单基因单行格式<br><code>cat *.iprscan.tsv |cut -f1,12,13|grep &quot;IPR&quot;|awk &#39;&#123;print $1&quot;\t&quot;$2&quot;: &quot;$3&#125;&#39;|sort -k 1.3n|uniq |awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n |sed &#39;1i\gene\tinterpro&#39; &gt;iprs.interpro.anno</code></li>
</ol>
<p>notes：</p>
<ul>
<li>注意排序和去重。</li>
<li>因为我的geneID是tg00001这种格式，只有两个字母，所以sort排序按数字从第三个<code>sort -k 1.3n</code>开始排序，可以根据不同geneID更改。</li>
<li>其中<code>awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n</code>部分是单基因多行注释转化成单基因单行注释的命令。</li>
<li>用cut截取1，12，13三列是因为有些行没有12列及之后数据，直接用awk会失效。</li>
<li>用grep搜索IPR是因为直接用awk匹配第12列不能匹配完全，比如SUPERFAMILY数据库的比对就匹配不了。</li>
</ul>
<ol start="2">
<li><p>提取Pfam数据库注释结果并转化成单基因单行格式<br><code>cat *.iprscan.tsv  |awk &#39;$4 == &quot;Pfam&quot; &#123;print $1&quot;\t&quot;$5&quot;: &quot;$6&#125;&#39; |sort -k 1.3n |uniq|awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n |sed &#39;1i\gene\tinterproscan.pfam&#39;&gt;iprs.pfam.anno</code></p>
</li>
<li><p>提取PANTHER数据库注释结果并转化成单基因单行格式<br><code>cat *.iprscan.tsv |awk &#39;$4 == &quot;PANTHER&quot; &#123;print $1&quot;\t&quot;$5&quot;: &quot;$6&#125;&#39;|sort -k 1.3n|uniq |awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n |sed &#39;1i\gene\tinterproscan.panther&#39;&gt;iprs.panther.anno</code></p>
</li>
</ol>
<h2 id="2-4-PANNZER-Protein-ANNotation-with-Z-scoRE"><a href="#2-4-PANNZER-Protein-ANNotation-with-Z-scoRE" class="headerlink" title="2.4. PANNZER(Protein ANNotation with Z-scoRE)"></a>2.4. PANNZER(Protein ANNotation with Z-scoRE)</h2><h3 id="2-4-1-PANNZER"><a href="#2-4-1-PANNZER" class="headerlink" title="2.4.1. PANNZER"></a>2.4.1. PANNZER</h3><p>网站注释：<a target="_blank" rel="noopener" href="http://ekhidna2.biocenter.helsinki.fi/sanspanz/">http://ekhidna2.biocenter.helsinki.fi/sanspanz/</a></p>
<ul>
<li>用于原核和真核蛋白序列的功能注释。</li>
<li>使用的Uniprot数据库，与数据库更新一致。</li>
<li>用于预测functional description (DE) 和 GO classes。</li>
</ul>
<p>注释速度：7万条序列4个小时。</p>
<h3 id="2-4-2-结果文件"><a href="#2-4-2-结果文件" class="headerlink" title="2.4.2. 结果文件"></a>2.4.2. 结果文件</h3><p>notes: </p>
<ul>
<li>网页版的结果，注意需要把进度条拉到最下面再复制保存文本，否则没有显示完全，ctrl+a全选只会选中已显示的内容，未显示的未被选中。</li>
<li>(20210813)今天发现不需要拉进度条这么繁琐，在网页右键另存为可以报错文本格式的结果。</li>
</ul>
<ol>
<li>*.anno.out.txt是蛋白序列的注释信息，包含六列：</li>
</ol>
<ul>
<li>qpid：序列ID；</li>
<li>type：original_DE,qseq（前两种类型是每个基因都有的）,DE,MF_ARGOT,BP_ARGOT,CC_ARGOT,EC_ARGOT等（后面的类型是注释到的基因才有的）；</li>
<li>score：匹配分数；</li>
<li>PPV：不知道是啥指标；</li>
<li>id：数字；</li>
<li>desc：功能描述；每个序列都有多行信息，没被注释到的序列只有两行第二列type分别为original_DE和qseq的基本信息。</li>
</ul>
<p>每个基因有只有一行第二列type为DE的数据，此时第六列的描述是对基因的功能描述，即我们想要的基因功能注释信息。</p>
<ol start="2">
<li>*.DE_prediction.txt是注释到DE（functional description）的序列的注释信息。<br>重要的是第一列（qpid）：基因id；第九列（desc）：DE描述信息，即注释到uniprot数据库的描述信息；第十列（genename）：基因名字。</li>
<li>*.GO_prediction.txt是注释到GO的序列的注释信息。<br>重要的是第一列（qpid）：基因id；第二列（ontology）：GO类别，BP&#x2F;MF&#x2F;CC三个类别；第三列（goid）：GO ID；第四列（desc）：GO描述信息。</li>
</ol>
<h3 id="2-4-3-结果整理"><a href="#2-4-3-结果整理" class="headerlink" title="2.4.3. 结果整理"></a>2.4.3. 结果整理</h3><ol>
<li><p><code>cat pannzer2/anno.out.txt |awk &#39;$2 == &quot;DE&quot; &#123;print $1&quot;\t&quot;$6&#125;&#39; |sort -k 1.3n |uniq |sed &#39;1i\gene\tpannzer2.uniprot&#39; &gt;pannzer2.uniprot.anno</code>把Uniprot数据库功能注释信息提取出来。</p>
</li>
<li><p><code>cat pannzer2/GO.out.txt |awk &#39;&#123;print $1&quot;\tGO:&quot;$3,$4&#125;&#39; |awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n |uniq |sed &quot;1s/.*/gene\tpannzer2.go/&quot; &gt;pannzer2.go.anno</code>把Uniprot数据库的GO注释提取出来并转化成单基因单行信息。</p>
</li>
</ol>
<h2 id="2-5-Mercator4——植物基因组功能注释"><a href="#2-5-Mercator4——植物基因组功能注释" class="headerlink" title="2.5. Mercator4——植物基因组功能注释"></a>2.5. Mercator4——植物基因组功能注释</h2><p>网站注释：<a target="_blank" rel="noopener" href="https://www.plabipd.de/portal/web/guest/mercator4/-/wiki/Mercator4/FrontPage">https://www.plabipd.de/portal/web/guest/mercator4/-/wiki/Mercator4/FrontPage</a></p>
<h3 id="2-5-1-注释原理"><a href="#2-5-1-注释原理" class="headerlink" title="2.5.1. 注释原理"></a>2.5.1. 注释原理</h3><ul>
<li>针对植物基因组的功能注释，以整个植物界的高质量带注释基因组为种子，找直系同源蛋白。</li>
<li>基于隐马尔可夫模型（HMM），用HMMER3软件的hmmscan对每条蛋白做HMM测试，然后被分到Bin-50类别下。</li>
<li>其余未被分类的蛋白与Swiss-Prot数据库做blastp比较，然后注释。</li>
</ul>
<p>注释速度：7万条蛋白3min注释完，神速！</p>
<h3 id="2-5-2-注释结果"><a href="#2-5-2-注释结果" class="headerlink" title="2.5.2. 注释结果"></a>2.5.2. 注释结果</h3><p>两个文件，点击即可下载。</p>
<ol>
<li>Mercator4 annotation results<br>一共有五列，每列信息都在单引号内：</li>
</ol>
<ul>
<li>BINCODE：有层级的有序数字列表</li>
<li>NAME：这一层级的功能名字，代表蛋白数据库里的功能层级</li>
<li>IDENTIFIER：基因ID，在末级层级才会有基因ID</li>
<li>DESCRIPTION：功能描述</li>
<li>TYPE：类型</li>
</ul>
<ol start="2">
<li>Mercator4 annotated fasta file</li>
</ol>
<h3 id="2-5-3-结果整理"><a href="#2-5-3-结果整理" class="headerlink" title="2.5.3. 结果整理"></a>2.5.3. 结果整理</h3><ol>
<li>从Mercator4 annotation results提取注释结果出来并转化成单基因单行信息格式，其中geneIDprefix根据基因名的前缀替换：</li>
</ol>
<p><code>cat mercator4v3/*.annotations.results.txt |awk -F&quot;\t&quot; &#39;$3 ~ &quot;geneIDprefix&quot; &#123;print $3&quot;\t&quot;$2&quot;: &quot;$4&#125;&#39; |sed &quot;s/&#39;//g&quot; |sort -k 1.3n |uniq |grep -v &quot;no hits&quot;|awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;; &quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39;|sed &quot;s/; $//g&quot; |sort -k 1.3n |sed &quot;s/not assigned.annotated/-/g&quot; |sed &quot;1i\gene\tmercator&quot;&gt;mercator.anno</code></p>
<p>注意检查是否还有no hits或者not assigned.annotated，有的话都替换掉。</p>
<ol start="2">
<li>从Mercator4 annotated fasta file中以”not annotated”作为关键词搜索，可以获取未被注释到的基因。</li>
</ol>
<h1 id="3-基因功能注释的整合"><a href="#3-基因功能注释的整合" class="headerlink" title="3. 基因功能注释的整合"></a>3. 基因功能注释的整合</h1><p>提交功能注释文件有两种方案：</p>
<ol>
<li>整合功能注释和结构注释：把功能注释整理后添加到结构注释文件sample.gff第九列的product属性，与结构注释一同提交给NCBI等公共数据库</li>
<li>单独提供功能注释：通过Dryad【收费】或Figshare【免费】等文件分享网站单独提供功能注释文件</li>
</ol>
<h2 id="3-1-方案一：整合功能注释和结构注释"><a href="#3-1-方案一：整合功能注释和结构注释" class="headerlink" title="3.1. 方案一：整合功能注释和结构注释"></a>3.1. 方案一：整合功能注释和结构注释</h2><p>功能注释以product属性值的形式提供，添加到结构注释gff文件的第九列。</p>
<h3 id="3-1-1-基因功能注释整理"><a href="#3-1-1-基因功能注释整理" class="headerlink" title="3.1.1. 基因功能注释整理"></a>3.1.1. 基因功能注释整理</h3><p>不同软件的基因功能注释结果都预先进行了整理（参考各个软件后的注释整理部分），整理成两列的格式，首行为表头，第一列为基因ID（gene），第二列为基因功能注释的描述信息（标明注释来源），每个注释软件&#x2F;数据库的描述信息单独为一列。</p>
<p>目前共有9个文件被整理出来：</p>
<ol>
<li>eggnog.anno</li>
<li>eggnog.go.anno</li>
<li>eggnog.kegg.anno</li>
<li>iprs.interpro.anno</li>
<li>iprs.panther.anno</li>
<li>iprs.pfam.anno</li>
<li>mercator.anno</li>
<li>pannzer2.go.anno</li>
<li>pannzer2.uniprot.anno</li>
</ol>
<h3 id="3-1-2-合并功能注释和结构注释"><a href="#3-1-2-合并功能注释和结构注释" class="headerlink" title="3.1.2. 合并功能注释和结构注释"></a>3.1.2. 合并功能注释和结构注释</h3><p>合并步骤：</p>
<ol>
<li><p>整理注释结果<br>先把所有功能注释转换成单基因单注释的格式(单基因多行注释)，确保数据只有两列(基因ID列和功能注释列)，并删除除了product值以外的内容，比如iprs.panther.anno文件中的pantherid号，整理成pannzer2.uniprot.anno文件的形式，两列数据，第一列基因ID，第二列注释到的产物product值。</p>
</li>
<li><p>合并多个注释结果<br>cat *anno|sort -f|uniq -i|awk ‘$2 !&#x3D; “” {print $0}’ &gt;merge.anno<br>合并多个注释结果，排序sort，去重uniq，和去除空值行等处理；按照基因ID排序，生成单基因多行注释的merge.anno文件</p>
</li>
<li><p>过滤注释<br>最好的办法是在这一步做过滤filter。</p>
</li>
</ol>
<p><code>sed -i -E -e &quot;s/[,|()[]&#123;&#125;]//g&quot; -e &quot;s/^[-:/]//g&quot; -e &quot;s/[-:/]$//g&quot; -e &quot;s/[^a-zA-Z]+//g&quot; merge.anno</code></p>
<p>product值的不规范情况：</p>
<ul>
<li>包含逗号,（因为逗号是分隔符）</li>
<li>包含竖杠|</li>
<li>短横杠-开头或结尾</li>
<li>冒号:开头或结尾</li>
<li>斜杠&#x2F;开头或结尾</li>
<li>不包含字母</li>
<li>括号（包括圆括号,中括号和大括号）不完整的</li>
</ul>
<ol start="4">
<li><p>单注释转多注释<br>用<code>cat merge.anno |awk -F&quot;\t&quot; &#39;&#123;a[$1]=a[$1]$2&quot;,&quot;&#125;END&#123;for( i in a)&#123;print i&quot;\t&quot;a[i]&#125;&#125;&#39; |sed &quot;s/,$/;/g&quot;|sort -gk 1.3n|awk &#39;&#123;print $1&quot;\tproduct=&quot;$2&#125;&#39; &gt;product.list</code>命令从merge.anno生成product.list，每个基因单行的格式，还是两列，第一列基因ID，第二列若有多个注释用逗号隔开，用分号结尾。</p>
</li>
<li><p>添加product属性值<br>sample.gff与product.list合并，把结构注释和功能注释文件合并，功能注释以product属性值的形式加入gff文件的第九列内容。</p>
</li>
</ol>
<h2 id="3-2-方案二：单独提供功能注释"><a href="#3-2-方案二：单独提供功能注释" class="headerlink" title="3.2. 方案二：单独提供功能注释"></a>3.2. 方案二：单独提供功能注释</h2><h3 id="3-2-1-基因功能注释整理"><a href="#3-2-1-基因功能注释整理" class="headerlink" title="3.2.1. 基因功能注释整理"></a>3.2.1. 基因功能注释整理</h3><p>不同软件的基因功能注释结果都预先进行了整理（参考各个软件后的注释整理部分），整理成两列的格式，首行为表头，第一列为基因ID（gene），第二列为基因功能注释的描述信息（标明注释来源），每个注释软件&#x2F;数据库的描述信息单独为一列。</p>
<p>目前共有9个文件被整理出来：</p>
<ol>
<li>eggnog.anno</li>
<li>eggnog.go.anno</li>
<li>eggnog.kegg.anno</li>
<li>iprs.interpro.anno</li>
<li>iprs.panther.anno</li>
<li>iprs.pfam.anno</li>
<li>mercator.anno</li>
<li>pannzer2.go.anno</li>
<li>pannzer2.uniprot.anno</li>
</ol>
<h3 id="3-2-2-基因功能注释整合"><a href="#3-2-2-基因功能注释整合" class="headerlink" title="3.2.2. 基因功能注释整合"></a>3.2.2. 基因功能注释整合</h3><p>需要整合所有注释软件&#x2F;数据库的信息，只需根据第一列进行文件的合并。</p>
<h4 id="3-2-2-1-join命令"><a href="#3-2-2-1-join命令" class="headerlink" title="3.2.2.1. join命令"></a>3.2.2.1. join命令</h4><ol>
<li><p>实现合并a.anno和b.anno的方法：a.anno和b.anno都只有两列的情况</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">join -t $&#x27;\t&#x27; -j 1 a.anno b.anno &gt;ab.tem #合并两个注释文件都注释到基因的行</span><br><span class="line">join -t $&#x27;\t&#x27; -j 1 -v 1 a.anno b.anno |awk -F&quot;\t&quot; &#x27;&#123;print $1&quot;\t&quot;$2&quot;\t-&quot;&#125;&#x27; &gt;a.tem #提取a.anno单独注释到基因的行</span><br><span class="line">join -t $&#x27;\t&#x27; -j 1 -v 2 a.anno b.anno |awk -F&quot;\t&quot; &#x27;&#123;print $1&quot;\t-\t&quot;$2&#125;&#x27; &gt;b.tem #提取b.anno单独注释到基因的行</span><br><span class="line">cat ab.tem a.tem b.tem |sort -k 1.3n &gt;ab.anno #合并注释文件</span><br><span class="line">rm ab.tem a.tem b.tem # 删除临时文件</span><br></pre></td></tr></table></figure>
</li>
<li><p>实现合并ab.anno和c.anno的方法：ab.anno有三列，c.anno有两列的情况</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">join -t $&#x27;\t&#x27; -j 1 ab.anno c.anno &gt;abc.tem #合并两个注释文件都注释到基因的行</span><br><span class="line">join -t $&#x27;\t&#x27; -j 1 -v 1 ab.anno c.anno |awk -F&quot;\t&quot; &#x27;&#123;print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t-&quot;&#125;&#x27; &gt;ab.tem #提取ab.anno单独注释到基因的行</span><br><span class="line">join -t $&#x27;\t&#x27; -j 1 -v 2 ab.anno c.anno |awk -F&quot;\t&quot; &#x27;&#123;print $1&quot;\t-\t-\t&quot;$2&#125;&#x27; &gt;c.tem #提取c.anno单独注释到基因的行</span><br><span class="line">cat abc.tem ab.tem c.tem |sort -k 1.3n &gt;abc.anno #合并注释文件</span><br><span class="line">rm abc.tem ab.tem c.tem # 删除临时文件</span><br></pre></td></tr></table></figure></li>
</ol>
<p>如果多个文件，则两两依次合并。还没想到什么更简单的方法，就这样用join手动合并所有注释软件&#x2F;数据库的功能注释到一个文件吧。</p>
<h4 id="3-2-2-2-merge-file-key-property-title-cpp程序"><a href="#3-2-2-2-merge-file-key-property-title-cpp程序" class="headerlink" title="3.2.2.2. merge_file_key_property.title.cpp程序"></a>3.2.2.2. merge_file_key_property.title.cpp程序</h4><p>请大佬写了个合并功能注释的程序，merge_file_key_property.title.cpp，可以一次合并任意多个文件。</p>
<ol>
<li>编译<br><code>g++ -std=c++11 ./merge_file_key_property.cpp -o merge_file_key_property.o</code>生成执行文件*.o</li>
<li>执行<br><code> ./merge_file_key_property.o -h</code>查看参数</li>
</ol>
<p><code>./merge_file_key_property.o -a .anno -n 7 -p mc -q gene -o merged</code> 使用示范：-a指定合并的文件类型（比如.anno）；-n指定第一列字段长度（比如sp00001长度为7）；-p指定第一列关键词的前缀（基因名前缀，比如sp）；-q指定第一列关键词的标题（比如gene），作为表头；-o指定合并文件名的前缀。</p>
<h4 id="3-2-2-3-waiting…"><a href="#3-2-2-3-waiting…" class="headerlink" title="3.2.2.3. waiting…"></a>3.2.2.3. waiting…</h4><p>如果文件多且特征复杂，还是琢磨一下python或者R的merge功能吧。</p>
<h1 id="4-references"><a href="#4-references" class="headerlink" title="4. references"></a>4. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/67dbafa86334">https://www.jianshu.com/p/67dbafa86334</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/e646c0fa6443">https://www.jianshu.com/p/e646c0fa6443</a></li>
<li><a target="_blank" rel="noopener" href="http://xuzhougeng.top/archives/Function-anotation-with-swiss-prot-database">徐洲更文章</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/4f4819f385d2">https://www.jianshu.com/p/4f4819f385d2</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/05/17/omics_genome.functional.annotation/" data-id="cl1aiqcrw003wm4vk58k171bc" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-blog" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/04/20/blog/" class="article-date">
  <time datetime="2021-04-20T08:22:52.000Z" itemprop="datePublished">2021-04-20</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/blog/">blog</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/04/20/blog/">blog</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=20707476&auto=1&height=32"></iframe><music URL></div>

<h1 id="1-blog的categories和tags"><a href="#1-blog的categories和tags" class="headerlink" title="1. blog的categories和tags"></a>1. blog的categories和tags</h1><p>categories和tags的记录</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br></pre></td><td class="code"><pre><span class="line">---</span><br><span class="line">title: blog</span><br><span class="line">date: 2021-04-20 16:50:00</span><br><span class="line">categories: </span><br><span class="line"></span><br><span class="line">- bio</span><br><span class="line">	- concept</span><br><span class="line">	- taxon</span><br><span class="line">	- bioinfo</span><br><span class="line">	- experiment</span><br><span class="line">	- theory</span><br><span class="line">	- knowledge</span><br><span class="line"></span><br><span class="line">- biosoft</span><br><span class="line"></span><br><span class="line">- omics</span><br><span class="line">	- genome</span><br><span class="line">	- transcriptome</span><br><span class="line">	- plastome</span><br><span class="line">	- mitochondrion</span><br><span class="line"></span><br><span class="line">- plot</span><br><span class="line">	- R</span><br><span class="line"></span><br><span class="line">- computer</span><br><span class="line">	- system</span><br><span class="line">		- windows</span><br><span class="line">		- linux</span><br><span class="line">	- program language</span><br><span class="line">		- python</span><br><span class="line">		- R</span><br><span class="line">		- perl</span><br><span class="line">		- java</span><br><span class="line">		- C</span><br><span class="line">	- IDE</span><br><span class="line">		- vim</span><br><span class="line">		- VScode</span><br><span class="line">		- git</span><br><span class="line">	- script</span><br><span class="line">	- web</span><br><span class="line"></span><br><span class="line">- linux</span><br><span class="line">	- basics</span><br><span class="line">	- shell</span><br><span class="line">	- text processing</span><br><span class="line">	- operation and maintenance</span><br><span class="line"></span><br><span class="line">- blog</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">tags: </span><br><span class="line">- genome assemble</span><br><span class="line">- genome annotation</span><br><span class="line">- phylogeny</span><br><span class="line">- divergence time</span><br><span class="line">- WGD</span><br><span class="line">- HGT</span><br><span class="line">- molecular experiment</span><br><span class="line">- homolog</span><br><span class="line">- ortholog</span><br><span class="line">- paralog</span><br><span class="line">- xenolog</span><br><span class="line">- analog</span><br><span class="line">- orthology</span><br><span class="line">- orthogroup</span><br><span class="line">- gene family</span><br><span class="line">---  </span><br></pre></td></tr></table></figure>

<h1 id="2-日常blog撰写和备份操作"><a href="#2-日常blog撰写和备份操作" class="headerlink" title="2. 日常blog撰写和备份操作"></a>2. 日常blog撰写和备份操作</h1><p>在做好blog搭建后，blog撰写和日常管理可参考这部分内容。</p>
<h2 id="2-1-blog同步"><a href="#2-1-blog同步" class="headerlink" title="2.1. blog同步"></a>2.1. blog同步</h2><p>养成习惯，每次开始撰写blog前都通过git bash进入工作区，进行<code>git pull</code>命令把github端的hexo分支的更新（更新可能是其他终端上提交的）同步到本地，实现多终端的内容完全同步。<br>但如果本地有未提交的更新，则千万不要用<code>git pull</code>，否则会覆盖本地更新；直接进入下一步；直到使用<code>git add .</code>，<code>git commit -m &quot;submit&quot;</code>，<code>git push origin hexo</code>提交备份本地更新到github端的hexo分支后才可以使用<code>git pull</code>(一般是在其他终端，把github的hexo分支更新拉到其他终端设备使用)。</p>
<h2 id="2-2-blog撰写"><a href="#2-2-blog撰写" class="headerlink" title="2.2. blog撰写"></a>2.2. blog撰写</h2><p>在本地source&#x2F;_posts下添加和修改md文档实现blog的日常撰写和修改。</p>
<p>使用命令<code>hexo new &quot;newpostname&quot;</code>可以在hexo&#x2F;source&#x2F;_posts下新建一个newpostname.md的文件，这个文件以scaffolds&#x2F;post.md为模板，修改scaffolds&#x2F;post.md文件可以修改hexo new命令生成的新blog文件样式。</p>
<h2 id="2-3-blog备份"><a href="#2-3-blog备份" class="headerlink" title="2.3. blog备份"></a>2.3. blog备份</h2><p>只要blog有更改或者新增，或者配置文件有修改，即工作区（即本地的hexo目录或github.io目录）有文件修改，则建议对文件进行备份到GitHub端的hexo分支。<br>用三条命令<code>git add .</code>，<code>git commit -m &quot;submit&quot;</code>，<code>git push origin hexo</code>备份工作区，包括md博客源文件和hexo部署到github端的hexo分支。三条命令执行前建议通过<code>hexo clean</code>清除缓存和public目录，以免备份不需要的文件。</p>
<h2 id="2-4-blog发布"><a href="#2-4-blog发布" class="headerlink" title="2.4. blog发布"></a>2.4. blog发布</h2><p>可根据自身需求决定是否发布blog到github.io网站，一般写的blog完整程度比较高时可以发布。使用<code>hexo clean &amp; hexo g -d</code>命令，根据source&#x2F;_posts下的博客源文件生成public目录（网站html并同步到github端的master分支，即发布blog到github.io网站。</p>
<p>总结一下，在配置好写作环境后的任意一台终端的日常工作流应该是：</p>
<ol>
<li><code>git pull</code>同步远程github库的hexo更新到本地</li>
<li><code>hexo new &quot;newblog&quot;</code>在source&#x2F;_posts&#x2F;下添加md格式的blog，或者修改已有的blog</li>
<li><code>git add .</code>,<code>git commit -m &quot;commit notes&quot;</code>,<code>git push</code>把修改备份到github端</li>
<li>下次写作重复以上三个步骤</li>
<li>直至blog完善成熟后，用命令<code>hexo clean &amp; hexo g -d</code>生成网站并部署到github.io</li>
</ol>
<h1 id="3-攥写blog"><a href="#3-攥写blog" class="headerlink" title="3. 攥写blog"></a>3. 攥写blog</h1><h2 id="3-1-插入图片"><a href="#3-1-插入图片" class="headerlink" title="3.1. 插入图片"></a>3.1. 插入图片</h2><p>若是本地图片，就放在特定位置，然后在blog中引用；若是网络图片，就直接复制图片的url地址引用；</p>
<h3 id="3-1-1-推荐用法"><a href="#3-1-1-推荐用法" class="headerlink" title="3.1.1. 推荐用法"></a>3.1.1. 推荐用法</h3><p>若是自己的图片，可以统一保存在sources&#x2F;images&#x2F;目录下，在github上同步后，复制github上图片地址，就可以像引用网络图片一样引用图片url地址。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">&lt;img src=&quot;https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/R_plot_venn_2.png?raw=true&quot; width=50% title=&quot;venn_2&quot; align=&quot;center&quot; /&gt;</span><br><span class="line"></span><br><span class="line">**&lt;p align=&quot;center&quot;&gt;Figure 1. Venn diagram with 2 groups**</span><br><span class="line">from [github: yanzhongsino](https://github.com/yanzhongsino/yanzhongsino.github.io)&lt;/p&gt;</span><br></pre></td></tr></table></figure>

<p><code>&lt;img /&gt;</code>用于图片显示，src填写图片网址，width指图片占屏幕的比例（同时还有类似的height参数，但建议不设置height以保持图片长宽比例），title是图片名称(如果网址失效会显示title值的文字)，align设置对齐方式。<br><code>&lt;p align=&quot;center&quot;&gt;Figure 1. text&lt;/p&gt;</code>用于图的标题居中显示，from指示图源(引用其他来源的图时)，**用于加粗显示。</p>
<p>效果如下：<br><img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/R_plot_venn_2.png?raw=true" width=50% title="venn_2" align="center" /></p>
<p><strong><p align="center">Figure 1. Venn diagram with 2 groups</strong><br>from <a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/yanzhongsino.github.io">github: yanzhongsino</a></p></p>
<h3 id="3-1-2-markdown语法"><a href="#3-1-2-markdown语法" class="headerlink" title="3.1.2. markdown语法"></a>3.1.2. markdown语法</h3><ol>
<li>统一放在images目录下<br>当Hexo项目中只用到少量图片时，可以将图片统一放在source&#x2F;images文件夹中，在blog中通过markdown语法访问它们，即<code>![图片注释](/images/image.jpg &quot;图片标题&quot;)</code>。</li>
</ol>
<p>图片既可以在首页内容中访问到，也可以在文章正文中访问到。</p>
<ol start="2">
<li>放在文章各自目录下<br>图片除了可以放在统一的images文件夹中，还可以放在文章自己的目录中。文章的目录可以通过配置_config.yml来生成。</li>
</ol>
<ul>
<li>将_config.yml文件中的配置项post_asset_folder设为true<code>post_asset_folder: true</code>;</li>
<li>执行命令<code>hexo new postname</code>，在source&#x2F;_posts中会生成文章postname.md和同名文件夹postname;</li>
<li>将图片资源放在post_name中，文章就可以使用相对路径引用图片资源了。</li>
<li>在文章中添加markdown语法代码<code>![图片注释](image.jpg &quot;图片标题&quot;)</code>即可。</li>
</ul>
<p>图片只能在文章中显示，但无法在首页中正常显示。</p>
<h3 id="3-1-3-HTML语法【推荐】"><a href="#3-1-3-HTML语法【推荐】" class="headerlink" title="3.1.3. HTML语法【推荐】"></a>3.1.3. HTML语法【推荐】</h3><p>用markdown语法无法指定图片的尺寸和对齐方式，建议用HTML语法插入图片，以实现更好的控制。<br><code>&lt;img src=&quot;/images/image.png&quot; width=80% height=80% title=&quot;picture&quot; alt=&quot;picture&quot; align=center/&gt;</code></p>
<p><code>&lt;img src=&quot;url&quot; title=&quot;title&quot; width=&quot;80%&quot; height=&quot;80%&quot; /&gt;</code></p>
<h3 id="3-1-4-标签插件语法引用"><a href="#3-1-4-标签插件语法引用" class="headerlink" title="3.1.4. 标签插件语法引用"></a>3.1.4. 标签插件语法引用</h3><p>如果希望图片在文章和首页中同时显示，可以使用标签插件语法，本地和网络图片都适用。</p>
<ul>
<li>本地图片资源，不限制图片尺寸，使用 <code>&#123;% asset_img image.jpg This is an image %&#125;</code>；</li>
<li>网络图片资源，限制图片显示尺寸，<code>&#123;% img http://www.something.gif 200 400 vi-vim-cheat-sheet %&#125;</code></li>
</ul>
<h3 id="3-1-5-CDN引用"><a href="#3-1-5-CDN引用" class="headerlink" title="3.1.5. CDN引用"></a>3.1.5. CDN引用</h3><ol>
<li>除了在本地存储图片，还可以将图片上传到一些免费的CDN服务中。比如Cloudinary提供的图片CDN服务，在Cloudinary中上传图片后，会生成对应的url地址，将地址直接拿来引用即可。</li>
<li>【实测这种方法不生效】把代码<code>&lt;div align=&quot;middle&quot; style=&quot;width:200px; margin:auto&quot;&gt;这里粘贴生成的url地址&lt;/div&gt;</code>粘贴到文章中即可；align为了美观设置成居中。</li>
</ol>
<h3 id="3-1-6-fancybox"><a href="#3-1-6-fancybox" class="headerlink" title="3.1.6. fancybox"></a>3.1.6. fancybox</h3><ol>
<li>启用fancybox<br>启用fancybox：点击查看图片大图。</li>
</ol>
<p>Hexo的NexT主题中提供了fancybox的方便接口。</p>
<p>Usage：<a target="_blank" rel="noopener" href="https://github.com/theme-next/theme-next-fancybox3">https://github.com/theme-next/theme-next-fancybox3</a><br>markdown用法：<code>&#123;% img http://www.viemu.com/vi-vim-cheat-sheet.gif 600 600 "点击查看大图:vi/vim-cheat-sheet" %&#125;</code></p>
<ol start="2">
<li>Hexo部分图片禁用fancybox</li>
</ol>
<p>hexo在使用fancybox插件时，图片的效果还是很可观的，但是我们往往是不需要所有的图片都用fancybox；<br>例如：hexo next主题下，添加某些图片的时候，有些事不需要可点击的<br>修改theme\next\source\js\src\utils.js 红色字体部分；</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">diff --git a/source/js/src/utils.js b/source/js/src/utils.js</span><br><span class="line">index 0f3704e..8516665 100644</span><br><span class="line">--- a/source/js/src/utils.js</span><br><span class="line">+++ b/source/js/src/utils.js</span><br><span class="line">@@ -11,6 +11,7 @@ NexT.utils = NexT.$u = &#123;</span><br><span class="line">       .not(&#x27;.group-picture img, .post-gallery img&#x27;)</span><br><span class="line">       .each(function() &#123;</span><br><span class="line">         var $image = $(this);</span><br><span class="line">+        if ($(this).hasClass(&#x27;nofancybox&#x27;)) return;</span><br><span class="line">         var imageTitle = $image.attr(&#x27;title&#x27;);</span><br><span class="line">         var $imageWrapLink = $image.parent(&#x27;a&#x27;);</span><br></pre></td></tr></table></figure>

<p>在img标签使用的时候加上class&#x3D;”nofancybox”即可。<code>&lt;img src=&quot;http://www.viemu.com/vi-vim-cheat-sheet.gif&quot; class=&quot;nofancybox&quot; /&gt;</code></p>
<h2 id="3-2-插入背景音乐"><a href="#3-2-插入背景音乐" class="headerlink" title="3.2. 插入背景音乐"></a>3.2. 插入背景音乐</h2><ol>
<li>打开网易云网页版，找到想听的歌曲，然后点击<strong>生成外链播放器</strong>，然后复制网易云音乐的插件页面的HTML代码；</li>
<li>把代码<code>&lt;div align=&quot;middle&quot;&gt;这里粘贴刚刚复制的代码&lt;/div&gt;</code>粘贴到文章中即可；align为了美观设置成居中；</li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/04/20/blog/" data-id="cl1aiqcrp0033m4vkfo2na0rn" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-linux_PATH" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/04/20/linux_PATH/" class="article-date">
  <time datetime="2021-04-20T08:20:15.000Z" itemprop="datePublished">2021-04-20</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/linux/">linux</a>►<a class="article-category-link" href="/categories/linux/basics/">basics</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/04/20/linux_PATH/">在Linux系统中设置环境变量PATH</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=5264843&auto=1&height=32"></iframe><music URL></div>



<h1 id="1-环境变量PATH设置"><a href="#1-环境变量PATH设置" class="headerlink" title="1. 环境变量PATH设置"></a>1. 环境变量PATH设置</h1><h2 id="1-1-PATH的解释"><a href="#1-1-PATH的解释" class="headerlink" title="1.1. PATH的解释"></a>1.1. PATH的解释</h2><p>PATH是环境变量，一个存放有（可执行）命令和程序的目录集合，目录间以冒号:分隔。</p>
<p>操作系统接到用户输入命令时，会<strong>从前到后依次</strong>查看PATH存储的目录，查找与输入命令同名的可执行文件，<strong>一旦查找到就停止查找</strong>，全部目录都查不到就报错。</p>
<p><code>echo $PATH</code>可以查看当前存放了哪些目录。</p>
<h2 id="1-2-设置环境变量"><a href="#1-2-设置环境变量" class="headerlink" title="1.2. 设置环境变量"></a>1.2. 设置环境变量</h2><h3 id="1-2-1-临时设置，当前终端有效"><a href="#1-2-1-临时设置，当前终端有效" class="headerlink" title="1.2.1. 临时设置，当前终端有效"></a>1.2.1. 临时设置，当前终端有效</h3><ol>
<li>添加目录到环境变量<br><code>PATH=$PATH:/home/usrname/opt/software/</code>命令把目录<code>/home/software/</code>添加到PATH后，并用冒号:分隔，即重新给PATH赋值。</li>
</ol>
<p><code>PATH=$PATH:/home/software/</code>和<code>PATH=/home/software/:$PATH</code>区别在于把目录<code>/home/software/</code>置于PATH的最前面还是最后面，最前面代表最优先使用这个目录下的命令。</p>
<ol start="2">
<li>声明环境变量<br><code>export PATH</code>命令声明变量，使其对系统shell可识别</li>
</ol>
<p>以上两条命令可以合并成<code>export PATH=$PATH:/home/software/</code>或者<code>export PATH=/home/software/:$PATH</code>，此时目录<code>/home/software/</code>下的命令可以被当前shell直接调用。</p>
<p>但这种设置方式只在当前shell有效，关掉shell就失效。</p>
<h3 id="1-2-2-当前用户永久设置"><a href="#1-2-2-当前用户永久设置" class="headerlink" title="1.2.2. 当前用户永久设置"></a>1.2.2. 当前用户永久设置</h3><p>登录服务器时，会自动运行当前用户下<code>~/.bashrc</code>和<code>~/.bash_profile</code>文件内的命令。</p>
<p>所以把<code>export PATH=$PATH:/home/software/</code>写入<code>~/.bashrc</code>或者<code>~/.bash_profile</code>文件就可以使目录<code>/home/software/</code>对于当前用户永久生效。</p>
<p>第一次写入后需要用命令<code>source ~/.bashrc</code>运行~&#x2F;.bashrc文件来刷新环境变量，之后就不再需要。</p>
<h3 id="1-2-3-所有用户永久设置"><a href="#1-2-3-所有用户永久设置" class="headerlink" title="1.2.3. 所有用户永久设置"></a>1.2.3. 所有用户永久设置</h3><p>与前一步同理，不过写入的文件是&#x2F;etc&#x2F;profile或者&#x2F;etc&#x2F;bashrc，需要root用户修改。</p>
<p>第一次修改需要<code>source /etc/profile</code>刷新，之后不再需要。</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/04/20/linux_PATH/" data-id="cl1aiqcrt003lm4vkazb26dgm" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-scripts" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/04/01/scripts/" class="article-date">
  <time datetime="2021-04-01T02:56:00.000Z" itemprop="datePublished">2021-04-01</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer/">computer</a>►<a class="article-category-link" href="/categories/computer/scripts/">scripts</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/04/01/scripts/">scripts</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=476751845&auto=1&height=32"></iframe></div>

<h1 id="1-我写的脚本"><a href="#1-我写的脚本" class="headerlink" title="1. 我写的脚本"></a>1. 我写的脚本</h1><h2 id="1-1-多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测"><a href="#1-1-多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测" class="headerlink" title="1.1. 多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测"></a>1.1. 多个RNA-seq数据，基于参考基因组的转录本组装和ORF预测</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/myscripts/multi-transcriptome_assembly_with_ref.sh">multi-transcriptome_assembly_with_ref.sh</a></p>
<h2 id="1-2-提取，合并，和比对单拷贝基因和蛋白"><a href="#1-2-提取，合并，和比对单拷贝基因和蛋白" class="headerlink" title="1.2. 提取，合并，和比对单拷贝基因和蛋白"></a>1.2. 提取，合并，和比对单拷贝基因和蛋白</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/my_scripts/extract_single_copy_genes.sh">extract_single_copy_genes.sh</a></p>
<h1 id="2-我修改的脚本（其他人写的）"><a href="#2-我修改的脚本（其他人写的）" class="headerlink" title="2. 我修改的脚本（其他人写的）"></a>2. 我修改的脚本（其他人写的）</h1><h2 id="2-1-从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息"><a href="#2-1-从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息" class="headerlink" title="2.1. 从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息"></a>2.1. 从包含mRNA和CDS的gff注释文件中提取intron的位置和长度信息</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/modifiedscripts/extract_intron_info.pl">extract_intron_info.pl</a></p>
<h1 id="3-保存的脚本"><a href="#3-保存的脚本" class="headerlink" title="3. 保存的脚本"></a>3. 保存的脚本</h1><h2 id="3-1-batch-run-sh-——-多线程并行运行批量化命令"><a href="#3-1-batch-run-sh-——-多线程并行运行批量化命令" class="headerlink" title="3.1. batch_run.sh —— 多线程并行运行批量化命令"></a>3.1. batch_run.sh —— 多线程并行运行批量化命令</h2><p><a target="_blank" rel="noopener" href="https://github.com/yanzhongsino/bioscripts/blob/main/saved_scripts/batch_run.sh">batch_run.sh</a></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">!/bin/bash</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">batch_run.sh脚本用于多线程并行批量化循环命令，并控制并行数量始终为<span class="variable">$&#123;thread&#125;</span>。</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">usage: 修改thread=后的值为想要设定的并行线程数，把需要并行的命令替换23行的run something。运行脚本batch_run.sh，实现并行运行循环任务。</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">ref：网上找的脚本，来源找不到了，侵删。</span></span><br><span class="line"></span><br><span class="line">thread=16</span><br><span class="line"></span><br><span class="line">start_time=`date +%s`              # 定义脚本运行的开始时间</span><br><span class="line">echo $&#123;start_time&#125;</span><br><span class="line">tmpFifo=/tmp/$$.fifo # 声明管道名称，$$表示脚本当前运行的进程PID</span><br><span class="line">mkfifo $&#123;tmpFifo&#125; # 创建有名管道</span><br><span class="line">exec 3&lt;&gt;$&#123;tmpFifo&#125;                   #创建文件描述符，以可读（&lt;）可写（&gt;）的方式关联管道文件，这时候文件描述符3就有了有名管道文件的所有特性</span><br><span class="line">rm -rf $&#123;tmpFifo&#125;                    #关联后的文件描述符拥有管道文件的所有特性,所以这时候管道文件可以删除，我们留下文件描述符来用就可以了</span><br><span class="line">for ((i=1;i&lt;=$&#123;thread&#125;;i++))</span><br><span class="line">do</span><br><span class="line">        echo &quot;&quot; &gt;&amp;3                   #&amp;3代表引用文件描述符3，这条命令代表往管道里面放入了一个&quot;令牌&quot;</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">for sample in $(cat sample_list.txt) #并行运行的循环</span><br><span class="line">do</span><br><span class="line">        read -u3                           #代表从管道中读取一个令牌</span><br><span class="line">        &#123;</span><br><span class="line">                echo $&#123;sample&#125;</span><br><span class="line">                run something # 需要并行运行的真实命令</span><br><span class="line">                echo &#x27;success&#x27; $&#123;sample&#125;       </span><br><span class="line">                echo &quot;&quot; &gt;&amp;3                   #代表我这一次命令执行到最后，把令牌放回管道</span><br><span class="line">        &#125; &amp;</span><br><span class="line">done</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash"><span class="comment"># ps:可以使用while read line; do echo $line; done &lt; file.txt代替for i in $(cat file.txt);do echo $i; done循环。但是不能用cat file.txt | while read line; do echo $line; done代替。</span></span></span><br><span class="line"></span><br><span class="line">wait</span><br><span class="line"> </span><br><span class="line">stop_time=`date +%s`  #定义脚本运行的结束时间</span><br><span class="line"> </span><br><span class="line">echo &quot;TIME:`expr $stop_time - $start_time`&quot;</span><br><span class="line">exec 3&lt;&amp;-                       #关闭文件描述符的读</span><br><span class="line">exec 3&gt;&amp;-                       #关闭文件描述符的写</span><br></pre></td></tr></table></figure>

<h2 id="3-2-ROUSFinder2-0-py-——-植物线粒体基因组重复序列注释脚本"><a href="#3-2-ROUSFinder2-0-py-——-植物线粒体基因组重复序列注释脚本" class="headerlink" title="3.2. ROUSFinder2.0.py —— 植物线粒体基因组重复序列注释脚本"></a>3.2. ROUSFinder2.0.py —— 植物线粒体基因组重复序列注释脚本</h2><ol>
<li>背景<br>文章<a target="_blank" rel="noopener" href="https://academic.oup.com/g3journal/article/9/2/549/6026745">Repeats of Unusual Size in plant mitochondrial genomes: identification, incidence and evolution</a>总结了植物线粒体基因组的重复序列的特征。研究表明，植物线粒体基因组比动物的要大，包含大量的非编码DNA，突变率低，重排率高。</li>
</ol>
<p>文章里提供了一个python脚本<a target="_blank" rel="noopener" href="https://gsajournals.figshare.com/articles/figure/Supplemental_Material_for_Wynn_and_Christensen_2018/7425680?file=13750433">ROUSFinder2.0.py</a>用于注释植物线粒体基因组中的非串联重复序列(non-tandem repeat)。</p>
<ol start="2">
<li>脚本<br>脚本是python2写的，依赖主要是blastn。</li>
</ol>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ROUSFinder2.0.py用于植物线粒体基因组重复序列的注释</span></span><br><span class="line"><span class="comment">#! /usr/bin/env python  </span></span><br><span class="line"><span class="keyword">import</span> sys, math, os, argparse, csv  </span><br><span class="line">csv.field_size_limit(sys.maxsize)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Version 2.0, November 21, 2018  </span></span><br><span class="line"><span class="comment"># Changes: uses variable parameters  </span></span><br><span class="line"><span class="comment"># Find dispersed repeated sequences in genomes.   </span></span><br><span class="line"><span class="comment"># Designed for plant mitochondrial genomes of up to a few Mbp.  </span></span><br><span class="line"><span class="comment"># May be very slow with larger genomes.   </span></span><br><span class="line"><span class="comment"># Blast can also sometimes give odd results with large or highly repetetive genomes.  </span></span><br><span class="line"><span class="comment"># Gaps, or runs of &#x27;N&#x27;s in the sequence will definitely give weird results.   </span></span><br><span class="line"><span class="comment"># The program assumes there aren&#x27;t any, and that the longest repeat will be the full sequence to itself.  </span></span><br><span class="line"><span class="comment"># If there are long repeats in the output that are listed as being only at one location, this is probably what happened.  </span></span><br><span class="line"><span class="comment"># If there are a lot of repeats within repeats the results can also be odd.  </span></span><br><span class="line"><span class="comment"># Copyright Alan C. Christensen, University of Nebraska, 2018  </span></span><br><span class="line"><span class="comment"># No guarantees, warranties, support, or anything else is implicit or explicit.  </span></span><br><span class="line"><span class="comment"># Input is a fasta format file of a sequence. Genbank format works but generates lots of error messages in stdout.  </span></span><br><span class="line"><span class="comment"># Output is a list of unique, ungapped repeated sequences, fasta formatted.  </span></span><br><span class="line"><span class="comment"># The names are in the format &#x27;&gt;Repeat/ROUS_name_start_end_length&#x27;.  </span></span><br><span class="line"><span class="comment"># A table of repeats with the coordinates of each one is generated.  </span></span><br><span class="line"><span class="comment"># A list of repeat name, length and copy number is generated.  </span></span><br><span class="line"><span class="comment"># A binned table of the total number of repeats in size ranges is generated.  </span></span><br><span class="line"><span class="comment">#  </span></span><br><span class="line"><span class="comment"># PARAMETERS  </span></span><br><span class="line"><span class="comment">#   REQUIRED:  </span></span><br><span class="line"><span class="comment">#      input file in fasta format  </span></span><br><span class="line"><span class="comment">#   Optional  </span></span><br><span class="line"><span class="comment">#      -o output file name  </span></span><br><span class="line"><span class="comment">#      -m minimum length of exact matches to keep  </span></span><br><span class="line"><span class="comment">#      -b path to blastn (default is /usr/bin/)  </span></span><br><span class="line"><span class="comment">#      -k keep temp files  </span></span><br><span class="line"><span class="comment">#      -gb to write the repeats to a genbank format file  </span></span><br><span class="line"><span class="comment">#      -rew reward for match (default is 1)  </span></span><br><span class="line"><span class="comment">#      -pen penalty for mismatch (default is 20)  </span></span><br><span class="line">  </span><br><span class="line">parser = argparse.ArgumentParser(description=<span class="string">&#x27;Find repeats in a fasta sequence file&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;infile&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Input .fasta file&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-o&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;outfile&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Output file name seed, default is input_repeats&#x27;</span>, default=<span class="string">&#x27;default&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-m&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;minlen&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Minimum length of matches to keep, default=50&#x27;</span>, default=<span class="string">&#x27;50&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-b&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;blast_path&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Path to blastn program, default is /usr/bin/&#x27;</span>, default=<span class="string">&#x27;/usr/bin/&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-k&#x27;</span>, action=<span class="string">&#x27;store_true&#x27;</span>, dest=<span class="string">&#x27;keep&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;True to keep temp files&#x27;</span>, default=<span class="literal">False</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-gb&#x27;</span>, action=<span class="string">&#x27;store_true&#x27;</span>, dest=<span class="string">&#x27;genbank&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;True to write GenBank format file&#x27;</span>, default=<span class="literal">False</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-rew&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;reward&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Reward for match&#x27;</span>, default=<span class="string">&#x27;1&#x27;</span>)  </span><br><span class="line">parser.add_argument(<span class="string">&#x27;-pen&#x27;</span>, action=<span class="string">&#x27;store&#x27;</span>, dest=<span class="string">&#x27;penalty&#x27;</span>, <span class="built_in">help</span>=<span class="string">&#x27;Penalty for mismatch&#x27;</span>, default=<span class="string">&#x27;20&#x27;</span>)  </span><br><span class="line">results = parser.parse_args()  </span><br><span class="line">infile = results.infile  </span><br><span class="line">outfile = results.outfile  </span><br><span class="line">minlen = <span class="built_in">int</span>(results.minlen)  </span><br><span class="line">blast_path = results.blast_path  </span><br><span class="line">keep = results.keep  </span><br><span class="line">genbank = results.genbank  </span><br><span class="line">reward = results.reward  </span><br><span class="line">penalty = results.penalty  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># It might be useful to define the wordsize as something less than minlen, so both variables are used.  </span></span><br><span class="line"><span class="comment"># Wordsize smaller than minlen would give smaller core identical sequences in the middle of repeats.  </span></span><br><span class="line"><span class="comment"># An example might be to change this to wordsize = str(int(minlen/2)).  </span></span><br><span class="line">wordsize = <span class="built_in">str</span>(minlen)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># If no output file seed is specified, make one by stripping leading directory information  </span></span><br><span class="line"><span class="comment"># and stripping trailing .fa or .fasta from the input file name and using that.  </span></span><br><span class="line"><span class="keyword">if</span> outfile == <span class="string">&#x27;default&#x27;</span>:  </span><br><span class="line">    outfile = infile  </span><br><span class="line">    <span class="keyword">if</span> outfile.count(<span class="string">&#x27;/&#x27;</span>) &gt; <span class="number">0</span>:  </span><br><span class="line">        <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(outfile.count(<span class="string">&#x27;/&#x27;</span>)):  </span><br><span class="line">            index = outfile.index(<span class="string">&#x27;/&#x27;</span>)  </span><br><span class="line">            outfile = outfile[index+<span class="number">1</span>:]  </span><br><span class="line">    <span class="keyword">if</span> outfile.endswith(<span class="string">&#x27;.fa&#x27;</span>) <span class="keyword">or</span> outfile.endswith(<span class="string">&#x27;.fasta&#x27;</span>):  </span><br><span class="line">        outfile = outfile.rstrip(<span class="string">&#x27;fasta&#x27;</span>)  </span><br><span class="line">    outfile = outfile.rstrip(<span class="string">&#x27;.&#x27;</span>)  </span><br><span class="line">outfa = outfile+<span class="string">&#x27;_rep.fasta&#x27;</span>  </span><br><span class="line">outtab = outfile+<span class="string">&#x27;_rep_table.txt&#x27;</span>  </span><br><span class="line">outbin = outfile+<span class="string">&#x27;_binned.txt&#x27;</span>  </span><br><span class="line">outcount = outfile+<span class="string">&#x27;_rep_counts.txt&#x27;</span>  </span><br><span class="line">outgb = outfile+<span class="string">&#x27;_repeats.gb.txt&#x27;</span>  </span><br><span class="line">tempblast = outfile+<span class="string">&#x27;_tempblast.txt&#x27;</span>  </span><br><span class="line">temprepeats = outfile+<span class="string">&#x27;_temprepeats.txt&#x27;</span>  </span><br><span class="line">tempparse = outfile+<span class="string">&#x27;_sequence_parsing.txt&#x27;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Get sequence name and length from fasta file.  </span></span><br><span class="line">seq = <span class="built_in">open</span>(infile, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">seqname = seq.readline()  </span><br><span class="line">seqname = seqname.lstrip(<span class="string">&#x27;&gt; &#x27;</span>)  </span><br><span class="line">seqname = seqname.rstrip()  </span><br><span class="line">seqlen = <span class="number">0</span>   </span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> seq:  </span><br><span class="line">    <span class="keyword">if</span>(line[<span class="number">0</span>] == <span class="string">&quot;&gt;&quot;</span>):  </span><br><span class="line">        <span class="keyword">continue</span>  </span><br><span class="line">    seqlen += <span class="built_in">len</span>(line.strip())  </span><br><span class="line">seq.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># run blastn with query file plus strand (removing first line which is full length sequence), minus strand, and concatenate  </span></span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Performing self-blastn comparison with &#x27;</span>+seqname      </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+infile+<span class="string">&#x27; -strand plus -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward &#x27;</span>+reward+<span class="string">&#x27; -penalty -&#x27;</span>+penalty+<span class="string">&#x27; -ungapped -dust no -soft_masking false -evalue 10  -outfmt &quot;10 qstart qend length sstart send mismatch sstrand qseq&quot; | tail -n+2 &gt; tempblast1.txt&#x27;</span>)  </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+infile+<span class="string">&#x27; -strand minus -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward &#x27;</span>+reward+<span class="string">&#x27; -penalty -&#x27;</span>+penalty+<span class="string">&#x27; -ungapped -dust no -soft_masking false -evalue 10 -outfmt &quot;10 qstart qend length sstart send mismatch sstrand qseq&quot; &gt; tempblast2.txt&#x27;</span>)  </span><br><span class="line">os.system(<span class="string">&#x27;cat tempblast1.txt tempblast2.txt &gt; &#x27;</span>+tempblast)  </span><br><span class="line">os.system(<span class="string">&#x27;rm tempblast1.txt tempblast2.txt&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># open tempblast.txt, convert to list of lists, and sort by length and position descending  </span></span><br><span class="line"><span class="comment"># This is necessary because blastn does not output every possible pair of hits when there are more than 2 copies of a repeat  </span></span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Sorting alignments...&#x27;</span>  </span><br><span class="line">f = <span class="built_in">open</span>(tempblast, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">reader = csv.reader(f)  </span><br><span class="line">alignments = <span class="built_in">list</span>(reader)  </span><br><span class="line">f.close()  </span><br><span class="line">alignments = <span class="built_in">sorted</span>(alignments, key=<span class="keyword">lambda</span> x: (-<span class="number">1</span>*<span class="built_in">int</span>(x[<span class="number">2</span>]), -<span class="number">1</span>*<span class="built_in">int</span>(x[<span class="number">0</span>])))  </span><br><span class="line">alignments.append([<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;1&#x27;</span>,<span class="string">&#x27;0&#x27;</span>,<span class="string">&#x27;A&#x27;</span>,<span class="string">&#x27;X&#x27;</span>])  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># New list of uniques  </span></span><br><span class="line"><span class="comment"># Text file &#x27;_sequence_parsing.txt&#x27; includes the information on how duplicates were found.  </span></span><br><span class="line"><span class="comment"># Start at row 0. Compare to subsequent rows.   </span></span><br><span class="line"><span class="comment"># If repeat length is different from the next row, it has passed all the tests, write it to the file.  </span></span><br><span class="line"><span class="comment"># If query or subject coordinates are the same as the query or subject or reversed coordinates  </span></span><br><span class="line"><span class="comment"># of a subsequent row, it is not unique, so go to the next row and do the comparisons again.  </span></span><br><span class="line"><span class="comment"># Thanks to Alex Kozik for repeatedly testing and finding bugs in the algorithm.  </span></span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Finding unique repeats...&#x27;</span>  </span><br><span class="line">uniques = []  </span><br><span class="line">sp = <span class="built_in">open</span>(tempparse, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> row <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(alignments)):  </span><br><span class="line">    sp.write(<span class="string">&#x27;row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">      </span><br><span class="line">    <span class="keyword">if</span> <span class="built_in">int</span>(alignments[row][<span class="number">2</span>]) &lt; minlen:  </span><br><span class="line">        <span class="comment"># This won&#x27;t happen unless the word_size is defined as something other than minlen.  </span></span><br><span class="line">        <span class="comment"># That could be useful under some circumstances.  </span></span><br><span class="line">        sp.write(<span class="string">&#x27;row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is less than minlength&#x27;</span>)  </span><br><span class="line">        <span class="keyword">break</span>  </span><br><span class="line">    <span class="keyword">else</span>:  </span><br><span class="line">      </span><br><span class="line">        <span class="keyword">for</span> compare <span class="keyword">in</span> <span class="built_in">range</span>(row+<span class="number">1</span>,<span class="built_in">len</span>(alignments)):  </span><br><span class="line">            <span class="keyword">if</span> alignments[row][<span class="number">2</span>] != alignments[compare][<span class="number">2</span>]:   </span><br><span class="line">                uniques.append(alignments[row])  </span><br><span class="line">                sp.write(<span class="string">&#x27;\tadding row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; to unique list\n&#x27;</span>)  </span><br><span class="line">                <span class="keyword">break</span>  </span><br><span class="line">            <span class="keyword">else</span>:  </span><br><span class="line">                sp.write(<span class="string">&#x27;\tcomparing to &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">      </span><br><span class="line">                <span class="keyword">if</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">0</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">1</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; and &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27; are the same\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">1</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">0</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qend and qstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">3</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">4</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as sstart and send of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">0</span>] == alignments[compare][<span class="number">4</span>] <span class="keyword">and</span> alignments[row][<span class="number">1</span>] == alignments[compare][<span class="number">3</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tqstart and qend of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as send and sstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">0</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">1</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qstart and qend of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">1</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">0</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as qend and qstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">3</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">4</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as sstart and send of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">elif</span> alignments[row][<span class="number">3</span>] == alignments[compare][<span class="number">4</span>] <span class="keyword">and</span> alignments[row][<span class="number">4</span>] == alignments[compare][<span class="number">3</span>]:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\tsstart and send of row &#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is the same as send and sstart of &#x27;</span>+<span class="built_in">str</span>(compare)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">                    <span class="keyword">break</span>  </span><br><span class="line">                <span class="keyword">else</span>:  </span><br><span class="line">                    sp.write(<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(row)+<span class="string">&#x27; is different\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">sp.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Write uniques into output file  </span></span><br><span class="line"><span class="comment"># Start list for copy number table  </span></span><br><span class="line">rous_count = <span class="number">0</span>  </span><br><span class="line">g = <span class="built_in">open</span>(outfa, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">repcopies = []  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(uniques)):  </span><br><span class="line">    qstart = uniques[i][<span class="number">0</span>]  </span><br><span class="line">    qend = uniques[i][<span class="number">1</span>]  </span><br><span class="line">    length = uniques[i][<span class="number">2</span>]  </span><br><span class="line">    seq = uniques[i][<span class="number">7</span>]  </span><br><span class="line">      </span><br><span class="line">    rous_count += <span class="number">1</span>  </span><br><span class="line">    g.write(<span class="string">&#x27;&gt;Repeat_&#x27;</span>+<span class="built_in">str</span>(rous_count)+<span class="string">&#x27;\n&#x27;</span>+seq+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">    repcopies.append([<span class="string">&#x27;Repeat_&#x27;</span>+<span class="built_in">str</span>(rous_count),length])  </span><br><span class="line">          </span><br><span class="line"><span class="keyword">if</span> rous_count == <span class="number">0</span>:  </span><br><span class="line">    <span class="built_in">print</span> <span class="string">&quot;\tRepeats of unusual size? I don&#x27;t think they exist&quot;</span>  </span><br><span class="line">g.close()  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;Repeat fasta file is done, as you wish.&#x27;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Now find each copy of each repeat. Again, this is because the blastn output file does not have every possible alignment.  </span></span><br><span class="line"><span class="comment"># It is also because the information on locations and strand is not organized well in the blastn output.  </span></span><br><span class="line"><span class="comment"># In addition, this subroutine eliminates duplicates of nested repeats.  </span></span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Finding all copies of repeats...&quot;</span>  </span><br><span class="line">g = <span class="built_in">open</span>(outfa, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">os.system(blast_path+<span class="string">&#x27;blastn -query &#x27;</span>+outfa+<span class="string">&#x27; -strand both -subject &#x27;</span>+infile+<span class="string">&#x27; -word_size &#x27;</span>+wordsize+<span class="string">&#x27; -reward 1 -penalty -20 -ungapped -dust no -soft_masking false -evalue 1000 -outfmt &quot;10 qseqid length sstart send sstrand qcovhsp&quot; &gt; &#x27;</span>+temprepeats)  </span><br><span class="line">g.close()  </span><br><span class="line">  </span><br><span class="line">tempr = <span class="built_in">open</span>(temprepeats, <span class="string">&#x27;r&#x27;</span>)  </span><br><span class="line">reader = csv.reader(tempr)  </span><br><span class="line">replist = <span class="built_in">list</span>(reader)  </span><br><span class="line">tempr.close()  </span><br><span class="line">  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Making a table of the repeats...&quot;</span>  </span><br><span class="line">sum_rep_len = <span class="number">0</span>  </span><br><span class="line">bin_dict = &#123;&#125;  </span><br><span class="line">binned = [seqname,seqlen,<span class="number">0</span>]  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># defining the bins  </span></span><br><span class="line">i = <span class="number">0</span>  </span><br><span class="line">j = <span class="number">50</span>  </span><br><span class="line"><span class="keyword">while</span> j &lt; <span class="number">1000</span>:  </span><br><span class="line">    bin_dict[i] = j  </span><br><span class="line">    binned.append(<span class="number">0</span>)  </span><br><span class="line">    i += <span class="number">1</span>  </span><br><span class="line">    j += <span class="number">50</span>  </span><br><span class="line"><span class="keyword">while</span> j &lt;= <span class="number">10000</span>:  </span><br><span class="line">    bin_dict[i] = j  </span><br><span class="line">    binned.append(<span class="number">0</span>)  </span><br><span class="line">    i +=<span class="number">1</span>  </span><br><span class="line">    j += <span class="number">250</span>  </span><br><span class="line">      </span><br><span class="line"><span class="comment"># make list for entire sequence, set each position as 0  </span></span><br><span class="line">posit = []  </span><br><span class="line"><span class="keyword">for</span> n <span class="keyword">in</span> <span class="built_in">range</span>(seqlen):  </span><br><span class="line">    posit.append(<span class="number">0</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Thanks to Emily Wynn for suggesting qcovhsp for this loop.  </span></span><br><span class="line"><span class="comment"># if qcovhsp is &gt;98%, write to the file  </span></span><br><span class="line"><span class="comment"># write tab separated values of repeat name, length, start, end, strand to outtab  </span></span><br><span class="line"><span class="comment"># make list for genbank file  </span></span><br><span class="line"><span class="comment"># Keep stats on lengths  </span></span><br><span class="line">rt = <span class="built_in">open</span>(outtab, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">rt.write(seqname+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(seqlen)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">templist = []  </span><br><span class="line">gblist =[]  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># look at each repeat in turn  </span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(replist)):  </span><br><span class="line">    <span class="comment"># if repeat is good (&gt;98% identical to another one), write it to the file, and put the name in a list  </span></span><br><span class="line">    <span class="keyword">if</span> <span class="built_in">int</span>(replist[i][<span class="number">5</span>])&gt;<span class="number">98</span>:  </span><br><span class="line">        rt.write(<span class="built_in">str</span>(replist[i][<span class="number">0</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">1</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">2</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">3</span>])+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(replist[i][<span class="number">4</span>])+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">        <span class="keyword">if</span> replist[i][<span class="number">4</span>] == <span class="string">&#x27;minus&#x27;</span>:  </span><br><span class="line">            location = <span class="string">&#x27;complement(&#x27;</span>+replist[i][<span class="number">3</span>]+<span class="string">&#x27;..&#x27;</span>+replist[i][<span class="number">2</span>]+<span class="string">&#x27;)&#x27;</span>  </span><br><span class="line">        <span class="keyword">else</span>:  </span><br><span class="line">            location = replist[i][<span class="number">2</span>]+<span class="string">&#x27;..&#x27;</span>+replist[i][<span class="number">3</span>]  </span><br><span class="line">        gblist.append(<span class="string">&#x27;     repeat_region   &#x27;</span>+location+<span class="string">&#x27;\n                     /rpt_type=dispersed\n                     /label=&#x27;</span>+replist[i][<span class="number">0</span>]+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">        templist.append(replist[i][<span class="number">0</span>])  </span><br><span class="line">        <span class="comment"># then write 1&#x27;s at every position in the sequence covered by that repeat  </span></span><br><span class="line">        <span class="comment"># these can then be summed to get total bases of repeats  </span></span><br><span class="line">        <span class="comment"># bases in overlapping repeats are only counted once  </span></span><br><span class="line">        <span class="keyword">for</span> n <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">int</span>(replist[i][<span class="number">2</span>]), <span class="built_in">int</span>(replist[i][<span class="number">3</span>])):  </span><br><span class="line">            posit[n] = <span class="number">1</span>  </span><br><span class="line">        <span class="comment"># then scan through bin sizes and if a repeat is greater than the  </span></span><br><span class="line">        <span class="comment"># bin_dict size cutoff, add one to the bin  </span></span><br><span class="line">        <span class="keyword">for</span> j <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(binned)-<span class="number">4</span>, -<span class="number">1</span>, -<span class="number">1</span>):  </span><br><span class="line">            <span class="keyword">if</span> <span class="built_in">int</span>(replist[i][<span class="number">1</span>]) &gt;= bin_dict[j]:  </span><br><span class="line">                binned[j+<span class="number">3</span>] +=<span class="number">1</span>  </span><br><span class="line">                <span class="keyword">break</span>  </span><br><span class="line">sum_rep_len = posit.count(<span class="number">1</span>)  </span><br><span class="line">binned[<span class="number">2</span>] = sum_rep_len  </span><br><span class="line">rt.close()  </span><br><span class="line"><span class="keyword">if</span> genbank == <span class="literal">True</span>:  </span><br><span class="line">    gb = <span class="built_in">open</span>(outgb, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">    <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(gblist)):  </span><br><span class="line">        gb.write(gblist[i])  </span><br><span class="line">    gb.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># write tab separated values of repeat name, length, copy number to outcount  </span></span><br><span class="line"><span class="comment"># first two lines are also a table of stats on repeats  </span></span><br><span class="line">rc = <span class="built_in">open</span>(outcount,<span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">rc.write(<span class="string">&#x27;Sequence\tGenome_size\tNumROUS\tAvgSize\tAvgCopyNum\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">numrous = <span class="number">0</span>  </span><br><span class="line">sizerous = <span class="number">0</span>  </span><br><span class="line">copyrous = <span class="number">0</span>  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(repcopies)):  </span><br><span class="line">    repname = repcopies[i][<span class="number">0</span>]  </span><br><span class="line">    replen = <span class="built_in">float</span>(repcopies[i][<span class="number">1</span>])  </span><br><span class="line">    repcop = <span class="built_in">float</span>(templist.count(repname))  </span><br><span class="line">  </span><br><span class="line">    numrous += <span class="number">1</span>  </span><br><span class="line">    sizerous += replen  </span><br><span class="line">    copyrous += repcop  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">if</span> numrous == <span class="number">0</span>:  </span><br><span class="line">    avsizerous = <span class="string">&#x27;NA&#x27;</span>  </span><br><span class="line">    avcopyrous = <span class="string">&#x27;NA&#x27;</span>  </span><br><span class="line"><span class="keyword">else</span>:  </span><br><span class="line">    avsizerous = sizerous/numrous  </span><br><span class="line">    avcopyrous = copyrous/numrous  </span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">rc.write(seqname+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(seqlen)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(numrous)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(avsizerous)+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(avcopyrous)+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(repcopies)):  </span><br><span class="line">    rc.write(repcopies[i][<span class="number">0</span>]+<span class="string">&#x27;\t&#x27;</span>+repcopies[i][<span class="number">1</span>]+<span class="string">&#x27;\t&#x27;</span>+<span class="built_in">str</span>(templist.count(repcopies[i][<span class="number">0</span>]))+<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">  </span><br><span class="line">rc.close()  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Write binned table headers, then stats for this sequence.  </span></span><br><span class="line">binfile = <span class="built_in">open</span>(outbin, <span class="string">&#x27;w&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;Sequence\tSeq_len\tRep_len\t&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(bin_dict)):  </span><br><span class="line">    binfile.write(<span class="built_in">str</span>(bin_dict[i])+<span class="string">&#x27;\t&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(<span class="built_in">len</span>(binned)):  </span><br><span class="line">    binfile.write(<span class="built_in">str</span>(binned[i])+<span class="string">&#x27;\t&#x27;</span>)  </span><br><span class="line">binfile.write(<span class="string">&#x27;\n&#x27;</span>)  </span><br><span class="line">binfile.close()  </span><br><span class="line"><span class="built_in">print</span> <span class="string">&quot;Repeat tables are done, as you wish.&quot;</span>  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Removing temp files if necessary  </span></span><br><span class="line"><span class="keyword">if</span> keep == <span class="literal">False</span>:  </span><br><span class="line">    os.system(<span class="string">&#x27;rm &#x27;</span>+tempblast+<span class="string">&#x27; &#x27;</span>+temprepeats+<span class="string">&#x27; &#x27;</span>+tempparse)  </span><br><span class="line">  </span><br><span class="line"><span class="comment"># Rachael Schulte, William Goldman and Rob Reiner inspired this section of code  </span></span><br><span class="line">quote_dict = &#123;<span class="number">0</span>:<span class="string">&quot;48656c6c6f2e204d79206e616d6520697320496e69676f204d6f6e746f79612e20596f75206b696c6c6564206d79206661746865722e205072657061726520746f206469652e&quot;</span>, <span class="number">1</span>:<span class="string">&quot;5768656e20492077617320796f7572206167652c2074656c65766973696f6e207761732063616c6c656420626f6f6b732e&quot;</span>, <span class="number">2</span>:<span class="string">&quot;486176652066756e2073746f726d696e2720646120636173746c6521&quot;</span>, <span class="number">3</span>:<span class="string">&quot;4d79207761792773206e6f7420766572792073706f7274736d616e6c696b652e&quot;</span>, <span class="number">4</span>:<span class="string">&quot;596f75206b656570207573696e67207468617420776f72642e204920646f206e6f74207468696e6b206974206d65616e73207768617420796f75207468696e6b206974206d65616e732e&quot;</span>, <span class="number">5</span>:<span class="string">&quot;4d75726465726564206279207069726174657320697320676f6f642e&quot;</span>,<span class="number">6</span>:<span class="string">&quot;496e636f6e6365697661626c6521&quot;</span>, <span class="number">7</span>:<span class="string">&quot;5468657265277320612062696720646966666572656e6365206265747765656e206d6f73746c79206465616420616e6420616c6c20646561642e&quot;</span>, <span class="number">8</span>:<span class="string">&quot;596f7520727573682061206d697261636c65206d616e2c20796f752067657420726f7474656e206d697261636c65732e&quot;</span>, <span class="number">9</span>:<span class="string">&quot;476f6f64206e696768742c20576573746c65792e20476f6f6420776f726b2e20536c6565702077656c6c2e2049276c6c206d6f7374206c696b656c79206b696c6c20796f7520696e20746865206d6f726e696e672e&quot;</span>,<span class="number">10</span>:<span class="string">&quot;4e6f206d6f7265207268796d65732c2049206d65616e2069742120416e79626f64792077616e742061207065616e75743f&quot;</span>&#125;  </span><br><span class="line"><span class="keyword">import</span> random, binascii  </span><br><span class="line">z = random.randint(<span class="number">0</span>,<span class="number">10</span>)  </span><br><span class="line"><span class="built_in">print</span> binascii.unhexlify(quote_dict[z])+<span class="string">&#x27;\n&#x27;</span>  </span><br></pre></td></tr></table></figure>

<ol start="3">
<li>运行<br><code>python2 ROUSFinder2.0.py mito.genome.fa</code></li>
</ol>
<ul>
<li>-m参数指定注释重复序列的最小长度，默认是50bp</li>
<li>-b参数指定blastn的所在路径</li>
<li>-o指定输出文件</li>
<li>-gb生成GenBank格式文件</li>
</ul>
<ol start="4">
<li>输出文件<br>输出文件有四个，mito.genome.fa_binned.txt, mito.genome.fa_rep.fasta, mito.genome.fa_rep_counts.txt, mito.genome.fa_rep_table.txt。</li>
</ol>
<p>如果使用了-gb还会生成mito.genome.fa_repeats.gb.txt。</p>
<ul>
<li>mito.genome.fa_rep_table.txt文件，内容包括重复序列的长度，起始位置，正负链等信息。</li>
<li>mito.genome.fa_rep.fasta是鉴定出来的重复序列。</li>
<li>mito.genome.fa_rep_counts.txt是重复序列的数量信息，包括序列ID-Sequence,基因组尺寸-Genome_size,重复序列数量-NumROUS,重复序列平均尺寸-AvgSize,平均重复次数-AvgCopyNum，以及每个重复序列的长度信息。</li>
<li>mito.genome.fa_binned.txt，其他统计信息。</li>
<li>mito.genome.fa_repeats.gb.txt是GeneBank格式的重复序列文件。</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/04/01/scripts/" data-id="cl1aiqcs1004em4vkf1wqg29d" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bio_experiment" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/03/27/bio_experiment/" class="article-date">
  <time datetime="2021-03-27T08:50:00.000Z" itemprop="datePublished">2021-03-27</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/experiment/">experiment</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/03/27/bio_experiment/">分子生物学实验</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=36307139&auto=1&height=32"></iframe></div>


<h2 id="常用PCR参数"><a href="#常用PCR参数" class="headerlink" title="常用PCR参数"></a>常用PCR参数</h2><ol>
<li>PCR Easy tag 体系(20μl)</li>
</ol>
<ul>
<li>ddH2O 12.2μl</li>
<li>10Xbuffer 2.0μl</li>
<li>dNTP 1.6μl</li>
<li>Easy tag酶 0.2μl</li>
<li>DNA 2μl</li>
<li>Primer-F 1μl</li>
<li>Primer-R 1μl</li>
</ul>
<ol start="2">
<li>PCR常用程序</li>
</ol>
<ul>
<li>预热 94℃ 10min</li>
</ul>
<p>以下步骤循环28-34次</p>
<ul>
<li>解螺旋94℃ 0.5-1min</li>
<li>退火53℃ 30-45seconds</li>
<li>72℃ 1min15seconds</li>
</ul>
<p>循环结束继续以下步骤</p>
<ul>
<li>72℃ 10min</li>
<li>16℃ 10min</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/03/27/bio_experiment/" data-id="cl1aiqcr3001cm4vkdu2qbxcs" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-taxon_Bauhinia" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/03/27/taxon_Bauhinia/" class="article-date">
  <time datetime="2021-03-27T08:40:00.000Z" itemprop="datePublished">2021-03-27</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/taxon/">taxon</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/03/27/taxon_Bauhinia/">羊蹄甲属常见三种类群的分类特征</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=26090155&auto=1&height=32"></iframe></div>

<p>羊蹄甲属（Bauhinia）的羊蹄甲（B. purpurea）、洋紫荆（B. variegata ）和红花羊蹄甲（B. blakeana）是华南地区三种园林植物，在广州尤为普遍，常植为行道树。</p>
<p><strong>羊蹄甲属常见三种类群的分类特征</strong></p>
<table>
<thead>
<tr>
<th>羊蹄甲属（Bauhinia）学名</th>
<th>B. purpurea</th>
<th>B. variegata</th>
<th>B. variegata var. candida</th>
<th>Bauhinia purpurea x variegata ‘Blakeana’, cv. nov.</th>
</tr>
</thead>
<tbody><tr>
<td>中文正名&#x2F;中国大陆常用俗名</td>
<td>羊蹄甲</td>
<td>洋紫荆</td>
<td>白花洋紫荆</td>
<td>红花羊蹄甲</td>
</tr>
<tr>
<td>香港常用俗名</td>
<td>红花羊蹄甲</td>
<td>宫粉羊蹄甲</td>
<td>白花羊蹄甲</td>
<td>洋紫荆</td>
</tr>
<tr>
<td>台湾常用俗名</td>
<td>洋紫荆</td>
<td>羊蹄甲</td>
<td>–</td>
<td>艶&#x2F;豔&#x2F;艳（yan4）紫荆</td>
</tr>
<tr>
<td>主要区别</td>
<td>能育雄蕊3，退化雄蕊5-6，花瓣较狭窄，具长柄</td>
<td>能育雄蕊5，无退化雄蕊，落叶后盛花，花色粉红，总状花序极短缩，正常结果</td>
<td>洋紫荆的白花变种，花白色</td>
<td>雄蕊5，花叶同期，花色紫红，总状花序开展，不结果</td>
</tr>
<tr>
<td>雄蕊</td>
<td>能育雄蕊3，退化雄蕊5-6</td>
<td>能育雄蕊5(7)，无退化雄蕊</td>
<td>同洋紫荆</td>
<td>雄蕊5(7)，退化雄蕊2-5</td>
</tr>
<tr>
<td>花期</td>
<td>花期9-11月</td>
<td>花期1-4月（3月盛花期）</td>
<td>同洋紫荆</td>
<td>花期全年（3-4月盛花期）</td>
</tr>
<tr>
<td>花色</td>
<td>花色桃红，中国发现一棵白花</td>
<td>花色粉红到淡紫，有白花、黄花变种</td>
<td>花白色</td>
<td>花色紫红</td>
</tr>
<tr>
<td>花着生</td>
<td>集中顶生</td>
<td>分散腋生</td>
<td>同洋紫荆</td>
<td>集中顶生</td>
</tr>
<tr>
<td>花瓣形态</td>
<td>窄而相互分离</td>
<td>宽而相互重叠</td>
<td>同洋紫荆</td>
<td>稍宽而分离</td>
</tr>
<tr>
<td>物候</td>
<td></td>
<td>落叶后盛花</td>
<td>同洋紫荆</td>
<td>花叶同期</td>
</tr>
<tr>
<td>其他</td>
<td>树皮、花和根可药用外用，根皮剧毒</td>
<td>花芽、嫩叶和幼果可食</td>
<td>同洋紫荆，云南有白花变种的花瓣食用史</td>
<td>香港市花，红花羊蹄甲是羊蹄甲和洋紫荆杂交起源的杂交种，不结果。</td>
</tr>
</tbody></table>
<p>如果同时和台湾人香港人讨论香港市花，一定很有趣。</p>
<ul>
<li>羊蹄甲&#x3D;台湾的洋紫荆</li>
<li>洋紫荆&#x3D;台湾的羊蹄甲</li>
<li>红花羊蹄甲&#x3D;香港的洋紫荆</li>
<li>台湾的洋紫荆&#x3D;香港的红花羊蹄甲</li>
</ul>
<p>你记住了吗？</p>
<p>课后习题，香港市花是什么？</p>
<p>reference</p>
<ol>
<li><a target="_blank" rel="noopener" href="https://zh.wikipedia.org/wiki/%E7%BE%8A%E8%B9%84%E7%94%B2%E5%B1%9E#%E5%90%8D%E7%A8%B1%E6%B7%B7%E6%B7%86">wiki: Bauhinia</a></li>
<li><a target="_blank" rel="noopener" href="https://bsapubs.onlinelibrary.wiley.com/doi/full/10.3732/ajb.92.3.525">paper: Hybrid origin of Bauhinia blakeana</a></li>
<li><a target="_blank" rel="noopener" href="http://www.iplant.cn/info/Bauhinia%20purpurea?t=z">Bauhinia purpuprea: 羊蹄甲</a></li>
<li><a target="_blank" rel="noopener" href="http://www.iplant.cn/info/Bauhinia%20variegata?t=z">Bauhinia variegata： 洋紫荆</a></li>
<li><a target="_blank" rel="noopener" href="http://www.iplant.cn/info/Bauhinia%20%C3%97%20blakeana">Bauhinia × blakeana: 红花羊蹄甲</a></li>
<li>《羊蹄甲属植物资源培育与利用》魏丹著 2021年12月</li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/03/27/taxon_Bauhinia/" data-id="cl1aiqcs2004mm4vk9se84o9g" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_caculate.divergence.time" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/03/25/bioinfo_caculate.divergence.time/" class="article-date">
  <time datetime="2021-03-25T08:50:00.000Z" itemprop="datePublished">2021-03-25</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/03/25/bioinfo_caculate.divergence.time/">估算系统树分歧时间（paml.mcmctree,r8s)</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=330 height=86 src="//music.163.com/outchain/player?type=2&id=27853347&auto=1&height=66"></iframe></div>

<p>估算分歧时间是在已经建好物种的系统发育树，根据系统发育树上已知类群间分化时间（一般来自化石或者其他研究的标定），或者根据权威研究中对特定类群的碱基替换速率的估计（例如蕨类4.79e-9 subst.&#x2F;syn. site&#x2F;year，水稻2.5e-8 subst.&#x2F;syn. site&#x2F;year)进行估算系统发育树上其他类群的分歧时间。</p>
<p>一般来说，我们尽量使用化石证据的时间（更可靠），可以在网站<a target="_blank" rel="noopener" href="http://www.timetree.org/">timetree</a>搜索化石或者其他文章标定的类群分化时间；实在没有化石可用时，则使用碱基替换速率常数。</p>
<h1 id="1-原理"><a href="#1-原理" class="headerlink" title="1. 原理"></a>1. 原理</h1><p>尝试用数学思维讲讲估算分歧时间的原理：即路程&#x3D;速度*时间，经过多长<strong>时间</strong>（地质时间），通过多大的<strong>速度</strong>（碱基替换速率），完成从祖先物种的DNA到现在物种的DNA的变化，总体上的碱基替换积累即<strong>路程</strong>。</p>
<p>假设下面Figure 1. 系统发育树示意图(A4,(A3,(A2,A1)B)C)D;是有枝长有根树，A1-A4都是现在的物种（代表时间也已知），目的是要估算B和C节点的时间。</p>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_caculate.divergence.time_phylogeny.jpg?raw=true" width=100% height=80% title="phylogeny示意图(A4,(A3,(A2,A1)B)C)D" align=center/>

<p><strong><p align="center">Figure 1. 系统发育示意图</p></strong></p>
<ul>
<li>第一种情况，当已知D节点时间时，求B节点时间（C节点同理）</li>
</ul>
<p>由于A1的时间点已知（即现在），那么从D到A1的时间跨度是已知的；从D到A1的枝长是已知的，枝长代表的是这个过程的碱基替换数量，即路程；时间和路程已知，我们可以算出速度；同时，从D到B的路程也是已知的（因为枝长已知），那么在假设匀速的前提下，我们也可以算出从D到B的时间，这样即可标定B（即A1和A2分歧时间）的时间。</p>
<ul>
<li>第二种情况，当已知速度（碱基替换速率，常数）时，求B节点时间</li>
</ul>
<p>由于从B到A1的枝长是已知的，枝长代表的是这个过程的碱基替换数量，即路程；速度已知，那么我们可以算得B到A1的时间跨度，由于A1的时间即现在，时间跨度就是B节点的时间。</p>
<p>以上解释是非常简化的理解这个估算过程，实际情况复杂得多，物种不会匀速进行碱基替换，所以软件会使用算法进行精细化估算。</p>
<h1 id="2-估算分歧时间常用软件"><a href="#2-估算分歧时间常用软件" class="headerlink" title="2. 估算分歧时间常用软件"></a>2. 估算分歧时间常用软件</h1><ul>
<li>beast：Linux也调用图形界面</li>
<li>paml的mcmctree：推荐</li>
<li>r8s：非常快</li>
</ul>
<h1 id="3-paml的mcmctree估算分歧时间"><a href="#3-paml的mcmctree估算分歧时间" class="headerlink" title="3. paml的mcmctree估算分歧时间"></a>3. paml的mcmctree估算分歧时间</h1><h2 id="3-1-paml介绍"><a href="#3-1-paml介绍" class="headerlink" title="3.1. paml介绍"></a>3.1. paml介绍</h2><p>PAML (Phylogenetic Analysis by Maximum Likelihood)，是杨子恒开发的一款利用DNA或Protein数据使用最大似然法进行系统发育分析的软件。</p>
<p>注意事项：</p>
<ol>
<li>PAML要求的输入文件是多序列比对结果。PAML软件不能进行多序列比对，需要使用其他软件进行分析。</li>
<li>PAML软件进行基于密码子的分析要求输入的序列必须是去除了Introns等非编码区的DNA序列，且其长度是3的整数倍，且第一个核酸位点是密码子的第一位。PAML软件不能进行基因预测，密码子的比对结果需要自行编写程序分析得到。</li>
<li>对于较大数据（物种数&gt;&#x3D;10）量的分析，输入的树文件最好使用其他软件计算。</li>
</ol>
<h2 id="3-2-paml软件包含的程序"><a href="#3-2-paml软件包含的程序" class="headerlink" title="3.2. paml软件包含的程序"></a>3.2. paml软件包含的程序</h2><ul>
<li>baseml 用于对核酸序列进行最大似然法分析。</li>
<li>codeml 用于对密码子和蛋白序列进行最大似然法分析。该程序整合了以前的两支程- 序codonml和aaml，前者用于对密码子序列进行分析，后者对蛋白序列进行分析。在新的codeml程序的配置文件中设置seqtype&#x3D;1，表示使用codonml命令，设置- seqtype&#x3D;2，表示使用aaml命令。</li>
<li>evolver 用于模拟核酸、密码子或蛋白序列。</li>
<li>basemlg 相比于baseml，该程序应用了连续的gamma模型。当数据中的物种数多于6- 到7个时，该程序运行非常慢以至得不到结果。这时，需要使用设置离散gamma模型的- baseml命令。</li>
<li>mcmctree 该程序应用Bayesian MCMC算法计算物种分歧时间。</li>
<li>pamp  用于简约法分析。</li>
<li>yn00  用于对两条编码蛋白的DNA序列进行比较并计算dnds。</li>
<li>chi2  用于在likelihood ratio test中计算卡方临界值和p值。</li>
</ul>
<p>mcmctree通过调用baseml(核苷酸数据)、codeml（密码子或者氨基酸数据）估算模型参数，之后估算分歧时间；</p>
<p><strong>指南一和指南二只适用于核苷酸数据；氨基酸数据看指南三</strong></p>
<h2 id="3-3-指南一：paml常规方法估算分歧时间"><a href="#3-3-指南一：paml常规方法估算分歧时间" class="headerlink" title="3.3. 指南一：paml常规方法估算分歧时间"></a>3.3. 指南一：paml常规方法估算分歧时间</h2><p>paml常规方法只适用于核苷酸数据；</p>
<h3 id="3-3-1-input-files"><a href="#3-3-1-input-files" class="headerlink" title="3.3.1. input files"></a>3.3.1. input files</h3><p>准备3个输入文件</p>
<h4 id="3-3-1-1-input-tre-带有校准点的有根树文件"><a href="#3-3-1-1-input-tre-带有校准点的有根树文件" class="headerlink" title="3.3.1.1. input.tre - 带有校准点的有根树文件"></a>3.3.1.1. input.tre - 带有校准点的有根树文件</h4><p>把iqtree&#x2F;raxml-ng等建树软件中得到的树文件中的支持率和枝长信息删除，添加化石校准点时间信息（格式是时间范围’&gt;0.23&lt;0.26’或者时间点‘@0.245’)，单位时百万年前100Ma；再在首行添加两个数字（物种数量和树的数量），空格隔开，可得到input.tre文件。</p>
<figure class="highlight plaintext"><figcaption><span>#cat input.tre</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">5 1</span><br><span class="line">((((A,B),C),D)&#x27;&gt;0.23&lt;0.26&#x27;,E);</span><br></pre></td></tr></table></figure>

<ul>
<li>文件内容包含两行：第一行表述树中有n个物种，共计1个树，两个数值之间用空分割；第二行则是Newick格式树信息。</li>
<li>其中第二行包含有校准点信息。校准点信息一般指95%HPD（Highest Posterior Density）对应的置信区间；校准点单位是100MYA（软件说明文档中使用该单位，也推荐使用该单位，若使用其它单位，后续配置文件中的相关参数也需要对应修改）。</li>
<li>此外，Newick格式的树尾部一定要有分号，没有的话程序可能不能正常运行。</li>
</ul>
<h4 id="3-3-1-2-input-phy-多序列比对文件（phylip格式）"><a href="#3-3-1-2-input-phy-多序列比对文件（phylip格式）" class="headerlink" title="3.3.1.2. input.phy - 多序列比对文件（phylip格式）"></a>3.3.1.2. input.phy - 多序列比对文件（phylip格式）</h4><ul>
<li>PAML要求输入的Phylip格式，其物种名和后面的序列之间至少间隔两个空格（是为了允许物种名的属名和种名之间有一个空格）。</li>
<li>python的SeqIO的转换格式模块获得的phylip和nexus格式都不行。</li>
<li>推荐用<code>echo $(cat input.align.fa) |sed &quot;s/ &gt;/\n/g&quot; |sed &quot;s/&gt;//g&quot;|sed &quot;s/ /  /g&quot; &gt;input.phy</code>手动转化align好的fas格式文件；再在首行添加两个数值，空格隔开，物种数量和碱基数量；实现fasta2phylip。</li>
</ul>
<p>如果有<strong>多个区域的序列</strong>，比如exon和intron，LSC、SSC和IR，不同的基因，密码子的第一二三位，需要不同的模型分开估算，那可以把各自区域分别align之后制作多个phy文件，再合并到一起，用空行隔开，组成input.phy文件。（此时mcmctree.ctl的ndata值为区域的个数）。</p>
<p>写了一个脚本来制作多区域的phylip文件，其中OG.length包含两列，一列OGID，一列比对后序列长度。</p>
<figure class="highlight shell"><figcaption><span>## cat fa2phylip.sh</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">cat ./OG.length | while read line</span><br><span class="line">do</span><br><span class="line">        sample_id=$(echo $line |awk &#x27;&#123;print $1&#125;&#x27;) #获取OG.ID</span><br><span class="line">        sample_a=$(echo $line |awk &#x27;&#123;print $2&#125;&#x27;) #获取序列长度</span><br><span class="line">        sed &quot;s/_.*//g&quot; ../singlegenetree/$&#123;sample_id&#125;/$&#123;sample_id&#125;.mafft.pep|seqkit seq -w 0|sed -E &quot;:a;N;s/\n/ /g;ta&quot; |sed &quot;s/ &gt;/\n/g&quot; |sed &quot;s/&gt;//g&quot;|sed &quot;s/ /  /g&quot;|sed &quot;1i\5  $&#123;sample_a&#125;&quot; |sed &#x27;1i\ &#x27; &gt; ./phy/$&#123;sample_id&#125;.phy #把上一步获取的$&#123;sample_id&#125;.mafft.pep改为phylip格式，并在首行添加空格行（为了合并后每个OG用空行隔开）</span><br><span class="line"></span><br><span class="line">done</span><br><span class="line">cat ./phy/*phy &gt;input.phy #合并所有phy文件为一个Ane.phy</span><br></pre></td></tr></table></figure>

<h4 id="3-3-1-3-mcmctree-ctl-mcmctree程序的配置文件"><a href="#3-3-1-3-mcmctree-ctl-mcmctree程序的配置文件" class="headerlink" title="3.3.1.3. mcmctree.ctl - mcmctree程序的配置文件"></a>3.3.1.3. mcmctree.ctl - mcmctree程序的配置文件</h4><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br></pre></td><td class="code"><pre><span class="line">          seed = -1 *设置随机数作为seed，-1代表使用系统当前时间作为随机数</span><br><span class="line">       seqfile = input.phy *输入多序列比对文件</span><br><span class="line">      treefile = input.tre *带校准点（化石时间）的有根树文件</span><br><span class="line">       outfile = mcmc.out *输出文件</span><br><span class="line">      mcmcfile = mcmc.txt *输出的mcmc信息文件，可用Tracer软件查看</span><br><span class="line"></span><br><span class="line">       seqtype = 0  * 设置多序列比对数据类型；0：核酸数据；1：密码子比对数据；2：氨基酸数据；</span><br><span class="line">	   usedata = 1</span><br><span class="line">     * 是否利用多序列比对数据；</span><br><span class="line">	   * 0: no data不使用，不会进行likelihood估算，会快速得到mcmc树，但分歧时间不可用; </span><br><span class="line">	   * 1:seq like，使用多序列比对数据进行likelihood估算，正常进行mcmc; usedata=1时model无法选择；</span><br><span class="line">	   * 2:normal 进行正常的approximation likelihood分析，不读取多序列比对数据，直接读取当前目录的in.BV文件，in.BV是由usedata = 3时生成的out.BV重命名得来；此外，由于程序BUG，当设置usedata = 2时，一定要在改行参数后加 *，否则程序报错 Error: file name empty..；</span><br><span class="line">	   * 3：程序利用多序列比对数据调用baseml/codeml命令对数据进行分析，生成out.BV文件。由于mcmctree调用baseml/codeml进行估算的参数设置可能不太好（特别时对蛋白序列进行估算时），推荐自己修改软件自动生成的baseml/codeml配置文件，然后再手动运行baseml/codeml命令，再整合其结果文件为out.BV文件。</span><br><span class="line"></span><br><span class="line">		 ndata = 1    * 输入的多序列比对的数据区域的数量；</span><br><span class="line">         clock = 2    * 设置分子钟算法，1: global clock，表示所有分支进化速率一致; 2: independent rates，各分支的进化速率独立且进化速率的对数log(r)符合正态分布; 3，correlated rates方法，和方法2类似，但是log(r)的方差和时间t相关。</span><br><span class="line">*       TipDate = 1 100  *当外部节点由取样时间时使用该参数进行设置，同时该参数也设置了时间单位。具体数据示例请见examples/TipData文件夹。</span><br><span class="line">        RootAge = &#x27;&lt;10&#x27;  * constraint on root age, used if no fossil for root.设置root节点的分歧时间，一般设置一个最大值。</span><br><span class="line"></span><br><span class="line">         model = 4    * models for DNA:</span><br><span class="line">                        * 0:JC69, 1:K80, 2:F81, 3:F84, 4:HKY85；*设置碱基替换模型；当设置usedata = 1时，model不能使用超过4的模型，所以usedata = 1时用model = 4；usedata不等于1时，用model = 7，即GTR模型；</span><br><span class="line">                      * models for codons:</span><br><span class="line">                        * 0:one 恒定速率模型, 1:b 中性模型, 2:2 or more dN/dS ratios for branches 选择模型。</span><br><span class="line">                      * models for AAs or codon-translated AAs:</span><br><span class="line">                        * 0:poisson, 1:proportional, 2:Empirical, 3:Empirical+F</span><br><span class="line">                        * 6:FromCodon, 7:AAClasses, 8:REVaa_0, 9:REVaa(nr=189)</span><br><span class="line">         alpha = 0.5   * alpha for gamma rates at sites；*核酸序列中不同位点，其进化速率不一致，其变异速率服从GAMMA分布。一般设置GAMMA分布的alpha值为0.5。若该参数值设置为0，则表示所有位点的进化速率一致。此外，当userdata = 2时，alpha、ncatG、alpha_gamma、kappa_gamma这4个GAMMA参数无效。因为userdata = 2时，不会利用到多序列比对的数据。</span><br><span class="line">         ncatG = 5    * No. categories in discrete gamma；设置离散型GAMMA分布的categories值。</span><br><span class="line"></span><br><span class="line">     cleandata = 0    * remove sites with ambiguity data (1:yes, 0:no)?</span><br><span class="line"></span><br><span class="line">       BDparas = 1 1 0.1   * birth, death, sampling；*设置出生率、死亡率和取样比例。若输入有根树文件中的时间单位发生改变，则需要相应修改出生率和死亡率的值。例如，时间单位由100Myr变换为1Myr，则要设置成&quot;.01 .01 0.1&quot;。</span><br><span class="line">   kappa_gamma = 6 2      * gamma prior for kappa；设置kappa（转换/颠换比率）的GAMMA分布参数。</span><br><span class="line">   alpha_gamma = 1 1      * gamma prior for alpha；设置GAMMA形状参数alpha的GAMMA分布参数。</span><br><span class="line"></span><br><span class="line">   rgene_gamma = 2 20 1   * gamma prior for rate for genes；设置序列中所所有位点平均[碱基/密码子/氨基酸]替换率的Dirichlet-GAMMA分布参数：alpha=2、beta=20、初始平均替换率为每100million年（取决于输入有根树文件中的时间单位）1个替换。若时间单位由100Myr变换为1Myr，则要设置成&quot;2 2000 1&quot;。总体上的平均进化速率为：2 / 20 = 0.1 个替换 / 每100Myr，即每个位点每年的替换数为 1e-9。</span><br><span class="line">  sigma2_gamma = 1 10 1    * gamma prior for sigma^2     (for clock=2 or 3)；设置所有位点进化速率取对数后方差（sigma的平方）的Dirichlet-GAMMA分布参数：alpha=1、beta=10、初始方差值为1。当clock参数值为1时，表示使用全局的进化速率，各分枝的进化速率没有差异，即方差为0，该参数无效；当clock参数值为2时，若修改了时间单位，该参数值不需要改变；当clock参数值为3时，若修改了时间单位，该参数值需要改变。</span><br><span class="line"></span><br><span class="line">      finetune = 1: .1 .1 .1 .1 .1 .1  * times, rates, mixing, paras, RateParas；冒号前的值设置是否自动进行finetune，一般设置成1，然程序自动进行优化分析；冒号后面设置各个参数的步进值：times, musigma2, rates, mixing, paras, FossilErr。由于有了自动设置，该参数不像以前版本那么重要了，可能以后会取消该参数。</span><br><span class="line"></span><br><span class="line">         print = 1  *设置打印mcmc的取样信息：0，不打印mcmc结果；1，打印除了分支进化速率的其它信息（各内部节点分歧时间、平均进化速率、sigma2值）；2，打印所有信息。 </span><br><span class="line">        burnin = 1000000  *将前1000000次迭代burnin后，再进行取样（即打印出该次迭代估算的结果信息，各内部节点分歧时间、平均进化速率、sigma2值和各分支进化速率等）。</span><br><span class="line">      sampfreq = 10  *每10次迭代则取样一次</span><br><span class="line">       nsample = 500000  *当取样次数达到该次数时，则取样结束，同时结束程序。</span><br><span class="line"></span><br><span class="line">*** Note: Make your window wider (100 columns) when running this program.</span><br></pre></td></tr></table></figure>

<h3 id="3-3-2-mcmctree运行"><a href="#3-3-2-mcmctree运行" class="headerlink" title="3.3.2. mcmctree运行"></a>3.3.2. mcmctree运行</h3><p>运行<code>mcmctree mcmctree.ctl</code>即可获得结果。</p>
<h2 id="3-4-指南二：用approximate-likelihood-calculation估算分歧时间【推荐】"><a href="#3-4-指南二：用approximate-likelihood-calculation估算分歧时间【推荐】" class="headerlink" title="3.4. 指南二：用approximate likelihood calculation估算分歧时间【推荐】"></a>3.4. 指南二：用approximate likelihood calculation估算分歧时间【推荐】</h2><p>approximate likelihood法比常规方法快很多，而且可以选择更复杂的GTR模型（model &#x3D; 7）；推荐使用。</p>
<h3 id="3-4-1-input-files"><a href="#3-4-1-input-files" class="headerlink" title="3.4.1. input files"></a>3.4.1. input files</h3><p>输入文件和指南一一致，只是mcmctree.ctl的usedata需要修改为，usedata &#x3D; 3【一定要改】；建议model修改为model &#x3D; 7【建议改】。</p>
<h3 id="3-4-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）"><a href="#3-4-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）" class="headerlink" title="3.4.2. 运行mcmctree mcmctree3.ctl，（mcmctree usedata &#x3D; 3）"></a>3.4.2. 运行<code>mcmctree mcmctree3.ctl</code>，（mcmctree usedata &#x3D; 3）</h3><p>运行<code>mcmctree mcmctree3.ctl</code>；用Maximum Likelihood方法估算枝长和Hessian信息；</p>
<ol>
<li>若比对数据只有一个区域，即ndata &#x3D; 1，则等待运行完成，生成out.BV；</li>
<li>若比对数据多于一个区域，即ndata &gt; 1，mcmctree会多次调用baseml依次处理每个区域，可等待运行完成，也可以做手动并行运行，以加快分析速度；</li>
<li>手动并行多个区域的操作：</li>
</ol>
<ul>
<li>mcmctree运行起来后，从第一区域开始，生成一套tmp0001文件(包括tmp0001.ctl,tmp0001.txt,tmp0001.trees），ctrl+C切断程序，会自动进入下一区域分析，生成一套tmp0002文件，继续ctrl+C切断程序，直到最后一个区域的tmp文件也生成。</li>
<li>对每个区域的tmp文件分别放置一个单独的目录，并在单独目录内运行<code>baseml tmp*.ctl</code>，从而实现多区域并行运行。</li>
<li>所有区域的baseml运行结束后，会在各自目录生成rst2文件，把所有目录的rst2文件cat合并，就是out.BV文件；</li>
<li>或者用命令<code>for i in $(ls tmp*.ctl);do baseml $&#123;i&#125; ; mv rst2 $&#123;i&#125;.rst2; done</code>调用codeml进行多次分析（每次codeml都生成rst2文件，所以每次生成后把rst2改名，否则rst2被覆盖）；然后<code>cat *rst2 &gt;out.BV</code>合并所有rst2文件到out.BV文件；</li>
</ul>
<h3 id="3-4-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）"><a href="#3-4-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）" class="headerlink" title="3.4.3. 运行mcmctree mcmctree2.ctl，（mcmctree usedata &#x3D; 2）"></a>3.4.3. 运行<code>mcmctree mcmctree2.ctl</code>，（mcmctree usedata &#x3D; 2）</h3><ul>
<li>把input.phy，input.tre，mcmctree3.ctl，out.BV四个文件复制到新建目录下，mcmctree3.ctl的usedata改为usedata &#x3D; 2并重命名为mcmctree2.ctl，out.BV重命名为in.BV；</li>
<li>运行<code>mcmctree mcmctree2.ctl</code>即可获得结果。</li>
</ul>
<h2 id="3-5-指南三：用approximate-likelihood-calculation估算氨基酸数据的分歧时间"><a href="#3-5-指南三：用approximate-likelihood-calculation估算氨基酸数据的分歧时间" class="headerlink" title="3.5. 指南三：用approximate likelihood calculation估算氨基酸数据的分歧时间"></a>3.5. 指南三：用approximate likelihood calculation估算氨基酸数据的分歧时间</h2><p>氨基酸数据不能用usedata &#x3D; 1这种模式，只能用approximate likelihood calculation估算分歧时间；方法与指南二类似。</p>
<h3 id="3-5-1-input-files"><a href="#3-5-1-input-files" class="headerlink" title="3.5.1. input files"></a>3.5.1. input files</h3><ul>
<li>输入文件和指南一一致，mcmctree.ctl的参数修改：usedata &#x3D; 3，model &#x3D; 2，seqtype &#x3D; 2；增加一行aaRatefile &#x3D; wag.dat；</li>
<li>同时把paml软件安装位置下的wag.dat复制一份到当前目录；</li>
<li>wag.dat是氨基酸替换速率的数据，与model &#x3D; 2对应，也可以选用其他的氨基酸替换率数据，就用其他的替换率文件(*.dat)；</li>
</ul>
<h3 id="3-5-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）"><a href="#3-5-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）" class="headerlink" title="3.5.2. 运行mcmctree mcmctree3.ctl，（mcmctree usedata &#x3D; 3）"></a>3.5.2. 运行<code>mcmctree mcmctree3.ctl</code>，（mcmctree usedata &#x3D; 3）</h3><p>运行<code>mcmctree mcmctree3.ctl</code>；用Maximum Likelihood方法估算枝长和Hessian信息；</p>
<ol>
<li>若比对数据只有一个区域，即ndata &#x3D; 1，则等待运行完成，生成out.BV；</li>
<li>若比对数据多于一个区域，即ndata &gt; 1，mcmctree会多次调用codeml依次处理每个区域，可等待运行完成，也可以做手动并行运行，以加快分析速度；</li>
<li>手动并行多个区域的操作：</li>
</ol>
<ul>
<li>mcmctree运行起来后，从第一区域开始，生成一套tmp0001文件(包括tmp0001.ctl,tmp0001.txt,tmp0001.trees），ctrl+C切断程序，会自动进入下一区域分析，生成一套tmp0002文件，继续ctrl+C切断程序，直到最后一个区域的tmp文件也生成。</li>
<li>对每个区域的tmp文件分别放置一个单独的目录，修改所有tmp*.ctl内的aaRatefile &#x3D; wag.dat为wag.dat的相对位置或绝对，如aaRatefile &#x3D; ..&#x2F;wag.dat，然后在每一个单独目录内运行<code>codeml tmp*.ctl</code>；</li>
<li>所有区域的codeml运行结束后，会在各自目录生成rst2文件，把所有目录的rst2文件cat合并，就是out.BV文件；</li>
<li>或者用命令<code>for i in $(ls tmp*.ctl);do codeml $&#123;i&#125; ; mv rst2 $&#123;i&#125;.rst2; done</code>调用codeml进行多次分析（每次codeml都生成rst2文件，所以每次生成后把rst2改名，否则rst2被覆盖）；然后<code>cat *rst2 &gt;out.BV</code>合并所有rst2文件到out.BV文件；</li>
</ul>
<h3 id="3-5-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）"><a href="#3-5-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）" class="headerlink" title="3.5.3. 运行mcmctree mcmctree2.ctl，（mcmctree usedata &#x3D; 2）"></a>3.5.3. 运行<code>mcmctree mcmctree2.ctl</code>，（mcmctree usedata &#x3D; 2）</h3><ul>
<li>把input.phy，input.tre，mcmctree3.ctl，out.BV四个文件复制到新建目录下，mcmctree3.ctl的usedata改为usedata &#x3D; 2并重命名为mcmctree2.ctl，out.BV重命名为in.BV；</li>
<li>运行<code>mcmctree mcmctree2.ctl</code>即可获得结果。</li>
</ul>
<h2 id="3-6-指南四：codons数据处理方式"><a href="#3-6-指南四：codons数据处理方式" class="headerlink" title="3.6. 指南四：codons数据处理方式"></a>3.6. 指南四：codons数据处理方式</h2><p>如果是密码子codons数据，实测设置文件里seqtype &#x3D; 1 会报错 “Error: dN&#x2F;dS ratios among branches not implemented for gamma.”。<br>有两个方案处理codons数据。</p>
<ol>
<li>转化codons成第一、第二、第三密码子的三组数据之后，并设置ndata&#x3D;3与seqtype &#x3D; 0结合用核苷酸模式使用codons数据。</li>
<li>把codons翻译成氨基酸数据，用氨基酸模型分析codons数据。</li>
</ol>
<h2 id="3-7-结果文件"><a href="#3-7-结果文件" class="headerlink" title="3.7. 结果文件"></a>3.7. 结果文件</h2><p>程序在运行过程中，会在屏幕生生成一些信息。比较耗时间的步骤主要在于取样的百分比进度：</p>
<ul>
<li>第一列：取样的百分比进度。</li>
<li>第2-6列：参数的接受比例。一般，其值应该在30%左右。20-40%是很好的结果，15-70%是可以接受的范围。若这些值在开始时变动较大，则表示burnin数设置太小。</li>
<li>第7-x列：各内部节点的平均分歧时间，第7列则是root节点的平均分歧时间。若有y个物种，则总共有y-1个内部节点，从第7列开始后的y-1列对应这些内部节点。</li>
<li>倒数第3-x列：r_left值。若输入3各多序列比对结果，则有3列。x列的前一列是中划线 - 。</li>
<li>倒数第1-2列：likelihood值和时间消耗。</li>
</ul>
<p>屏幕信息最后，给出各个内部节点的分歧时间(t)、平均变异速率(mu)、变异速率方差(sigma2)和r_left的Posterior信息：均值(mean)、95%双侧置信区间(95% Equal-tail CI)和95% HPD置信区间(95% HPD CI)等信息。此外，倒数第二列给出了各个内部节点的Posterior mean time信息，可以用于收敛分析。</p>
<p>在当前目录下，生成几个主要结果：</p>
<ul>
<li><p>FigTree.tre    生成含有枝长（分歧时间）的超度量树文件，分歧时间和分歧时间的95%HPD区间。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#NEXUS</span><br><span class="line">BEGIN TREES;</span><br><span class="line"></span><br><span class="line">        UTREE 1 = (((B: 0.408365, (A1: 0.107333, A2: 0.107333) [&amp;95%=&#123;0.0749432, 0.126365&#125;]: 0.301032) [&amp;95%=&#123;0.285365, 0.478163&#125;]: 0.362000, C: 0.770366) [&amp;95%=&#123;0.53864, 0.90012&#125;]: 0.047786, D: 0.818151) [&amp;95%=&#123;0.572406, 0.958116&#125;];</span><br><span class="line"></span><br><span class="line">END;</span><br></pre></td></tr></table></figure>
<p>FigTree.tre文件的解释：其中A1或A2冒号:后的0.107333为A1和A2的枝长，代表A1和A2分歧时间，单位是亿年，(A1,A2)外的[&amp;95%&#x3D;{0.0749432, 0.126365}]为A1和A2分歧时间的95%置信区间；A1或A2的枝长0.107333加上(A1,A2)之后的祖先枝长0.301032之和等于B的枝长0.408363，即B与A1和A2祖先种的分歧时间为0.408363亿年，95%置信区间为[&amp;95%&#x3D;{0.285365, 0.478163}]；依此类推。</p>
</li>
<li><p>mcmc.txt       MCMC取样信息，包含各内部节点分歧时间、平均进化速率、sigma2值等信息，可以在Tracer软件中打开。通过查看各参数的ESS值，若ESS值大于200，则从一定程度上表示MCMC过程能收敛，结果可靠。</p>
</li>
<li><p>out.txt        包含由较多信息的结果文件。例如，各碱基频率、节点命名信息。</p>
</li>
</ul>
<h2 id="3-8-一些注意事项"><a href="#3-8-一些注意事项" class="headerlink" title="3.8. 一些注意事项"></a>3.8. 一些注意事项</h2><p>以下来自<a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2974">陈连福的教程</a>。</p>
<ol>
<li>如何设置burnin、sampfreq和nsample值？</li>
</ol>
<ul>
<li>一般推荐设置nsample值不小于20k(官方示例数据中设置的值)，只有当该值较大时，求得的均值才有意义。</li>
<li>sampfreq表示取样间隔，一般设置为10，100或1000。nsample 和sampfreq 的乘积表示有效迭代的次数，这些迭代越准确，次数越大，则结果越好，越趋于收敛。当然，次数越大，越消耗时间。若数据量很小，可以考虑设置sampfreq为1000；若数据量大，每次迭代耗时多，则考虑设置sampfre为10。</li>
<li>(burnin)其结果，则设置burnin值。要设置足够大的burnin值，保证在程序运行时当迭代比例达到0% (即burnin结束) 时，其参数的接受比例值较好，在30%左右且随迭代次数增多时变动幅度不大。推荐设置burnin的迭代次数为有效次数的10~40%。PAML软件时先burnin后再记录有效迭代的参数值；其它软件如BEAST则记录所有的参数值后，最后汇总时burnin掉一定比例的数据。</li>
<li>总体上，其实就是两个参数：burnin掉的迭代次数和用于估算结果的有效迭代次数。由于需要根据有效迭代数据来进行均值估算，若记录每次迭代的参数，则生成的mcmc.txt文件行数过多，文件太大，汇总时也消耗估算。于是设置没隔一定迭代轮数取样一次，这样生成的mcmc.txt文件不会太大，对最后的均值估算影响不大。</li>
<li>个人推荐有效迭代次数 1<del>10M 次，burnin 0.2</del>4M次。按时间消耗从少到多的参数值：<br>数据简单时，进行0.5M迭代次数，burnin比例40%：{ burnin 200k; sampfreq 10; nsample 50k }<br>数据中等时，进行1M迭代次数，burnin比例40%：{ burnin 400k; sampfreq 10; nsample 100k }<br>数据复杂时，进行5M迭代次数，burnin比例20%：{ burnin 1000k; sampfreq 10; nsample 500k }<br>数据巨大时，进行10M迭代次数，burnin比例20%：{ burnin 2000k; sampfreq 100; nsample 100k }<br>使用相应参数进行分析后，若不满意其收敛程度，则选用更多迭代次数的参数。</li>
</ul>
<ol start="2">
<li>如何检测MCMC的结果是否达到收敛状态？</li>
</ol>
<p>收敛的意思，即经过很多次迭代后，得到的MCMC树的各枝长趋于一个定值，变动非常小。于是，最直接的检测方法是：分别使用不同的Seed值进行mcmctree或infinitesites进行两次或多次分析，然后比较两个结果树是否一致，实际就是比较树文件中各内部节点的Height值（分歧时间 &#x2F; Posterior time）。比较方法可以有多种: </p>
<ul>
<li>结果文件mcmc.txt中每行记录了一个取样的结果，包含各内部节点的Posterior time值，即进化树的Height值，对该mcmc.txt文件进行分析，得到每个内部节点的Posterior mean time值。然后作图 （官方说明文档第6页图示），横坐标表示第一次运行的各内部节点的分歧时间均值、纵坐标表示第二次运行的各内部节点的分歧时间均值。该散点图要表现出非常明显的对角线，才认为达到收敛。这时可以考虑估算其相关系数来判断是否符合线性。</li>
<li>我觉得第一种官方文档给的方法比较麻烦且是否符合对角线没有明确的定义，于是就直接比较两次结果树文件中的各枝长。估算各枝长总的偏差百分比，当偏差百分比低于0.1%，则认为两次结果非常吻合，差异低于0.1%，认为达到收敛。</li>
<li>此外，还可以使用Tracer分析mcmc.txt文件，检测其ESS值，一般认为该值高于200，则可能达到收敛。该方法可用于辅助检测。最后，若不收敛，则需要提高burnin、nsample值，重新运行程序。</li>
</ul>
<ol start="3">
<li>使用infinitesites输入含有枝长信息的系统进化树快速进行分歧时间估算<br>infinitesites程序进行分歧时间估算时，其假设前提为多序列比对的序列长度为无限长。相比于正常的mcmctree命令，要求额外多输入一个名为 FixedDsClock23.txt 的文件。该文件中存放了相应的带有枝长信息的系统发育树。该文件中系统发育树的拓扑结构要和input.tre一致。推荐使用baseml命令输入input.trees和多序列比对结果估算枝长。</li>
</ol>
<p>FixedDsClock23.txt 文件内容示例：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">7</span><br><span class="line">((((human: 0.029043, (chimpanzee: 0.014557, bonobo: 0.010908): 0.016729): 0.015344, gorilla: 0.033888): 0.033816, (orangutan: 0.026872, sumatran: 0.022437): 0.069648): 0.073309, gibbon: 0.024637);</span><br><span class="line">((((human: 0.012463, (chimpanzee: 0.002782, bonobo: 0.003835): 0.003331): 0.004490, gorilla: 0.014278): 0.006308, (orangutan: 0.010818, sumatran: 0.008845): 0.030551): 0.004363, gibbon: 0.029246);</span><br><span class="line">((((human: 0.270862, (chimpanzee: 0.066698, bonobo: 0.056883): 0.124104): 0.139082, gorilla: 0.310797): 0.391342, (orangutan: 0.152555, sumatran: 0.114176): 0.696518): 0.017607, gibbon: 1.394718);</span><br></pre></td></tr></table></figure>
<p>运行infinitesites命令进行分歧时间估算：<code>infinitesites mcmctree.ctl</code><br>使用infinitesites进行分歧时间估算时，程序要求输入多序列比对文件。虽然程序读取了序列信息，但在估算时会忽略其序列信息。</p>
<h1 id="4-r8s估算系统分歧时间"><a href="#4-r8s估算系统分歧时间" class="headerlink" title="4. r8s估算系统分歧时间"></a>4. r8s估算系统分歧时间</h1><h2 id="4-1-r8s"><a href="#4-1-r8s" class="headerlink" title="4.1. r8s"></a>4.1. r8s</h2><p>r8s 是美国加利福尼亚大学戴维斯分校的进化生物学家 Mike Sanderson 在2003年编写的用于估算进化树分化时间的软件，在进化生物学、分子生物地理学等学科有着广泛的应用，是估算分化时间常用的软件之一。该软件中的一些方法如 NPRS 和 PL 是软件作者最先提出的，目前在同类的其他软件中还难以实现。</p>
<h2 id="4-2-r8s的安装"><a href="#4-2-r8s的安装" class="headerlink" title="4.2. r8s的安装"></a>4.2. r8s的安装</h2><p>wget <a target="_blank" rel="noopener" href="http://loco.biosci.arizona.edu/r8s/r8s.dist.tgz">http://loco.biosci.arizona.edu/r8s/r8s.dist.tgz</a><br>tar zxf r8s.dist.tgz</p>
<p>或者sourceforge下载<a target="_blank" rel="noopener" href="https://sourceforge.net/projects/r8s/">https://sourceforge.net/projects/r8s/</a></p>
<h2 id="4-3-r8s的输入文件"><a href="#4-3-r8s的输入文件" class="headerlink" title="4.3. r8s的输入文件"></a>4.3. r8s的输入文件</h2><h3 id="4-3-1-输入文件r8s-input"><a href="#4-3-1-输入文件r8s-input" class="headerlink" title="4.3.1. 输入文件r8s.input"></a>4.3.1. 输入文件r8s.input</h3><p>一个例子，包括两个模块，一个树模块，一个r8s参数设置模块。<br>树模块可以用raxml-ng等建树软件获得的树，包括枝长信息。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">#nexus</span><br><span class="line">begin trees;</span><br><span class="line">tree tree_1 = (A:0.5069367934,(((((B:0.5881634293,C:0.2730677988)100:0.0325689044,D:0.2516482074)100:0.0150403964,E:0.2672133160)86:0.0239229988,((G:0.4074076834,H:0.4308871688)100:0.0260782585,I:0.2360982873)100:0.0237486276)100:0.0147845808,(J:0.5746285099,K:0.2089809581)92:0.0360035242)100:0.1832776220,L:0.2440847991);</span><br><span class="line">end;</span><br><span class="line"></span><br><span class="line">begin r8s;</span><br><span class="line">blformat lengths=persite nsites=300000 ulrametric=no;</span><br><span class="line">mrca pointa A B;</span><br><span class="line">mrca pointb K L;</span><br><span class="line">fixage taxon=pointa age=520;</span><br><span class="line">constrain taxon=pointb min_age=350 max_age=410;</span><br><span class="line">divtime method=PL algorithm=TN;</span><br><span class="line">set smoothing=100;</span><br><span class="line">showage;</span><br><span class="line">describe plot=chronogram;</span><br><span class="line">describe plot=chrono_description;</span><br><span class="line">end;</span><br></pre></td></tr></table></figure>

<h3 id="4-3-2-r8s参数设置模块"><a href="#4-3-2-r8s参数设置模块" class="headerlink" title="4.3.2. r8s参数设置模块"></a>4.3.2. r8s参数设置模块</h3><ol>
<li>blformat命令：进化树基本信息，需要在计算divergence time&#x2F;rates analysis之前运行。</li>
</ol>
<ul>
<li>lengths选项：total|persite<br>  描述进化树长度的单位。MP简约法获得的进化树，枝长单位是碱基替换个数，lengths选项应该用total；ML极大似然法获得的进化树，枝长单位是期望的碱基替换数，lengths选项应该用persite。</li>
<li>nsites选项：number<br>  在进化分析中用到的碱基位点数量。</li>
<li>ultrametric选项：yes|no<br>  表示该进化树是否经过了校正。如果选yes，直接给出进化时间，不对内部进行任何更改。如果要校对，则应该用calibrate命令。如果不想对ultrametric的进化树的分化时间全部重新估算，应该选yes。</li>
<li>round选项：yes|no<br>  是否对枝长信息四舍五入。只针对nonultrametric树（因为ultrametric树的枝长经过四舍五入会转变成not truly ultrametric树。）。round选项将每个分枝上的替换数转换为整数。如果用户自己提供了已经校对时间的进化树，chronograms，需要设定round&#x3D;no。</li>
</ul>
<ol start="2">
<li><p>mrca命令：为节点定名。<br>mrca(most recent common ancestor)可以命名两个或多个类群的最近共同祖先。<br>例如想要给树 tree test&#x3D;(a,(b,c));中a、b的共同祖先定名为node1，可以通过命令 mrca node1 a b来实现。</p>
</li>
<li><p>fixage命令：设定节点的分化时间。<br>在mrca命令的基础上，设定已知的节点分化时间点，r8s需要至少一个内部节点进行fixage。<br>用法fixage taxon&#x3D;node1 age&#x3D;100;</p>
</li>
<li><p>constrain命令：设定节点的分化时间。<br>在mrca命令的基础上，设定已知的节点分化时间上下限。<br>用法constrain taxon&#x3D;node1 min_age&#x3D;50 max_age&#x3D;200;</p>
</li>
<li><p>divtime命令：分化时间估算<br>依据设定好的分化时间，对未知节点的分化时间进行估算。<br>用法 divtime method&#x3D;LF|NPRS|PL algorithm&#x3D;POWELL|TN|QNEWT</p>
</li>
</ol>
<ul>
<li>method选项（计算方法）：LF|NPRS|PL<br>  分化时间估算的方式。推荐PL罚分似然法：如果各枝长变化速率差异很大，则smoothing值应该取的很小；如果各枝长变化速率差异很小，则smoothing值应该取的很大。</li>
<li>algorithm选项（数学算法）：POWELL|TN|QNEWT<br>  算法，作者推荐TN算法。</li>
<li>crossv选项：基于化石的进化树校正</li>
</ul>
<p>divtime method&#x3D;PL crossv&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;<br>上述命令，是设置 smoothing 的值从 1, 10, 100, 1000 … 1e17, 来计算，最后得到最佳的 smoothing 值。</p>
<p>若使用 fixage 对 2 个节点的分歧时间进行了固定，则可以运行命令：<br>divtime method&#x3D;PL crossv&#x3D;yes fossilfixed&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;</p>
<p>若使用 fixage 对 1 个节点进行分歧时间固定，同时使用 constrain 对 2 个节点进行了约束，则可以运行命令：<br>divtime method&#x3D;PL crossv&#x3D;yes fossilconstrained&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;</p>
<p>得到最优的smoothing值后，再用set设置，然后进行divtime分歧时间和替换速率的计算。</p>
<ol start="6">
<li><p>smoothing<br>divtime使用PL方法，需要设置参数smoothing的值。通过设置多个smoothing的值来进行一些计算，选择最优的值之后用set设置。</p>
</li>
<li><p>showage命令：显示分化时间和分化速率</p>
</li>
<li><p>describe命令：显示进化树及树的说明</p>
</li>
</ol>
<ul>
<li><p>cladogram<br>  得到分支树的图，图上有各个节点的编号，和showage的结果结合观察。</p>
</li>
<li><p>phylogram<br>  得到进化树的图，枝长表示替换数。</p>
</li>
<li><p>chronogram<br>  得到超度量树的图，枝长表示时间。</p>
</li>
<li><p>ratogram<br>  得到树图，枝长表示替换速率。</p>
</li>
<li><p>phylo_description<br>  得到树的ASCII文字结果，枝长表示替换数。</p>
</li>
<li><p>chrono_description<br>  得到树的ASCII文字结果，枝长表示时间。</p>
</li>
<li><p>rato_description<br>  得到树的ASCII文字结果，枝长表示替换速率。</p>
</li>
<li><p>node_info<br>  得到节点的信息表格</p>
</li>
</ul>
<p>做超度量时间树，一般选择<br>describe plot&#x3D;chronogram;<br>describe plot&#x3D;chrono_description;</p>
<h2 id="4-4-r8s运行"><a href="#4-4-r8s运行" class="headerlink" title="4.4. r8s运行"></a>4.4. r8s运行</h2><p><code>r8s -b -f r8s.in &gt; r8s.out</code></p>
<p>-b是batch process the datafile，-f用于指定输入文件</p>
<p>结果保存在r8s.out文件中</p>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><p><a target="_blank" rel="noopener" href="http://abacus.gene.ucl.ac.uk/software/MCMCtree.Tutorials.pdf">mcmctree manual</a>;<br><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2974">陈连福的paml mcmctree教程</a><br><a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/19/2/301/372781">r8s paper</a><br><a target="_blank" rel="noopener" href="https://naturalis.github.io/mebioda/doc/week1/w1d5/r8s1.7.manual.pdf">r8s manual</a><br><a target="_blank" rel="noopener" href="http://image.sciencenet.cn/olddata/kexue.com.cn/upload/blog/file/2010/3/201032420201531842.0.pdf">r8s manual 中文</a><br><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2280">陈连福的r8s教程</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/03/25/bioinfo_caculate.divergence.time/" data-id="cl1aiqcrh002cm4vk9yfmboyf" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-system_reinstall.windows" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2020/12/29/system_reinstall.windows/" class="article-date">
  <time datetime="2020-12-29T13:05:00.000Z" itemprop="datePublished">2020-12-29</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer/">computer</a>►<a class="article-category-link" href="/categories/computer/system/">system</a>►<a class="article-category-link" href="/categories/computer/system/Windows/">Windows</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2020/12/29/system_reinstall.windows/">重装windows系统</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=330 height=86 src="//music.163.com/outchain/player?type=2&id=418550511&auto=1&height=66"></iframe></div>

<h1 id="1-小白就一键重装"><a href="#1-小白就一键重装" class="headerlink" title="1. 小白就一键重装"></a>1. 小白就一键重装</h1><p>自己用过黑鲨装机软件，360装机软件等，都有绑定，需要重装之后卸载绑定的软件。<br>一键重装只能在电脑正常可进入桌面和使用软件的情况下进行，许多情况下不适用。</p>
<h1 id="2-U盘重装"><a href="#2-U盘重装" class="headerlink" title="2. U盘重装"></a>2. U盘重装</h1><p>但强烈不建议一键重装，除非纯小白，建议在正常运行的电脑上制作启动U盘，然后使用U盘启动和安装需要更换系统的电脑：</p>
<h2 id="2-1-【推荐】微软官方工具"><a href="#2-1-【推荐】微软官方工具" class="headerlink" title="2.1. 【推荐】微软官方工具"></a>2.1. 【推荐】微软官方工具</h2><p>可用微软官方的制作工具，<a target="_blank" rel="noopener" href="https://www.microsoft.com/zh-cn/software-download/windows10">微软官方工具</a>，速度比msdn网站快。</p>
<h3 id="2-1-1-创建启动U盘的步骤"><a href="#2-1-1-创建启动U盘的步骤" class="headerlink" title="2.1.1. 创建启动U盘的步骤"></a>2.1.1. 创建启动U盘的步骤</h3><p>进入网站，根据<strong>使用该工具创建安装介质(USB闪存驱动器、DVD或ISO文件），以在其他电脑上安装Windows10</strong>指引操作，用一个&gt;&#x3D;8G的空U盘制作U盘启动盘。</p>
<ol>
<li>插上&gt;&#x3D;8G的空白U盘（U盘会被格式化）</li>
<li>选择<strong>立即下载工具</strong>，选择<strong>运行</strong>（需要管理员权限）</li>
<li>同意许可条款</li>
<li>选择<strong>为另一台电脑创建安装介质</strong>，选择<strong>下一步</strong></li>
<li>选择Window10的语言、版本和体系结构（64&#x2F;32位）</li>
<li>选择介质类型（USB闪存驱动器或ISO文件，其中ISO文件用于创建启动DVD），选择USB闪存驱动器。</li>
<li>等待创建完成</li>
</ol>
<h3 id="2-1-2-U盘启动安装"><a href="#2-1-2-U盘启动安装" class="headerlink" title="2.1.2. U盘启动安装"></a>2.1.2. U盘启动安装</h3><p>根据网站上接下来的<strong>使用您所创建的安装介质</strong>，为电脑安装新的Window10系统，步骤如下</p>
<ol>
<li>需要安装系统的电脑的所有硬盘需要备份。</li>
<li>把制作好的启动U盘连接到等待安装系统的电脑上</li>
<li>重启电脑</li>
<li>有时电脑会自动引导至USB介质启动安装；如果电脑没有自动引导至 USB 或 DVD 介质，可能需要打开引导菜单或在电脑 BIOS 或 UEFI 设置中更改引导顺序。</li>
<li>若要打开引导菜单或更改引导顺序，需要在重启&#x2F;开机后，在出现Dell&#x2F;Lenovo等商家界面后立即按下按键（例如 F2、F12、Delete 或 ESC）进入BIOS界面（不同品牌电脑不一样，可查询各厂家进入BIOS界面的快捷键），选择U盘对应名称，一步步按照指导进行系统安装。如果没有看到 USB 或 DVD 介质设备在引导选项中列出，可能需要联系电脑制造商来获取在 BIOS 设置中暂时禁用“安全引导”的说明。</li>
<li>在<strong>安装Windows</strong>页面上，选择语言、时间和键盘首选项，然后选择<strong>下一步</strong>；</li>
<li>选择<strong>安装Windows</strong>。</li>
</ol>
<h2 id="2-2-msdn下载系统镜像-rufus制作启动U盘"><a href="#2-2-msdn下载系统镜像-rufus制作启动U盘" class="headerlink" title="2.2. msdn下载系统镜像+rufus制作启动U盘"></a>2.2. msdn下载系统镜像+rufus制作启动U盘</h2><p>这里可以替代上一步中的<strong>创建启动U盘的步骤</strong>；制作完成后<strong>U盘启动安装</strong>与上一步一致。</p>
<p>在<a target="_blank" rel="noopener" href="https://msdn.itellyou.cn/">msdn网站</a>下载需要的操作系统版本的ISO文件；然后用<a target="_blank" rel="noopener" href="https://rufus.ie/">Rufus</a>软件制作U盘。</p>
<p>下好系统镜像和rufus软件后，插上U盘在rufus软件上依次操作制作启动U盘。</p>
<p>几个需要注意的选项：</p>
<ul>
<li>目标系统类型选择UEFI启动(Legacy BIOS启动时间更长)，分区类型选择GPT(比MBR好在可以识别和使用2T以上磁盘)，Legacy BIOS+MBR组合较旧，不推荐。</li>
<li>用软件Rufus制作U盘启动盘（大约30-60分钟），制作完成后插入需装系统的电脑（台式机建议插后面的USB接口）</li>
</ul>
<h2 id="2-3-notes："><a href="#2-3-notes：" class="headerlink" title="2.3. notes："></a>2.3. notes：</h2><ul>
<li>不同电脑主板进入BIOS界面的快捷键不一样，需要查询。</li>
<li>进入BIOS界面后，有些主板不显示U盘，可以UEFI设置的“Secure Boot”选项disable掉。</li>
<li>有些主板不支持U盘是NTFS文件格式使用UEFI启动，但FAT32最大支持4G文件，所以网上有建议U盘分区，把系统文件放在NTFS区，启动文件放在FAT32区，但我亲试用Rufus制作的NTFS格式U盘可以在Dell inspire 3881上使用（UltraISO制作的就不行）。</li>
<li>如果选择U盘名称后键盘鼠标不工作，可以换个USB接口试试，可能系统内的USB驱动版本不匹配。</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/12/29/system_reinstall.windows/" data-id="cl1aiqcs2004im4vk7dvq95d8" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/windows/" rel="tag">windows</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-web_react" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2018/06/14/web_react/" class="article-date">
  <time datetime="2018-06-14T06:53:00.000Z" itemprop="datePublished">2018-06-14</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer/">computer</a>►<a class="article-category-link" href="/categories/computer/web/">web</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2018/06/14/web_react/">react</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <h1 id="1-react"><a href="#1-react" class="headerlink" title="1. react"></a>1. react</h1><h2 id="1-1-JSX"><a href="#1-1-JSX" class="headerlink" title="1.1. JSX"></a>1.1. JSX</h2><p>react使用JSX代替javascript，JSX是有点像XML的javascript语法扩展。</p>
<h3 id="1-1-1-基本语法"><a href="#1-1-1-基本语法" class="headerlink" title="1.1.1. 基本语法"></a>1.1.1. 基本语法</h3><p>ReactDOM.render(<br>  div<br>    h1 react教程 &#x2F;h1<br>    h2 欢迎学习 React &#x2F;h2<br>    p 这是一个很不错的 JavaScript 库! &#x2F;p<br>  &#x2F;div<br>document.getElementById(‘example’))</p>
<p>多个html标签需要全包含在一个div中，以上代码表示在id&#x3D;example的html标签位置渲染一个div，里面有两个标题和一个段落。</p>
<h3 id="1-1-2-嵌入javascript表达式"><a href="#1-1-2-嵌入javascript表达式" class="headerlink" title="1.1.2. 嵌入javascript表达式"></a>1.1.2. 嵌入javascript表达式</h3><ul>
<li>JSX中可以嵌入javascript表达式，但需要写在花括号{}内。</li>
<li>注释也需要写在花括号中。</li>
<li>花括号中的数组会自动展开所有成员。</li>
<li>表达式不能使用<code>if else</code>语句，但可以用<code>conditional</code>三元运算表达式。<br><code>&lt;h1&gt;&#123;i == 1 ? &#39;True!&#39; : &#39;False&#39;&#125;&lt;/h1&gt;</code></li>
</ul>
<h3 id="1-1-3-内联样式"><a href="#1-1-3-内联样式" class="headerlink" title="1.1.3. 内联样式"></a>1.1.3. 内联样式</h3><p>使用驼峰法设置内联样式</p>
<p>ReactDOM.render(<br>  h1 style &#x3D; {myStyle} 菜鸟教程 &#x2F;h1<br>  document.getElementById(‘example’))</p>
<h3 id="1-1-4-渲染HTML标签（strings）和React组件（classes）"><a href="#1-1-4-渲染HTML标签（strings）和React组件（classes）" class="headerlink" title="1.1.4. 渲染HTML标签（strings）和React组件（classes）"></a>1.1.4. 渲染HTML标签（strings）和React组件（classes）</h3><p>React 可以渲染 HTML 标签 (strings) 或 React 组件 (classes)。</p>
<p>要渲染 HTML 标签，只需在 JSX 里使用小写字母的标签名。</p>
<p>要渲染 React 组件，只需创建一个大写字母开头的本地变量。</p>
<h3 id="1-1-5-tips"><a href="#1-1-5-tips" class="headerlink" title="1.1.5. tips"></a>1.1.5. tips</h3><p>由于 JSX 就是 JavaScript，一些标识符像 class 和 for 不建议作为 XML 属性名。作为替代，React DOM 使用 className 和 htmlFor 来做对应的属性。</p>
<h2 id="1-2-组件"><a href="#1-2-组件" class="headerlink" title="1.2. 组件"></a>1.2. 组件</h2><p>组件API的7个方法：</p>
<ul>
<li>设置状态：setState</li>
<li>替换状态：replaceState</li>
<li>设置属性：setProps</li>
<li>替换属性：replaceProps</li>
<li>强制更新：forceUpdate</li>
<li>获取DOM节点：findDOMNode</li>
<li>判断组件挂载状态：isMounted</li>
</ul>
<p>react元素是DOM标签或者用户自定义组件，当是自定义组件时，会将JSX属性作为单个对象传递给该组件，这个对象是“props”。<br>组件名称必须以大写字母开头。<br>组件的返回值只能有一个根元素。<br>所有的React组件必须像纯函数（即不能改变输入值的函数）那样使用它们的props。<br>React应用中，按钮、表单、对话框、整个屏幕的内容等，这些通常都被表示为组件。</p>
<h2 id="1-3-生命周期"><a href="#1-3-生命周期" class="headerlink" title="1.3. 生命周期"></a>1.3. 生命周期</h2><ol>
<li>组件的生命周期可分成三个状态：</li>
</ol>
<ul>
<li>Mounting：已插入真实 DOM</li>
<li>Updating：正在被重新渲染</li>
<li>Unmounting：已移出真实 DOM</li>
</ul>
<ol start="2">
<li>生命周期的方法有：</li>
</ol>
<ul>
<li>componentWillMount 在渲染前调用,在客户端也在服务端。</li>
<li>componentDidMount : 在第一次渲染后调用，只在客户端。之后组件已经生成了对应的DOM结构，可以通过this.getDOMNode()来进行访问。 如果你想和其他JavaScript框架一起使用，可以在这个方法中调用setTimeout, setInterval或者发送AJAX请求等操作(防止异部操作阻塞UI)。</li>
<li>componentWillReceiveProps 在组件接收到一个新的 prop (更新后)时被调用。这个方法在初始化render时不会被调用。</li>
<li>shouldComponentUpdate 返回一个布尔值。在组件接收到新的props或者state时被调用。在初始化时或者使用forceUpdate时不被调用。</li>
<li>可以在你确认不需要更新组件时使用。</li>
<li>componentWillUpdate在组件接收到新的props或者state但还没有render时被调用。在初始化时不会被调用。</li>
<li>componentDidUpdate 在组件完成更新后立即调用。在初始化时不会被调用。</li>
<li>componentWillUnmount在组件从 DOM 中移除的时候立刻被调用。</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2018/06/14/web_react/" data-id="cl1aiqcsa005dm4vk3yfl6e4f" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/react/" rel="tag">react</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-web_ES6" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2018/06/12/web_ES6/" class="article-date">
  <time datetime="2018-06-12T06:53:00.000Z" itemprop="datePublished">2018-06-12</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer/">computer</a>►<a class="article-category-link" href="/categories/computer/web/">web</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2018/06/12/web_ES6/">ES6</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <h2 id="常量"><a href="#常量" class="headerlink" title="常量"></a>常量</h2><p>const pi &#x3D; 3.1415926<br>常量是只读的，不可修改。</p>
<h2 id="作用域"><a href="#作用域" class="headerlink" title="作用域"></a>作用域</h2><p>ES6中有块级作用域<br>const、let，在每次循环内都有自己的闭包和作用域。<br>{}内就是块级作用域。</p>
<h2 id="箭头函数"><a href="#箭头函数" class="headerlink" title="箭头函数"></a>箭头函数</h2><ol>
<li>语法：(参数)&#x3D;&gt;{函数体}<br>当只有一个参数时，可省略()；当函数体只有return something时，可以省略{}。eg：i&#x3D;&gt;i+1</li>
<li>箭头函数的this是定义该名称时this的指向</li>
</ol>
<h2 id="默认参数"><a href="#默认参数" class="headerlink" title="默认参数"></a>默认参数</h2><p>function func(x&#x3D;1,y&#x3D;5,z&#x3D;20){<br>return x+y+z<br>}</p>
<h3 id="可变参数"><a href="#可变参数" class="headerlink" title="可变参数"></a>可变参数</h3><p>不知数量参数的求和<br>function func(…a){<br>a.forEach(item&#x3D;&gt;{<br>sum+&#x3D;item*1<br>});<br>return sum<br>}</p>
<h3 id="利用扩展运算符合并数组"><a href="#利用扩展运算符合并数组" class="headerlink" title="利用扩展运算符合并数组"></a>利用扩展运算符合并数组</h3><p>var params &#x3D; [‘hello’,true,1,23]<br>var concatlist &#x3D; [1,2,’world’,…params]<br>相当于concatlist为[1,2,’world’,’hello’,true,1,23]</p>
<h2 id="对象代理"><a href="#对象代理" class="headerlink" title="对象代理"></a>对象代理</h2><p>私有属性<br>let Person &#x3D; {<br>name: ‘karen’,<br>sex: ‘female’,<br>age: 18<br>}</p>
<p>let person &#x3D; new Proxy(Person,{<br>get(target,key){<br>return target[key]<br>},<br>set(target,key,value){<br>if(key!&#x3D;&#x3D;’sex’){<br>target[key]&#x3D;value;<br>}<br>}<br>})<br>通过Proxy对象代理实现保护Person的sex属性，使其不能修改。</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2018/06/12/web_ES6/" data-id="cl1aiqcs4004sm4vkb79t9hpw" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li></ul>

    </footer>
  </div>
  
</article>


  


  <nav id="page-nav">
    
    <a class="extend prev" rel="prev" href="/page/3/">&amp;laquo; 上一页</a><a class="page-number" href="/">1</a><a class="page-number" href="/page/2/">2</a><a class="page-number" href="/page/3/">3</a><span class="page-number current">4</span><a class="page-number" href="/page/5/">5</a><a class="extend next" rel="next" href="/page/5/">下一页 &amp;raquo;</a>
  </nav>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>