<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>估算系统树分歧时间（paml.mcmctree,r8s) | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="用paml的mcmctree模块估算系统树上节点分歧时间的教程。">
<meta property="og:type" content="article">
<meta property="og:title" content="估算系统树分歧时间（paml.mcmctree,r8s)">
<meta property="og:url" content="http://example.com/2021/03/25/bioinfo_caculate.divergence.time/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="用paml的mcmctree模块估算系统树上节点分歧时间的教程。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_caculate.divergence.time_phylogeny.jpg?raw=true">
<meta property="article:published_time" content="2021-03-25T08:50:00.000Z">
<meta property="article:modified_time" content="2022-03-28T02:33:51.718Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="tutorial">
<meta property="article:tag" content="biosoft">
<meta property="article:tag" content="phylogeny">
<meta property="article:tag" content="paml">
<meta property="article:tag" content="divergence time">
<meta property="article:tag" content="mcmctree">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_caculate.divergence.time_phylogeny.jpg?raw=true">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-bioinfo_caculate.divergence.time" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/03/25/bioinfo_caculate.divergence.time/" class="article-date">
  <time datetime="2021-03-25T08:50:00.000Z" itemprop="datePublished">2021-03-25</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      估算系统树分歧时间（paml.mcmctree,r8s)
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=330 height=86 src="//music.163.com/outchain/player?type=2&id=27853347&auto=1&height=66"></iframe></div>

<p>估算分歧时间是在已经建好物种的系统发育树，根据系统发育树上已知类群间分化时间（一般来自化石或者其他研究的标定），或者根据权威研究中对特定类群的碱基替换速率的估计（例如蕨类4.79e-9 subst.&#x2F;syn. site&#x2F;year，水稻2.5e-8 subst.&#x2F;syn. site&#x2F;year)进行估算系统发育树上其他类群的分歧时间。</p>
<p>一般来说，我们尽量使用化石证据的时间（更可靠），可以在网站<a target="_blank" rel="noopener" href="http://www.timetree.org/">timetree</a>搜索化石或者其他文章标定的类群分化时间；实在没有化石可用时，则使用碱基替换速率常数。</p>
<h1 id="1-原理"><a href="#1-原理" class="headerlink" title="1. 原理"></a>1. 原理</h1><p>尝试用数学思维讲讲估算分歧时间的原理：即路程&#x3D;速度*时间，经过多长<strong>时间</strong>（地质时间），通过多大的<strong>速度</strong>（碱基替换速率），完成从祖先物种的DNA到现在物种的DNA的变化，总体上的碱基替换积累即<strong>路程</strong>。</p>
<p>假设下面Figure 1. 系统发育树示意图(A4,(A3,(A2,A1)B)C)D;是有枝长有根树，A1-A4都是现在的物种（代表时间也已知），目的是要估算B和C节点的时间。</p>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/bioinfo_caculate.divergence.time_phylogeny.jpg?raw=true" width=100% height=80% title="phylogeny示意图(A4,(A3,(A2,A1)B)C)D" align=center/>

<p><strong><p align="center">Figure 1. 系统发育示意图</p></strong></p>
<ul>
<li>第一种情况，当已知D节点时间时，求B节点时间（C节点同理）</li>
</ul>
<p>由于A1的时间点已知（即现在），那么从D到A1的时间跨度是已知的；从D到A1的枝长是已知的，枝长代表的是这个过程的碱基替换数量，即路程；时间和路程已知，我们可以算出速度；同时，从D到B的路程也是已知的（因为枝长已知），那么在假设匀速的前提下，我们也可以算出从D到B的时间，这样即可标定B（即A1和A2分歧时间）的时间。</p>
<ul>
<li>第二种情况，当已知速度（碱基替换速率，常数）时，求B节点时间</li>
</ul>
<p>由于从B到A1的枝长是已知的，枝长代表的是这个过程的碱基替换数量，即路程；速度已知，那么我们可以算得B到A1的时间跨度，由于A1的时间即现在，时间跨度就是B节点的时间。</p>
<p>以上解释是非常简化的理解这个估算过程，实际情况复杂得多，物种不会匀速进行碱基替换，所以软件会使用算法进行精细化估算。</p>
<h1 id="2-估算分歧时间常用软件"><a href="#2-估算分歧时间常用软件" class="headerlink" title="2. 估算分歧时间常用软件"></a>2. 估算分歧时间常用软件</h1><ul>
<li>beast：Linux也调用图形界面</li>
<li>paml的mcmctree：推荐</li>
<li>r8s：非常快</li>
</ul>
<h1 id="3-paml的mcmctree估算分歧时间"><a href="#3-paml的mcmctree估算分歧时间" class="headerlink" title="3. paml的mcmctree估算分歧时间"></a>3. paml的mcmctree估算分歧时间</h1><h2 id="3-1-paml介绍"><a href="#3-1-paml介绍" class="headerlink" title="3.1. paml介绍"></a>3.1. paml介绍</h2><p>PAML (Phylogenetic Analysis by Maximum Likelihood)，是杨子恒开发的一款利用DNA或Protein数据使用最大似然法进行系统发育分析的软件。</p>
<p>注意事项：</p>
<ol>
<li>PAML要求的输入文件是多序列比对结果。PAML软件不能进行多序列比对，需要使用其他软件进行分析。</li>
<li>PAML软件进行基于密码子的分析要求输入的序列必须是去除了Introns等非编码区的DNA序列，且其长度是3的整数倍，且第一个核酸位点是密码子的第一位。PAML软件不能进行基因预测，密码子的比对结果需要自行编写程序分析得到。</li>
<li>对于较大数据（物种数&gt;&#x3D;10）量的分析，输入的树文件最好使用其他软件计算。</li>
</ol>
<h2 id="3-2-paml软件包含的程序"><a href="#3-2-paml软件包含的程序" class="headerlink" title="3.2. paml软件包含的程序"></a>3.2. paml软件包含的程序</h2><ul>
<li>baseml 用于对核酸序列进行最大似然法分析。</li>
<li>codeml 用于对密码子和蛋白序列进行最大似然法分析。该程序整合了以前的两支程- 序codonml和aaml，前者用于对密码子序列进行分析，后者对蛋白序列进行分析。在新的codeml程序的配置文件中设置seqtype&#x3D;1，表示使用codonml命令，设置- seqtype&#x3D;2，表示使用aaml命令。</li>
<li>evolver 用于模拟核酸、密码子或蛋白序列。</li>
<li>basemlg 相比于baseml，该程序应用了连续的gamma模型。当数据中的物种数多于6- 到7个时，该程序运行非常慢以至得不到结果。这时，需要使用设置离散gamma模型的- baseml命令。</li>
<li>mcmctree 该程序应用Bayesian MCMC算法计算物种分歧时间。</li>
<li>pamp  用于简约法分析。</li>
<li>yn00  用于对两条编码蛋白的DNA序列进行比较并计算dnds。</li>
<li>chi2  用于在likelihood ratio test中计算卡方临界值和p值。</li>
</ul>
<p>mcmctree通过调用baseml(核苷酸数据)、codeml（密码子或者氨基酸数据）估算模型参数，之后估算分歧时间；</p>
<p><strong>指南一和指南二只适用于核苷酸数据；氨基酸数据看指南三</strong></p>
<h2 id="3-3-指南一：paml常规方法估算分歧时间"><a href="#3-3-指南一：paml常规方法估算分歧时间" class="headerlink" title="3.3. 指南一：paml常规方法估算分歧时间"></a>3.3. 指南一：paml常规方法估算分歧时间</h2><p>paml常规方法只适用于核苷酸数据；</p>
<h3 id="3-3-1-input-files"><a href="#3-3-1-input-files" class="headerlink" title="3.3.1. input files"></a>3.3.1. input files</h3><p>准备3个输入文件</p>
<h4 id="3-3-1-1-input-tre-带有校准点的有根树文件"><a href="#3-3-1-1-input-tre-带有校准点的有根树文件" class="headerlink" title="3.3.1.1. input.tre - 带有校准点的有根树文件"></a>3.3.1.1. input.tre - 带有校准点的有根树文件</h4><p>把iqtree&#x2F;raxml-ng等建树软件中得到的树文件中的支持率和枝长信息删除，添加化石校准点时间信息（格式是时间范围’&gt;0.23&lt;0.26’或者时间点‘@0.245’)，单位时百万年前100Ma；再在首行添加两个数字（物种数量和树的数量），空格隔开，可得到input.tre文件。</p>
<figure class="highlight plaintext"><figcaption><span>#cat input.tre</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">5 1</span><br><span class="line">((((A,B),C),D)&#x27;&gt;0.23&lt;0.26&#x27;,E);</span><br></pre></td></tr></table></figure>

<ul>
<li>文件内容包含两行：第一行表述树中有n个物种，共计1个树，两个数值之间用空分割；第二行则是Newick格式树信息。</li>
<li>其中第二行包含有校准点信息。校准点信息一般指95%HPD（Highest Posterior Density）对应的置信区间；校准点单位是100MYA（软件说明文档中使用该单位，也推荐使用该单位，若使用其它单位，后续配置文件中的相关参数也需要对应修改）。</li>
<li>此外，Newick格式的树尾部一定要有分号，没有的话程序可能不能正常运行。</li>
</ul>
<h4 id="3-3-1-2-input-phy-多序列比对文件（phylip格式）"><a href="#3-3-1-2-input-phy-多序列比对文件（phylip格式）" class="headerlink" title="3.3.1.2. input.phy - 多序列比对文件（phylip格式）"></a>3.3.1.2. input.phy - 多序列比对文件（phylip格式）</h4><ul>
<li>PAML要求输入的Phylip格式，其物种名和后面的序列之间至少间隔两个空格（是为了允许物种名的属名和种名之间有一个空格）。</li>
<li>python的SeqIO的转换格式模块获得的phylip和nexus格式都不行。</li>
<li>推荐用<code>echo $(cat input.align.fa) |sed &quot;s/ &gt;/\n/g&quot; |sed &quot;s/&gt;//g&quot;|sed &quot;s/ /  /g&quot; &gt;input.phy</code>手动转化align好的fas格式文件；再在首行添加两个数值，空格隔开，物种数量和碱基数量；实现fasta2phylip。</li>
</ul>
<p>如果有<strong>多个区域的序列</strong>，比如exon和intron，LSC、SSC和IR，不同的基因，密码子的第一二三位，需要不同的模型分开估算，那可以把各自区域分别align之后制作多个phy文件，再合并到一起，用空行隔开，组成input.phy文件。（此时mcmctree.ctl的ndata值为区域的个数）。</p>
<p>写了一个脚本来制作多区域的phylip文件，其中OG.length包含两列，一列OGID，一列比对后序列长度。</p>
<figure class="highlight shell"><figcaption><span>## cat fa2phylip.sh</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">cat ./OG.length | while read line</span><br><span class="line">do</span><br><span class="line">        sample_id=$(echo $line |awk &#x27;&#123;print $1&#125;&#x27;) #获取OG.ID</span><br><span class="line">        sample_a=$(echo $line |awk &#x27;&#123;print $2&#125;&#x27;) #获取序列长度</span><br><span class="line">        sed &quot;s/_.*//g&quot; ../singlegenetree/$&#123;sample_id&#125;/$&#123;sample_id&#125;.mafft.pep|seqkit seq -w 0|sed -E &quot;:a;N;s/\n/ /g;ta&quot; |sed &quot;s/ &gt;/\n/g&quot; |sed &quot;s/&gt;//g&quot;|sed &quot;s/ /  /g&quot;|sed &quot;1i\5  $&#123;sample_a&#125;&quot; |sed &#x27;1i\ &#x27; &gt; ./phy/$&#123;sample_id&#125;.phy #把上一步获取的$&#123;sample_id&#125;.mafft.pep改为phylip格式，并在首行添加空格行（为了合并后每个OG用空行隔开）</span><br><span class="line"></span><br><span class="line">done</span><br><span class="line">cat ./phy/*phy &gt;input.phy #合并所有phy文件为一个Ane.phy</span><br></pre></td></tr></table></figure>

<h4 id="3-3-1-3-mcmctree-ctl-mcmctree程序的配置文件"><a href="#3-3-1-3-mcmctree-ctl-mcmctree程序的配置文件" class="headerlink" title="3.3.1.3. mcmctree.ctl - mcmctree程序的配置文件"></a>3.3.1.3. mcmctree.ctl - mcmctree程序的配置文件</h4><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br></pre></td><td class="code"><pre><span class="line">          seed = -1 *设置随机数作为seed，-1代表使用系统当前时间作为随机数</span><br><span class="line">       seqfile = input.phy *输入多序列比对文件</span><br><span class="line">      treefile = input.tre *带校准点（化石时间）的有根树文件</span><br><span class="line">       outfile = mcmc.out *输出文件</span><br><span class="line">      mcmcfile = mcmc.txt *输出的mcmc信息文件，可用Tracer软件查看</span><br><span class="line"></span><br><span class="line">       seqtype = 0  * 设置多序列比对数据类型；0：核酸数据；1：密码子比对数据；2：氨基酸数据；</span><br><span class="line">	   usedata = 1</span><br><span class="line">     * 是否利用多序列比对数据；</span><br><span class="line">	   * 0: no data不使用，不会进行likelihood估算，会快速得到mcmc树，但分歧时间不可用; </span><br><span class="line">	   * 1:seq like，使用多序列比对数据进行likelihood估算，正常进行mcmc; usedata=1时model无法选择；</span><br><span class="line">	   * 2:normal 进行正常的approximation likelihood分析，不读取多序列比对数据，直接读取当前目录的in.BV文件，in.BV是由usedata = 3时生成的out.BV重命名得来；此外，由于程序BUG，当设置usedata = 2时，一定要在改行参数后加 *，否则程序报错 Error: file name empty..；</span><br><span class="line">	   * 3：程序利用多序列比对数据调用baseml/codeml命令对数据进行分析，生成out.BV文件。由于mcmctree调用baseml/codeml进行估算的参数设置可能不太好（特别时对蛋白序列进行估算时），推荐自己修改软件自动生成的baseml/codeml配置文件，然后再手动运行baseml/codeml命令，再整合其结果文件为out.BV文件。</span><br><span class="line"></span><br><span class="line">		 ndata = 1    * 输入的多序列比对的数据区域的数量；</span><br><span class="line">         clock = 2    * 设置分子钟算法，1: global clock，表示所有分支进化速率一致; 2: independent rates，各分支的进化速率独立且进化速率的对数log(r)符合正态分布; 3，correlated rates方法，和方法2类似，但是log(r)的方差和时间t相关。</span><br><span class="line">*       TipDate = 1 100  *当外部节点由取样时间时使用该参数进行设置，同时该参数也设置了时间单位。具体数据示例请见examples/TipData文件夹。</span><br><span class="line">        RootAge = &#x27;&lt;10&#x27;  * constraint on root age, used if no fossil for root.设置root节点的分歧时间，一般设置一个最大值。</span><br><span class="line"></span><br><span class="line">         model = 4    * models for DNA:</span><br><span class="line">                        * 0:JC69, 1:K80, 2:F81, 3:F84, 4:HKY85；*设置碱基替换模型；当设置usedata = 1时，model不能使用超过4的模型，所以usedata = 1时用model = 4；usedata不等于1时，用model = 7，即GTR模型；</span><br><span class="line">                      * models for codons:</span><br><span class="line">                        * 0:one 恒定速率模型, 1:b 中性模型, 2:2 or more dN/dS ratios for branches 选择模型。</span><br><span class="line">                      * models for AAs or codon-translated AAs:</span><br><span class="line">                        * 0:poisson, 1:proportional, 2:Empirical, 3:Empirical+F</span><br><span class="line">                        * 6:FromCodon, 7:AAClasses, 8:REVaa_0, 9:REVaa(nr=189)</span><br><span class="line">         alpha = 0.5   * alpha for gamma rates at sites；*核酸序列中不同位点，其进化速率不一致，其变异速率服从GAMMA分布。一般设置GAMMA分布的alpha值为0.5。若该参数值设置为0，则表示所有位点的进化速率一致。此外，当userdata = 2时，alpha、ncatG、alpha_gamma、kappa_gamma这4个GAMMA参数无效。因为userdata = 2时，不会利用到多序列比对的数据。</span><br><span class="line">         ncatG = 5    * No. categories in discrete gamma；设置离散型GAMMA分布的categories值。</span><br><span class="line"></span><br><span class="line">     cleandata = 0    * remove sites with ambiguity data (1:yes, 0:no)?</span><br><span class="line"></span><br><span class="line">       BDparas = 1 1 0.1   * birth, death, sampling；*设置出生率、死亡率和取样比例。若输入有根树文件中的时间单位发生改变，则需要相应修改出生率和死亡率的值。例如，时间单位由100Myr变换为1Myr，则要设置成&quot;.01 .01 0.1&quot;。</span><br><span class="line">   kappa_gamma = 6 2      * gamma prior for kappa；设置kappa（转换/颠换比率）的GAMMA分布参数。</span><br><span class="line">   alpha_gamma = 1 1      * gamma prior for alpha；设置GAMMA形状参数alpha的GAMMA分布参数。</span><br><span class="line"></span><br><span class="line">   rgene_gamma = 2 20 1   * gamma prior for rate for genes；设置序列中所所有位点平均[碱基/密码子/氨基酸]替换率的Dirichlet-GAMMA分布参数：alpha=2、beta=20、初始平均替换率为每100million年（取决于输入有根树文件中的时间单位）1个替换。若时间单位由100Myr变换为1Myr，则要设置成&quot;2 2000 1&quot;。总体上的平均进化速率为：2 / 20 = 0.1 个替换 / 每100Myr，即每个位点每年的替换数为 1e-9。</span><br><span class="line">  sigma2_gamma = 1 10 1    * gamma prior for sigma^2     (for clock=2 or 3)；设置所有位点进化速率取对数后方差（sigma的平方）的Dirichlet-GAMMA分布参数：alpha=1、beta=10、初始方差值为1。当clock参数值为1时，表示使用全局的进化速率，各分枝的进化速率没有差异，即方差为0，该参数无效；当clock参数值为2时，若修改了时间单位，该参数值不需要改变；当clock参数值为3时，若修改了时间单位，该参数值需要改变。</span><br><span class="line"></span><br><span class="line">      finetune = 1: .1 .1 .1 .1 .1 .1  * times, rates, mixing, paras, RateParas；冒号前的值设置是否自动进行finetune，一般设置成1，然程序自动进行优化分析；冒号后面设置各个参数的步进值：times, musigma2, rates, mixing, paras, FossilErr。由于有了自动设置，该参数不像以前版本那么重要了，可能以后会取消该参数。</span><br><span class="line"></span><br><span class="line">         print = 1  *设置打印mcmc的取样信息：0，不打印mcmc结果；1，打印除了分支进化速率的其它信息（各内部节点分歧时间、平均进化速率、sigma2值）；2，打印所有信息。 </span><br><span class="line">        burnin = 1000000  *将前1000000次迭代burnin后，再进行取样（即打印出该次迭代估算的结果信息，各内部节点分歧时间、平均进化速率、sigma2值和各分支进化速率等）。</span><br><span class="line">      sampfreq = 10  *每10次迭代则取样一次</span><br><span class="line">       nsample = 500000  *当取样次数达到该次数时，则取样结束，同时结束程序。</span><br><span class="line"></span><br><span class="line">*** Note: Make your window wider (100 columns) when running this program.</span><br></pre></td></tr></table></figure>

<h3 id="3-3-2-mcmctree运行"><a href="#3-3-2-mcmctree运行" class="headerlink" title="3.3.2. mcmctree运行"></a>3.3.2. mcmctree运行</h3><p>运行<code>mcmctree mcmctree.ctl</code>即可获得结果。</p>
<h2 id="3-4-指南二：用approximate-likelihood-calculation估算分歧时间【推荐】"><a href="#3-4-指南二：用approximate-likelihood-calculation估算分歧时间【推荐】" class="headerlink" title="3.4. 指南二：用approximate likelihood calculation估算分歧时间【推荐】"></a>3.4. 指南二：用approximate likelihood calculation估算分歧时间【推荐】</h2><p>approximate likelihood法比常规方法快很多，而且可以选择更复杂的GTR模型（model &#x3D; 7）；推荐使用。</p>
<h3 id="3-4-1-input-files"><a href="#3-4-1-input-files" class="headerlink" title="3.4.1. input files"></a>3.4.1. input files</h3><p>输入文件和指南一一致，只是mcmctree.ctl的usedata需要修改为，usedata &#x3D; 3【一定要改】；建议model修改为model &#x3D; 7【建议改】。</p>
<h3 id="3-4-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）"><a href="#3-4-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）" class="headerlink" title="3.4.2. 运行mcmctree mcmctree3.ctl，（mcmctree usedata &#x3D; 3）"></a>3.4.2. 运行<code>mcmctree mcmctree3.ctl</code>，（mcmctree usedata &#x3D; 3）</h3><p>运行<code>mcmctree mcmctree3.ctl</code>；用Maximum Likelihood方法估算枝长和Hessian信息；</p>
<ol>
<li>若比对数据只有一个区域，即ndata &#x3D; 1，则等待运行完成，生成out.BV；</li>
<li>若比对数据多于一个区域，即ndata &gt; 1，mcmctree会多次调用baseml依次处理每个区域，可等待运行完成，也可以做手动并行运行，以加快分析速度；</li>
<li>手动并行多个区域的操作：</li>
</ol>
<ul>
<li>mcmctree运行起来后，从第一区域开始，生成一套tmp0001文件(包括tmp0001.ctl,tmp0001.txt,tmp0001.trees），ctrl+C切断程序，会自动进入下一区域分析，生成一套tmp0002文件，继续ctrl+C切断程序，直到最后一个区域的tmp文件也生成。</li>
<li>对每个区域的tmp文件分别放置一个单独的目录，并在单独目录内运行<code>baseml tmp*.ctl</code>，从而实现多区域并行运行。</li>
<li>所有区域的baseml运行结束后，会在各自目录生成rst2文件，把所有目录的rst2文件cat合并，就是out.BV文件；</li>
<li>或者用命令<code>for i in $(ls tmp*.ctl);do baseml $&#123;i&#125; ; mv rst2 $&#123;i&#125;.rst2; done</code>调用codeml进行多次分析（每次codeml都生成rst2文件，所以每次生成后把rst2改名，否则rst2被覆盖）；然后<code>cat *rst2 &gt;out.BV</code>合并所有rst2文件到out.BV文件；</li>
</ul>
<h3 id="3-4-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）"><a href="#3-4-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）" class="headerlink" title="3.4.3. 运行mcmctree mcmctree2.ctl，（mcmctree usedata &#x3D; 2）"></a>3.4.3. 运行<code>mcmctree mcmctree2.ctl</code>，（mcmctree usedata &#x3D; 2）</h3><ul>
<li>把input.phy，input.tre，mcmctree3.ctl，out.BV四个文件复制到新建目录下，mcmctree3.ctl的usedata改为usedata &#x3D; 2并重命名为mcmctree2.ctl，out.BV重命名为in.BV；</li>
<li>运行<code>mcmctree mcmctree2.ctl</code>即可获得结果。</li>
</ul>
<h2 id="3-5-指南三：用approximate-likelihood-calculation估算氨基酸数据的分歧时间"><a href="#3-5-指南三：用approximate-likelihood-calculation估算氨基酸数据的分歧时间" class="headerlink" title="3.5. 指南三：用approximate likelihood calculation估算氨基酸数据的分歧时间"></a>3.5. 指南三：用approximate likelihood calculation估算氨基酸数据的分歧时间</h2><p>氨基酸数据不能用usedata &#x3D; 1这种模式，只能用approximate likelihood calculation估算分歧时间；方法与指南二类似。</p>
<h3 id="3-5-1-input-files"><a href="#3-5-1-input-files" class="headerlink" title="3.5.1. input files"></a>3.5.1. input files</h3><ul>
<li>输入文件和指南一一致，mcmctree.ctl的参数修改：usedata &#x3D; 3，model &#x3D; 2，seqtype &#x3D; 2；增加一行aaRatefile &#x3D; wag.dat；</li>
<li>同时把paml软件安装位置下的wag.dat复制一份到当前目录；</li>
<li>wag.dat是氨基酸替换速率的数据，与model &#x3D; 2对应，也可以选用其他的氨基酸替换率数据，就用其他的替换率文件(*.dat)；</li>
</ul>
<h3 id="3-5-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）"><a href="#3-5-2-运行mcmctree-mcmctree3-ctl，（mcmctree-usedata-x3D-3）" class="headerlink" title="3.5.2. 运行mcmctree mcmctree3.ctl，（mcmctree usedata &#x3D; 3）"></a>3.5.2. 运行<code>mcmctree mcmctree3.ctl</code>，（mcmctree usedata &#x3D; 3）</h3><p>运行<code>mcmctree mcmctree3.ctl</code>；用Maximum Likelihood方法估算枝长和Hessian信息；</p>
<ol>
<li>若比对数据只有一个区域，即ndata &#x3D; 1，则等待运行完成，生成out.BV；</li>
<li>若比对数据多于一个区域，即ndata &gt; 1，mcmctree会多次调用codeml依次处理每个区域，可等待运行完成，也可以做手动并行运行，以加快分析速度；</li>
<li>手动并行多个区域的操作：</li>
</ol>
<ul>
<li>mcmctree运行起来后，从第一区域开始，生成一套tmp0001文件(包括tmp0001.ctl,tmp0001.txt,tmp0001.trees），ctrl+C切断程序，会自动进入下一区域分析，生成一套tmp0002文件，继续ctrl+C切断程序，直到最后一个区域的tmp文件也生成。</li>
<li>对每个区域的tmp文件分别放置一个单独的目录，修改所有tmp*.ctl内的aaRatefile &#x3D; wag.dat为wag.dat的相对位置或绝对，如aaRatefile &#x3D; ..&#x2F;wag.dat，然后在每一个单独目录内运行<code>codeml tmp*.ctl</code>；</li>
<li>所有区域的codeml运行结束后，会在各自目录生成rst2文件，把所有目录的rst2文件cat合并，就是out.BV文件；</li>
<li>或者用命令<code>for i in $(ls tmp*.ctl);do codeml $&#123;i&#125; ; mv rst2 $&#123;i&#125;.rst2; done</code>调用codeml进行多次分析（每次codeml都生成rst2文件，所以每次生成后把rst2改名，否则rst2被覆盖）；然后<code>cat *rst2 &gt;out.BV</code>合并所有rst2文件到out.BV文件；</li>
</ul>
<h3 id="3-5-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）"><a href="#3-5-3-运行mcmctree-mcmctree2-ctl，（mcmctree-usedata-x3D-2）" class="headerlink" title="3.5.3. 运行mcmctree mcmctree2.ctl，（mcmctree usedata &#x3D; 2）"></a>3.5.3. 运行<code>mcmctree mcmctree2.ctl</code>，（mcmctree usedata &#x3D; 2）</h3><ul>
<li>把input.phy，input.tre，mcmctree3.ctl，out.BV四个文件复制到新建目录下，mcmctree3.ctl的usedata改为usedata &#x3D; 2并重命名为mcmctree2.ctl，out.BV重命名为in.BV；</li>
<li>运行<code>mcmctree mcmctree2.ctl</code>即可获得结果。</li>
</ul>
<h2 id="3-6-指南四：codons数据处理方式"><a href="#3-6-指南四：codons数据处理方式" class="headerlink" title="3.6. 指南四：codons数据处理方式"></a>3.6. 指南四：codons数据处理方式</h2><p>如果是密码子codons数据，实测设置文件里seqtype &#x3D; 1 会报错 “Error: dN&#x2F;dS ratios among branches not implemented for gamma.”。<br>有两个方案处理codons数据。</p>
<ol>
<li>转化codons成第一、第二、第三密码子的三组数据之后，并设置ndata&#x3D;3与seqtype &#x3D; 0结合用核苷酸模式使用codons数据。</li>
<li>把codons翻译成氨基酸数据，用氨基酸模型分析codons数据。</li>
</ol>
<h2 id="3-7-结果文件"><a href="#3-7-结果文件" class="headerlink" title="3.7. 结果文件"></a>3.7. 结果文件</h2><p>程序在运行过程中，会在屏幕生生成一些信息。比较耗时间的步骤主要在于取样的百分比进度：</p>
<ul>
<li>第一列：取样的百分比进度。</li>
<li>第2-6列：参数的接受比例。一般，其值应该在30%左右。20-40%是很好的结果，15-70%是可以接受的范围。若这些值在开始时变动较大，则表示burnin数设置太小。</li>
<li>第7-x列：各内部节点的平均分歧时间，第7列则是root节点的平均分歧时间。若有y个物种，则总共有y-1个内部节点，从第7列开始后的y-1列对应这些内部节点。</li>
<li>倒数第3-x列：r_left值。若输入3各多序列比对结果，则有3列。x列的前一列是中划线 - 。</li>
<li>倒数第1-2列：likelihood值和时间消耗。</li>
</ul>
<p>屏幕信息最后，给出各个内部节点的分歧时间(t)、平均变异速率(mu)、变异速率方差(sigma2)和r_left的Posterior信息：均值(mean)、95%双侧置信区间(95% Equal-tail CI)和95% HPD置信区间(95% HPD CI)等信息。此外，倒数第二列给出了各个内部节点的Posterior mean time信息，可以用于收敛分析。</p>
<p>在当前目录下，生成几个主要结果：</p>
<ul>
<li><p>FigTree.tre    生成含有枝长（分歧时间）的超度量树文件，分歧时间和分歧时间的95%HPD区间。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#NEXUS</span><br><span class="line">BEGIN TREES;</span><br><span class="line"></span><br><span class="line">        UTREE 1 = (((B: 0.408365, (A1: 0.107333, A2: 0.107333) [&amp;95%=&#123;0.0749432, 0.126365&#125;]: 0.301032) [&amp;95%=&#123;0.285365, 0.478163&#125;]: 0.362000, C: 0.770366) [&amp;95%=&#123;0.53864, 0.90012&#125;]: 0.047786, D: 0.818151) [&amp;95%=&#123;0.572406, 0.958116&#125;];</span><br><span class="line"></span><br><span class="line">END;</span><br></pre></td></tr></table></figure>
<p>FigTree.tre文件的解释：其中A1或A2冒号:后的0.107333为A1和A2的枝长，代表A1和A2分歧时间，单位是亿年，(A1,A2)外的[&amp;95%&#x3D;{0.0749432, 0.126365}]为A1和A2分歧时间的95%置信区间；A1或A2的枝长0.107333加上(A1,A2)之后的祖先枝长0.301032之和等于B的枝长0.408363，即B与A1和A2祖先种的分歧时间为0.408363亿年，95%置信区间为[&amp;95%&#x3D;{0.285365, 0.478163}]；依此类推。</p>
</li>
<li><p>mcmc.txt       MCMC取样信息，包含各内部节点分歧时间、平均进化速率、sigma2值等信息，可以在Tracer软件中打开。通过查看各参数的ESS值，若ESS值大于200，则从一定程度上表示MCMC过程能收敛，结果可靠。</p>
</li>
<li><p>out.txt        包含由较多信息的结果文件。例如，各碱基频率、节点命名信息。</p>
</li>
</ul>
<h2 id="3-8-一些注意事项"><a href="#3-8-一些注意事项" class="headerlink" title="3.8. 一些注意事项"></a>3.8. 一些注意事项</h2><p>以下来自<a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2974">陈连福的教程</a>。</p>
<ol>
<li>如何设置burnin、sampfreq和nsample值？</li>
</ol>
<ul>
<li>一般推荐设置nsample值不小于20k(官方示例数据中设置的值)，只有当该值较大时，求得的均值才有意义。</li>
<li>sampfreq表示取样间隔，一般设置为10，100或1000。nsample 和sampfreq 的乘积表示有效迭代的次数，这些迭代越准确，次数越大，则结果越好，越趋于收敛。当然，次数越大，越消耗时间。若数据量很小，可以考虑设置sampfreq为1000；若数据量大，每次迭代耗时多，则考虑设置sampfre为10。</li>
<li>(burnin)其结果，则设置burnin值。要设置足够大的burnin值，保证在程序运行时当迭代比例达到0% (即burnin结束) 时，其参数的接受比例值较好，在30%左右且随迭代次数增多时变动幅度不大。推荐设置burnin的迭代次数为有效次数的10~40%。PAML软件时先burnin后再记录有效迭代的参数值；其它软件如BEAST则记录所有的参数值后，最后汇总时burnin掉一定比例的数据。</li>
<li>总体上，其实就是两个参数：burnin掉的迭代次数和用于估算结果的有效迭代次数。由于需要根据有效迭代数据来进行均值估算，若记录每次迭代的参数，则生成的mcmc.txt文件行数过多，文件太大，汇总时也消耗估算。于是设置没隔一定迭代轮数取样一次，这样生成的mcmc.txt文件不会太大，对最后的均值估算影响不大。</li>
<li>个人推荐有效迭代次数 1<del>10M 次，burnin 0.2</del>4M次。按时间消耗从少到多的参数值：<br>数据简单时，进行0.5M迭代次数，burnin比例40%：{ burnin 200k; sampfreq 10; nsample 50k }<br>数据中等时，进行1M迭代次数，burnin比例40%：{ burnin 400k; sampfreq 10; nsample 100k }<br>数据复杂时，进行5M迭代次数，burnin比例20%：{ burnin 1000k; sampfreq 10; nsample 500k }<br>数据巨大时，进行10M迭代次数，burnin比例20%：{ burnin 2000k; sampfreq 100; nsample 100k }<br>使用相应参数进行分析后，若不满意其收敛程度，则选用更多迭代次数的参数。</li>
</ul>
<ol start="2">
<li>如何检测MCMC的结果是否达到收敛状态？</li>
</ol>
<p>收敛的意思，即经过很多次迭代后，得到的MCMC树的各枝长趋于一个定值，变动非常小。于是，最直接的检测方法是：分别使用不同的Seed值进行mcmctree或infinitesites进行两次或多次分析，然后比较两个结果树是否一致，实际就是比较树文件中各内部节点的Height值（分歧时间 &#x2F; Posterior time）。比较方法可以有多种: </p>
<ul>
<li>结果文件mcmc.txt中每行记录了一个取样的结果，包含各内部节点的Posterior time值，即进化树的Height值，对该mcmc.txt文件进行分析，得到每个内部节点的Posterior mean time值。然后作图 （官方说明文档第6页图示），横坐标表示第一次运行的各内部节点的分歧时间均值、纵坐标表示第二次运行的各内部节点的分歧时间均值。该散点图要表现出非常明显的对角线，才认为达到收敛。这时可以考虑估算其相关系数来判断是否符合线性。</li>
<li>我觉得第一种官方文档给的方法比较麻烦且是否符合对角线没有明确的定义，于是就直接比较两次结果树文件中的各枝长。估算各枝长总的偏差百分比，当偏差百分比低于0.1%，则认为两次结果非常吻合，差异低于0.1%，认为达到收敛。</li>
<li>此外，还可以使用Tracer分析mcmc.txt文件，检测其ESS值，一般认为该值高于200，则可能达到收敛。该方法可用于辅助检测。最后，若不收敛，则需要提高burnin、nsample值，重新运行程序。</li>
</ul>
<ol start="3">
<li>使用infinitesites输入含有枝长信息的系统进化树快速进行分歧时间估算<br>infinitesites程序进行分歧时间估算时，其假设前提为多序列比对的序列长度为无限长。相比于正常的mcmctree命令，要求额外多输入一个名为 FixedDsClock23.txt 的文件。该文件中存放了相应的带有枝长信息的系统发育树。该文件中系统发育树的拓扑结构要和input.tre一致。推荐使用baseml命令输入input.trees和多序列比对结果估算枝长。</li>
</ol>
<p>FixedDsClock23.txt 文件内容示例：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">7</span><br><span class="line">((((human: 0.029043, (chimpanzee: 0.014557, bonobo: 0.010908): 0.016729): 0.015344, gorilla: 0.033888): 0.033816, (orangutan: 0.026872, sumatran: 0.022437): 0.069648): 0.073309, gibbon: 0.024637);</span><br><span class="line">((((human: 0.012463, (chimpanzee: 0.002782, bonobo: 0.003835): 0.003331): 0.004490, gorilla: 0.014278): 0.006308, (orangutan: 0.010818, sumatran: 0.008845): 0.030551): 0.004363, gibbon: 0.029246);</span><br><span class="line">((((human: 0.270862, (chimpanzee: 0.066698, bonobo: 0.056883): 0.124104): 0.139082, gorilla: 0.310797): 0.391342, (orangutan: 0.152555, sumatran: 0.114176): 0.696518): 0.017607, gibbon: 1.394718);</span><br></pre></td></tr></table></figure>
<p>运行infinitesites命令进行分歧时间估算：<code>infinitesites mcmctree.ctl</code><br>使用infinitesites进行分歧时间估算时，程序要求输入多序列比对文件。虽然程序读取了序列信息，但在估算时会忽略其序列信息。</p>
<h1 id="4-r8s估算系统分歧时间"><a href="#4-r8s估算系统分歧时间" class="headerlink" title="4. r8s估算系统分歧时间"></a>4. r8s估算系统分歧时间</h1><h2 id="4-1-r8s"><a href="#4-1-r8s" class="headerlink" title="4.1. r8s"></a>4.1. r8s</h2><p>r8s 是美国加利福尼亚大学戴维斯分校的进化生物学家 Mike Sanderson 在2003年编写的用于估算进化树分化时间的软件，在进化生物学、分子生物地理学等学科有着广泛的应用，是估算分化时间常用的软件之一。该软件中的一些方法如 NPRS 和 PL 是软件作者最先提出的，目前在同类的其他软件中还难以实现。</p>
<h2 id="4-2-r8s的安装"><a href="#4-2-r8s的安装" class="headerlink" title="4.2. r8s的安装"></a>4.2. r8s的安装</h2><p>wget <a target="_blank" rel="noopener" href="http://loco.biosci.arizona.edu/r8s/r8s.dist.tgz">http://loco.biosci.arizona.edu/r8s/r8s.dist.tgz</a><br>tar zxf r8s.dist.tgz</p>
<p>或者sourceforge下载<a target="_blank" rel="noopener" href="https://sourceforge.net/projects/r8s/">https://sourceforge.net/projects/r8s/</a></p>
<h2 id="4-3-r8s的输入文件"><a href="#4-3-r8s的输入文件" class="headerlink" title="4.3. r8s的输入文件"></a>4.3. r8s的输入文件</h2><h3 id="4-3-1-输入文件r8s-input"><a href="#4-3-1-输入文件r8s-input" class="headerlink" title="4.3.1. 输入文件r8s.input"></a>4.3.1. 输入文件r8s.input</h3><p>一个例子，包括两个模块，一个树模块，一个r8s参数设置模块。<br>树模块可以用raxml-ng等建树软件获得的树，包括枝长信息。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">#nexus</span><br><span class="line">begin trees;</span><br><span class="line">tree tree_1 = (A:0.5069367934,(((((B:0.5881634293,C:0.2730677988)100:0.0325689044,D:0.2516482074)100:0.0150403964,E:0.2672133160)86:0.0239229988,((G:0.4074076834,H:0.4308871688)100:0.0260782585,I:0.2360982873)100:0.0237486276)100:0.0147845808,(J:0.5746285099,K:0.2089809581)92:0.0360035242)100:0.1832776220,L:0.2440847991);</span><br><span class="line">end;</span><br><span class="line"></span><br><span class="line">begin r8s;</span><br><span class="line">blformat lengths=persite nsites=300000 ulrametric=no;</span><br><span class="line">mrca pointa A B;</span><br><span class="line">mrca pointb K L;</span><br><span class="line">fixage taxon=pointa age=520;</span><br><span class="line">constrain taxon=pointb min_age=350 max_age=410;</span><br><span class="line">divtime method=PL algorithm=TN;</span><br><span class="line">set smoothing=100;</span><br><span class="line">showage;</span><br><span class="line">describe plot=chronogram;</span><br><span class="line">describe plot=chrono_description;</span><br><span class="line">end;</span><br></pre></td></tr></table></figure>

<h3 id="4-3-2-r8s参数设置模块"><a href="#4-3-2-r8s参数设置模块" class="headerlink" title="4.3.2. r8s参数设置模块"></a>4.3.2. r8s参数设置模块</h3><ol>
<li>blformat命令：进化树基本信息，需要在计算divergence time&#x2F;rates analysis之前运行。</li>
</ol>
<ul>
<li>lengths选项：total|persite<br>  描述进化树长度的单位。MP简约法获得的进化树，枝长单位是碱基替换个数，lengths选项应该用total；ML极大似然法获得的进化树，枝长单位是期望的碱基替换数，lengths选项应该用persite。</li>
<li>nsites选项：number<br>  在进化分析中用到的碱基位点数量。</li>
<li>ultrametric选项：yes|no<br>  表示该进化树是否经过了校正。如果选yes，直接给出进化时间，不对内部进行任何更改。如果要校对，则应该用calibrate命令。如果不想对ultrametric的进化树的分化时间全部重新估算，应该选yes。</li>
<li>round选项：yes|no<br>  是否对枝长信息四舍五入。只针对nonultrametric树（因为ultrametric树的枝长经过四舍五入会转变成not truly ultrametric树。）。round选项将每个分枝上的替换数转换为整数。如果用户自己提供了已经校对时间的进化树，chronograms，需要设定round&#x3D;no。</li>
</ul>
<ol start="2">
<li><p>mrca命令：为节点定名。<br>mrca(most recent common ancestor)可以命名两个或多个类群的最近共同祖先。<br>例如想要给树 tree test&#x3D;(a,(b,c));中a、b的共同祖先定名为node1，可以通过命令 mrca node1 a b来实现。</p>
</li>
<li><p>fixage命令：设定节点的分化时间。<br>在mrca命令的基础上，设定已知的节点分化时间点，r8s需要至少一个内部节点进行fixage。<br>用法fixage taxon&#x3D;node1 age&#x3D;100;</p>
</li>
<li><p>constrain命令：设定节点的分化时间。<br>在mrca命令的基础上，设定已知的节点分化时间上下限。<br>用法constrain taxon&#x3D;node1 min_age&#x3D;50 max_age&#x3D;200;</p>
</li>
<li><p>divtime命令：分化时间估算<br>依据设定好的分化时间，对未知节点的分化时间进行估算。<br>用法 divtime method&#x3D;LF|NPRS|PL algorithm&#x3D;POWELL|TN|QNEWT</p>
</li>
</ol>
<ul>
<li>method选项（计算方法）：LF|NPRS|PL<br>  分化时间估算的方式。推荐PL罚分似然法：如果各枝长变化速率差异很大，则smoothing值应该取的很小；如果各枝长变化速率差异很小，则smoothing值应该取的很大。</li>
<li>algorithm选项（数学算法）：POWELL|TN|QNEWT<br>  算法，作者推荐TN算法。</li>
<li>crossv选项：基于化石的进化树校正</li>
</ul>
<p>divtime method&#x3D;PL crossv&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;<br>上述命令，是设置 smoothing 的值从 1, 10, 100, 1000 … 1e17, 来计算，最后得到最佳的 smoothing 值。</p>
<p>若使用 fixage 对 2 个节点的分歧时间进行了固定，则可以运行命令：<br>divtime method&#x3D;PL crossv&#x3D;yes fossilfixed&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;</p>
<p>若使用 fixage 对 1 个节点进行分歧时间固定，同时使用 constrain 对 2 个节点进行了约束，则可以运行命令：<br>divtime method&#x3D;PL crossv&#x3D;yes fossilconstrained&#x3D;yes cvstart&#x3D;0 cvinc&#x3D;1 cvnum&#x3D;18;</p>
<p>得到最优的smoothing值后，再用set设置，然后进行divtime分歧时间和替换速率的计算。</p>
<ol start="6">
<li><p>smoothing<br>divtime使用PL方法，需要设置参数smoothing的值。通过设置多个smoothing的值来进行一些计算，选择最优的值之后用set设置。</p>
</li>
<li><p>showage命令：显示分化时间和分化速率</p>
</li>
<li><p>describe命令：显示进化树及树的说明</p>
</li>
</ol>
<ul>
<li><p>cladogram<br>  得到分支树的图，图上有各个节点的编号，和showage的结果结合观察。</p>
</li>
<li><p>phylogram<br>  得到进化树的图，枝长表示替换数。</p>
</li>
<li><p>chronogram<br>  得到超度量树的图，枝长表示时间。</p>
</li>
<li><p>ratogram<br>  得到树图，枝长表示替换速率。</p>
</li>
<li><p>phylo_description<br>  得到树的ASCII文字结果，枝长表示替换数。</p>
</li>
<li><p>chrono_description<br>  得到树的ASCII文字结果，枝长表示时间。</p>
</li>
<li><p>rato_description<br>  得到树的ASCII文字结果，枝长表示替换速率。</p>
</li>
<li><p>node_info<br>  得到节点的信息表格</p>
</li>
</ul>
<p>做超度量时间树，一般选择<br>describe plot&#x3D;chronogram;<br>describe plot&#x3D;chrono_description;</p>
<h2 id="4-4-r8s运行"><a href="#4-4-r8s运行" class="headerlink" title="4.4. r8s运行"></a>4.4. r8s运行</h2><p><code>r8s -b -f r8s.in &gt; r8s.out</code></p>
<p>-b是batch process the datafile，-f用于指定输入文件</p>
<p>结果保存在r8s.out文件中</p>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><p><a target="_blank" rel="noopener" href="http://abacus.gene.ucl.ac.uk/software/MCMCtree.Tutorials.pdf">mcmctree manual</a>;<br><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2974">陈连福的paml mcmctree教程</a><br><a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/19/2/301/372781">r8s paper</a><br><a target="_blank" rel="noopener" href="https://naturalis.github.io/mebioda/doc/week1/w1d5/r8s1.7.manual.pdf">r8s manual</a><br><a target="_blank" rel="noopener" href="http://image.sciencenet.cn/olddata/kexue.com.cn/upload/blog/file/2010/3/201032420201531842.0.pdf">r8s manual 中文</a><br><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2280">陈连福的r8s教程</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/03/25/bioinfo_caculate.divergence.time/" data-id="cl1aidmwt00284gvk9stkhqyg" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/03/27/taxon_Bauhinia/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          羊蹄甲属常见三种类群的分类特征
        
      </div>
    </a>
  
  
    <a href="/2020/12/29/system_reinstall.windows/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">重装windows系统</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>