<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>基因组注释（一）：重复序列注释 | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="记录了基因组注释的第一步，重复序列的注释，转座子的相关知识，注释常用软件和用法。">
<meta property="og:type" content="article">
<meta property="og:title" content="基因组注释（一）：重复序列注释">
<meta property="og:url" content="http://example.com/2021/08/02/omics_genome.repeat.annotation/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="记录了基因组注释的第一步，重复序列的注释，转座子的相关知识，注释常用软件和用法。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://github.com/oushujun/EDTA/raw/master/development/EDTA%20workflow.png?raw=true">
<meta property="article:published_time" content="2021-08-02T02:30:00.000Z">
<meta property="article:modified_time" content="2022-03-28T03:36:09.574Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="tutorial">
<meta property="article:tag" content="genome annotation">
<meta property="article:tag" content="repeat sequences">
<meta property="article:tag" content="TE">
<meta property="article:tag" content="RepeatMasker">
<meta property="article:tag" content="RepeatModeler">
<meta property="article:tag" content="EDTA">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/oushujun/EDTA/raw/master/development/EDTA%20workflow.png?raw=true">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-omics_genome.repeat.annotation" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/08/02/omics_genome.repeat.annotation/" class="article-date">
  <time datetime="2021-08-02T02:30:00.000Z" itemprop="datePublished">2021-08-02</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/genome/">genome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      基因组注释（一）：重复序列注释
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=365639&auto=1&height=32"></iframe></div>

<p><strong>防坑指南</strong></p>
<ol>
<li>基因组在组装后，注释前。<code>seqkit seq -u lower.geno &gt;upper.geno</code>改为碱基全部大写的形式，因为小写碱基有些软件是识别成soft-masked的形式，影响注释预测，edta的TIR预测和geta软件都有可能报错。（这个坑我已经踩两次了，唉~）</li>
</ol>
<p><strong>写在前面</strong></p>
<p>多种参数和条件用下来推荐的方案是：</p>
<ol>
<li>RepeatModeler自我训练</li>
<li>RepeatMasker加载Repbase数据库，并用-species参数指定近缘种，参考RepeatModeler自我训练结果做重复序列预测</li>
<li>EDTA，提供转录组组装的CDS序列做过滤，实现Denovo的TE预测。</li>
<li>结合RepeatMasker和EDTA的结果作为最终基因组的重复序列注释。</li>
</ol>
<h1 id="1-重复序列注释"><a href="#1-重复序列注释" class="headerlink" title="1. 重复序列注释"></a>1. 重复序列注释</h1><p>重复序列占基因组非常高的比例，对重复序列的注释一般是做基因组注释的第一步。常用的基因组重复序列注释软件有RepeatMasker, RepeatModeler, EDTA。</p>
<p>转座子(transposable elements,TE)是可以在基因组内改变位置的一段DNA序列，通常由DNA复制造成，TE是基因组的重要组成部分。</p>
<table>
    <caption><h4>转座子(transposable elements,TEs)的分类</h4></caption>
    <thead>
    <tr>
        <th>转座子</br>(transposable elements,TEs)</th>
        <th>功能</th>
        <th colspan="2">分类</th>
        <th>结构特征</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td rowspan="5"><h6>I 类元件</br>/逆转录转座子(retrotransposons)</h6></td>
        <td rowspan="5">在“复制和粘贴”转座机制中作为RNA中间体</td>
        <td rowspan="3">长末端重复(long terminal repeat, LTR)逆转录转座子</td>
        <td>Ty3-gypsy类</td>
        <td>LTR 元件具有 5 bp 目标位点重复(TSD)；LTR-Gypsy元件为 5'-TG...CA-3'。</td>
    </tr>
    <tr>
        <td>Ty1-copia类</td>
        <td>LTR 元件具有 5 bp 目标位点重复(TSD)；LTR-Copia元件的标准末端序列为 5'-TG...C/G/TA-3'</td>
    </tr>
    <tr>
        <td>其他，比如BEL-Pap类，内源性反转录病毒(HERV,MER4)等</td>
        <td>---</td>
    </tr>
    <tr>
        <td rowspan="2">非长末端重复(non-LTR)逆转录转座子</td>
        <td>长散布核元件</br>(LINEs,long interspersed nuclear element)</td>
        <td rowspan="2">non-LTR 具有可变长度的 TSD 或完全缺乏 TSD，而是与插入时侧翼序列的缺失相关。non-LTR 通常在元件的 3' 末端有一个末端 poly-A 尾</td>
    </tr>
    <tr>
        <td>短散布核元件</br>(SINEs,short interspersed nuclear element)</td>
    </tr>
    <tr>
        <td rowspan="3"><h6>II 类元件</br>/DNA 转座子(DNA transposons)</h6></td>
        <td rowspan="3">在“剪切和粘贴”转座机制中作为DNA中间体</td>
        <td colspan="2">TIR元件</br>(TIR末端反向重复序列,这里指具有TIRs结构的DNA转座子，包括微型反向转座元件MITEs)</td>
        <td>---</td>
    </tr>
    <tr>
        <td colspan="2">Helitron</td>
        <td>通过滚环机制复制，因此不产生 TSD 序列，也没有 TIR，但具有特征 5'-TC…CTRR-3' 末端序列，通常在元素的 3' 端附近有一个富含 GC 的短茎环结构。</td>
    </tr>
    <tr>
        <td colspan="2">Mavericks</br>非TE重复序列</br>(non-TE repeat sequence)</td>
        <td>大尺寸，15-40 kbp</br>包含长的TIRs(几百对碱基)</br>包含保守的5'-AG...TC-3'末端</td>
    </tr>
    </tbody>
</table>

<h1 id="2-RepeatMasker"><a href="#2-RepeatMasker" class="headerlink" title="2. RepeatMasker"></a>2. RepeatMasker</h1><h2 id="2-1-RepeatMasker介绍"><a href="#2-1-RepeatMasker介绍" class="headerlink" title="2.1. RepeatMasker介绍"></a>2.1. RepeatMasker介绍</h2><p>RepeatMasker是基因组重复序列检测的常用工具。一般依赖于已有的重复序列参考库Repbase作同源预测。对于绝大部分目标真核物种，都收录在Repbase中。</p>
<h2 id="2-2-RepeatMasker安装"><a href="#2-2-RepeatMasker安装" class="headerlink" title="2.2. RepeatMasker安装"></a>2.2. RepeatMasker安装</h2><ol>
<li><p>conda安装<br><code>conda install -c bioconda repeatmasker</code></p>
</li>
<li><p>加载Repbase数据库【推荐】</p>
</li>
</ol>
<ul>
<li><p>强烈建议加载<a target="_blank" rel="noopener" href="https://www.girinst.org/server/RepBase/index.php">Repbase数据库</a>，遗传信息研究所发布的重复DNA数据库，收录了非常多的物种基因组的重复序列信息，目前是需要付费的，加载后重复序列注释率得到很大提升。</p>
</li>
<li><p>我上传了一个20181026的版本到百度云盘，可以下载：<br>链接：<a target="_blank" rel="noopener" href="https://pan.baidu.com/s/1g43z1DyHtiGxAGJMiC4nZQ">https://pan.baidu.com/s/1g43z1DyHtiGxAGJMiC4nZQ</a><br>提取码：ckc4</p>
</li>
<li><p>下载解压后把目录下两个文件RMRBSeqs.embl和README.RMRBSeqs放到RepeatMasker安装目录的Libraries目录下，如果是conda安装的在pathto&#x2F;anaconda3&#x2F;share&#x2F;RepeatMasker&#x2F;Libraries&#x2F;下。</p>
</li>
</ul>
<ol start="3">
<li>配置依赖</li>
</ol>
<ul>
<li><p>在RepeatMasker目录下运行<code>perl ./configure</code>，依次配置trf的位置，选择搜索引擎（4选1，推荐RMBlast或者HMMER（nhmmer命令），可以配置多个，但运行时只能用一种，选完再选5）。</p>
</li>
<li><p>一般如果依赖的软件安装成功加入环境变量了，会在配置依赖时自动出现依赖软件的路径，如果没有则手动输入依赖软件的路径。最后提示RepeatMasker is now ready to use即表示配置成功。</p>
</li>
</ul>
<h2 id="2-3-RepeatMasker使用"><a href="#2-3-RepeatMasker使用" class="headerlink" title="2.3. RepeatMasker使用"></a>2.3. RepeatMasker使用</h2><h3 id="2-3-1-查看数据库中的物种"><a href="#2-3-1-查看数据库中的物种" class="headerlink" title="2.3.1. 查看数据库中的物种"></a>2.3.1. 查看数据库中的物种</h3><ol>
<li>Repbase数据库配置好以后，用<code>RepeatMasker/util/queryRepeatDatabse.pl -tree &gt;species.txt</code>可以在species.txt中查看数据库中的物种列表。</li>
<li>查看RepeatMasker&#x2F;Libraries&#x2F;taxonomy.dat文件，也有所有已收录物种的名称。</li>
</ol>
<ul>
<li>找到数据库中已有训练的物种或者近缘种后，在使用RepeatMasker时用参数-species指定库中已有物种作为参考。</li>
</ul>
<h3 id="2-3-2-RepeatModeler自我训练"><a href="#2-3-2-RepeatModeler自我训练" class="headerlink" title="2.3.2. RepeatModeler自我训练"></a>2.3.2. RepeatModeler自我训练</h3><ul>
<li>大部分情况下数据库中没有待注释物种的近缘种，此时最好用RepeatModeler等软件进行自我训练，用denovo预测来构建重复序列数据库。</li>
<li>即使数据库中已有近缘种，有时候Repbase注释重复区的效果不是很好，也推荐进行自我训练。</li>
</ul>
<h4 id="2-3-2-1-RepeatModeler安装"><a href="#2-3-2-1-RepeatModeler安装" class="headerlink" title="2.3.2.1. RepeatModeler安装"></a>2.3.2.1. RepeatModeler安装</h4><ol>
<li>RepeatModeler依赖：</li>
</ol>
<ul>
<li>配置好的RepeatMasker</li>
<li>RECON：Denovo Repeat Finder</li>
<li>RepeatScout：Denovo Repeat Finder</li>
<li>NSEG</li>
</ul>
<ol start="2">
<li><p>RepeatModeler支持conda安装：<code>conda install -c bioconda repeatmodeler</code></p>
</li>
<li><p>或者下载源码编译，再一步步指定依赖软件路径，类似RepeatMasker的安装过程。</p>
</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">wget http://repeatmasker.org/RepeatModeler/RepeatModeler-open-1.0.11.tar.gz</span><br><span class="line">tar xzvf RepeatModeler-open-1.0.11.tar.gz</span><br><span class="line"> </span><br><span class="line">cd RepeatModeler-open-1.0.11</span><br><span class="line">chmod -R 755 *</span><br><span class="line">perl ./configure</span><br></pre></td></tr></table></figure>

<h4 id="2-3-2-2-RepeatModeler使用"><a href="#2-3-2-2-RepeatModeler使用" class="headerlink" title="2.3.2.2. RepeatModeler使用"></a>2.3.2.2. RepeatModeler使用</h4><ol>
<li>建库</li>
</ol>
<p>生成sample.nhr,.nin,.nnd,.nni,.nog,.nsq,.translation 7个文件</p>
<p><code>mkdir db &amp;&amp; cd db</code><br><code>BuildDatabase -name sample -engine ncbi sample.fa</code></p>
<ol start="2">
<li>self-training</li>
</ol>
<p><code>nohup RepeatModeler -pa 8 -database db/sample -engine ncbi &amp;</code></p>
<ul>
<li>耗时数天。</li>
<li>-pa 限制的是core的进程，如果服务器是4核，想设置32个线程跑，应该设置-pa 8。</li>
<li>-databse指定库</li>
<li>-engine指定引擎<br>运行跑6 rounds，前4轮有几分钟到十几分钟的-pa线程×cores的超负荷运行，后两轮每轮约1-2h的-pa线程×cores的超负荷运行。</li>
</ul>
<ol start="3">
<li>结果文件</li>
</ol>
<ul>
<li>sample-families.fa （用于repeatmasker指定lib）</li>
<li>sample-families.stk（用于上传到Dfam数据库）</li>
<li>生成目录下RM_21945.SatAug11650032020，内含consensi.fa文件（自身比对找到的一致性序列）</li>
<li>consensi.fa.classified文件（fa格式），包含所有repeat序列和分类，序列id里包含了序列名称和是否能被repeatmasker识别的状态，如果不能被识别就标记Unknown。</li>
</ul>
<p>consensi.fa.classified文件即为训练结果，重复序列数据库，用作其他软件的输入。</p>
<ol start="4">
<li>处理</li>
</ol>
<p>如果想要更加可信的结果，可以把标记为Unknown的去除，留下的保存为ModelerID.lib，作为后续使用。</p>
<ul>
<li><p><code>seqkit grep -r -i -p &quot;Unknown&quot; -v ./RM_21945.SatAug11650032020/consensi.fa.classified &gt; ModelerID.lib</code>  #从consensi.fa.classified 文件提取可以被识别（就是序列id不含Unknown的序列）</p>
</li>
<li><p><code>seqkit grep -r -i -p &quot;Unknown&quot;  ./RM_21945.SatAug11650032020/consensi.fa.classified &gt; Modelerunknown.lib</code>  #从consensi.fa.classified 文件提取不能被识别（就是序列id含Unknown的序列）</p>
</li>
</ul>
<h3 id="2-3-3-RepeatMasker运行"><a href="#2-3-3-RepeatMasker运行" class="headerlink" title="2.3.3. RepeatMasker运行"></a>2.3.3. RepeatMasker运行</h3><p><code>RepeatMasker sample.fa -species &quot;Arachis ipaensis&quot; -pa 12 -lib consensi.fa.classified -poly -html -gff -dir repeatmasker</code></p>
<p><strong>参数解释：</strong></p>
<ul>
<li>-species 选择RepeatMasker数据库中已有近缘物种，比如”Arachis ipaensis”，可用RepeatMasker&#x2F;util&#x2F;queryRepeatDatabase.pl -tree &gt;species.txt获取已有物种列表。但RepeatMasker-4.1.1版本及以后不再提供queryRepeatDatabase.pl脚本，而是由famdb.py代替。</li>
<li>-pa 线程，用RMBlast引擎会用到4核，即设置-pa 12会用到48线程。</li>
<li>-lib RepeatModeler.consensi.fa.classified RepeatModeler自我训练的库</li>
<li>-poly 在file.poly中保存多态的polymorphic的简单重复序列，即微卫星重复序列。</li>
<li>-html输出xhtml格式文件</li>
<li>-gff输出gff格式文件</li>
<li>-dir 指定输出文件的目录，默认时当下目录</li>
</ul>
<h3 id="2-3-4-RepeatMasker结果"><a href="#2-3-4-RepeatMasker结果" class="headerlink" title="2.3.4. RepeatMasker结果"></a>2.3.4. RepeatMasker结果</h3><ul>
<li>sample.fa.tbl：结果报告，统计了基因组长度，GC含量，重复序列长度及各类重复序列占比等。</li>
<li>sample.fa.out：将基因组中预测得到的重复序列和参考序列相比的碱基替换频率、插入&#x2F;删除率，以及重复序列的位置、结构、类型等信息展示出。</li>
<li>sample.fa.out.gff：sample.fa.out的gff格式。</li>
<li>sample.fa.out.html：sample.fa.out的网页形式。</li>
<li>sample.fa.polyout：通过-poly参数，额外将sample.fa.out中的微卫星注释提取出来。微卫星注释是一种Simple_repeat序列，可以通过*.polyout将*.out的微卫星注释删除。有研究者认为微卫星不能算作严格的重复序列类型，还有研究者甚至认为Simple_repeat全都不能算。</li>
<li>sample.fa.cat.ga：记录了基因组序列和数据库中参考重复序列的比对详情，其中“i”和“v”分别代表了碱基转换（transitions）和颠换（transversions），“-”表示该位点存在碱基插入&#x2F;删除。。</li>
<li>sample.fa.masked：将重复序列屏蔽成N碱基的masked基因组。</li>
</ul>
<p>其中，sample.fa.out每一列含义：</p>
<ul>
<li>第一列：比对分值，SW score</li>
<li>第二列：替代率 perc div.</li>
<li>第三列：碱基缺失百分率</li>
<li>第四列：在重复序列中碱基缺失百分率</li>
<li>第五列：query sequence</li>
<li>第六列：查询序列起始位置</li>
<li>第七列：查询序列终止位置</li>
<li>第八列：查询区域中超出比对区域碱基的数- 目，也就是没有比对上的碱基数</li>
<li>第九列：+&#x2F;-(C)</li>
<li>第十列：比上的重复序列名称，类型命名</li>
<li>第十一列：比上重复序列的分类，和- repeatmolder 中*.classed 是一样的</li>
<li>第十二列：比上的在数据库中的起始位置</li>
<li>第十三列：比上的在数据库中的终止位置</li>
<li>第十四列：在第十列上超出比对区域碱基的数- 目，也就是没有比对上的碱基数</li>
<li>第十五列：比对区域的ID，随机给的</li>
</ul>
<h1 id="3-EDTA-The-Extensive-de-novo-TE-Annotator"><a href="#3-EDTA-The-Extensive-de-novo-TE-Annotator" class="headerlink" title="3. EDTA(The Extensive de novo TE Annotator)"></a>3. EDTA(The Extensive de novo TE Annotator)</h1><h2 id="3-1-EDTA介绍"><a href="#3-1-EDTA介绍" class="headerlink" title="3.1. EDTA介绍"></a>3.1. EDTA介绍</h2><p><a target="_blank" rel="noopener" href="https://github.com/oushujun/EDTA#quick-installation-using-conda-linux64">EDTA</a>是一个用于自动化全基因组的转座元件(transposable elements,TE)从头注释(de-novo)注释和评估TE注释的综合软件。</p>
<img src="https://github.com/oushujun/EDTA/raw/master/development/EDTA%20workflow.png?raw=true" title="EDTA流程图" width="90%" />

<p><strong><p align="center">Figure 1. EDTA流程图</strong><br>from <a target="_blank" rel="noopener" href="https://github.com/oushujun/EDTA#quick-installation-using-conda-linux64">github: EDTA</a></p></p>
<p><strong>EDTA运行和调用软件的流程：</strong></p>
<ol>
<li>LTR预测</li>
</ol>
<ul>
<li>LTR_FINDER预测LTR</li>
<li>LTRharvest预测LTR</li>
<li>LTR_retriever确认前两个软件预测的LTR</li>
</ul>
<ol start="2">
<li>TIR预测</li>
</ol>
<ul>
<li>Generic Repeat Finder</li>
<li>TIR-Learner预测TIR</li>
</ul>
<ol start="3">
<li>Helitrons预测</li>
</ol>
<ul>
<li>HelitronScanner</li>
</ul>
<ol start="4">
<li>filter过滤</li>
</ol>
<ul>
<li>以上三种TE预测结束后做basic filter获得full-length TEs</li>
<li>进一步做advance filter，获得raw library</li>
<li>与指定的curated library合并</li>
<li>用RepeatModeler做训练</li>
<li>用指定的CDS做最后的filters</li>
</ul>
<ol start="5">
<li>获得最终的final TE library</li>
<li>Annotation</li>
<li>Evaluation</li>
</ol>
<h2 id="3-2-EDTA-安装"><a href="#3-2-EDTA-安装" class="headerlink" title="3.2. EDTA 安装"></a>3.2. EDTA 安装</h2><p>参考<a target="_blank" rel="noopener" href="https://github.com/oushujun/EDTA#installation">开发者推荐的安装方式</a></p>
<p><strong>注意RepeatMasker安装的Repbase库的加入，参考RepeatMasker的安装。</strong></p>
<h3 id="3-2-1-quick-installation-using-conda"><a href="#3-2-1-quick-installation-using-conda" class="headerlink" title="3.2.1. quick installation using conda"></a>3.2.1. quick installation using conda</h3><ol>
<li><p>下载EDTA<br><code>git clone https://github.com/oushujun/EDTA.git</code></p>
</li>
<li><p>用EDTA.yml创建新的环境和安装依赖<br><code>conda env create -f EDTA.yml</code></p>
</li>
</ol>
<h3 id="3-2-2-another-installation-using-conda"><a href="#3-2-2-another-installation-using-conda" class="headerlink" title="3.2.2. another installation using conda"></a>3.2.2. another installation using conda</h3><ol>
<li><p>【推荐】创建新的环境并进入<br><code>conda create -n EDTA</code><br><code>conda activate EDTA</code></p>
</li>
<li><p>三种方式安装EDTA：</p>
</li>
</ol>
<ul>
<li><p>安装EDTA<br><code>conda install -c bioconda -c conda-forge edta</code></p>
</li>
<li><p>安装EDTA和指定版本依赖<br><code>conda install -c conda-forge -c bioconda edta python=3.6 tensorflow=1.14 &#39;h5py&lt;3&#39;</code></p>
</li>
<li><p>用mamba加速安装<br><code>conda install -c conda-forge mamba</code><br><code>mamba install -c conda-forge -c bioconda edta python=3.6 tensorflow=1.14 &#39;h5py&lt;3&#39;</code></p>
</li>
</ul>
<h3 id="3-2-3-singularity安装"><a href="#3-2-3-singularity安装" class="headerlink" title="3.2.3. singularity安装"></a>3.2.3. singularity安装</h3><p><code>singularity pull EDTA.sif docker://oushujun/edta:&lt;tag&gt;</code></p>
<h3 id="3-2-4-docker安装"><a href="#3-2-4-docker安装" class="headerlink" title="3.2.4. docker安装"></a>3.2.4. docker安装</h3><p><code>docker pull docker://oushujun/edta:&lt;tag&gt;</code></p>
<p>singularity和docker安装我没用过，参考开发者说明吧。</p>
<h2 id="3-3-EDTA-使用"><a href="#3-3-EDTA-使用" class="headerlink" title="3.3. EDTA 使用"></a>3.3. EDTA 使用</h2><p><code>nohup EDTA.pl --genome sample.fa --cds sample.cds --species others --step all --sensitive 1 --anno 1 --evaluate 1 -t 36 &amp;</code></p>
<p><strong>参数解释：</strong></p>
<ul>
<li>–genome 基因组文件</li>
<li>–cds 提供这个种或近缘种的CDS序列（不能包括introns和UTR），用于最终过滤。</li>
<li>–rmout 提供其他软件做的同源TE注释（比如repeatmasker的.out文件），如果不提供则默认使用EDTA - library用于masking。</li>
<li>–species [Rice|Maize|others]三种可选</li>
<li>–step    [all|filter|final|anno]</li>
<li>-sensitive: 是否用RepeatModeler分析剩下的TE，默认是0，不要。RepeatModeler会增加运行时间。</li>
<li>-anno: 是否在构建TE文库后进行全基因组预测，默认是0.</li>
<li>-evalues: 默认是0，需要同时设置-anno 1才能使用。能够评估注释质量，但会显著增加分析时间。</li>
<li>–overwrite默认是0，设定为1会删除已有结果重新运行，建议保持默认，运行中断可以继续运行。</li>
</ul>
<p>简化推荐版：<br><code>nohup EDTA.pl --genome sample.fa --cds sample.cds --species others --step all --anno 1 -t 36 &amp;</code></p>
<h2 id="3-4-结果文件"><a href="#3-4-结果文件" class="headerlink" title="3.4. 结果文件"></a>3.4. 结果文件</h2><ul>
<li>genome.mod.EDTA.TElib.fa：最终结果，非冗余的TE库。如果在输入文件中用–curatedlib指定的修正版TE库，则该文件中也将包含这部分序列。</li>
<li>genome.mod.EDTA.TElib.novel.fa: 新TE类型。该文件包括在输入文件中用–curatedlib指定的修正版TE库没有的TE序列，即genome.mod.EDTA.TElib.fa减去–curatedlib指定库(需要–curatedlib参数)。</li>
<li>genome.mod.EDTA.TEanno.gff: 全基因组TE的注释。该文件包括结构完整和结构不完整的TE的注释（需要–anno 1参数）。</li>
<li>genome.mod.EDTA.TEanno.sum: 对全基因组TE注释的总结（需要–anno 1参数）。</li>
<li>genome.mod.MAKER.masked: 低阈值TE的屏蔽。该文件中仅包括长TE（&gt;&#x3D; 1 kb）序列(需要–anno 1参数)。</li>
<li>genome.mod.EDTA.TE.fa.stat.redun.sum: 简单TE的注释偏差(需要–evaluate 1参数)。</li>
<li>genome.mod.EDTA.TE.fa.stat.nested.sum：嵌套型TE注释的偏差（需要–evaluate 1参数）。</li>
<li>genome.mod.EDTA.TE.fa.stat.all.sum: 注释偏差的概述（需要–evaluate 1参数）。</li>
</ul>
<h2 id="3-5-评估已有TE注释"><a href="#3-5-评估已有TE注释" class="headerlink" title="3.5. 评估已有TE注释"></a>3.5. 评估已有TE注释</h2><p>当已有TE库结果，想要与其他软件（比如RepeatMasker）预测结果比较，可以用EDTA的脚本<code>lib-test.pl</code>进行，评估是通过与参考注释进行比较来进行的。</p>
<ol>
<li>用RepeatMasker根据已有TE库进行注释<br><code>RepeatMasker -pa 36 -q -no_is -norna -nolow -div 40 -lib sample.TE.lib.fasta -cutoff 225 sample.fa</code></li>
<li>用lib-test.pl进行TE注释的比较和评估<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">perl lib-test.pl -genome genome.fasta -std genome.stdlib.RM.out -tst genome.testlib.RM.out -cat [options]</span><br><span class="line">	-genome	[file]	FASTA format genomesequence</span><br><span class="line">	-std	[file]	RepeatMasker .out file of the standard library</span><br><span class="line">	-tst	[file]	RepeatMasker .out file of the test library</span><br><span class="line">	-cat	[string]	Testing TE category. Use one of LTR|nonLTR|LINE|SINE|TIR|MITEHelitron|Total|Classified</span><br><span class="line">	-N	[0|1]	Include Ns in total length of the genome. Defaule: 0 (not include Ns).</span><br><span class="line">	-unknown	[0|1]	Include unknownannotations to the testing category. This should be used when</span><br><span class="line">				the test library has no classification and you assume they all belong to the</span><br><span class="line">				target category specified by -cat. 	Default: 0 (not include unknowns)</span><br><span class="line">	-rand	[int]	A randum number used to identify the current run. (default:generate automatically)</span><br><span class="line">	-threads|-t	[int]	Number of threads torun this program. Default: 4</span><br></pre></td></tr></table></figure></li>
</ol>
<p><code>lib-test.pl -genome sample.fa -std genome.stdlib.RM.out -tst genome.testlib.RM.out -cat LTR</code></p>
<h2 id="3-6-notes"><a href="#3-6-notes" class="headerlink" title="3.6. notes"></a>3.6. notes</h2><ul>
<li>基因组的碱基全部大写，因为小写在一些软件是被认为进行了soft-masked的，在TIR预测时会报错。</li>
<li>-t设置线程，当运行到调用RMBlast引擎时，默认使用四核，如果服务器是四核，则实际使用线程为设置的-t的四倍（RepeatMasker和RepeatModeler的-pa设置线程有一样的问题），EDTA支持断点续跑，可以先设置-t 高一点，运行到RMBlast时，kill掉，重新设置低一点的-t。</li>
<li>预测有时运行会报错缺一些perl的module，例如Bio，安装缺失的module，再次运行即可。</li>
<li>关于–rmout，当使用这个参数指定同源TE注释时，不使用EDTA的同源注释，而是用这个参数指定的库做同源注释，最终结果是这个同源注释加上EDTA其他注释的完整的intact TEs结果。所以当提供的repeatmasker文件预测结果非常全面时，使用这个参数；如果提供的库对物种TE的预测结果不够全面，使用这个参数反而会降低最终预测结果，不建议使用这个参数。不知道是否提供的库是否全面，有时间就有这个参数和没这个参数的结果比较选用，没时间就直接不用这个参数。参考<a target="_blank" rel="noopener" href="https://github.com/oushujun/EDTA/issues/193">issue</a>。</li>
<li>HelitronScanner预测Helitrons默认需要400G运行内存，若不够会出现如下报错，把运行内存上限升高后再次运行即可：</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ERROR: Raw Helitron results not found in sample.fa.mod.EDTA.raw/sample.fa.mod.Helitron.raw.fa</span><br><span class="line">	If you believe the program is working properly, this may be caused by the lack of intact Helitrons in your genome. Consider to use the --force 1 parameter to overwrite this check</span><br></pre></td></tr></table></figure>

<p><strong>ref：</strong></p>
<ol>
<li><a target="_blank" rel="noopener" href="http://www.repeatmasker.org/RepeatModeler/">http://www.repeatmasker.org/RepeatModeler/</a></li>
<li><a target="_blank" rel="noopener" href="https://www.cnblogs.com/zhanmaomao/p/12345462.html">https://www.cnblogs.com/zhanmaomao/p/12345462.html</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/ddd1c9a74fde">https://www.jianshu.com/p/ddd1c9a74fde</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/dfa89f394882">https://www.jianshu.com/p/dfa89f394882</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/08/02/omics_genome.repeat.annotation/" data-id="cl1aiqcry0042m4vkbjkl9iek" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/08/10/omics_genome.management/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          基因组组装和注释结果的整理
        
      </div>
    </a>
  
  
    <a href="/2021/07/24/biosoft_BUSCO/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">BUSCO评估基因组组装和注释完整性</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>