<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>提交基因组到公共数据库 —— 以GenBank为例 | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="此文记录提交基因组到NCBI的GenBank数据库的步骤，常见错误和修改，以及注意事项。">
<meta property="og:type" content="article">
<meta property="og:title" content="提交基因组到公共数据库 —— 以GenBank为例">
<meta property="og:url" content="http://example.com/2022/03/22/omics_genome.submit/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="此文记录提交基因组到NCBI的GenBank数据库的步骤，常见错误和修改，以及注意事项。">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2022-03-22T13:30:00.000Z">
<meta property="article:modified_time" content="2022-03-28T07:04:51.433Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="genome">
<meta property="article:tag" content="genome submit">
<meta property="article:tag" content="GenBank">
<meta property="article:tag" content="table2asn">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-omics_genome.submit" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2022/03/22/omics_genome.submit/" class="article-date">
  <time datetime="2022-03-22T13:30:00.000Z" itemprop="datePublished">2022-03-22</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/omics/">omics</a>►<a class="article-category-link" href="/categories/omics/genome/">genome</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      提交基因组到公共数据库 —— 以GenBank为例
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=31721697&auto=1&height=32"></iframe></div>

<h1 id="1-基因组数据库"><a href="#1-基因组数据库" class="headerlink" title="1. 基因组数据库"></a>1. 基因组数据库</h1><p>提交基因组组装的数据到公共数据库以实现共享，在发表提供基因组数据文章时在文章内提供基因组的accession number。</p>
<p>常用的公共数据库包括：</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/">NCBI-GenBank</a>：美国生物信息学国家中心</li>
<li><a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/">EMBL-EBI</a>：欧洲生物信息学研究所</li>
<li><a target="_blank" rel="noopener" href="https://www.ddbj.nig.ac.jp/index-e.html">DDBJ</a>：日本DNA数据银行</li>
<li><a target="_blank" rel="noopener" href="https://ngdc.cncb.ac.cn/?lang=zh">中国的国家基因组科学数据中心, National Genomics Data Center, NGDC</a>: 中国</li>
</ul>
<h1 id="2-基因组提交"><a href="#2-基因组提交" class="headerlink" title="2. 基因组提交"></a>2. 基因组提交</h1><p>可以单独提交基因组，也可以基因组和基因组注释一同提交。</p>
<p>单独提交基因组会比加上注释信息简单很多，因为加上注释信息一般都会面临多种注释错误(我的例子是4万个基因注释中包含了将近一千个错误需要手动修改)，但是现在基因组注释是非常基础的操作，所以还是建议加上注释文件更完整。如果发表时间紧张，可以考虑先提交基因组获得accession number(在文章里提供)，再提交注释(accession number不变)。</p>
<ul>
<li>单独提交基因组只需要准备基因组文件(fasta格式)，在提交网站填写基因组相关信息，上传基因组文件即可。</li>
<li>基因组与注释文件共同提交则需要对注释文件(gff或者bed格式)做检查和过滤后，用table2asn软件从基因组文件(fasta格式)和注释文件(gff或者bed格式)生成一个sqn文件，再提交sqn文件到提交网站上。当然，单独的基因组文件也可以转化成sqn格式再上传。</li>
</ul>
<p>在<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/">NCBI的提交网站</a>可以选择各种数据类型，提交基因组就输入<strong>genome</strong>。</p>
<h1 id="3-文件准备"><a href="#3-文件准备" class="headerlink" title="3. 文件准备"></a>3. 文件准备</h1><p>在整理基因组文件，基因组结构注释和功能注释文件时，可以先用table2asn检测错误，以指导基因组改善(例如可能删除污染片段)，以及合并注释文件。</p>
<h2 id="3-1-基因组文件"><a href="#3-1-基因组文件" class="headerlink" title="3.1. 基因组文件"></a>3.1. 基因组文件</h2><p>基因组文件用fa格式储存，最好做一做污染过滤。</p>
<h2 id="3-2-注释文件"><a href="#3-2-注释文件" class="headerlink" title="3.2. 注释文件"></a>3.2. 注释文件</h2><ul>
<li>注释文件可以是tbl格式或者gff格式，我的习惯是用gff格式(许多注释软件结果也是gff格式)。</li>
<li>提交的注释文件可以只有结构注释(不提供功能注释，或者通过Dryad【收费】或Figshare【免费】等文件分享网站提供功能注释文件)，也可以把功能注释整理后添加到结构注释文件的product属性值。</li>
</ul>
<h3 id="3-2-1-功能注释"><a href="#3-2-1-功能注释" class="headerlink" title="3.2.1. 功能注释"></a>3.2.1. 功能注释</h3><p>功能注释以product属性值的形式提供，添加到结构注释gff文件的第九列。</p>
<p>合并功能注释和结构注释可以参考博客<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/05/17/omics_genome.functional.annotation/">基因组注释(三)：基因功能注释</a></p>
<p>product值的不规范情况：</p>
<ul>
<li>包含逗号,（因为逗号是分隔符）</li>
<li>包含竖杠|</li>
<li>短横杠-开头或结尾</li>
<li>冒号:开头或结尾</li>
<li>斜杠&#x2F;开头或结尾</li>
<li>不包含字母</li>
<li>括号（包括圆括号,中括号和大括号）不完整的</li>
</ul>
<h1 id="4-提交基因组"><a href="#4-提交基因组" class="headerlink" title="4. 提交基因组"></a>4. 提交基因组</h1><h2 id="4-1-提交基因组到NCBI的GenBank"><a href="#4-1-提交基因组到NCBI的GenBank" class="headerlink" title="4.1. 提交基因组到NCBI的GenBank"></a>4.1. 提交基因组到NCBI的GenBank</h2><p>单独提交基因组只需要基因组(fasta格式)文件，在基因组提交网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/subs/genome/">genome Submit Portal</a>操作。</p>
<ol>
<li>注册和登录NCBI账号</li>
<li>选择提交类型Submission Type：单个基因组Single genome或者批量多个基因组Batch&#x2F;multiple genomes</li>
</ol>
<p>然后进入信息填写页面：</p>
<ol>
<li><p>提交者信息Submitter</p>
<ul>
<li>包括姓名First and Last name，邮箱Email，单位Submitting organization(学校)，部门Department(学院)，街道Street，城市City，省&#x2F;州State&#x2F;Province，邮编Postal code，国家Country。</li>
<li>在最后可以选择<strong>Update my contact information in profile</strong>会更新填写的信息到账号，下一次新填写就默认填入提交者信息了。</li>
</ul>
</li>
<li><p>一般性的信息General info</p>
<ul>
<li>BioProject和BioSample可以先注册在这里提供accession number；也可以选择还未注册提交之后会增加填写BioProject和BioSample信息的步骤然后自动注册新的；</li>
<li>释放日期Release date可以指定日期，也可以提交通过后立即释放；释放日期可以提交后发邮件更改的；</li>
<li>基因组信息Genome info：如果保存到了sqn文件也可以选择**Genome Assembly structured comment is in the contig .sqn file(s)**就不需要填写(但我选了之后不识别还是不选填入)。包括组装日期Assembly date，组装方法Assembly method和版本，组装名称Assembly name，基因组覆盖度Genome coverage(这里其实是depth)，测序技术Sequencing technology；</li>
<li>确认信息：是否完整基因组，是否最终版，是否de novo从头组装即无参考组装，是否已有提交的更新，是否自动移除被系统认为是污染的序列(建议自动移除，否则可能发邮件让手动移除污染序列)；</li>
<li>Submission title可选填，建议填上以区别于其他提交。</li>
</ul>
</li>
<li><p>BioProject General Info</p>
<ul>
<li>提供创建新的BioProject的信息，项目题目Project title，项目描述Project description，和领域Relevance</li>
<li>可选项：相关的外部链接External links，基金grants</li>
</ul>
</li>
<li><p>Publications【only batch&#x2F;multiple genomes】：选填，发表文章的PubMed ID或者DOI号</p>
</li>
<li><p>BioSample Type：样品类型，选择Human&#x2F;Plant等</p>
</li>
<li><p>BioSample Attributes</p>
<ul>
<li>如果是批量提交基因组，可以用excel或csv表格提交多个BioSamples信息</li>
<li>样品名称Sample Name：区别于其他样品的唯一名称</li>
<li>生物名称Organism，可以填物种学名；个体描述isolate&#x2F;栽培名称cultivar&#x2F;生态型ecotype三选一填，年龄age&#x2F;发育阶段development stage二选一填</li>
<li>地理位置geographic location：可以填国家China</li>
<li>组织tissue：填写取样部位，eg leaf，flower，root，stem</li>
</ul>
</li>
<li><p>Genome Info【only batch&#x2F;multiple genomes】</p>
<ul>
<li>批量提交的基因组应该有相同的类型：要么只有染色体序列，要么都包含非染色体的序列(会被处理到WGD genome程序)</li>
<li>确认信息：是否完整基因组，是否最终版，是否de novo从头组装即无参考组装，是否已有提交的更新，是否自动移除被系统认为是污染的序列(建议自动移除，否则可能发邮件让手动移除污染序列)；</li>
</ul>
</li>
<li><p>Files</p>
<ul>
<li>提交基因组文件，单个基因组提交fasta格式或者sqn格式，批量基因组提交fasta格式或者asn.1格式</li>
<li>FTP或Aspera命令行上传文件夹(&gt;10GB)，或者网页提交(&lt;2GB)或Aspera Connect plugin(2-10GB)上传文件</li>
<li>提交之后会检查信息是否错误，提示修改</li>
</ul>
</li>
<li><p>空隙Gaps【only batch&#x2F;multiple genomes】</p>
<ul>
<li>基因组序列中的N的含义，是否随机合并序列(不用组装软件)，感觉一般都不会随机合并吧</li>
<li>指定代表gap的N的最小数量，是否有代表未知长度gap的N的数量(常用大于100个N代表未知长度的gap)，组装gap的证据的数据源</li>
</ul>
</li>
<li><p>Assignment【only single genomes】：确认是否有染色体序列，填写染色体序列信息，是否有质体或线粒体或质粒序列    </p>
</li>
<li><p>References【only single genomes】：序列作者和相关文章信息</p>
</li>
<li><p>Review&amp;Submit：最后检查一遍信息没错误就确认提交</p>
</li>
</ol>
<h2 id="4-2-提交基因组和注释"><a href="#4-2-提交基因组和注释" class="headerlink" title="4.2. 提交基因组和注释"></a>4.2. 提交基因组和注释</h2><p>提交带有注释的基因组的大致步骤：</p>
<ol>
<li>注册新的BioProject和BioSample</li>
<li>生成template.sbt文件</li>
<li>用table2asn把基因组(fsa)和注释文件(gff&#x2F;tbl)生成用于提交的sqn文件</li>
<li>根据生成的验证结果文件查看table2asn检查出来的错误，并修正基因组和注释文件中包含的错误</li>
<li>重新运行table2asn和修正错误(重复步骤3和4)直至没有错误</li>
<li>根据上一个部分<strong>提交基因组到NCBI的GenBank</strong>的步骤提交sqn文件到NCBI</li>
<li>提交后可能会收到邮件需要再次更改，更改后重新运行table2asn和修正错误(重复步骤3和4)直至没有错误在NCBI修正错误(提交新的sqn文件)</li>
</ol>
<h3 id="4-2-1-准备工作"><a href="#4-2-1-准备工作" class="headerlink" title="4.2.1. 准备工作"></a>4.2.1. 准备工作</h3><h4 id="4-2-1-1-注册新的BioProject和BioSample"><a href="#4-2-1-1-注册新的BioProject和BioSample" class="headerlink" title="4.2.1.1. 注册新的BioProject和BioSample"></a>4.2.1.1. 注册新的BioProject和BioSample</h4><ul>
<li>在网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/subs/">submit</a>提交新的BioProject和BioSample</li>
<li>根据项目和样品信息，计划发表的信息填写，勾选自动生成Locus Tag Prefixes。</li>
<li>提交后马上可以获得BioProject ID(PRJNAxxxxxx)和BioSample ID(SAMNxxxxxxxx)，以及Locus Tag Prefixes(xxxxx)三个编号信息用于genome的提交填写。</li>
</ul>
<h4 id="4-2-1-2-生成template-sbt文件"><a href="#4-2-1-2-生成template-sbt文件" class="headerlink" title="4.2.1.2. 生成template.sbt文件"></a>4.2.1.2. 生成template.sbt文件</h4><p>在网站<a target="_blank" rel="noopener" href="https://submit.ncbi.nlm.nih.gov/genbank/template/submission/">template</a>提交基因组信息和BioProject ID(PRJNAxxxxxx)和BioSample ID(SAMNxxxxxxxx)，生成template.sbt文件。</p>
<h3 id="4-2-2-生成提交文件"><a href="#4-2-2-生成提交文件" class="headerlink" title="4.2.2. 生成提交文件"></a>4.2.2. 生成提交文件</h3><p>用table2asn把基因组(fsa)和注释文件(gff&#x2F;tbl)生成用于提交的sqn文件。</p>
<p>table2asn还有一个功能，是在整理基因组文件，基因组结构注释和功能注释文件时，可以先用table2asn检测错误，以指导基因组(例如可能删除污染片段)，以及合并注释文件。</p>
<h4 id="4-2-2-1-table2asn简介"><a href="#4-2-2-1-table2asn简介" class="headerlink" title="4.2.2.1. table2asn简介"></a>4.2.2.1. table2asn简介</h4><p>table2asn用于把基因组和其他信息转换成NCBI的GenBank认可的ASN.1 (Abstract Syntax Notation 1)文本格式文件(sqn后缀)和其他用于协助修改的文件，是命令行格式的软件，取代了原来的老款软件tbl2asn。</p>
<p>另一个相同功能的图形界面软件是Genome Workbench(尝试了一下，数据量大的情况下Window版本会卡顿，还是推荐table2asn)。</p>
<h4 id="4-2-2-2-安装table2asn"><a href="#4-2-2-2-安装table2asn" class="headerlink" title="4.2.2.2. 安装table2asn"></a>4.2.2.2. 安装table2asn</h4><p>在网址下载对应系统(linux64,mac,win64)版本的table2asn安装包，解压缩，重命名为table2asn并设置权限(linux系统下运行<code>chmod 711 table2asn</code>)。</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/">table2asn下载网址</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/table2asn/">table2asn 用法简介</a></li>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/DOCUMENTATION/table2asn_readme.txt">table2asn read me</a></li>
</ul>
<h4 id="4-2-2-3-运行table2asn"><a href="#4-2-2-3-运行table2asn" class="headerlink" title="4.2.2.3. 运行table2asn"></a>4.2.2.3. 运行table2asn</h4><ol>
<li>用法</li>
</ol>
<ul>
<li>最基本的用法：<code>table2asn -t template.sbt -indir ./</code></li>
<li>注释的基因组提交的基础用法：<code>table2asn -t template.sbt -indir ./ -outdir ./ -M n -Z -locus-tag-prefix L6164</code></li>
<li>注释的基因组提交推荐用法：<code>table2asn -t template.sbt -indir ./ -outdir ./ -M n -Z -a s -V vb -c fx -euk -augustus-fix -l paired-ends -gaps-min 1 -gaps-unknown 100 -locus-tag-prefix L6164 -j &quot;[organism=Bauhinia variegata][isolate=pink_petal]&quot; -logfile table2asn.log</code></li>
</ul>
<ol start="2">
<li>参数解释</li>
</ol>
<ul>
<li>-indir .&#x2F;: 指定存放sample.fsa和sample.gff&#x2F;.tbl文件的目录。自动识别目录下所有*.fsa和*.gff&#x2F;tbl文件，需要sample.fsa和sample.gff有相同前缀</li>
<li>-outdir .&#x2F;: 指定结果文件目录，默认与输入一致</li>
<li>-t: 指定template.sbt文件</li>
<li>-i sample.fsa：指定目录下的单个sample.fsa文件(单个基因组提交)，有基因组的最大限制（大概是2G内），超出限制可以只用-indir不用-i -f指定</li>
<li>-f sample.gff：指定目录下的sample.gff&#x2F;tbl文件，不能和-indir同用</li>
<li>-M n：用作原核或真核基因组提交，代替(-a s -V v -c f)，加上-Z就包含基因组特定的差异验证; -M t则是用作TSA提交，代替(-a s -V v -c f，加上TSA特定的验证)</li>
<li>-Z: 生成差异报告(Discrepancy report)，保存到sample.dr中；只推荐注释基因组或者转录组提交使用</li>
<li>-a s: 指定文件类型；-a a指任意格式，包括单个fasta或ASN.1(default)； -a s指一套不相关的序列，例如一个基因组组装；-a s1指物种内的居群集；-a s2指不同物种的系统集</li>
<li>-V: 验证数据，后面可以跟vbr的任意组合；v验证数据记录，结果保存在sample.val(含有错误的类型和严重程度)，错误总结保存在sample.stats；b生成sample.gbf的GenBank文件；r在不做国家检查的情况下验证(Validates without Country Check)</li>
<li>-c: 清理数据，后面可以跟fxdD的任意组合；f修正差异报告中特定类别的产物名称，被修正的产物名称保存在sample.fixedproducts；x扩展注释的features的1-2个核苷酸边界，使得连接gaps或序列末端；D&#x2F;d修正采样日期</li>
<li>-euk：为差异报告的测试指定是真核生物(eukaryotic lineage)</li>
<li>-augustus-fix：针对augustus的错误进行修正</li>
<li>-locus-tag-prefix L6164: 在向NCBI申请BioProject(以及BioSample)后会分配一个locus-tag-prefix前缀，在这里指定，不指定可能在sample.dr中报错FATAL: INCONSISTENT_PROTEIN_ID。</li>
<li>-gaps-min 1 -gaps-unknown 100: 指定序列中的N的含义，如未正确指定，可能在submit.stats中报错SEQ_INST.InternalNsInSeqRaw。-gaps-min 指定代表gap的连续Ns的最小数量，1指把大于等于1个的连续Ns碱基都转换成assembly_gaps。-gaps-unknown 指定代表未知长度的gap的连续Ns的</li>
<li>-l paired-ends: </li>
<li>-j “[organism&#x3D;Bauhinia variegata][isolate&#x3D;pink_petal]”: 用于指定样本名[organism]和个体名[isolate]，会添加进每一条序列的ID里；不指定可能在sample.stats中报错SEQ_DESCR.BioSourceMissing和SEQ_DESCR.NoSourceDescriptor。</li>
<li>-logfile bv.log:指定log文件，如果运q行有错误存放在log文件</li>
<li>-y：添加评论到每一条提交记录，例如-y “Contigs larger than 2kb have been annotated, representing approximately 87% of the total genome”</li>
</ul>
<h3 id="4-2-3-修正错误"><a href="#4-2-3-修正错误" class="headerlink" title="4.2.3. 修正错误"></a>4.2.3. 修正错误</h3><p>根据生成的验证结果修正注释错误，重复table2asn生成sqn文件和修正错误的步骤直至没有错误。</p>
<h4 id="4-2-3-1-查看错误"><a href="#4-2-3-1-查看错误" class="headerlink" title="4.2.3.1. 查看错误"></a>4.2.3.1. 查看错误</h4><p>查看运行table2asn生成的sample.stats，如果有ERROR-level的错误和部分特定的WARNING-level的错误就需要修正，通过sample.val查看错误的具体描述。<br>另外差异报告文件sample.dr中标注为FATAL的错误也需要修正。</p>
<p>下面是一个sample.stats文件的例子：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">Total messages:		86696</span><br><span class="line"></span><br><span class="line">=================================================================</span><br><span class="line">86259 WARNING-level messages exist</span><br><span class="line"></span><br><span class="line">SEQ_FEAT.CDSmRNArange:	219</span><br><span class="line">SEQ_FEAT.NotSpliceConsensusDonor:	300</span><br><span class="line">SEQ_FEAT.NotSpliceConsensusAcceptor:	585</span><br><span class="line">SEQ_FEAT.SeqFeatXrefFeatureMissing:	84578</span><br><span class="line">SEQ_FEAT.ShortExon:	577</span><br><span class="line"></span><br><span class="line">=================================================================</span><br><span class="line">437 ERROR-level messages exist</span><br><span class="line"></span><br><span class="line">SEQ_INST.StopInProtein:	109</span><br><span class="line">SEQ_INST.InternalNsInSeqRaw:	169</span><br><span class="line">SEQ_DESCR.BioSourceMissing:	47</span><br><span class="line">SEQ_DESCR.NoSourceDescriptor:	1</span><br><span class="line">SEQ_FEAT.InternalStop:	109</span><br><span class="line">SEQ_FEAT.NoStop:	2</span><br></pre></td></tr></table></figure>

<h4 id="4-2-3-2-注释规范"><a href="#4-2-3-2-注释规范" class="headerlink" title="4.2.3.2. 注释规范"></a>4.2.3.2. 注释规范</h4><p>报错常常是注释文件的注释不规范导致的，根据报错我总结出几点注释规范：</p>
<ol>
<li>同一个mRNA包含的exons与CDSs的数量和位置几乎一致；区别在于同一个mRNA的exons的起始和终止位置(第一个exon的首和最后一个exon的尾)与mRNA的起始和终止相同，包含5’UTR和3’UTR，CDSs则不包含5’UTR和3’UTR，所以CDSs的起始和终止位置不一定和exon相同。</li>
<li>第9列的产物属性product值不能包括逗号,(因为逗号是多个属性值的分隔符)，竖杠|，不能不包含字母，不能以短横杠-或冒号:结尾。最好先做好product值的过滤filter，再添加到gff注释文件。</li>
</ol>
<p>product值的不规范情况：</p>
<ul>
<li>包含逗号,（因为逗号是分隔符）</li>
<li>包含竖杠|</li>
<li>短横杠-开头或结尾</li>
<li>冒号:开头或结尾</li>
<li>斜杠&#x2F;开头或结尾</li>
<li>不包含字母</li>
<li>括号（包括圆括号和中括号）不完整的</li>
</ul>
<p>在整理基因组结构注释和功能注释时，可以先用table2asn检测错误，以指导注释合并。</p>
<h4 id="4-2-3-3-报错和修正方案"><a href="#4-2-3-3-报错和修正方案" class="headerlink" title="4.2.3.3. 报错和修正方案"></a>4.2.3.3. 报错和修正方案</h4><p>table2asn运行结束后要根据输出文件的报错进行调整，并重新运行直至没有报错再提交给NCBI。</p>
<p>NCBI给出了<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/validation/">Validation and Discrepancy Report Error Explanations</a>和<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genome_validation/">Validation Error Explanations for Genomes</a>可以参考报错含义和解决方案。</p>
<p>下面我总结了一下我遇到的报错类型和修正方案：</p>
<h5 id="4-2-3-3-1-sample-stats-x2F-sample-val文件中报错ERROR"><a href="#4-2-3-3-1-sample-stats-x2F-sample-val文件中报错ERROR" class="headerlink" title="4.2.3.3.1. sample.stats&#x2F;sample.val文件中报错ERROR"></a>4.2.3.3.1. sample.stats&#x2F;sample.val文件中报错ERROR</h5><p>添加<code>pseudo=true</code>属性值的命令举例：<code>sed -i &#39;/ID=Bv00639;/ &#123;s/$/pseudo=true;/g&#125;&#39; sample.gff</code></p>
<ol>
<li><strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong><ul>
<li>当gff的CDS位置有中间终止密码子时报错</li>
<li>可以修改CDS序列位置，去除中间终止密码子。如果无法去除中间终止密码子，可以在sample.gff文件对应的gene行(第三列为gene的那行)的第九列添加<code>pseudo=true</code>属性值代表序列无法被翻译。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.NoStop</strong><ul>
<li>当gff的CDS位置无终止密码子时</li>
<li>可以延长CDS序列位置，直至出现终止密码子。如果无法找到终止密码子，可以在sample.gff文件对应的gene行(第三列为gene的那行)的第九列添加<code>pseudo=true</code>属性值代表序列无法被翻译。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.AbuttingIntervals</strong><ul>
<li>当gff注释的一个基因的exons位置相邻时。</li>
<li>可以合并两个相邻的exons，更改位置。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.SeqLocOrder</strong><ul>
<li>当gff的exon位置有重叠时。</li>
<li>可以参考mRNA和CDS序列的位置更改重叠的exon位置；exons的边界和mRNA一致，数量和CDS一致。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.CDSmRNAXrefLocationProblem</strong>或<strong>SEQ_FEAT.CDSwithMultipleMRNAs</strong><ul>
<li>当gff注释CDS和exon的位置或数量不一致时</li>
<li>可以更改exon或CDS位置和数量。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.ShortIntron</strong><ul>
<li>含有短于11bp的内含子时</li>
<li>可以修正intron的位置使它长于11bp，或者在gene那行第九列添加<code>pseudo=true</code>属性值，或者在table2asn的命令中中加上参数<code>-c s</code>代表标记为”LOW QUALITY PROTEIN”。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.TransLen</strong><ul>
<li>表示蛋白质长度与预测的蛋白质长度不匹配，运行错误</li>
<li>建议重跑table2asn，报错持续存在就写邮件把sample.sqn和运行的命令行发给NCBI(<a href="mailto:&#103;&#101;&#x6e;&#x6f;&#109;&#101;&#115;&#64;&#110;&#x63;&#98;&#105;&#x2e;&#110;&#x6c;&#x6d;&#46;&#110;&#105;&#104;&#x2e;&#103;&#x6f;&#118;">&#103;&#101;&#x6e;&#x6f;&#109;&#101;&#115;&#64;&#110;&#x63;&#98;&#105;&#x2e;&#110;&#x6c;&#x6d;&#46;&#110;&#105;&#104;&#x2e;&#103;&#x6f;&#118;</a>)让帮忙修改这个错误。</li>
</ul>
</li>
<li><strong>SEQ_FEAT.BadInternalCharacter</strong><ul>
<li>表示gff的第9列的属性值attributes中包含不符合规定的字符，例如”|”符号。</li>
<li>删除第9列属性值包含的”|”符号</li>
</ul>
</li>
<li><strong>SEQ_FEAT.BadTrailingHyphen</strong></li>
</ol>
<ul>
<li>当gff的第9列的属性值attributes中以连字符”-“结尾时</li>
<li>删除第9列属性值包含的”-“符号</li>
</ul>
<ol start="11">
<li><strong>SEQ_FEAT.BadTrailingCharacter</strong><ul>
<li>当gff的第9列的属性值attributes中以连字符”:”结尾时。</li>
<li>删除第9列属性值包含的”:”结尾符号</li>
</ul>
</li>
<li>当gff的product属性不符合蛋白产物名称规定时，会生成Adiantum.nelumboides.fixedproducts文件，记录把不符合规定的product属性值改为<strong>hypothetical protein</strong>。可以提取不符合规定的product属性值用于product的filter。</li>
</ol>
<h5 id="4-2-3-3-2-sample-stats-x2F-sample-val文件中的报警WARNING"><a href="#4-2-3-3-2-sample-stats-x2F-sample-val文件中的报警WARNING" class="headerlink" title="4.2.3.3.2. sample.stats&#x2F;sample.val文件中的报警WARNING"></a>4.2.3.3.2. sample.stats&#x2F;sample.val文件中的报警<strong>WARNING</strong></h5><p>部分WARNING也最好修复；比如<strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong>和<strong>SEQ_FEAT.DuplicateFeat</strong>虽然是在WARNING-level messages中，但上传给NCBI后会发邮件让修改后重新提交。</p>
<ol>
<li><strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong><ul>
<li>exon和CDS的位置或数量不一致且有中间终止密码子导致</li>
<li>可以调整exon和CDS的位置和数量；可能引入<strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong>错误，若引入则在gene行的第九列添加<code>pseudo=true</code>属性值；</li>
</ul>
</li>
<li><strong>SEQ_FEAT.DuplicateFeat</strong><ul>
<li>当gff在同样的位置注释了多个gene时</li>
<li>需要删除其中一个。</li>
</ul>
</li>
</ol>
<h5 id="4-2-3-3-3-sample-dr文件中的报错FATAL"><a href="#4-2-3-3-3-sample-dr文件中的报错FATAL" class="headerlink" title="4.2.3.3.3. sample.dr文件中的报错FATAL"></a>4.2.3.3.3. sample.dr文件中的报错FATAL</h5><p><code>cat sample.dr |grep &quot;FATAL&quot;</code>查看FATAL报错信息，如果物种不是细菌则可以忽略BACTERIAL_开头的报错信息；CONTAINED_CDS开头的信息不影响提交，也可暂时忽略。<br>如果product属性值中有错误格式会报错，比如含有不完整的括号，会显示<strong>FATAL: SUSPECT_PRODUCT_NAMES: 9 features contain unbalanced brackets or parentheses</strong>，需要处理(添加括号或者删除括号)。</p>
<h3 id="4-2-4-在线提交"><a href="#4-2-4-在线提交" class="headerlink" title="4.2.4. 在线提交"></a>4.2.4. 在线提交</h3><p>根据上一个部分<strong>提交基因组到NCBI的GenBank</strong>的步骤提交sqn文件到NCBI</p>
<h3 id="4-2-5-提交后修改"><a href="#4-2-5-提交后修改" class="headerlink" title="4.2.5. 提交后修改"></a>4.2.5. 提交后修改</h3><p>提交给NCBI后会收到邮件通知处理进度，如果有错误，会在提交网站显示Error提交状态，并收到需要更改的邮件。根据邮件提示修正注释错误，重复table2asn生成sqn文件和修正错误的步骤直至没有错误在NCBI修正错误(提交新的sqn文件)。</p>
<p>下面记录了我遇到的邮件中提到的错误和修改方案：</p>
<ol>
<li><p><strong>SEQ_FEAT.SeqFeatXrefNotReciprocal</strong><br>exon和CDS的位置或数量不一致且有中间终止密码子导致。可以调整exon和CDS的位置和数量；可能引入<strong>SEQ_FEAT.InternalStop</strong> 和 <strong>SEQ_INST.StopInProtein</strong>错误，若引入则在gene行的第九列添加<code>pseudo=true</code>属性值；</p>
</li>
<li><p><strong>SEQ_FEAT.DuplicateFeat</strong><br>这个错误是同一位置注释到多个基因引起的，可以提取邮件中的gene ID(重复基因的第二个)信息，保存成gene ID list，然后用命令<code>grep -v -f geneID.list sample.gff &gt;sample.revised.gff</code>删除list里的注释信息。</p>
</li>
<li><p><strong>Discrepancy</strong><br>Discrepancy.txt文件中保存了一些不符合标准的报错，比如product属性值中有横杠起始的元素，或者括号不完全，根据Discrepancy.txt文件指出的错误一条条修改。</p>
</li>
</ol>
<p><code>cat sample.dr |grep &quot;FATAL&quot;</code>查看FATAL报错信息，如果物种不是细菌则可以忽略BACTERIAL_开头的报错信息；CONTAINED_CDS开头的信息不影响提交，也可暂时忽略。<br>如果product属性值中有不完整的括号，会显示<strong>FATAL: SUSPECT_PRODUCT_NAMES: 9 features contain unbalanced brackets or parentheses</strong>，需要处理需要处理(添加括号或者删除括号)。</p>
<h2 id="4-3-提交状态"><a href="#4-3-提交状态" class="headerlink" title="4.3. 提交状态"></a>4.3. 提交状态</h2><p>在提交网页，提交后会显示提交状态，经过一轮机器自动验证和两轮人工检查后可公开数据。</p>
<p>提交状态含义：</p>
<ol>
<li>Unfinished</li>
</ol>
<ul>
<li>提交未完成</li>
<li>提交如果没有最后确认，会保存已填写内容，随时可以登陆账号点击未完成的提交号继续提交。</li>
</ul>
<ol start="2">
<li>Queued</li>
</ol>
<ul>
<li>提交在等待自动验证</li>
<li>刚提交的任务会进入自动验证环节</li>
</ul>
<ol start="3">
<li>Error</li>
</ol>
<ul>
<li>错误</li>
<li>有文件有错误需要修正或者再次提交；会提供错误具体信息的文件和邮件通知。</li>
</ul>
<ol start="4">
<li>Processing</li>
</ol>
<ul>
<li>处理中</li>
<li>提交通过自动验证，等待NCBI员工手动检查</li>
</ul>
<ol start="5">
<li>Processing and accession number present</li>
</ol>
<ul>
<li>处理中，并且显示了序列号</li>
<li>提交通过了NCBI员工的首次检查，等待NCBI员工的最终检查</li>
<li>如果提交有问题最终检查员工会发邮件告知</li>
<li>如果指定了释放日期，提交会在释放日期前再做最终检查；也就是说释放日期前都保持在这个提交状态</li>
</ul>
<ol start="6">
<li>Processed</li>
</ol>
<ul>
<li>已处理</li>
<li>数据已被公开释放，在这之前做的任何更改都会包含在释放数据中，但在提交网站不会显示修改。</li>
</ul>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genomesubmit/">genomes submission guide</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/table2asn/">table2asn</a></li>
<li><a target="_blank" rel="noopener" href="https://ftp.ncbi.nlm.nih.gov/asn1-converters/by_program/table2asn/DOCUMENTATION/">table2asn documentation</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/validation/">Validation and Discrepancy Report Error Explanations</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genome_validation/">Validation Error Explanations for Genomes</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/asndisc/">Discrepancy Report</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/genome-vendor/sequin/blob/master/errmsg/valid.msg">MODULE valid</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genbank/genomes_gff/">annotating genomes with gff3</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2022/03/22/omics_genome.submit/" data-id="cl1aiqcrz0046m4vke1i4cmup" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
  
    <a href="/2022/03/20/bioinfo_geneflow_treemix/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">用TreeMix推断居群间基因流</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>