<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单介绍了mapping法的其中一个评估指标：mapping rate。通过samtools计算mapping r">
<meta property="og:type" content="article">
<meta property="og:title" content="基因组质量评估：（五）mapping法：2. samtools计算mapping rate">
<meta property="og:url" content="http://example.com/2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单介绍了mapping法的其中一个评估指标：mapping rate。通过samtools计算mapping r">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2022-07-22T16:00:00.000Z">
<meta property="article:modified_time" content="2022-07-27T07:31:52.880Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="quality assessment">
<meta property="article:tag" content="genome">
<meta property="article:tag" content="mapping">
<meta property="article:tag" content="sam">
<meta property="article:tag" content="bam">
<meta property="article:tag" content="samtools">
<meta property="article:tag" content="transcriptome">
<meta property="article:tag" content="HiSat2">
<meta property="article:tag" content="mapping rate">
<meta property="article:tag" content="samtools flagstat">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">

<link rel="canonical" href="http://example.com/2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>基因组质量评估：（五）mapping法：2. samtools计算mapping rate | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --></head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          基因组质量评估：（五）mapping法：2. samtools计算mapping rate
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-07-23 00:00:00" itemprop="dateCreated datePublished" datetime="2022-07-23T00:00:00+08:00">2022-07-23</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-07-27 15:31:52" itemprop="dateModified" datetime="2022-07-27T15:31:52+08:00">2022-07-27</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/" itemprop="url" rel="index"><span itemprop="name">omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/" itemprop="url" rel="index"><span itemprop="name">genome</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/quality-assessment/" itemprop="url" rel="index"><span itemprop="name">quality assessment</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>4.3k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>4 mins.</span>
            </span>
            <div class="post-description">mapping法评估基因组组装质量。mapping法是指把测序的reads（包括Pacbio，Illumina，RNA-seq 等reads）映射回组装好的基因组，评估mapping rate，genome coverage，depth分布等指标，用这些指标评估基因组组装质量。这篇文章简单介绍了mapping法的其中一个评估指标：mapping rate。通过samtools计算mapping rate，和HiSat2对RNA-seq进行mapping时把mapping rate统计在log文件中，两者间的差异和转化。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=18600061&auto=1&height=32"></iframe></div>

<h1 id="1-mapping-rate"><a href="#1-mapping-rate" class="headerlink" title="1. mapping rate"></a>1. mapping rate</h1><p>通过mapping把reads与组装好的基因组进行alignment，然后分析mapped reads的sam&#x2F;bam格式文件，统计mapping rate来评估基因组组装质量。期望mapping rate越接近100%，组装质量越高。</p>
<h1 id="2-Hisat2统计的mapping-rate"><a href="#2-Hisat2统计的mapping-rate" class="headerlink" title="2. Hisat2统计的mapping rate"></a>2. Hisat2统计的mapping rate</h1><p>运行Hisat2对RNA-seq进行mapping时生成的log文件<code>hisat.log</code>会保存着比对的mapping rate信息。</p>
<ol>
<li>hisat2比对统计结果hisat.log示例</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">19429766 reads; of these: # reads总数</span><br><span class="line">  19429766 (100.00%) were paired; of these: # 配对的reads数量</span><br><span class="line">    1066192 (5.49%) aligned concordantly 0 times # 一致地比对了0次的reads数量</span><br><span class="line">    14853850 (76.45%) aligned concordantly exactly 1 time # 一致地比对了1次的reads数量</span><br><span class="line">    3509724 (18.06%) aligned concordantly &gt;1 times # 一致地比对了大于1次的reads数量</span><br><span class="line">    ----</span><br><span class="line">    1066192 pairs aligned concordantly 0 times; of these: # 一致地比对了0次的reads数量中：</span><br><span class="line">      54954 (5.15%) aligned discordantly 1 time # 不一致地比对了1次的reads数量</span><br><span class="line">    ----</span><br><span class="line">    1011238 pairs aligned 0 times concordantly or discordantly; of these: #一致或不一致地比对了0次的reads数量中：</span><br><span class="line">      2022476 mates make up the pairs; of these: # 配对的reads数量中：</span><br><span class="line">        1211647 (59.91%) aligned 0 times #比对0次的数量</span><br><span class="line">        607196 (30.02%) aligned exactly 1 time #比对1次的数量</span><br><span class="line">        203633 (10.07%) aligned &gt;1 times #比对大于1次的数量</span><br><span class="line">96.88% overall alignment rate # mapping rate，由mapped reads number/total reads number的比例计算得到</span><br><span class="line">[bam_sort_core] merging from 20 files and 4 in-memory blocks...</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>hisat2结果解释</li>
</ol>
<ul>
<li>hisat.log结果中，<code>19429766 reads; of these:</code>及大部分包含的信息中，双端测序的reads是只统计一次的。比如19429766 reads代表的是有19429766对双端测序的reads，总reads数量是$19429766*2&#x3D;38859532$条。</li>
<li>在<code>2022476 mates make up the pairs; of these:</code>及之后包含的信息中，代表配对的reads数量，双端测序的reads是统计了配对的所有reads，总reads数量就是2022476条。</li>
</ul>
<ol start="3">
<li>hisat2的mapping rate的计算</li>
</ol>
<p>96.88%的overall alignment rate即为mapping rate，计算方法是：</p>
<p>$$mapping rate&#x3D;mapped reads number&#x2F;total reads number$$</p>
<ul>
<li>total reads number用的$19429766*2$。</li>
<li>mapped reads number包含：concordantly exactly 1 time(14853850*2)，aligned concordantly &gt;1 times(3509724*2)，aligned discordantly 1 time(54954*2)，mates make up the pairs中的aligned exactly 1 time(607196)和aligned &gt;1 times(203633)。</li>
</ul>
<p>  $$mapping rate&#x3D;((14853850+3509724+54954)*2+607196+203633)&#x2F;(19429766*2) &#x3D; 37647885&#x2F;38859532*100%&#x3D;96.88%$$</p>
<ul>
<li>mapped reads number的另一种计算方法：concordantly exactly 1 time(14853850)，aligned concordantly &gt;1 times(3509724)，aligned concordantly 0 times(1066192)中aligned到的所有reads，即除了aligned concordantly 0 times(1066192)中的aligned 0 times(1211647&#x2F;2)以外的所有reads。</li>
</ul>
<p>  $$mapping rate&#x3D;(14853850+3509724+54954+1066192-(1211647&#x2F;2))&#x2F;19429766*100%&#x3D;96.88%$$</p>
<h1 id="3-samtools-flagstat统计mapping-rate"><a href="#3-samtools-flagstat统计mapping-rate" class="headerlink" title="3. samtools flagstat统计mapping rate"></a>3. samtools flagstat统计mapping rate</h1><ol>
<li>samtools flagstat</li>
</ol>
<ul>
<li>如果hisat2运行时未保存log文件，也可以用<code>samtools flagstat</code>来计算reads的mapping统计值。</li>
<li>illumina reads和Pacbio reads等的sam&#x2F;bam文件也可以用这种方式统计mapping rate。</li>
<li>flagstat统计结果中，记录的是sam&#x2F;bam文件中reads的记录数量，即mapping record rate（双端测序包含配对的所有reads）。</li>
</ul>
<ol start="2">
<li>samtools flagstat统计</li>
</ol>
<ul>
<li><code>samtools flagstat output.bam &gt; output.flagstat</code></li>
</ul>
<ol start="3">
<li>output.flagstat的结果示例</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">51231959 + 0 in total (QC-passed reads + QC-failed reads) #共有51231959条reads通过QC+0条reads未通过QC，后面的信息行中+后的都是代表QC没通过的reads的数量。</span><br><span class="line">12372427 + 0 secondary</span><br><span class="line">0 + 0 supplementary</span><br><span class="line">0 + 0 duplicates</span><br><span class="line">50020312 + 0 mapped (97.63% : N/A) # 97.63%比例的reads mapping到参考序列上，这就是mapping record rate</span><br><span class="line">38859532 + 0 paired in sequencing</span><br><span class="line">19429766 + 0 read1 # 双端reads中read1的总数</span><br><span class="line">19429766 + 0 read2 # 双端reads中read2的总数</span><br><span class="line">36727148 + 0 properly paired (94.51% : N/A) # 94.51%比例的reads成对的映射上</span><br><span class="line">37160066 + 0 with itself and mate mapped # read映射上但配对read没映射上的数量</span><br><span class="line">487819 + 0 singletons (1.26% : N/A) # 1.26%比例的read没映射上的同时，配对read映射上了</span><br><span class="line">289948 + 0 with mate mapped to a different chr # reads和配对reads映射到不同染色体的情况下的reads数量</span><br><span class="line">205767 + 0 with mate mapped to a different chr (mapQ&gt;=5) # reads和配对reads映射到不同染色体，且映射质量大于等于5的情况下的reads数量</span><br></pre></td></tr></table></figure>

<ol start="3">
<li>samtools flagstat的mapping record rate的计算方法</li>
</ol>
<p>$$mapping record rate&#x3D;mapped recorder number&#x2F;total recorder number&#x3D;((primary) mapped reads number + secondary mapped reads number)&#x2F;(total reads number + secondary mapped reads number)$$</p>
<p>其中，recorder number代表sam文件中去除header部分的比对记录数量（每行一条比对记录，即行数）。</p>
<p>同一reads可能多次mapping，有多条记录，所以recorder number的数量会比reads number多。</p>
<ul>
<li>mapped recorder number用的是50020312；</li>
<li>total recorder number用的是51231959；</li>
<li>secondary mapped reads number是12372427；</li>
</ul>
<p>$$mapping record rate&#x3D;50020312&#x2F;51231959*100%&#x3D;97.63%$$</p>
<p>有文章直接用mapping record rate，但建议用mapping rate来代表mapped reads的比例。</p>
<ol start="4">
<li>与hisat2的统计结果的不同</li>
</ol>
<ul>
<li>samtools flagstat的mapping record rate（97.63%）比hisat2的mapping rate（96.88%）高一些，原因在于计算方式的区别。</li>
</ul>
<ol start="5">
<li>计算mapping rate</li>
</ol>
<ul>
<li>通常我们在文章中使用reads的比例来代表mapping rate（即hisat2的计算方式），通过计算公式，可以利用samtools flagsta的统计数据计算mapping rate。</li>
</ul>
<p>$$mapping rate &#x3D; mapped reads number&#x2F;total reads number &#x3D; (mapped recorder number - secondary mapped reads number)&#x2F;(total recorder number - secondary mapped reads number) &#x3D; (50020312-12372427)&#x2F;(51231959-12372427) &#x3D; 37647885&#x2F;38859532*100% &#x3D; 96.88%$$</p>
<h1 id="4-references"><a href="#4-references" class="headerlink" title="4. references"></a>4. references</h1><ol>
<li>hisat2和samtools flagstat计算的mapping rate不同的解释：<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/73208822">https://zhuanlan.zhihu.com/p/73208822</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=50% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/quality-assessment/" rel="tag"># quality assessment</a>
              <a href="/tags/genome/" rel="tag"># genome</a>
              <a href="/tags/mapping/" rel="tag"># mapping</a>
              <a href="/tags/sam/" rel="tag"># sam</a>
              <a href="/tags/bam/" rel="tag"># bam</a>
              <a href="/tags/samtools/" rel="tag"># samtools</a>
              <a href="/tags/transcriptome/" rel="tag"># transcriptome</a>
              <a href="/tags/HiSat2/" rel="tag"># HiSat2</a>
              <a href="/tags/mapping-rate/" rel="tag"># mapping rate</a>
              <a href="/tags/samtools-flagstat/" rel="tag"># samtools flagstat</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/07/20/omics_genome_quality.assessment_LAI/" rel="prev" title="基因组质量评估：（四）用LAI评估基因组组装的连贯性">
      <i class="fa fa-chevron-left"></i> 基因组质量评估：（四）用LAI评估基因组组装的连贯性
    </a></div>
      <div class="post-nav-item">
    <a href="/2022/07/23/omics_genome_quality.assessment_mapping_intro/" rel="next" title="基因组质量评估：（五）mapping法：1. 简介">
      基因组质量评估：（五）mapping法：1. 简介 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-mapping-rate"><span class="nav-number">1.</span> <span class="nav-text">1. mapping rate</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-Hisat2%E7%BB%9F%E8%AE%A1%E7%9A%84mapping-rate"><span class="nav-number">2.</span> <span class="nav-text">2. Hisat2统计的mapping rate</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-samtools-flagstat%E7%BB%9F%E8%AE%A1mapping-rate"><span class="nav-number">3.</span> <span class="nav-text">3. samtools flagstat统计mapping rate</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-references"><span class="nav-number">4.</span> <span class="nav-text">4. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">71</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">34</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">241</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">404k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">6:07</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  
      

<script>
  if (typeof MathJax === 'undefined') {
    window.MathJax = {
      loader: {
        source: {
          '[tex]/amsCd': '[tex]/amscd',
          '[tex]/AMScd': '[tex]/amscd'
        }
      },
      tex: {
        inlineMath: {'[+]': [['$', '$']]},
        tags: 'ams'
      },
      options: {
        renderActions: {
          findScript: [10, doc => {
            document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
              const display = !!node.type.match(/; *mode=display/);
              const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
              const text = document.createTextNode('');
              node.parentNode.replaceChild(text, node);
              math.start = {node: text, delim: '', n: 0};
              math.end = {node: text, delim: '', n: 0};
              doc.math.push(math);
            });
          }, '', false],
          insertedScript: [200, () => {
            document.querySelectorAll('mjx-container').forEach(node => {
              let target = node.parentNode;
              if (target.nodeName.toLowerCase() === 'li') {
                target.parentNode.classList.add('has-jax');
              }
            });
          }, '', false]
        }
      }
    };
    (function () {
      var script = document.createElement('script');
      script.src = '//cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js';
      script.defer = true;
      document.head.appendChild(script);
    })();
  } else {
    MathJax.startup.document.state(0);
    MathJax.texReset();
    MathJax.typeset();
  }
</script>

    

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/";
    this.page.identifier = "2022/07/23/omics_genome_quality.assessment_mapping_rate_samtools/";
    this.page.title = "基因组质量评估：（五）mapping法：2. samtools计算mapping rate";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
