<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="学习生物信息学的笔记本">
<meta property="og:type" content="website">
<meta property="og:title" content="生信技工">
<meta property="og:url" content="http://example.com/page/2/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="学习生物信息学的笔记本">
<meta property="og:locale" content="zh_CN">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoftware">
<meta property="article:tag" content="skills">
<meta property="article:tag" content="tools">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main">
  
    <article id="post-biosoft_Circos" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/16/biosoft_Circos/" class="article-date">
  <time datetime="2021-11-16T02:10:00.000Z" itemprop="datePublished">2021-11-16</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/biosoft/">biosoft</a>►<a class="article-category-link" href="/categories/biosoft/circos/">circos</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/16/biosoft_Circos/">Circos圈图绘制</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=4278314&auto=1&height=32"></iframe></div>

<h1 id="1-介绍circos圈图"><a href="#1-介绍circos圈图" class="headerlink" title="1. 介绍circos圈图"></a>1. 介绍circos圈图</h1><p>介绍基因组的文章里常常出现下面的circos圈图，这篇博客记录了如何绘制这种圈图。</p>
<img src="http://circos.ca/img/circos-sample-panel.png" title="Circos圈图" width="100%"/>

<p><strong><p align="center">Figure 1. circos圈图示例</strong><br>from <a target="_blank" rel="noopener" href="http://circos.ca/">circos官网</a></p></p>
<p>circos圈图的内核是把矩形掰弯成环状，矩形信息的展示以行和列为基础，环形除了行和列，还可以增加环的不同位置之间的关系，比如基因组特征的Circos图中间常见的染色体共线性关系的展示。</p>
<h1 id="2-绘制circos圈图的软件和包"><a href="#2-绘制circos圈图的软件和包" class="headerlink" title="2. 绘制circos圈图的软件和包"></a>2. 绘制circos圈图的软件和包</h1><p>有许多可以绘制circos圈图的软件和包。常见的有<a target="_blank" rel="noopener" href="http://circos.ca/">Circos软件</a>,TBtools，R包circlize，python版本的Circos等。</p>
<h2 id="2-1-Circos-——-从名字就可以看出是circos绘制软件的霸王"><a href="#2-1-Circos-——-从名字就可以看出是circos绘制软件的霸王" class="headerlink" title="2.1. Circos —— 从名字就可以看出是circos绘制软件的霸王"></a>2.1. Circos —— 从名字就可以看出是circos绘制软件的霸王</h2><p><a target="_blank" rel="noopener" href="http://circos.ca/">Circos软件</a>是2009年Michael Smith Genome Sciences Centre的Martin Krzywinski开发的一款基于perl的做比较基因组Circos圈图绘制的软件，功能完备，被文章里使用的非常多。Martin Krzywinski是生物信息学家和专业摄影师双重身份，所以Circos软件的审美设计是非常可靠的（当然得在合理的配色和布局下，我就用过同一套数据生成两个美学差异极大的图）。</p>
<p>Circos有在线版本<a target="_blank" rel="noopener" href="http://mkweb.bcgsc.ca/tableviewer/">Circos Online</a>，在线使用时是把表格转为圈图，不过只允许最大75行和75列。</p>
<p>下面记录了用Circos绘制基因组圈图的基础操作。</p>
<h3 id="2-1-1-Circos安装"><a href="#2-1-1-Circos安装" class="headerlink" title="2.1.1. Circos安装"></a>2.1.1. Circos安装</h3><p><code>conda create -c bioconda -n circos circos</code> #单独环境安装</p>
<p><code>circos -V</code> #确认版本</p>
<h3 id="2-1-2-Circos运行"><a href="#2-1-2-Circos运行" class="headerlink" title="2.1.2. Circos运行"></a>2.1.2. Circos运行</h3><p><code>circos -conf circos.conf</code></p>
<p>会生成circos.png和circos.svg两个文件。</p>
<h3 id="2-1-3-Circos模块化绘制"><a href="#2-1-3-Circos模块化绘制" class="headerlink" title="2.1.3. Circos模块化绘制"></a>2.1.3. Circos模块化绘制</h3><p>Circos是模块化绘制的方式，即通过增添数据和设置来实现每个部分的绘制。</p>
<p>基因组圈图中的几个基础模块：染色体模块，特征图模块，共线性模块。</p>
<ul>
<li>染色体模块显示了染色体的数量和长度，常常添加标尺，通常在最外圈。</li>
<li>特征图模块可以有多圈，可以是折线图(line)，散点图(scatter)，直方图(histogram)和热图(heatmap)等。通常用直方图或者热图展示基因密度，重复序列密度，GC占比等信息。</li>
<li>共线性模块通常放在中心，代表不同染色体间的共线性关系。如果有多个物种，则可以体现物种间的共线性关系。</li>
</ul>
<h3 id="2-1-4-Circos输入文件"><a href="#2-1-4-Circos输入文件" class="headerlink" title="2.1.4. Circos输入文件"></a>2.1.4. Circos输入文件</h3><p>输入文件包括数据文件和配置文件。</p>
<p>需要展示的模块数据都保存在单独的数据文件中，然后在配置文件circos.conf中调用数据文件。</p>
<p>有些教程会建议多制作几个不同类别的配置文件，然后在circos.conf调用其他配置文件。<br>比如单独设置ticks.conf,links.conf，然后在circos.conf里加入下面内容调用</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&lt;&lt;include ticks.conf&gt;&gt;</span><br><span class="line">&lt;&lt;include links.conf&gt;&gt;</span><br></pre></td></tr></table></figure>

<p>比如单独设置plots_histogram.conf，plots_heatmap.conf，plots_text.conf，然后在circos.conf里加入下面内容调用</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&lt;plots&gt;</span><br><span class="line">&lt;&lt;include plots_histogram.conf&gt;&gt;</span><br><span class="line"> &lt;&lt;include plots_heatmap.conf&gt;&gt;</span><br><span class="line"> &lt;&lt;include plots_text.conf&gt;&gt;</span><br><span class="line">&lt;/plots&gt;</span><br></pre></td></tr></table></figure>

<p>但如果不是设置特别复杂，直接使用circos.conf一个配置文件即可，所有设置都可以写在circos.conf。</p>
<h4 id="2-1-4-1-主配置文件circos-conf"><a href="#2-1-4-1-主配置文件circos-conf" class="headerlink" title="2.1.4.1. 主配置文件circos.conf"></a>2.1.4.1. 主配置文件circos.conf</h4><h5 id="2-1-4-1-1-circos-conf由几个部分组成"><a href="#2-1-4-1-1-circos-conf由几个部分组成" class="headerlink" title="2.1.4.1.1. circos.conf由几个部分组成"></a>2.1.4.1.1. circos.conf由几个部分组成</h5><ul>
<li><ideogram>绘制染色体</li>
<li><ticks>绘制染色体刻度</li>
<li><links>绘制染色体间连线（通常代表共线性）</li>
<li><plots>绘制图形</li>
</ul>
<h5 id="2-1-4-1-2-circos-conf示例"><a href="#2-1-4-1-2-circos-conf示例" class="headerlink" title="2.1.4.1.2. circos.conf示例"></a>2.1.4.1.2. circos.conf示例</h5><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"># 1. 染色体和刻度的设置</span><br><span class="line">karyotype = karyotype.txt #指定染色体数据文件</span><br><span class="line">chromosomes_order = scaf001,scaf002,scaf003,scaf004,contig003,contig006 #指定染色体顺序，若不指定则按照染色体数据文件karyotype.txt的顺序</span><br><span class="line">chromosomes_reverse = /scaf/ #染色体坐标默认是从0到最大值，通过正则匹配把染色体坐标改成最大值到0</span><br><span class="line">chromosomes_scale = /scaf/=0.5rn,/contig/=0.5rn #设置染色体缩放，scaf的所有染色体占据50%的空间，contig的所有染色体占据50%的空间。</span><br><span class="line"></span><br><span class="line">&lt;&lt;include etc/colors_fonts_patterns.conf&gt;&gt;</span><br><span class="line">&lt;&lt;include etc/housekeeping.conf&gt;&gt;</span><br><span class="line"></span><br><span class="line">chromosomes_units = 1000000 #定义长度单位，后面使用u赋值长度</span><br><span class="line">show_ticks = yes #显示染色体刻度</span><br><span class="line">show_tick_labels = yes #显示刻度标签</span><br><span class="line"></span><br><span class="line"># 2. ideogram模块，用于染色体设置</span><br><span class="line">&lt;ideogram&gt;</span><br><span class="line">&lt;spacing&gt;</span><br><span class="line">default = 1u #染色体间间距</span><br><span class="line"> &lt;pairwise scaf001 contig001&gt;</span><br><span class="line">  spacing = 10u</span><br><span class="line"> &lt;/pairwise&gt; # 在scaf001和contig001之间流出10u的间距（其他染色体默认是1u）</span><br><span class="line"> &lt;pairwise scaf004 contig006&gt;</span><br><span class="line">  spacing = 10u</span><br><span class="line"> &lt;/pairwise&gt; # 在scaf004和contig006之间流出10u的间距（其他染色体默认是1u）</span><br><span class="line">&lt;/spacing&gt;</span><br><span class="line">radius = 0.90r #染色体位置</span><br><span class="line">thickness = 20p #染色体的厚</span><br><span class="line">fill = yes</span><br><span class="line">stroke_color = dgrey #颜色</span><br><span class="line">stroke_thickness = 2p</span><br><span class="line"></span><br><span class="line">show_label = yes #显示label，染色体名称</span><br><span class="line">label_font = default #字体</span><br><span class="line">label_radius = dims(ideogram,radius) + 0.05r #染色体标签位置</span><br><span class="line">label_size = 48 #字体大小</span><br><span class="line">label_parallel = yes #是否平行</span><br><span class="line">label_format = eval(sprintf(&quot;%s&quot;,var(chr))) #格式</span><br><span class="line"></span><br><span class="line">&lt;/ideogram&gt;</span><br><span class="line"></span><br><span class="line"># 3. ticks模块，用于染色体刻度设置</span><br><span class="line">&lt;ticks&gt;</span><br><span class="line">radius = 1r #刻度位置</span><br><span class="line">color = black #刻度颜色</span><br><span class="line">thickness = 2p #刻度厚度</span><br><span class="line">multiplier = 1e-6 #输出标签为实际长度与其相乘，实际位置为chromosomes_units*multiplier</span><br><span class="line">format = %d # %d表示显示整数</span><br><span class="line"></span><br><span class="line">&lt;tick&gt;</span><br><span class="line">spacing = 1u #刻度间距离，1u表示一个长度单位，即1chromosomes_units</span><br><span class="line">size = 5p #tick的长度</span><br><span class="line">&lt;/tick&gt;</span><br><span class="line"></span><br><span class="line">&lt;tick&gt;</span><br><span class="line">spacing = 5u</span><br><span class="line">size = 10p</span><br><span class="line">show_label = yes #是否展示标签</span><br><span class="line">label_size = 20p #标签尺寸</span><br><span class="line">label_offset = 10p #让标签往外偏移一点</span><br><span class="line">format = %d #显示整数</span><br><span class="line">&lt;/tick&gt;</span><br><span class="line">&lt;/ticks&gt;</span><br><span class="line"></span><br><span class="line"># 4. image模块，用于输出设置</span><br><span class="line">&lt;image&gt;</span><br><span class="line">dir* = . #输出目录，参数后*表示覆盖已有设置</span><br><span class="line">radius* = 1500p #输出图片半径</span><br><span class="line">angle_offset* = -85 #默认情况下，第一个染色体从-90°位置开始，但设置scaf001和contig001之间的空隙为10u会导致图看起来不对称，通过angle_offset调整角度使图对称。</span><br><span class="line">&lt;&lt;include etc/image.conf&gt;&gt;</span><br><span class="line">&lt;/image&gt;</span><br><span class="line"></span><br><span class="line"># 5. plots模块，用于特征图形设置</span><br><span class="line">&lt;plots&gt;</span><br><span class="line">&lt;plot&gt;</span><br><span class="line">show = True #是否展示该图形，默认展示</span><br><span class="line">type = scatter #展示的图形类型，折线图(line)，散点图(scatter)，直方图(histogram)和热图(heatmap)</span><br><span class="line">file = genes_num.txt #数据文件</span><br><span class="line">r1 = 0.98r #在圈图中的位置，外径</span><br><span class="line">r0 = 0.89r #在圈图中的位置，内径</span><br><span class="line">min = 10 #展示的数据下限</span><br><span class="line">max = 1000 #展示的数据上限</span><br><span class="line">glyph = circle #散点图的符号类型，圆(circle), 矩形(rectangle), 三角形(triangle)</span><br><span class="line">glyph_size = 2p #散点图符号大小，单位为p</span><br><span class="line">color = pastel1-5-qual-2 #散点图符号颜色，直方图外部线的颜色</span><br><span class="line">stroke_color = pastel1-5-qual-2 #对于散点图，符号外部的颜色</span><br><span class="line">stroke_thickness = 1p #对于散点图，符号外部线的厚度</span><br><span class="line">&lt;/plot&gt;</span><br><span class="line"></span><br><span class="line">&lt;plot&gt;</span><br><span class="line">type = histogram #直方图</span><br><span class="line">file = gc_rate.txt #数据文件</span><br><span class="line">fill_color = pastel1-5-qual-1 #直方图填充色</span><br><span class="line">color = pastel1-5-qual-2 #散点图符号颜色，直方图外部线的颜色</span><br><span class="line">r1 = 0.78r</span><br><span class="line">r0 = 0.69r</span><br><span class="line">&lt;/plot&gt;</span><br><span class="line"></span><br><span class="line">&lt;plot&gt;</span><br><span class="line">type = heatmap #热图</span><br><span class="line">file = repeats_num.txt #数据文件</span><br><span class="line">fill_color = pastel1-5-qual-2</span><br><span class="line">r1 = 0.88r</span><br><span class="line">r0 = 0.79r</span><br><span class="line">&lt;/plot&gt;</span><br><span class="line"></span><br><span class="line">&lt;plot&gt;</span><br><span class="line">type = line #折线图</span><br><span class="line">file = col.txt #数据文件</span><br><span class="line">fill_color = pastel1-5-qual-4</span><br><span class="line">r1 = 0.68r</span><br><span class="line">r0 = 0.59r</span><br><span class="line">&lt;/plot&gt;</span><br><span class="line">&lt;/plots&gt;</span><br><span class="line"></span><br><span class="line"># 6. links模块，用于共线性设置</span><br><span class="line">&lt;links&gt;</span><br><span class="line">&lt;link&gt;</span><br><span class="line">file = anchors.simple_link.txt #共线性数据文件</span><br><span class="line">radius = 0.58r #位置外界</span><br><span class="line">color = blue_a4 #颜色</span><br><span class="line">ribbon = yes</span><br><span class="line"></span><br><span class="line"># 6plus. rules模块，放在link内部，用于为共线性模块不同染色体用不同颜色渲染</span><br><span class="line">&lt;rules&gt;</span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf001&quot; #判断link文件中左侧染色体`var(chr1)`的名字是不是`eq` &quot;scaf001&quot;，如果是颜色就进行下面设置。</span><br><span class="line">color = pastel1-9-qual-9 #颜色</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf002&quot;</span><br><span class="line">color = pastel1-9-qual-7</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf003&quot;</span><br><span class="line">color = pastel1-9-qual-6</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf004&quot;</span><br><span class="line">color = pastel1-9-qual-8</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf005&quot;</span><br><span class="line">color = pastel1-9-qual-5</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf006&quot;</span><br><span class="line">color = pastel1-9-qual-4</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf007&quot;</span><br><span class="line">color = pastel1-9-qual-3</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf008&quot;</span><br><span class="line">color = pastel1-9-qual-2</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf009&quot;</span><br><span class="line">color = pastel1-9-qual-1</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf010&quot;</span><br><span class="line">color = pastel1-5-qual-5</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf011&quot;</span><br><span class="line">color = pastel1-5-qual-4</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf012&quot;</span><br><span class="line">color = pastel1-5-qual-3</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf013&quot;</span><br><span class="line">color = pastel1-5-qual-2</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line"></span><br><span class="line">&lt;rule&gt;</span><br><span class="line">condition = var(chr1) eq &quot;scaf014&quot;</span><br><span class="line">color = pastel1-5-qual-1</span><br><span class="line">&lt;/rule&gt;</span><br><span class="line">&lt;/rules&gt;</span><br><span class="line"></span><br><span class="line">&lt;/link&gt;</span><br><span class="line">&lt;/links&gt;</span><br></pre></td></tr></table></figure>

<h5 id="2-1-4-1-3-circos-conf参数详解"><a href="#2-1-4-1-3-circos-conf参数详解" class="headerlink" title="2.1.4.1.3. circos.conf参数详解"></a>2.1.4.1.3. circos.conf参数详解</h5><ol>
<li>颜色选择<br>Circos的颜色用的是<a target="_blank" rel="noopener" href="https://colorbrewer2.org/">colorbrewer2网站</a>的颜色，也可以参考<a target="_blank" rel="noopener" href="http://circos.ca/documentation/tutorials/configuration/colors/images">circos的colors网址</a>。</li>
</ol>
<p>Circos中颜色的命名格式为PALETTE-NUMCOLORS-TYPE-IDX：</p>
<ul>
<li>PALETTE:调色版名，如rdylbu</li>
<li>NUMCOLORS: 颜色数目, 11</li>
<li>调色版类型: div(diverging), seq(sequential), qual(qualitative)</li>
<li>IDX: 调色版中的颜色索引</li>
</ul>
<p>所以brbg-3-div-1代表的是<a target="_blank" rel="noopener" href="http://circos.ca/documentation/tutorials/configuration/colors/images">circos的colors网址</a>里的brbg-div的3组的第一个颜色。</p>
<ol start="2">
<li>长度单位<br>控制图形不同元素大小的三个单位，p,r,u。</li>
</ol>
<ul>
<li>p(pixels), 表示绝对大小；如果使用p作为单位，需要参考circos.conf中设置输出图形模块的<image>里定义的radius输出图片半径。</li>
<li>r(relative), 相对大小；会随着最终图形大小而发生变换。</li>
<li>u(chromosome unit)，定义chromosome unit之后使用；一般在显示刻度时使用。。</li>
</ul>
<h4 id="2-1-4-2-数据文件-——-染色体文件karyotype-txt"><a href="#2-1-4-2-数据文件-——-染色体文件karyotype-txt" class="headerlink" title="2.1.4.2. 数据文件 —— 染色体文件karyotype.txt"></a>2.1.4.2. 数据文件 —— 染色体文件karyotype.txt</h4><p>染色体文件karyotype.txt用于生成染色体模块。</p>
<h5 id="2-1-4-2-1-染色体文件karyotype-txt的获取"><a href="#2-1-4-2-1-染色体文件karyotype-txt的获取" class="headerlink" title="2.1.4.2.1. 染色体文件karyotype.txt的获取"></a>2.1.4.2.1. 染色体文件karyotype.txt的获取</h5><p>从genome.fa.fai里生成karyotype.txt文件<code>head -12 genome.fa.fai |awk &#39;&#123;print &quot;chr - &quot;$1&quot; &quot;$1&quot; 0 &quot;$2&quot; set3-12-qual-12&quot;&#125;&#39; &gt;karyotype.txt</code>，取前12条染色体。</p>
<p>修改最后一列来定义染色体颜色，推荐<code>set3-12-qual</code>系列，一共12种，可以依次给到12条染色体。</p>
<h4 id="2-1-4-3-数据文件-——-线性关系文件-link-txt"><a href="#2-1-4-3-数据文件-——-线性关系文件-link-txt" class="headerlink" title="2.1.4.3. 数据文件 —— 线性关系文件_link.txt"></a>2.1.4.3. 数据文件 —— 线性关系文件_link.txt</h4><p>线性关系文件_link.txt用于生成共线性模块。</p>
<p>格式是六列数据，用tab分隔，每行都代表一条线，由具有共线性的两组染色体ID、起始位置和终止位置组成。</p>
<p><code>contig003  7719    355757  contig014   3459    349346</code></p>
<h5 id="2-1-4-3-1-线性关系文件-link-txt的获取"><a href="#2-1-4-3-1-线性关系文件-link-txt的获取" class="headerlink" title="2.1.4.3.1. 线性关系文件_link.txt的获取"></a>2.1.4.3.1. 线性关系文件_link.txt的获取</h5><p>可以用jcvi生成simple文件，然后用脚本simple2links.py转换成_link.txt文件。</p>
<p><code>pip install jcvi</code>安装jcvi，可以参考博文<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/05/bioinfo_MCscan/">MCscan</a>。</p>
<ol>
<li><p>准备sample.bed和sample.cds两个文件</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">python -m jcvi.formats.gff bed --type=gene --key=ID sample.gff3 &gt;sample.bed #gff2bed</span><br><span class="line">python -m jcvi.formats.bed uniq sample.bed #生成sample.uniq.bed</span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sample.uniq.bed )  sample.cds.fa | seqkit seq -i &gt; sample.cds #提取cds</span><br><span class="line">mv sample.uniq.bed sample.bed</span><br><span class="line">rm sample.leftover.bed</span><br></pre></td></tr></table></figure>
</li>
<li><p>运行共线性分析</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">python -m jcvi.compara.catalog ortholog --no_strip_names --cpu=1 sample sample #生成sample.sample.anchors，单个物种就用sample sample，两个物种就准备两套.bed和.cds文件，用sample1 sample2。</span><br><span class="line">python -m jcvi.compara.synteny screen --minspan=30 --simple sample.sample.anchors sample.sample.anchors.new #生成sample.sample.anchors.simple</span><br></pre></td></tr></table></figure>
</li>
<li><p>MCScanX<br>从MCScanX结果sample.collinearity中生成类似MCSCAN结果的bv.simple。</p>
</li>
</ol>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">grep -C 1 &quot;#&quot; sample.collinearity |grep -v &quot;^-&quot;  &gt;sample.col</span><br><span class="line">tail -1 sample.collinearity &gt;&gt; sample.col</span><br><span class="line">cat sample.col |sed -e &quot;s/#.* /#/g&quot; -e &quot;s/[^b]*bv/bv/&quot; |tr &#x27;\n&#x27; &#x27; &#x27;|sed &quot;s/ #/\n/g&quot;|awk &#x27;&#123;print $2,$3,$5,$6&quot; 200 &quot;$1&#125;&#x27;|sed -e &quot;s/plus/+/g&quot; -e &quot;s/minus/-/g&quot; |grep -v &quot;#&quot; &gt;sample.simple</span><br></pre></td></tr></table></figure>

<ol start="4">
<li>simple2links<br><code>python simple2links.py sample.sample.anchors.simple #生成sample.sample.anchors.simple_link.txt</code></li>
</ol>
<p>sample.sample.anchors.simple_link.txt可以作为circos画共线性的输入文件</p>
<p><a target="_blank" rel="noopener" href="https://github.com/xuzhougeng/myscripts/blob/master/simple2links.py">simple2links.py脚本地址</a></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/env python</span></span><br><span class="line"><span class="comment"># simple2links.py</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">from</span> sys <span class="keyword">import</span> argv</span><br><span class="line"></span><br><span class="line">simple_file = argv[<span class="number">1</span>]</span><br><span class="line"></span><br><span class="line">ref_bed = simple_file.split(<span class="string">&quot;.&quot;</span>)[<span class="number">0</span>] + <span class="string">&quot;.bed&quot;</span></span><br><span class="line">qry_bed = simple_file.split(<span class="string">&quot;.&quot;</span>)[<span class="number">1</span>] + <span class="string">&quot;.bed&quot;</span></span><br><span class="line"></span><br><span class="line">ref_dict = &#123;line.split(<span class="string">&quot;\t&quot;</span>)[<span class="number">3</span>]:line.split(<span class="string">&quot;\t&quot;</span>)[<span class="number">0</span>:<span class="number">3</span>] <span class="keyword">for</span> line <span class="keyword">in</span> <span class="built_in">open</span>(ref_bed)&#125;</span><br><span class="line">qry_dict = &#123;line.split(<span class="string">&quot;\t&quot;</span>)[<span class="number">3</span>]:line.split(<span class="string">&quot;\t&quot;</span>)[<span class="number">0</span>:<span class="number">3</span>] <span class="keyword">for</span> line <span class="keyword">in</span> <span class="built_in">open</span>(qry_bed)&#125;</span><br><span class="line"></span><br><span class="line">fo = <span class="built_in">open</span>(simple_file + <span class="string">&quot;_link.txt&quot;</span>, <span class="string">&quot;w&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> line <span class="keyword">in</span> <span class="built_in">open</span>(simple_file):</span><br><span class="line">    <span class="keyword">if</span> line.startswith(<span class="string">&quot;#&quot;</span>):</span><br><span class="line">        <span class="keyword">continue</span></span><br><span class="line">    items = line.strip().split(<span class="string">&quot;\t&quot;</span>) </span><br><span class="line">    ref_start_gene = items[<span class="number">0</span>]</span><br><span class="line">    ref_end_gene = items[<span class="number">1</span>]</span><br><span class="line">    qry_start_gene = items[<span class="number">2</span>]</span><br><span class="line">    qry_end_gene = items[<span class="number">3</span>]</span><br><span class="line">    </span><br><span class="line">    ref_chr, ref_start = ref_dict[ref_start_gene][<span class="number">0</span>:<span class="number">2</span>]</span><br><span class="line">    ref_end = ref_dict[ref_end_gene][<span class="number">2</span>]</span><br><span class="line">    qry_chr, qry_start = qry_dict[qry_start_gene][<span class="number">0</span>:<span class="number">2</span>]</span><br><span class="line">    qry_end = qry_dict[qry_end_gene][<span class="number">2</span>]</span><br><span class="line">    </span><br><span class="line">    circos_input = [ref_chr, ref_start, ref_end, qry_chr, qry_start, qry_end]</span><br><span class="line">    fo.writelines(<span class="string">&#x27;\t&#x27;</span>.join(circos_input) + <span class="string">&#x27;\n&#x27;</span>)</span><br><span class="line">    </span><br><span class="line">fo.close()</span><br></pre></td></tr></table></figure>

<h4 id="2-1-4-4-数据文件-——-特征图文件"><a href="#2-1-4-4-数据文件-——-特征图文件" class="headerlink" title="2.1.4.4. 数据文件 —— 特征图文件"></a>2.1.4.4. 数据文件 —— 特征图文件</h4><p>特征图文件用于生成折线图(line)，散点图(scatter)，直方图(histogram)和热图(heatmap)等特征图模块。</p>
<h5 id="2-1-4-4-1-获取特征图文件"><a href="#2-1-4-4-1-获取特征图文件" class="headerlink" title="2.1.4.4.1. 获取特征图文件"></a>2.1.4.4.1. 获取特征图文件</h5><p>特征图文件格式相同，由四列组成，染色体ID(chr)，起始位置(start)，终止位置(end)，特征数据(value[options])。</p>
<p>前三列是bed格式，最后一列是特征数据，比如在这个位置有几个基因。通常用法是展示染色体的基因密度信息、重复序列密度和GC含量等信息。</p>
<ol>
<li><p>准备</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cut -d &#x27; &#x27; -f 3,6 karyotype.txt | tr &#x27; &#x27; &#x27;\t&#x27; &gt;genome.txt #获取基因组文件</span><br><span class="line">bedtools makewindows -g genome.txt -w 100000 &gt;genome.windows #以100kb为滑窗，沿染色体创建窗口</span><br></pre></td></tr></table></figure></li>
<li><p>基因密度</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">zgrep &#x27;[[:blank:]]gene[[:blank:]]&#x27; sample.gff3 |awk &#x27;&#123;print $1&quot;\t&quot;$4&quot;\t&quot;$5&#125;&#x27; &gt;genes.bed #提取基因位置信息</span><br><span class="line">bedtools coverage -a genome.windows -b genes.bed | cut -f 1-4 &gt;genes_num.txt #按照滑窗统计基因密度</span><br></pre></td></tr></table></figure></li>
<li><p>重复序列密度</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">awk &#x27;&#123;print $1&quot;\t&quot;$4&quot;\t&quot;$5&#125;&#x27; repeats.gff |grep -v &quot;#&quot; &gt;repeats.bed #提取重复位置信息</span><br><span class="line">bedtools coverage -a genome.windows -b repeats.bed | cut -f 1-4 &gt;repeats_num.txt #按照滑窗统计重复密度</span><br></pre></td></tr></table></figure></li>
<li><p>GC含量<br><code>bedtools nuc -fi genome.fa -bed genome.windows | cut -f 1-3,5 | sed &#39;1d&#39; &gt; gc_rate.txt #按照滑窗统计GC含量</code></p>
</li>
<li><p>共线性区块的基因密度</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cat bv.col|sed -e &quot;s/#.* /#/g&quot; -e &quot;s/[^b]*bv/bv/&quot; |tr &#x27;\n&#x27; &#x27; &#x27;|sed &quot;s/ #/\n/g&quot;|awk &#x27;&#123;print $2,$3,$5,$6&#125;&#x27;|grep -v &quot;#&quot; |sed &quot;s/ /\n/g&quot;|sed &quot;/^$/d&quot; |sort |uniq &gt;bv.col.list #获取共线性区块包含的基因列表</span><br><span class="line">grep -f col.list Bauhinia.variegata.gene.gff3 |awk &#x27;$3 == &quot;gene&quot; &#123;print $1&quot;\t&quot;$4&quot;\t&quot;$5&#125;&#x27; &gt;col.bed #获取基因的位置信息</span><br><span class="line">bedtools coverage -a genome.windows -b col.bed |cut -f 1-4 &gt;col.txt #统计共线性基因密度</span><br></pre></td></tr></table></figure></li>
</ol>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><p><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2752132/">Circos paper</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/3a31ceef711b">xuzhougeng 1</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/4b3d3809ac07">xuzhougeng 2</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/ea3a8933ace9">xuzhougeng 3</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/1658e702ba17">xuzhougeng 4</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/31f26d1e5974">xuzhougeng 5</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/16/biosoft_Circos/" data-id="cl1aiqcrn002um4vk49630twn" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/plot/" rel="tag">plot</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_GSEA_topGO" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/13/bioinfo_GSEA_topGO/" class="article-date">
  <time datetime="2021-11-13T08:00:00.000Z" itemprop="datePublished">2021-11-13</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/13/bioinfo_GSEA_topGO/">基因富集分析(gene set enrichment analysis, GSEA)—— topGO</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=116744&auto=1&height=32"></iframe></div>

<h1 id="1-基因富集分析-gene-set-enrichment-analysis-GSEA"><a href="#1-基因富集分析-gene-set-enrichment-analysis-GSEA" class="headerlink" title="1. 基因富集分析(gene set enrichment analysis, GSEA)"></a>1. 基因富集分析(gene set enrichment analysis, GSEA)</h1><p>基因富集分析(gene set enrichment analysis, GSEA)的介绍参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/12/bioinfo_GSEA/">博文基因富集分析</a>。</p>
<h1 id="2-topGO"><a href="#2-topGO" class="headerlink" title="2. topGO"></a>2. topGO</h1><h2 id="2-1-topGO简介"><a href="#2-1-topGO简介" class="headerlink" title="2.1. topGO简介"></a>2.1. topGO简介</h2><p>topGO是一个R包，用于半自动的GO terms的基因富集分析。</p>
<p>因为topGO只用于GO的富集分析，且是半自动化的，推荐使用更方便的在线工具<a target="_blank" rel="noopener" href="http://kobas.cbi.pku.edu.cn/">KOBAS-i</a>；<a target="_blank" rel="noopener" href="http://bioinfo.org/kobas">KOBAS-i 备用</a>；<a target="_blank" rel="noopener" href="http://omicslab.genetics.ac.cn/GOEAST/index.php">GOEAST</a>; 或者功能更完善的clusterProfiler包，参考<a href="">博文clusterProfiler包</a>。</p>
<h2 id="2-2-GO-term"><a href="#2-2-GO-term" class="headerlink" title="2.2. GO term"></a>2.2. GO term</h2><p>GO term分为三大类：</p>
<ul>
<li>cellular component(CC)-细胞成分（其中基因产物位于细胞内部）</li>
<li>molecular function(MF)-分子功能（基因产物的功能是什么）</li>
<li>biology process(BP)-生物过程（即基因产物参与的一系列事件）<br>三类都可以用topGO做富集分析。</li>
</ul>
<h2 id="2-3-topGO支持的统计检验方法"><a href="#2-3-topGO支持的统计检验方法" class="headerlink" title="2.3. topGO支持的统计检验方法"></a>2.3. topGO支持的统计检验方法</h2><p>topGO包默认算法用的是weight01，是elim和权重算法的混合。</p>
<table>
<thead>
<tr>
<th>topGO支持的统计方法</th>
<th>fisher</th>
<th>ks</th>
<th>t</th>
<th>globaltest</th>
<th>sum</th>
</tr>
</thead>
<tbody><tr>
<td>classic</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
</tr>
<tr>
<td>elim</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
</tr>
<tr>
<td>weight</td>
<td>Y</td>
<td>N</td>
<td>N</td>
<td>N</td>
<td>N</td>
</tr>
<tr>
<td>weight01</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
</tr>
<tr>
<td>lea</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
<td>Y</td>
</tr>
<tr>
<td>parentchild</td>
<td>Y</td>
<td>N</td>
<td>N</td>
<td>N</td>
<td>N</td>
</tr>
</tbody></table>
<h2 id="2-4-安装topGO"><a href="#2-4-安装topGO" class="headerlink" title="2.4. 安装topGO"></a>2.4. 安装topGO</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 安装topGO软件包</span></span><br><span class="line"><span class="keyword">if</span> <span class="punctuation">(</span><span class="operator">!</span>requireNamespace<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">,</span> quietly <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">  install.packages<span class="punctuation">(</span><span class="string">&quot;BiocManager&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;topGO&quot;</span><span class="punctuation">,</span> version <span class="operator">=</span> <span class="string">&quot;3.14&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;Rgraphviz&quot;</span><span class="punctuation">,</span> version <span class="operator">=</span> <span class="string">&quot;3.8&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;GO.db&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;biomaRt&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p>或者</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">source<span class="punctuation">(</span><span class="string">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="punctuation">)</span></span><br><span class="line">biocLite<span class="punctuation">(</span><span class="string">&quot;topGO&quot;</span><span class="punctuation">)</span></span><br><span class="line">biocLite<span class="punctuation">(</span><span class="string">&quot;GO.db&quot;</span><span class="punctuation">)</span></span><br><span class="line">biocLite<span class="punctuation">(</span><span class="string">&quot;biomaRt&quot;</span><span class="punctuation">)</span></span><br><span class="line">biocLite<span class="punctuation">(</span><span class="string">&quot;Rgraphviz&quot;</span><span class="punctuation">)</span></span><br><span class="line"> </span><br><span class="line"><span class="comment"># Load the required R packages</span></span><br><span class="line">library<span class="punctuation">(</span>topGO<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>GO.db<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>biomaRt<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>Rgraphviz<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h2 id="2-5-topGO做基因富集分析-GSEA"><a href="#2-5-topGO做基因富集分析-GSEA" class="headerlink" title="2.5. topGO做基因富集分析(GSEA)"></a>2.5. topGO做基因富集分析(GSEA)</h2><h3 id="2-5-1-输入文件"><a href="#2-5-1-输入文件" class="headerlink" title="2.5.1. 输入文件"></a>2.5.1. 输入文件</h3><p>两个输入文件</p>
<ul>
<li>genes.list：需要做富集分析的geneID的list，一个基因ID一行</li>
<li>sample.anno：基因及GO注释信息，第一列是geneID，第二列是GO注释，空格分隔，GO注释可以有多个，格式为GO:0000428,GO:0003677,GO:0005506,</li>
</ul>
<h3 id="2-5-2-GSEA分析"><a href="#2-5-2-GSEA分析" class="headerlink" title="2.5.2. GSEA分析"></a>2.5.2. GSEA分析</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 设置工作目录，后面读取文件什么的就可以直接读取不需要那么长的路径</span></span><br><span class="line">setwd<span class="punctuation">(</span><span class="string">&#x27;D:/test_data&#x27;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 加载包</span></span><br><span class="line">rm<span class="punctuation">(</span><span class="built_in">list</span><span class="operator">=</span>ls<span class="punctuation">(</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>topGO<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>Rgraphviz<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 设置输入文件</span></span><br><span class="line">input<span class="operator">=</span><span class="string">&quot;genes.list&quot;</span>  <span class="comment">#待分析基因名称的列表</span></span><br><span class="line">mapfile<span class="operator">=</span><span class="string">&quot;sample.anno&quot;</span>    <span class="comment">#所有基因GO注释结果</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 开始分析</span></span><br><span class="line">gene_id <span class="operator">=</span> readMappings<span class="punctuation">(</span>file <span class="operator">=</span> mapfile<span class="punctuation">)</span> <span class="comment">#如果是读取其他文件格式，后面参数还需要修改</span></span><br><span class="line">gene_names <span class="operator">=</span> <span class="built_in">names</span><span class="punctuation">(</span>gene_id<span class="punctuation">)</span></span><br><span class="line">my_genes <span class="operator">=</span> read.table<span class="punctuation">(</span>input<span class="punctuation">)</span><span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span></span><br><span class="line"></span><br><span class="line">gene_list <span class="operator">=</span> <span class="built_in">rep</span><span class="punctuation">(</span><span class="number">1</span><span class="punctuation">,</span><span class="built_in">length</span><span class="punctuation">(</span>gene_id<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>gene_list<span class="punctuation">)</span> <span class="operator">=</span> <span class="built_in">names</span><span class="punctuation">(</span>gene_id<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">gene_list<span class="punctuation">[</span>match<span class="punctuation">(</span>my_genes<span class="punctuation">,</span><span class="built_in">names</span><span class="punctuation">(</span>gene_list<span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">]</span> <span class="operator">=</span> <span class="number">0</span></span><br><span class="line">top_diff_genes <span class="operator">=</span> <span class="keyword">function</span><span class="punctuation">(</span>allScore<span class="punctuation">)</span><span class="punctuation">&#123;</span><span class="built_in">return</span><span class="punctuation">(</span>allScore<span class="operator">&lt;</span><span class="number">0.01</span><span class="punctuation">)</span><span class="punctuation">&#125;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 1. BP 富集分析</span></span><br><span class="line"><span class="comment">## new() 创建一个 topGO 的对象，然后对这个对象做检验</span></span><br><span class="line">bp_go <span class="operator">=</span> new<span class="punctuation">(</span><span class="string">&quot;topGOdata&quot;</span><span class="punctuation">,</span></span><br><span class="line">                   nodeSize <span class="operator">=</span> <span class="number">6</span><span class="punctuation">,</span></span><br><span class="line">                   ontology<span class="operator">=</span><span class="string">&quot;BP&quot;</span><span class="punctuation">,</span></span><br><span class="line">                   allGenes <span class="operator">=</span> gene_list<span class="punctuation">,</span></span><br><span class="line">                   annot <span class="operator">=</span> annFUN.gene2GO<span class="punctuation">,</span></span><br><span class="line">                   gene2GO <span class="operator">=</span> gene_id<span class="punctuation">,</span></span><br><span class="line">                   geneSel<span class="operator">=</span>top_diff_genes<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 做显著性检验，使用的是elim 的算法，使用 ks 的统计量。可以理解为 p 值</span></span><br><span class="line">result_KS.elim <span class="operator">=</span> runTest<span class="punctuation">(</span>bp_go<span class="punctuation">,</span></span><br><span class="line">                         algorithm <span class="operator">=</span> <span class="string">&quot;elim&quot;</span><span class="punctuation">,</span></span><br><span class="line">                         statistic <span class="operator">=</span> <span class="string">&quot;ks&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 提取基因 table</span></span><br><span class="line">allres <span class="operator">=</span> GenTable<span class="punctuation">(</span>bp_go<span class="punctuation">,</span></span><br><span class="line">                  KS <span class="operator">=</span> result_KS.elim<span class="punctuation">,</span></span><br><span class="line">                  ranksOf <span class="operator">=</span> <span class="string">&quot;classic&quot;</span><span class="punctuation">,</span></span><br><span class="line">                  topNodes <span class="operator">=</span> <span class="built_in">attributes</span><span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="operator">$</span>geneData<span class="punctuation">[</span><span class="number">4</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 生成文件，后面画图都可以用这个表</span></span><br><span class="line">write.table<span class="punctuation">(</span>allres<span class="punctuation">,</span></span><br><span class="line">            file <span class="operator">=</span> paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.BP.xls&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">            sep<span class="operator">=</span><span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> <span class="built_in">quote</span><span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">,</span> col.names<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span> row.names<span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出矢量图</span></span><br><span class="line">pdf<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.BP.pdf&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>bp_go<span class="punctuation">,</span></span><br><span class="line">               score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">               firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span></span><br><span class="line">               useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span> <span class="comment">#设置节点数量，10个或者20个更多都可以</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出像素图</span></span><br><span class="line">png<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.BP.png&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>bp_go<span class="punctuation">,</span> score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span> firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span> useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. MF 富集分析（同理）</span></span><br><span class="line"><span class="comment">## 创建一个 topGO 的对象</span></span><br><span class="line">mf_go <span class="operator">=</span> new<span class="punctuation">(</span><span class="string">&quot;topGOdata&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    nodeSize <span class="operator">=</span> <span class="number">6</span><span class="punctuation">,</span></span><br><span class="line">                    ontology<span class="operator">=</span><span class="string">&quot;MF&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    allGenes <span class="operator">=</span> gene_list<span class="punctuation">,</span></span><br><span class="line">                    annot <span class="operator">=</span> annFUN.gene2GO<span class="punctuation">,</span></span><br><span class="line">                    gene2GO <span class="operator">=</span> gene_id<span class="punctuation">,</span></span><br><span class="line">                    geneSel<span class="operator">=</span>top_diff_genes<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 做检验，使用的是elim 的算法，使用 ks 的统计量。可以理解为 p 值</span></span><br><span class="line">result_KS.elim <span class="operator">=</span> runTest<span class="punctuation">(</span>mf_go<span class="punctuation">,</span></span><br><span class="line">                         algorithm <span class="operator">=</span> <span class="string">&quot;elim&quot;</span><span class="punctuation">,</span></span><br><span class="line">                         statistic <span class="operator">=</span> <span class="string">&quot;ks&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 提取基因 table</span></span><br><span class="line">allres <span class="operator">=</span> GenTable<span class="punctuation">(</span>mf_go <span class="punctuation">,</span></span><br><span class="line">                  KS <span class="operator">=</span> result_KS.elim<span class="punctuation">,</span></span><br><span class="line">                  ranksOf <span class="operator">=</span> <span class="string">&quot;classic&quot;</span><span class="punctuation">,</span></span><br><span class="line">                  topNodes <span class="operator">=</span> <span class="built_in">attributes</span><span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="operator">$</span>geneData<span class="punctuation">[</span><span class="number">4</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 生成文件，后面画图都可以用这个表</span></span><br><span class="line">write.table<span class="punctuation">(</span>allres<span class="punctuation">,</span></span><br><span class="line">            file <span class="operator">=</span> paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.MF.xls&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">            sep<span class="operator">=</span><span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> <span class="built_in">quote</span><span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">,</span> col.names<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span> row.names<span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出矢量图</span></span><br><span class="line">pdf<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.MF.pdf&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>mf_go<span class="punctuation">,</span></span><br><span class="line">               score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">               firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span></span><br><span class="line">               useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span> <span class="comment">#设置节点数量，10个或者20个更多都可以</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出像素图</span></span><br><span class="line">png<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.MF.png&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>mf_go<span class="punctuation">,</span> score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span> firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span> useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. CC节点的富集分析(同理)</span></span><br><span class="line"><span class="comment">## 创建一个 topGO 的对象</span></span><br><span class="line">cc_go <span class="operator">=</span> new<span class="punctuation">(</span><span class="string">&quot;topGOdata&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    nodeSize <span class="operator">=</span> <span class="number">6</span><span class="punctuation">,</span></span><br><span class="line">                    ontology<span class="operator">=</span><span class="string">&quot;CC&quot;</span><span class="punctuation">,</span></span><br><span class="line">                    allGenes <span class="operator">=</span> gene_list<span class="punctuation">,</span></span><br><span class="line">                    annot <span class="operator">=</span> annFUN.gene2GO<span class="punctuation">,</span></span><br><span class="line">                    gene2GO <span class="operator">=</span> gene_id<span class="punctuation">,</span></span><br><span class="line">                    geneSel<span class="operator">=</span>top_diff_genes<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 做检验，使用的是elim 的算法，使用 ks 的统计量。可以理解为 p 值</span></span><br><span class="line">result_KS.elim <span class="operator">=</span> runTest<span class="punctuation">(</span>cc_go<span class="punctuation">,</span></span><br><span class="line">                         algorithm <span class="operator">=</span> <span class="string">&quot;elim&quot;</span><span class="punctuation">,</span></span><br><span class="line">                         statistic <span class="operator">=</span> <span class="string">&quot;ks&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 提取基因 table</span></span><br><span class="line">allres <span class="operator">=</span> GenTable<span class="punctuation">(</span>cc_go<span class="punctuation">,</span></span><br><span class="line">                  KS <span class="operator">=</span> result_KS.elim<span class="punctuation">,</span></span><br><span class="line">                  ranksOf <span class="operator">=</span> <span class="string">&quot;classic&quot;</span><span class="punctuation">,</span></span><br><span class="line">                  topNodes <span class="operator">=</span> <span class="built_in">attributes</span><span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="operator">$</span>geneData<span class="punctuation">[</span><span class="number">4</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 生成文件，后面画图都可以用这个表</span></span><br><span class="line">write.table<span class="punctuation">(</span>allres<span class="punctuation">,</span></span><br><span class="line">            file <span class="operator">=</span> paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.CC.xls&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">            sep<span class="operator">=</span><span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> <span class="built_in">quote</span><span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">,</span> col.names<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span> row.names<span class="operator">=</span><span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出矢量图</span></span><br><span class="line">pdf<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.CC.pdf&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>cc_go<span class="punctuation">,</span></span><br><span class="line">               score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">               firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span></span><br><span class="line">               useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span> <span class="comment">#设置节点数量，10个或者20个更多都可以</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 输出像素图</span></span><br><span class="line">png<span class="punctuation">(</span>paste<span class="punctuation">(</span>input<span class="punctuation">,</span><span class="string">&quot;.CC.png&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">showSigOfNodes<span class="punctuation">(</span>cc_go<span class="punctuation">,</span> score<span class="punctuation">(</span>result_KS.elim<span class="punctuation">)</span><span class="punctuation">,</span> firstSigNodes <span class="operator">=</span> <span class="number">10</span><span class="punctuation">,</span> useInfo <span class="operator">=</span> <span class="string">&quot;all&quot;</span><span class="punctuation">)</span></span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h3 id="2-5-3-结果解释"><a href="#2-5-3-结果解释" class="headerlink" title="2.5.3. 结果解释"></a>2.5.3. 结果解释</h3><ol>
<li>.xls结果文件中每一列的含义</li>
</ol>
<ul>
<li><p>GO.ID：富集的GO ID</p>
</li>
<li><p>Term：GO ID的描述</p>
</li>
<li><p>Annotated : number of genes in go.db which are annotated with the GO-term.在go.db中被注释到GO-term的基因数量。</p>
</li>
<li><p>Significant : number of genes belonging to your input which are annotated with the GO-term. GO-term被注释到的基因中包含输入的基因的数量</p>
</li>
<li><p>Expected : show an estimate of the number of genes a node of size Annotated would have if the significant genes were to be randomly selected from the gene universe. 对节点基因数量的预期</p>
</li>
<li><p>KS：用KS（Kolmogorov-Smirnov）算法计算得到的p-value值。</p>
<p>KS全称是：Kolmogorov-Smirnov，KS值是通过KS检验所得，KS检验是一种算法。统计方法如下：</p>
<ul>
<li>首先计算每个go节点对应的gene个数，</li>
<li>如果某个节点的子节点也有gene比对上，那么父节点对应的gene个数也要加上子节点的基因数</li>
<li>使用KS统计检验进行p值的计算，文件中的KS值，就是常说的p value，叫KS值的原因，是体现使用的KS检验方法。</li>
</ul>
</li>
</ul>
<ol start="2">
<li>DAG图（矢量图&#x2F;像素图）<br>topGO有向无环图(Directed acyclic graph, DAG)能直观展示差异表达基因富集的GO节点（Term）及其层级关系，是差异表达基因GO富集分析的结果图形化展示，分支代表包含关系，从上至下所定义的功能描述范围越来越具体。在有向无环图中，箭头代表包含关系，即该节点的所有基因同样注释到其上级节点中。</li>
</ol>
<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/fe/Tred-G.svg/800px-Tred-G.svg.png" width=40% title="DAG图" alt="DAG图" align=center/>

<p><strong><p align="center">Figure 1. DAG图示例</strong><br>from <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Directed_acyclic_graph">wikipedia:directed acyclic graph</a></p></p>
<p>注：对每个GO节点进行富集，在图中用方框表示显著度最高的10个节点，图中还包含其各层对应关系。每个方框（或椭圆）内给出了该GO节点的内容描述和富集显著性值。不同颜色代表不同的富集显著性，颜色越深，显著性越高。</p>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><p><a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf">topGO tutorial</a><br><a target="_blank" rel="noopener" href="https://datacatz.wordpress.com/2018/01/19/gene-set-enrichment-analysis-with-topgo-part-1/">topGO blog</a><br><a target="_blank" rel="noopener" href="https://www.codenong.com/cs105162324/">R topGO</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/13/bioinfo_GSEA_topGO/" data-id="cl1aiqcr6001lm4vk2fy85qw0" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_GSEA" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/12/bioinfo_GSEA/" class="article-date">
  <time datetime="2021-11-12T08:00:00.000Z" itemprop="datePublished">2021-11-12</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/12/bioinfo_GSEA/">基因富集分析(gene set enrichment analysis, GSEA)</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe width="298" height="52" src="https://www.youtube.com/embed/wcOM3Rx43ko" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe></div>

<h1 id="1-基因富集分析-gene-set-enrichment-analysis-GSEA"><a href="#1-基因富集分析-gene-set-enrichment-analysis-GSEA" class="headerlink" title="1. 基因富集分析(gene set enrichment analysis, GSEA)"></a>1. 基因富集分析(gene set enrichment analysis, GSEA)</h1><h2 id="1-1-定义"><a href="#1-1-定义" class="headerlink" title="1.1. 定义"></a>1.1. 定义</h2><p>基因富集分析(gene set enrichment analysis, GSEA)，或者功能富集分析(functional enrichment analysis)是对大量的序列(基因或者蛋白质)进行按照基因功能或通路等分组，并鉴定哪些组包含比背景比例更多的基因(过度表达，over-represented)或更少的基因(表达不足，under-represented)的一种统计学方法。</p>
<h2 id="1-2-方法"><a href="#1-2-方法" class="headerlink" title="1.2. 方法"></a>1.2. 方法</h2><ol>
<li>通过将输入基因集和先验基因集(比如GO,KEGG)中的每个类别进行比较，来鉴定先验基因集中的哪一类基因集富含输入基因，即哪一类基因被富集。</li>
<li>通常有三个步骤</li>
</ol>
<ul>
<li>计算富集分数(enrichment score,ES)</li>
<li>估计ES的统计显著性</li>
<li>调整多重假设检验，归一化ES，计算错误发现率。<h2 id="1-3-应用"><a href="#1-3-应用" class="headerlink" title="1.3. 应用"></a>1.3. 应用</h2>通常会使用其他分析的结果作为输入基因集，比如比较基因组分析中得到的某物种特有的基因集，比较转录组分析得到的差异表达基因集，基因家族收缩扩张分析得到的基因组中扩张的基因集，基因组共线性分析中在全基因组复制事件附近的Ks值的基因集等各种分析得到的基因集，做基因富集分析来查看这些基因集是否主要集中在某些类别，这些类别代表的功能是否与表型或者进化事件有关联。</li>
</ul>
<h1 id="2-基因富集分析常用基因集"><a href="#2-基因富集分析常用基因集" class="headerlink" title="2. 基因富集分析常用基因集"></a>2. 基因富集分析常用基因集</h1><h2 id="2-1-Gene-Ontology-GO"><a href="#2-1-Gene-Ontology-GO" class="headerlink" title="2.1. Gene Ontology(GO)"></a>2.1. Gene Ontology(GO)</h2><h3 id="2-1-1-GO"><a href="#2-1-1-GO" class="headerlink" title="2.1.1. GO"></a>2.1.1. GO</h3><p>GO是一个国际标准化的基因功能分类体系，由基因本体联合会(Gene Ontology Consortium，GOC) 负责。它提供了一套动态并可控的词汇表（controlled vocabulary）来全面描述生物体中基因和基因产物的属性，它由一组预先定义好的GO术语（GO term）组成，这组术语对基因产物的功能进行限定和描述。</p>
<p>GO由三个ontology（本体）组成，是由独立的术语表示的，分别描述基因的细胞组分（cellular component，CC）、分子功能（molecular function，MF）、参与的生物过程（biological process，BP）。</p>
<p>GO这三个本体的含义：</p>
<ol>
<li>细胞组成（cellular component，CC）：描述基因产物执行功能的细胞结构相关的位置，比如一个蛋白可能定位在细胞核中，也可能定位在核糖体中；</li>
<li>分子功能（Molecular Function，MF）：描述基因产物发生在分子水平上的活动，例如催化或运输。通常对应于单个基因产物（即蛋白质或RNA）可以进行的活动。常见的宽泛的分子功能描述是催化活性(catalytic activity)和转运活动(transporter activity)。为了避免与基因产物名称混淆，通常分子功能描述后加上”activity”一词。</li>
<li>生物过程（biological process，BP）：描述的是指基因产物所关联的一个大的生物功能，或者说是多个分子活动完成的一个大的生物程序。例如有丝分裂或嘌呤代谢；</li>
</ol>
<h3 id="2-1-2-GO-terms"><a href="#2-1-2-GO-terms" class="headerlink" title="2.1.2. GO terms"></a>2.1.2. GO terms</h3><p>GO terms，它提供生物过程的逻辑结构与相关关系，不同GO terms之间的关系可以通过一个有向无环图来表示。</p>
<p>此处需要注意的是，GO terms是对<strong>基因产物</strong>，而不是基因本身进行描述，因为基因本身的产物有时候不止一种。GO数据库中的GO分类相关信息会得到不断地更新与增加，这个特点要记住，因为不同的GO分析工具使用的数据库版本有可能不一样，造成GO分析结果出现不同。</p>
<h3 id="2-1-3-GO-annotations"><a href="#2-1-3-GO-annotations" class="headerlink" title="2.1.3. GO annotations"></a>2.1.3. GO annotations</h3><p>GO注释（GO annotations）是关于特定基因功能的声明，它主要是将GO terms和基因或基因产物相关联来提供注释，也就是描述这个GO terms关联的基因产物是什么（蛋白质，还是非编码RNA，还是大分子等），有什么功能，如何在分子水平发挥作用，在细胞中的哪个位置发挥作用，以及它有助于执行哪些生物过程(途径、程序)。</p>
<h2 id="2-2-Kyoto-Encyclopedia-of-Genes-and-Genomes-KEGG"><a href="#2-2-Kyoto-Encyclopedia-of-Genes-and-Genomes-KEGG" class="headerlink" title="2.2. Kyoto Encyclopedia of Genes and Genomes(KEGG)"></a>2.2. Kyoto Encyclopedia of Genes and Genomes(KEGG)</h2><h3 id="2-2-1-KEGG"><a href="#2-2-1-KEGG" class="headerlink" title="2.2.1. KEGG"></a>2.2.1. KEGG</h3><p>KEGG是处理基因组，生物通路，疾病，药物，化学物质的数据库集合，于1995年由京都大学化学研究所教授Minoru Kanehisa在当时正在进行的日本人类基因组计划下发起。</p>
<p>KEGG 是一种数据库资源，用于从基因组和分子级信息了解生物系统（例如细胞、生物体和生态系统）的高级功能和效用。它是生物系统的计算机表示，由基因和蛋白质（基因组信息）和化学物质（化学信息）的分子构建块组成，它们与相互作用、反应和关系网络（系统信息）的分子接线图知识相结合。它还包含疾病和药物信息（健康信息）作为对生物系统的扰动。</p>
<h3 id="2-2-2-KEGG-Database"><a href="#2-2-2-KEGG-Database" class="headerlink" title="2.2.2. KEGG Database"></a>2.2.2. KEGG Database</h3><p>KEGG 是一个集成的数据库资源，由如下所示的 16 个数据库组成。它们大致分为系统信息、基因组信息、化学信息和健康信息，它们通过网页的颜色编码来区分。</p>
<table>
<caption><h4>KEGG Database</h4></caption>
<thead>
<tr>
<th>Category</th>
<th>Database</th>
<th>Content</th>
<th>Color</th>
</tr>
</thead>
</tbody>
<tr>
<td rowspan="3">Systems information</td>
<td>KEGG PATHWAY</td>
<td>KEGG pathway maps</td>
<td rowspan="3"><font color=green>kegg3</br>green</font></td>
</tr>
<tr>
<td>KEGG BRITE</td>
<td>BRITE hierarchies and tables</td>
</tr>
<tr>
<td>KEGG MODULE</td>
<td>KEGG modules and reaction modules</td>
</tr>
<tr>
<td rowspan="3">Genomic information</td>
<td>KEGG ORTHOLOGY (KO)</td>
<td>Functional orthologs</td>
<td><font color=yellow>kegg4 yellow</font></td>
</tr>
<tr>
<td>KEGG GENES</td>
<td>Genes and proteins</td>
<td rowspan="2"><font color=red>kegg1</br>red</font></td>
</tr>
<tr>
<td>KEGG GENOME</td>
<td>KEGG organisms and viruses</td>
</tr>
<tr>
<td rowspan="4">Chemical information</td>
<td>KEGG COMPOUND</td>
<td>Metabolites and other chemical substances</td>
<td rowspan="4"><font color=blue>kegg2</br>blue</font></td>
</tr>
<tr>
<td>KEGG GLYCAN</td>
<td>Glycans</td>
</tr>
<tr>
<td>KEGG REACTION</br>KEGG RCLASS</td>
<td>Biochemical reactions</br>Reaction class</td>
</tr>
<tr>
<td>KEGG ENZYME</td>
<td>Enzyme nomenclature</td>
</tr>
<tr>
<td rowspan="4">Health information</td>
<td>KEGG NETWORK</td>
<td>Disease-related network variations</td>
<td rowspan="4"><font color=purple>kegg5</br>purle</font></td>
</tr>
<tr>
<td>KEGG VARIANT</td>
<td>Human gene variants</td>
</tr>
<tr>
<td>KEGG DISEASE</td>
<td>Human diseases</td>
</tr>
<tr>
<td>KEGG DRUG</br>KEGG DGROUP</td>
<td>Drugs</br>Drug groups</td>
</tr>
</tbody>
</table>

<h3 id="2-2-3-KEGG-PATHWAY-Database"><a href="#2-2-3-KEGG-PATHWAY-Database" class="headerlink" title="2.2.3. KEGG PATHWAY Database"></a>2.2.3. KEGG PATHWAY Database</h3><p>KEGG PATHWAY Database是KEGG资源的核心，是一组手工绘制的KEGG通路图，代表细胞和生物体的新陈代谢和各种其他功能的实验知识。每个通路图都包含一个分子相互作用和反应网络，旨在将基因组中的基因与通路中的基因产物（主要是蛋白质）联系起来。</p>
<h1 id="3-基因富集分析程序"><a href="#3-基因富集分析程序" class="headerlink" title="3. 基因富集分析程序"></a>3. 基因富集分析程序</h1><p>目前有许多程序可以用于基因富集分析。包括NASQAR,PlantRegMap,MSigDB,Broad Institute,WebGestalt,Enrichr,GeneSCF,DAVID,Metascape,AmiGO 2,GREAT,FunRich,FuncAssociate,InterMine,ToppGene Suite,QuSAGE,Blast2GO,g:Profiler。</p>
<p>可以根据需要分析的物种类别和流行度选择分析平台，根据需要选择先验基因集。</p>
<p>下面介绍几种常见的。</p>
<h2 id="3-1-NASQAR"><a href="#3-1-NASQAR" class="headerlink" title="3.1. NASQAR"></a>3.1. NASQAR</h2><p>NASQAR (Nucleic Acid SeQuence Analysis Resource)是一个开源的网页平台，可以用R包clusterProfiler做GSEA分析，支持<a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">Org.Db数据库</a>的所有物种的GO Term和KEGG Pathway富集分析。</p>
<h2 id="3-2-PlantRegMap"><a href="#3-2-PlantRegMap" class="headerlink" title="3.2. PlantRegMap"></a>3.2. PlantRegMap</h2><p>支持165种植物的GO注释和GO富集分析。</p>
<h2 id="3-3-Enrichr"><a href="#3-3-Enrichr" class="headerlink" title="3.3. Enrichr"></a>3.3. Enrichr</h2><p>Enrichr是针对哺乳动物的基因富集分析工具。可以通过API使用，并提供可视化结果。</p>
<h2 id="3-4-GeneSCF"><a href="#3-4-GeneSCF" class="headerlink" title="3.4. GeneSCF"></a>3.4. GeneSCF</h2><p>GeneSCF支持多个物种，实时的功能富集分析工具。不需要额外更新数据库，GeneSCF是实时最新的数据库，并且支持多物种富集分析。结果以文本呈现。</p>
<h2 id="3-5-DAVID"><a href="#3-5-DAVID" class="headerlink" title="3.5. DAVID"></a>3.5. DAVID</h2><p>DAVID是做注释，富集和可视化的整合的数据库。但注释数据库自从2016年10月就没有更新。</p>
<h2 id="3-6-Blast2GO"><a href="#3-6-Blast2GO" class="headerlink" title="3.6. Blast2GO"></a>3.6. Blast2GO</h2><p>Blast2GO可以做组学数据的功能注释和GSEA分析。</p>
<h2 id="3-7-KOBAS-i"><a href="#3-7-KOBAS-i" class="headerlink" title="3.7. KOBAS-i"></a>3.7. KOBAS-i</h2><h3 id="3-7-1-KOBAS-i简介"><a href="#3-7-1-KOBAS-i简介" class="headerlink" title="3.7.1. KOBAS-i简介"></a>3.7.1. KOBAS-i简介</h3><ol>
<li>KOBAS是中科院和北大联合开发的做功能注释和GSEA分析的工具。</li>
<li>目前已经发表了第三个版本，KOBAS-intelligence(KOBAS-i)，KOBAS-i引入了之前发布的基于机器学习的新方法CGPS，并扩展了可视化功能，支持的物种增加到5944个。</li>
<li>KOBAS包含注释模块和富集模块。</li>
</ol>
<ul>
<li>注释模块接受基因列表作为输入，包括 ID 或序列，并根据通路、疾病和GO信息等多个数据库为每个基因生成注释。</li>
<li>富集模块给出了关于哪些通路和 GO 术语与输入基因列表或表达在统计上显着相关的结果。有两种不同的富集分析可用，命名为基因列表富集和 exp-data 富集。</li>
</ul>
<ol start="4">
<li>KOBAS-i有网页版，也有本地版。</li>
</ol>
<h3 id="3-7-2-KOBAS-i网页版【推荐】"><a href="#3-7-2-KOBAS-i网页版【推荐】" class="headerlink" title="3.7.2. KOBAS-i网页版【推荐】"></a>3.7.2. KOBAS-i网页版【推荐】</h3><p>给了两个可用的网址：<a target="_blank" rel="noopener" href="http://kobas.cbi.pku.edu.cn/%EF%BC%8Chttp://bioinfo.org/kobas%E3%80%82">http://kobas.cbi.pku.edu.cn/，http://bioinfo.org/kobas。</a><br><a target="_blank" rel="noopener" href="http://kobas.cbi.pku.edu.cn/">KOBAS-i</a>。</p>
<h4 id="3-7-2-1-注释-annotation-和富集-Enrichment-的步骤"><a href="#3-7-2-1-注释-annotation-和富集-Enrichment-的步骤" class="headerlink" title="3.7.2.1. 注释(annotation)和富集(Enrichment)的步骤"></a>3.7.2.1. 注释(annotation)和富集(Enrichment)的步骤</h4><ol>
<li>输入<br>注释只有三个输入项：</li>
</ol>
<ul>
<li>选择物种。目前支持5944个物种。</li>
<li>选择输入基因集数据类型。支持核苷酸或者蛋白质的fasta序列，blast的Tabular输出结果，Ensembl Gene ID，Entrez Gene ID，UniProtKB AC，Gene Symbol。</li>
<li>输入基因集。可以粘贴文本，也可以上传文件。</li>
</ul>
<p>富集除了需要上面三个输入项外，还需要选择用哪写先验基因集的数据库来做富集分析。包括PATHWAY,DISEASE,GO。其中KEGG Pathway是所有物种都可以用，其他的只有部份物种可以选。</p>
<ul>
<li>PATHWAY。有四个选项，KEGG Pathway(K),Reactome(R),BioCyc(B),PANTHER(p)。</li>
<li>DISEASE。有三个选项，OMIM(o),KEGG Disease(k),NHGRI GWAS Catalog(N)</li>
<li>GO(G)。<br>此外还有高级选项，可以选择统计学方法和纠错方法。</li>
</ul>
<p>然后点击Run，就可以等待运行结果了。</p>
<ol start="2">
<li>结果<br>运行结束后，点击右上角的Download total terms就可以下载到结果。<br>富集分析还可以点击Visualization得到结果的可视化图。</li>
</ol>
<h2 id="3-8-GOEAST"><a href="#3-8-GOEAST" class="headerlink" title="3.8. GOEAST"></a>3.8. GOEAST</h2><p>在线工具<a target="_blank" rel="noopener" href="http://omicslab.genetics.ac.cn/GOEAST/index.php">GOEAST</a></p>
<h1 id="4-富集分析的R包"><a href="#4-富集分析的R包" class="headerlink" title="4. 富集分析的R包"></a>4. 富集分析的R包</h1><p>常见的有topGO，clusterProfiler，有一些进行富集分析的程序使用了这些包。</p>
<h2 id="4-1-topGO"><a href="#4-1-topGO" class="headerlink" title="4.1. topGO"></a>4.1. topGO</h2><p>用topGO做GSEA分析的具体教程可以查看博文<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/11/13/bioinfo_GSEA_topGO/">blog_topGO</a>。</p>
<p>topGO是一个R包，用于半自动的GO terms的基因富集分析。topGO的结果可以展示为有向无环图。</p>
<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/fe/Tred-G.svg/800px-Tred-G.svg.png" width=40% title="DAG图" alt="DAG图" align=center/>

<p><strong><p align="center">Figure 1. DAG图示例</strong><br>from <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Directed_acyclic_graph">wikipedia:directed acyclic graph</a></p></p>
<p>注：对每个GO节点进行富集，在图中用方框表示显著度最高的10个节点，图中还包含其各层对应关系。每个方框（或椭圆）内给出了该GO节点的内容描述和富集显著性值。不同颜色代表不同的富集显著性，颜色越深，显著性越高。</p>
<h2 id="4-2-clusterProfiler"><a href="#4-2-clusterProfiler" class="headerlink" title="4.2. clusterProfiler"></a>4.2. clusterProfiler</h2><p>clusterProfiler包的具体使用参考博文<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/12/13/bioinfo_GSEA_clusterProfiler/">blog_clusterProfiler</a>。</p>
<h3 id="4-2-1-clusterProfiler"><a href="#4-2-1-clusterProfiler" class="headerlink" title="4.2.1. clusterProfiler"></a>4.2.1. clusterProfiler</h3><p>clusterProfiler是一个R包，是一个解释组学数据的通用富集工具，支持Gene Ontology(GO), Kyoto Encyclopedia of Genes and Genomes(KEGG), Disease Ontology(DO), Disease Gene Network(DisGeNET), Molecular Signatures Database(MSigDb), wikiPathways和许多其他的基因集的功能注释和富集分析，以及富集分析结果的可视化。2021年07月发布了clusterProfiler 4.0版本。</p>
<h3 id="4-2-2-clusterProfiler支持的基因集-gene-sets"><a href="#4-2-2-clusterProfiler支持的基因集-gene-sets" class="headerlink" title="4.2.2. clusterProfiler支持的基因集(gene sets)"></a>4.2.2. clusterProfiler支持的基因集(gene sets)</h3><ol>
<li>Gene Ontology(GO)</li>
<li>Kyoto Encyclopedia of Genes and Genomes(KEGG)</li>
<li>Disease Ontology(DO)</li>
<li>Disease Gene Network(DisGeNET)</li>
<li>Molecular Signatures Database(MSigDb)</li>
<li>wikiPathways</li>
</ol>
<h3 id="4-2-3-clusterProfiler功能-——-enrichment-analysis"><a href="#4-2-3-clusterProfiler功能-——-enrichment-analysis" class="headerlink" title="4.2.3. clusterProfiler功能 —— enrichment analysis"></a>4.2.3. clusterProfiler功能 —— enrichment analysis</h3><ol>
<li>Over Representation Analysis, ORA<br>ORA是用于判断已知的生物功能或过程在实验产生的基因列表（例如差异表达基因列表, differentially expressed genes, DEGs）中是否过表达(over-represented&#x3D;enriched)的常用方法。</li>
<li>Gene Set Enrichment Analysis, GSEA</li>
<li>Leading edge analysis and core enriched genes</li>
</ol>
<h1 id="5-references"><a href="#5-references" class="headerlink" title="5. references"></a>5. references</h1><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Gene_set_enrichment_analysis">GSEA wiki</a><br><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247484456&idx=1&sn=bbcd0b5d10ba9312d92b7baae777ccde&scene=21#wechat_redirect">GOEAST introduction</a><br><a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf">topGO tutorial</a><br><a target="_blank" rel="noopener" href="https://datacatz.wordpress.com/2018/01/19/gene-set-enrichment-analysis-with-topgo-part-1/">topGO blog</a><br><a target="_blank" rel="noopener" href="https://www.codenong.com/cs105162324/">R topGO</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/47b5ea646932">enrichment</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/7177c372243f">GO explanation</a><br><a target="_blank" rel="noopener" href="http://geneontology.org/docs/ontology-documentation/">GO overview</a><br><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/KEGG">KEGG</a><br><a target="_blank" rel="noopener" href="https://github.com/YuLab-SMU/clusterProfiler">clusterProfiler github</a><br><a target="_blank" rel="noopener" href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">universal enrichment analysis using clusterProfiler</a><br><a target="_blank" rel="noopener" href="https://www.cell.com/the-innovation/fulltext/S2666-6758(21)00066-7?_returnURL=https://linkinghub.elsevier.com/retrieve/pii/S2666675821000667?showall=true">clusterProfiler paper</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/12/bioinfo_GSEA/" data-id="cl1aiqcr4001fm4vkbz9t0vvq" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_MCscan" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/11/05/bioinfo_MCscan/" class="article-date">
  <time datetime="2021-11-05T07:10:00.000Z" itemprop="datePublished">2021-11-05</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/11/05/bioinfo_MCscan/">用MCscan做基因组间的同线性分析</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=5250805&auto=1&height=32"></iframe></div>

<h1 id="1-MCscan背景"><a href="#1-MCscan背景" class="headerlink" title="1. MCscan背景"></a>1. MCscan背景</h1><h2 id="1-1-jcvi介绍"><a href="#1-1-jcvi介绍" class="headerlink" title="1.1. jcvi介绍"></a>1.1. jcvi介绍</h2><p><a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi">jcvi</a>是美国佐治亚大学唐海宝开发的一个生物信息学python库。主要有几个常见pipeline用到了jcvi库，一个是做基因组scaffoldding的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/ALLMAPS">ALLMAPS</a>，另一个是做基因组同线性分析的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan</a>，以及提取植物表型数据的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/GRABSEEDS">GRABSEEDS</a>。</p>
<h2 id="1-2-MCscan介绍"><a href="#1-2-MCscan介绍" class="headerlink" title="1.2. MCscan介绍"></a>1.2. MCscan介绍</h2><p>用MCscan pipeline做两个基因组间的同线性分析，包括成对同线性(pairwise synteny)分析，宏观同线性(macrosynteny)分析，微观同线性(microsynteny)分析和结果的可视化(visualization)。</p>
<p>成对同线性分析主要是制作同线性点图(dotplot),同线性深度直方图；宏观同线性分析主要是制作同线性染色体图。其中同线性深度直方图，同线性染色体图需要用到同线性点图的结果文件作为输入。</p>
<p>微观同线性分析等后面用到再补坑。</p>
<h1 id="2-MCscan使用"><a href="#2-MCscan使用" class="headerlink" title="2. MCscan使用"></a>2. MCscan使用</h1><h2 id="2-1-安装jcvi"><a href="#2-1-安装jcvi" class="headerlink" title="2.1. 安装jcvi"></a>2.1. 安装jcvi</h2><h3 id="2-1-1-依赖"><a href="#2-1-1-依赖" class="headerlink" title="2.1.1. 依赖"></a>2.1.1. 依赖</h3><p>jcvi依赖包括biopython,numpy,matplotlib。</p>
<p>MCscan依赖包括LASTAL。</p>
<p>LASTAL可以用<code>conda install -c bioconda last</code>安装，但我用conda安装的lastal和lastdb不能多线程运行，所以在运行<code>python -m jcvi.compara.catalog ortholog --cpu=1</code>时用参数<code>--cpu=1</code>来限制单线程运行。</p>
<h3 id="2-1-2-安装jcvi"><a href="#2-1-2-安装jcvi" class="headerlink" title="2.1.2. 安装jcvi"></a>2.1.2. 安装jcvi</h3><p>任意选一种安装方式</p>
<ol>
<li><code>pip install jcvi</code></li>
<li><code>pip install git+git://github.com/tanghaibao/jcvi.git</code></li>
<li><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cd jcvi</span><br><span class="line">git clone git://github.com/tanghaibao/jcvi.git</span><br><span class="line">pip install -e .</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="2-2-准备输入文件"><a href="#2-2-准备输入文件" class="headerlink" title="2.2. 准备输入文件"></a>2.2. 准备输入文件</h2><p>输入文件包括两个基因组的CDS&#x2F;氨基酸序列文件（fa格式），两个基因组的注释坐标文件（bed格式）。</p>
<h3 id="2-2-1-jcvi支持直接从phytozome下载数据"><a href="#2-2-1-jcvi支持直接从phytozome下载数据" class="headerlink" title="2.2.1. jcvi支持直接从phytozome下载数据"></a>2.2.1. jcvi支持直接从phytozome下载数据</h3><ol>
<li><p>连接phytozome（需要登录，提前注册账号）<br><code>python -m jcvi.apps.fetch phytozome</code> 输入账户登录信息后，出现可以下载的基因组list。</p>
</li>
<li><p>下载CDS序列和注释文件(gff格式)，例如下载这两个物种:grape and peach<br><code>python -m jcvi.apps.fetch phytozome Vvinifera,Ppersica</code></p>
</li>
<li><p>转换GFF为BED格式<br><code>python -m jcvi.formats.gff bed --type=mRNA --key=Name  --primary_only Vvinifera_145_gene.gff3.gz -o grape.bed</code></p>
</li>
</ol>
<p><code>python -m jcvi.formats.gff bed --type=mRNA --key=Name  --primary_only Ppersica_139_gene.gff3.gz -o peach.bed</code></p>
<p>其中<code>--primary_only</code>是用于去除多个转录本的，只保留一个。（原文是多个isoforms）一般不去也不影响后续步骤，加上更好吧。</p>
<ol start="4">
<li>整理一下，获得CDS文件<br><code>python -m jcvi.formats.fasta format Vvinifera_145_Genoscope.12X.cds.fa.gz grape.cds</code></li>
</ol>
<p><code>python -m jcvi.formats.fasta format Ppersica_298_v2.1.cds.fa.gz peach.cds</code></p>
<h3 id="2-2-2-自己的数据"><a href="#2-2-2-自己的数据" class="headerlink" title="2.2.2. 自己的数据"></a>2.2.2. 自己的数据</h3><ol>
<li>gff2bed<br><code>python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only sampleA.gff3.gz &gt; sampleA.bed</code></li>
</ol>
<p><code>python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only sampleB.gff3.gz &gt; sampleB.bed</code></p>
<p>其中<code>--primary_only</code>是用于去除多个转录本的，只保留一个。（原文是多个isoforms）一般不去也不影响后续步骤，加上更好吧。</p>
<ol start="2">
<li><p>去重，会生成sampleA.uniq.bed和sampleB.uniq.bed，然后重命名</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">python -m jcvi.formats.bed uniq sampleA.bed</span><br><span class="line">mv sampleA.uniq.bed sampleA.bed</span><br><span class="line">python -m jcvi.formats.bed uniq sampleB.bed</span><br><span class="line">mv sampleB.uniq.bed sampleB.bed</span><br></pre></td></tr></table></figure>
</li>
<li><p>提取cds和pep</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">seqkit grep -f &lt;(cut -f 4 sampleA.uniq.bed ) sampleA.cdna.all.fa.gz | seqkit seq -i &gt; sampleA.cds</span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleA.uniq.bed ) sampleB.pep.all.fa.gz | seqkit seq -i &gt; sampleA.pep </span><br><span class="line"></span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleB.uniq.bed )  sampeB.cdna.fa.gz | seqkit seq -i  &gt; sampleB.cds</span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleB.uniq.bed ) sampleB.pep.fa.gz | seqkit seq -i  &gt; sampleB.pep</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="2-3-同线性点图"><a href="#2-3-同线性点图" class="headerlink" title="2.3. 同线性点图"></a>2.3. 同线性点图</h2><h3 id="2-3-1-CDS的同线性点图dotplot"><a href="#2-3-1-CDS的同线性点图dotplot" class="headerlink" title="2.3.1. CDS的同线性点图dotplot"></a>2.3.1. CDS的同线性点图dotplot</h3><ol>
<li>确保<code>sampleA.cds</code>,<code>sampleA.bed</code>,<code>sampleB.cds</code>,<code>sampleB.bed</code>四个文件在同一目录下。</li>
<li>运行<code>python -m jcvi.compara.catalog ortholog --no_strip_names --cpu=1 sampleA sampleB</code>制作同线性点图。</li>
</ol>
<p><code>--cpu=1</code>是因为conda安装的lastal不支持多线程，所以用单线程。</p>
<ol start="3">
<li>运行时如果报错<code>ERROR savefig failed. Reset usetex to False</code>是缺少dvipng包没能生成png格式图，但看看结果文件sampleA.sampleB.anchors还是完整生成了就可以进行下一步。</li>
</ol>
<h3 id="2-3-2-蛋白的同线性点图"><a href="#2-3-2-蛋白的同线性点图" class="headerlink" title="2.3.2. 蛋白的同线性点图"></a>2.3.2. 蛋白的同线性点图</h3><p>与CDS一样，cds数据换成pep数据，在运行参数里加上数据类型<code>--dbtype prot</code>即可。</p>
<p><code>python -m jcvi.compara.catalog ortholog --dbtype prot --no_strip_names --cpu=1 sampleA sampleB</code></p>
<p><code>--cpu=1</code>是因为conda安装的lastal不支持多线程，所以用单线程。</p>
<h3 id="2-3-3-结果"><a href="#2-3-3-结果" class="headerlink" title="2.3.3. 结果"></a>2.3.3. 结果</h3><p>会生成五个结果文件。</p>
<ul>
<li>sampleA.sampleB.last: 基于LAST的比对结果</li>
<li>sampleA.sampleB.last.filtered: LAST的比对结果过滤串联重复和低分比对</li>
<li>sampleA.sampleB.anchors: 高质量的同线性区块</li>
<li>sampleA.sampleB.lifted.anchors:增加了额外的锚点，形成最终的同线性区块</li>
<li>sampleA.sampleB.pdf:同线性点图</li>
</ul>
<p>同线性点图一般可以看点连成的比较长的线，在同一横坐标范围有几根线，或者在同一纵坐标范围有几根线，从而判断两个基因组的倍数关系。</p>
<img src="https://www.dropbox.com/s/32mmo0kfhtx5b8g/grape.peach.cscore.99.png?raw=1" width=100% title="同线性点图" alt="同线性点图" align=center/>

<p><strong><p align="center">Figure 1. 同线性点图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi viki</a></p></p>
<h2 id="2-4-同线性深度直方图"><a href="#2-4-同线性深度直方图" class="headerlink" title="2.4. 同线性深度直方图"></a>2.4. 同线性深度直方图</h2><ol>
<li>在同线性点图运行成功获得sampleA.sampleB.anchors结果文件的前提下</li>
<li>运行<code>python -m jcvi.compara.synteny depth --histogram sampleA.sampleB.anchors</code></li>
<li>生成sampleA.sampleB.depth.pdf，显示了同线性深度比例。</li>
</ol>
<img src="https://www.dropbox.com/s/hhx2dtryrum2gyo/grape.peach.depth.png?raw=1" width=100% title="同线性深度直方图" alt="同线性深度直方图" align=center/>

<p><strong><p align="center">Figure 2. 同线性深度直方图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi viki</a></p></p>
<h2 id="2-5-同线性染色体图"><a href="#2-5-同线性染色体图" class="headerlink" title="2.5. 同线性染色体图"></a>2.5. 同线性染色体图</h2><h3 id="2-5-1-输入文件"><a href="#2-5-1-输入文件" class="headerlink" title="2.5.1. 输入文件"></a>2.5.1. 输入文件</h3><ol>
<li>seqids文件</li>
</ol>
<ul>
<li>指定展示的染色体ID，两行对应两个物种，每行的染色体之间逗号隔开，不能有空行。</li>
<li>seqids文件染色体的顺序对应同线性染色体图的顺序，可以根据前面得到的同线性点图来调整顺序使两个基因组的同线性对应，使图更清晰。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">scaf001,scaf002,scaf003,scaf004,scaf005,scaf006,scaf007,scaf008,scaf009,scaf010,scaf011,scaf012,scaf013,scaf014</span><br><span class="line">scaffold1,scaffold2,scaffold3,scaffold4,scaffold5,scaffold6,scaffold7</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>.simple文件</li>
</ol>
<ul>
<li>每行代表一个同线性区块</li>
<li>一共六列，前四列是两个物种的同线性区块的上下限基因，第五列是评分，第六列是方向。</li>
<li>用前面分析同线性点图的结果文件.anchors创建。</li>
<li>运行<code>python -m jcvi.compara.synteny screen --minspan=30 --simple sampleA.sampleB.anchors sampleA.sampleB.anchors.new</code>会生成sampleA.sampleB.anchors.simple文件。</li>
</ul>
<ol start="3">
<li>layout文件</li>
</ol>
<ul>
<li>设置绘制参数。前三列是设置图的位置坐标信息，整个图的x和y轴都是[0-1]范围。第四列是设置旋转角度，第五列是设置染色体颜色，第六列是展示的标签，第七列是对齐方式，第八列指定bed文件。</li>
<li>注意, #edges下的每一行开头都不能有空格。用于指定simple文件。<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"># y, xstart, xend, rotation, color, label, va,  bed</span><br><span class="line"> .6,     .2,    .8,       0,      lightblue, sample_A, top, sampleA.bed</span><br><span class="line"> .4,     .2,    .8,       0,      lightpink, sample_B, top, sampleB.bed</span><br><span class="line"># edges</span><br><span class="line">e, 0, 1, sampleA.sampleB.anchors.simple</span><br></pre></td></tr></table></figure></li>
</ul>
<h3 id="2-5-2-运行"><a href="#2-5-2-运行" class="headerlink" title="2.5.2. 运行"></a>2.5.2. 运行</h3><p><code>python -m jcvi.graphics.karyotype seqids layout</code></p>
<p>生成karyotype.pdf文件，为同线性双染色体图</p>
<img src="https://www.dropbox.com/s/51k8jujyzage3oa/grape.peach.karyotype.png?raw=1" width=100% title="同线性双染色体图" alt="同线性双染色体图" align=center/>

<p><strong><p align="center">Figure 3. 同线性双染色体图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi wiki</a></p></p>
<h3 id="2-5-3-tips"><a href="#2-5-3-tips" class="headerlink" title="2.5.3. tips"></a>2.5.3. tips</h3><ul>
<li>运行虽然报错<code>ERROR    savefig failed. Reset usetex to False.</code>，应该是没能生成png图，但还是生成了pdf图文件。</li>
<li>当label用的名称过长时，label和图容易重叠，没有好的办法解决，最好用短一点的label。</li>
</ul>
<h3 id="2-5-4-个性化"><a href="#2-5-4-个性化" class="headerlink" title="2.5.4. 个性化"></a>2.5.4. 个性化</h3><h4 id="2-5-4-1-highlight指定同线性区块"><a href="#2-5-4-1-highlight指定同线性区块" class="headerlink" title="2.5.4.1. highlight指定同线性区块"></a>2.5.4.1. highlight指定同线性区块</h4><p>如果要突出显示指定的同线性区块（一般用于展示染色体倍数关系），则在sampleA.sampleB.anchors.sample文件中的对应同线性区域行的前面加上<code>yellowgreen*</code>,表示这行的颜色用yellowgreen展示。</p>
<p>然后再运行<code>python -m jcvi.graphics.karyotype seqids layout</code>就会生成突出显示的图。</p>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><p><a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version)">https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version)</a><br><a target="_blank" rel="noopener" href="https://xuzhougeng.top/archives/Comparative-genomics-using-JCVI-part-one">https://xuzhougeng.top/archives/Comparative-genomics-using-JCVI-part-one</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/11/05/bioinfo_MCscan/" data-id="cl1aiqcr9001pm4vk9ys5b87t" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_align.pep2cds" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/10/29/bioinfo_align.pep2cds/" class="article-date">
  <time datetime="2021-10-29T09:50:00.000Z" itemprop="datePublished">2021-10-29</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/10/29/bioinfo_align.pep2cds/">蛋白质比对转换成CDS比对(ParaAT,PAL2NAL)</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=17059176&auto=1&height=32"></iframe></div>

<h1 id="1-序列比对-sequence-alignment"><a href="#1-序列比对-sequence-alignment" class="headerlink" title="1. 序列比对(sequence alignment)"></a>1. 序列比对(sequence alignment)</h1><p>序列比对的知识可以参考另一篇博文的内容。<br><a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/09/06/bioinfo_MSA/">Multiple Sequence Alignment</a></p>
<p>如果序列间差异较大，为了获得更精准的比对，有时我们会先做蛋白质的比对，然后根据蛋白质比对，转化成CDS比对。这篇博客就是记录把蛋白质比对转换成CDS比对的软件和使用。</p>
<h1 id="2-蛋白比对转CDS比对"><a href="#2-蛋白比对转CDS比对" class="headerlink" title="2. 蛋白比对转CDS比对"></a>2. 蛋白比对转CDS比对</h1><h2 id="2-1-成对比对-pairwise-alignment"><a href="#2-1-成对比对-pairwise-alignment" class="headerlink" title="2.1. 成对比对(pairwise alignment)"></a>2.1. 成对比对(pairwise alignment)</h2><h3 id="2-1-1-应用场景"><a href="#2-1-1-应用场景" class="headerlink" title="2.1.1. 应用场景"></a>2.1.1. 应用场景</h3><p>如果是两条序列比对的转化，批量处理，可以用ParaAT脚本。</p>
<p>比如用MCScanX软件做了共线性分析找到物种间的homologs基因对，或者WGD分析找到物种内的paralogs基因对，想要根据蛋白比对做对应的CDS比对。比对之后用于计算Ka和Ks。</p>
<h3 id="2-1-2-成对比对软件"><a href="#2-1-2-成对比对软件" class="headerlink" title="2.1.2. 成对比对软件"></a>2.1.2. 成对比对软件</h3><h4 id="2-1-2-1-ParaAT"><a href="#2-1-2-1-ParaAT" class="headerlink" title="2.1.2.1. ParaAT"></a>2.1.2.1. ParaAT</h4><p>ParaAT是中科院基因组所张章课题组在2012年开发，2014年更新了2.0版本，是一个perl脚本。</p>
<p><a target="_blank" rel="noopener" href="https://ngdc.cncb.ac.cn/tools/paraat">ParaAT download</a><br><a target="_blank" rel="noopener" href="https://www.sciencedirect.com/science/article/pii/S0006291X12003518">ParaAT paper</a></p>
<h4 id="2-1-2-2-ParaAT下载"><a href="#2-1-2-2-ParaAT下载" class="headerlink" title="2.1.2.2. ParaAT下载"></a>2.1.2.2. ParaAT下载</h4><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://download.big.ac.cn/bigd/tools/ParaAT2.0.tar.gz</span><br><span class="line">tar -zxf ParaAT2.0.tar.gz</span><br><span class="line">cd ParaAT2.0</span><br><span class="line">ParaAT.pl -h</span><br></pre></td></tr></table></figure>

<p>ParaAT2.0目录下有两个脚本，ParaAT.pl用于成对比对的转换（可以批量处理），目录下还有另一个多序列转换脚本Epal2nal.pl（好像是pal2nal.pl的V13版本）。</p>
<h4 id="2-1-2-3-ParaAT使用"><a href="#2-1-2-3-ParaAT使用" class="headerlink" title="2.1.2.3. ParaAT使用"></a>2.1.2.3. ParaAT使用</h4><ol>
<li>输入文件<br>三个输入文件,sample.id,cds.fa,pep.fa，三个文件的序列id要一致。</li>
</ol>
<ul>
<li>sample.id文件<br>两列，每行对应两条要做成对比对的序列ID，任意行，ParaAT可以批量处理多个成对比对。</li>
</ul>
<p><code>cat sample.collinearity |grep &quot;species_prefix&quot;|cut -f2,3 &gt;sample.id</code> 用MCScanX的结果文件提取blocks的同源gene对，获得sample.id文件。</p>
<ul>
<li>cds.fa<br>包括所有需要比对的cds序列的文件</li>
<li>pep.fa<br>包括所有需要比对的蛋白序列的文件</li>
</ul>
<ol start="2">
<li>运行</li>
</ol>
<ul>
<li><p><code>echo &quot;12&quot; &gt;proc</code><br>指定ParaAT.pl使用线程，也可以不指定，默认是6个线程</p>
</li>
<li><p><code>ParaAT.pl -h sample.id -n cds.fa -a pep.fa -m mafft -p proc -f axt -o paraat.out 2&gt;paraat.log</code><br>先根据sample.id指定的id做蛋白的成对比对，然后根据蛋白的比对结果转换成对应的CDS的比对结果。</p>
</li>
</ul>
<p>-h,-n,-a分别指定三个输入文件，-m指定比对软件，-p指定线程，-f指定输出的cds序列的格式（WGD计算KaKs一般用axt格式比较多），-o指定输出文件目录(如果不存在会新建目录)。</p>
<p>其他参数：</p>
<ul>
<li>-n：好像可以包含多核苷酸序列比对；</li>
<li>-g：移除比对后包含gap的密码子；</li>
<li>-t：移除mismatched codons；</li>
<li>-k：用KaKs_Calculator计算(需要输出axt格式)Ka和Ks，获得axt文件后自动计算kaks值，使用MA模型，比YN模型慢，推荐输出axt后自己用KaKs_Calculator计算并用YN模型。</li>
</ul>
<h2 id="2-2-多序列比对-multiple-sequence-alignment"><a href="#2-2-多序列比对-multiple-sequence-alignment" class="headerlink" title="2.2. 多序列比对(multiple sequence alignment)"></a>2.2. 多序列比对(multiple sequence alignment)</h2><h3 id="2-2-1-应用场景"><a href="#2-2-1-应用场景" class="headerlink" title="2.2.1. 应用场景"></a>2.2.1. 应用场景</h3><p>如果是多序列的转化，可以使用PAL2NAL脚本。</p>
<p>比如做了orthofinder2找到多个物种的orthogroups，要对每一组orthogroups进行蛋白比对转换成CDS比对。</p>
<h3 id="2-2-2-多序列比对软件"><a href="#2-2-2-多序列比对软件" class="headerlink" title="2.2.2. 多序列比对软件"></a>2.2.2. 多序列比对软件</h3><h4 id="2-2-2-1-PAL2NAL"><a href="#2-2-2-1-PAL2NAL" class="headerlink" title="2.2.2.1. PAL2NAL"></a>2.2.2.1. PAL2NAL</h4><p><a target="_blank" rel="noopener" href="http://www.bork.embl.de/pal2nal/">PAL2NAL介绍</a></p>
<ul>
<li>PAL2NAL可以将蛋白的多序列比对转换成CDS比对，如果输入的是一对序列，还会通过paml的codeml程序自动计算dn和ds。</li>
<li>如果数据量少，可以通过上面的网页进行转换。</li>
</ul>
<h4 id="2-2-2-2-PAL2NAL下载"><a href="#2-2-2-2-PAL2NAL下载" class="headerlink" title="2.2.2.2. PAL2NAL下载"></a>2.2.2.2. PAL2NAL下载</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget http://www.bork.embl.de/pal2nal/distribution/pal2nal.v14.tar.gz</span><br><span class="line">tar -zxvf pal2nal.v14.tar.gz</span><br><span class="line">pal2nal.pl -h</span><br></pre></td></tr></table></figure>

<h4 id="2-2-2-3-PAL2NAL使用"><a href="#2-2-2-3-PAL2NAL使用" class="headerlink" title="2.2.2.3. PAL2NAL使用"></a>2.2.2.3. PAL2NAL使用</h4><p><code>pal2nal.pl -nogap -nomismatch pep.aln nuc.fa -output fasta &gt;nuc.aln</code></p>
<p>需要已经做好比对的蛋白质序列pep.aln和ID一致的cds序列nuc.fa</p>
<p>参数：</p>
<ul>
<li>-output (clustal|paml|fasta|codon) # Output format; default &#x3D; clustal</li>
<li>-nogap  # remove columns with gaps and inframe stop codons</li>
<li>-nomismatch # remove mismatched codons (mismatch between pep and cDNA) from the output</li>
</ul>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><p><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Sequence_alignment">https://en.wikipedia.org/wiki/Sequence_alignment</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/10/29/bioinfo_align.pep2cds/" data-id="cl1aiqcre0024m4vk24ohcydi" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-biosoft_CAFE5" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/10/29/biosoft_CAFE5/" class="article-date">
  <time datetime="2021-10-29T08:50:00.000Z" itemprop="datePublished">2021-10-29</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/biosoft/">biosoft</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/10/29/biosoft_CAFE5/">CAFE5分析基因家族扩张和收缩</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=17059176&auto=1&height=32"></iframe></div>

<h1 id="1-CAFE"><a href="#1-CAFE" class="headerlink" title="1. CAFE"></a>1. CAFE</h1><p>CAFE(Computational Analysis of gene Family Evolution)是一款以解释系统发育历史的方式分析基因家族大小变化的软件，这种分析常被称为基因家族收缩扩张(Gene family expansions and contractions)分析。</p>
<p>CAFE使用出生和死亡过程来模拟用户指定的系统发育树中的基因获得和丢失，可计算由父节点到子节点的基因家族大小转移率，也可推断祖先物种的基因家族大小，在该模型下生成的基因家族规模分布可以为评估观察到的类群之间家族规模差异的显著性提供基础。</p>
<p>自2005年Hahn课题组提出评估基因家族进化速度和模式的算法，2006年第一个版本CAFE发表后，2020年推出了最新版本<a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article-abstract/36/22-23/5516/6039105?redirectedFrom=fulltext">CAFE5</a>，之前旧版本的基本模型假设所有基因家族都具有相同的进化速率。新版本支持伽马分布速率类别对家族之间的速率变化进行显式建模。</p>
<h1 id="2-CAFE5安装"><a href="#2-CAFE5安装" class="headerlink" title="2. CAFE5安装"></a>2. CAFE5安装</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/hahnlab/CAFE5.git</span><br><span class="line">cd CAFE5</span><br><span class="line">./configure</span><br><span class="line">make</span><br></pre></td></tr></table></figure>

<p>conda可以安装CAFE4<br><code>conda install -c bioconda cafe</code>安装的是CAFE4</p>
<h1 id="3-CAFE5使用"><a href="#3-CAFE5使用" class="headerlink" title="3. CAFE5使用"></a>3. CAFE5使用</h1><h2 id="3-1-CAFE5输入文件"><a href="#3-1-CAFE5输入文件" class="headerlink" title="3.1. CAFE5输入文件"></a>3.1. CAFE5输入文件</h2><p>CAFE5需要至少两个输入文件，一个是基因家族计数文件gene_families.txt，一个是树文件tree.txt</p>
<h3 id="3-1-1-gene-families-txt"><a href="#3-1-1-gene-families-txt" class="headerlink" title="3.1.1. gene_families.txt"></a>3.1.1. gene_families.txt</h3><ol>
<li>gene_families.txt文件格式<br>制表符分隔的基因家族计数文件，通常用OrthoMCL, SwiftOrtho, FastOrtho, OrthAgogue, OrthoFinder等软件获取计数信息。</li>
</ol>
<p>第一列是功能描述Desc，可以为null，第二列是Family ID，其余列是每个物种以及包含对应的基因家族基因的数量。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">Desc	Family ID	human	chimp	orang	baboon	gibbon	macaque	marmoset rat	mouse	cat	horse	cow</span><br><span class="line">ATPase	ORTHOMCL1	 52	 55	 54	 57	 54	  56	  56	 53	 52	57	55	 54</span><br><span class="line">(null)	ORTHOMCL2	 76	 51	 41	 39	 45	  36	  37	 67	 79	37	41	 49</span><br><span class="line">HMG box	ORTHOMCL3	 50	 49	 48	 48	 46	  49	  48	 55	 52	51	47	 55</span><br><span class="line">(null)	ORTHOMCL4	 43	 43	 47	 53	 44	  47	  46	 59	 58	51	50	 55</span><br><span class="line">Dynamin	ORTHOMCL5	 43	 40	 43	 44	 31	  46	  33	 79	 70	43	49	 50</span><br></pre></td></tr></table></figure>

<ol start="2">
<li><p>用orthofinder2的结果文件Orthogroups.GeneCount.tsv转换成gene_families.txt文件<br><code>awk -v OFS=&quot;\t&quot; &#39;&#123;$NF=null;print $1,$0&#125;&#39; Orthogroups.GeneCount.tsv |sed -E -e &#39;s/Orthogroup/desc/&#39; -e &#39;s/_[^\t]+//g&#39; &gt;gene_families.txt</code></p>
</li>
<li><p>如果是MCL获得的基因家族数据。cafe5的tutorials目录下有脚本mcl2rawcafe.py用于把mcl的输出转化成cafe5的输入。<br><code>python mcl2rawcafe.py -i dump.blast_output.mci.I30 -o gene_families.txt -sp &quot;ENSG00 ENSPTR ENSPPY ENSPAN ENSNLE ENSMMU ENSCJA ENSRNO ENSMUS ENSFCA ENSECA ENSBTA&quot;</code>。</p>
</li>
<li><p>剔除不同物种间的基因拷贝数变异特别大的基因家族。<br>否则可能无法估计参数，运行中断报错Failed to initialize any reasonable values</p>
</li>
</ol>
<ul>
<li>cafe5的tutorial目录下脚本clade_and_size_filter.py可以筛除一个或以上物种有超过100个基因拷贝的基因家族。<br><code>python clade_and_size_filter.py -s -i gene_families.txt -o gene_familie_filter.txt</code></li>
</ul>
<p>脚本运行失败。</p>
<ul>
<li>也可以用awk命令来筛除3-13列中基因家族数量大于等于100的行，可以根据自己的数据更改。<br><code>awk &#39;NR==1 || $3&lt;100 &amp;&amp; $4&lt;100 &amp;&amp; $5&lt;100 &amp;&amp; $6&lt;100 &amp;&amp; $7&lt;100 &amp;&amp; $8&lt;100 &amp;&amp; $9&lt;100 &amp;&amp; $10&lt;100 &amp;&amp; $11&lt;100 &amp;&amp; $12&lt;100 &amp;&amp; $13&lt;100 &#123;print $0&#125;&#39; gene_families.txt &gt;gene_families_filter.txt</code></li>
</ul>
<h3 id="3-1-2-tree-txt"><a href="#3-1-2-tree-txt" class="headerlink" title="3.1.2. tree.txt"></a>3.1.2. tree.txt</h3><ol>
<li>tree.txt文件格式<br>tree.txt是二叉的（binary），有根的（rooted），超度量(时间树，ultrametric)的newick格式树。</li>
</ol>
<p>可以用R包Ape的函数<code>is.binary</code>, <code>is.rooted</code>, <code>is.ultrametric</code>对树是否是二叉有根超度量做检验。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">((((cat:68.710507,horse:68.710507):4.566782,cow:73.277289):20.722711,(((((chimp:4.444172,human:4.444172):6.682678,orang:11.126850):2.285855,gibbon:13.412706):7.211527,(macaque:4.567240,baboon:4.567240):16.056992):16.060702,marmoset:36.684935):57.315065):38.738021,(rat:36.302445,mouse:36.302445):96.435575);</span><br></pre></td></tr></table></figure>

<ol start="2">
<li><p>用paml的Figtree转换成tree.txt文件（newick格式）<br><code>grep -E -v &quot;NEXUS|BEGIN|END&quot; FigTree.tre|sed -E -e &quot;s/\[[^]]*\]//g&quot; -e &quot;s/[ \t]//g&quot; -e &quot;/^$/d&quot; -e &quot;s/UTREE/tree tree/&quot; &gt;tree.txt</code></p>
</li>
<li><p>CAFE5软件下脚本prep_r8s.py用于构建r8s的批量运行脚本，可用批量运行脚本获得tree。</p>
</li>
</ol>
<h2 id="3-2-运行CAFE5"><a href="#3-2-运行CAFE5" class="headerlink" title="3.2. 运行CAFE5"></a>3.2. 运行CAFE5</h2><p><code>cafe5 -i gene_families_filter.txt -t tree.txt -p -k 2 -o k2p</code><br>参数：<br>-p代表指定root frequency distribution为泊松分布（默认是均匀分布uniform distribution）。<br>-k 3代表使用GAMMA模型（默认是base模型）并且使用3种gamma rate（代表不同基因家族有着不同的进化速率）。-k的值需要运行多次比较likelihood并确保收敛后才知道使用哪个最好，一般来说2-5之间试一试。<br>-o 指定输出目录，默认是results。</p>
<h2 id="3-3-CAFE5的结果"><a href="#3-3-CAFE5的结果" class="headerlink" title="3.3. CAFE5的结果"></a>3.3. CAFE5的结果</h2><h3 id="3-3-1-结果文件"><a href="#3-3-1-结果文件" class="headerlink" title="3.3.1. 结果文件"></a>3.3.1. 结果文件</h3><ul>
<li>model_asr.tre：每个基因家族的树的合集</li>
<li>model_clade_results.txt：进化树上每个节点扩张或者收缩了多少基因家族</li>
<li>model_family_results.txt：基因家族变化的p值和是否显著的结果</li>
<li>model_results.txt：模型，最终似然值，最终Lambda值等参数信息。</li>
</ul>
<p>还生成了一些其他文件。</p>
<h3 id="3-3-2-选择最优结果"><a href="#3-3-2-选择最优结果" class="headerlink" title="3.3.2. 选择最优结果"></a>3.3.2. 选择最优结果</h3><p>k值的结果比较：<br>查看k2p，k3p，k5p，k6p等不同的结果文件Gamma_results.txt文件中的第一行信息，Model Gamma Final Likelihood (-lnL)值，挑选最大的为最优结果。</p>
<h3 id="结果整理"><a href="#结果整理" class="headerlink" title="结果整理"></a>结果整理</h3><p>对特定物种扩张和收缩基因的提取</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">cat Gamma_change.tab |cut -f1,15|grep &quot;+[1-9]&quot; &gt;sample.expanded #提取Gamma_change.tab第15列代表物种sample的扩张的orthogroupsID</span><br><span class="line">cat Gamma_change.tab |cut -f1,15|grep &quot;-&quot; &gt;sample.contracted  #提取Gamma_change.tab第15列代表物种sample的收缩的orthogroupsID</span><br><span class="line">cat Gamma_family_results.txt |grep &quot;y&quot;|cut -f1 &gt;p0.05.significant #提取显著扩张或收缩的orthogroupsID</span><br><span class="line">grep -f p0.05.significant sample.expanded |cut -f1&gt;sample.expanded.significant #提取显著扩张的sample物种的orthogroupsID</span><br><span class="line">grep -f p0.05.significant sample.contracted |cut -f1 &gt;sample.contracted.significant #提取显著收缩的sample物种的orthogroupsID</span><br><span class="line"></span><br><span class="line">grep -f sample.contracted.significant ./OrthoFinder/Results_Oct14/Orthogroups/Orthogroups.txt |sed -E -e &quot;s/: [^b]+bv/ bv/g&quot; -e &quot;s/ [^b]+//g&quot; &gt;sample.contracted.significant.ortho #提取显著收缩的基因</span><br><span class="line">grep -f sample.expanded.significant ./OrthoFinder/Results_Oct14/Orthogroups/Orthogroups.txt |sed -E -e &quot;s/: [^b]+bv/ bv/g&quot; -e &quot;s/ [^b]+//g&quot; &gt;sample.expanded.significant.ortho #提取显著扩张的基因</span><br><span class="line"></span><br><span class="line">cat sample.expanded.significant.ortho |sed &quot;s/ /\n/g&quot;|grep &quot;bv&quot; |sort -k 1.3n |uniq &gt;sample.expanded.significant.genes</span><br><span class="line">cat sample.contracted.significant.ortho |sed &quot;s/ /\n/g&quot;|grep &quot;bv&quot; |sort -k 1.3n |uniq &gt;sample.contracted.significant.genes</span><br><span class="line"></span><br><span class="line">seqkit grep -f sample.expanded.significant.genes sample.pep.fa &gt;sample.expanded.significant.pep.fa</span><br><span class="line">seqkit grep -f sample.contracted.significant.genes sample.pep.fa &gt;sample.contracted.significant.pep.fa</span><br></pre></td></tr></table></figure>

<p>提取出指定物种的显著扩张和收缩的蛋白序列之后，就可以拿去做GO注释和基因富集分析。</p>
<h2 id="3-4-把每个节点收缩扩张的基因数量画在树上"><a href="#3-4-把每个节点收缩扩张的基因数量画在树上" class="headerlink" title="3.4. 把每个节点收缩扩张的基因数量画在树上"></a>3.4. 把每个节点收缩扩张的基因数量画在树上</h2><p>有看到一个画图脚本。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">python python_scripts/cafetutorial_draw_tree.py -i reports/summary_run1_node.txt</span><br><span class="line"> -t <span class="string">&#x27;((((cat:68.7105,horse:68.7105):4.56678,cow:73.2773):20.7227,(((((chimp:4.444</span></span><br><span class="line"><span class="string">17,human:4.44417):6.68268,orang:11.1268):2.28586,gibbon:13.4127):7.21153,(macaque</span></span><br><span class="line"><span class="string">:4.56724,baboon:4.56724):16.057):16.0607,marmoset:36.6849):57.3151):38.738,(rat:3</span></span><br><span class="line"><span class="string">6.3024,mouse:36.3024):96.4356)&#x27;</span> -d <span class="string">&#x27;((((cat&lt;0&gt;,horse&lt;2&gt;)&lt;1&gt;,cow&lt;4&gt;)&lt;3&gt;,(((((chimp</span></span><br><span class="line"><span class="string">&lt;6&gt;,human&lt;8&gt;)&lt;7&gt;,orang&lt;10&gt;)&lt;9&gt;,gibbon&lt;12&gt;)&lt;11&gt;,(macaque&lt;14&gt;,baboon&lt;16&gt;)&lt;15&gt;)&lt;13&gt;,</span></span><br><span class="line"><span class="string">marmoset&lt;18&gt;)&lt;17&gt;)&lt;5&gt;,(rat&lt;20&gt;,mouse&lt;22&gt;)&lt;21&gt;)&lt;19&gt;&#x27;</span> -o reports/summary_run1_tree_</span><br><span class="line">rapid.png -y Rapid</span><br></pre></td></tr></table></figure>

<p>没有试过这个脚本，发表的图还是用R包ggtree自己画。</p>
<h1 id="4-references"><a href="#4-references" class="headerlink" title="4. references"></a>4. references</h1><p><a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article-abstract/36/22-23/5516/6039105?redirectedFrom=fulltext">https://academic.oup.com/bioinformatics/article-abstract/36/22-23/5516/6039105?redirectedFrom=fulltext</a><br><a target="_blank" rel="noopener" href="https://github.com/hahnlab/CAFE5">https://github.com/hahnlab/CAFE5</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/10/29/biosoft_CAFE5/" data-id="cl1aiqcrm002qm4vk51cwgpkg" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-bioinfo_extract.4dtv.from.cds" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/10/10/bioinfo_extract.4dtv.from.cds/" class="article-date">
  <time datetime="2021-10-10T02:39:54.000Z" itemprop="datePublished">2021-10-10</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/10/10/bioinfo_extract.4dtv.from.cds/">从codon比对的cds序列中提取四倍简并位点</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <h1 id="1-从condon比对的cds序列中提取四倍简并位点"><a href="#1-从condon比对的cds序列中提取四倍简并位点" class="headerlink" title="1. 从condon比对的cds序列中提取四倍简并位点"></a>1. 从condon比对的cds序列中提取四倍简并位点</h1><p>输入文件是根据密码子规则（即三个三个对齐）比对好的cds序列，输出文件是以其中一个物种的4dtv为标准提取所有物种的4dtv位点，提取的也是按照比对前的顺序排列，结果保存在4dtv.aln文件中。</p>
<h2 id="1-1-输入数据和变量定义"><a href="#1-1-输入数据和变量定义" class="headerlink" title="1.1. 输入数据和变量定义"></a>1.1. 输入数据和变量定义</h2><ol>
<li><code>cds=/path/to/cds.aln</code> # 定义变量cds为codon模式比对好的cds序列</li>
<li><code>species=Athaliana</code> #以哪个物种的4dtv为标准提取，就定义变量species为那个物种的序列ID</li>
</ol>
<h2 id="1-2-三种方案【选择一种】"><a href="#1-2-三种方案【选择一种】" class="headerlink" title="1.2. 三种方案【选择一种】"></a>1.2. 三种方案【选择一种】</h2><h3 id="1-2-1-一步生成"><a href="#1-2-1-一步生成" class="headerlink" title="1.2.1. 一步生成"></a>1.2.1. 一步生成</h3><ol>
<li>一步生成运行文件，生成从cds获取4dtv的命令，很长，所以储存在文件4dtv.sh中。运行4dtv.sh就可以获取<br><code>seqkit locate -V 0 -i -d -p GCN -p CGN -p GGN -p CTN -p CCN -p TCN -p ACN -p GTN $&#123;cds&#125; |awk -v awka=&quot;$&#123;species&#125;&quot; &#39;&#123;if ($1 == awka &amp;&amp; $6%3== 0) print $6&#125;&#39;|sort -k 1n |uniq |awk -v awkb=&quot;$&#123;cds&#125;&quot; &#39;&#123;print &quot;&lt;(seqkit subseq -r &quot;$1&quot;:&quot;$1&quot; &quot;awkb&quot;)&quot;&#125;&#39; |sed -e &#39;1i\seqkit concat -w 0&#39; -e &#39;$a\&gt; 4dtv.aln&#39;|xargs echo &gt;4dtv.sh</code></li>
</ol>
<p>代码含义：用seqkit locate根据4dtv的规则提取位置，然后筛选指定物种的位置，筛选第三位位置是3的整数倍的位置，把第三位位置(即4dtv位点)提取出来，排序去重；再按照seqkit concat要求的格式规则把seqkit concat命令写入提取脚本4dtv.sh中。</p>
<p>除非位点在1000个以内，否则一步生成运行文件可能会运行失败，推荐用下一种方案。</p>
<ol start="2">
<li>运行生成的4dtv.sh，获得4dtv.aln结果<br><code>bash 4dtv.sh</code></li>
</ol>
<h3 id="1-2-2-切割分开运行【推荐】"><a href="#1-2-2-切割分开运行【推荐】" class="headerlink" title="1.2.2. 切割分开运行【推荐】"></a>1.2.2. 切割分开运行【推荐】</h3><p>如果需要合并的位点超过1000个，一步生成运行文件占用内存，运行较慢，可以分步骤进行。</p>
<p>如果因为4dtv.sh太长无法运行，可以把4dtv.temp分成1-n几部分（比如1000行作为一个子文件），然后切割各部分分别生成4dtv_n.sh，之后分别运行再合并。</p>
<ol>
<li><p>生成临时文件4dtv.temp<br><code>seqkit locate -V 0 -i -d -p GCN -p CGN -p GGN -p CTN -p CCN -p TCN -p ACN -p GTN $&#123;cds&#125; |awk -v awka=&quot;$&#123;species&#125;&quot; &#39;&#123;if ($1 == awka &amp;&amp; $6%3== 0) print $6&#125;&#39; |sort -k 1n |uniq &gt;4dtv.temp</code></p>
</li>
<li><p>分割临时文件4dtv.temp<br><code>split -l 1000 4dtv.temp -d -a 4 4dtv.temp_</code><br>命令含义：分割文件4dtv.temp成多个文件，每个子文件1000行（-l 1000），以数字（-d）而非字母命名子文件，数字长度为4（-a 4，default是2）；会生成4dtv.temp_0000-nnnn的子文件。</p>
</li>
</ol>
<p>可以先看看4dtv.temp的长度，我尝试了下按1000行切割比较合适，再多就达到能同时打开文件数量上限无法运行了。</p>
<ol start="3">
<li><p>生成4dtv.temp.n.sh<br><code>for i in $(ls 4dtv.temp_*);do cat $&#123;i&#125; |awk -v awkb=&quot;$&#123;cds&#125;&quot; &#39;&#123;print &quot;&lt;(seqkit subseq -r &quot;$1&quot;:&quot;$1&quot; &quot;awkb&quot;)&quot;&#125;&#39; |sed -e &#39;1i\seqkit concat -w 0&#39; -e &#39;$a\&gt;&#39;|sed &quot;s/^&gt;$/&gt; $&#123;i&#125;.aln/&quot; |sed &#39;s/$/ \\/g&#39; |sed &#39;$ s/.$//&#39; &gt;$&#123;i&#125;.sh; done</code></p>
</li>
<li><p>运行生成的4dtv.temp.n.sh，获得4dtv.aln_n结果【耗时步骤】<br><code>for i in $(ls 4dtv.temp_*.sh);do bash $&#123;i&#125;;done</code></p>
</li>
</ol>
<p>建议用nohup放到后台运行<br><code>nohup sh -c &#39;for i in $(ls 4dtv.temp_*.sh);do bash $&#123;i&#125;;done&#39; &amp;</code></p>
<ol start="5">
<li>合并子文件，获得最终结果<br><code>seqkit concat ./4dtv.temp_*.aln &gt;4dtv.aln</code></li>
</ol>
<p>可以<code>seqkit stats *aln</code>检查是否都是1000长度的序列；以及<code>seqkit stats 4dtv.aln</code>检查是否序列长度与4dtv.temp的行数一致。</p>
<ol start="6">
<li>删除中间文件<br><code>rm 4dtv.temp*</code></li>
</ol>
<h3 id="1-2-3-分物种运行【不推荐】"><a href="#1-2-3-分物种运行【不推荐】" class="headerlink" title="1.2.3. 分物种运行【不推荐】"></a>1.2.3. 分物种运行【不推荐】</h3><p>一开始没发现seqkit concat这个工具，然后自己写的合并，有时会有aln的问题，不建议用。</p>
<ol>
<li>定义数组sps为物种列表<br><code>sps=($(seqkit seq -n $&#123;cds&#125;|xargs))</code></li>
<li>以第六个物种${sps[5]}的4dtv为标准，为每个物种保存一份4dtv.bed文件<br><code>for i in $(echo $&#123;sps[*]&#125;); do cat $&#123;sps[5]&#125;.4dtv|sed &quot;s/$&#123;species[5]&#125;/$&#123;i&#125;/g&quot; &gt;$&#123;i&#125;.4dtv.bed; done</code></li>
<li>为每个物种提取4dtv位点，保存到4dtv.aln.temp【耗时步骤】<br><code>for i in $(echo $&#123;sps[*]&#125;); do seqkit subseq --bed $&#123;i&#125;.4dtv.bed $&#123;cds&#125; | grep -v &quot;&gt;&quot;|xargs echo |sed &quot;1i\&gt;$&#123;i&#125;&quot; &gt;$&#123;i&#125;.4dtv.aln.temp; done</code></li>
<li>合并物种的4dtv位点<br><code>cat *.4dtv.aln.temp &gt;4dtv.aln</code> #合并</li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/10/10/bioinfo_extract.4dtv.from.cds/" data-id="cl1aiqcrj002gm4vkecwyfflk" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-linux_loop" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/10/10/linux_loop/" class="article-date">
  <time datetime="2021-10-10T01:39:54.000Z" itemprop="datePublished">2021-10-10</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/linux/">linux</a>►<a class="article-category-link" href="/categories/linux/shell/">shell</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/10/10/linux_loop/">shell的几种循环语法</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1697043&auto=1&height=32"></iframe></div>

<h1 id="1-shell的几种循环语法间的差异"><a href="#1-shell的几种循环语法间的差异" class="headerlink" title="1. shell的几种循环语法间的差异"></a>1. shell的几种循环语法间的差异</h1><h2 id="1-1-for-loop"><a href="#1-1-for-loop" class="headerlink" title="1.1. for-loop"></a>1.1. for-loop</h2><ol>
<li><code>for i in $(cat file.txt);do echo $i; done</code></li>
</ol>
<ul>
<li>for循环每次都是以空格&#x2F;制表符为分割符输出，所以i变量不是一整行内容，而是单个单元格的内容。想要获取一整行内容，可以先将空格换成别的字符，输出时再替换回来。比如<code>for i in $(sed &#39;s/\t/#/g&#39; file.txt); do echo $i |sed &#39;s/#/\t/g&#39;; done</code></li>
<li>【更耗内存】cat会一次性把file.txt所有内容输入到内存中；</li>
</ul>
<h2 id="1-2-while-loop"><a href="#1-2-while-loop" class="headerlink" title="1.2. while-loop"></a>1.2. while-loop</h2><ol>
<li><code>while read line; do echo $line; done &lt; file.txt</code></li>
</ol>
<ul>
<li>【更省内存】按行读取文件，读取到line变量是一整行内容。</li>
</ul>
<ol start="2">
<li><code>cat file.txt | while read line; do echo $line; done</code></li>
</ol>
<ul>
<li>【更耗内存】cat会一次性把file.txt所有内容输入到内存中；<br>因为管道|的使用，while read line在子shell中按行调用，line变量也是一整行内容。<br>这个和多线程控制脚本batch_run.sh冲突。</li>
</ul>
<h1 id="2-循环语法常用命令符号"><a href="#2-循环语法常用命令符号" class="headerlink" title="2. 循环语法常用命令符号"></a>2. 循环语法常用命令符号</h1><h2 id="2-1-amp-amp"><a href="#2-1-amp-amp" class="headerlink" title="2.1. &amp;&amp; ;"></a>2.1. &amp;&amp; ;</h2><p>一行for&#x2F;while循环中，执行多条命令的代码常写成这种形式：<code>for i in $(cat file.txt);do grep $i sum.txt &amp;&amp; echo $i; j=$i+3;done</code>。</p>
<p>包含了三条命令：</p>
<ul>
<li><code>grep $i sum.txt</code></li>
<li><code>break</code></li>
<li><code>echo $i</code></li>
</ul>
<p>其中<code>grep $i sum.txt</code>和<code>echo $i</code>以<code>&amp;&amp;</code>相连，<code>j=$i+3</code>与其他两条命令以<code>;</code>相连。</p>
<p><code>&amp;&amp;</code>和<code>;</code>都是用于连接同一行的两条命令的；代表前一条命令执行完再接着执行下一条命令；区别在于<code>&amp;&amp;</code>需要上一条命令执行成功才执行下一条命令；<code>;</code>则只是命令分隔符，上一条命令执行成功与否不影响下一条命令的执行。</p>
<p>再多介绍一个命令符<code>||</code>，当用<code>||</code>连接两条命令时，表示前一条命令执行成功时，停止执行后面的命令；只有当前一条命令执行失败时，才执行后一条命令。</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/10/10/linux_loop/" data-id="cl1aiqcrv003rm4vkcxt19gff" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/while/" rel="tag">while</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-linux_replace.line.separator" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/10/10/linux_replace.line.separator/" class="article-date">
  <time datetime="2021-10-10T01:39:54.000Z" itemprop="datePublished">2021-10-10</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/linux/">linux</a>►<a class="article-category-link" href="/categories/linux/text-processing/">text processing</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/10/10/linux_replace.line.separator/">在Linux系统中替换文本换行符</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1697043&auto=1&height=32"></iframe></div>

<h1 id="替换换行符-x2F-多行变单行的几种方法"><a href="#替换换行符-x2F-多行变单行的几种方法" class="headerlink" title="替换换行符&#x2F;多行变单行的几种方法"></a>替换换行符&#x2F;多行变单行的几种方法</h1><ol>
<li><code>cat file.txt | xargs echo</code> #用xargs为echo传入参数，默认是把换行符替换成空格。【推荐】</li>
<li><code>cat file.txt | tr &#39;\n&#39; &#39; &#39;</code> #用tr把换行符替换成空格。【推荐】</li>
<li><code>cat file.txt | paste -sd &quot; &quot;</code> #用paste替换；-s是一次处理文件的所有行，而非并行处理每一行；-d指定分割符。</li>
<li><code>sed &#39;:a;N;s/\n/ /g;ta&#39; file.txt</code> #用sed处理；sed按行处理所以每次处理会自动添加换行符，<code>:a</code>在代码开始处设置标记a，代码执行结尾处通过跳转命令<code>ta</code>重新跳转到标号a处，重新执行代码，递归每行；N表示读入下一行。【很慢】</li>
<li><code>awk &#39;ORS=&quot; &quot;&#39; file.txt |head -c -1</code> #用awk处理；ORS(Output Record Separator)设置输出分隔符，把换行符换成空格。head -c -1代表截取文件除了最后一个字符的字符，用于去掉文本末多的分隔符，根据情况使用。</li>
<li><code>awk BEGIN&#123;RS=EOF&#125;&#39;&#123;gsub(/\n/,&quot; &quot;);print&#125;&#39; file.txt</code> #用awk处理，将RS(record separator)设置成EOF(end of file)，即将文件视为一个记录；再通过gsub函数将换行符\n替换成空格。</li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/10/10/linux_replace.line.separator/" data-id="cl1aiqcrw003tm4vk8ox09bf4" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li></ul>

    </footer>
  </div>
  
</article>


  
    <article id="post-R_plot_upset" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/09/29/R_plot_upset/" class="article-date">
  <time datetime="2021-09-29T07:56:36.000Z" itemprop="datePublished">2021-09-29</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/computer-language/">computer language</a>►<a class="article-category-link" href="/categories/computer-language/R/">R</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2021/09/29/R_plot_upset/">集合可视化 —— upset图</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1697043&auto=1&height=32"></iframe></div>

<h1 id="1-多集合展示UpSet图"><a href="#1-多集合展示UpSet图" class="headerlink" title="1. 多集合展示UpSet图"></a>1. 多集合展示UpSet图</h1><p>当有多个群组（集合）的数据，想要展示在不同的事物群组（集合）之间的数学或逻辑联系时，可使用韦恩Venn图和UpSet图展示多集合直接的共享关系和各自的独享关系。</p>
<p>当集合数量少于等于5个时，多用韦恩图；当集合数量大于5个时，多用upset图。</p>
<p>韦恩图的绘制推荐使用R包gplots，参考博文<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/09/29/R_plot_venn/">韦恩图</a>；upset图的绘制推荐使用R包UpSetR，参考博文。</p>
<h2 id="1-1-应用场景"><a href="#1-1-应用场景" class="headerlink" title="1.1. 应用场景"></a>1.1. 应用场景</h2><ol>
<li>利用orthofinder找到的不同物种的orthogroups的结果来绘制Venn Diagram韦恩图或者UpSet plot图，查看不同物种间共享的orthogroups的数量关系。</li>
</ol>
<h2 id="1-2-数据准备"><a href="#1-2-数据准备" class="headerlink" title="1.2. 数据准备"></a>1.2. 数据准备</h2><p>UpSetR的输入文件是表格形式，第一列是数据分类信息，后面列是每一个集合占一列（每个物种占一列），表格内容是1&#x2F;0表示集合在每一个类别是否有数据。</p>
<h3 id="1-2-1-UpSetR自带数据"><a href="#1-2-1-UpSetR自带数据" class="headerlink" title="1.2.1. UpSetR自带数据"></a>1.2.1. UpSetR自带数据</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">movies <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span>system.file<span class="punctuation">(</span><span class="string">&quot;extdata&quot;</span><span class="punctuation">,</span> <span class="string">&quot;movies.csv&quot;</span><span class="punctuation">,</span> package <span class="operator">=</span> <span class="string">&quot;UpSetR&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span> sep<span class="operator">=</span><span class="string">&quot;;&quot;</span><span class="punctuation">)</span> <span class="comment"># 载入UpSetR包提供的电影数据用作示例。第一行是电影的类别名称，第一列是电影名，第二列是电影上映年度，后面列中的1/0是代表电影是否属于相应的类别。</span></span><br><span class="line">mutations <span class="operator">&lt;-</span> read.csv<span class="punctuation">(</span>system.file<span class="punctuation">(</span><span class="string">&quot;extdata&quot;</span><span class="punctuation">,</span> <span class="string">&quot;mutations.csv&quot;</span><span class="punctuation">,</span> package <span class="operator">=</span> <span class="string">&quot;UpSetR&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&quot;,&quot;</span><span class="punctuation">)</span> <span class="comment"># 载入UpsetR包提供的突变数据用作示例。第一行是不同的基因简称，第一列是基因特征ID，数据中的1/0是代表基因是否有对应的特征。</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>movies<span class="punctuation">)</span> <span class="comment"># 大致浏览一下该数据集,数据集太长，就只看前几列</span></span><br><span class="line">View<span class="punctuation">(</span>movies<span class="punctuation">)</span> <span class="comment"># 弹出窗口，可查看数据。</span></span><br><span class="line"></span><br><span class="line">head<span class="punctuation">(</span>mutations<span class="punctuation">)</span> <span class="comment"># 大致浏览一下该数据集,数据集太长，就只看前几列</span></span><br><span class="line">View<span class="punctuation">(</span>mutations<span class="punctuation">)</span> <span class="comment"># 弹出窗口，可查看数据。</span></span><br></pre></td></tr></table></figure>
<br>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>movies<span class="punctuation">)</span></span><br><span class="line">                                Name ReleaseDate Action Adventure Children Comedy Crime Documentary Drama Fantasy Noir Horror Musical Mystery Romance SciFi Thriller War Western AvgRating Watches</span><br><span class="line"><span class="number">1</span>                   Toy Story <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">0</span>         <span class="number">0</span>        <span class="number">1</span>      <span class="number">1</span>     <span class="number">0</span>           <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">4.15</span>    <span class="number">2077</span></span><br><span class="line"><span class="number">2</span>                     Jumanji <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">0</span>         <span class="number">1</span>        <span class="number">1</span>      <span class="number">0</span>     <span class="number">0</span>           <span class="number">0</span>     <span class="number">0</span>       <span class="number">1</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">3.20</span>     <span class="number">701</span></span><br><span class="line"><span class="number">3</span>            Grumpier Old Men <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">0</span>         <span class="number">0</span>        <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>           <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">1</span>     <span class="number">0</span>        <span class="number">0</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">3.02</span>     <span class="number">478</span></span><br><span class="line"><span class="number">4</span>           Waiting to Exhale <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">0</span>         <span class="number">0</span>        <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>           <span class="number">0</span>     <span class="number">1</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">2.73</span>     <span class="number">170</span></span><br><span class="line"><span class="number">5</span> Father of the Bride Part II <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">0</span>         <span class="number">0</span>        <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>           <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">3.01</span>     <span class="number">296</span></span><br><span class="line"><span class="number">6</span>                        Heat <span class="punctuation">(</span><span class="number">1995</span><span class="punctuation">)</span>        <span class="number">1995</span>      <span class="number">1</span>         <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">1</span>           <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>     <span class="number">0</span>        <span class="number">1</span>   <span class="number">0</span>       <span class="number">0</span>      <span class="number">3.88</span>     <span class="number">940</span></span><br><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>mutations<span class="punctuation">)</span></span><br><span class="line">  Identifier TTN PTEN TP53 EGFR MUC16 FLG RYR2 PCLO PIK3R1 PIK3CA NF1 MUC17 HMCN1 SPTA1 USH2A RB1 PKHD1 OBSCN AHNAK2 RYR3 RELN FRAS1 GPR98 DNAH5 ATRX APOB TCHH SYNE1 LRP2 KEL HRNR DNAH3 COL6A3 MUC5B LAMA1 DSP</span><br><span class="line"><span class="number">1</span>    <span class="number">02</span><span class="operator">-</span><span class="number">0003</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">1</span>    <span class="number">1</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">1</span>      <span class="number">0</span>   <span class="number">0</span>     <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">1</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line"><span class="number">2</span>    <span class="number">02</span><span class="operator">-</span><span class="number">0033</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>      <span class="number">1</span>   <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line"><span class="number">3</span>    <span class="number">02</span><span class="operator">-</span><span class="number">0047</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>      <span class="number">0</span>      <span class="number">1</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line"><span class="number">4</span>    <span class="number">02</span><span class="operator">-</span><span class="number">0055</span>   <span class="number">1</span>    <span class="number">1</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">1</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line"><span class="number">5</span>    <span class="number">02</span><span class="operator">-</span><span class="number">2470</span>   <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line"><span class="number">6</span>    <span class="number">02</span><span class="operator">-</span><span class="number">2483</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>    <span class="number">1</span>      <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">1</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>   <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span></span><br><span class="line">  DNAH8 CNTNAP2 SDK1 NBPF10 DNAH2 NLRP5 MLL3 IDH1 HCN1 FCGBP DOCK5 RIMS2 PCDHA1 MXRA5 HEATR7B2 GRIN2A FGD5 TMEM132D STAG2 SEMA3C SCN9A PRDM9 POM121L12 PIK3CG PDGFRA GABRA6 FLG2 FBN3 FBN2 FAT2 DNAH11 DMD COL1A2</span><br><span class="line"><span class="number">1</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">1</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">2</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">3</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">1</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">4</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">1</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">5</span>     <span class="number">0</span>       <span class="number">1</span>    <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">6</span>     <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>    <span class="number">1</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>        <span class="number">0</span>     <span class="number">0</span>      <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>         <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>   <span class="number">0</span>      <span class="number">0</span></span><br><span class="line">  ABCC9 XIRP2 TSHZ2 TEX15 SLIT3 RBM47 PIK3C2G PCDH11X MYH2 MACF1 KSR2 DNAH9 DCHS2 CSMD3 CDH18 BCOR AHNAK ZAN TRRAP THSD7B TAF1L SPAG17 SLCO5A1 SCN10A RYR1 RIMBP2 PLEKHG4B PCDHB7 NPTX2 NOS1 LZTR1</span><br><span class="line"><span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span></span><br><span class="line"><span class="number">2</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span></span><br><span class="line"><span class="number">3</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span></span><br><span class="line"><span class="number">4</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>        <span class="number">1</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">1</span>     <span class="number">0</span></span><br><span class="line"><span class="number">5</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">1</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">0</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">1</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span></span><br><span class="line"><span class="number">6</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>       <span class="number">0</span>       <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span>   <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>      <span class="number">0</span>       <span class="number">1</span>      <span class="number">0</span>    <span class="number">0</span>      <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>     <span class="number">0</span>    <span class="number">0</span>     <span class="number">0</span></span><br></pre></td></tr></table></figure>

<h3 id="1-2-2-orthofinder数据"><a href="#1-2-2-orthofinder数据" class="headerlink" title="1.2.2. orthofinder数据"></a>1.2.2. orthofinder数据</h3><p>从orthofinder的结果文件Results_Aug14&#x2F;Orthogroups&#x2F;Orthogroups.GeneCount.tsv稍加处理就可以作为输入文件，展示不同物种的orthogroups集合的共享情况。</p>
<p><code>sed -E &quot;s/\t[1-9][0-9]*/\t1/g&quot; Orthogroups.GeneCount.tsv |sed &quot;s/\.pep//g&quot; &gt;orthogroups.upset</code> # 把Orthogroups.GeneCount.tsv中的非零数字替换成1。</p>
<p>在R中用<code>mutations &lt;- read.csv(&quot;orthogroups.upset&quot;, header=TRUE, sep = &quot;\t&quot;)</code>读取orthogroups.upset。</p>
<h2 id="1-3-UpSetR包安装"><a href="#1-3-UpSetR包安装" class="headerlink" title="1.3. UpSetR包安装"></a>1.3. UpSetR包安装</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">install.packages(&quot;UpSetR&quot;); # 安装</span><br><span class="line">library(UpSetR); # 载入UpSetR</span><br><span class="line">require(ggplot2); require(plyr); require(gridExtra); require(grid); # 载入包</span><br></pre></td></tr></table></figure>

<h2 id="1-4-UpSetR包使用"><a href="#1-4-UpSetR包使用" class="headerlink" title="1.4. UpSetR包使用"></a>1.4. UpSetR包使用</h2><h3 id="1-4-1-upset函数"><a href="#1-4-1-upset函数" class="headerlink" title="1.4.1. upset函数"></a>1.4.1. upset函数</h3><ol>
<li><p><code>upset(mutations)</code>可以看到upset图的效果</p>
</li>
<li><p>调整参数做指定数据显示</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">upset<span class="punctuation">(</span>mutations<span class="punctuation">,</span> </span><br><span class="line">sets <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;MUC16&quot;</span><span class="punctuation">,</span><span class="string">&quot;EGFR&quot;</span><span class="punctuation">,</span><span class="string">&quot;TP53&quot;</span><span class="punctuation">,</span><span class="string">&quot;TTN&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span><span class="comment"># 查看特定的几个集合/几种电影类别</span></span><br><span class="line">nset <span class="operator">=</span> <span class="number">4</span><span class="punctuation">,</span> <span class="comment"># 最多展示多少个集合数据</span></span><br><span class="line">nintersects <span class="operator">=</span> <span class="number">20</span><span class="punctuation">,</span> <span class="comment"># 展示数量多的前多少个交集</span></span><br><span class="line">mb.ratio <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">0.55</span><span class="punctuation">,</span> <span class="number">0.45</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 控制上方条形图以及下方点图的比例</span></span><br><span class="line">order.by <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;degree&quot;</span><span class="punctuation">,</span> <span class="string">&quot;freq&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 交集如何排序，这里先根据degree(数量，交集包含的数字大小)，然后再根据freq(频率，涉及到的交集个数)</span></span><br><span class="line">keep.order <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">,</span> <span class="comment"># keep.order按照sets参数的顺序排序</span></span><br><span class="line">decreasing <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="literal">TRUE</span><span class="punctuation">,</span><span class="literal">FALSE</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 变量如何排序；对应order.by参数。这里表示degree降序，freq升序。</span></span><br><span class="line">number.angles <span class="operator">=</span> <span class="number">30</span><span class="punctuation">,</span> <span class="comment"># 调整柱形图上数字角度</span></span><br><span class="line">point.size <span class="operator">=</span> <span class="number">2</span><span class="punctuation">,</span> line.size <span class="operator">=</span> <span class="number">1</span><span class="punctuation">,</span> <span class="comment"># 点和线的大小</span></span><br><span class="line">mainbar.y.label <span class="operator">=</span> <span class="string">&quot;Intersection size of gene family&quot;</span><span class="punctuation">,</span>sets.x.label <span class="operator">=</span> <span class="string">&quot;genome size&quot;</span><span class="punctuation">,</span> <span class="comment"># 坐标轴名称</span></span><br><span class="line">text.scale <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">1.3</span><span class="punctuation">,</span> <span class="number">1.3</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">,</span> <span class="number">1.5</span><span class="punctuation">,</span> <span class="number">1</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="comment"># 六个数字，分别控制c(intersectionsize title, intersection size tick labels, set size title, set size ticklabels, set names, numbers above bars)</span></span><br></pre></td></tr></table></figure>

<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/3cabd7b9fd4f1552a69bf5f1797f489be352717a/source/images/R_plot_upset_movies.png?raw=true" width=100% height=60% title="upset_movies" align=center/>

<p><strong><p align="center">Figure 1. movies upset</p></strong></p>
<h3 id="1-4-2-queries参数"><a href="#1-4-2-queries参数" class="headerlink" title="1.4.2. queries参数"></a>1.4.2. queries参数</h3><p>upset函数中可以添加queries参数，用于突出显示（上色）部分数据。</p>
<ol>
<li><p>queries<br>queries是一个由多个query组成的list；每个query也是一个list，作为一次查询数据并突出显示的请求。</p>
</li>
<li><p>query<br>query也是list格式，由查询函数query和其他参数（param,color,active,query.name）组成，其中query,param是必须设置的参数。</p>
<ul>
<li>query: 指定查询函数，UpSetR有内置(比如intersects)，也可以自定义函数后调用。</li>
<li>param: list格式, 指定query作用的数据。</li>
<li>color：设置颜色，可选设置。</li>
<li>active：显示类型，TRUE&#x2F;T表示用颜色覆盖条形图，FALSE&#x2F;F表示在条形图顶端显示三角形。</li>
<li>query.name：添加query图例名称。</li>
</ul>
</li>
<li><p>queries参数示例</p>
</li>
</ol>
<ul>
<li>把”EGFR”和”TP53”两个基因共同拥有的突变标上蓝色；把”TTN”基因特有的突变在直方图上标为红色。</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">upset<span class="punctuation">(</span>mutations<span class="punctuation">,</span> sets<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;MUC16&quot;</span><span class="punctuation">,</span><span class="string">&quot;EGFR&quot;</span><span class="punctuation">,</span><span class="string">&quot;TP53&quot;</span><span class="punctuation">,</span><span class="string">&quot;TTN&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> </span><br><span class="line">queries <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span>  </span><br><span class="line">params <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;EGFR&quot;</span><span class="punctuation">,</span> <span class="string">&quot;TP53&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 指定作用的数据</span></span><br><span class="line">color <span class="operator">=</span> <span class="string">&quot;blue&quot;</span><span class="punctuation">,</span> <span class="comment"># 设置颜色，未设置会调用默认调色板</span></span><br><span class="line">active <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span>   <span class="comment"># 条形图被颜色覆盖</span></span><br><span class="line">query.name <span class="operator">=</span> <span class="string">&quot;share EGFR and TP53&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 添加query图例</span></span><br><span class="line"><span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span> params<span class="operator">=</span><span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;TTN&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&quot;red&quot;</span><span class="punctuation">,</span> active<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/R_plot_upset_mutations.png?raw=true" width=100% height=60% title="upset_mutations" align=center/>

<p><strong><p align="center">Figure 2. mutations upset</p></strong></p>
<ul>
<li>把同属Drama和Thriller的电影突出显示，把1970-1980的电影标红。<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">between &lt;- function(row, min, max)&#123;</span><br><span class="line">  newData &lt;- (row[&quot;ReleaseDate&quot;] &lt; max) &amp; (row[&quot;ReleaseDate&quot;] &gt; min)</span><br><span class="line">&#125; # 自定义between函数</span><br><span class="line"></span><br><span class="line">upset(movies, sets=c(&quot;Drama&quot;,&quot;Comedy&quot;,&quot;Action&quot;,&quot;Thriller&quot;,&quot;Western&quot;,&quot;Documentary&quot;),</span><br><span class="line">      queries = list(list(query = intersects, params = list(&quot;Drama&quot;, &quot;Thriller&quot;)),</span><br><span class="line">                     list(query = between, params=list(1970,1980), color=&quot;red&quot;, active=TRUE)))</span><br></pre></td></tr></table></figure></li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/R_plot_upset_movies2.png?raw=true" width=100% height=60% title="upset_movies2" align=center/>

<p><strong><p align="center">Figure 3. movies upset 2</p></strong></p>
<h3 id="1-4-3-添加属性图"><a href="#1-4-3-添加属性图" class="headerlink" title="1.4.3. 添加属性图"></a>1.4.3. 添加属性图</h3><ol>
<li>添加箱线图<br>每次最多添加两个箱线图<br><code>upset(movies, boxplot.summary = c(&quot;AvgRating&quot;, &quot;ReleaseDate&quot;)) </code></li>
</ol>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/R_plot_upset_boxplot.png?raw=true" width=100% height=70% title="upset_boxplot" align=center/>

<p><strong><p align="center">Figure 4. movies upset boxplot</p></strong></p>
<h4 id="1-4-3-1-attribute-plots参数"><a href="#1-4-3-1-attribute-plots参数" class="headerlink" title="1.4.3.1. attribute.plots参数"></a>1.4.3.1. attribute.plots参数</h4><p>attribute.plots参数用于添加属性图，内置有柱形图，散点图，热图等。<br>如果想添加密度曲线图，可以自定义plot函数后添加。</p>
<ol>
<li>添加柱形图和散点图<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">upset<span class="punctuation">(</span>movies<span class="punctuation">,</span> sets<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;Drama&quot;</span><span class="punctuation">,</span><span class="string">&quot;Comedy&quot;</span><span class="punctuation">,</span><span class="string">&quot;Action&quot;</span><span class="punctuation">,</span><span class="string">&quot;Thriller&quot;</span><span class="punctuation">,</span><span class="string">&quot;Western&quot;</span><span class="punctuation">,</span><span class="string">&quot;Documentary&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">      queries <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span> params <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;Drama&quot;</span><span class="punctuation">,</span> <span class="string">&quot;Thriller&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                     <span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> between<span class="punctuation">,</span> params<span class="operator">=</span><span class="built_in">list</span><span class="punctuation">(</span><span class="number">1970</span><span class="punctuation">,</span><span class="number">1980</span><span class="punctuation">)</span><span class="punctuation">,</span> color<span class="operator">=</span><span class="string">&quot;red&quot;</span><span class="punctuation">,</span> active<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">      attribute.plots<span class="operator">=</span><span class="built_in">list</span><span class="punctuation">(</span>gridrows<span class="operator">=</span><span class="number">60</span><span class="punctuation">,</span> <span class="comment"># 添加属性图</span></span><br><span class="line">      plots<span class="operator">=</span><span class="built_in">list</span><span class="punctuation">(</span></span><br><span class="line">        <span class="built_in">list</span><span class="punctuation">(</span>plot<span class="operator">=</span>scatter_plot<span class="punctuation">,</span> <span class="comment"># 散点图</span></span><br><span class="line">        x<span class="operator">=</span><span class="string">&quot;ReleaseDate&quot;</span><span class="punctuation">,</span> y<span class="operator">=</span><span class="string">&quot;AvgRating&quot;</span><span class="punctuation">,</span> <span class="comment"># 指定横纵坐标</span></span><br><span class="line">        queries <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># T表示显示queries定义的颜色</span></span><br><span class="line">        <span class="built_in">list</span><span class="punctuation">(</span>plot<span class="operator">=</span> histogram<span class="punctuation">,</span> x<span class="operator">=</span><span class="string">&quot;ReleaseDate&quot;</span><span class="punctuation">,</span> queries <span class="operator">=</span> <span class="built_in">F</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 直方图</span></span><br><span class="line">        ncols <span class="operator">=</span> <span class="number">2</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="comment"># 添加的图分两列</span></span><br><span class="line">      query.legend <span class="operator">=</span> <span class="string">&quot;top&quot;</span><span class="punctuation">)</span> <span class="comment"># query图例放在上方</span></span><br></pre></td></tr></table></figure></li>
</ol>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/3cabd7b9fd4f1552a69bf5f1797f489be352717a/source/images/R_plot_upset_scatter_histogram.png?raw=true" width=100% height=70% title="upset_scatter_histograms" align=center/>

<p><strong><p align="center">Figure 5. movies upset scatter histograms</p></strong></p>
<ol start="2">
<li>添加密度曲线图</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#自定义密度曲线</span></span><br><span class="line">another.plot <span class="operator">&lt;-</span> <span class="keyword">function</span><span class="punctuation">(</span>data<span class="punctuation">,</span> x<span class="punctuation">,</span> y<span class="punctuation">)</span> <span class="punctuation">&#123;</span></span><br><span class="line">    data<span class="operator">$</span>decades <span class="operator">&lt;-</span> round_any<span class="punctuation">(</span><span class="built_in">as.integer</span><span class="punctuation">(</span>unlist<span class="punctuation">(</span>data<span class="punctuation">[</span>y<span class="punctuation">]</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="number">10</span><span class="punctuation">,</span> <span class="built_in">ceiling</span><span class="punctuation">)</span></span><br><span class="line">    data <span class="operator">&lt;-</span> data<span class="punctuation">[</span>which<span class="punctuation">(</span>data<span class="operator">$</span>decades <span class="operator">&gt;=</span> <span class="number">1970</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="punctuation">]</span></span><br><span class="line">    myplot <span class="operator">&lt;-</span> <span class="punctuation">(</span>ggplot<span class="punctuation">(</span>data<span class="punctuation">,</span> aes_string<span class="punctuation">(</span>x <span class="operator">=</span> x<span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span> geom_density<span class="punctuation">(</span>aes<span class="punctuation">(</span>fill <span class="operator">=</span> factor<span class="punctuation">(</span>decades<span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> </span><br><span class="line">        alpha <span class="operator">=</span> <span class="number">0.4</span><span class="punctuation">)</span> <span class="operator">+</span> theme<span class="punctuation">(</span>plot.margin <span class="operator">=</span> unit<span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="number">0</span><span class="punctuation">,</span> <span class="number">0</span><span class="punctuation">,</span> <span class="number">0</span><span class="punctuation">,</span> <span class="number">0</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="string">&quot;cm&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> legend.key.size <span class="operator">=</span> unit<span class="punctuation">(</span><span class="number">0.4</span><span class="punctuation">,</span> <span class="string">&quot;cm&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">&#125;</span></span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>plyr<span class="punctuation">)</span></span><br><span class="line">upset<span class="punctuation">(</span>movies<span class="punctuation">,</span> main.bar.color <span class="operator">=</span> <span class="string">&quot;black&quot;</span><span class="punctuation">,</span> mb.ratio <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">0.5</span><span class="punctuation">,</span> <span class="number">0.5</span><span class="punctuation">)</span><span class="punctuation">,</span> queries <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span> </span><br><span class="line">    params <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;Drama&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> color <span class="operator">=</span> <span class="string">&quot;red&quot;</span><span class="punctuation">,</span> active <span class="operator">=</span> <span class="built_in">F</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span> </span><br><span class="line">    params <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;Action&quot;</span><span class="punctuation">,</span> <span class="string">&quot;Drama&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> active <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="built_in">list</span><span class="punctuation">(</span>query <span class="operator">=</span> intersects<span class="punctuation">,</span> </span><br><span class="line">    params <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="string">&quot;Drama&quot;</span><span class="punctuation">,</span> <span class="string">&quot;Comedy&quot;</span><span class="punctuation">,</span> <span class="string">&quot;Action&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> color <span class="operator">=</span> <span class="string">&quot;orange&quot;</span><span class="punctuation">,</span> active <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> </span><br><span class="line">    attribute.plots <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span>gridrows <span class="operator">=</span> <span class="number">50</span><span class="punctuation">,</span> plots <span class="operator">=</span> <span class="built_in">list</span><span class="punctuation">(</span><span class="built_in">list</span><span class="punctuation">(</span>plot <span class="operator">=</span> histogram<span class="punctuation">,</span> </span><br><span class="line">        x <span class="operator">=</span> <span class="string">&quot;ReleaseDate&quot;</span><span class="punctuation">,</span> queries <span class="operator">=</span> <span class="built_in">F</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="built_in">list</span><span class="punctuation">(</span>plot <span class="operator">=</span> scatter_plot<span class="punctuation">,</span> x <span class="operator">=</span> <span class="string">&quot;ReleaseDate&quot;</span><span class="punctuation">,</span> </span><br><span class="line">        y <span class="operator">=</span> <span class="string">&quot;AvgRating&quot;</span><span class="punctuation">,</span> queries <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="built_in">list</span><span class="punctuation">(</span>plot <span class="operator">=</span> another.plot<span class="punctuation">,</span> x <span class="operator">=</span> <span class="string">&quot;AvgRating&quot;</span><span class="punctuation">,</span> </span><br><span class="line">        y <span class="operator">=</span> <span class="string">&quot;ReleaseDate&quot;</span><span class="punctuation">,</span> queries <span class="operator">=</span> <span class="built_in">F</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="punctuation">,</span> ncols <span class="operator">=</span> <span class="number">3</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/3cabd7b9fd4f1552a69bf5f1797f489be352717a/source/images/R_plot_upset_density.png?raw=true" width=100% height=70% title="upset_density" align=center/>

<p><strong><p align="center">Figure 6. movies upset density</p></strong></p>
<h1 id="2-references"><a href="#2-references" class="headerlink" title="2. references"></a>2. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://github.com/hms-dbmi/UpSetR">https://github.com/hms-dbmi/UpSetR</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/324aae3d5ea4">https://www.jianshu.com/p/324aae3d5ea4</a></li>
<li><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/35303590">https://zhuanlan.zhihu.com/p/35303590</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/09/29/R_plot_upset/" data-id="cl1aiqcr00010m4vk8gkw9ndh" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/plot/" rel="tag">plot</a></li></ul>

    </footer>
  </div>
  
</article>


  


  <nav id="page-nav">
    
    <a class="extend prev" rel="prev" href="/">&amp;laquo; 上一页</a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><a class="page-number" href="/page/4/">4</a><a class="page-number" href="/page/5/">5</a><a class="extend next" rel="next" href="/page/3/">下一页 &amp;raquo;</a>
  </nav>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>