<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="用MCscan pipeline做两个基因组间的同线性分析，包括制作同线性点图(dotplot)，同线性深度直方图，同线性染色体图。">
<meta property="og:type" content="article">
<meta property="og:title" content="基因组间的同线性分析 —— MCscan">
<meta property="og:url" content="http://example.com/2021/11/05/bioinfo_synteny_MCscan/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="用MCscan pipeline做两个基因组间的同线性分析，包括制作同线性点图(dotplot)，同线性深度直方图，同线性染色体图。">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://www.dropbox.com/s/32mmo0kfhtx5b8g/grape.peach.cscore.99.png?raw=1">
<meta property="og:image" content="https://www.dropbox.com/s/hhx2dtryrum2gyo/grape.peach.depth.png?raw=1">
<meta property="og:image" content="https://www.dropbox.com/s/51k8jujyzage3oa/grape.peach.karyotype.png?raw=1">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2021-11-05T07:10:00.000Z">
<meta property="article:modified_time" content="2024-06-15T03:52:11.679Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="biosoft">
<meta property="article:tag" content="synteny">
<meta property="article:tag" content="colinearity">
<meta property="article:tag" content="MCscan">
<meta property="article:tag" content="jcvi">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://www.dropbox.com/s/32mmo0kfhtx5b8g/grape.peach.cscore.99.png?raw=1">

<link rel="canonical" href="http://example.com/2021/11/05/bioinfo_synteny_MCscan/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>基因组间的同线性分析 —— MCscan | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2021/11/05/bioinfo_synteny_MCscan/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          基因组间的同线性分析 —— MCscan
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-11-05 15:10:00" itemprop="dateCreated datePublished" datetime="2021-11-05T15:10:00+08:00">2021-11-05</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2024-06-15 11:52:11" itemprop="dateModified" datetime="2024-06-15T11:52:11+08:00">2024-06-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/" itemprop="url" rel="index"><span itemprop="name">bioinfo</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bioinfo/synteny/" itemprop="url" rel="index"><span itemprop="name">synteny</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span><br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>4.8k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>4 mins.</span>
            </span>
            <div class="post-description">用MCscan pipeline做两个基因组间的同线性分析，包括制作同线性点图(dotplot)，同线性深度直方图，同线性染色体图。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=5250805&auto=1&height=32"></iframe></div>

<h1 id="1-MCscan背景"><a href="#1-MCscan背景" class="headerlink" title="1. MCscan背景"></a>1. MCscan背景</h1><h2 id="1-1-jcvi介绍"><a href="#1-1-jcvi介绍" class="headerlink" title="1.1. jcvi介绍"></a>1.1. jcvi介绍</h2><p><a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi">jcvi</a>是美国佐治亚大学唐海宝开发的一个生物信息学python库。主要有几个常见pipeline用到了jcvi库，一个是做基因组scaffoldding的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/ALLMAPS">ALLMAPS</a>，另一个是做基因组同线性分析的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan</a>，以及提取植物表型数据的<a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/GRABSEEDS">GRABSEEDS</a>。</p>
<h2 id="1-2-MCscan介绍"><a href="#1-2-MCscan介绍" class="headerlink" title="1.2. MCscan介绍"></a>1.2. MCscan介绍</h2><p>用MCscan pipeline做两个基因组间的同线性分析，包括成对同线性(pairwise synteny)分析，宏观同线性(macrosynteny)分析，微观同线性(microsynteny)分析和结果的可视化(visualization)。</p>
<p>成对同线性分析主要是制作同线性点图(dotplot),同线性深度直方图；宏观同线性分析主要是制作同线性染色体图。其中同线性深度直方图，同线性染色体图需要用到同线性点图的结果文件作为输入。</p>
<p>微观同线性分析等后面用到再补坑。</p>
<h1 id="2-MCscan使用"><a href="#2-MCscan使用" class="headerlink" title="2. MCscan使用"></a>2. MCscan使用</h1><h2 id="2-1-安装jcvi"><a href="#2-1-安装jcvi" class="headerlink" title="2.1. 安装jcvi"></a>2.1. 安装jcvi</h2><h3 id="2-1-1-依赖"><a href="#2-1-1-依赖" class="headerlink" title="2.1.1. 依赖"></a>2.1.1. 依赖</h3><p>jcvi依赖包括biopython,numpy,matplotlib。</p>
<p>MCscan依赖包括LASTAL。</p>
<p>LASTAL可以用<code>conda install -c bioconda last</code>安装，但我用conda安装的lastal和lastdb不能多线程运行，所以在运行<code>python -m jcvi.compara.catalog ortholog --cpu=1</code>时用参数<code>--cpu=1</code>来限制单线程运行。</p>
<h3 id="2-1-2-安装jcvi"><a href="#2-1-2-安装jcvi" class="headerlink" title="2.1.2. 安装jcvi"></a>2.1.2. 安装jcvi</h3><p>任意选一种安装方式</p>
<ol>
<li><code>pip install jcvi</code></li>
<li><code>pip install git+git://github.com/tanghaibao/jcvi.git</code></li>
<li><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cd jcvi</span><br><span class="line">git clone git://github.com/tanghaibao/jcvi.git</span><br><span class="line">pip install -e .</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="2-2-准备输入文件"><a href="#2-2-准备输入文件" class="headerlink" title="2.2. 准备输入文件"></a>2.2. 准备输入文件</h2><p>输入文件包括两个基因组的CDS&#x2F;氨基酸序列文件（fa格式），两个基因组的注释坐标文件（bed格式）。</p>
<h3 id="2-2-1-jcvi支持直接从phytozome下载数据"><a href="#2-2-1-jcvi支持直接从phytozome下载数据" class="headerlink" title="2.2.1. jcvi支持直接从phytozome下载数据"></a>2.2.1. jcvi支持直接从phytozome下载数据</h3><ol>
<li><p>连接phytozome（需要登录，提前注册账号）<br><code>python -m jcvi.apps.fetch phytozome</code> 输入账户登录信息后，出现可以下载的基因组list。</p>
</li>
<li><p>下载CDS序列和注释文件(gff格式)，例如下载这两个物种:grape and peach<br><code>python -m jcvi.apps.fetch phytozome Vvinifera,Ppersica</code></p>
</li>
<li><p>转换GFF为BED格式<br><code>python -m jcvi.formats.gff bed --type=mRNA --key=Name  --primary_only Vvinifera_145_gene.gff3.gz -o grape.bed</code></p>
</li>
</ol>
<p><code>python -m jcvi.formats.gff bed --type=mRNA --key=Name  --primary_only Ppersica_139_gene.gff3.gz -o peach.bed</code></p>
<p>其中<code>--primary_only</code>是用于去除多个转录本的，只保留一个。（原文是多个isoforms）一般不去也不影响后续步骤，加上更好吧。</p>
<ol start="4">
<li>整理一下，获得CDS文件<br><code>python -m jcvi.formats.fasta format Vvinifera_145_Genoscope.12X.cds.fa.gz grape.cds</code></li>
</ol>
<p><code>python -m jcvi.formats.fasta format Ppersica_298_v2.1.cds.fa.gz peach.cds</code></p>
<h3 id="2-2-2-自己的数据"><a href="#2-2-2-自己的数据" class="headerlink" title="2.2.2. 自己的数据"></a>2.2.2. 自己的数据</h3><ol>
<li>gff2bed<br><code>python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only sampleA.gff3.gz &gt; sampleA.bed</code></li>
</ol>
<p><code>python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only sampleB.gff3.gz &gt; sampleB.bed</code></p>
<p>其中<code>--primary_only</code>是用于去除多个转录本的，只保留一个。（原文是多个isoforms）一般不去也不影响后续步骤，加上更好吧。</p>
<ol start="2">
<li><p>去重，会生成sampleA.uniq.bed和sampleB.uniq.bed，然后重命名</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">python -m jcvi.formats.bed uniq sampleA.bed</span><br><span class="line">mv sampleA.uniq.bed sampleA.bed</span><br><span class="line">python -m jcvi.formats.bed uniq sampleB.bed</span><br><span class="line">mv sampleB.uniq.bed sampleB.bed</span><br></pre></td></tr></table></figure>
</li>
<li><p>提取cds和pep</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">seqkit grep -f &lt;(cut -f 4 sampleA.uniq.bed ) sampleA.cdna.all.fa.gz | seqkit seq -i &gt; sampleA.cds</span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleA.uniq.bed ) sampleB.pep.all.fa.gz | seqkit seq -i &gt; sampleA.pep </span><br><span class="line"></span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleB.uniq.bed )  sampeB.cdna.fa.gz | seqkit seq -i  &gt; sampleB.cds</span><br><span class="line">seqkit grep -f &lt;(cut -f 4 sampleB.uniq.bed ) sampleB.pep.fa.gz | seqkit seq -i  &gt; sampleB.pep</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id="2-3-同线性点图"><a href="#2-3-同线性点图" class="headerlink" title="2.3. 同线性点图"></a>2.3. 同线性点图</h2><h3 id="2-3-1-CDS的同线性点图dotplot"><a href="#2-3-1-CDS的同线性点图dotplot" class="headerlink" title="2.3.1. CDS的同线性点图dotplot"></a>2.3.1. CDS的同线性点图dotplot</h3><ol>
<li>确保<code>sampleA.cds</code>,<code>sampleA.bed</code>,<code>sampleB.cds</code>,<code>sampleB.bed</code>四个文件在同一目录下。</li>
<li>运行<code>python -m jcvi.compara.catalog ortholog --no_strip_names --cpu=1 sampleA sampleB</code>制作同线性点图。</li>
</ol>
<p><code>--cpu=1</code>是因为conda安装的lastal不支持多线程，所以用单线程。</p>
<ol start="3">
<li>运行时如果报错<code>ERROR savefig failed. Reset usetex to False</code>是缺少dvipng包没能生成png格式图，但看看结果文件sampleA.sampleB.anchors还是完整生成了就可以进行下一步。</li>
</ol>
<h3 id="2-3-2-蛋白的同线性点图"><a href="#2-3-2-蛋白的同线性点图" class="headerlink" title="2.3.2. 蛋白的同线性点图"></a>2.3.2. 蛋白的同线性点图</h3><p>与CDS一样，cds数据换成pep数据，在运行参数里加上数据类型<code>--dbtype prot</code>即可。</p>
<p><code>python -m jcvi.compara.catalog ortholog --dbtype prot --no_strip_names --cpu=1 sampleA sampleB</code></p>
<p><code>--cpu=1</code>是因为conda安装的lastal不支持多线程，所以用单线程。</p>
<h3 id="2-3-3-结果"><a href="#2-3-3-结果" class="headerlink" title="2.3.3. 结果"></a>2.3.3. 结果</h3><p>会生成五个结果文件。</p>
<ul>
<li>sampleA.sampleB.last: 基于LAST的比对结果</li>
<li>sampleA.sampleB.last.filtered: LAST的比对结果过滤串联重复和低分比对</li>
<li>sampleA.sampleB.anchors: 高质量的同线性区块</li>
<li>sampleA.sampleB.lifted.anchors:增加了额外的锚点，形成最终的同线性区块</li>
<li>sampleA.sampleB.pdf:同线性点图</li>
</ul>
<p>同线性点图一般可以看点连成的比较长的线，在同一横坐标范围有几根线，或者在同一纵坐标范围有几根线，从而判断两个基因组的倍数关系。</p>
<img src="https://www.dropbox.com/s/32mmo0kfhtx5b8g/grape.peach.cscore.99.png?raw=1" width=100% title="同线性点图" alt="同线性点图" align=center/>

<p><strong><p align="center">Figure 1. 同线性点图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi viki</a></p></p>
<h2 id="2-4-同线性深度直方图"><a href="#2-4-同线性深度直方图" class="headerlink" title="2.4. 同线性深度直方图"></a>2.4. 同线性深度直方图</h2><ol>
<li>在同线性点图运行成功获得sampleA.sampleB.anchors结果文件的前提下</li>
<li>运行<code>python -m jcvi.compara.synteny depth --histogram sampleA.sampleB.anchors</code></li>
<li>生成sampleA.sampleB.depth.pdf，显示了同线性深度比例。</li>
</ol>
<img src="https://www.dropbox.com/s/hhx2dtryrum2gyo/grape.peach.depth.png?raw=1" width=100% title="同线性深度直方图" alt="同线性深度直方图" align=center/>

<p><strong><p align="center">Figure 2. 同线性深度直方图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi viki</a></p></p>
<h2 id="2-5-同线性染色体图"><a href="#2-5-同线性染色体图" class="headerlink" title="2.5. 同线性染色体图"></a>2.5. 同线性染色体图</h2><h3 id="2-5-1-输入文件"><a href="#2-5-1-输入文件" class="headerlink" title="2.5.1. 输入文件"></a>2.5.1. 输入文件</h3><ol>
<li>seqids文件</li>
</ol>
<ul>
<li>指定展示的染色体ID，两行对应两个物种，每行的染色体之间逗号隔开，不能有空行。</li>
<li>seqids文件染色体的顺序对应同线性染色体图的顺序，可以根据前面得到的同线性点图来调整顺序使两个基因组的同线性对应，使图更清晰。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">scaf001,scaf002,scaf003,scaf004,scaf005,scaf006,scaf007,scaf008,scaf009,scaf010,scaf011,scaf012,scaf013,scaf014</span><br><span class="line">scaffold1,scaffold2,scaffold3,scaffold4,scaffold5,scaffold6,scaffold7</span><br></pre></td></tr></table></figure>

<ol start="2">
<li>.simple文件</li>
</ol>
<ul>
<li>每行代表一个同线性区块</li>
<li>一共六列，前四列是两个物种的同线性区块的上下限基因，第五列是评分，第六列是方向。</li>
<li>用前面分析同线性点图的结果文件.anchors创建。</li>
<li>运行<code>python -m jcvi.compara.synteny screen --minspan=30 --simple sampleA.sampleB.anchors sampleA.sampleB.anchors.new</code>会生成sampleA.sampleB.anchors.simple文件。</li>
</ul>
<ol start="3">
<li>layout文件</li>
</ol>
<ul>
<li>设置绘制参数。前三列是设置图的位置坐标信息，整个图的x和y轴都是[0-1]范围。第四列是设置旋转角度，第五列是设置染色体颜色，第六列是展示的标签，第七列是对齐方式，第八列指定bed文件。</li>
<li>注意, #edges下的每一行开头都不能有空格。用于指定simple文件。<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"># y, xstart, xend, rotation, color, label, va,  bed</span><br><span class="line"> .6,     .2,    .8,       0,      lightblue, sample_A, top, sampleA.bed</span><br><span class="line"> .4,     .2,    .8,       0,      lightpink, sample_B, top, sampleB.bed</span><br><span class="line"># edges</span><br><span class="line">e, 0, 1, sampleA.sampleB.anchors.simple</span><br></pre></td></tr></table></figure></li>
</ul>
<h3 id="2-5-2-运行"><a href="#2-5-2-运行" class="headerlink" title="2.5.2. 运行"></a>2.5.2. 运行</h3><p><code>python -m jcvi.graphics.karyotype seqids layout</code></p>
<p>生成karyotype.pdf文件，为同线性双染色体图</p>
<img src="https://www.dropbox.com/s/51k8jujyzage3oa/grape.peach.karyotype.png?raw=1" width=100% title="同线性双染色体图" alt="同线性双染色体图" align=center/>

<p><strong><p align="center">Figure 3. 同线性双染色体图结果示例</strong><br>from <a target="_blank" rel="noopener" href="https://github-wiki-see.page/m/tanghaibao/jcvi/wiki/MCscan-%28Python-version%29">MCscan jcvi wiki</a></p></p>
<h3 id="2-5-3-tips"><a href="#2-5-3-tips" class="headerlink" title="2.5.3. tips"></a>2.5.3. tips</h3><ul>
<li>运行虽然报错<code>ERROR    savefig failed. Reset usetex to False.</code>，应该是没能生成png图，但还是生成了pdf图文件。</li>
<li>当label用的名称过长时，label和图容易重叠，没有好的办法解决，最好用短一点的label。</li>
</ul>
<h3 id="2-5-4-个性化"><a href="#2-5-4-个性化" class="headerlink" title="2.5.4. 个性化"></a>2.5.4. 个性化</h3><h4 id="2-5-4-1-highlight指定同线性区块"><a href="#2-5-4-1-highlight指定同线性区块" class="headerlink" title="2.5.4.1. highlight指定同线性区块"></a>2.5.4.1. highlight指定同线性区块</h4><p>如果要突出显示指定的同线性区块（一般用于展示染色体倍数关系），则在sampleA.sampleB.anchors.sample文件中的对应同线性区域行的前面加上<code>yellowgreen*</code>,表示这行的颜色用yellowgreen展示。</p>
<p>然后再运行<code>python -m jcvi.graphics.karyotype seqids layout</code>就会生成突出显示的图。</p>
<h1 id="3-references"><a href="#3-references" class="headerlink" title="3. references"></a>3. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version)">https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version)</a></li>
<li><a target="_blank" rel="noopener" href="https://xuzhougeng.top/archives/Comparative-genomics-using-JCVI-part-one">https://xuzhougeng.top/archives/Comparative-genomics-using-JCVI-part-one</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=20% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    
        <div class="reward-container">
  <div>真诚赞赏，手留余香</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    Donate
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/images/wechatpay.jpg" alt="南方枝叶 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
      
      <div style="display: inline-block;">
        <img src="/images/alipay.jpg" alt="南方枝叶 Alipay">
        <p>Alipay</p>
      </div>

  </div>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/biosoft/" rel="tag"># biosoft</a>
              <a href="/tags/synteny/" rel="tag"># synteny</a>
              <a href="/tags/colinearity/" rel="tag"># colinearity</a>
              <a href="/tags/MCscan/" rel="tag"># MCscan</a>
              <a href="/tags/jcvi/" rel="tag"># jcvi</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2021/10/29/bioinfo_alignment_align_pep2cds/" rel="prev" title="蛋白质比对转换成CDS比对 —— ParaAT,PAL2NAL">
      <i class="fa fa-chevron-left"></i> 蛋白质比对转换成CDS比对 —— ParaAT,PAL2NAL
    </a></div>
      <div class="post-nav-item">
    <a href="/2021/11/11/bioinfo_enrichment_intro/" rel="next" title="富集分析：（一）概述">
      富集分析：（一）概述 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-MCscan%E8%83%8C%E6%99%AF"><span class="nav-text">1. MCscan背景</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#1-1-jcvi%E4%BB%8B%E7%BB%8D"><span class="nav-text">1.1. jcvi介绍</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-2-MCscan%E4%BB%8B%E7%BB%8D"><span class="nav-text">1.2. MCscan介绍</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-MCscan%E4%BD%BF%E7%94%A8"><span class="nav-text">2. MCscan使用</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#2-1-%E5%AE%89%E8%A3%85jcvi"><span class="nav-text">2.1. 安装jcvi</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-1-1-%E4%BE%9D%E8%B5%96"><span class="nav-text">2.1.1. 依赖</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-1-2-%E5%AE%89%E8%A3%85jcvi"><span class="nav-text">2.1.2. 安装jcvi</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-2-%E5%87%86%E5%A4%87%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6"><span class="nav-text">2.2. 准备输入文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-2-1-jcvi%E6%94%AF%E6%8C%81%E7%9B%B4%E6%8E%A5%E4%BB%8Ephytozome%E4%B8%8B%E8%BD%BD%E6%95%B0%E6%8D%AE"><span class="nav-text">2.2.1. jcvi支持直接从phytozome下载数据</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-2-2-%E8%87%AA%E5%B7%B1%E7%9A%84%E6%95%B0%E6%8D%AE"><span class="nav-text">2.2.2. 自己的数据</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-3-%E5%90%8C%E7%BA%BF%E6%80%A7%E7%82%B9%E5%9B%BE"><span class="nav-text">2.3. 同线性点图</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-3-1-CDS%E7%9A%84%E5%90%8C%E7%BA%BF%E6%80%A7%E7%82%B9%E5%9B%BEdotplot"><span class="nav-text">2.3.1. CDS的同线性点图dotplot</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-3-2-%E8%9B%8B%E7%99%BD%E7%9A%84%E5%90%8C%E7%BA%BF%E6%80%A7%E7%82%B9%E5%9B%BE"><span class="nav-text">2.3.2. 蛋白的同线性点图</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-3-3-%E7%BB%93%E6%9E%9C"><span class="nav-text">2.3.3. 结果</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-4-%E5%90%8C%E7%BA%BF%E6%80%A7%E6%B7%B1%E5%BA%A6%E7%9B%B4%E6%96%B9%E5%9B%BE"><span class="nav-text">2.4. 同线性深度直方图</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-5-%E5%90%8C%E7%BA%BF%E6%80%A7%E6%9F%93%E8%89%B2%E4%BD%93%E5%9B%BE"><span class="nav-text">2.5. 同线性染色体图</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-5-1-%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6"><span class="nav-text">2.5.1. 输入文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-5-2-%E8%BF%90%E8%A1%8C"><span class="nav-text">2.5.2. 运行</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-5-3-tips"><span class="nav-text">2.5.3. tips</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-5-4-%E4%B8%AA%E6%80%A7%E5%8C%96"><span class="nav-text">2.5.4. 个性化</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#2-5-4-1-highlight%E6%8C%87%E5%AE%9A%E5%90%8C%E7%BA%BF%E6%80%A7%E5%8C%BA%E5%9D%97"><span class="nav-text">2.5.4.1. highlight指定同线性区块</span></a></li></ol></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-references"><span class="nav-text">3. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">141</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">95</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">510</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="/atom.xml" title="RSS → &#x2F;atom.xml"><i class="fa fa-rss fa-fw"></i>RSS</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

      <div class="wechat_channel">
        <br>
        <!-- 这里添加你的二维码图片 -->
        <img src ="/images/wechat_channel.jpg?raw=true">
        <span>欢迎关注微信公众号</span>
      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2025</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">736k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">11:09</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

</body>
</html>
