<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>多序列比对(Multiple Sequence Alignment,MSA) | 生信技工</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="记录了多序列比对(Multiple Sequence Alignment,MSA)及比对后过滤(trimming)的软件。">
<meta property="og:type" content="article">
<meta property="og:title" content="多序列比对(Multiple Sequence Alignment,MSA)">
<meta property="og:url" content="http://example.com/2021/09/06/bioinfo_MSA/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="记录了多序列比对(Multiple Sequence Alignment,MSA)及比对后过滤(trimming)的软件。">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2021-09-06T10:00:00.000Z">
<meta property="article:modified_time" content="2021-10-29T08:59:54.388Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="Multiple Sequence Alignment">
<meta property="article:tag" content="MSA">
<meta property="article:tag" content="sequence alignment">
<meta property="article:tag" content="trimming">
<meta property="article:tag" content="Clustal">
<meta property="article:tag" content="MUSCLE">
<meta property="article:tag" content="MAFFT">
<meta property="article:tag" content="PRANK">
<meta property="article:tag" content="Gblocks">
<meta property="article:tag" content="trimAl">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">生信技工</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="搜索"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-bioinfo_MSA" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2021/09/06/bioinfo_MSA/" class="article-date">
  <time datetime="2021-09-06T10:00:00.000Z" itemprop="datePublished">2021-09-06</time>
</a>
    
  <div class="article-category">
    <a class="article-category-link" href="/categories/bio/">bio</a>►<a class="article-category-link" href="/categories/bio/bioinfo/">bioinfo</a>
  </div>

  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      多序列比对(Multiple Sequence Alignment,MSA)
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=1697043&auto=1&height=32"></iframe></div>

<p><strong>写在前面</strong>：</p>
<ul>
<li>比对软件的选择：<br>  要是是基因组大数据的序列做比对，优先选择<strong>MAFFT</strong>（快速且兼顾准确）；要是少量序列比对，优先选择<strong>PRANK</strong>（准确）；要是极少数序列比对，哪个方便选哪个，一般<strong>Clustal和MUSCLE</strong>（方便）会整合到许多软件里，会更方便使用。</li>
<li>过滤软件的选择：<br>  目前暂时优先选<strong>trimAl</strong>。</li>
</ul>
<p>update-20211006:今天发现PRANK这个软件比对过后序列的顺序会被打乱，后续处理要小心。</p>
<h1 id="1-序列比对-sequence-alignment"><a href="#1-序列比对-sequence-alignment" class="headerlink" title="1. 序列比对(sequence alignment)"></a>1. 序列比对(sequence alignment)</h1><h2 id="1-1-概念"><a href="#1-1-概念" class="headerlink" title="1.1. 概念"></a>1.1. 概念</h2><p>在生物信息学中，序列比对(sequence alignment)是指根据相似性排列DNA,RNA或者protein序列，通过在序列中插入空格(gap)来使得相似的碱基&#x2F;氨基酸在相同列对齐的一种方法。</p>
<h2 id="1-2-方法"><a href="#1-2-方法" class="headerlink" title="1.2. 方法"></a>1.2. 方法</h2><p>比对方法通常包括全局比对(global alignments)和局部比对(local alignments)。</p>
<ol>
<li>全局比对(global alignments)<br>全局比对强制要求比对跨越所有序列的整个长度，尽量对齐每条序列的每个碱基，当序列碱基和长度大致相似时最有用。通常全局比对用Needleman-Wunsch算法，是一种动态规划算法。</li>
<li>局部比对(local alignments)<br>局部比对则确认通常有较大分歧的长序列内的相似性，有更多的计算相似性区域的额外的挑战。通常在较长序列包含相似序列基序的差异较大序列中更有用，常见的是Smith-Waterman算法，也是一种动态规划算法。</li>
<li>混合比对(Hybrid methods)<br>全局和局部比对都使用的方法。</li>
</ol>
<h2 id="1-3-类型"><a href="#1-3-类型" class="headerlink" title="1.3. 类型"></a>1.3. 类型</h2><p>按照序列的数量分为成对比对(pairwise alignment)和多序列比对(multiple sequence alignment)。</p>
<ol>
<li>成对比对(pairwise alignment)</li>
</ol>
<ul>
<li>两条序列的比对。因为计算效率很高，通常不需要高精度方法。</li>
<li>常见的三种方法是点阵法(dot-matrix methods)，动态规划法(dynamic programming)和字节法(word methods)。</li>
</ul>
<ol start="2">
<li>多序列比对(multiple sequence alignment)</li>
</ol>
<ul>
<li>三条及以上序列的比对，是成对比对的扩展，算法上有些差异，以及更复杂。是这篇博客主要讲的内容。</li>
</ul>
<h1 id="2-多序列比对-Multiple-Sequence-Alignment-MSA"><a href="#2-多序列比对-Multiple-Sequence-Alignment-MSA" class="headerlink" title="2. 多序列比对(Multiple Sequence Alignment,MSA)"></a>2. 多序列比对(Multiple Sequence Alignment,MSA)</h1><h2 id="2-1-概念"><a href="#2-1-概念" class="headerlink" title="2.1. 概念"></a>2.1. 概念</h2><p>多序列比对(Multiple Sequence Alignment,MSA)是对三个以上的生物学序列（biological sequence），如蛋白质序列、DNA序列或RNA序列所作的序列比对。<br>目的是在不改变序列顺序的前提下，尽可能地把不同序列的相同碱基或者氨基酸排在同一列，并认为同一列的序列在进化上是同源的，有共同的祖先。</p>
<h2 id="2-2-用途"><a href="#2-2-用途" class="headerlink" title="2.2. 用途"></a>2.2. 用途</h2><p>比对之后的序列，可用于进化学研究。比如，比较序列差异来推断序列进化历史（eg. 分化时间）和进化方式(突变&#x2F;插入缺失&#x2F;倒置等)。</p>
<p>常见的应用：</p>
<ul>
<li>构建系统发育树：亲缘关系的分析。</li>
<li>比较基因功能：比较不同物种间的基因功能。</li>
<li>预测蛋白质结构，分类和注释蛋白和ncRNA：通过与已知功能的基因序列的相似性比对，进行基因功能的预测和注释。</li>
<li>序列拼接：通过比对，组装基因组。</li>
<li>突变分析：不同个体基因组间的差异；不同物种基因组间的共线性和结构变异。</li>
<li>检测适应性进化</li>
</ul>
<h2 id="2-3-重要性"><a href="#2-3-重要性" class="headerlink" title="2.3. 重要性"></a>2.3. 重要性</h2><p>多序列比对的准确性与进化分析的准确性息息相关，不同比对软件和过滤软件的结果不同。</p>
<p>Blackburner和Whelan在2013年发表在MBE上的文章《Class of Multiple Sequence Alignment Algorithm Affects Genomic Analysis》对不同的比对算法和软件进行了比较分析，提出多序列比对MSA方法对下游分析至关重要，系统发育树和分支长度的推断高度依赖使用的多序列比对MSA类别，基于相似性的比对软件倾向于识别适应性进化（正选择）。</p>
<p>简单的对齐后过滤不能解决不同MSA方法间的差异。</p>
<h1 id="3-多序列比对-Multiple-Sequence-Alignment-MSA-的算法"><a href="#3-多序列比对-Multiple-Sequence-Alignment-MSA-的算法" class="headerlink" title="3. 多序列比对(Multiple Sequence Alignment,MSA)的算法"></a>3. 多序列比对(Multiple Sequence Alignment,MSA)的算法</h1><p>多序列比对是双序列比对的扩展。</p>
<h2 id="3-1-动态规划算法-dynamic-programming-——-主要用于双序列比对"><a href="#3-1-动态规划算法-dynamic-programming-——-主要用于双序列比对" class="headerlink" title="3.1. 动态规划算法(dynamic programming) —— 主要用于双序列比对"></a>3.1. 动态规划算法(dynamic programming) —— 主要用于双序列比对</h2><p>算法特点：</p>
<ul>
<li>可以给出最优解。</li>
<li>将一个二维的动态规划矩阵扩展到三维或多维，即可用于多序列比对。</li>
<li>比对序列超过3个，需要的存储空间和运算时间都使得比对应用无法实现。<h3 id="3-1-1-基于全局比对的算法：Needleman-Wunsch算法"><a href="#3-1-1-基于全局比对的算法：Needleman-Wunsch算法" class="headerlink" title="3.1.1. 基于全局比对的算法：Needleman-Wunsch算法"></a>3.1.1. 基于全局比对的算法：Needleman-Wunsch算法</h3></li>
</ul>
<ol>
<li><p><strong>全局比对</strong><br> 全局比对是指将参与比对的两条序列里面的所有字符进行比对。全局比对在全局范围内对两条序列进行比对打分，找出最佳比对，主要被用来寻找关系密切的序列。其可以用来鉴别或证明新序列与已知序列家族的同源性，是进行分子进化分析的重要前提。其代表是Needleman-Wunsch算法。</p>
</li>
<li><p>全局比对算法特点：</p>
</li>
</ol>
<ul>
<li>打分矩阵：罚分分值不同，比对结果不同。</li>
<li>计算比对最高得分的算法：常用Needleman-Wunsch算法。</li>
<li>打印出最高得分相应的序列比对结果：根据得分矩阵回溯，找到最高得分结果。</li>
</ul>
<h3 id="3-1-2-基于局部比对的算法：Smith-Waterman算法"><a href="#3-1-2-基于局部比对的算法：Smith-Waterman算法" class="headerlink" title="3.1.2. 基于局部比对的算法：Smith-Waterman算法"></a>3.1.2. 基于局部比对的算法：Smith-Waterman算法</h3><ol>
<li><p><strong>局部比对</strong><br> 与全局比对不同，局部比对不必对两个完整的序列进行比对，而是在每个序列中使用某些局部区域片段进行比对。其产生的需求在于、人们发现有的蛋白序列虽然在序列整体上表现出较大的差异性，但是在某些局部区域能独立的发挥相同的功能，序列相当保守。这时候依靠全局比对明显不能得到这些局部相似序列的。其次，在真核生物的基因中，内含子片段表现出了极大变异性，外显子区域却较为保守，这时候全局比对表现出了其局限性，无法找出这些局部相似性序列。其代表是Smith-Waterman局部比对算法。BLAST是局部比对的一种推广。</p>
</li>
<li><p>Smith-Waterman算法特点</p>
</li>
</ol>
<ul>
<li>是一种用来寻找并比较具有局部相似性区域的动态规划算法，适用于亲缘关系较远，整体上不具有相似性而在一些较小区域存在局部相似性的两条序列。</li>
<li>使用迭代方法计算出每个序列的相似分值，保存在一个得分矩阵M中，然后根据这个得分矩阵，通过动态规划的方法回溯找到最优的比对序列。</li>
</ul>
<h2 id="3-2-启发式搜索算法-heuristic-algorithm-：BWT算法和BLAST算法"><a href="#3-2-启发式搜索算法-heuristic-algorithm-：BWT算法和BLAST算法" class="headerlink" title="3.2. 启发式搜索算法(heuristic algorithm)：BWT算法和BLAST算法"></a>3.2. 启发式搜索算法(heuristic algorithm)：BWT算法和BLAST算法</h2><p>一个基于直观或经验构造的算法，在可接受的花费下给出待解决组合优化问题每一个实例的一个可行解，该可行解与最优解的偏离程度一般不能被预计；目前绝大多数应用的比对软件都是启发式算法。</p>
<ul>
<li>渐进式算法(progressive methods): Clustal W,MUSCLE,T-Coffee,DiAlign</li>
<li>迭代算法(iterative methods): MUSCLE,PRRP算法</li>
<li>星对比算法</li>
</ul>
<h2 id="3-3-随机算法"><a href="#3-3-随机算法" class="headerlink" title="3.3. 随机算法"></a>3.3. 随机算法</h2><p>对构造好的目标函数进行最优解搜索。</p>
<ul>
<li>遗传算法：借鉴生物界进化规律演化来的全局意义上的自适应随机搜索方法。</li>
<li>模拟退火算法：用一物质系统的退火过程来模拟优化问题的寻优方法，当物质系统达到最小能量状态时，优化问题的目标函数也相应地达到了全局最优解。</li>
<li>粒子群算法</li>
</ul>
<h2 id="3-4-其他算法"><a href="#3-4-其他算法" class="headerlink" title="3.4. 其他算法"></a>3.4. 其他算法</h2><p>还有一些其他不常见的算法</p>
<ol>
<li>分治算法</li>
<li>蚁群算法</li>
</ol>
<h1 id="4-序列选择和比对前处理"><a href="#4-序列选择和比对前处理" class="headerlink" title="4. 序列选择和比对前处理"></a>4. 序列选择和比对前处理</h1><p>一般来说，同源序列、同一家族的序列才会进行比对。</p>
<p><strong>通用的序列选择标准</strong>：</p>
<ul>
<li>如果是编码区序列，优先选蛋白序列。因为蛋白序列短且含有20种氨基酸信息，比DNA的4种核苷酸信息更多。</li>
<li>如果是数据库里选序列，尽量选择有详细注释的，可以在后续分析中提供更多信息。</li>
<li>多序列比对选用10-15条开始比对。如果比对结果不错，再加其他序列比对；如果比对结果不好，对现有序列处理（比如删除，剪辑等）。比对序列数量不是越多越好，多了可能增加比对软件出错概率。</li>
<li>如果有一条序列与半数以上其他序列一致性低于30%，比对会有问题。（一般氨基酸序列一致性在30-70%，E-value在10^-40-10^-5，仅供参考）。</li>
<li>如果有序列间一致性太高的，多序列比对也没什么价值，结果一直不错。</li>
<li>很多比对软件擅长比对总长度类似的序列，对长短不一的序列可以尝试比对前剪辑。</li>
<li>一般的比对软件对有重复片段的多序列比对时会有问题，尤其是序列间重复次数不同时问题更大，需要人工提取这部分手工比对。</li>
</ul>
<h1 id="5-多序列比对软件"><a href="#5-多序列比对软件" class="headerlink" title="5. 多序列比对软件"></a>5. 多序列比对软件</h1><h2 id="5-1-比对软件概况"><a href="#5-1-比对软件概况" class="headerlink" title="5.1. 比对软件概况"></a>5.1. 比对软件概况</h2><ol>
<li>基于相似性的比对(similarity-MSAMs)</li>
</ol>
<ul>
<li>经典渐进式比对(progressive alignment)软件：ClustalW</li>
<li>现代渐进式比对软件：DIALIGN-TX, MUSCLE</li>
<li>一致性比对软件(Consistency methods)：MUMMALS,ProbAlign,ProbCons,MAFFT,T-Coffee。</li>
</ul>
<ol start="2">
<li>基于进化关系的比对</li>
</ol>
<ul>
<li>PRANK</li>
<li>BAli-Phy</li>
<li>StatAlign</li>
</ul>
<ol start="3">
<li>其他</li>
</ol>
<ul>
<li>MACSE</li>
<li>基于隐马尔可夫模型(Profile HMM Methods)的比对：SEPP, TIPP, UPP, HIPPI</li>
</ul>
<h2 id="5-2-基于相似性的比对-similarity-MSAMs"><a href="#5-2-基于相似性的比对-similarity-MSAMs" class="headerlink" title="5.2. 基于相似性的比对(similarity-MSAMs)"></a>5.2. 基于相似性的比对(similarity-MSAMs)</h2><h3 id="5-2-1-Clustal-1994"><a href="#5-2-1-Clustal-1994" class="headerlink" title="5.2.1. Clustal(1994)"></a>5.2.1. Clustal(1994)</h3><p>Clustal版本有X、W系列，目前最新的Omega表现最好。</p>
<p>基于渐进式比对(progressive alignment)，由引导树(guide tree)确定将序列添加到不断增长的MSA中的顺序，此算法通过成对的序列(sequence)-序列，序列-剖面(profile)，剖面-剖面的比对实现。</p>
<ol>
<li>算法流程</li>
</ol>
<ul>
<li>先进行序列两两比对，构建距离矩阵；</li>
<li>基于两两比对距离矩阵，通过计算构建系统进化引导树(guide tree)，对关系密切的序列进行加权；</li>
<li>从关系最近的两条序列开始，逐渐加入关系远的序列并不断重新构建多序列比对，直到所有序列都被加入为止。</li>
</ul>
<ol start="2">
<li>算法特点</li>
</ol>
<ul>
<li>是一种试探算法，渐进式比对算法都不能保证得到最优的比对。</li>
<li>比对准确性高度依赖开始的两两比对，适用于<strong>亲缘关系较近</strong>的序列。</li>
</ul>
<h3 id="5-2-2-MUSCLE-2004"><a href="#5-2-2-MUSCLE-2004" class="headerlink" title="5.2.2. MUSCLE(2004)"></a>5.2.2. MUSCLE(2004)</h3><p>特点：</p>
<ul>
<li>采用迭代方法进行比对运算，每一次最优化过程就是迭代过程，通过不断使用动态规划法重排来纠正错误，同时对这些亚类群进行比对以获得所有序列的全局比对。</li>
<li>相对Clustal，在比对速度和精度上都更优。</li>
<li>相对Clustal，添加了额外的步骤，包括改进起始树(starting tree)，重新选择树的分支和重新对齐任一剖面的改进步骤。</li>
</ul>
<h3 id="5-2-3-MAFFT-2002"><a href="#5-2-3-MAFFT-2002" class="headerlink" title="5.2.3. MAFFT(2002)"></a>5.2.3. MAFFT(2002)</h3><p>特点：</p>
<ul>
<li>基于渐进式比对的算法，用快速傅里叶变换(fast Fourier transform)对序列进行聚类，再实现成对比对(pairwise alignments)；</li>
<li>之后的版本增加了其他算法和操作模式，包括快速、精确模式，非编码RNA序列比对及在现有比对中添加新的序列。</li>
<li>实现了两种算法，累进方法（FFT‐NS‐2）和迭代优化方法（FFT‐NS‐i）</li>
<li>比对精度比MUSCLE高，速度也快。</li>
</ul>
<h3 id="5-2-4-T-Coffee-2000"><a href="#5-2-4-T-Coffee-2000" class="headerlink" title="5.2.4. T-Coffee (2000)"></a>5.2.4. T-Coffee (2000)</h3><p>与MAFFT功能类似，但包括更复杂的成对比对评分函数，MSA必须描述一组一致的成对比对。</p>
<h2 id="5-3-基于进化关系的比对-——-统计比对-statistical-alignment"><a href="#5-3-基于进化关系的比对-——-统计比对-statistical-alignment" class="headerlink" title="5.3. 基于进化关系的比对 —— 统计比对(statistical alignment)"></a>5.3. 基于进化关系的比对 —— 统计比对(statistical alignment)</h2><h3 id="5-3-1-统计比对的特点"><a href="#5-3-1-统计比对的特点" class="headerlink" title="5.3.1. 统计比对的特点"></a>5.3.1. 统计比对的特点</h3><ul>
<li>统计比对(statistical alignment)是在进化学语境下描述序列的变化，能够确定各序列在与其共同祖先分化后发生的插入、替换和删除的过程，在系统发育树上明确建模。</li>
<li>可以根据序列的差异性灵活地使用不同地比对标准（评分矩阵与罚分），基于渐进式的比对无法做到</li>
<li>基于进化关系，准确性优于渐进式比对，但由于计算量大，很少使用在基因组数据上。</li>
</ul>
<h3 id="5-3-2-BAli-Phy-2005"><a href="#5-3-2-BAli-Phy-2005" class="headerlink" title="5.3.2. BAli-Phy(2005)"></a>5.3.2. BAli-Phy(2005)</h3><ul>
<li>贝叶斯后验比对</li>
<li>利用马尔科夫链蒙特卡罗来探索给定分子序列数据的比对和系统发育的联合空间，同时估计消除了对不准确的比对引导树的偏差，在比对过程中采用了更复杂的替换模型，并自动利用共享插入&#x2F;删除中的信息来帮助推断系统发育关系</li>
<li>准确度和PRANK差不多？</li>
</ul>
<h3 id="5-3-3-PRANK-2008"><a href="#5-3-3-PRANK-2008" class="headerlink" title="5.3.3. PRANK(2008)"></a>5.3.3. PRANK(2008)</h3><p>特点：</p>
<ul>
<li>针对DNA,密码子(codon)和氨基酸序列的概率多重比对软件（有密码子和氨基酸模式）；</li>
<li>可以重新构建祖先序列，有DNA翻译(DNA translation)和回译(back-translation)选项；</li>
<li>对完全统计对齐的启发式(heuristic)方法；</li>
<li>准确度高，但非常耗时，不适合基因组数据；</li>
</ul>
<p>p.s.：PRANK比对过后序列顺序会被打乱，与比对前不一致，做后续处理时要小心。</p>
<h3 id="5-3-4-StatAlign-2008"><a href="#5-3-4-StatAlign-2008" class="headerlink" title="5.3.4. StatAlign(2008)"></a>5.3.4. StatAlign(2008)</h3><h2 id="5-4-MACSE"><a href="#5-4-MACSE" class="headerlink" title="5.4. MACSE"></a>5.4. MACSE</h2><p>了解得不多，先摘录在这里。</p>
<ul>
<li>第一个可以用于自动调整含有移码变异以及假基因的蛋白编码序列，而不破坏潜在密码子结构。</li>
<li>在核苷酸水平上对DNA序列进行比对，但有可能包括不是三个碱基倍数的间隙长度（即产生移码），同时基于其氨基酸翻译对产生的核苷酸比对进行评分。</li>
<li>这可以产生保留潜在密码子结构的核苷酸比对，同时受益于氨基酸序列的更高相似性。</li>
<li>已被用于基因组分析中</li>
<li>OrthoMAM v8&#x2F;v10数据库采用这个软件的分析流程</li>
</ul>
<h2 id="5-5-基于隐马尔可夫模型-Profile-HMM-Methods-的比对"><a href="#5-5-基于隐马尔可夫模型-Profile-HMM-Methods-的比对" class="headerlink" title="5.5. 基于隐马尔可夫模型(Profile HMM Methods)的比对"></a>5.5. 基于隐马尔可夫模型(Profile HMM Methods)的比对</h2><h3 id="5-5-1-SEPP-SATe-enabled-Phylogenetic-Placement"><a href="#5-5-1-SEPP-SATe-enabled-Phylogenetic-Placement" class="headerlink" title="5.5.1. SEPP(SATe-enabled Phylogenetic Placement)"></a>5.5.1. SEPP(SATe-enabled Phylogenetic Placement)</h3><p>解决将short reads放进参考序列和树的系统发育问题</p>
<h3 id="5-5-2-TIPP-Taxonomic-Identification-and-Phylogenetic-Profiling"><a href="#5-5-2-TIPP-Taxonomic-Identification-and-Phylogenetic-Profiling" class="headerlink" title="5.5.2. TIPP(Taxonomic Identification and Phylogenetic Profiling)"></a>5.5.2. TIPP(Taxonomic Identification and Phylogenetic Profiling)</h3><p>解决元组数据的分类识别和丰度分析问题</p>
<h3 id="5-5-3-UPP-Ultra-large-alignments-using-Phylogeny-aware-Profiles"><a href="#5-5-3-UPP-Ultra-large-alignments-using-Phylogeny-aware-Profiles" class="headerlink" title="5.5.3. UPP(Ultra-large alignments using Phylogeny-aware Profiles)"></a>5.5.3. UPP(Ultra-large alignments using Phylogeny-aware Profiles)</h3><p>解决非常大的数据集对齐的问题，这些数据集可能包括零碎数据，可以将数据集多达1,000,000条序列对齐；</p>
<h3 id="5-5-4-HIPPI-Highly-Accurate-Protein-Family-Classificatio-with-Ensembles-of-HMMs"><a href="#5-5-4-HIPPI-Highly-Accurate-Protein-Family-Classificatio-with-Ensembles-of-HMMs" class="headerlink" title="5.5.4. HIPPI(Highly Accurate Protein Family Classificatio with Ensembles of HMMs)"></a>5.5.4. HIPPI(Highly Accurate Protein Family Classificatio with Ensembles of HMMs)</h3><p>解决蛋白质家族分类问题</p>
<h1 id="6-Trimming-质量过滤软件"><a href="#6-Trimming-质量过滤软件" class="headerlink" title="6. Trimming - 质量过滤软件"></a>6. Trimming - 质量过滤软件</h1><p>在比对完成后，使用质量过滤软件过滤一些低质量以及高变异度的序列区域，仅保留进化保守的区域用于后续分析。</p>
<h2 id="6-1-产生低质量区域的原因"><a href="#6-1-产生低质量区域的原因" class="headerlink" title="6.1. 产生低质量区域的原因"></a>6.1. 产生低质量区域的原因</h2><ul>
<li>序列特征：远缘物种的序列在一些位置（例如第一个或最后一个外显子）有很大差异，这种真实存在的差异由于受到选择的影响，不适用于推断系统发育树（希望用保守区域推断系统发育关系）。</li>
<li>各种错误：包括测序错误、基因组组装错误、基因预测错误和多序列比对错误等。</li>
</ul>
<h2 id="6-2-多序列比对后过滤软件"><a href="#6-2-多序列比对后过滤软件" class="headerlink" title="6.2. 多序列比对后过滤软件"></a>6.2. 多序列比对后过滤软件</h2><h3 id="6-2-1-常见过滤软件"><a href="#6-2-1-常见过滤软件" class="headerlink" title="6.2.1. 常见过滤软件"></a>6.2.1. 常见过滤软件</h3><ol>
<li><strong>block-filtering</strong>: trimAl,Gblocks</li>
<li><strong>segment filtering</strong>: HmmCleaner,PREQUAL（基于隐马尔可夫模型的算法）</li>
<li><strong>sliding window analysis</strong>: FasParser2</li>
</ol>
<h3 id="6-2-2-Gblocks-2000"><a href="#6-2-2-Gblocks-2000" class="headerlink" title="6.2.2. Gblocks(2000)"></a>6.2.2. Gblocks(2000)</h3><ul>
<li>删除大片段非保守性或非同源性片段（6-10bp的非同源片段识别得不好），还对block(即一段连续不含Gap的列)的长度进行了限制。</li>
<li>不足：武断地规定了某个具体阈值来判断比对片段的保留或删除，对所有基因和所有片段一刀切式的处理，没有考虑不同片段的不同进化速率。</li>
</ul>
<h3 id="6-2-3-trimAl-2009"><a href="#6-2-3-trimAl-2009" class="headerlink" title="6.2.3. trimAl(2009)"></a>6.2.3. trimAl(2009)</h3><ul>
<li>优点：速度快，准确度高。</li>
<li>比起Gblocks，可自动选择每个特定比对中的参数，包括Gap比例和氨基酸相似性水平，从而优化signal-to-noise ratio。</li>
</ul>
<h3 id="6-2-4-HmmCleaner"><a href="#6-2-4-HmmCleaner" class="headerlink" title="6.2.4. HmmCleaner"></a>6.2.4. HmmCleaner</h3><ul>
<li>基于隐马尔可夫模型的新过滤算法。</li>
<li>只能过滤氨基酸序列。</li>
</ul>
<h3 id="6-2-5-PREQUAL"><a href="#6-2-5-PREQUAL" class="headerlink" title="6.2.5. PREQUAL"></a>6.2.5. PREQUAL</h3><ul>
<li>基于隐马尔可夫模型的新过滤算法。</li>
<li>针对非比对的蛋白编码序列及氨基酸序列。</li>
</ul>
<h3 id="6-2-6-FasParser2-2018"><a href="#6-2-6-FasParser2-2018" class="headerlink" title="6.2.6. FasParser2(2018)"></a>6.2.6. FasParser2(2018)</h3><ul>
<li>有图形界面。</li>
<li>包括有效过滤序列比对质量模块（滑窗分析），可实现大规模检测序列中经历正选择的位点并清除。</li>
</ul>
<h1 id="7-references"><a href="#7-references" class="headerlink" title="7. references"></a>7. references</h1><ol>
<li><a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Sequence_alignment">https://en.wikipedia.org/wiki/Sequence_alignment</a></li>
<li><a target="_blank" rel="noopener" href="https://zh.wikipedia.org/zh-hans/%E5%A4%9A%E9%87%8D%E5%BA%8F%E5%88%97%E6%AF%94%E5%B0%8D">https://zh.wikipedia.org/zh-hans/%E5%A4%9A%E9%87%8D%E5%BA%8F%E5%88%97%E6%AF%94%E5%B0%8D</a></li>
<li><a target="_blank" rel="noopener" href="https://academic.oup.com/mbe/article/30/3/642/1038709?login=true">https://academic.oup.com/mbe/article/30/3/642/1038709?login=true</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/430f21cce4f5">https://www.jianshu.com/p/430f21cce4f5</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/31fb919f1c91">https://www.jianshu.com/p/31fb919f1c91</a></li>
<li><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/150579075">https://zhuanlan.zhihu.com/p/150579075</a></li>
</ol>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/09/06/bioinfo_MSA/" data-id="cl1aiqcra001sm4vkb3my79g8" class="article-share-link">Share</a>
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2021/09/11/biosoft_WGDI/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          WGDI分析基因组共线性、计算Ks和鉴定WGD
        
      </div>
    </a>
  
  
    <a href="/2021/08/23/biosoft_QUAST/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">QUAST评估基因组组装</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">分类</h3>
    <div class="widget">
      <ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/">bio</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/bio/bioinfo/">bioinfo</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/biosoft/">biosoft</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/concept/">concept</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/evolution/">evolution</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/experiment/">experiment</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/bio/taxon/">taxon</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/">biosoft</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/biosoft/circos/">circos</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/blog/">blog</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/">computer</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/scripts/">scripts</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/">system</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer/system/Windows/">Windows</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer/web/">web</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/">computer language</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/computer-language/R/">R</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/">linux</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/linux/basics/">basics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/shell/">shell</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/linux/text-processing/">text processing</a></li></ul></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/">omics</a><ul class="category-list-child"><li class="category-list-item"><a class="category-list-link" href="/categories/omics/genome/">genome</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/omics/transcriptome/">transcriptome</a></li></ul></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/4dtv/" rel="tag">4dtv</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">;</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/BUSCO/" rel="tag">BUSCO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bauhinia/" rel="tag">Bauhinia</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Bioconductor/" rel="tag">Bioconductor</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CAFE/" rel="tag">CAFE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CDS/" rel="tag">CDS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CRAN/" rel="tag">CRAN</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/CSS/" rel="tag">CSS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Circos/" rel="tag">Circos</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Clustal/" rel="tag">Clustal</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/DESeq2/" rel="tag">DESeq2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/EDTA/" rel="tag">EDTA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ES6/" rel="tag">ES6</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GATK/" rel="tag">GATK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GOEAST/" rel="tag">GOEAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GSEA/" rel="tag">GSEA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Gblocks/" rel="tag">Gblocks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/GenBank/" rel="tag">GenBank</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/HISAT2/" rel="tag">HISAT2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KOBAS-i/" rel="tag">KOBAS-i</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/KaKs-Calculator2-0/" rel="tag">KaKs_Calculator2.0</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Ks/" rel="tag">Ks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LoRDEC/" rel="tag">LoRDEC</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MAFFT/" rel="tag">MAFFT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCScanX/" rel="tag">MCScanX</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MCscan/" rel="tag">MCscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MSA/" rel="tag">MSA</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MUSCLE/" rel="tag">MUSCLE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Multiple-Sequence-Alignment/" rel="tag">Multiple Sequence Alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Oases/" rel="tag">Oases</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoFinder/" rel="tag">OrthoFinder</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/OrthoMCL/" rel="tag">OrthoMCL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PAL2NAL/" rel="tag">PAL2NAL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PANNZER2/" rel="tag">PANNZER2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PATH/" rel="tag">PATH</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PCR/" rel="tag">PCR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/PRANK/" rel="tag">PRANK</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT/" rel="tag">ParaAT</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ParaAT-pl/" rel="tag">ParaAT.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QUAST/" rel="tag">QUAST</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R-package/" rel="tag">R package</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-seq/" rel="tag">RNA-seq</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RNA-sequencing/" rel="tag">RNA-sequencing</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ROUSFinder2-0-py/" rel="tag">ROUSFinder2.0.py</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RStudio/" rel="tag">RStudio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatMasker/" rel="tag">RepeatMasker</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/RepeatModeler/" rel="tag">RepeatModeler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Rtools/" rel="tag">Rtools</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/STARlong/" rel="tag">STARlong</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Salmon-SMEM/" rel="tag">Salmon-SMEM</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/StringTie/" rel="tag">StringTie</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TE/" rel="tag">TE</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TreeMix/" rel="tag">TreeMix</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/UpSetR/" rel="tag">UpSetR</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Venn/" rel="tag">Venn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/VennDiagram/" rel="tag">VennDiagram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGD/" rel="tag">WGD</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/WGDI/" rel="tag">WGDI</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\&&</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/" rel="tag">\||</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/analog/" rel="tag">analog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/awk/" rel="tag">awk</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/batch-run-sh/" rel="tag">batch_run.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bifurcated-tree/" rel="tag">bifurcated tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/biosoft/" rel="tag">biosoft</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/blog/" rel="tag">blog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bootstrap/" rel="tag">bootstrap</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/build-website/" rel="tag">build website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/chronogram/" rel="tag">chronogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladistics/" rel="tag">cladistics</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/cladogram/" rel="tag">cladogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/clusterProfiler/" rel="tag">clusterProfiler</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/codon/" rel="tag">codon</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/colinearity/" rel="tag">colinearity</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/combine/" rel="tag">combine</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/concept/" rel="tag">concept</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/crown-groups/" rel="tag">crown groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/deploy-website/" rel="tag">deploy website</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/divergence-time/" rel="tag">divergence time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/eggnog-mapper/" rel="tag">eggnog-mapper</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/evolutionary-tree/" rel="tag">evolutionary tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/experiment/" rel="tag">experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract/" rel="tag">extract</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-intron-info-pl/" rel="tag">extract_intron_info.pl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/extract-single-copy-genes-sh/" rel="tag">extract_single_copy_genes.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/facet-plot/" rel="tag">facet_plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/flex/" rel="tag">flex</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/for/" rel="tag">for</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/fossil-time/" rel="tag">fossil time</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/functional-annotation/" rel="tag">functional annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-family/" rel="tag">gene family</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-flow/" rel="tag">gene flow</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gene-set-enrichment-analysis/" rel="tag">gene set enrichment analysis</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome/" rel="tag">genome</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-annotation/" rel="tag">genome annotation</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-assessment/" rel="tag">genome assessment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/genome-submit/" rel="tag">genome submit</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/geom-taxalink/" rel="tag">geom_taxalink</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggdensitree/" rel="tag">ggdensitree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggplot2/" rel="tag">ggplot2</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggstance/" rel="tag">ggstance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ggtree/" rel="tag">ggtree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/" rel="tag">github</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gplots/" rel="tag">gplots</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/gulp/" rel="tag">gulp</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/homolog/" rel="tag">homolog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/html/" rel="tag">html</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/interproscan/" rel="tag">interproscan</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/itol/" rel="tag">itol</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jcvi/" rel="tag">jcvi</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/jquery/" rel="tag">jquery</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/line-separator/" rel="tag">line separator</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/loop/" rel="tag">loop</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/mcmctree/" rel="tag">mcmctree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/molecular-experiment/" rel="tag">molecular experiment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/monophyly/" rel="tag">monophyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multi-transcriptome-assembly-with-ref-sh/" rel="tag">multi-transcriptome_assembly_with_ref.sh</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/multiPhylo/" rel="tag">multiPhylo</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthogroup/" rel="tag">orthogroup</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/ortholog/" rel="tag">ortholog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/orthology/" rel="tag">orthology</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paml/" rel="tag">paml</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paralog/" rel="tag">paralog</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paraphyly/" rel="tag">paraphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paste/" rel="tag">paste</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-reticulum/" rel="tag">phylogenetic reticulum</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogenetic-tree/" rel="tag">phylogenetic tree</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogeny/" rel="tag">phylogeny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/phylogram/" rel="tag">phylogram</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/plot/" rel="tag">plot</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/polyphyly/" rel="tag">polyphyly</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/population-networks/" rel="tag">population networks</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/protein/" rel="tag">protein</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/react/" rel="tag">react</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/reinstall/" rel="tag">reinstall</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/rename/" rel="tag">rename</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/repeat-sequences/" rel="tag">repeat sequences</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/scripts/" rel="tag">scripts</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sed/" rel="tag">sed</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sequence-alignment/" rel="tag">sequence alignment</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sort/" rel="tag">sort</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/stem-groups/" rel="tag">stem groups</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/sync/" rel="tag">sync</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/synteny/" rel="tag">synteny</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/system/" rel="tag">system</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/table2asn/" rel="tag">table2asn</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/topGO/" rel="tag">topGO</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tr/" rel="tag">tr</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/treeio/" rel="tag">treeio</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimAl/" rel="tag">trimAl</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/trimming/" rel="tag">trimming</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tutorial/" rel="tag">tutorial</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/website-maintenance/" rel="tag">website maintenance</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/while/" rel="tag">while</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/windows/" rel="tag">windows</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xargs/" rel="tag">xargs</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/xenolog/" rel="tag">xenolog</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">标签云</h3>
    <div class="widget tagcloud">
      <a href="/tags/4dtv/" style="font-size: 10px;">4dtv</a> <a href="/tags/" style="font-size: 10px;">;</a> <a href="/tags/BUSCO/" style="font-size: 10px;">BUSCO</a> <a href="/tags/Bauhinia/" style="font-size: 10px;">Bauhinia</a> <a href="/tags/Bioconductor/" style="font-size: 10px;">Bioconductor</a> <a href="/tags/CAFE/" style="font-size: 10px;">CAFE</a> <a href="/tags/CDS/" style="font-size: 10px;">CDS</a> <a href="/tags/CRAN/" style="font-size: 10px;">CRAN</a> <a href="/tags/CSS/" style="font-size: 10px;">CSS</a> <a href="/tags/Circos/" style="font-size: 10px;">Circos</a> <a href="/tags/Clustal/" style="font-size: 10px;">Clustal</a> <a href="/tags/DESeq2/" style="font-size: 10px;">DESeq2</a> <a href="/tags/EDTA/" style="font-size: 10px;">EDTA</a> <a href="/tags/ES6/" style="font-size: 10px;">ES6</a> <a href="/tags/GATK/" style="font-size: 10px;">GATK</a> <a href="/tags/GOEAST/" style="font-size: 10px;">GOEAST</a> <a href="/tags/GSEA/" style="font-size: 14px;">GSEA</a> <a href="/tags/Gblocks/" style="font-size: 10px;">Gblocks</a> <a href="/tags/GenBank/" style="font-size: 10px;">GenBank</a> <a href="/tags/HISAT2/" style="font-size: 10px;">HISAT2</a> <a href="/tags/KOBAS-i/" style="font-size: 10px;">KOBAS-i</a> <a href="/tags/KaKs-Calculator2-0/" style="font-size: 10px;">KaKs_Calculator2.0</a> <a href="/tags/Ks/" style="font-size: 12px;">Ks</a> <a href="/tags/LoRDEC/" style="font-size: 10px;">LoRDEC</a> <a href="/tags/MAFFT/" style="font-size: 10px;">MAFFT</a> <a href="/tags/MCScanX/" style="font-size: 10px;">MCScanX</a> <a href="/tags/MCscan/" style="font-size: 10px;">MCscan</a> <a href="/tags/MSA/" style="font-size: 10px;">MSA</a> <a href="/tags/MUSCLE/" style="font-size: 10px;">MUSCLE</a> <a href="/tags/Multiple-Sequence-Alignment/" style="font-size: 10px;">Multiple Sequence Alignment</a> <a href="/tags/Oases/" style="font-size: 10px;">Oases</a> <a href="/tags/OrthoFinder/" style="font-size: 10px;">OrthoFinder</a> <a href="/tags/OrthoMCL/" style="font-size: 10px;">OrthoMCL</a> <a href="/tags/PAL2NAL/" style="font-size: 10px;">PAL2NAL</a> <a href="/tags/PANNZER2/" style="font-size: 10px;">PANNZER2</a> <a href="/tags/PATH/" style="font-size: 10px;">PATH</a> <a href="/tags/PCR/" style="font-size: 10px;">PCR</a> <a href="/tags/PRANK/" style="font-size: 10px;">PRANK</a> <a href="/tags/ParaAT/" style="font-size: 10px;">ParaAT</a> <a href="/tags/ParaAT-pl/" style="font-size: 10px;">ParaAT.pl</a> <a href="/tags/QUAST/" style="font-size: 10px;">QUAST</a> <a href="/tags/R/" style="font-size: 16px;">R</a> <a href="/tags/R-package/" style="font-size: 14px;">R package</a> <a href="/tags/RNA-seq/" style="font-size: 10px;">RNA-seq</a> <a href="/tags/RNA-sequencing/" style="font-size: 10px;">RNA-sequencing</a> <a href="/tags/ROUSFinder2-0-py/" style="font-size: 10px;">ROUSFinder2.0.py</a> <a href="/tags/RStudio/" style="font-size: 10px;">RStudio</a> <a href="/tags/RepeatMasker/" style="font-size: 10px;">RepeatMasker</a> <a href="/tags/RepeatModeler/" style="font-size: 10px;">RepeatModeler</a> <a href="/tags/Rtools/" style="font-size: 10px;">Rtools</a> <a href="/tags/STARlong/" style="font-size: 10px;">STARlong</a> <a href="/tags/Salmon-SMEM/" style="font-size: 10px;">Salmon-SMEM</a> <a href="/tags/StringTie/" style="font-size: 10px;">StringTie</a> <a href="/tags/TE/" style="font-size: 10px;">TE</a> <a href="/tags/TreeMix/" style="font-size: 10px;">TreeMix</a> <a href="/tags/UpSetR/" style="font-size: 10px;">UpSetR</a> <a href="/tags/Venn/" style="font-size: 10px;">Venn</a> <a href="/tags/VennDiagram/" style="font-size: 10px;">VennDiagram</a> <a href="/tags/WGD/" style="font-size: 12px;">WGD</a> <a href="/tags/WGDI/" style="font-size: 10px;">WGDI</a> <a href="/tags/" style="font-size: 10px;">\&&</a> <a href="/tags/" style="font-size: 10px;">\||</a> <a href="/tags/analog/" style="font-size: 10px;">analog</a> <a href="/tags/awk/" style="font-size: 10px;">awk</a> <a href="/tags/batch-run-sh/" style="font-size: 10px;">batch_run.sh</a> <a href="/tags/bifurcated-tree/" style="font-size: 10px;">bifurcated tree</a> <a href="/tags/biosoft/" style="font-size: 18px;">biosoft</a> <a href="/tags/blog/" style="font-size: 12px;">blog</a> <a href="/tags/bootstrap/" style="font-size: 10px;">bootstrap</a> <a href="/tags/build-website/" style="font-size: 10px;">build website</a> <a href="/tags/chronogram/" style="font-size: 10px;">chronogram</a> <a href="/tags/cladistics/" style="font-size: 10px;">cladistics</a> <a href="/tags/cladogram/" style="font-size: 10px;">cladogram</a> <a href="/tags/clusterProfiler/" style="font-size: 12px;">clusterProfiler</a> <a href="/tags/codon/" style="font-size: 10px;">codon</a> <a href="/tags/colinearity/" style="font-size: 10px;">colinearity</a> <a href="/tags/combine/" style="font-size: 10px;">combine</a> <a href="/tags/concept/" style="font-size: 12px;">concept</a> <a href="/tags/crown-groups/" style="font-size: 10px;">crown groups</a> <a href="/tags/deploy-website/" style="font-size: 10px;">deploy website</a> <a href="/tags/divergence-time/" style="font-size: 10px;">divergence time</a> <a href="/tags/eggnog-mapper/" style="font-size: 10px;">eggnog-mapper</a> <a href="/tags/evolutionary-tree/" style="font-size: 12px;">evolutionary tree</a> <a href="/tags/experiment/" style="font-size: 10px;">experiment</a> <a href="/tags/extract/" style="font-size: 10px;">extract</a> <a href="/tags/extract-intron-info-pl/" style="font-size: 10px;">extract_intron_info.pl</a> <a href="/tags/extract-single-copy-genes-sh/" style="font-size: 10px;">extract_single_copy_genes.sh</a> <a href="/tags/facet-plot/" style="font-size: 10px;">facet_plot</a> <a href="/tags/flex/" style="font-size: 10px;">flex</a> <a href="/tags/for/" style="font-size: 10px;">for</a> <a href="/tags/fossil-time/" style="font-size: 10px;">fossil time</a> <a href="/tags/functional-annotation/" style="font-size: 10px;">functional annotation</a> <a href="/tags/gene-family/" style="font-size: 10px;">gene family</a> <a href="/tags/gene-flow/" style="font-size: 10px;">gene flow</a> <a href="/tags/gene-set-enrichment-analysis/" style="font-size: 14px;">gene set enrichment analysis</a> <a href="/tags/genome/" style="font-size: 10px;">genome</a> <a href="/tags/genome-annotation/" style="font-size: 14px;">genome annotation</a> <a href="/tags/genome-assessment/" style="font-size: 12px;">genome assessment</a> <a href="/tags/genome-submit/" style="font-size: 10px;">genome submit</a> <a href="/tags/geom-taxalink/" style="font-size: 10px;">geom_taxalink</a> <a href="/tags/ggdensitree/" style="font-size: 10px;">ggdensitree</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/ggstance/" style="font-size: 10px;">ggstance</a> <a href="/tags/ggtree/" style="font-size: 10px;">ggtree</a> <a href="/tags/github/" style="font-size: 12px;">github</a> <a href="/tags/gplots/" style="font-size: 10px;">gplots</a> <a href="/tags/gulp/" style="font-size: 10px;">gulp</a> <a href="/tags/hexo/" style="font-size: 12px;">hexo</a> <a href="/tags/homolog/" style="font-size: 10px;">homolog</a> <a href="/tags/html/" style="font-size: 10px;">html</a> <a href="/tags/interproscan/" style="font-size: 10px;">interproscan</a> <a href="/tags/itol/" style="font-size: 10px;">itol</a> <a href="/tags/jcvi/" style="font-size: 10px;">jcvi</a> <a href="/tags/jquery/" style="font-size: 10px;">jquery</a> <a href="/tags/line-separator/" style="font-size: 10px;">line separator</a> <a href="/tags/linux/" style="font-size: 16px;">linux</a> <a href="/tags/loop/" style="font-size: 10px;">loop</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/mcmctree/" style="font-size: 10px;">mcmctree</a> <a href="/tags/molecular-experiment/" style="font-size: 10px;">molecular experiment</a> <a href="/tags/monophyly/" style="font-size: 10px;">monophyly</a> <a href="/tags/multi-transcriptome-assembly-with-ref-sh/" style="font-size: 10px;">multi-transcriptome_assembly_with_ref.sh</a> <a href="/tags/multiPhylo/" style="font-size: 10px;">multiPhylo</a> <a href="/tags/orthogroup/" style="font-size: 10px;">orthogroup</a> <a href="/tags/ortholog/" style="font-size: 10px;">ortholog</a> <a href="/tags/orthology/" style="font-size: 12px;">orthology</a> <a href="/tags/paml/" style="font-size: 12px;">paml</a> <a href="/tags/paralog/" style="font-size: 10px;">paralog</a> <a href="/tags/paraphyly/" style="font-size: 10px;">paraphyly</a> <a href="/tags/paste/" style="font-size: 10px;">paste</a> <a href="/tags/phylogenetic-reticulum/" style="font-size: 10px;">phylogenetic reticulum</a> <a href="/tags/phylogenetic-tree/" style="font-size: 10px;">phylogenetic tree</a> <a href="/tags/phylogeny/" style="font-size: 16px;">phylogeny</a> <a href="/tags/phylogram/" style="font-size: 10px;">phylogram</a> <a href="/tags/plot/" style="font-size: 14px;">plot</a> <a href="/tags/polyphyly/" style="font-size: 10px;">polyphyly</a> <a href="/tags/population-networks/" style="font-size: 10px;">population networks</a> <a href="/tags/protein/" style="font-size: 10px;">protein</a> <a href="/tags/react/" style="font-size: 10px;">react</a> <a href="/tags/reinstall/" style="font-size: 10px;">reinstall</a> <a href="/tags/rename/" style="font-size: 10px;">rename</a> <a href="/tags/repeat-sequences/" style="font-size: 10px;">repeat sequences</a> <a href="/tags/scripts/" style="font-size: 10px;">scripts</a> <a href="/tags/sed/" style="font-size: 10px;">sed</a> <a href="/tags/sequence-alignment/" style="font-size: 12px;">sequence alignment</a> <a href="/tags/sort/" style="font-size: 10px;">sort</a> <a href="/tags/stem-groups/" style="font-size: 10px;">stem groups</a> <a href="/tags/sync/" style="font-size: 12px;">sync</a> <a href="/tags/synteny/" style="font-size: 10px;">synteny</a> <a href="/tags/system/" style="font-size: 10px;">system</a> <a href="/tags/table2asn/" style="font-size: 10px;">table2asn</a> <a href="/tags/topGO/" style="font-size: 12px;">topGO</a> <a href="/tags/tr/" style="font-size: 10px;">tr</a> <a href="/tags/treeio/" style="font-size: 10px;">treeio</a> <a href="/tags/trimAl/" style="font-size: 10px;">trimAl</a> <a href="/tags/trimming/" style="font-size: 10px;">trimming</a> <a href="/tags/tutorial/" style="font-size: 20px;">tutorial</a> <a href="/tags/website-maintenance/" style="font-size: 10px;">website maintenance</a> <a href="/tags/while/" style="font-size: 10px;">while</a> <a href="/tags/windows/" style="font-size: 10px;">windows</a> <a href="/tags/xargs/" style="font-size: 10px;">xargs</a> <a href="/tags/xenolog/" style="font-size: 10px;">xenolog</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">归档</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/03/">三月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2022/01/">一月 2022</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/12/">十二月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/11/">十一月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/10/">十月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/09/">九月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/08/">八月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/07/">七月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/06/">六月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/05/">五月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/04/">四月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/03/">三月 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">十二月 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">最新文章</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2022/03/22/omics_genome.submit/">提交基因组到公共数据库 —— 以GenBank为例</a>
          </li>
        
          <li>
            <a href="/2022/03/20/bioinfo_geneflow_treemix/">用TreeMix推断居群间基因流</a>
          </li>
        
          <li>
            <a href="/2022/01/24/bioinfo_evolutionary.tree/">绘制进化树 —— R包treeio+ggtree</a>
          </li>
        
          <li>
            <a href="/2022/01/16/linux_command/">shell的常用命令</a>
          </li>
        
          <li>
            <a href="/2021/12/28/R_install/">R及相关软件的安装</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2022 南方枝叶<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/script.js"></script>




  </div>
</body>
</html>