<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="介绍GenomeScope，用GenomeScope做基因组调查(genome survey)的基因组特征评估。用GenomeScope1.0评估二倍体物种的基因组大小、杂合度、重复率等基因组特征，用GenomeScope2.0评估多倍体物种的基因组大小、杂合度、重复率、基因型比例和基因组结构(同源&#x2F;异源多倍体)等基因组特征，用Smudgeplot估计物种的倍性。">
<meta property="og:type" content="article">
<meta property="og:title" content="用k-mer分析进行基因组调查：（五）用GenomeScope评估基因组特征">
<meta property="og:url" content="http://example.com/2022/06/05/omics_genome.survey_05.GenomeScope/index.html">
<meta property="og:site_name" content="生信技工">
<meta property="og:description" content="介绍GenomeScope，用GenomeScope做基因组调查(genome survey)的基因组特征评估。用GenomeScope1.0评估二倍体物种的基因组大小、杂合度、重复率等基因组特征，用GenomeScope2.0评估多倍体物种的基因组大小、杂合度、重复率、基因型比例和基因组结构(同源&#x2F;异源多倍体)等基因组特征，用Smudgeplot估计物种的倍性。">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/omics_genome.survey_GenomeScope1.0.png?raw=true">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/omics_genome.survey_GenomeScope2.0.png?raw=true">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-020-14998-3/MediaObjects/41467_2020_14998_Fig2_HTML.png?as=webp">
<meta property="og:image" content="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-020-14998-3/MediaObjects/41467_2020_14998_Fig6_HTML.png?as=webp">
<meta property="og:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true">
<meta property="article:published_time" content="2022-06-04T16:00:00.000Z">
<meta property="article:modified_time" content="2024-06-15T03:53:38.278Z">
<meta property="article:author" content="南方枝叶">
<meta property="article:tag" content="genome survey">
<meta property="article:tag" content="genome">
<meta property="article:tag" content="k-mer">
<meta property="article:tag" content="KMC">
<meta property="article:tag" content="GenomeScope">
<meta property="article:tag" content="Smudgeplot">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/omics_genome.survey_GenomeScope1.0.png?raw=true">

<link rel="canonical" href="http://example.com/2022/06/05/omics_genome.survey_05.GenomeScope/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>用k-mer分析进行基因组调查：（五）用GenomeScope评估基因组特征 | 生信技工</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a7db9ab8b9b3ef3b239a82e4a58e43dc";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<!-- hexo injector head_end start -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css">

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-math@4.0.0/dist/style.css">
<!-- hexo injector head_end end --><link rel="alternate" href="/atom.xml" title="生信技工" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">生信技工</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
        <li class="menu-item menu-item-resume">

    <a href="/resume/index.html" rel="section"><i class="fa fa-file fa-fw"></i>resume</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2022/06/05/omics_genome.survey_05.GenomeScope/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="南方枝叶">
      <meta itemprop="description" content="学习生物信息学的笔记本">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="生信技工">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          用k-mer分析进行基因组调查：（五）用GenomeScope评估基因组特征
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-06-05 00:00:00" itemprop="dateCreated datePublished" datetime="2022-06-05T00:00:00+08:00">2022-06-05</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2024-06-15 11:53:38" itemprop="dateModified" datetime="2024-06-15T11:53:38+08:00">2024-06-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/" itemprop="url" rel="index"><span itemprop="name">omics</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/" itemprop="url" rel="index"><span itemprop="name">genome</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/omics/genome/genome-survey/" itemprop="url" rel="index"><span itemprop="name">genome survey</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Disqus: </span>
    
    <a title="disqus" href="/2022/06/05/omics_genome.survey_05.GenomeScope/#disqus_thread" itemprop="discussionUrl">
      <span class="post-comments-count disqus-comment-count" data-disqus-identifier="2022/06/05/omics_genome.survey_05.GenomeScope/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>4.8k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>4 mins.</span>
            </span>
            <div class="post-description">介绍GenomeScope，用GenomeScope做基因组调查(genome survey)的基因组特征评估。用GenomeScope1.0评估二倍体物种的基因组大小、杂合度、重复率等基因组特征，用GenomeScope2.0评估多倍体物种的基因组大小、杂合度、重复率、基因型比例和基因组结构(同源/异源多倍体)等基因组特征，用Smudgeplot估计物种的倍性。</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <div align="middle"><iframe frameborder="no" border="0" marginwidth="0" marginheight="0" width=298 height=52 src="//music.163.com/outchain/player?type=2&id=283098&auto=1&height=32"></iframe> </div>

<p><strong>用k-mer分析进行基因组调查系列：</strong></p>
<ul>
<li>（一）基本原理：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/05/25/omics_genome.survey_01.intro/">https://yanzhongsino.github.io/2022/05/25/omics_genome.survey_01.intro/</a></li>
<li>（二）用Smudgeplot估计倍性：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/12/31/omics_genome.survey_02.Smudgeplot/">https://yanzhongsino.github.io/2022/12/31/omics_genome.survey_02.Smudgeplot/</a></li>
<li>（三）用jellyfish进行k-mer频数统计：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/05/27/omics_genome.survey_03.jellyfish/">https://yanzhongsino.github.io/2022/05/27/omics_genome.survey_03.jellyfish/</a></li>
<li>（四）用KMC进行k-mer频数统计：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/06/05/omics_genome.survey_04.KMC/">https://yanzhongsino.github.io/2022/06/05/omics_genome.survey_04.KMC/</a></li>
<li>（五）用GenomeScope评估基因组特征：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/06/05/omics_genome.survey_05.GenomeScope/">https://yanzhongsino.github.io/2022/06/05/omics_genome.survey_05.GenomeScope/</a></li>
<li>（六）用GCE分步实现：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/06/07/omics_genome.survey_06.GCE/">https://yanzhongsino.github.io/2022/06/07/omics_genome.survey_06.GCE/</a></li>
<li>（七）用KmerGenie一步实现：<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/06/19/omics_genome.survey_07.KmerGenie/">https://yanzhongsino.github.io/2022/06/19/omics_genome.survey_07.KmerGenie/</a></li>
</ul>
<p><strong>【推荐】用Smudgeplot评估物种倍性后，用组合jellyfish+GenomeScope1.0做二倍体物种的基因组调查，用组合KMC+GenomeScope2.0做多倍体物种的基因组调查。</strong></p>
<h1 id="1-k-mer进行基因组调查的软件概况"><a href="#1-k-mer进行基因组调查的软件概况" class="headerlink" title="1. k-mer进行基因组调查的软件概况"></a>1. k-mer进行基因组调查的软件概况</h1><p>k-mer进行基因组调查分为<strong>k-mer频数统计</strong>和<strong>基因组特征评估</strong>两步。</p>
<ul>
<li>GenomeScope可以实现第二步基因组特征评估。</li>
<li>需要在jellyfish&#x2F;KMC等软件的第一步结果k-mer频数分布表的基础上，GenomeScope才可实现。</li>
</ul>
<p>推荐第一步获取k-mer频数分布表的命令：</p>
<ol>
<li><p>jellyfish</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">jellyfish count -C -m 21 -s 1000000000 -t 10 *.fastq -o sample.jf #计算k-mer频率，生成sample.jf</span><br><span class="line">jellyfish histo -t 10 sample.jf &gt; sample.histo #生成k-mer频数直方表sample.histo和k-mer直方图</span><br></pre></td></tr></table></figure>
</li>
<li><p>KMC</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">mkdir tmp</span><br><span class="line">ls *.fastq.gz &gt; FILES</span><br><span class="line">kmc -k21 -t16 -m64 -ci1 -cs10000 @FILES kmcdb tmp #计算k-mer频率</span><br><span class="line">kmc_tools transform kmcdb histogram sample.histo -cx10000 #生成k-mer直方图</span><br></pre></td></tr></table></figure></li>
</ol>
<h1 id="2-GenomeScope概况"><a href="#2-GenomeScope概况" class="headerlink" title="2. GenomeScope概况"></a>2. GenomeScope概况</h1><ul>
<li>GenomeScope可以利用第一步jellyfish或KMC等其他软件分析得到k-mer频数分布表(sample.histo文件)实现第二步<strong>基因组特征评估</strong>。</li>
<li>GenomeScope1.0在2017年发表，用于二倍体物种的基因组调查；2020年又发表了GenomeScope 2.0版本，用于多倍体物种的基因组调查，并发布了用于判断物种倍性的Smudgeplot。</li>
<li><strong>GenomeScope1.0</strong>的基因组特征结果包括基因组大小(genome size)，杂合度(heterozygosity)，重复序列比例，GC含量等。</li>
<li><strong>GenomeScope2.0</strong>的基因组特征结果包括基因组大小(genome size)，杂合度(heterozygosity)，重复序列比例，GC含量，基因型比例，和基因组结构(同源&#x2F;异源多倍体)等。</li>
<li>GenomeScope有网页版和Linux本地版，功能一样；推荐网页版，免去安装的麻烦。</li>
</ul>
<h1 id="3-GenomeScope1-0-网页版【推荐】——-适用于二倍体物种"><a href="#3-GenomeScope1-0-网页版【推荐】——-适用于二倍体物种" class="headerlink" title="3. GenomeScope1.0 网页版【推荐】—— 适用于二倍体物种"></a>3. GenomeScope1.0 网页版【推荐】—— 适用于二倍体物种</h1><p><a target="_blank" rel="noopener" href="http://qb.cshl.edu/genomescope">GenomeScope1.0 网页版</a>：<a target="_blank" rel="noopener" href="http://qb.cshl.edu/genomescope">http://qb.cshl.edu/genomescope</a></p>
<ol>
<li><p>上传第一步获得的k-mer频数分布表sample.histo文件；</p>
</li>
<li><p>设置参数Kmer length为第一步选择的k-mer长度值，这里是17；</p>
</li>
<li><p>参数Read length为序列读长，一般为150；</p>
</li>
<li><p>参数Max kmer coverage默认是1000。</p>
<p> 建议按照物种情况修改，比如10000，以统计更准确。</p>
<p> 这个参数太小，可能造成过滤过多的Kmer，导致估计的基因组大小偏小的情况。</p>
<p> 这个参数太大则可能把高拷贝数量的DNA，比如叶绿体DNA，包括进Kmer的统计，造成GenomeScope算法的误差，所以还是不推荐使用-1或太大的值。</p>
</li>
<li><p>提交后几分钟就可以得到结果，保存结果图片可用于发表。</p>
</li>
</ol>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/omics_genome.survey_GenomeScope1.0.png?raw=true" width=70% title="GenomeScope1.0结果示例" align=center/>

<p><strong><p align="center">Figure 1. GenomeScope1.0结果示例</p></strong></p>
<h1 id="4-GenomeScope2-0-网页版-【推荐】-——-适用于多倍体物种"><a href="#4-GenomeScope2-0-网页版-【推荐】-——-适用于多倍体物种" class="headerlink" title="4. GenomeScope2.0 网页版 【推荐】 —— 适用于多倍体物种"></a>4. GenomeScope2.0 网页版 【推荐】 —— 适用于多倍体物种</h1><p><a target="_blank" rel="noopener" href="http://qb.cshl.edu/genomescope/genomescope2.0">GenomeScope2.0 网页版</a>：<a target="_blank" rel="noopener" href="http://qb.cshl.edu/genomescope/genomescope2.0">http://qb.cshl.edu/genomescope/genomescope2.0</a></p>
<p>GenomeScope2.0版本相较于1.0，进行了许多改进，主要是增加了多倍体物种的基因组调查，并提出Smudgeplot方法来估计基因组的倍性和基因组结构。</p>
<h2 id="4-1-GenomeScope-2-0-使用步骤"><a href="#4-1-GenomeScope-2-0-使用步骤" class="headerlink" title="4.1. GenomeScope 2.0 使用步骤"></a>4.1. GenomeScope 2.0 使用步骤</h2><ol>
<li>上传第一步获得的k-mer频数分布表histo文件；</li>
<li>设置参数Kmer length为第一步选择的k-mer长度值，这里是17；</li>
<li>参数倍性Ploidy根据物种的倍性设定，默认是二倍体，设置成2；</li>
<li>参数Max k-mer coverage默认是-1，即不限制最大k-mer深度。建议按照物种情况修改，比如10000，以统计更准确。这个参数太小，可能造成过滤过多的Kmer，导致估计的基因组大小偏小的情况。这个参数太大则可能把高拷贝数量的DNA，比如叶绿体DNA，包括进Kmer的统计，造成GenomeScope算法的误差，所以还是不推荐使用-1或太大的值。</li>
<li>参数Average k-mer coverage for polyploid genome默认是-1，即不进行筛选，可以根据情况调整。</li>
<li>提交后几分钟就可以得到结果，保存结果图片可用于发表。</li>
</ol>
<h2 id="4-2-GenomeScope-2-0-结果"><a href="#4-2-GenomeScope-2-0-结果" class="headerlink" title="4.2. GenomeScope 2.0 结果"></a>4.2. GenomeScope 2.0 结果</h2><h3 id="4-2-1-二倍体结果"><a href="#4-2-1-二倍体结果" class="headerlink" title="4.2.1. 二倍体结果"></a>4.2.1. <strong>二倍体</strong>结果</h3><p>二倍体的GenomeScope 2.0 结果与GenomeScope 1.0 结果的主要不同之处在于杂合度结果(het)变成了2.0版本的代表基因型的aa和ab的比例，其中杂合基因型ab的比例即为杂合度。2.0结果中的p值代表设置的物种倍性。</p>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/images/omics_genome.survey_GenomeScope2.0.png?raw=true" width=70% title="GenomeScope2.0二倍体结果示例" align=center/>

<p><strong><p align="center">Figure 2. GenomeScope2.0 二倍体结果示例</p></strong></p>
<h3 id="4-2-2-区分异源四倍体和同源四倍体"><a href="#4-2-2-区分异源四倍体和同源四倍体" class="headerlink" title="4.2.2. 区分异源四倍体和同源四倍体"></a>4.2.2. 区分<strong>异源四倍体</strong>和<strong>同源四倍体</strong></h3><p>GenomeScope2.0添加了参数倍性<strong>Ploidy</strong>，可以评估多倍体的基因组特征。</p>
<ol>
<li>四倍体共有两种可能的拓扑结构，代表着同源四倍体和异源四倍体，每种拓扑包含三种杂合基因型和一种纯合基因型，共有五种基因型。(五倍体有五种可能的拓扑，六倍体有十六种)</li>
<li>根据结果中杂合基因型的分布模式可以区分异源四倍体和同源四倍体。</li>
</ol>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-020-14998-3/MediaObjects/41467_2020_14998_Fig2_HTML.png?as=webp" width=80% title="四倍体的两种拓扑结构可能性" align=center/>

<p><strong><p align="center">Figure 3. 四倍体的两种拓扑结构可能性 a异源四倍体，b同源四倍体。 图片来源：<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-020-14998-3">GenomeScope 2.0 paper</a></p></strong></p>
<ol start="3">
<li>GenomeScope2.0的<strong>四倍体</strong>结果</li>
</ol>
<p>在GenomeScope 2.0 的结果中，如果杂合基因型aaab的比例大于aabb，则认为该物种是异源四倍体；如果杂合基因型aaab的比例小于aabb，则认为该物种是同源四倍体。</p>
<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-020-14998-3/MediaObjects/41467_2020_14998_Fig6_HTML.png?as=webp" width=100% title="GenomeScope2.0多倍体结果示例" align=center/>

<p><strong><p align="center">Figure 4. GenomeScope2.0 多倍体结果示例 a异源四倍体，b同源四倍体。 图片来源：<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-020-14998-3">GenomeScope 2.0 paper</a></p></strong></p>
<h1 id="5-GenomeScope1-0本地版-——-适用于二倍体物种"><a href="#5-GenomeScope1-0本地版-——-适用于二倍体物种" class="headerlink" title="5. GenomeScope1.0本地版 —— 适用于二倍体物种"></a>5. GenomeScope1.0本地版 —— 适用于二倍体物种</h1><p>GenomeScope1.0的本地版是用一个R脚本实现的，在<a target="_blank" rel="noopener" href="https://github.com/schatzlab/genomescope">GenomeScope github</a>可以下载genomescope.R脚本，下载后把genomescope.R文件加入环境变量即可使用。。</p>
<p><code>Rscript genomescope.R sample.histo k-mer_length read_length output_dir [kmer_max] [verbose]</code></p>
<p>必需参数：</p>
<ul>
<li>sample.histo：频数分布直方表，jellyfish的结果。</li>
<li>k-mer_length：k-mer长度，通常是17，21，与jellyfish一致。</li>
<li>read_length：reads长度，这里是150bp的PE reads，所以是150。</li>
<li>output_dir：输出目录，结果图和文本都输出到这个目录。</li>
</ul>
<h1 id="6-GenomeScope2-0本地版-——-适用于多倍体物种"><a href="#6-GenomeScope2-0本地版-——-适用于多倍体物种" class="headerlink" title="6. GenomeScope2.0本地版 —— 适用于多倍体物种"></a>6. GenomeScope2.0本地版 —— 适用于多倍体物种</h1><h2 id="6-1-下载和安装"><a href="#6-1-下载和安装" class="headerlink" title="6.1. 下载和安装"></a>6.1. 下载和安装</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/tbenavi1/genomescope2.0.git #下载</span><br><span class="line">cd genomescope2.0/</span><br><span class="line">mkdir ~/R_libs #创建主目录下的R_libs文件夹用于安装本地R库</span><br><span class="line">echo &quot;R_LIBS=~/R_libs/&quot; &gt;&gt; ~/.Renviron #创建/编辑.Renviron文件，使得R在创建的R_libs文件夹加载库</span><br><span class="line">Rscript install.R #安装</span><br></pre></td></tr></table></figure>

<p>安装后把目录下的genomescope.R文件加入环境变量即可使用。</p>
<h2 id="6-2-使用"><a href="#6-2-使用" class="headerlink" title="6.2. 使用"></a>6.2. 使用</h2><p><code>genomescope.R -i histogram_file -o output_dir -k k-mer_length</code></p>
<p>参数：</p>
<ul>
<li>-i histogram_file：频数分布直方表，jellyfish或KMC的结果。</li>
<li>-k k-mer_length：k-mer长度，通常是17，21，与jellyfish&#x2F;KMC的设置一致。</li>
<li>-o output_dir：输出目录，结果图和文本都输出到这个目录。</li>
<li>-p ploidy：设置倍性。</li>
<li>-l lambda：设置测序的平均k-mer覆盖率的初始猜测。</li>
<li>-n name_prefix：设置输出文件的前缀。</li>
<li>-m max_kmercov：设置从分析中排除高频k-mers的截止值，根据物种情况确定，推荐1000或10000。</li>
</ul>
<h1 id="7-GenomeScope实践经验"><a href="#7-GenomeScope实践经验" class="headerlink" title="7. GenomeScope实践经验"></a>7. GenomeScope实践经验</h1><ol>
<li>实际使用中发现，GenomeScope1.0和2.0常常估算差异较大。建议二倍体还是使用GenomeScope1.0。</li>
</ol>
<ul>
<li>在估算一个约300Mb的二倍体基因组时，GenomeScope1.0估算出来267Mb，GenomeScope2.0估算出来149Mb。</li>
<li>在估算一个约6Gb的四倍体基因组时发现，GenomeScope1.0估算出来5.5Gb，GenomeScope2.0估算出来2.7Gb。</li>
</ul>
<h1 id="8-references"><a href="#8-references" class="headerlink" title="8. references"></a>8. references</h1><ol>
<li>GenomeScope 1.0 github：<a target="_blank" rel="noopener" href="https://github.com/schatzlab/genomescope">https://github.com/schatzlab/genomescope</a></li>
<li>GenomeScope 2.0 github：<a target="_blank" rel="noopener" href="https://github.com/tbenavi1/genomescope2.0">https://github.com/tbenavi1/genomescope2.0</a></li>
<li>GenomeScope 1.0 paper：<a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/33/14/2202/3089939">https://academic.oup.com/bioinformatics/article/33/14/2202/3089939</a></li>
<li>GenomeScope 2.0 + Smudgeplot paper：<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-020-14998-3">https://www.nature.com/articles/s41467-020-14998-3</a></li>
<li>Smudgeplot github：<a target="_blank" rel="noopener" href="https://github.com/KamilSJaron/smudgeplot">https://github.com/KamilSJaron/smudgeplot</a></li>
</ol>
<hr>
<ul>
<li>欢迎关注微信公众号：<strong>生信技工</strong></li>
<li>公众号主要分享生信分析、生信软件、基因组学、转录组学、植物进化、生物学概念等相关内容，包括生物信息学工具的基本原理、操作步骤和学习心得。</li>
</ul>
<img src="https://github.com/yanzhongsino/yanzhongsino.github.io/blob/hexo/source/wechat/Wechat_public_qrcode.jpg?raw=true" width=20% title="wechat_public_QRcode.png" align=center/>
    </div>

    
    
    
        <div class="reward-container">
  <div>真诚赞赏，手留余香</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    Donate
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/images/wechatpay.jpg" alt="南方枝叶 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
      
      <div style="display: inline-block;">
        <img src="/images/alipay.jpg" alt="南方枝叶 Alipay">
        <p>Alipay</p>
      </div>

  </div>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/genome-survey/" rel="tag"># genome survey</a>
              <a href="/tags/genome/" rel="tag"># genome</a>
              <a href="/tags/k-mer/" rel="tag"># k-mer</a>
              <a href="/tags/KMC/" rel="tag"># KMC</a>
              <a href="/tags/GenomeScope/" rel="tag"># GenomeScope</a>
              <a href="/tags/Smudgeplot/" rel="tag"># Smudgeplot</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/05/27/omics_genome.survey_03.jellyfish/" rel="prev" title="用k-mer分析进行基因组调查：（三）用jellyfish进行k-mer频数统计">
      <i class="fa fa-chevron-left"></i> 用k-mer分析进行基因组调查：（三）用jellyfish进行k-mer频数统计
    </a></div>
      <div class="post-nav-item">
    <a href="/2022/06/05/omics_genome.survey_04.KMC/" rel="next" title="用k-mer分析进行基因组调查：（四）用KMC进行k-mer频数统计">
      用k-mer分析进行基因组调查：（四）用KMC进行k-mer频数统计 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    
  <div class="comments">
    <div id="disqus_thread">
      <noscript>Please enable JavaScript to view the comments powered by Disqus.</noscript>
    </div>
  </div>
  

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-k-mer%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E7%BB%84%E8%B0%83%E6%9F%A5%E7%9A%84%E8%BD%AF%E4%BB%B6%E6%A6%82%E5%86%B5"><span class="nav-text">1. k-mer进行基因组调查的软件概况</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-GenomeScope%E6%A6%82%E5%86%B5"><span class="nav-text">2. GenomeScope概况</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-GenomeScope1-0-%E7%BD%91%E9%A1%B5%E7%89%88%E3%80%90%E6%8E%A8%E8%8D%90%E3%80%91%E2%80%94%E2%80%94-%E9%80%82%E7%94%A8%E4%BA%8E%E4%BA%8C%E5%80%8D%E4%BD%93%E7%89%A9%E7%A7%8D"><span class="nav-text">3. GenomeScope1.0 网页版【推荐】—— 适用于二倍体物种</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-GenomeScope2-0-%E7%BD%91%E9%A1%B5%E7%89%88-%E3%80%90%E6%8E%A8%E8%8D%90%E3%80%91-%E2%80%94%E2%80%94-%E9%80%82%E7%94%A8%E4%BA%8E%E5%A4%9A%E5%80%8D%E4%BD%93%E7%89%A9%E7%A7%8D"><span class="nav-text">4. GenomeScope2.0 网页版 【推荐】 —— 适用于多倍体物种</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#4-1-GenomeScope-2-0-%E4%BD%BF%E7%94%A8%E6%AD%A5%E9%AA%A4"><span class="nav-text">4.1. GenomeScope 2.0 使用步骤</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-2-GenomeScope-2-0-%E7%BB%93%E6%9E%9C"><span class="nav-text">4.2. GenomeScope 2.0 结果</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#4-2-1-%E4%BA%8C%E5%80%8D%E4%BD%93%E7%BB%93%E6%9E%9C"><span class="nav-text">4.2.1. 二倍体结果</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-2-2-%E5%8C%BA%E5%88%86%E5%BC%82%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%92%8C%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93"><span class="nav-text">4.2.2. 区分异源四倍体和同源四倍体</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-GenomeScope1-0%E6%9C%AC%E5%9C%B0%E7%89%88-%E2%80%94%E2%80%94-%E9%80%82%E7%94%A8%E4%BA%8E%E4%BA%8C%E5%80%8D%E4%BD%93%E7%89%A9%E7%A7%8D"><span class="nav-text">5. GenomeScope1.0本地版 —— 适用于二倍体物种</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-GenomeScope2-0%E6%9C%AC%E5%9C%B0%E7%89%88-%E2%80%94%E2%80%94-%E9%80%82%E7%94%A8%E4%BA%8E%E5%A4%9A%E5%80%8D%E4%BD%93%E7%89%A9%E7%A7%8D"><span class="nav-text">6. GenomeScope2.0本地版 —— 适用于多倍体物种</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#6-1-%E4%B8%8B%E8%BD%BD%E5%92%8C%E5%AE%89%E8%A3%85"><span class="nav-text">6.1. 下载和安装</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#6-2-%E4%BD%BF%E7%94%A8"><span class="nav-text">6.2. 使用</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-GenomeScope%E5%AE%9E%E8%B7%B5%E7%BB%8F%E9%AA%8C"><span class="nav-text">7. GenomeScope实践经验</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-references"><span class="nav-text">8. references</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">南方枝叶</p>
  <div class="site-description" itemprop="description">学习生物信息学的笔记本</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">125</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">77</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">442</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="/atom.xml" title="RSS → &#x2F;atom.xml"><i class="fa fa-rss fa-fw"></i>RSS</a>
      </span>
      <span class="links-of-author-item">
        <a href="https://github.com/yanzhongsino" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yanzhongsino" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:yan.zhong.sino@gmail.com" title="E-Mail → mailto:yan.zhong.sino@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

      <div class="wechat_channel">
        <br>
        <!-- 这里添加你的二维码图片 -->
        <img src ="/images/wechat_channel.jpg?raw=true">
        <span>欢迎关注微信公众号</span>
      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2018 – 
  <span itemprop="copyrightYear">2024</span>
  <span class="with-love">
    <i class="snowflake-o"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">南方枝叶</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="Symbols count total">661k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="Reading time total">10:01</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

  

<script>
  function loadCount() {
    var d = document, s = d.createElement('script');
    s.src = 'https://yanzhongsino.disqus.com/count.js';
    s.id = 'dsq-count-scr';
    (d.head || d.body).appendChild(s);
  }
  // defer loading until the whole page loading is completed
  window.addEventListener('load', loadCount, false);
</script>
<script>
  var disqus_config = function() {
    this.page.url = "http://example.com/2022/06/05/omics_genome.survey_05.GenomeScope/";
    this.page.identifier = "2022/06/05/omics_genome.survey_05.GenomeScope/";
    this.page.title = "用k-mer分析进行基因组调查：（五）用GenomeScope评估基因组特征";
    };
  NexT.utils.loadComments(document.querySelector('#disqus_thread'), () => {
    if (window.DISQUS) {
      DISQUS.reset({
        reload: true,
        config: disqus_config
      });
    } else {
      var d = document, s = d.createElement('script');
      s.src = 'https://yanzhongsino.disqus.com/embed.js';
      s.setAttribute('data-timestamp', '' + +new Date());
      (d.head || d.body).appendChild(s);
    }
  });
</script>

</body>
</html>
